* Biotools                                                           :career:
:PROPERTIES:
:CREATED:  [2020-07-02 Thu 13:14]
:ID:       b03025b7-4b60-4196-8d7f-d947c7d64b28
:END:
:LOGBOOK:
CLOCK: [2021-05-20 Thu 13:14]--[2021-05-20 Thu 13:38] =>  0:24
CLOCK: [2020-12-16 Wed 07:08]--[2020-12-16 Wed 07:25] =>  0:17
CLOCK: [2020-12-15 Tue 20:02]--[2020-12-15 Tue 21:02] =>  1:00
CLOCK: [2020-10-14 Wed 16:00]--[2020-10-14 Wed 17:09] =>  1:09
CLOCK: [2020-09-21 Mon 08:42]--[2020-09-21 Mon 08:56] =>  0:14
CLOCK: [2020-09-21 Mon 07:01]--[2020-09-21 Mon 07:05] =>  0:04
:END:
https://slowkow.com/publications/
** [[id:39F11977-BE11-47CF-98B1-14CE6D9B97C4][LINK TO Information Science / Data Science]]
** [[file:~/repos/basecamp/basecamp.org::*Basecamp][LINK TO Basecamp]]
:PROPERTIES:
:ID:       76100c2b-a077-4c21-a398-16d88722e1c6
:END:
** References, databases, and data sets
:PROPERTIES:
:ID:       FCBA0863-010A-448C-A33C-941D2AA45AD5
:END:
:LOGBOOK:
CLOCK: [2020-11-17 Tue 12:07]--[2020-11-17 Tue 13:23] =>  1:16
CLOCK: [2019-07-07 Sun 09:58]--[2019-07-07 Sun 10:01] =>  0:03
CLOCK: [2019-02-22 Fri 07:58]--[2019-02-22 Fri 08:39] =>  0:41
CLOCK: [2018-03-03 Sat 20:28]--[2018-03-03 Sat 20:56] =>  0:28
:END:
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19

- https://www.ncbi.nlm.nih.gov/tools/vecscreen/univec/
- https://science.sciencemag.org/content/369/6509/1318
*** By substrate
:PROPERTIES:
:ID:       ffe7a0f8-5e6d-45ff-adec-efbba7c44783
:END:
**** miRNA data
:PROPERTIES:
:ID:       95b3c7f3-3b84-45b1-b8bf-5e2e6e2a6f88
:END:
- cite:kozomara2019
- https://exrna-atlas.org/
- [[id:fd462ab0-6aaf-4732-953b-141ad5e2ee72][LINK TO small RNA sequencing]]
- see mirbase in [[id:FCBA0863-010A-448C-A33C-941D2AA45AD5][databases, data sets, and database browsers]]
- Human miRNA tissue atlas
  - cite:fehlmann2016
  - https://ccb-web.cs.uni-saarland.de/tissueatlas/*** Adapter sequences
:PROPERTIES:
:ID:       cf2ea58e-8751-4d90-ab08-a63445925c89
:END:
- https://support.illumina.com/downloads/illumina-adapter-sequences-document-1000000002694.html
- https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html
*** code associated with reference data and databases
:PROPERTIES:
:ID:       767e5e30-8490-4cda-8018-8b0bd91c4ef3
:END:
- pull down common reference sequences
  #+name: reference_pull
  #+begin_src bash :tangle no
#########1#########2#########3#########4#########5#########6#########7#########8
reference_pull(){
    function usage(){
        printf "
reference_pull 
Checks reference files appropriate for host.   
If necessary, downloads and sets up reference files in a common directory structure.
Takes two parameters: 
1) The main reference data directory of the host
2) Desired reference file size 
	    "
            }

    if [ -z "$1" ] || [[ $1 == -h ]] || [[ $1 == --help ]]; then
        usage
        exit
    fi

    mkdir -p $1
    cd $1
    mkdir -p mouse
    mkdir -p rat
    mkdir -p human
    mkdir -p small    
}
end_src
#
#
wget --no-clobber https://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.ensGene.gtf.gz -O /mnt/users/jeszyman/data/rn6.ensGene.gtf.gz
#
gunzip -c /mnt/users/jeszyman/data/rn6.ensGene.gtf.gz > /mnt/users/jeszyman/data/rn6.ensGene.gtf
#

cd /mnt/users/jeszyman/data
rsem-prepare-reference --gtf ./rn6.ensGene.gtf \
#                       --bowtie \
                       ./rn6.fa \
                       rsem_ref

#+end_src
*** [[id:c0c0ee28-2e41-41a7-9a3b-ae195117a93e][LINK TO common bioinformatics file manipulation]]
:PROPERTIES:
:ID:       fc44f2e8-c32f-4169-ba9c-5032a35327a6
:END:
*** by organism
:PROPERTIES:
:CREATED:  [2020-10-26 Mon 12:18]
:ID:       33638cdf-113c-4248-aff4-b0d8ea7cbc09
:END:
**** rat
:PROPERTIES:
:ID:       f4ff99c8-9149-4183-89ff-56a09e68943b
:END:
- https://useast.ensembl.org/Rattus_norvegicus/Info/Index
- most recent assembly is Rnor_6.0
  - https://rgd.mcw.edu/rgdweb/report/genomeInformation/genomeInformation.html?species=Rat&mapKey=360&details=true
- https://rgd.mcw.edu/
**** human
:PROPERTIES:
:ID:       8a910eca-5f96-4a74-ad3e-efc6eb748b7c
:END:
- http://browser.1000genomes.org/index.html
**** mouse
:PROPERTIES:
:ID:       23fc30cd-7e45-4333-8740-1d8ec0755dab
:END:
- https://github.com/czbiohub/tabula-muris
- [[http://www.informatics.jax.org/vocab/gene_ontology][jax mouse gene ontology browser]]
- cite:lai2016
- https://www.biorxiv.org/content/10.1101/082511v1#disqus_thread
*** by data type
:PROPERTIES:
:ID:       5f2913ae-1f4a-44d5-90e2-9ee045c37b78
:END:
- extracellular RNA
  - https://exrna.org/resources/data/
- protein - http://www.proteinatlas.org/about
- integrated
  - https://www.nature.com/articles/s41586-019-1629-x
- [[id:333ebad6-9ed4-4d5b-a50e-f9fe9a8c3a92][gene expression]]
- [[id:629e512e-60ac-4283-bc0e-028b8cad3dcc][sequence variation]]
- miRNA
  - http://www.mirbase.org/cgi-bin/sentences.pl?acc=MI0001445
*** reference data 
:PROPERTIES:
:ID:       4dc2f252-12fe-409e-bef0-aab2d765de7a
:END:
- homology 
  - http://bioinf.wehi.edu.au/MSigDB/
  - cite:lai2016
    - https://bioconductor.org/packages/release/data/experiment/vignettes/gskb/inst/doc/gskb.pdf
    - http://ge-lab.org/gskb/
  - http://www.informatics.jax.org/homology.shtml
- gene sets and pathways
  - cite:liberzon2015
  - cite:lai2016
  - KEGG- note mostly orthology mapping for non-human organisms
  - http://pathguide.org/
  - cite:chowdhury2015
  - https://www.bioconductor.org/packages/devel/data/experiment/vignettes/gskb/inst/doc/gskb.pdf
  - - https://www.bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
  - MSigDB
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4707969/
  - [[https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/MSigDB_v7.0_Release_Notes#Changes_to_MSigDB_Gene_Symbol_Mapping_Procedures][orthology mapping]] 
- https://github.com/genome/gms/wiki/HCC1395-WGS-Exome-RNA-Seq-Data
- [[https://github.com/genome-in-a-bottle][Genome-in-a-bottle]]
  - https://github.com/genome-in-a-bottle
- http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/
- http://www.mirbase.org/
- annotaiton
  - http://daid.abcc.ncifcrf.gov/
- alignment 
  - terminology
    - abinitio is predicted genes
  - organizations
    - Ensembl
  - [[https://support.illumina.com/sequencing/sequencing_software/igenome.html][illumina iGenomes refrence sequence and annotation]]
  - [[https://software.broadinstitute.org/gatk/documentation/article?id=11010][gatk:  Human genome reference builds - GRCh38/hg38 - b37 - hg19]]
  - https://www.ncbi.nlm.nih.gov/grc/help/definitions/
  - https://www.ccdatalab.org/blog/2019/3/29/gene-expression-repos-explained
  - https://useast.ensembl.org/info/data/ftp/index.html
  - Genome Reference Consortium
    - https://www.slideshare.net/ShaojunXie/hg19-grch37-vs-hg38-grch38
  - Ensembl
    - [[https://useast.ensembl.org/info/data/ftp/index.html][Ensembl reference FTP]]
    - https://useast.ensembl.org/info/genome/variation/
*** individual subject / sample level data
:PROPERTIES:
:ID:       55d196dd-c99f-4ce8-b8e6-290b778e15b3
:END:
- [[https://www.ebi.ac.uk/ena][European Nucleotide Archive]]
- https://www.ncbi.nlm.nih.gov/geo/
- [[http://www.ebi.ac.uk/gwas/][The NHGRI-EBI Catalog of published genome-wide association studies]]
- https://www.ebi.ac.uk/gxa/experiments/E-MTAB-513/Results
- SRA (sequence read archive)
  - https://edwards.sdsu.edu/research/fastq-dump/
- dbGaP
  - https://www.ncbi.nlm.nih.gov/books/NBK99225/#DArequest.could_you_give_a_demo_about_th
- [[https://www.ccdatalab.org/blog/2019/3/29/gene-expression-repos-explained][gene expression repos explained]]
- [[https://www.encodeproject.org/][ENCODE]]
  - http://www.reddit.com/r/askscience/comments/znlk6/askscience_special_ama_we_are_the_encyclopedia_of/

- cite:beck2015
- [[https://www.reddit.com/r/bioinformatics/comments/601fzr/where_can_i_access_free_sequencing_data/][Where can I access free sequencing data]]
- [[https://www.aacr.org/professionals/research/aacr-project-genie/][GENIE]]
**** GDC: Genomic Data Commons
:PROPERTIES:
:ID:       99eace6a-6b4e-4acf-9381-1b48a59a4b61
:END:
- [[https://gdc.cancer.gov/][NCI Genomic Data Commons]]
  - [[https://docs.gdc.cancer.gov/][GDC documentation]]
  - [[https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization/genomic-data-alignment/rna-seq-pipeline][GDC RNA-seq pipeline]]
  - [[https://cbiit.nci.nih.gov/ncip/nci-cancer-genomics-cloud-pilots/nci-cloud-initiative][GDC cloud pilots]]
  - - https://gdc.cancer.gov/developers
  - https://docs.gdc.cancer.gov/
  - - https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization/genomic-data-alignment/rna-seq-pipeline
  - https://cbiit.nci.nih.gov/ncip/nci-cancer-genomics-cloud-pilots/nci-cloud-initiative
- http://www.cbioportal.org/tutorials
*** sequence variation
:PROPERTIES:
:ID:       629e512e-60ac-4283-bc0e-028b8cad3dcc
:END:
- [[http://firebrowse.org/?cohort=ACC&download_dialog=true][firehose]]
- http://browser.1000genomes.org/index.html
- Variance
  - gnomAD
    - - https://mail.google.com/mail/u/0/#inbox/FMfcgxwBWKXrqjHPFNdbSqJRRRdhzdGf
    - https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad/
    - (Subsumes ExAC)
  - ClinVar
    - [[https://www.ncbi.nlm.nih.gov/projects/clinvar/ClinVarDataDictionary.pdf][ClinVar Data Dictionary]]
  - phenotype association
    - https://genematcher.org/?tab=submissionsTab
    - [[https://www.ncbi.nlm.nih.gov/gap][dbGaP: database of Genotypes and Phenotypes]]
  - cBioPortal
    - http://www.cbioportal.org/tutorials
    - http://www.cbioportal.org/rmatlab
  - [[https://www.ncbi.nlm.nih.gov/projects/SNP/index.html][dbSNP]]
  - [[http://exac.broadinstitute.org/][ExAC Exome Aggregation Consortium]]
- polymorphism
  - [[http://www.internationalgenome.org/data#download][IGSR at 1000 Genomes]]
- [[https://civic.genome.wustl.edu/#/home][CIViC]]
  - cite:griffith2017civic
*** gene expression 
:PROPERTIES:
:ID:       333ebad6-9ed4-4d5b-a50e-f9fe9a8c3a92
:END:
- https://github.com/czbiohub/tabula-muris
- [[http://firebrowse.org/?cohort=ACC&download_dialog=true][firehose]]
- [[https://gtexportal.org/home/datasets][GTEx]]
  - [[https://biospecimens.cancer.gov/resources/sops/][GTEx SOPs]]
  - GTEx currently uses the
    - GRCh37/Hg19 genome
    - GENCODE v19 annotation
      - The GENCODE annotation should be patched to use Ensembl chromosome names: zcat gencode.v19.annotation.gtf.gz | sed 's/chrM/chrMT/;s/chr//' > gencode.v19.annotation.patched_contigs.gtf
  - sample ID
    - such as GTEX-14753-1626-SM-5NQ9L?
    - GTEX-YYYYY GTEx donor ID
    - 2nd number mostly refers to the tissue site, BUT we do not recommend using it for tissue site designation.
    - 3rd number is the RNA or DNA aliquot ID used for sequencing.
  - Unstranded 
  - GTEx
    - [[https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000424.v7.p2][at dbGAP]]
*** public data, data mapping
:PROPERTIES:
:CREATED:  [2020-07-02 Thu 13:18]
:ID:       7b386a21-a6b8-49d1-ad4b-f5976bf30666
:END:
:LOGBOOK:
CLOCK: [2020-07-02 Thu 13:19]--[2020-07-02 Thu 13:29] =>  0:10
:END:
- https://www.r-bloggers.com/converting-mouse-to-human-gene-names-with-biomart-package/
- geo submission   
  - https://www.ncbi.nlm.nih.gov/geo/info/seq.html
  - [[http://yeolab.github.io/onboarding/geo.html][see for useful ftp commands]]
  - [[https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE138253][example from Rentschler lab]]
*** Sequence Databases and Data Sets
:PROPERTIES:
:ID:       C137CB28-0517-4A98-8FB4-6D312DDBC325
:END:
- https://tabula-muris.ds.czbiohub.org/
- [[http://www.mitomap.org/MITOMAP][MITOMAP human mitochondrial genome]]
- https://portal.gdc.cancer.gov/
- [[https://www.ebi.ac.uk/ega/datasets/EGAD00001000078][EBI Data Browser]]
- [[http://www.internationalgenome.org/sites/1000genomes.org/files/documents/1000genomes_browser_quickstart.pdf][100s genomes browser]]
- [[https://genome.ucsc.edu/training/vids/index.html#vid08][ucsc training videos]]
- [[http://useast.ensembl.org/info/website/tutorials/sequence.html][ensembl sequence tutorials]]
- General Feature Format (GFF) / GTF Annotation Files
  - [[http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=494995959_n1VrbO9xpH51JE8qLga6fCvOyV2R&clade=mammal&org=Mouse&db=mm9&hgta_group=genes&hgta_track=refGene&hgta_table=0&hgta_regionType=genome&position=chr12%3A57795963-57815592&hgta_outputType=primaryTable&hgta_outFileName=][make your own]]
  - a tab-separated value file
  - see for standard: http://mblab.wustl.edu/GTF22.html

**** Stored on neuro server at G:/reference
:PROPERTIES:
:ID:       F08101C9-F675-4A35-B9A8-D0159FD74AFF
:END:

- mouse
  - gencode fa.gz
- human
  - hg38.fa and 

**** Genomes
:PROPERTIES:
:ID:       277AAE94-CFE5-400E-B5B9-291677A513D0
:END:
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19#:~:text=The%20latest%20build%20of%20the,the%20repertoire%20of%20ALT%20contigs.
- [[https://www.biostars.org/p/4405/][reasons to avoid soft-masked genomes for alignment]]
- [[http://hgdownload.cse.ucsc.edu/downloads.html][UCSC goldenPath bigZips]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/][Mouse mm10]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/mm9/bigZips/][Mouse mm9]]]]
***** Custom Genomes
:PROPERTIES:
:ID:       F5506BF8-8A6E-4949-9F5E-15B21511E87E
:END:

- No hits for "custom" in cite:star_2.5_dobin or cite:dobin2015mapping



https://www.biostars.org/p/111012/
http://mblab.wustl.edu/GTF2.html
http://useast.ensembl.org/info/website/upload/gff.html#fields
https://www.biostars.org/p/182230/
https://wiki.galaxyproject.org/Learn/CustomGenomes
http://wiki.bits.vib.be/index.php/Create_a_valid_gtf_file_for_tophat
https://support.bioconductor.org/p/66379/

****** Generating a Custom Genome
:PROPERTIES:
:ID:       11F1FF38-7F23-4B81-AEA3-AA760DC62C9F
:END:

******* Trial w/ EWSR1-FLLI1
:PROPERTIES:
:ID:       D293BED1-C843-476E-891B-E8FEFB4917F3
:END:

- Looking for help
  - no info
    - cite:beadling16_multip_amplic_approac_detec_gene
  - bad ideas
    - don't align to transcriptome- this doesn't save any time
    - don't try to make a gtf from a fa- the info isn't there
- Sequence sources
  - Genome- Downloaded GRCh37.85rm fasta from [[http://ftp.ensembl.org/pub/grch37/release-85/fasta/homo_sapiens/dna/][here]]
  - Fusions
    - from ensembl
      - [[http://useast.ensembl.org/Homo_sapiens/Transcript/Exons?db=core;g=ENSG00000182944;r=22:29268009-29300522;t=ENST00000332050][EWSR1 Exon 7]]
        - coord- 29,286,923	29,287,134	2	1	212
        - seq- CTATTCCTCTACACAGCCGACTAGTTATGATCAGAGCAGTTACTCTCAGCAGAACACCTATGGGCAACCGAGCAGCTATGGACAGCAGAGTAGCTATGGTCAACAAAGCAGCTATGGGCAGCAGCCTCCCACTAGTTACCCACCCCAAACTGGATCCTACAGCCAAGCTCCAAGTCAATATAGCCAACAGAGCAGCAGCTACGGGCAGCAGA
      - FLI1 exon - _complicated by multiple splice variants_
    - failed
      - not from UCSC (yet), it will output the whole gene sequence 
      - ?from bed files? 
      - EWSR1-FLI1 fusion sequence- COSMIC-Gene-Analysis-Fusions-by prevalance : http://cancer.sanger.ac.uk/cosmic/fusion/summary?id=166 
- Thought about splitting this to make a little genome to play with, but splits just get me way too many files, a lot with NNNNN only 
  - [[Split text file by percentage  the file from 533,942,310 lines to 10,000 lines- 53394 files
- Adding sequences to a fasta][Added fusion sequence programmatically

***** Human genome builds
:PROPERTIES:
:ID:       D0BD8791-0AE3-472E-A346-C0DD0D79AFD9
:END:

- [[http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/][Human hg38]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/][Human hg19]]
- [[http://useast.ensembl.org/info/website/tutorials/grch37.html][for converting GRCh37 to GRCh38]]
- https://www.biostars.org/p/113100/
  - The actual sequences you'll get from NCBI/UCSC/Ensembl will be identical, but their annotations will be different and (importantly) updated at different frequencies. NCBI's annotation is the "refseq" dataset (the "refGene" track in UCSC), which is essentially a subset of the UCSC and Ensembl annotations. UCSC's annotations are kind of a mess.
****** hg19 
:PROPERTIES:
:ID:       5a7cdb22-c982-4daa-b0ef-799f7347a683
:END:
:LOGBOOK:
CLOCK: [2020-10-05 Mon 13:24]--[2020-10-05 Mon 13:31] =>  0:07
:END:
- hg19 copy-number accessible whitelist
  - subset to autosomes
  - subset against duke
****** Ensembl GRCh37 fastas
:PROPERTIES:
:ID:       B6066026-81E3-49F3-9A0E-C5E610CB86AB
:END:

[[ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/README][Readme]]

hard masked- low complexity regions by RepeatMasker replaced with N's
soft masked- nucleotide replaced w/ lower case
https://www.biostars.org/p/122305/
[[https://groups.google.com/forum/#!topic/rna-star/2wdHXaPv_vU][Dobin suggests hard masking?]]

***** by Institute
:PROPERTIES:
:ID:       48D2EFA9-39A0-4D5F-8AFA-4726E9488D60
:END:

    - genome reference consortium
      - [[http://www.slideshare.net/GenomeRef][genome reference consortium presentations]]
      - [[http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/human/][human genome overview]]

        GRC distributes the reference genome in two versions: one as contigs and the other as assembled chromosomes. The latter is in the "assembled_chromosome" directory. I do not know who are using the contigs, but nearly everyone I know use assembled chromosomes only.

        GRCh37/hg19 and GRCh38 are genome builds rather than annotations, which describe where features are in a given genome build. The actual sequences you'll get from NCBI/UCSC/Ensembl will be identical, but their annotations will be different and (importantly) updated at different frequencies. NCBI's annotation is the "refseq" dataset (the "refGene" track in UCSC), which is essentially a subset of the UCSC and Ensembl annotations. UCSC's annotations are kind of a mess. You'll find genes with the same ID on multiple strand and multiple chromosomes, which makes them a bit useless. Ensembl's annotations typically contain more features than UCSC (so a bit more noise), but they're otherwise much better put together (e.g., you'll never find a gene ID on different strand or different chromosomes) and their IDs are typically easier to map to other things (e.g., gene names, GO and pathway memberships). Ensembl also updates its annotation fairly often and versions everything nicely, so it's quite convenient to report what version you used in a paper (reproducibility is always a good thing). Given the choice, use the Ensembl annotation.

        BTW, don't forget that the various sources can use different names for chromosomes (e.g., chr1 in UCSC is just 1 in Ensembl), so don't mix and match them.

***** 37=hg19 sort of
:PROPERTIES:
:ID:       604441C8-6BCE-426F-9DE9-5D960F75C3D3
:END:

https://www.biostars.org/p/123767/
I believe the genomic content for the two is identical, except for the mitochondrial contig.

The contig names are also different. GRCh37 names them `chr1`, `chr2`,,`chr3`, etc, while hg19 just has `1`, `2`, `3`.

Thus you can use the same GTF file for both (excluding mitochondrial, of course) if you do a simple replace operation for the contig names.

https://wiki.dnanexus.com/Scientific-Notes/human-genome

***** [[http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/][GRC GR/scratch/jeszyman/f/scratch/jeszyman/fishseq/set1/S14-43147_2_sequence.fastq ishseq/set1/S14-43147_2_sequence.fastq Cm38]]
:PROPERTIES:
:ID:       A57F0012-6870-4F79-92A3-C580DB1D33AD
:END:

***** [[http://useast.ensembl.org/info/data/ftp/index.html?redirect=no][Ensembl]]
:PROPERTIES:
:ID:       D18360E5-0B07-4AB6-92F6-D4B717A2E159
:END:

******
:PROPERTIES:
:ID:       B94A8093-3229-41A6-92A6-104D03128BAE
:END:

*******
:PROPERTIES:
:ID:       D47888B2-1210-4C9D-B30C-81672143987A
:END:

*******
:PROPERTIES:
:ID:       68A22065-D788-4653-9BBC-2B5F67AD218A
:END:

*******
:PROPERTIES:
:ID:       687708C8-78BF-4EB9-8C48-F5456E9CF63D
:END:

*******
:PROPERTIES:
:ID:       B0449C36-57B8-4A2A-AE44-67C96C51706D
:END:

*** Clinical Fusion Protein Databases
:PROPERTIES:
:ID:       51EC0C78-7164-43CA-97B1-1FF2205B46A7
:END:

chimer- 
cite:Kim01012010

cite:wang2015fusioncancer

cite:korla2015fare

see Wellcome Trust db and Chimerdb 2.0- cite:scolnick2015efficient

*** Other Databases and Data Sets
:PROPERTIES:
:ID:       396F7A49-ACB4-4F81-8C73-ACC907207BB8
:END:
- DGIdb
  - - https://pubmed.ncbi.nlm.nih.gov/33237278/
- mouse
  - [[http://www.informatics.jax.org/][Jackson Lab mouse genome informatics]]
  - [[http://web.stanford.edu/group/barres_lab/brain_rnaseq.html][mouse cerebral cortex RNA Seq database]]
  - mouse Ensembl genome files
- any
  - for datasets
  - GEO and SRA
  - [[http://www.ebi.ac.uk/arrayexpress/about.html][ArrayExpress]]
** File Formats
:PROPERTIES:
:ID:       5e2c4e39-99bc-4cc8-86a6-c39bea5ac2d8
:END:
- [[https://genome.ucsc.edu/FAQ/FAQformat.html#format1][Nucleic acid data formats]]
- https://help.basespace.illumina.com/articles/descriptive/fastq-files/
- alignment formats 
  - [[https://software.broadinstitute.org/gatk/documentation/article?id=11014][GATK explaination of alignment formats]]
  - https://samtools.github.io/hts-specs/CRAMv3.pdf
- https://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp_v2/Content/Vault/Informatics/Sequencing_Analysis/BS/swSEQ_mBS_FASTQFiles.htm
- bcl file format? 
- mutation annotation format
  - maf format viewer https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html
- gff / gtf
  - [[http://www.gencodegenes.org/gencodeformat.html][Gencode gtf fields]]
  - [[https://www.ensembl.org/info/genome/genebuild/genome_annotation.html#Names][Ensembl annotation]]
    - note that chr.gtf and gtf are same https://www.biostars.org/p/217700/
  - [[https://useast.ensembl.org/info/website/upload/gff.html][Ensembl GFF/GTF format]]
- [[https://genome.ucsc.edu/goldenpath/help/bigWig.html][bigWig format]]
- Variant call format (VCF)
  - http://samtools.github.io/hts-specs/VCFv4.3.pdf
  - http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/VCF%20(Variant%20Call%20Format)%20version%204.0/encoding-structural-variants
  - [[http://samtools.github.io/hts-specs/VCFv4.3.pdf][Current documentation]]
  - [[http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/][VCF page at IGSR]]
- [[https://www.biostars.org/p/84686/][Obi on 1 vs 0 based coordinates]]
- cite:li2009 
*** fastq
:PROPERTIES:
:ID:       59bc8f81-cf57-4e56-a5ff-3b0ff4095a91
:END:

- [[https://www.biostars.org/p/198143/][get sequencer type from fastq header]]

- https://github.com/10XGenomics/supernova/blob/master/tenkit/lib/python/tenkit/illumina_instrument.py#L12-L45
- https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/FileFormat_FASTQ-files_swBS.htm

*** .bed
:PROPERTIES:
:ID:       6DB65B82-D88C-43EA-9D2B-169D9184C9D5
:END:
*** CIGAR- a subformat in SAM w/ three operations
:PROPERTIES:
:ID:       A664A1C3-9130-400D-BE32-D7D58CF80537
:END:

Original CIGAR
match/mismatch
insertion 
deletion
Extended CIGAR
N for skipped bases on the reference- how is this different than an insertion? 
S for soft clipping
H for hard clipping
P for padding 


read in IGV
parameters
reads up to 128 Mbp
tab delimited
per-alignment fields
mandatory

*** Sequence alignment/Map format specification (.sam/.bam)
:PROPERTIES:
:ID:       44889954-63D8-4416-9F28-0691ECF99A9B
:END:
- [[https://genome.sph.umich.edu/wiki/SAM][nice visualization of fields]]
- cite:samspec2017
  - 1
    - 1.3 The header section
      - read groups
        - 
- See [[Samtools]]
- sam- fastq with alignment column 
  - BAM- companion to SAM in compressed binary 
    - .bai is a companion file which contains the index
      - index acts like an external table of contents, and allows programs to jump directly to specific parts of the bam file without reading through all of the sequences. Without the corresponding bam file, your bai file is useless, since it doesn't actually contain any sequence data.
- read groups
  - https://gatkforums.broadinstitute.org/gatk/discussion/6472/read-groups
  - effectively treated as a separate run of the NGS instrument in tools like base quality score recalibration -- all reads within a read group are assumed to come from the same instrument run and to therefore share the same error model
  - GATK tools treat all read groups with the same SM value as containing sequencing data for the same sample
  - https://gatkforums.broadinstitute.org/gatk/discussion/6472/read-groups


** Processes
:PROPERTIES:
:CREATED:  [2020-04-23 Thu 08:54]
:ID:       eaac4fb0-ebc3-43c3-9b27-d0633f728fc9
:END:
:LOGBOOK:
CLOCK: [2020-10-08 Thu 10:05]--[2020-10-08 Thu 10:15] =>  0:10
:END:
*** Setup
- R setup
  #+begin_src R :tangle ./src/setup.R
#
# Check Packages
##
## Bioconductor
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
##
packages = c(
  "ComplexHeatmap",
  "fgsea",
  "limma"
)
##
##NOTE requires libcurl4-openssl-dev https://stackoverflow.com/questions/11471690/curl-h-no-such-file-or-directory/11471743
package.check = lapply(
  packages,
  FUN = function(x) {
    if (!require (x, character.only = T)) {      
      BiocManager::install(x, quietly =T, ask=F)
      library(x, character.only = T)
    }
  }
)
#
#+end_src
**** Biotools docker setup
:PROPERTIES:
:ID:       7ed344ce-373c-4f9e-b24f-87b57374e1a0
:END:
:LOGBOOK:
CLOCK: [2021-03-03 Wed 19:56]--[2021-03-03 Wed 21:25] =>  1:29
:END:
- [[file:~/repos/biotools/Dockerfile]]
- build biotools docker container
  #+begin_src bash :tangle no
cd ~/repos/biotools/
docker build . --file Dockerfile --tag jeszyman/biotools
docker push jeszyman/biotools
#+end_src
- run docker interactively 
  #+name: docker_interactive
  #+begin_src bash :tangle ./src/functions.sh
docker_interactive() {
    if [ -f /.dockerenv ]; then
        echo "shell already in docker, exiting"
        exit 1
    else
        account="USER INPUT"
        container="USER INPUT"
        read -p "docker account name: " account
        account="${account:=jeszyman}"
        read -p "container name: " container
        container="${container:=biotools}"
        case $HOSTNAME in
            (radonc-cancerbio) docker run -it \
                                      --env HOME=${HOME} \
                                      --hostname ${HOSTNAME} \
                                      --user $(id -u ${USER}) \
                                      --volume /home/:/home/ \
                                      --volume /mnt/:/mnt/ \
                                      --volume /tmp/:/tmp/ \
                                      $account/$container \
                                      /bin/bash;;
            (acl*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /drive3/:/drive3/ \
                          -v /duo4/:/duo4/ \
                          -v /home/:/home/ \
                          -u $(id -u ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (ACL*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /home/:/home/ \
                          -v /duo4/:/duo4/ \
                          -u $(id -u ${USER}):$(id -g ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (virtual-workstation*.gsc.wustl.edu) bsub -Is -q docker-interactive -a 'docker($account/'"$container"')' /bin/bash;;
            (*) docker run -it \
                       --env HOME=/home/${USER} \
                       --hostname ${HOSTNAME} \
                       --user $(id -u ${USER}) \
                       --volume /home/:/home/ \
                       --volume /mnt/:/mnt/ \
                       --volume /tmp/:/tmp/ \
                       $account/$container \
                       /bin/bash;;
        esac
    fi
}
#+end_src
**** Setup common references
:PROPERTIES:
:ID:       813ed980-1c55-44d7-84d8-67afdad2e5bc
:END:
  #+name: setup_common_references
  #+begin_src bash :tangle no
#
# Setup
/bin/bash
alias wget_common=wget --noclobber
refdata=
#
# Download reference genomes
## Human
### UCSC Hg38
wget_common https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz -O $refdata/hg38.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.toplevel.fa.gz
#### Primary assembly
wget --no-clobber ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
##
## Rattus norvegicus
### UCSC
wget_common ftp://hgdownload.cse.ucsc.edu/goldenPath/rn6/bigZips/rn6.fa.gz -O $refdata/rn6.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz -O $refdata/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz
#### NOTE: No Rattus primary assembly
#### cDNA
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/cdna/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
gunzip /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
#### GTF
wget_common ftp://ftp.ensembl.org/pub/release-102/gtf/rattus_norvegicus/Rattus_norvegicus.Rnor_6.0.102.gtf.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.102.gtf.gz
#
# Unzip
for file in $refdata/*.fa.gz $refdata/*.gtf.gz;
do
    gunzip_common $file
done
#

#########1#########2#########3#########4#########5#########6#########7#########8
# ideas
#in/out from csv
# get readmes
#########1#########2#########3#########4#########5#########6#########7#########8
#+end_src
*** Platform and Workflows
:PROPERTIES:
:ID:       b7c65054-f648-4579-9c5f-7bedbe272b5c
:END:
- https://sites.google.com/view/williamgilks/computational-analysis-of-whole-genome-sequencing-data/flow-chart-in-latex

https://www.reddit.com/r/bioinformatics/comments/ahbzg5/how_do_people_turn_their_scripts_into_a_program/

- http://www.blopig.com/blog/2018/03/running-jupyter-notebook-on-a-remote-server-via-ssh/
- https://www.reddit.com/r/bioinformatics/comments/7gxsk0/does_anyone_use_cwl_does_it_actually_help_you_get/
- https://cromwell.readthedocs.io/en/stable/
- nextflow
  - https://www.youtube.com/watch?v=jsxTC8pNPUc
    - workflows are fragile
  - https://www.nextflow.io/example1.html
  - https://github.com/nf-core/rnaseq
  - https://www.ncbi.nlm.nih.gov/pubmed/?term=nextflow
- [[https://software.broadinstitute.org/wdl/userguide/][widdle]]
- http://rabix.io/
- http://bioboxes.org/docs/example-biobox-use/
**** Computer clusters
:PROPERTIES:
:CREATED:  [2020-12-18 Fri 18:50]
:ID:       10c078fd-a281-4612-a28b-5ceea1fa7af5
:END:
https://singularity.lbl.gov/install-linux
*** Probability and statistics related to bioinformatics
- clustering
  - http://statweb.stanford.edu/~susan/courses/stats305c/examplesClusters.html
  - https://bookdown.org/rdpeng/exdata/k-means-clustering.html
  - https://www.datanovia.com/en/lessons/cluster-analysis-example-quick-start-r-code/
- https://cran.r-project.org/web/packages/NMF/vignettes/NMF-vignette.pdf
- Pavlou M, Ambler G, Seaman SR, Guttmann O, Elliott P, King M, et al. How to develop a more accurate risk prediction model when there are few events. BMJ 2015;351:h3868.

- https://www.liebertpub.com/doi/abs/10.1089/cmb.2004.11.714
*** Common bioinformatics file manipulation
:PROPERTIES:
:ID:       c0c0ee28-2e41-41a7-9a3b-ae195117a93e
:END:
- [[Question: Selecting Random Pairs From Fastq?]]  
- https://biodbnet-abcc.ncifcrf.gov/db/db2db.php
- [[https://www.biostars.org/p/68283/][count total length of regions in bed file]]
- https://gatkforums.broadinstitute.org/gatk/discussion/6484/how-to-generate-an-unmapped-bam-from-fastq-or-aligned-bam

- https://www.biostars.org/p/247903/
- https://gatkforums.broadinstitute.org/gatk/discussion/2909/how-to-fix-a-badly-formatted-bam
- [[https://github.com/lh3/seqtk][seqtk]] 
- sequence file manipulation
  - https://gist.github.com/davfre/8596159
  - [[https://www.biostars.org/p/6544/][look at all the ways to split a bam file!]]
*** Massively parallel sequencing 
:PROPERTIES:
:ID:       3BC41E6A-00D3-4498-A48C-463F68CFD76B
:END:
:LOGBOOK:
CLOCK: [2021-05-27 Thu 11:25]--[2021-05-27 Thu 11:38] =>  0:13
:END:
**** General reference
- https://docs.nvidia.com/clara/parabricks/v3.0/index.html#
- https://support.illumina.com/training.html
- cite:goodwin2016
- cite:lee2016
**** High Throughput Sequencing Platforms
:PROPERTIES:
:ID:       DF6FFA19-8867-49AC-947D-FD57B403761A
:END:
- https://www.broadinstitute.org/genome-sequencing/broadillumina-genome-analyzer-boot-camp
- commercial sequencing entities
  - medgeneome
    - https://research.medgenome.com/qc-reports/
    - https://research.medgenome.com/ngs-services/whole-genome-exome-sequencing/
- https://www.illumina.com/systems/sequencing-platforms/comparison-tool.html
- https://dnatech.genomecenter.ucdavis.edu/
- [[https://www.youtube.com/watch?v=v1DbcJD4Ry0][youtube Mardis NGS]]
- [[http://www.broadinstitute.org/scientific-community/science/platforms/genome-sequencing/broadillumina-genome-analyzer-boot-camp][Broad / Illumina Genome Analyzer Boot Camp]]
- [[file:~/Box%20Sync/library/Mardis,08%20Next-Generation%20DNA%20%20Sequencing%20Methods.pdf][Mardis,08 Next-Generation DNA  Sequencing Methods]]
- 150924.1600Mardis talk
  - Illumina read length limit is signal to noise
  - Ion torrent difficult for indels with runs of single nucleotide
  - Paired end allows placement of repetitive sequences where one side might be repetitive, the other not
  - Evaluate coverage by comparison to snp array
  - Clinical sequencing false positive if variants have patterns
- Sequencing platform overview and limitations
  - cite:payton2016
- cite:glenn2011     
***** Ion Torrent
:PROPERTIES:
:ID:       EC631BCD-1A9B-4B8B-877E-7861C27D948B
:END:

***** Illumina
:PROPERTIES:
:ID:       D1A3EF94-7933-4854-99ED-C63DE8F7741D
:END:
- adapters
  - only 3' adapters need trim https://support.illumina.com/bulletins/2016/04/adapter-trimming-why-are-adapter-sequences-trimmed-from-only-the--ends-of-reads.html
  - indexing goes on outside of adapter seq- so routine trim will find and clip
  - https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html
- cite:illumina2015nextseq 
- table
    | series  | platform |
    |---------+----------|
    | NovaSeq |     6000 |
    | HiSeq   |     3000 |
    | HiSeq   |     2500 |
    | MiSeq   |          |
    | Nexseq  |          |
    - https://www.illumina.com/systems/sequencing-platforms/comparison-tool.html
https://www.illumina.com/systems/sequencing-platforms/novaseq/specifications.html
output- *.bcl files- base call and quality score per cycle
convert *.bcl to .fastq.gz w/ CASAVA
also demultiplexes 
- [[https://www.illumina.com/systems.html][system specs]]
- PhiX phasing and cluster ctrl https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/hiseq-phix-control-v3-technical-note.pdf
- library preparation kits
  - [[https://www.illumina.com/library-prep-array-kit-selector.html][selector]]
  - https://support.illumina.com/sequencing/sequencing_kits/truseq-small-rna-kit.html
****** Illumina TruSeq
:PROPERTIES:
:ID:       3CE663C4-BF37-4E2E-9A0D-0E822241D05E
:END:

- See Kukurba,15
- Steps
  - RNA isolation
  - Poly-A purification
  - Fragmentation
  - cDNA synthesis
  - Adaptor ligation
  - Size selection
  - PCR amplification

******* Removal of rRNA and tRNA
:PROPERTIES:
:ID:       D509473A-536B-46D2-8639-22DC19DF41E2
:END:

**** Sequencing testing, modeling, and read simulation
:PROPERTIES:
:ID:       f44b9dce-d7fa-4b6a-9949-49ff3c02547c
:END:
- Read simulation 
  - https://medium.com/computational-biology/a-simple-introduction-to-read-simulators-bbeff4f0c0c6
  - RNA-seq reads https://rdrr.io/bioc/polyester/
  - https://github.com/lh3/wgsim
- variant simulator http://bioinform.github.io/varsim/
- Downsample
  - at fastq
    - https://www.biostars.org/p/6544/
  - at bam
    - subsample function for reads instead of fraction
- Reference
  - [[https://www.reddit.com/r/bioinformatics/comments/c313uk/what_are_some_common_tasks_that_i_should_brush_up/][reddit: What are some common tasks that I should brush up on, that pertain to NGS?]]
# #https://www.biostars.org/p/76791/
# function SubSample {
# FACTOR=$(samtools idxstats $1 | cut -f3 | awk -v COUNT=$2 'BEGIN {total=0} {total += $1} END {print COUNT/total}')
# }
# SubSample $localdata/bams/lib080.dedup.bam
# if [[ $FACTOR > 1 ]]
#   then 
#   echo '[ERROR]: Requested number of reads exceeds total read count in' $1 '-- exiting' && exit 1
# fi

# sambamba view -s $FACTOR -t 2 -f bam -l 5 $1

# }

# export -f SubSample

# ls *.bam | parallel "SubSample {} 50000000 > {.}_subsampled.bam"

    - https://bioinformatics.stackexchange.com/questions/402/how-can-i-downsample-a-bam-file-while-keeping-both-reads-in-pairs
    - (no pair preservation) https://www.biostars.org/p/44527/#126428 https://www.biostars.org/p/145820/
    - w/ sambamba # bam_sample(){
#     base=`basename -s .bam $1`
#     sambamba view -f sam -t 30 $1 | shuf -n $2 > $3/${base}_${2}_sample
# }

***** modeling
:PROPERTIES:
:ID:       00983e26-ec16-4f68-be0d-2f59eb31def4
:END:
- http://www.nytimes.com/2012/07/21/science/in-a-first-an-entire-organism-is-simulated-by-software.html?_r=2
- https://www.ebi.ac.uk/biomodels-main/
- https://precision.fda.gov/challenges/
- Synthetic biology- cite:caschera2014
***** Generating Simulated Reads
:PROPERTIES:
:ID:       8877C684-C217-4E7C-8E19-3D972A4015B2
:END:

Generating Simulated Reads
First, we need a small set of sample read data. A number of tools, including ArtificialFastqGenerator, and SimSeq, will generate artificial or simulated sequence data for you. For this tutorial, I chose to use the wgsim tool (created by Heng Li, also the creator of SAMtools).
**** Adapter design
- https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/user-guide-manual/xgen-prism-dna-library-prep-kit-processing-sequencing-data-with-umis.pdf
- https://github.com/pughlab/ConsensusCruncher
**** Library preparation
- https://www.future-science.com/doi/10.2144/000114133
- [[http://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-2-r18][amplification bias]]
**** Sequencing errors
- seq error correction
  - as an information model, propogation of errors, random vs. nonrandom
  - random start-stop- allows ID of PCR duplicates- by start/stop regardless of internal seq
  - ligation umi
    - umi strategies
    - umi actaully adds back 'false' pcr duplicates which can now be positively IDed as unique molecules
  - duplexing and asymmetric strand artifacts
    - distinguishing strands 
- 1st exon GC-rich, poorly seq by illumina 
- https://www.nature.com/articles/s41598-018-29325-6
**** Raw sequence QC and processing
:PROPERTIES:
:ID:       1EE731A3-83DD-46AF-B988-2311307C335F
:CREATED:  [2020-10-14 Wed 08:47]
:END:
:LOGBOOK:
CLOCK: [2020-10-13 Tue 14:00]--[2020-10-13 Tue 14:47] =>  0:47
CLOCK: [2016-04-05 Tue 17:21]--[2016-04-05 Tue 17:32] =>  0:11
CLOCK: [2016-04-05 Tue 17:06]--[2016-04-05 Tue 17:15] =>  0:09
CLOCK: [2016-04-05 Tue 10:44]--[2016-04-05 Tue 11:30] =>  0:46
CLOCK: [2019-07-23 Tue 20:35]--[2019-07-23 Tue 21:13] =>  0:38
CLOCK: [2019-07-23 Tue 19:27]--[2019-07-23 Tue 19:40] =>  0:13
CLOCK: [2019-07-16 Tue 14:46]--[2019-07-16 Tue 15:03] =>  0:17
CLOCK: [2019-07-07 Sun 11:18]--[2019-07-07 Sun 11:34] =>  0:16
CLOCK: [2019-07-06 Sat 08:11]--[2019-07-06 Sat 09:07] =>  0:56
CLOCK: [2019-07-05 Fri 21:00]--[2019-07-05 Fri 22:04] =>  1:04
CLOCK: [2019-06-05 Wed 09:07]--[2019-06-05 Wed 09:47] =>  0:40
CLOCK: [2019-06-03 Mon 18:40]--[2019-06-03 Mon 20:30] =>  1:22
CLOCK: [2019-06-04 Tue 12:39]--[2019-06-04 Tue 13:12] =>  0:33
CLOCK: [2020-10-13 Tue 13:38]--[2020-10-13 Tue 14:00] =>  0:22
:END:
- https://genomics.sschmeier.com/ngs-qc/index.html?highlight=phix#phix-genome
161. Lo,C.-C. and Chain,P.S.G. (2014) Rapid evaluation and quality
control of next generation sequencing data with FaQCs. BMC
Bioinformatics, 15, doi:10.1186/s12859-014-0366-2.


. Chen,C., Khaleel,S.S., Huang,H. and Wu,C.H. (2014) Software for
pre-processing Illumina next-generation sequencing short read
sequences. Source Code Biol. Med., 9, doi:10.1186/1751-0473-9-8.


165. Del Fabbro,C., Scalabrin,S., Morgante,M. and Giorgi,F.M. (2013)
An extensive evaluation of read trimming effects on Illumina NGS
data analysis. PLoS One, 8, e85024.


- raw read trimming and filtering
  - https://www.biostars.org/p/368420/
  - adapter
    - https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary
    - https://www.biostars.org/p/267388/
    - http://seqanswers.com/forums/showthread.php?t=51226
    - https://github.com/griffithlab/rnaseq_tutorial/wiki/Adapter-Trim
    - do all aligners need adaptor trimming? at least common to do so for all
    - https://www.biostars.org/p/368420/
  - trimming- removing bad ends
    - adaptors
    - low quality bases
  - filering- removing bad reads
    - low quality reads
    - contamination
    - low complexity reads
    - short reads
  - https://www.biostars.org/p/212136/
  - https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary
- https://sequencing.qcfail.com/software/fastqc/
- RNA
  - Do not
    - quality-trim RNA- cite:williams2016a
      - see also - http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0085024
        - w/ commentary here- http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/
      - - https://www.michaelchimenti.com/2016/06/trim-rna-seq-reads/
    - remove duplicate- no necessarly PCR artifact
      - https://www.biostars.org/p/55648/
      - https://www.biostars.org/p/14283/
  - https://www.slideshare.net/joachimjacob/2rna-seqpart2investigatin-rawdata
  - https://sequencing.qcfail.com/applications/mrna-seq/
- [[id:c0c0ee28-2e41-41a7-9a3b-ae195117a93e][LINK TO common sequence file manipulation]]
- QC outputs explained- https://sequencing.qcfail.com/
- see - cite:chen2013 for GC bias and effects on assembly
- [[http://bioinformatics-core-shared-training.github.io/cruk-bioinf-sschool/Day1/fastqc_sweave.pdf][de Santiago,15]]
- [[https://www.danielecook.com/double-checking-fastqs/][nice fastq python]]
- https://wikis.utexas.edu/display/bioiteam/Evaluating+your+raw+sequencing+data
- [[id:1165bc2b-9373-4313-94bd-2a1d7be34fb8][LINK TO aligned sequence qc]]
- reference
  - yas
  - [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][docker]] 
    - dockerfile - include current build status in comments
    - sequencing dockerfile repos
      - https://github.com/broadinstitte/gtex-pipeline.git
      - https://hub.docker.com/u/mgibio/
      - https://hub.docker.com/u/biocontainers
  - https://github.com/genome/gms
  - best practices
    - each tool install as standalone
    - Version control 
    - File organization
      - https://libraries.mit.edu/data-management/
      - See [[file:~/maintenance.org::Information Maintenance][information maintenance]] Organization -> File tree structure
    - http://www.broadinstitute.org/gatk/guide/best-practices
      best practices include recording everything
      record of all commands
      record of all software versions
      see Emacs, rpubs.com, sessionInfo and knitR
    - [[http://www.nature.com/sdata/for-authors/submission-guidelines][Nature data submission guidelines]]
    - https://kieranhealy.org/resources/
    - http://kieranhealy.org/files/misc/workflow-apps.pdf
    - see Anders,13
    - no docker in docker for dev- confuses vols, etc. instead add emacs to each container and have local repos of .emacs and org
  - associated mods
    - [[id:43331BB8-4234-48FE-B510-E92F631D5D7B][Code]]
  - compendia
    - [[file:~/Box%20Sync/org/compend.org::*UNIX][UNIX]]
    - [[file:~/.emacs.d/settings.org::*Emacs][Emacs]]
    - git
  - [ ] information management- see cite:griffith2015b           
  - https://github.com/cbcrg/lncrna-annotation-nf
  - set alias https://emacs.stackexchange.com/questions/34022/error-initialization-user-has-no-home-directory
  - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
  - https://hub.docker.com/r/mgibio/mark_duplicates-cwl/dockerfile
  - https://hub.docker.com/r/broadinstitute/picard/dockerfile
  - stack
    - ubuntu
    - git and github
    - docker
      - https://github.com/bioinformatics-core-shared-training/docker-4-bioinformatics
      - https://gatkforums.broadinstitute.org/firecloud/discussion/6886/guidelines-for-working-with-docker-images-and-dockerfiles-broadies-only
    - nextflow
      - https://www.youtube.com/watch?v=jsxTC8pNPUc&t=579s
      - https://github.com/cbcrg/lncrna-annotation-nf
      - https://www.nextflow.io/example1.html
      - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
      - commands
        |----------+---------|
        | function | command |
        |----------+---------|
        | <35>     | <35>    |
        |          |         |
      - processes
      - documentation
- ideas
  - Consider more emacs shell work: 
  - add to gatk docker
    - https://hub.docker.com/u/broadinstitute
    - https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php
  - ideas
    - link to documentation- maybe an org and/or markdown file?
    - script comment design
  - https://gatkforums.broadinstitute.org/gatk/discussion/2909/how-to-fix-a-badly-formatted-bam

**** DNA-seq
:PROPERTIES:
:ID:       728e8984-696f-4531-8e76-bf6d78bfa3b7
:END:
- index, adapter, barcode
- https://www.genscript.com/molecular-biology-glossary/11941/genomic-equivalent
- ATAC-Seq
  - https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/atac-seq.html
- DNA 
  - for immuno regulatory regions- https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-016-0220-7
  - for chromatin accessibility
    - ATAC-seq
      - cite:buenrostro2015
      - FAST ATAC- ATAC-Seq optimized for blood
  - DNA for sequence variation 
    - variation
      - visualizations
        - - https://mail.google.com/mail/u/0/#inbox/FMfcgxwHNCtwvpXGcckxJfDWPTPKHTst
      - clingen
        - https://www.youtube.com/channel/UCsn4nEVUTpVQz70rClgMMsQ?view_as=subscriber
    - https://gatk.broadinstitute.org/hc/en-us/articles/360035535892-Somatic-copy-number-variant-discovery-CNVs-
    - https://gatk.broadinstitute.org/hc/en-us/articles/360035535912-Data-pre-processing-for-variant-discovery
    - https://gatk.broadinstitute.org/hc/en-us/sections/360007226651-Best-Practices-Workflows
    - https://mail.google.com/mail/u/0/#inbox/FMfcgxwHNWHJNLRlZwfGkGJHwJnBxFlh
  - for sequence homology- cite:stoeger2018 
  - cfDNA
    - cite:huang2019
- marking duplicates :process: 
  - consider picard markduplicates - https://gatkforums.broadinstitute.org/gatk/discussion/6945/picard-tools-markduplicates-using-cram-format-how-to-pass-a-valid-cram-reference
  - samtools
    - https://samtools.github.io/hts-specs/SAMv1.pdf
    - sort
      - https://www.biostars.org/p/247588/
    - http://www.htslib.org/algorithms/duplicate.html
  - https://qcb.ucla.edu/wp-content/uploads/sites/14/2016/03/GATKwr12-2-Marking_duplicates.pdf
***** Paired tumor-normal sequencing
- https://github.com/mskcc/facets
***** Tumor-only whole exome sequencing
:PROPERTIES:
:CREATED:  [2021-05-01 Sat 07:26]
:END:
- https://ascopubs.org/doi/full/10.1200/CCI.19.00130
- https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)31635-8
- cite:carter2012 ABSOLUTE
  - https://software.broadinstitute.org/cancer/cga/absolute
  - https://www.biostars.org/p/175131/
***** DNA Bioinformatics
:PROPERTIES:
:ID:       1C24E3B6-CE16-47C8-8D7B-95EB530560F1
:END:

- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4669575/
- https://github.com/kevinblighe/ClinicalGradeDNAseq
- https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html

- https://software.broadinstitute.org/gatk/documentation/tooldocs/current/picard_illumina_IlluminaBasecallsToFastq.php

- ffpe
  - cite:munchel2015
  - cite:spencer2013
- Fonseca,12 Tools for mapping high throughput sequencing data(2)
- [[file:~/Box%20Sync/scientific_technique/Cruchaga%20lab%20eQTLs.pptx][Cruchaga eQTLs]]

****** Post-alignment Samtools
:PROPERTIES:
:ID:       2D153E08-9F4A-4564-ACA7-5B4932D1022B
:END:

- nice header review https://sarahpenir.github.io/bioinformatics/awk/calculating-mapping-stats-from-a-bam-file-using-samtools-and-awk/
- Post-alignment
  - samtools
    - [[http://www.htslib.org/][website]]
      - [[http://www.htslib.org/workflow/][workflows]]
      - [[http://www.htslib.org/doc/][documentation]]
        - [[http://www.htslib.org/doc/#publications][publications]]
    - [[https://github.com/samtools/][github]]
    - cite:li2009 
    - samtools view sorted.bam | awk '{sum+=$5} END { print "Mean MAPQ =",sum/NR}'
    - [[https://www.biostars.org/p/15953/][biostars: calculate MAPQ]]
    - calc mapq w/ samtools- https://www.biostars.org/p/15953/][Malachi Griffith biostars
- [[http://www.htslib.org/doc/samtools-1.2.html][Documentation]]
- Subset a bam
  - By chromosome
    - e.g.: ​samtools view -b in.bam chr1 > in_chr1.bam
- [[http://barcwiki.wi.mit.edu/wiki/SOPs/miningSAMBAM][some common Samtools tasks]]
- [[http://davetang.org/wiki/tiki-index.php?page=SAMTools][a Samtools tutorial]]
- [[http://biobits.org/samtools_primer.html][another Samtools tutorial]]
- [[http://broadinstitute.github.io/picard/explain-flags.html][SAM flags]]
- [[https://www.biostars.org/p/17361/][Picard is Samtools!]]
- Find ummapped reads by flags
- [[http://davetang.org/wiki/tiki-index.php?page=SAMTools#Converting_a_SAM_file_to_a_BAM_file][Sam to Bam, indexing ]]   

****** Ideas
:PROPERTIES:
:ID:       F6511EA3-A5B4-455E-99F1-D536EE81F22D
:END:

- [[http://zvfak.blogspot.com/2012/02/samtools-in-parallel.html][run in parallel]]

- incorporate samtools in R workflow via Rsamtools

****** Exome Sequencing Pipeline
:PROPERTIES:
:ID:       57992D53-D767-4E44-9E3F-B4A05E5B4EF1
:END:

****** Specialized Applications
:PROPERTIES:
:ID:       7A9FC47A-2E9E-46D5-8E3E-E7C7C5C5AD9E
:END:

******* Detecting Internal Tandem Duplications
:PROPERTIES:
:ID:       4B340D6D-E38D-42E5-AD1F-3CB0BDF62545
:END:

[[file:~/Box%20Sync/library/Spencer,12%20Detection%20of%20FLT3%20ITD%20in%20short%20read%20length%20NGS.pdf][Spencer,12 Detection of FLT3 ITD in short read length NGS]]

****** DNA Alignment
:PROPERTIES:
:ID:       B2F17B06-C1BA-487A-93EC-D9737584BC0E
:END:

- https://www.youtube.com/watch?v=4n7NPk5lwbI 
- - https://www.youtube.com/watch?v=hpb-mH-yjLc&list=PL2mpR0RYFQsBiCWVJSvVAO3OJ2t7DzoHA
- tutorials
  - http://seqinformatics.com/?page_id=160
    - near perfect matches (BLAST not specific)
  - https://www.youtube.com/watch?v=zMAa9gFd2Gs
  - http://www.ngscourse.org/Course_Materials/alignment/tutorial/example.html
  - https://wikis.utexas.edu/display/bioiteam/Mapping+tutorial

******* Novoalign
:PROPERTIES:
:ID:       22097A02-1E18-40BD-BA2F-0D3BC645F0D5
:END:

- cite:Novocraft
- [[http://www.novocraft.com/support/download/][Novoalign]]

******* BWA
:PROPERTIES:
:ID:       83F91393-2108-4785-9696-BDE8EC09FA88
:END:

-"bwa concatenates all the references sequences together before aligning. So if a read hangs off of one sequence onto the next, it's given the appropriate mapping position, and the unmapped flag is also set, as a sign that something is off about the alignment."

- sequences should be <3% difference from reference
- A block sorting lossless data compression algorithm
- Compresses the genome as a BWT index
- Identical sequences (genome repeats) only occur once in the index, reads that match the genome in multiple locations get a mapping quality of 0
- output is SAM
- [[http://bio-bwa.sourceforge.net/][Burrows-Wheeler Aligner]]
- [[https://www.biostars.org/t/bwa%20mem/][Biostars BWA Posts]]
- [[http://bio-bwa.sourceforge.net/bwa.shtml][bwa manual]]
  - algorithms
    - backtrack illumina to 100 bp
    - 70 bp to 1 mb
      - sw 
      - mem
    - mem is generally recommended
    - 

****** Post-Alignment QC
:PROPERTIES:
:ID:       F689D6FA-071C-4167-BC21-85A7EC20DA41
:END:

filtering
noncoding, known mis-spliced, mitochondrial, x/y in mixed sex, unannotated 

******* Inspection
:PROPERTIES:
:ID:       07602EB8-5EFD-4179-9F55-320190ECE670
:END:

******* Processing
:PROPERTIES:
:ID:       2A4B2B42-C088-4479-BC65-670CC8377D39
:END:

******* FASTX-Toolkit
:PROPERTIES:
:ID:       8169C9DC-FDB6-473E-9B54-E154261186EA
:END:

  - [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][documentation]]
  - brew installed on mac, but says, "cannot execute binary file", will have to run from cluster
  - [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][Command-line Usage]]
  - the clip can be a .fa file 
  - [[http://mgt2.chpc.wustl.edu/wiki119/index.php/FASTX_Toolkit][on cluster]]
  - Clipper 
    https://github.com/NCBI-Hackathons/RNA_mapping

****** Broad Best Practices for Variant Discovery in DNAseq
:PROPERTIES:
:ID:       850AA38C-75E9-42E2-8DEE-61F4950BCFDE
:END:

https://software.broadinstitute.org/gatk/documentation/topic?name=methods

***** Adaptive TCR-beta immunosequencing 
:PROPERTIES:
:CREATED:  [2020-12-20 Sun 16:45]
:ID:       61aac189-1770-4d80-a367-e7c39b46e0d1
:END:
***** aligned sequence qc
:PROPERTIES:
:ID:       1165bc2b-9373-4313-94bd-2a1d7be34fb8
:END:
- https://sarahpenir.github.io/bioinformatics/awk/calculating-mapping-stats-from-a-bam-file-using-samtools-and-awk/
- coverage
  - cite:illumina2014
  - Picard CollectWgsMetrics
****** coverage and depth
:PROPERTIES:
:ID:       a4dcf0d6-1619-4f82-989d-2bd599cdf061
:END:
- https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/hiseq-x-30x-coverage-technical-note-770-2014-042.pdf
- https://www.biostars.org/p/331957/
- https://www.htslib.org/doc/samtools-depth.1.html#OPTIONS
- for coverage https://www.biostars.org/p/104063/
***** capture probes and barcoding
:PROPERTIES:
:ID:       675eba72-76e5-4b49-81e9-37bdc02cce9b
:END:
https://support.illumina.com/bulletins/2018/08/understanding-unique-dual-indexes--udi--and-associated-library-p.html
https://www.illumina.com/content/dam/illumina-marketing/documents/products/whitepapers/index-hopping-white-paper-770-2017-004.pdf
https://www.google.com/search?q=calculation+probability+of+sequence+mismatch&oq=calculation+probability+of+sequence+mismatch&aqs=chrome..69i57j33.12480j0j7&client=ubuntu&sourceid=chrome&ie=UTF-8
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245947/
- capture probe design
  - https://ascopubs.org/doi/full/10.1200/CCI.19.00077?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed
   Contigs: group of sequences representing overlapping regions from a genome or transcriptome.
- [[https://wikis.utexas.edu/display/GSAF/Illumina+-+all+flavors][illumina adaptors explained]]
- adaptor strategies
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/case-study/idt_improved-coverage-uniformity-with-xgen-lockdown-panels-case-study.pdf?sfvrsn=c33d1907_6
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/protocol/idt_xgen-prism-dna-library-prep-kit-quick-reference-guide.pdf?sfvrsn=6fe71207_4
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/user-guide-manual/xgen-prism-dna-library-prep-kit-processing-sequencing-data-with-umis.pdf?sfvrsn=22991407_10
  - https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4428-5#MOESM1
- UMI
  - - [[http://www.genomics.agilent.com/article.jsp?pageId=3081][HaloPlex]]
  - - https://cgatoxford.wordpress.com/2015/08/14/unique-molecular-identifiers-the-problem-the-solution-and-the-proof/
- barcoding
  - 10x
- https://pdfs.semanticscholar.org/310b/3bac42989485c98406848217418ff22c22e7.pdf
- barcodes and adaptors for increased sensitivity 
  - cite:kennedy2014
  - cite:newman2016 
  - cite:wang2019
  - https://www.youtube.com/watch?v=HIVCQ7iaowk
  - newman2016
    - adaptor informatics
      - consensus building
        - set "family size"
        - 
    - error base signature
      - C>A C>T 
  - wang2019
    - IDT xGen
      - blocks adaptor to limit off-target
      - Cot-1 blocks repetitve regions
    - method explanation followed by experiments to validate
      - present experiments as condition vs. condition 

- see edit distance in [[id:A91BF233-F14E-4E91-A1E9-B39802A9ED2D][Computers, Computer Science, and Information Technology]]
**** Whole-genome sequencing
:PROPERTIES:
:ID:       ccab4617-0399-4ba3-828e-b5fbbb972ba1
:END:
- cite:campbell2020
- https://www.nejm.org/doi/10.1056/NEJMoa2024534?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub++0pubmed
- duplication rate as a measure of lib complexity
*** Nucleic acid sequence alignment
:PROPERTIES:
:ID:       326ecd60-8cd4-4815-a389-967b2c3fef0a
:END:

73. Li,H. and Homer,N. (2010) A survey of sequence alignment
algorithms for next-generation sequencing. Brief. Bioinform., 11,
473–483.
174. Bao,S., Jiang,R., Kwan,W., Wang,B., Ma,X. and Song,Y.-Q. (2011)
Evaluation of next-generation sequencing software in mapping and
assembly. J. Hum. Genet., 56, 406–414.

- hash table like BLAST or suffic/prefix like Burrows-Wheeler Transform
- https://davetang.org/muse/2017/10/25/getting-started-hisat-stringtie-ballgown/
- prerequisites
  - adaptor identification and trimming
- pseudoaligner
  - kallisto
  - Salmon
- Aligners
  - TopHat2
  - Specialized aligners
    - Fusions
      - TopHat Fusion
        - tophat fusion documentation http://ccb.jhu.edu/software/tophat/fusion_manual.html 
      - STAR for chimeric reads / STAR-Fusion
        - [[https://github.com/STAR-Fusion/STAR-Fusion/wiki][website]]
        - From cite:star_2.5_dobin Chapter 5
        - From cite:dobin2015mapping Alternate Protocol 6
          - --chimSegmentMin <N> is minimum allowed length for each of the chimeric segments
        - additional outputs are
          - Chimeric.out.sam
          - Chimeric.out.junction- a STAR-specific format
    - To transcriptome
      - 
  - [[*STAR][STAR]]
- HISAT2
*** Sequence variation and variant calling
:PROPERTIES:
:ID:       57E81A44-3233-479B-8264-2CD5F1904A60
:END:
:LOGBOOK:
CLOCK: [2020-07-07 Tue 12:17]--[2020-07-07 Tue 12:52] =>  0:35
CLOCK: [2019-04-26 Fri 11:28]--[2019-04-26 Fri 11:44] =>  0:16
CLOCK: [2019-04-26 Fri 11:03]--[2019-04-26 Fri 11:13] =>  0:10
CLOCK: [2019-04-26 Fri 10:43]--[2019-04-26 Fri 10:57] =>  0:14
:END:
- cite:kocakavuk2021 
- https://www.bioconductor.org/packages/devel/bioc/vignettes/TCGAbiolinks/inst/doc/index.html
- https://github.com/raerose01/deconstructSigs
- https://github.com/google/deepvariant 
- variant calling
  - fusion
    - - https://www.biorxiv.org/content/biorxiv/early/2019/01/16/521039.full.pdf
- cite:kulkarni2014 
- [ ] cite:leonard2016 chapter 60 
- https://www.broadinstitute.org/partnerships/education/broade/best-practices-variant-calling-gatk-1
- BreakDancer
  - cite:chen2009
  - [[http://gmt.genome.wustl.edu/packages/breakdancer/][MGI website]]
  - cite:chen2009
- by input
  - wgs
    - [[http://zzz.bwh.harvard.edu/plink/profile.shtml#dose][plink]]
- by variant class
  - structural 
  - indel
  - SNP / SNV
    - cite:nielsen2011 
  - cna
    - - for CNV method see prichard paper- cite:armenia2018
    - [ ] ("copy number variation" OR "copy number alteration") AND (ctDNA or "circulating tumor DNA" ) SEARCH TERMS ([[https://www.ncbi.nlm.nih.gov/pubmed/?term=(%22copy+number+variation%22+OR+%22copy+number+alteration%22)+AND+(ctDNA+or+%22circulating+tumor+DNA%22+)][PubMed]]) 21 hits on 2019-04-17
      - https://www.ncbi.nlm.nih.gov/pubmed/30371735       
      - irrelevant
        - AR-specific gain https://www.ncbi.nlm.nih.gov/pubmed/27191887
        - CTCs https://www.pnas.org/content/110/52/21083.long
        - mlpa https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5593570/
        - https://www.ncbi.nlm.nih.gov/pubmed/28448765
        - partek software w/ normalizer pop https://febs.onlinelibrary.wiley.com/doi/full/10.1002/1878-0261.12077
        - ddpcr https://www.ncbi.nlm.nih.gov/pubmed/29717451
        - uses guardant CNA https://ascopubs.org/doi/full/10.1200/JCO.18.00328?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%3dpubmed
        - https://www.liebertpub.com/doi/full/10.1089/dna.2017.4112?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed&
        - heme malig https://www.ncbi.nlm.nih.gov/pubmed/30371735
        - casadio- uses MLPA
    - comparitor
      - single normal
      - none- assumed 0 across bins
      - copycat uses paired tumor/normal PER sample 
    - tools
      - https://cnvkit.readthedocs.io/en/stable/pipeline.html#reference
        - cite:talevich2016 
        - cite:mansukhani2018
        - cite:mayrhofer2018
        - https://www.biostars.org/p/265951/
      - used in cite:quigley2018
        - https://github.com/chrisamiller/copyCat/commits?author=chrisamiller
        - canvas
          - https://github.com/Illumina/canvas
          - https://academic.oup.com/bioinformatics/article/32/15/2375/1743834
        - circular binary segmentation? 
      - other 
        - https://github.com/bcbio/bcbio-nextgen/issues/2254
        - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870863/pdf/btx284.pdf
        - https://academic.oup.com/bioinformatics/article/30/17/i639/201399
        - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0617-1
        - EXCAVATOR2- https://www.ncbi.nlm.nih.gov/pubmed/27507884
    - seq - depth? cite:mansukhani2018
    - cite:rieber2017
    - cite:miller2011a
- [[https://gatkforums.broadinstitute.org/gatk/discussion/39/variant-quality-score-recalibration][gatk: variant quality score recalibration]]
- http://www.nature.com/nrg/journal/v12/n6/full/nrg2986.html
- Splice site prediction
  - cite:jian2014
- Variant Effect Predictor (VEP)
  - [[https://useast.ensembl.org/info/docs/tools/vep/index.html][website]]
  - cite:mclaren2016

**** structural variation 
:PROPERTIES:
:ID:       0f679231-b782-4001-b7c3-91ec3a8816d2
:END:
https://www.nature.com/articles/s41598-020-71168-7
***** sv
:PROPERTIES:
:CREATED:  [2019-08-15 Thu 15:43]
:ID:       5ff1bbbb-3471-4f2a-b9eb-43b293eee807
:END:
:LOGBOOK:
CLOCK: [2019-10-11 Fri 11:03]--[2019-10-11 Fri 11:38] =>  0:35
CLOCK: [2019-10-10 Thu 17:16]--[2019-10-10 Thu 18:16] =>  1:00
CLOCK: [2019-10-10 Thu 15:53]--[2019-10-10 Thu 16:54] =>  1:01
CLOCK: [2019-10-07 Mon 17:34]--[2019-10-07 Mon 17:43] =>  0:09
:END:
[[file:~/repos/sv/sv.org]]
[[file:~/repos/sv/Dockerfile]]
- sv
  - https://github.com/stat-lab/EvalSVcallers/tree/master/scripts
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5
- [-] build docker image into fat bioinformatics-toolkit container 
- [ ] build slim docker image 
- reference
  - [[~/repos/.emacs.d/snippets/org-mode/biopipe][YAS]] 
  - [[id:333E498B-E0EA-460C-95EC-1931ACE19473][mod]]
    - attributes
      | isa | module                     |
      | isa | work module                |
      | isa | engineering module         |
      | isa | code module                |
      | isa | process development module |
  - principles and philosophy
    - antifragility
    - minimization
    - transparency
    - reproducibility
    - efficiency
    - ?intpretabilty? - code logging, etc 
  - best practices
    - each tool install as standalone
    - Version control - git / github
    - File organization
      - https://libraries.mit.edu/data-management/
      - See [[file:~/maintenance.org::Information Maintenance][information maintenance]] Organization -> File tree structure
    - https://github.com/kylelobo/The-Documentation-Compendium/
    - http://www.broadinstitute.org/gatk/guide/best-practices
      best practices include recording everything
      record of all commands
      record of all software versions
      see Emacs, rpubs.com, sessionInfo and knitR
    - [[http://www.nature.com/sdata/for-authors/submission-guidelines][Nature data submission guidelines]]
    - https://kieranhealy.org/resources/
    - http://kieranhealy.org/files/misc/workflow-apps.pdf
    - see Anders,13
    - no docker in docker for dev- confuses vols, etc. instead add emacs to each container and have local repos of .emacs and org
    - prefer binaries over source- only real advantage to source is custom build
  - stack
    - [[id:c28ef36f-5b5c-4789-90da-270e56fc4898][ubuntu]]
    - [[id:E56ADCB0-F6DD-4D58-8B68-CEDF7DC176E6][git and github]]
    - [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][docker]]
      - [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][docker]] 
        - dockerfile - include current build status in comments
        - sequencing dockerfile repos
          - https://github.com/broadinstitte/gtex-pipeline.git
          - https://hub.docker.com/u/mgibio/
          - https://hub.docker.com/u/biocontainers
    - nextflow
      - https://www.youtube.com/watch?v=jsxTC8pNPUc&t=579s
      - https://github.com/cbcrg/lncrna-annotation-nf
      - https://www.nextflow.io/example1.html
      - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
      - commands
        |----------+---------|
        | function | command |
        |----------+---------|
        | <35>     | <35>    |
        |          |         |
      - processes
      - documentation
    - compendia
      - [[file:~/Box%20Sync/org/compend.org::*UNIX][UNIX]]
      - [[file:~/.emacs.d/settings.org::*Emacs][Emacs]]
      - git
    - set alias https://emacs.stackexchange.com/questions/34022/error-initialization-user-has-no-home-directory
    - https://github.com/cbcrg/lncrna-annotation-nf
    - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
    - https://github.com/bioinformatics-core-shared-training/docker-4-bioinformatics
    - https://gatkforums.broadinstitute.org/firecloud/discussion/6886/guidelines-for-working-with-docker-images-and-dockerfiles-broadies-only
    - [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][Docker]]
      - https://www.boynux.com/containers-fat-thin/
  - examples 
    - https://wiki.cyverse.org/wiki/display/TUT/Biocontainers+for+the+Discovery+Environment
    - https://blog.bananacoding.com/2016/08/29/how-to-build-bioinformatic-tools-image-for-docker/#.XSvInN-YW02
  - package managers
    - bioconda-
      - EZ to add to dockerfile,
      - source difficult to understand, but can see sh and yaml- e.g. https://github.com/bioconda/bioconda-recipes/tree/master/recipes/samtools
      - https://www.nature.com/articles/s41592-018-0046-7
  - modules
  - https://www.reddit.com/r/bioinformatics/comments/d260ok/no_your_ten_rules_arent_simple_and_they_wont_fix/
  - [X] module instance startup- upon initial entry into work heading of a module
    - identify and invoke most specific applicable module 
    - invoke module hierarchy
  - [X] setup repository artifact
    - github and local 
    - Dockerfile
    - org file 
  - https://pmbio.org/module-05-somatic/0005/03/02/Somatic_SV_FilteringAnnotationReview/
- ideas
  - module improvement
    - work module
      - focus on minimal work to closeout this objective
        - move unnecessary work to ideas
  - https://nextflow-io.github.io/nf-hack18/training.html
  - Consider more emacs shell work: 
  - add to gatk docker
    - https://hub.docker.com/u/broadinstitute
    - https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php
  - optimization
    - Lock pipe
    - [[https://www.biostars.org/p/241437/][bioinformatics software distribution]] 
  - extend    
  - [ ] information management- see cite:griffith2015b           
    - https://github.com/genome/gms
  - https://stackoverflow.com/questions/31578446/running-a-script-inside-a-docker-container-using-shell-script
  - https://software.broadinstitute.org/gatk/documentation/article?id=11101
- [ ] closeout
  - [ ] update mod- consolidate and functionalize common tasks
  - [ ] export


:PROPERTIES:
:ID:       2D39899E-F415-4B16-B5ED-4A36465F4CB4
:END:
https://lbgi.fr/AnnotSV/
- cite:kumar2016
  [[http://svviz.readthedocs.io/en/latest/install.html][documentation]]
  [[http://breakdancer.sourceforge.net/][Breakdancer]]

***** Fusion Proteins
:PROPERTIES:
:ID:       C3E4A442-77C6-4A0C-8A9A-3FAA04DAF632
:END:

- [ ] cite:newman2014a 
http://ignyta.com/providers/dx-molecular-diagnostic-cancer-testing/
- cite:beadling16_multip_amplic_approac_detec_gene
- cite:zhao2014detecting
- cite:zheng2013oncogenic
- cite:bunting2013end
- cite:maher2009transcriptome
- cite:scolnick2015efficient


- cite:kumar2016comparative
- cite:korla2015fare
- cite:wang2015fusioncancer
- cite:webster2015mining
- cite:pekar2015fluorescence
- cite:alves14_next_gener_sequen_reveal_novel
- cite:qadir2014childseq
- cite:greaves14_janet_rowley
- cite:heydt2014comparison
- cite:yoshihara2014landscape

- cite:smallridge2013rna

- cite:duncavage2012targeted
- cite:edgren2011identification



- [[file:~/Box%20Sync/clin_research/fishseq/Fusion_Gene_Detection.pdf][Partek fusion detection book]]

****** Low Yield
:PROPERTIES:
:ID:       65A89F9B-9FCE-420F-906A-BD7EC0C23919
:END:

- cite:wheeler2013human
- cite:wilhelm2009rna
- purple book

****** Alignment
:PROPERTIES:
:ID:       CDEC529A-B11D-4890-B75D-9B9470D8F856
:END:

cite:williams2014rna

https://www.biostars.org/p/45986/#144082
[[https://www.biostars.org/p/158039/][visualization tools]]
[[https://sourceforge.net/p/samtools/mailman/message/30210230/][Dobin on SAM handling of chimerica alignment]]
https://igor.sbgenomics.com/public/pipelines/540dd2fad79f00766c174eb0/

cite:yoshihara2014landscape

cite:dobin12_star

******* Aligners, unranked
:PROPERTIES:
:ID:       5DE5AD4D-7634-46D7-86F4-6111B2A3A3B5
:END:

  - [[https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3691734/][FusionQ, 2013]] https://sites.google.com/site/fusionq1/home/
  - [[http://genome.cshlp.org/content/early/2015/11/10/gr.186114.114?top=1][using both DNA and RNA-seq]]
  - [[https://github.com/ndaniel/fusioncatcher/blob/master/doc/manual.md#2---hardware-requirements-and-dependencies][FusionCatcher]]
  - [[https://github.com/Oshlack/JAFFA/wiki][JAFFA]]
  - SOAPfuse
    - https://www.biostars.org/p/144077/
    - vs star https://www.biostars.org/p/171531/#171544

    comparisons
    cite:kumar2016comparative

******** Chimerascan
:PROPERTIES:
:ID:       A6FFE9F2-4004-429D-8057-30D663E4069C
:END:

https://code.google.com/archive/p/chimerascan/wikis/Installation.wiki

******** Ericscript
:PROPERTIES:
:ID:       5B8010B9-93DB-4CC2-9321-210E7CB045F8
:END:

    - NOTE- the ericscript samtools version wasn't going b/c I was asking for samtools, not ./samtools !
      - [[https://sites.google.com/site/bioericscript/home][website]]
      - [[https://groups.google.com/forum/#!forum/ericscript][google group]]
      - 
        cite:Benelli01122012

****** Reviews
:PROPERTIES:
:ID:       EE65AF13-DB95-4F64-8640-4FC890197DCD
:END:

cite:Latysheva02062016
cite:stransky2014landscape




http://bib.oxfordjournals.org/content/early/2012/08/09/bib.bbs044.short?rss=1
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3595110/
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3713712/
https://nar.oxfordjournals.org/content/early/2015/11/17/nar.gkv1234.full

******* Clinical
:PROPERTIES:
:ID:       9D02E90C-AE0B-4805-8F78-7FF6787B4606
:END:

cite:maher2009transcriptome

cite:mitelman2007impact

cite:lieber2016mechanisms

****** Human Pathological Translocations
:PROPERTIES:
:ID:       92C7145A-9535-4E74-B4FF-88A90EE6A78F
:END:

cite:lipson2012identification 

cite:bunting2013end

cite:zheng2013oncogenic

cite:greaves14_janet_rowley

cite:wheeler2013human

****** [[Clinical Fusion Protein Databases]]
:PROPERTIES:
:ID:       4AE77A12-C491-4590-9F8B-6C6EC4077242
:END:

****** Clinical RNA-seq Assays to Detect Fusion Proteins
:PROPERTIES:
:ID:       9786E312-B8BC-4F14-BAF2-C9B682703FB7
:END:

cite:beadling16_multip_amplic_approac_detec_gene

cite:qadir2014childseq


cite:scolnick2015efficient

******* Specific Cancers
:PROPERTIES:
:ID:       E44CCF9D-6D9A-4F01-B90F-929D038CF4F4
:END:

cite:alves2015next

cite:smallridge2013rna

cite:edgren2011identification

cite:pekar2015fluorescence

cite:duncavage2012targeted

****** [[FISH]]
:PROPERTIES:
:ID:       F54BB35F-3BAA-4AA9-928C-18FAB543D7C6
:END:

****** Double Strand Break Repair
:PROPERTIES:
:ID:       642DB59E-D430-4E2A-9CBC-9DF12D6FF54B
:END:

****** Post-Alignment Analysis Programs
:PROPERTIES:
:ID:       EC14669D-1B71-45FD-9D4D-62E405CE6917
:END:

https://rdrr.io/bioc/chimera/man/starReads.html
http://www.bioconductor.org/packages/release/bioc/vignettes/chimera/inst/doc/chimera.pdf
https://github.com/STAR-Fusion/STAR-Fusion/wiki

cite:chimera_r

**** Variant Calling
:PROPERTIES:
:ID:       E3637686-6560-4088-9784-0A601F39C34D
:END:
- clonal evolution- https://cancerdiscovery.aacrjournals.org/content/8/11/1390.long
- HRDtools for homologous recombination index 
  - https://github.com/eyzhao/hrdtools
- cite:ainscough2018  
- - cite:barnell2018  
- [[https://software.broadinstitute.org/gatk/best-practices/bp_3step.php?case=GermShortWGS][GATK Best Practices Germline SNP and Indel]]
- [[http://www.htslib.org/workflow/][WGS/WES Mapping to Variant Calls - Version 1.0]]
- [[https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.php][Broad UnifiedGenotyper]]
- [[https://software.broadinstitute.org/gatk/documentation/article?id=3238][Broad best practices DNAseq variant]]
- cite:vcf
http://swog.org/Visitors/Fall13GpMtg/plenary/03-Corless-SWOG-Genomics-TM-ver1.pdf

***** GATK Tool Set
:PROPERTIES:
:ID:       B4100085-BD16-442F-9AB8-586241CD897A
:END:

****** GATK UnifiedGenotyper
:PROPERTIES:
:ID:       0D7356A4-E139-4BDF-BD6B-4170C15531F9
:END:

- one of several GATK variant discovery tools 

****** DONE Setup GATK
CLOSED: [2016-12-05 Mon 14:22]
:PROPERTIES:
:ID:       E38AAAE9-DB89-42C8-9CE3-C18003F720D3
:END:

- resuem here https://software.broadinstitute.org/gatk/documentation/quickstart.php

- [[https://software.broadinstitute.org/gatk/documentation/quickstart.php][Getting Started]]
  - "Unlike C-compiled programs such as Samtools, GATK cannot simply be added to your PATH, so we recommend setting up an environment variable to act as a shortcut."

**** sequence variation
:PROPERTIES:
:ID:       ec437673-e697-4c0d-a65f-46aec2641ed2
:END:
- https://ai.googleblog.com/2020/09/improving-accuracy-of-genomic-analysis.html?m=1
***** variant calling
:PROPERTIES:
:ID:       743b5990-53e6-4cce-8adc-a9d766d1c5e2
:END:
****** SCNA variants 
:PROPERTIES:
:ID:       179ce37c-f8cf-44b3-872b-e32677bcb49c
:END:
- SCNAs
  - https://pmbio.org/module-05-somatic/0005/04/02/Somatic_CNV_FilteringAnnotationReview/
  - https://www.annualreviews.org/doi/full/10.1146/annurev-genom-082410-101404?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed
  - cite:quigley2018
  - https://bioconductor.org/packages/release/bioc/html/DNAcopy.html
  - cite:talevich2016
  - [[file:~/repos/copy-number/]]
  - [[file:~/repos/copy-number/copy-number.org]]
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-S11-S1
  - https://www.youtube.com/watch?v=m9XPWJrMThY&t=8s
  - https://deeptools.readthedocs.io/en/latest/content/installation.html
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5737511/
- WGS callers
  - https://github.com/GavinHaLab
**** copy-number                                                    :biopipe:
:PROPERTIES:
:CREATED:  [2019-05-29 Wed 14:46]
:ID:       0206fb96-8785-47fb-87ca-1f597de0ee85
:END:
:LOGBOOK:
CLOCK: [2019-10-10 Thu 15:45]--[2019-10-10 Thu 15:53] =>  0:08
CLOCK: [2019-05-15 Wed 13:09]--[2019-05-15 Wed 13:24] =>  0:15
CLOCK: [2019-06-25 Tue 07:51]--[2019-06-25 Tue 08:00] =>  0:09
CLOCK: [2019-06-23 Sun 07:49]--[2019-06-23 Sun 07:53] =>  0:04
CLOCK: [2019-06-19 Wed 14:28]--[2019-06-19 Wed 14:49] =>  0:21
CLOCK: [2019-06-19 Wed 12:38]--[2019-06-19 Wed 13:33] =>  0:55
CLOCK: [2019-06-12 Wed 14:40]--[2019-06-12 Wed 14:58] =>  0:18
CLOCK: [2019-05-29 Wed 15:23]--[2019-05-29 Wed 15:34] =>  0:11
CLOCK: [2019-05-29 Wed 14:46]--[2019-05-29 Wed 14:47] =>  0:01
:END:
https://www.nature.com/scitable/topicpage/copy-number-variation-and-human-disease-741737/
https://www.frontiersin.org/articles/10.3389/fonc.2020.00321/full
- [[id:8c535c38-9310-4fd0-a8fe-c300639493da][ichorcna]]
- copy number classifiers
  - https://www.sciencedirect.com/science/article/pii/S0016508521002778?via%3Dihub
- https://www.sciencedirect.com/science/article/pii/S1535610818301119?via%3Dihub
- allele-specific copy number: the number of copies of each inherited allele, allows for the detection of copy-neutral loss of heterozygosity https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5627665/
- [[https://www.bioconductor.org/packages/devel/bioc/vignettes/TitanCNA/inst/doc/TitanCNA.pdf][see ploidy purity correction math]]
- [[https://www.biostars.org/p/383522/][good CN,purity,ploid discussion]]
***** cfDNA copy number calling
:PROPERTIES:
:ID:       21d876e8-8646-4773-91c1-c4385c3218c7
:END:
- by PCR
  - https://www.pnas.org/content/117/9/4858
- cite:feng2020 
- cite:ge2020
- cite:chen2019
- cite:hemming2019 
- cite:huang2019 
- cite:mansukhani2018
- cite:molparia2018
- cite:paoletti2018 
- cite:stover2018 
- cite:xu2018 
- cite:adalsteinsson2017
- cite:hovelson2017
- cite:molparia2017 
- https://bmccancer.biomedcentral.com/articles/10.1186/s12885-019-6003-8
- instability score
  - cite:li2017a  
**** [[id:5738AF1F-8621-40C3-AF80-0C396CDC889B][LINK TO Clinical NGS]]
:PROPERTIES:
:ID:       413414a5-651d-4d43-82c8-a6bb13de8ca3
:END:
*** Transcriptomics
:PROPERTIES:
:ID:       b9db0398-4481-4502-abb0-bd9edcd353f2
:END:
- 
- https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3016-x
**** Microarray
:PROPERTIES:
:ID:       294edcbb-f306-4f94-9eba-e8a1c7c512f1
:END:
- cite:lee2004
- cite:nguyen2002 
**** RNA-seq                                                         :career:
:PROPERTIES:
:ID:       4206be38-8d3c-43c4-a5c7-ca6a8bd8361a
:CREATED:  [2021-02-17 Wed 09:08]
:END:
:LOGBOOK:
CLOCK: [2021-02-17 Wed 08:34]--[2021-02-17 Wed 09:08] =>  0:34
CLOCK: [2021-01-22 Fri 06:42]--[2021-01-22 Fri 06:53] =>  0:11
CLOCK: [2020-12-28 Mon 07:46]--[2020-12-28 Mon 07:48] =>  0:02
CLOCK: [2020-11-30 Mon 12:40]--[2020-11-30 Mon 13:05] =>  0:25
CLOCK: [2020-10-08 Thu 10:15]--[2020-10-08 Thu 10:25] =>  0:10
CLOCK: [2020-10-08 Thu 09:17]--[2020-10-08 Thu 10:05] =>  0:48
CLOCK: [2020-06-22 Mon 12:16]--[2020-06-22 Mon 12:43] =>  0:27
CLOCK: [2016-05-09 Mon 15:01]--[2016-05-09 Mon 15:27] =>  0:26
CLOCK: [2020-06-22 Mon 11:26]--[2020-06-22 Mon 11:43] =>  0:17
CLOCK: [2016-04-25 Mon 15:07]--[2016-04-25 Mon 15:19] =>  0:12
CLOCK: [2016-04-25 Mon 14:26]--[2016-04-25 Mon 14:56] =>  0:30
CLOCK: [2016-05-31 Tue 09:31]--[2016-05-31 Tue 10:24] =>  0:53
CLOCK: [2017-07-08 Sat 12:26]--[2017-07-08 Sat 12:29] =>  0:03
CLOCK: [2017-02-24 Fri 13:04]--[2017-02-24 Fri 14:09] =>  1:05
CLOCK: [2017-02-24 Fri 12:13]--[2017-02-24 Fri 13:00] =>  0:47
:END:
https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/
cell of origin -- https://www.biorxiv.org/content/10.1101/2021.05.05.441859v1.full
cite:kukurba2015 
https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/lessons/data_visualization.html
https://www.bioconductor.org/packages/release/bioc/vignettes/DEGreport/inst/doc/DEGreport.html
- http://www.nathalievialaneix.eu/doc/pdf/tutorial-rnaseq.pdf
***** General guides, github repos, training, etc
:PROPERTIES:
:ID:       945cb660-b2c2-4416-be64-b24b9fb79a0f
:END:
- [[https://www.youtube.com/channel/UCQqQpIx3zQPaifBj67ocv1w][BabrahamBioinf youtube]]
- [[https://github.com/griffithlab/rnaseq_tutorial/blob/master/manuscript/supplementary_tables/supplementary_table_7.md][Griffith lab common RNASeq Q&A]]
- cite:griffith2015c
- https://pmbio.org/module-06-rnaseq/0006/02/01/RNAseq_Expression/
- cite:vandenberge2019
- cite:li2018rnastat
- https://rnabio.org/module-02-alignment/0002/02/01/Adapter_Trim/
- https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096346/
- [[https://www.bioconductor.org/packages/devel/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#differential-expression-analysis][easy as 1-2-3]]
- cite:picardi2015
  - intro
    - raw sequence metrics (d/n require alignment, ensure sequencing platform did it’s job)
    - read quality- Phred quality (Q) = -10 log10 (probability of erroneous base calling) reported as ASCII (or numbers for SOLiD)
      - no objective good/bad, but usually 20-30
      - variant requires greater quality than expression profiling
    - nucleotide composition
      - GC 3 vs AT 2 hydrogen bonds, exons have higher GC
      - first 12 bases (from 5’) will have large deviation from random hex priming
      - if there is GC bias, coverage uniformity and transcript abundance will be off
    - PCR duplication
      - chance of random duplication is 1/(4^(read length *2)), so still slim up to depth of 100s of millions
- cite:garve2012
- standards and best practices- cite:conesa2016
- cite:williams2014
- general resources- https://github.com/crazyhottommy/RNA-seq-analysis
- https://rnaseq.uoregon.edu/
- https://www.labome.com/method/RNA-seq.html
- http://mikelove.github.io/
- cite:dundar2018
- https://www.biostat.wisc.edu/bmi776/lectures/rnaseq.pdf
- https://github.com/crazyhottommy/RNA-seq-analysis
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096346/
- best practices
 - cite:conesa2016 
 - https://www.reddit.com/r/bioinformatics/comments/eahwak/an_update_on_rnaseq_differential_expression/
- cite:karow2017
- https://hub.docker.com/r/mgibio/rnaseq/dockerfile
- http://rseqc.sourceforge.net/
- cite:korpelainen2015
- cite:griffith2015
- https://www.rna-seqblog.com/blog/
- https://github.com/genome/gms/wiki/RnaSeq
- https://bioinformatics-core-shared-training.github.io/RNAseq_September_2018/slides/DAY_1.pdf
- standards and best practices
 - [[http://gatkforums.broadinstitute.org/gatk/discussion/4067/best-practices-for-variant-discovery-in-rnaseq][Broad best practices for variant discovery in RNA Seq]]
 - ENCODE RNA-Seq guidelines
 - [[http://www.fda.gov/ScienceResearch/BioinformaticsTools/MicroarrayQualityControlProject/][Sequencing Quality Control Project (SEQC)]]
   - [ ] [[http://www.ncbi.nlm.nih.gov/pubmed/?term=SEQC+sequencing+quality+control+project][pubmed: SEQC sequencing quality control project- 8 hits]]
     - cite:xu2016fda
       - SEQC Major Findings: 
         - Relative measurement is more consistent & reproducible than absolute
         - Comparison w/ microarry
           - 75% concordance for highly expressed, 35% for lowly expressed (w/ RNA-seq better at low via PCR)
         - Pipelines
           - Analyzed 12 pipelines, 278 permutations
           - Assessment is coming in Nature Methods
       - cite:Zheng_2015
       - cite:seqc2014comprehensive
- https://ycl6.gitbooks.io/rna-seq-data-analysis/rna-seq_analysis_workflow.html
- SEQC
 - cite:Li_2014
 - cite:seqc2014comprehensive
- genomeweb: [[file:~/Box%20Sync/career/Optimizing%20Whole-Transcriptome%20RNA-Seq%20with%20Improved%20Library%20Prep%20and%20Data%20Analysis-20170316%201700-1.arf][optimizing whole genome RNA seq]]
- [[https://blog.sbgenomics.com/rna-seq-the-first-wave/][RNA-seq: The first wave of papers]]
- cite:conesa2016
- cite:poplawski2015
- pipelines 
 - cite:kalari2014
 - [[https://hub.docker.com/r/mgibio/rnaseq/dockerfile][mgi rna-seq dockerfile]]
- cite:fonseca2014
- cite:wilhelm2009
****** Training
:PROPERTIES:
:ID:       10F920C2-3B28-4FC0-A91A-02F7229F2619
:END:

- - Garbe RNA-seq tutorial 1
- [[https://www.youtube.com/watch?v=hksQlJLwKqo][NYU Langone RNAseq workshop]]

******* Griffith AWS RNA-seq Tutorial
:PROPERTIES:
:ID:       520695E3-F98F-4195-8336-0A0E9589FB14
:END:
:LOGBOOK:
CLOCK: [2017-08-11 Fri 10:41]--[2017-08-11 Fri 11:17] =>  0:36
CLOCK: [2017-07-08 Sat 13:35]--[2017-07-08 Sat 14:20] =>  0:45
CLOCK: [2017-07-08 Sat 12:29]--[2017-07-08 Sat 12:33] =>  0:04
CLOCK: [2017-02-28 Tue 11:42]--[2017-02-28 Tue 11:44] =>  0:02
CLOCK: [2017-03-01 Wed 14:47]--[2017-03-01 Wed 14:55] =>  0:08
CLOCK: [2017-02-28 Tue 11:44]--[2017-02-28 Tue 11:54] =>  0:10
:END:
- [[https://us-west-2.console.aws.amazon.com/console/home?region=us-west-2#][my AWS]]
- [[https://bioinformatics.ca/workshops/2014/informatics-rna-sequence-analysis-qc-2014#material][videos on bioinformatics.ca]]
  - Module 1: Intro resume at 30 min 
    - RNA abundance 10^5 - 10^7
    - large genes have polyA selection bias
  - Module 2: RNA-seq alignment and visualization
  - Module 3: Expression and Differential Expression
  - Module 4: Isoform discovery and alternative expression 
- [ ] waiting on account confirmation
- [X] waiting to resolve AWS issue- https://console.aws.amazon.com/support/home?region=us-west-2#/case/?displayId=2093164191&language=en
- cite:Griffith_2015
  - Introduction to RNA Sequencing
    - accuracy = PCR see refs 1-2
    - workflow- isolation, enrichment, cDNA, fragmentation, sequencing, sequence preprocessing, alignment or assembly, adundance estimation, expression analysis, visualzation
    - downstream analysis
      - transcript discovery
      - genome annotation
      - gene regulation
      - expression analysis 
        - differential expression
        - allele-specific expression
      - RNA editing
      - viral detection
      - variant detection
        - gene fusion detection
    - clinical applications- redf 42, 43
    - limitations
      - single time-point steady state data
      - RNA expression may not correlate to protein expression- ref 46, 47
  - RNA Isolation, Library Preparation, and Sequencing Strategy
    - enrichment
      - total RNA
      - polyA enrichment
      - rRNA reduction
      - cDNA capture
    - size selection
    - use of linear amp
    - platform choice
    - reads- single vs. paired
  - Cloud Computing for RNA-Seq Analysis and Education- advantages
    - consistency
    - elasticity
    - multiple machine instances
  - RNA-Seq Data Formats, Qualty Contol, Trimming, Alignment, and Visualization
    - common file formats
      - FASTA
      - GTF
      - FASTQ
      - SAM/BAM
      - CIGAR
    - initial QC
      - k-mer-based 
        - adaptor contamination
        - rRNA contamination
        - short reads
      - base quality
      - ambiguous bases
      - read trimming
        - strategies
          - adaptor trimming
          - quality trimming
        - tools
          - skewer ref 57
          - trimmomatic ref 58
    - alignment or assembly
      - assembly- merging reads into contigs based on sequence similarity 
    - post-alignment QC
  - Expression and Differential Expression
    - biases affecting count based
      - gene or transcript size
      - CG bias
    - common analysis downstream of expression
      - classifying
        - Weka
        - RandomForests R package
      - pathway and gene set analysis
        - SeqGSEA
        - GAGE
        - PAthView
        - GoSeq
        - GSAASeqSP
        - Cytoscape
  - Isoform Discovery and Alternative Expression
    - dependent on correct inference of full-length transcripts from short reads
    - may be able to infer strand origin from splice site spanning reads
  - Challenges Specific to RNA-Seq
  - Conclusions and Future Work 
- [[http://www.rnaseq.wiki][Informatics for RNA-seq: A web resource for analysis on the cloud]] 
  - [[https://github.com/griffithlab/rnaseq_tutorial][github]]
  - Module 0 - Introduction and Cloud Computing
    - Authors
    - Citation and Supplementary Materials
    - Syntax
    - [[https://github.com/griffithlab/rnaseq_tutorial/wiki/Intro-to-AWS-Cloud-Computing][Intro to AWS Cloud Computing]]
      - Preamble
      - Acknowledgements
      - Glossary and abbreviations
      - What do I need to perform this tutorial
        - Creating an account
        - Logging into the AWS console
      - What is a Region?
      - How much does it cost to use AWS EC2 resources?
        - How does billing work?
      - Necessary steps for launching an instance
        - Step 1. Choosing an AMI
        - Step 2. Choosing an instance type
        - Step 3. Configuring instance details
        - Step 4. Adding storage
        - Storage volume options
        - Step 5. Tagging the instance
        - Step 6. Configuring a security group
        - Step 7. Reviewing the instance before launch
        - Step 8. Assigning a key pair
        - Step 9. Reviewing launch status
        - Step 10. Examining a new instance in the ec2 console
        - Step 11. Logging into an instance
      - Trouble-shooting and advanced topics
        - Can not login to EC2 instance - what might have gone wrong?
        - How do storage volumes appear within a linux instance on amazon EC2?
        - Taking stock of compute resources within an ubuntu linux instance
        - Basic setup and administration of an ubuntu linux instance
        - Setting up an Apache web server
        - What is difference between the start, stop, reboot and terminate instance states?
        - How do I create my own AMI, publish as a Community AMI, and what is a snapshot?
        - Tidying up and shutting down AWS resources
        - Further reading and preparing for more advanced AWS cloud computing concepts
    - Logging into Amazon Cloud
    - Unix Bootcamp
    - Environment
    - Resources
  - Module 1 - Introduction to RNA sequencing
    - Installation
    - Reference Genomes
    - Annotations
    - Indexing
    - RNA-seq Data
    - PreAlignment QC
  - Module 2 - RNA-seq Alignment and Visualization
    - Adapter Trim
    - Alignment
    - IGV
    - Alignment Visualization
    - Alignment QC
  - Module 3 - Expression and Differential Expression
    - Expression
    - Differential Expression
    - DE Visualization
    - Kallisto for Reference-Free Abundance Estimation
  - Module 4 - Isoform Discovery and Alternative Expression
    - Reference Guided Transcript Assembly
    - de novo Transcript Assembly
    - Transcript Assembly Merge
    - Differential Splicing
    - Splicing Visualization
  - Module 5 - De novo transcript reconstruction
    - De novo RNA-Seq Assembly and Analysis Using Trinity
  - Module 6 - Functional Annotation of Transcripts
    - Functional Annotation of Assembled Transcripts Using Trinotate
  - Appendix
    - Saving Results
    - Abbreviations
    - Lectures
    - Practical Exercise Solutions
    - Integrated Assignment
    - Proposed Improvements
    - AWS Setup
***** Experimental design
:PROPERTIES:
:ID:       19780d7c-afa7-4195-b1f9-df9c563de303
:END:
- https://www.degruyter.com/view/journals/sagmb/18/1/article-20180021.xml?language=en
- https://ucdavis-bioinformatics-training.github.io/2018-June-RNA-Seq-Workshop/tuesday/ExperimentalDesign.pdf
- http://scotty.genetics.utah.edu/help.html
- Depth
  - - determine directly using technical replicates
  - expression estimate correspondence among technical replicates (no biological variation) for genes with above-median coverage stabilized at about 10 million reads per sample (Wang et al. 2011).  
- Replicates and power
  - sufficient replication has to occur at every level of comparison. In a fully "factorial" design, for example, more than one experimental "factor" is of interest, each with two or more treatments, such that any individual receives one of multiple possible treatments at each factor. In this case, every possible combination of treatments across factors needs to be replicated sufficiently,
  - biological variation is more important than technical variation- See Bullard,10 Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments, fig. 4
  - adding depth vs. replicate- replicates preferred, gives less uncertainty 
  - remember, more comparisons require more replicates 
  - cite:schurch2016 
  - Power
    - https://cofactorgenomics.com/6-changes-thatll-make-big-difference-rna-seq-part-5/
    - cite:peixoto2015
    - the rewards in statistical power due to additional biological replicates will surpass the improved parameterization of technical variation garnered from additional technical replicates
    - low counts mean low power- see rna seqlopedia fig 1.1
  - [[http://scotty.genetics.utah.edu/help.html][Scotty]]
  - consider a "bank" of more sample replicates than are acutally sequenced 
- By specific experimental question
  - Differential expression
    - [[http://rnaseq.uoregon.edu/#exp-design][RNA-seqlopedia]]
      - qualitative or quantitative (DE, ect.) design- usually quant
      - need bio replicates
      - stranded and paired library prep not required
- Prefer “balanced block” design w/ barcode-> mixed prep-> all samples on all lanes
- cite:auer2010
- cite:williams2014
  - contains the 30-50 mil read for DE recommendation
- Examples of experimental objectives
  - cite:Cabanski_2014 RNA sequencing (RNA-Seq) approaches are designed to characterize the expressed genome in numerous ways1,2 from defining different types of RNA, such as long non- coding RNAs,3 to comparing RNA expression,4 splice iso- forms,5e7 allele-specific expression,8e10 fusions,11e14 RNA editing,15,16 
- sequencing platforms
  - cite:li2014
- cite:su2014
- cite:rajkumar2015
- http://www.slideshare.net/YaoyuEWang/rnaseq-analysis-ngsapplication2
- http://michelebusby.tumblr.com/post/26913184737/thinking-about-designing-rna-seq-experiments-to
- Parameters affecting sequencing
  - Variance is bimodal
    - low @ low counts b/c Poisson
    - low @ high counts b/c?
  - biological
    - transcript abundence 10^5-10^7 see cite:Griffith_2015 refs 85,86
  - technical @ platform
    - specialized
      - On target efficiency with bedtools 
    - read length
    - read depth
    - strandedness problem[[file:org.org_imgs/20170224_132238_567aRU.png]]
  - Strandedness
    - cite:Zhao_2015
- [[*testing, modeling, and read simulation][testing, modeling, and read simulation]]
***** [[id:7BF989EC-4DA7-4B0C-AAA9-0175156447FF][LINK TO RNA extraction, solubilization, and isolation]]
:PROPERTIES:
:ID:       b477ff77-85e8-4613-9bea-a703de899817
:END:
***** [[id:4E23127E-4ED6-4802-A03C-4103AA8CAF29][LINK TO RNA Quality and Quantification]]
:PROPERTIES:
:ID:       f8fb2eb4-e4dd-40f4-9ca6-f0582f4a3aac
:END:
***** Library preparation
:PROPERTIES:
:ID:       2111e30d-9183-40de-948e-3f76b70905d8
:END:
  - common kits
    - RNA exome 
    - pico
  - From input to library
    - cite:van_Dijk_2014
    - By workflow
      - Isolation
      - Enrichment
        - Methods
          - Hybridization
            - For selection of targets
              - via oligo-dT
                - will include some mitochondial 
            - For removal of non-targets
              - rRNA depletion- rely on specific oligo set, d/n work well if rRNA is degraded 
          - Size-selection enrichment- for ncRNA
          - DSN (duplex-specific nuclease digestion)- @ cDNA level, allow abundant transcripts to renature and are degraded by duplex-specific nuclease 
      - Fragmentation- RNase III, metal ion, heat, or sonication
      - Library preparation
        - FFPE
          - However, formalin treatment degrades RNA through cross-linking and fragmentation, which significantly impairs molecular analyses (Bass et al., 2014; Farragher et al., 2008; Klopfleisch et al., 2011). T
          - check webster 15 cites- (Auerbach et al., 2014; Hedegaard et al., 2014; Linton et al., 2012; Spencer et al., 2013; Zhao et al., 2014b). 
        - [[http://support.illumina.com/content/dam/illumina-support/courses/truseq-rna-access/story.html?iframe?iframe][TruSeq: RNA Access Library Prep]]
        - [[http://support.illumina.com/content/dam/illumina-support/courses/TruSeq_Sample_Prep_Best_Practices/index.html?iframe?iframe][TruSeq Sample Preparation Best Practices]]
        - First-strand synthesis
          - Priming options
            - oligo-dT
            - "random" hex
        - Second-strand synthesis
          - Also requires a primer
          - Usually via RNA displacement- RNase H and T4 ligase 
        - cDNA fragmentation
          - accounting for fragment bias- https://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-3-r22
        - Sequencing adaptors
          - Y-adaptor PCR: blunt cDNA ends, add adenosine overhangs
- cite:oneil2013
- stranded
  - https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work  
***** [[id:3BC41E6A-00D3-4498-A48C-463F68CFD76B][LINK TO massively parallel sequencing]]
:PROPERTIES:
:ID:       68056b25-1413-4bda-a1f2-6610e92c6765
:END:
***** [[id:1EE731A3-83DD-46AF-B988-2311307C335F][LINK TO raw sequence quality control]]
:PROPERTIES:
:ID:       7cc53256-8b7b-4c93-aaa5-2a35e41b7319
:END:
***** RNA-seq quality control
:PROPERTIES:
:ID:       e6cbde1b-22c3-4a23-b99e-d9d4fe0a6b95
:END:
:PROPERTIES:
:ID:       2d2f32cf-ef4a-4efe-90d1-ae85ad470e49
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110
:END:
- https://www.reddit.com/r/bioinformatics/comments/flqb8e/how_do_i_differentiate_outliers_from_ingroup/
- https://www.rna-seqblog.com/an-rna-seq-qc-overview/
- https://www.biostars.org/p/368420/
- https://bioinformatics.stackexchange.com/questions/3572/strange-per-sequence-gc-content-results
- cite:wang2012


- samtools sort (to postion) 

******* [[http://www-huber.embl.de/HTSeq/doc/count.html#count][website]]
:PROPERTIES:
:ID:       6B7F3881-DA7D-4700-BC30-9EF83150D916
:END:

htseq-count [options] <alignment_file> <gff_file>

simple code: htseq-count -m union -r pos -i gene_name -a 10 --stranded=no agcgccg.bam Mus_musculus.GRCm38.84.gtf > output.counts

Important options
--stranded=no (defualt is yes)
--format bam (defualt is sam)
  - for paired end, order is important
    - do samtools sort (default) to put in position order
    - in HTSeq-counts, specify -r pos
    -a 10 (minimum quality, 10 is default) 
    -m default is union, alternate intersection-strict and intersection-nonempty 

    Why union is default: consider two genes, which share a stretch of common sequence such that for a read mapping to this stretch, the aligner cannot decide which of the two genes the read originated from and hence reports a multiple alignment. If we discard all such reads, we undercount the total output of the genes, but the ratio of expression strength (the “fold change”) between samples or experimental condition will still be correct, because we discard the same fratcion of reads in all samples. On the other hand, if we counted these reads for both genes, a subsequent diffential-expression analysis might find false positives:


  followed by the special counters, which count reads that were not counted for any feature for various reasons

  [[http://www-huber.embl.de/HTSeq/doc/counting.html][algorithm details]]

****** Basic counting script
:PROPERTIES:
:ID:       AE616985-EA77-4CF0-8FEA-60FF56A04AC3
:END:

counting is computationally intensive- script HTseq and
summarizeOverlaps on the cluster 
variabilize RNA seq workflow- make into function 
***** [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][LINK TO Nucleic acid sequence alignment]]
:PROPERTIES:
:ID:       542988fa-307f-454e-927e-ee6a17187265
:END:
***** Read annotation
:PROPERTIES:
:ID:       8e435b88-6895-4e75-a728-cfdaa272bddb
:END:
https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1308-8
- https://github.com/griffithlab/rnaseq_tutorial/wiki/Annotation
***** Quantification and normalization
:PROPERTIES:
:ID:       853dd34c-27be-4be8-9794-19f04a1e22b0
:END:
https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
by spike-in  Jiang,L., Schlesinger,F., Davis,C.A., Zhang,Y., Li,R., Salit,M., Gingeras,T.R. and Oliver,B. (2011) Synthetic spike-in standards for RNA-seq experiments. Genome Res., 21, 1543–1551.

- choosing a strategy- Zyprych-Walczak,J., Szabelska,A., Handschuh,L., Górczak,K., Klamecka,K., Figlerowicz,M. and Siatkowski,I. (2015) The impactof normalization methods on RNA-Seq data analysis. Biomed Res. Int., 2015, doi:10.1155/2015/621690.
- what differences are controlled?
  - library size
  - batch effect
  - gene length
  - GC content
- methods
  - upper quartile- remove zero counts, scale based on 75th quartile of remaining reads Bullard,J.H., Purdom,E., Hansen,K.D. and Dudoit,S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics, 11, doi:10.1186/1471-2105-11-94.
  - quantile- read distribution assumed identical, counts adjusted by median of each quantile across all samples
  - TMM
  - pseudoalignment
    - kallisto, used with sleuth for DE
      - https://www.nature.com/articles/nbt.3519
      - https://pachterlab.github.io/sleuth/about
  - by spike-in  Jiang,L., Schlesinger,F., Davis,C.A., Zhang,Y., Li,R., Salit,M., Gingeras,T.R. and Oliver,B. (2011) Synthetic spike-in standards for RNA-seq experiments. Genome Res., 21, 1543–1551.
  - count-based
    - analysis only at the gene level
      - cite:pachter2015
    - edgeR 
  - TPM
  - FPKM
  - https://europepmc.org/article/PMC/6058388
  - abundance 
    - http://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#introduction
  - cite:pachter2015 
- Bullard,J.H., Purdom,E., Hansen,K.D. and Dudoit,S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics, 11, doi:10.1186/1471-2105-11-94.
- cite:peixoto2015
- https://www.the-scientist.com/news-opinion/technical-bias-widespread-in-rna-seq-datasets-66766?utm_campaign=TS_DAILY%20NEWSLETTER_2019&utm_source=hs_email&utm_medium=email&utm_content=79783848&_hsenc=p2ANqtz-_TvsXNNQSHgNWR9yE8Q6vZP04TRSx3T-W0rLk0NLucSkrSI0wQ0GDQ2L1Vs_Jf1x1_21ou8yVG1ur9nXtXbBYl91OsUg&_hsmi=79783848
- normalization to control for systemic error / bias
- shrinkage: pooling information across genes to obtain reliable variance estimates for small sample sizes, cite:pachter2015 
- feature counting
  - HTSeq
  - featureCounts 
- https://www.slideshare.net/mobile/jakonix/part-1-of-rnaseq-for-de-defining-the-goal
- https://www.biostars.org/p/404211/#:~:text=What%20is%20the%20difference%20between,as%20fragment%20length%20distribution%20etc.
- https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
- [[*rsem][rsem]]
- https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/
- https://github.com/griffithlab/rnaseq_tutorial/wiki/Transcript-Assembly-Merge
- https://www.reddit.com/r/bioinformatics/comments/i6qhrx/samples_from_different_studies_de_analysis_help/
- https://www.nature.com/articles/ng1032z.pdf
- evaluation
  - linearity
    - https://link.springer.com/article/10.1186/s12859-017-1526-y
  - https://bioinformatics.stackexchange.com/questions/92/confirm-success-or-failure-of-rna-seq-normalization
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/
- https://groups.google.com/forum/m/#!topic/rsem-users/GRyJfEOK1BQ
- https://www.biostars.org/p/68126/
- https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25
- https://cgatoxford.wordpress.com/2016/08/17/why-you-should-stop-using-featurecounts-htseq-or-cufflinks2-and-start-using-kallisto-salmon-or-sailfish/
- isoform abundance
  - https://cgatoxford.wordpress.com/2016/08/17/why-you-should-stop-using-featurecounts-htseq-or-cufflinks2-and-start-using-kallisto-salmon-or-sailfish/
- https://www.biostars.org/p/329625/
- sources of bias
  - 
- https://htseq.readthedocs.io/en/release_0.11.1/tour.html#counting-reads-by-genes
- RSEM
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323
****** featureCounts in Rsubread
:PROPERTIES:
:ID:       0FB3EE5E-DD17-4438-BF35-011555A41DF9
:END:

[[http://www.rdocumentation.org/packages/Rsubread/functions/featureCounts][nice featureCounts manual]]

****** [#Y] summarizeOverlaps in GenomicAlignments
:PROPERTIES:
:ID:       16408005-913F-4675-BA61-B7594A3FB4EC
:END:

****** HTseq-count
:PROPERTIES:
:ID:       E2262D77-424F-4943-8DA4-5649B2F0B4B2
:END:

[[file:~/Box%20Sync/comp_bio/Anders,15%20HTSeq%20a%20python%20framework%20to%20work%20with%20high%20throughput%20sequencing%20data.pdf][Anders,15 HTSeq a python framework to work with high throughput
sequencing data]]

Not on cluster 
- create R script to run stats on generic output file
- % no feature, ambiguous, too low qual, not aligned, alignment not
  unique 

******* Prerequisites
:PROPERTIES:
:ID:       D5102869-0833-4513-A716-69D44892FDDF
:END:

- samtools sort (to postion) 

******* [[http://www-huber.embl.de/HTSeq/doc/count.html#count][website]]
:PROPERTIES:
:ID:       6B7F3881-DA7D-4700-BC30-9EF83150D916
:END:

htseq-count [options] <alignment_file> <gff_file>

simple code: htseq-count -m union -r pos -i gene_name -a 10 --stranded=no agcgccg.bam Mus_musculus.GRCm38.84.gtf > output.counts

Important options
--stranded=no (defualt is yes)
--format bam (defualt is sam)
  - for paired end, order is important
    - do samtools sort (default) to put in position order
    - in HTSeq-counts, specify -r pos
    -a 10 (minimum quality, 10 is default) 
    -m default is union, alternate intersection-strict and intersection-nonempty 

    Why union is default: consider two genes, which share a stretch of common sequence such that for a read mapping to this stretch, the aligner cannot decide which of the two genes the read originated from and hence reports a multiple alignment. If we discard all such reads, we undercount the total output of the genes, but the ratio of expression strength (the “fold change”) between samples or experimental condition will still be correct, because we discard the same fratcion of reads in all samples. On the other hand, if we counted these reads for both genes, a subsequent diffential-expression analysis might find false positives:


  followed by the special counters, which count reads that were not counted for any feature for various reasons

  [[http://www-huber.embl.de/HTSeq/doc/counting.html][algorithm details]]


****** Expression Level Abundance, Normalization Units
:PROPERTIES:
:ID:       D4BFCCCA-8BBC-4D23-B71B-F8C2ED6287E3
:END:
https://ro-che.info/articles/2016-11-28-rna-seq-normalization
R/FPKM and C/TPM are "local" to the sample while TMM must be recalculated with different sample set

******* Raw- required input for DE analyses
:PROPERTIES:
:ID:       82A81565-9333-4E40-806E-474399BD73C4
:END:

******* R/FPKM
:PROPERTIES:
:ID:       9D6324B7-D9A2-44ED-A54F-B0C0C8DCECE2
:END:

- normalization for library size (i.e. sequencing depth) and transcript length
- use to
  - compare a gene across samples or
  - compare different genes within a sample
- see Mortazavi A., Williams B.A., McCue K., Schaeffer L., and Wold B. Mapping and quantifying mammalian transcriptomes by RNA-seq. Nat Methods 5(7):621–628, 2008.
- RPKM (reads per thousand nucleotides in transcript per million reads)
  - “For example, if a 2 kb transcript has 1000 reads and the total number of reads is 25 million, then RPKM = (1000/2)/25 = 20”
- FPKM (fragments per thousand nucleotides per million mapped reads)- equivalent of RPKM for paired-end (two reads per fragment)

******* CPM (counts per million) / TPM
:PROPERTIES:
:ID:       8C3481E5-B350-4D9D-A44C-DAB17FAEE1C6
:END:

  -   - see Wagner G.P., Kin K., and Lynch V.J. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. Theory Biosci 131(4):281–285, 2012.=
  - limma voom uses a logCPM 
  - corrects for transcript length distribution of different sample pools  
  - TPM (transcripts per million)- divides by sum of transcript-length-normalized reads
    i.e. normalizes to transcripts as opposed to reads
    useful if average transcript length differs b/w samples 

******* TMM (trimmed mean of M values)
:PROPERTIES:
:ID:       544B0EAB-4594-404B-9C83-B951E2244CFE
:END:

-attempts to normalize this problem:
[[file:work.org_imgs/20160419_132550_469_bk.png]]
- corrects for less "sequencing real estate" due to high expression per tissue 

***** Exploratory data analysis
:PROPERTIES:
:ID:       77d2a559-9858-453a-83c5-a646bbf473a3
:END:
Bergkvist,A., Rusnakova,V., Sindelka,R., Garda,J.M., Sjogreen,B.,
Lindh,D., Forootan,A. and Kubista,M. (2010) Gene expression
profiling–clusters of possibilities. Methods, 50, 323–335.
228. Beyene,J., Tritchler,D., Bull,S.B., Cartier,K.C., Jonasdottir,G.,
Kraja,A.T., Li,N., Nock,N.L., Parkhomenko,E., Rao,J.S. et al.
(2007) Multivariate analysis of complex gene expression and clinical
phenotypes with genetic marker data. Genet. Epidemiol., 31(Suppl.
1), S103–S109.

- cite:buschmann2016 pg 6011
- hierarchical clustering
  - eucledian
  - pearson 
- partitioning
  - k-means
  - self-organizing maps
- latent variable
  - PCA
    - add scores and loadings for initial variable
- [[id:588f6536-9db9-4279-9e67-800890c2de23][LINK TO High dimension data transformation]]
****** Clustering
:PROPERTIES:
:ID:       22a6e4a4-60c2-44cf-8e5a-3ff351a02aa3
:END:
https://uc-r.github.io/hc_clustering
https://www.biostars.org/p/287512/
- https://www.datanovia.com/en/blog/types-of-clustering-methods-overview-and-quick-start-r-code/

- cite:vidman2019
***** Differential expression
:PROPERTIES:
:ID:       ee030cef-74d4-435d-8228-b85f0e6203db
:END:
:LOGBOOK:
CLOCK: [2021-01-26 Tue 11:07]--[2021-01-26 Tue 11:53] =>  0:46
CLOCK: [2020-11-23 Mon 10:43]--[2020-11-23 Mon 10:57] =>  0:14
CLOCK: [2016-03-28 Mon 19:06]--[2016-03-28 Mon 19:29] =>  0:23
CLOCK: [2016-03-28 Mon 13:01]--[2016-03-28 Mon 13:06] =>  0:05
:END:
https://github.com/hbctraining/DGE_workshop_salmon_online
https://f1000research.com/articles/5-1438/v2
https://www.nature.com/articles/s41598-020-74567-y#Abs1
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6954399/
****** experimental design
:PROPERTIES:
:ID:       4b9eee0d-4dd2-4b31-a5c2-873663443671
:END:
  - cite:mccarthy2012
  - https://angus.readthedocs.io/en/2016/_static/DifferentialExpressionBasics_NGS2016_ID.pdf
****** specialized DE datasets
:PROPERTIES:
:ID:       722cd834-2504-4ae0-84ef-58554fa4da32
:END:
- long non-coding- https://europepmc.org/article/PMC/6058388
- transcript / isoform DE
  - https://davetang.org/muse/2017/10/25/getting-started-hisat-stringtie-ballgown/
  - sleuth https://www.nature.com/articles/nmeth.4324 cite:pachter2015
  - CuffDiff- isoform count-based
****** tutorials
:PROPERTIES:
:ID:       9bd06c29-45ba-42b8-b048-27b83a0fa47d
:END:
- http://master.bioconductor.org/packages/release/workflows/html/rnaseqGene.html
- https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs
- https://bioconductor.org/packages/devel/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#useful-graphical-representations-of-differential-expression-results
- [[https://github.com/griffithlab/rnaseq_tutorial/wiki/Citation][Griffith RNA-seq tutorial]]
****** statistical models
:PROPERTIES:
:ID:       8a788e6c-3899-4e6f-bf09-9c96d110aeaa
:END:

- overdispersion makes working under Poisson assumption prone to high false positives b/c sampling error underestimated


Di,Y., Schafer,D.W., Cumbie,J.S. and Chang,J.H. (2011) The NBP
negative binomial model for assessing differential gene expression
from RNA-Seq. Stat. Appl. Genet. Mol. Biol., 10,
doi:10.2202/1544-6115.1637.



  - cite:huang2015      
  - https://angus.readthedocs.io/en/2016/_static/DifferentialExpressionBasics_NGS2016_ID.pdf
  - https://bioinformatics-core-shared-training.github.io/RNAseq-R/slides/LinearModels.pdf
  - dispersion
    - https://support.bioconductor.org/p/75260/
  - no t-test
    - cite:jeanmougin2010 
    - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0012336
  - RNA expression varies about 10^5
  - RNA counts follow “overdispersed” poisson ( (large number of events, each event is rare) / negative binomial distribution (additional factor is dispersion)
    expression levels of same gene across tissues follow log-normal distribution 
    - base distribution is Poisson (b/c large number of events, each event is rare), variance = mean
    - shot noise- noise from sampling
    - zero inflation- many zero counts 
    - negative binomial (overdispersed Poisson), variance = mean + (1/r)*u^2, used by
      - DESeq2 (can also do continuous values like limma)
      - EdgeR
****** Methods and tools
:PROPERTIES:
:ID:       68BB9D3D-5879-4F2E-904B-54F70EAD2B5A
:END:
- count-based
  - negative binomial
    - [[id:32BB77F2-0283-4A52-86AC-CDF2CB6A7315][edgeR]], dispersion by weighted likelihood
    - [[id:2F642F69-A9C8-46DA-BE31-783AC8D4F4B6][DESeq2]], dispersion by modeling all genes via regression
- [[id:F9684168-A438-4F5E-81D6-D645B52B4074][limma voom]], voom transforms read count 
- Bayesian
  - BaySeq
  - BitSeq
  - ebSeq
- Non-parametrics- require 5-10 replicates per group
  - SAMSeq
  - NOISeq
- methods comparisons
  - cite:huang2015    
  - cite:seyednasrollah2015    
  - cite:tang2015 
  - cite:rapaport2013
  - https://www.nature.com/articles/s41598-020-59516-z#data-availability
  - comparison
    - https://academic.oup.com/bib/article/16/1/59/240754#26957696
- de novo
  - https://informatics.fas.harvard.edu/best-practices-for-de-novo-transcriptome-assembly-with-trinity.html
****** outputs
:PROPERTIES:
:ID:       48db2c9c-5d17-4438-aa40-43816245c490
:END:
  - log fold change
    - https://www.biostars.org/p/101727/


****** Notes
:PROPERTIES:
:ID:       558011C7-FB50-435E-B130-E3485CC25B90
:END:
- cite:anders2010
- https://pubmed.ncbi.nlm.nih.gov/22931062/
- bench methods
  - cite:parekh2016
***** [[*workflow][LINK TO workflow]]
:PROPERTIES:
:ID:       af099dea-affd-4fdf-8745-aaa9b22bc505
:END:
- http://bioinformatics.sdstate.edu/idep/
***** RNA-seq visualization
:PROPERTIES:
:ID:       b1351d51-3ac1-4193-b944-850c44a903fd
:END:
- [[id:DB8463CB-97CB-49E6-8EDD-A5ECA2F140BF][visualization tools]]
- examples
  - [[https://www.cell.com/cell-reports/pdf/S2211-1247(20)30669-0.pdf][good seq analysis and visual example]]
- volcano
  - https://www.biostars.org/p/203876/
- heatmaps
  - http://compbio.ucsd.edu/making-heat-maps-r/
  - https://www.rdocumentation.org/packages/gplots/versions/3.0.3/topics/heatmap.2 
- mapped reads were visualized by SeqMonk[[file:org.org_imgs/20161220_180037_4849srR.png]]

***** Methods by tissue source or isolation 
:PROPERTIES:
:ID:       4d4a89d1-e039-4a89-b968-81c80f38d884
:END:
- ffpe
  - cite:kashofer2013 
  - FFPE Pre-analytic variability
    - cite:ahlfen2007rnaqual
      - storage conditions
      - cold ischemia time
      - fixation
        - type
        - time
      - sample thickness
    - biospecimens.cancer.gov/researchnetwork 
    - cite:unlockingQiagen
    - https://www.ncbi.nlm.nih.gov/pubmed/25357115
    - By effect
      - Yield
        - - prolonged fix reduces yield- Bresters D, Schipper ME, Reesink HW, Boeser-Nunnink BD, Cuypers HT. The duration of fixation influences the yield of HCV cDNA-PCR products from formalin-fixed, paraffin- embedded liver tissue. J Virol Methods 1994;48:267–72.
        - cite:webster2015mining longer fixation times decrease RNA yields from paraffin-embedded tissues (herein 3 weeks in formalin resulted in half the RNA yield/sec- tion compared with 18 h) and increase RNA input needed during library preparation (see methods). This latter effect is likely due to ‘non-functional’ RNA (eg, adducts or damaged bases) that could be potentially mitigated by treatments like PreCR or other enzymatic and chemical methods. 
      - At microarray expression
        - cite:kashofer2013quality
    - "Formalin fixation introduced major changes into microarray gene expression data and led to marked gene-to-gene variations in delta-ct values of qRT-PCR. We found that qRT-PCR efficiency and gene-to- gene variations were mainly attributed to differences in the efficiency of cDNA synthesis as the most sensitive step. These differences could not be reliably detected by quality assessment of total RNA isolated from formalin-fixed tissues by electrophoresis or spectrophotometry. "
    - At sequencing
  - cite:graw2015 
  - Past expression tech
    - goals were biomarker discovery and patient stratification
    - cite:Zhao_2014 manifests its superiority in terms of the dynamic range, sampling depth, and has independence from pre-existing sequence information [1,2].
    - cite:webster2015mining When compared with conventional DNA microarrays, RNA-seq enriches for many additional fragments as it is not restricted to predefined probes and has (in principle) no limitations to dynamic range (Li et al., 2014b; SEQC/MAQC-III Consortium, 2014). 
    - - https://www.ncbi.nlm.nih.gov/pubmed/14695316
    - - [[https://www.ncbi.nlm.nih.gov/pubmed/15509548][DASL]]
    - https://www.ncbi.nlm.nih.gov/pubmed/21347257
  - FFPE microarray and RT-PCR was useful
    - AlloMap is a non-invasive gene expression-based blood test that is used to manage the clinical care of heart transplant recipients, providing a quantified score for the risk of rejection based on the measurement of expression of 20 genes, a subset of which are related to immune system activation and signalling15,16.
    - - cite:penland2007rna
    - - - https://www.ncbi.nlm.nih.gov/pubmed/11568892/
    - Mittempergher, L., de Ronde, J.J., Nieuwland, M., Kerkhoven, R.M., Simon, I., Th Rutgers, E.J., Wessels, L.F., Van't Veer, L.J., 2011. Gene expression profiles from formalin fixed paraffin embedded breast cancer tissue are largely comparable to fresh fro- zen matched tissue. PLoS One 6, e17163.
  - But this expression tech is inferior
    - In other work14, systematic evalu- ation of RNA-seq-based and microarray-based classifiers found that RNA-seq outperformed arrays in characteriz- ing the transcriptome of cancer and performed similarly to arrays in clinical endpoint prediction.
    - does not depend on genome annotation, or on predesigned species-specific or tran- script-specific probes, for transcript measurement.
    - microarray limited application to clinic- lthough powerful, microarray-based assays can have limitations in some environments, such as those related to laboratory-to-laboratory variation in sample preparation that can affect reproducibility. Moreover, for some applications, microarray signal-to-noise ratios can affect the limit of detection. Interestingly, a number of additional cancer multi-gene profile tests are clinically available, such as OncoTypeDX (Genome Health)135 for breast cancer recurrence risk and Prolaris (Myriad)136 for prostate cancer aggressiveness. These tests are based on qRT-PCR technologies, rather than microarrays, largely owing to the belief that qRT-PCR is more reliable, reproducible, sensitive and accurate.
    - " qRT-PCR is generally deemed the ‘gold standard’ method for measuring transcript levels, particularly in the clinical space; however, there are a number of limitations, including the fact that although it is a high sample throughput technology, relatively few markers or measurements can be made in a single assay. "
    - rnaseq superior for
      - snvs
      - indels
      - isoforms
      - splice vars
      - chimerics
        - fusions
    - semiquan
    - no SNV discovery
    - no fusion
    - "RNA-seq provides much more detailed information on specific transcript expression patterns in clinico-genetic neuroblastoma subgroups than microarrays. " cite:zhang2015comparison
- extracellular
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5426982/
  - [[id:51e5d70c-4a62-4960-880e-56f050194a0a][LINK TO circulating rna]]
***** Integrated miRNA-mRNA analysis
:PROPERTIES:
:ID:       82aafb6f-63a8-4ab7-867a-9add0e0c224a
:END:
- [[id:903f32d8-bd13-496c-8d5c-26a63455936a][LINK TO miRNA-seq]]
- example papers
  - https://bmccancer.biomedcentral.com/articles/10.1186/s12885-019-5983-8#Sec2
  - https://www.oncotarget.com/article/8713/text/
  - cite:li2017mirna
***** Methods by RNA population
:PROPERTIES:
:ID:       33823415-f2e4-4cdf-b3aa-a01000b35249
:END:
****** Bulk RNA
:PROPERTIES:
:ID:       933724d3-ef49-46f9-9b60-63372c236a98
:END:
******* Applications for bulk RNA-seq
:PROPERTIES:
:ID:       a9e9bbf7-8012-456a-9b45-0c86aacd84c7
:END:
- [[*cell enumeration through bulk RNA-seq][cell enumeration through bulk RNA-seq]]
- deconvolution
  - http://labs.pathology.jhu.edu/halushka/science/
  - tools
    - ISOpure
      - https://link.springer.com/article/10.1186/s12859-015-0597-x#Sec1
  - cite:newman2015 
- Clinical applications of RNA-seq
  - cite:byron2016translating
  - cite:Kalari_2014
  - cite:Van_Keuren_Jensen_2014
******** QTL mapping
:PROPERTIES:
:ID:       2CA85A6B-7B1E-4A38-96AA-717847A59168
:END:

********* Chemical- NextBio, see Webster,15ref Kupershmidt,10
:PROPERTIES:
:ID:       f9f486c9-cd9b-401a-a3c0-523a5d09d92b
:END:
:PROPERTIES:
:ID:       DD97187A-47FF-49D2-87D3-9DDFC9E1D351
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110
:END:
  - extracellular
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5426982/
  - stranded
    - https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work
  - ffpe
    - cite:kashofer2013 
    - FFPE Pre-analytic variability
      - cite:ahlfen2007rnaqual
        - storage conditions
        - cold ischemia time
        - fixation
          - type
          - time
        - sample thickness
      - biospecimens.cancer.gov/researchnetwork 
      - cite:unlockingQiagen
      - https://www.ncbi.nlm.nih.gov/pubmed/25357115
      - By effect
        - Yield
          - - prolonged fix reduces yield- Bresters D, Schipper ME, Reesink HW, Boeser-Nunnink BD, Cuypers HT. The duration of fixation influences the yield of HCV cDNA-PCR products from formalin-fixed, paraffin- embedded liver tissue. J Virol Methods 1994;48:267–72.
          - cite:webster2015mining longer fixation times decrease RNA yields from paraffin-embedded tissues (herein 3 weeks in formalin resulted in half the RNA yield/sec- tion compared with 18 h) and increase RNA input needed during library preparation (see methods). This latter effect is likely due to ‘non-functional’ RNA (eg, adducts or damaged bases) that could be potentially mitigated by treatments like PreCR or other enzymatic and chemical methods. 
        - At microarray expression
          - cite:kashofer2013quality
      - "Formalin fixation introduced major changes into microarray gene expression data and led to marked gene-to-gene variations in delta-ct values of qRT-PCR. We found that qRT-PCR efficiency and gene-to- gene variations were mainly attributed to differences in the efficiency of cDNA synthesis as the most sensitive step. These differences could not be reliably detected by quality assessment of total RNA isolated from formalin-fixed tissues by electrophoresis or spectrophotometry. "
      - At sequencing
    - cite:graw2015 
    - Past expression tech
      - goals were biomarker discovery and patient stratification
      - cite:Zhao_2014 manifests its superiority in terms of the dynamic range, sampling depth, and has independence from pre-existing sequence information [1,2].
      - cite:webster2015mining When compared with conventional DNA microarrays, RNA-seq enriches for many additional fragments as it is not restricted to predefined probes and has (in principle) no limitations to dynamic range (Li et al., 2014b; SEQC/MAQC-III Consortium, 2014). 
      - - https://www.ncbi.nlm.nih.gov/pubmed/14695316
      - - [[https://www.ncbi.nlm.nih.gov/pubmed/15509548][DASL]]
      - https://www.ncbi.nlm.nih.gov/pubmed/21347257
    - FFPE microarray and RT-PCR was useful
      - AlloMap is a non-invasive gene expression-based blood test that is used to manage the clinical care of heart transplant recipients, providing a quantified score for the risk of rejection based on the measurement of expression of 20 genes, a subset of which are related to immune system activation and signalling15,16.
      - - cite:penland2007rna
      - - - https://www.ncbi.nlm.nih.gov/pubmed/11568892/
      - Mittempergher, L., de Ronde, J.J., Nieuwland, M., Kerkhoven, R.M., Simon, I., Th Rutgers, E.J., Wessels, L.F., Van't Veer, L.J., 2011. Gene expression profiles from formalin fixed paraffin embedded breast cancer tissue are largely comparable to fresh fro- zen matched tissue. PLoS One 6, e17163.
    - But this expression tech is inferior
      - In other work14, systematic evalu- ation of RNA-seq-based and microarray-based classifiers found that RNA-seq outperformed arrays in characteriz- ing the transcriptome of cancer and performed similarly to arrays in clinical endpoint prediction.
      - does not depend on genome annotation, or on predesigned species-specific or tran- script-specific probes, for transcript measurement.
      - microarray limited application to clinic- lthough powerful, microarray-based assays can have limitations in some environments, such as those related to laboratory-to-laboratory variation in sample preparation that can affect reproducibility. Moreover, for some applications, microarray signal-to-noise ratios can affect the limit of detection. Interestingly, a number of additional cancer multi-gene profile tests are clinically available, such as OncoTypeDX (Genome Health)135 for breast cancer recurrence risk and Prolaris (Myriad)136 for prostate cancer aggressiveness. These tests are based on qRT-PCR technologies, rather than microarrays, largely owing to the belief that qRT-PCR is more reliable, reproducible, sensitive and accurate.
      - " qRT-PCR is generally deemed the ‘gold standard’ method for measuring transcript levels, particularly in the clinical space; however, there are a number of limitations, including the fact that although it is a high sample throughput technology, relatively few markers or measurements can be made in a single assay. "
      - rnaseq superior for
        - snvs
        - indels
        - isoforms
        - splice vars
        - chimerics
          - fusions
      - semiquan
      - no SNV discovery
      - no fusion
      - "RNA-seq provides much more detailed information on specific transcript expression patterns in clinico-genetic neuroblastoma subgroups than microarrays. " cite:zhang2015comparison
  - small RNA sequencing
    - sample processing
      - circulating miRNA stability
        - serum cite:gautam2016
        - plasma  cite:glinge2017
    - library preparation
      - with low input, use same kit for reproducibility cite:yeri2018 
      - cite:dard-dascot2018
    - https://rnajournal.cshlp.org/content/18/6/1279.long
    - qc
      - https://www.agilent.com/en/product/automated-electrophoresis/bioanalyzer-systems/bioanalyzer-rna-kits-reagents/bioanalyzer-small-rna-analysis-228257#howitworks
  - single cell
    - https://www.biorxiv.org/content/biorxiv/early/2019/02/05/541433.full.pdf
    - examples
      - - see fig 3 & methods https://genome.cshlp.org/content/27/2/208.full.pdf
    - https://www.nature.com/articles/s41587-020-0553-9
    - 10x
      - applications - https://pages.10xgenomics.com/nur-gen-research-area-literature.html?src=email&utm_medium=email&lss=nurture&utm_source=nurture&cnm=es-nurture-ra_g-general&utm_campaign=es-nurture-ra_g-general&utm_content=&utm_term=&useroffertype=marketing-literature&userresearcharea=ra_g&userregion=&userrecipient=customer&mktouserid=1265915&cid=&usercampaignid=
    - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0212031
    - http://www.nature.com/nbt/journal/v34/n2/full/nbt.3445.html
  - Clinical applications of RNA-seq
    - cite:byron2016translating
    - cite:Kalari_2014
    - cite:Van_Keuren_Jensen_2014
  - deconvolution
    - tools
      - ISOpure
        - https://link.springer.com/article/10.1186/s12859-015-0597-x#Sec1
    - cite:newman2015 
  - time course
    - cite:spies2017
    - https://support.bioconductor.org/p/65676/#66860
    - https://www.biostars.org/p/138988/
    - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#group-specific-condition-effects-individuals-nested-within-groups
    - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
    - http://seqanswers.com/forums/showthread.php?t=64039
    - https://www.sciencedirect.com/science/article/pii/S2001037015000392
- [[*cell enumeration through bulk RNA-seq][cell enumeration through bulk RNA-seq]]
- https://www.rna-seqblog.com/optimization-of-mirna-seq-data-preprocessing/
- [[id:e4351ef3-6d41-4121-a2cc-c63d89f997ac][pathway analysis]]
- [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- [[id:57E81A44-3233-479B-8264-2CD5F1904A60][Sequence variation and variant calling]]


- [[*omics integration][omics integration]]
********* Allele-specific Expression
:PROPERTIES:
:ID:       96C367A3-FEEB-4F6F-975A-A08A172FD277
:END:


********* RNA-fusion                                                :biopipe:
:PROPERTIES:
:CREATED:  [2019-06-25 Tue 09:05]
:ID:       96634f32-b19a-4390-8e50-e942abd35f90
:END:
:LOGBOOK:
CLOCK: [2019-07-09 Tue 09:19]--[2019-07-09 Tue 09:25] =>  0:06
CLOCK: [2019-07-02 Tue 08:20]--[2019-07-02 Tue 09:45] =>  1:25
CLOCK: [2019-06-27 Thu 08:50]--[2019-06-27 Thu 09:31] =>  0:41
CLOCK: [2019-06-27 Thu 08:17]--[2019-06-27 Thu 08:31] =>  0:14
CLOCK: [2019-06-26 Wed 13:35]--[2019-06-26 Wed 13:59] =>  0:24

http://101.datascience.community/tag/octave/
http://www.cbioportal.org/cgds_r.jsp
https://precision.fda.gov/
http://www.russpoldrack.org/?m=1
https://www.biostars.org/u/3737/
https://bioinformatics.ca/workshops/2014/informatics-rna-sequence-analysis-bc-2014
[[https://www.bioconductor.org/help/course-materials/][Bioconductor courses and conferences]]
[[https://www.ebi.ac.uk/training/online/course/embo-practical-course-analysis-high-throughput-seq/introduction-r-and-bioconductor][EMBO introduction to R and Biocinductor]]
[[https://www.edx.org/course/introduction-bioconductor-annotation-harvardx-ph525-5x][edX Intro to bioconductor]]
[[https://www.edx.org/course/high-dimensional-data-analysis-harvardx-ph525-4x][edX high-dimensional data analysis]]
[[https://www.coursera.org/learn/bioconductor][coursera bioconductor for genomic data science]]
http://bcb.dfci.harvard.edu/~aedin/courses/Bioconductor/

https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html

******** Gene Set Enrichment
:PROPERTIES:
:ID:       87734999-3139-4343-A323-925416811B12
:END:

- https://www.reddit.com/r/bioinformatics/comments/66374x/gsea_vs_go_enrichment/

********* [[http://www.bioconductor.org/packages/release/bioc/html/gage.html][GAGE]] (generally applicable gene-set enrichment for pathway analysis)
:PROPERTIES:
:ID:       441F7686-9969-4F8D-8092-C4429377214C
:END:

[[file:~/Box%20Sync/comp_bio/gsea_template/][Folder]]
[[file:~/Box%20Sync/comp_bio/gsea_template/gage.pdf][Manual]]
[[file:~/Box%20Sync/comp_bio/gsea_template/gsea_template.R][Rscript]]

********** RNA-Seq Workflow Example
:PROPERTIES:
:ID:       B687F931-3A5E-4CD0-B2B7-30DF026E0AB9
:END:

[[file:~/Box%20Sync/comp_bio/gsea_template/][Folder]]
[[file:~/Box%20Sync/comp_bio/gsea_template/RNA-seqWorkflow.pdf][pdf]]

********* Commercial
:PROPERTIES:
:ID:       417416B5-0169-41CA-8FCE-4D6E440C7E3A
:END:

http://www.ingenuity.com/
[[http://genome.ucsc.edu/cgi-bin/hgTables?command=start][create your own GTF]]
[[http://bedops.readthedocs.io/en/latest/index.html][BEDOPS]]
https://www.datacamp.com/

******** Genotype-Phenotype Associations
:PROPERTIES:
:ID:       69F82D91-6DE6-4332-AD0A-7ADF25263858
:END:

- cite:Denny_2011
- cite:Kho_2011


******** variant calling
:PROPERTIES:
:CREATED:  [2020-06-22 Mon 12:27]
:ID:       a5b408eb-77e2-4634-a5bb-950cefaafe18
:END:
- [[id:57E81A44-3233-479B-8264-2CD5F1904A60][Sequence variation and variant calling]]
********* fusions
:PROPERTIES:
:ID:       0ea6c74e-bcf8-44c2-b4c3-e725d930d4a3
:END:
  - cite:melsted2017  
********** Gene Fusion Detection
:PROPERTIES:
:ID:       492e0fad-ced4-407b-8913-80f3a0ac1202
:END:
:PROPERTIES:
:ID:       778C7C81-34E7-4BAE-94F9-5C33318BFDA5
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110
:END:
- [[https://github.com/jeszyman/rna-fusion][repository]]
- [ ] index script
- [ ] align script
- reference
  - https://www.nextflow.io/example4.html
  - gsnap example- - http://www.cureffi.org/2013/11/18/an-mrna-seq-pipeline-using-gsnap-samtools-cufflinks-and-bedtools/
  - https://github.com/genepattern/docker-tophat/blob/master/Dockerfile
  - https://github.com/broadinstitute/gtex-pipeline/blob/master/rnaseq/Dockerfile


************* NEXT INTEGRATE RNA aligners
:PROPERTIES:
:CREATED:  [2019-02-25 Mon 13:16]
:ID:       951F10BC-E844-45C5-8CB4-4FA8756D9F85
:END:

- are combined rna-seq alignment methods just sum of runs? 
- cite:zhang2016a
- [-] install
  - [ ] cite:wu2016 in cite:mathe2016
  - [X] hisat2
  - [X] star
  - [X] tophat - but prefer don't use- "TopHat has entered a low maintenance, low support stage as it is now largely superseded by HISAT2" https://ccb.jhu.edu/software/tophat/index.shtml
- [-] create indexes
  - [-] star
  - [ ] hisat2 https://ccb.jhu.edu/software/hisat2/manual.shtml see hista2-build
  - 

************ NEXT HCC1395 straight align and integrate and compare to published
:PROPERTIES:
:CREATED:  [2019-03-13 Wed 10:11]
:ID:       58F7BED2-DED1-4911-9E8C-104E12126A73
:END:
:LOGBOOK:
CLOCK: [2019-03-14 Thu 07:49]--[2019-03-14 Thu 08:45] =>  0:56
CLOCK: [2019-03-13 Wed 18:18]--[2019-03-13 Wed 19:54] =>  1:36
CLOCK: [2019-03-13 Wed 11:30]--[2019-03-13 Wed 12:00] =>  0:30
CLOCK: [2019-03-13 Wed 11:00]--[2019-03-13 Wed 11:01] =>  0:01
CLOCK: [2019-03-13 Wed 10:12]--[2019-03-13 Wed 10:47] =>  0:35
:END:
- log
  - from bam to fastq
    - by sample
      - tumor 
        - ./sambamba-0.6.8-linux-static sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam /gscmnt/gc2555/spencer/jeff/data/gerald_C1TD1ACXX_8_ACAGTG.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam
      - moved all this to ~data/HCC1395/
      - normal
        - (integrate image, 64gb) sambamba sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal-namesort.bam /gscmnt/gc2555/spencer/jeff/data/HCC1395/gerald_C2DBEACXX_3.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal.namesort.bam
        - 
  - alignment
    - bsub -Is -q docker-interactive -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/startest)' /bin/bash
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_2.fastq --outSAMunmapped Within
  - integrate
    - bsub -q research-hpc -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/integrate)' /bin/bash
    - prep each star output:
      - 
      - sambamba view -S -o Aligned.out.bam Aligned.out.sam 
      - sambamba sort Aligned.out.bam
      - sambamba index Aligned.out.sorted.bam
    - Integrate fusion /gscmnt/gc2555/spencer/jeff/data/reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa /gscmnt/gc2555/spencer/jeff/data/reference/annot-hg38-refeseq.txt /gscmnt/gc2555/spencer/jeff/data/reference/Integrate/bwts /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam
      - 

********* co-expression 
:PROPERTIES:
:ID:       91d46e60-eec3-48c7-97f1-b53eb1559b3f
:END:
- cite:rahmatallah2016
- [[id:9b222d1c-ab61-450c-8b77-a98ec3213867][gage]]
- https://bioconductor.org/packages/release/bioc/vignettes/pathview/inst/doc/pathview.pdf
- gscna
  - cite:rahmatallah2014
- gene set enrichment
  - https://david.ncifcrf.gov/gene2gene.jsp
  - https://genviz.org/module-04-expression/0004/03/01/pathwayAnalysis/
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - http://software.broadinstitute.org/gsea/msigdb/collections.jsp
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - pathway enrichment
    - - https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
    - - https://www.nature.com/articles/s41596-018-0103-9.pdf?proof=true19
    - https://bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
- [[id:948289d8-3a77-471a-b068-0507285e0bac][wgcna]]
  - wgcna
    - https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
    - cite:vanDam2018 
      #+begin_src bash 
    mkdir -p /tmp/wgnca-example
    cd /tmp/wgnca-example/
    wget https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip
    unzip FemaleLiver-Data.zip
    #+end_src
      #+begin_src R 
  setwd("/tmp/wgnca-example")
  source("~/repos/basecamp/bin/setup.R")
  #
  # ez install of wgcna and dependencies for R 3.6
  if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
  BiocManager::install("fastLiquidAssociation")
  #
  library(WGCNA)
  options(stringsAsFactors=F)
  femData=read.csv("./LiverFemale3600.csv")
  datExpr0=as.data.frame(t(femData[,-c(1:8)]))
  head(datExpr0)
  gsg=goodSamplesGenes(datExpr0, verbose=3)
  gsg$allOK
  if (!gsg$allOK)
  {
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
  printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
  printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
  }
  #
  sampleTree=hclust(dist(datExpr0),method="average")
  plot(sampleTree)
  #
  # Plot a line to show the cut
  abline(h = 15, col = "red");
  # Determine cluster under the line
  clust = cutreeStatic(sampleTree, cutHeight = 15, minSize = 10)
  table(clust)
  # clust 1 contains the samples we want to keep.
  keepSamples = (clust==1)
  datExpr = datExpr0[keepSamples, ]
  nGenes = ncol(datExpr)
  nSamples = nrow(datExpr)
  #
  traitData = read.csv("ClinicalTraits.csv")
  dim(traitData)
  names(traitData)
  # remove columns that hold information we do not need.
  allTraits = traitData[, -c(31, 16)]
  allTraits = allTraits[, c(2, 11:36) ]
  dim(allTraits)
  names(allTraits)
  # Form a data frame analogous to expression data that will hold the clinical traits.
  femaleSamples = rownames(datExpr)
  traitRows = match(femaleSamples, allTraits$Mice)
  datTraits = allTraits[traitRows, -1]
  rownames(datTraits) = allTraits[traitRows, 1]
  collectGarbage()
  #
  save(datExpr, datTraits, file = "FemaleLiver-01-dataInput.RData")
  #
  enableWGCNAThreads()
  lnames=load(file="FemaleLiver-01-dataInput.RData")
  #
  lnames
    #+end_src
- [[id:D68E1182-1F7B-4160-A6DD-BD2635F8085F][GSEA]]
********* Broad Calling variants in RNAseq
:PROPERTIES:
:ID:       DA64BC88-8BD6-40DF-9D80-5596BFA4D643
:END:

https://software.broadinstitute.org/gatk/documentation/article?id=3891
********* RNA-fusion                                                :biopipe:
:PROPERTIES:
:CREATED:  [2019-06-25 Tue 09:05]
:ID:       96634f32-b19a-4390-8e50-e942abd35f90
:END:
:LOGBOOK:
CLOCK: [2019-07-09 Tue 09:19]--[2019-07-09 Tue 09:25] =>  0:06
CLOCK: [2019-07-02 Tue 08:20]--[2019-07-02 Tue 09:45] =>  1:25
CLOCK: [2019-06-27 Thu 08:50]--[2019-06-27 Thu 09:31] =>  0:41
CLOCK: [2019-06-27 Thu 08:17]--[2019-06-27 Thu 08:31] =>  0:14
CLOCK: [2019-06-26 Wed 13:35]--[2019-06-26 Wed 13:59] =>  0:24
:END:
- [[https://github.com/jeszyman/rna-fusion][repository]]
- [ ] index script
- [ ] align script
- reference
  - https://www.nextflow.io/example4.html
  - gsnap example- - http://www.cureffi.org/2013/11/18/an-mrna-seq-pipeline-using-gsnap-samtools-cufflinks-and-bedtools/
  - https://github.com/genepattern/docker-tophat/blob/master/Dockerfile
  - https://github.com/broadinstitute/gtex-pipeline/blob/master/rnaseq/Dockerfile

************* NEXT INTEGRATE RNA aligners
:PROPERTIES:
:CREATED:  [2019-02-25 Mon 13:16]
:ID:       951F10BC-E844-45C5-8CB4-4FA8756D9F85
:END:

- are combined rna-seq alignment methods just sum of runs? 
- cite:zhang2016a
- [-] install
  - [ ] cite:wu2016 in cite:mathe2016
  - [X] hisat2
  - [X] star
  - [X] tophat - but prefer don't use- "TopHat has entered a low maintenance, low support stage as it is now largely superseded by HISAT2" https://ccb.jhu.edu/software/tophat/index.shtml
- [-] create indexes
  - [-] star
  - [ ] hisat2 https://ccb.jhu.edu/software/hisat2/manual.shtml see hista2-build
  - 

************ NEXT HCC1395 straight align and integrate and compare to published
:PROPERTIES:
:CREATED:  [2019-03-13 Wed 10:11]
:ID:       58F7BED2-DED1-4911-9E8C-104E12126A73
:END:
:LOGBOOK:
CLOCK: [2019-03-14 Thu 07:49]--[2019-03-14 Thu 08:45] =>  0:56
CLOCK: [2019-03-13 Wed 18:18]--[2019-03-13 Wed 19:54] =>  1:36
CLOCK: [2019-03-13 Wed 11:30]--[2019-03-13 Wed 12:00] =>  0:30
CLOCK: [2019-03-13 Wed 11:00]--[2019-03-13 Wed 11:01] =>  0:01
CLOCK: [2019-03-13 Wed 10:12]--[2019-03-13 Wed 10:47] =>  0:35
:END:
- log
  - from bam to fastq
    - by sample
      - tumor 
        - ./sambamba-0.6.8-linux-static sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam /gscmnt/gc2555/spencer/jeff/data/gerald_C1TD1ACXX_8_ACAGTG.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam
      - moved all this to ~data/HCC1395/
      - normal
        - (integrate image, 64gb) sambamba sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal-namesort.bam /gscmnt/gc2555/spencer/jeff/data/HCC1395/gerald_C2DBEACXX_3.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal.namesort.bam
        - 
  - alignment
    - bsub -Is -q docker-interactive -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/startest)' /bin/bash
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_2.fastq --outSAMunmapped Within
  - integrate
    - bsub -q research-hpc -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/integrate)' /bin/bash
    - prep each star output:
      - 
      - sambamba view -S -o Aligned.out.bam Aligned.out.sam 
      - sambamba sort Aligned.out.bam
      - sambamba index Aligned.out.sorted.bam
    - Integrate fusion /gscmnt/gc2555/spencer/jeff/data/reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa /gscmnt/gc2555/spencer/jeff/data/reference/annot-hg38-refeseq.txt /gscmnt/gc2555/spencer/jeff/data/reference/Integrate/bwts /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam
      - 

******** time course
:PROPERTIES:
:ID:       43f1772e-f8ce-4f6b-a4bd-d35d381b89c9
:END:
  - cite:spies2017
  - https://support.bioconductor.org/p/65676/#66860
  - https://www.biostars.org/p/138988/
  - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#group-specific-condition-effects-individuals-nested-within-groups
  - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
  - http://seqanswers.com/forums/showthread.php?t=64039
  - https://www.sciencedirect.com/science/article/pii/S2001037015000392
******** co-expression 
:PROPERTIES:
:ID:       91d46e60-eec3-48c7-97f1-b53eb1559b3f
:END:
- [[id:e4351ef3-6d41-4121-a2cc-c63d89f997ac][pathway analysis]]
- cite:rahmatallah2016
- [[id:9b222d1c-ab61-450c-8b77-a98ec3213867][gage]]
- https://bioconductor.org/packages/release/bioc/vignettes/pathview/inst/doc/pathview.pdf
- gscna
  - cite:rahmatallah2014
- gene set enrichment
  - https://david.ncifcrf.gov/gene2gene.jsp
  - https://genviz.org/module-04-expression/0004/03/01/pathwayAnalysis/
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - http://software.broadinstitute.org/gsea/msigdb/collections.jsp
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - pathway enrichment
    - - https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
    - - https://www.nature.com/articles/s41596-018-0103-9.pdf?proof=true19
    - https://bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
- [[id:948289d8-3a77-471a-b068-0507285e0bac][wgcna]]
  - wgcna
    - https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
    - cite:vanDam2018 
      #+begin_src bash 
    mkdir -p /tmp/wgnca-example
    cd /tmp/wgnca-example/
    wget https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip
    unzip FemaleLiver-Data.zip
    #+end_src
      #+begin_src R 
  setwd("/tmp/wgnca-example")
  source("~/repos/basecamp/bin/setup.R")
  #
  # ez install of wgcna and dependencies for R 3.6
  if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
  BiocManager::install("fastLiquidAssociation")
  #
  library(WGCNA)
  options(stringsAsFactors=F)
  femData=read.csv("./LiverFemale3600.csv")
  datExpr0=as.data.frame(t(femData[,-c(1:8)]))
  head(datExpr0)
  gsg=goodSamplesGenes(datExpr0, verbose=3)
  gsg$allOK
  if (!gsg$allOK)
  {
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
  printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
  printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
  }
  #
  sampleTree=hclust(dist(datExpr0),method="average")
  plot(sampleTree)
  #
  # Plot a line to show the cut
  abline(h = 15, col = "red");
  # Determine cluster under the line
  clust = cutreeStatic(sampleTree, cutHeight = 15, minSize = 10)
  table(clust)
  # clust 1 contains the samples we want to keep.
  keepSamples = (clust==1)
  datExpr = datExpr0[keepSamples, ]
  nGenes = ncol(datExpr)
  nSamples = nrow(datExpr)
  #
  traitData = read.csv("ClinicalTraits.csv")
  dim(traitData)
  names(traitData)
  # remove columns that hold information we do not need.
  allTraits = traitData[, -c(31, 16)]
  allTraits = allTraits[, c(2, 11:36) ]
  dim(allTraits)
  names(allTraits)
  # Form a data frame analogous to expression data that will hold the clinical traits.
  femaleSamples = rownames(datExpr)
  traitRows = match(femaleSamples, allTraits$Mice)
  datTraits = allTraits[traitRows, -1]
  rownames(datTraits) = allTraits[traitRows, 1]
  collectGarbage()
  #
  save(datExpr, datTraits, file = "FemaleLiver-01-dataInput.RData")
  #
  enableWGCNAThreads()
  lnames=load(file="FemaleLiver-01-dataInput.RData")
  #
  lnames
    #+end_src
- [[id:D68E1182-1F7B-4160-A6DD-BD2635F8085F][GSEA]]
******** [[*omics integration][omics integration]]
:PROPERTIES:
:ID:       db8191ed-937d-456b-90d6-63c39fa90758
:END:
****** single-cell RNA sequencing (scRNA-seq)
:PROPERTIES:
:ID:       2bbeb644-af54-4f07-80ad-3584e1e21b04
:END:
- https://www.10xgenomics.com/videos
- https://www.nature.com/articles/s41591-021-01276-y
- [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- single cell
  - https://www.biorxiv.org/content/biorxiv/early/2019/02/05/541433.full.pdf
  - examples
    - - see fig 3 & methods https://genome.cshlp.org/content/27/2/208.full.pdf
  - https://www.nature.com/articles/s41587-020-0553-9
  - 10x
    - applications - https://pages.10xgenomics.com/nur-gen-research-area-literature.html?src=email&utm_medium=email&lss=nurture&utm_source=nurture&cnm=es-nurture-ra_g-general&utm_campaign=es-nurture-ra_g-general&utm_content=&utm_term=&useroffertype=marketing-literature&userresearcharea=ra_g&userregion=&userrecipient=customer&mktouserid=1265915&cid=&usercampaignid=
  - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0212031
  - http://www.nature.com/nbt/journal/v34/n2/full/nbt.3445.html
- https://bioconductor.org/packages/release/bioc/vignettes/FlowSOM/inst/doc/FlowSOM.pdf
- https://chanzuckerberg.github.io/scRNA-python-workshop/analysis/05-diffexp.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6914766/
- https://www.nature.com/articles/s41467-019-12235-0
- http://www.kumc.edu/genomics/genome-sequencing-facility/services/chromium-single-cell-5-and-v(d)j-enrichment.html
- https://www.10xgenomics.com/solutions/vdj/
- https://research.medgenome.com/ngs-services/single-cell-sequencing/
- https://www.sciencedirect.com/science/article/pii/S1097276517300497#fig1
- https://www.biorxiv.org/content/10.1101/615013v1
- https://www.youtube.com/watch?v=Wbusqowyu6g&t=147s
- cite:monaco2019
- cite:see2018 
- toolkits
  - Seurat
  - Monocle 2
  - scVI
- clustering
  - SC3
  - SIMLR
  - CIDR
  - SNN-Cliq
  - Biscuit
- trajectory inference
  - Monocle 2
  - Slingshot
  - TSCAN
  - Wishbone
- imputing dropouts
  - MAGIC
  - scImpute
  - SAVER
- https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/mkfastq
- http://www.partek.com/free-trial/
- https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger
- https://github.com/koash/cellranger-docker
- https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html
- http://core-genomics.blogspot.com/2016/07/10x-genomics-single-cell-3mrna-seq.html
- https://www.frontiersin.org/articles/10.3389/fcell.2018.00108/full
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6292934/
- https://www.sciencedirect.com/science/article/pii/S1097276515002610?via%3Dihub
- https://www.ncbi.nlm.nih.gov/pubmed/26000846/
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5561556/
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6131190/
- https://www.nature.com/articles/s41598-019-46932-z
- https://www.biorxiv.org/content/biorxiv/early/2018/03/05/275859.full.pdf
- cite:wohnhaas2019 
- cite:papalexi2018
- https://support.10xgenomics.com/single-cell-gene-expression/datasets
- https://www.cell.com/cell-reports/pdf/S2211-1247(19)30059-2.pdf
- https://www.nature.com/articles/s41591-020-0844-1

- https://www.workcast.com/AuditoriumAuthenticator.aspx?cpak=8475927829006248&pak=1763506853003757
- cite:qi2019 
- applications
  - [[id:7bb87c84-6b06-44ef-9a1d-1bb9b9e57295][immunoprofiling]]
  - intra-tumor heterogeneity / [[id:12cfb78c-7912-4c68-b0f4-e2f90c112185][tumor microenvironment]]
    - cite:lawson2018 
- cite:stuart2019 
- [[id:1acf73b6-ca1e-458c-91c1-b93f17b80a02][single-cell illumina workshop [2019-09-10 Tue 09:00-15:30]​]]
- inputs and pre-processing
  - - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02048-6
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1171-9
  - https://www.nature.com/articles/s41591-020-0844-1
****** single-nucleus RNA-seq
:PROPERTIES:
:ID:       1149b091-eabe-4137-a3e9-222f21bfdd57
:END:
- https://www.nature.com/articles/s41591-020-0844-1
****** small RNA sequencing 
:PROPERTIES:
:ID:       fd462ab0-6aaf-4732-953b-141ad5e2ee72
:END:
:LOGBOOK:
CLOCK: [2021-02-17 Wed 11:08]--[2021-02-17 Wed 12:33] =>  1:25
CLOCK: [2021-02-17 Wed 09:46]--[2021-02-17 Wed 10:50] =>  1:04
CLOCK: [2021-02-17 Wed 09:23]--[2021-02-17 Wed 09:35] =>  0:12
CLOCK: [2021-02-17 Wed 09:08]--[2021-02-17 Wed 09:18] =>  0:10
CLOCK: [2021-02-05 Fri 15:43]--[2021-02-05 Fri 16:10] =>  0:27
:END:
******* [[id:51e5d70c-4a62-4960-880e-56f050194a0a][LINK TO Extracellular RNA biology]]
:PROPERTIES:
:ID:       49a0e52b-0bda-4237-8bab-ff2db54cf26d
:END:
******* Design
:PROPERTIES:
:ID:       4e8a8446-e84d-4c4f-8bd8-45ee611337b5
:END:
- spike-ins
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4538800/
- validation technologies
  - cannot distinguish isomiRs
  - qPCR expected to correlate >.9 w/ seq expression > 2^8 cite:yeri2018
  - qPCR more concordant than EdgeSeq or FirePlex cite:yeri2018 
- what depth? https://www.researchgate.net/post/how_many_million_reads_for_miRNA_discovery
- reads per sample
  - https://www.qiagen.com/us/resources/faq?id=a119d5f0-0a2a-4612-828e-d8fa3cc37891&lang=en
******* [[id:b2873d1a-837b-4927-b8cf-8a27e42d7bea][LINK TO small RNA extraction]]
:PROPERTIES:
:ID:       dfc3a8a2-e4c4-4043-af71-2db93128c18e
:END:
******* [[id:4E23127E-4ED6-4802-A03C-4103AA8CAF29][LINK TO RNA Quality and Quantification]]
:PROPERTIES:
:ID:       4df90313-39e4-44ed-83cd-b98453e18358
:END:
******* [[id:71289eeb-a929-44bd-a46e-a0c3dcd7a8f4][LINK TO small RNA library preparation]]
:PROPERTIES:
:ID:       a65f74ea-8211-4a51-97a3-a03f5bebb31b
:END:
******* [[id:3BC41E6A-00D3-4498-A48C-463F68CFD76B][LINK TO Massively parallel sequencing]]
:PROPERTIES:
:ID:       6584c294-78b9-42e5-a25b-28a069e112d8
:END:
******* Bioinformatics
:PROPERTIES:
:ID:       f46e278c-dd38-4603-9fa2-a973823e1333
:END:
- cite:buschmann2016
- https://link.springer.com/protocol/10.1007%2F978-1-4939-9042-9_4
******** Pre-processing
:PROPERTIES:
:ID:       0c7d62b9-6ef4-43d1-8e21-c3aa536125bd
:END:
- read filtering
  - adapter trimming
  - quality filtering
  - read length trimming
    - remove at less than 16 nt (see cite:buschmann2016 ref 157,158)
    - these are adapter dimers and degraded RNA   
- QC
  - filtered reads should have miRNA 21-23 nt peak and piRNA 30-32 nt peak 
  - overrepresented sequence- k-mer distribution- see cite:buschmann2016 pg 6007
- [[id:1EE731A3-83DD-46AF-B988-2311307C335F][LINK TO Raw sequence QC and processing]]    
******** Alignment
:PROPERTIES:
:ID:       d657a691-e338-44ad-96e4-47b4b18a03d2
:END:
- alignment to miRbase needs to be more conservative to avoid mismatches
******** QC
:PROPERTIES:
:ID:       fc7aeeb1-d387-4379-be37-46ebf1244cd7
:END:
https://exrna.org/resources/data/data-quality-control-standards/
******** [[id:853dd34c-27be-4be8-9794-19f04a1e22b0][LINK TO Quantification and normalization]]
:PROPERTIES:
:ID:       9a349ccd-3d29-46b7-b4fa-71693d32c674
:END:
******** [[id:ee030cef-74d4-435d-8228-b85f0e6203db][LINK TO Differential expression]]
:PROPERTIES:
:ID:       56d53b46-2a62-4777-b5c7-eedbc306d4b4
:END:
cite:buschmann2016
"y. Even though comparisons of software
performances on small RNA-Seq data are scarce, a num-
ber of independent and extensive evaluations for mRNAs
based on either synthetic data with clearly defined prop-
erties (216), or on biological datasets with validated gene
expressions (217,218) have been made recently. While it
was shown that statistical power of almost all methods is
heavily dependent on the number of samples per condition
and less on sequencing depth, the variability of expression
changes in biological datasets affects each analysis tool dif-
ferently. Outliers, ‘ON/OFF’ expression changes, where a
gene is detected in only one condition, and lopsided ex-
pression patterns, where upregulations drastically outweigh
downregulations or vice versa, influence specificity (false
positive rate) and sensitivity (false negative rate) of each
method unequally. Nonetheless, some methods appear to
capture the true expression status of small RNAs better
than others. Most independent evaluations seem to agree
that calling differential expression with SAMSeq works well
for datasets with sufficient sample sizes of 10 or more. For
smaller datasets, edgeR and especially the more conserva-
tive DESeq (or DESeq2) are found to be the methods of
choice. On"
******** Functional analysis
:PROPERTIES:
:ID:       e597d527-5538-4f69-bd28-a24a70969317
:END:
cite:buschmann2016 pg6011
******** Integrated tools
:PROPERTIES:
:ID:       07055c47-6e24-4f59-bce4-0ab91a7f5377
:END:
- miRDeep2 / miRDeep 
  - cite:friedlander2012
- exceRpt / genboree 
  - http://github.gersteinlab.org/exceRpt/
  - using 
    - cite:godoy2018
    - cite:freedman2016 
  - http://genboree.org/java-bin/workbench.jsp
- miRge / miRge2 WOS 29 times cited
  - https://mirge3.readthedocs.io/en/latest/quick_start.html
  - https://github.com/mhalushka/miRge3.0
  - cite:lu2018mirge  
  - https://github.com/mhalushka/miRge
    - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2287-y
- sRNAbench
  - https://academic.oup.com/nar/article/47/W1/W530/5494756
  - https://bioinfo2.ugr.es/ceUGR/srnabench/
  - sRNAtoolbox
    - https://arn.ugr.es/srnatoolbox/srnabench/
    - https://arn.ugr.es/srnatoolbox/static/sRNAtoolbox_manual.pdf
******* Applications
:PROPERTIES:
:ID:       55212621-170f-41aa-b9b5-b1c00385ffe6
:END:
******** Extracellular RNA bioinformatics
:PROPERTIES:
:ID:       cbf67ef7-39ec-45ec-b599-cdc0e8590cf7
:END:
- cite:coenen-stass2018
- https://www.zotero.org/jeszyman3/search/mu%20transcriptional%20analysis/titleCreatorYear

- spike ins use
  - cite:godoy2019 
  - cite:giraldez2018 
- pipelines- [[id:1db0ffb6-25b7-4c97-9aa0-263b8c96373b][exceRpt]]
- non-human plasma small RNA
  - cite:vliet2017
- human plasma small RNA
  - cite:godoy2019
  - cite:danielson2018
  - cite:godoy2018 
  - cite:max2018
  - cite:wong2019
  - cite:yeri2018
  - cite:danielson2017
  - cite:juzenas2017
  - cite:yeri2017
  - cite:freedman2016
- plasma transcriptome RNA
  - cite:zhao2018
  - cite:chen2019exrna
******** miRNA-seq
:PROPERTIES:
:ID:       903f32d8-bd13-496c-8d5c-26a63455936a
:END:
:LOGBOOK:
CLOCK: [2021-01-27 Wed 13:53]--[2021-01-27 Wed 13:57] =>  0:04
:END:
- https://www.nature.com/articles/srep05150#Sec11
- http://www.mircode.org/faq.php
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4652620/
- spike-ins
  - cite:qiagen2018mirnaspike cite:exiqon2016exiseq 
  - cite:locati2015
  - https://genome.cshlp.org/content/21/9/1543.long
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6161688/
- [[id:5d9593fd-f083-450f-a510-449aa15ae670][LINK TO miRNA biology]]
- [[id:95b3c7f3-3b84-45b1-b8bf-5e2e6e2a6f88][LINK TO miRNA data]]
- processes
  - preprocessing
    - read collapse just speads up downstream
      - see for read collapse https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0143066
  - QC
    - library complexity
    - https://arn.ugr.es/mirnaqc/
  - normalization
    - https://academic.oup.com/bib/article/16/6/950/226343
    - https://rnajournal.cshlp.org/content/18/6/1279.long
  - functional analysis
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4489228/
  - target identification
    - http://ophid.utoronto.ca/mirDIP/
  - [[id:82aafb6f-63a8-4ab7-867a-9add0e0c224a][LINK TO Integrated miRNA-mRNA analysis]]
- tools
  - manatee cite:handzlik2020
  - tool comparisons
    - https://www.frontiersin.org/articles/10.3389/fgene.2018.00022/full
    - cite:chen2019mirna
- general reference
  - cite:pritchard2012
    - citing, times cited WOS:        
      - cite:braunwald2013
      - cite:dong2013
      - cite:schwarzenbach2014
      - " Whereas high-speed centrifugation (14,000–16,000 g) can remove living cells and cell debris, separating exosomes and small shedding vesicles requires ultracentrifugation (>110,000 g).30 Blood cell count has a significant effect on miRNA expression levels in plasma;31 therefore, samples from patients with an inflammatory process, who should have high white blood cell count, should be excluded from analysis. Haemolysis is also known to markedly influence miRNA expression patterns, because of a contamination of abundant intracellular miRNAs.31"
  - [[http://apps.pathology.jhu.edu/mirnaprogram/links/http://apps.pathology.jhu.edu/mirnaprogram/links/][Johns Hopkins microRNA Bio lab]]
  - https://www.biostars.org/t/mirna/
- protocols
  - cite:fu2018mirna 
  - https://www.sciencedirect.com/science/article/pii/S266591312030131X#sec3
******* General reference
:PROPERTIES:
:ID:       65a85f59-afd5-4c6b-a785-9b6269cc7a51
:END:
- miRQC cite:mestdagh2014
- cite:mehta2014
- [[https://www.biostars.org/t/smallrna/?sort=votes&limit=all%20time][biostars small RNA]]
- cite:korpelainen2015 chapter 12
- best practices
  - https://www.encodeproject.org/data-standards/rna-seq/small-rnas/


***** reference
:PROPERTIES:
:ID:       70491293-d832-4947-b1bd-a0f4ae066c17
:END:
- my RNA-seq work
  - [[file:~/org/career.org::*high-grade head and neck characterization][high-grade head and neck characterization]]
  - [[id:e7a8362d-f4cf-45ec-a3f8-91b0ffbdc500][ffpe rna-seq project]]
  - [[file:~/data/career/pathology/molecular-pathology/ffpe-rna/ffpe-rnaseq.org]]
- [[https://www.nature.com/subjects/rna-sequencing/nmeth][Nature Methods RNA sequencing]]
- tool comparisons cite:sahraeian2017 
- data
  - public RNA sequencing data
    - [[https://webshare.bioinf.unc.edu/public/mRNAseq_TCGA/UNC_mRNAseq_summary.pdf][TCGA RSEM]]
      - https://support.bioconductor.org/p/131066/
      - http://zyxue.github.io/2017/06/02/understanding-TCGA-mRNA-Level3-analysis-results-files-from-firebrose.html
      - https://bsbludwig.com/post/94066296740/what-do-tcgas-rnaseq-files-actually-show
      - http://seqanswers.com/forums/showthread.php?t=42911
      - https://support.bioconductor.org/p/65890/
  - see [[http://www.cbil.upenn.edu/BEERS/][BEERS]] for generating fake RNA seq data
- cite:korpelainen2015 
  - [[http://rnaseq-book.blogspot.fi/][Korpelainen,15 web companion]]
- [[http://rnaseq.uoregon.edu/][RNAcyclopedia]]
- https://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools
- [[id:386d899a-9208-4d5c-93d8-3750b22d9748][RNA methods]]
***** Ideas
:PROPERTIES:
:ID:       B5E9166C-CB80-4578-BD90-1E694B46F192
:END:

[[http://bioinf.wehi.edu.au/marray/jsm2005/lab4/lab4.html][A limma workflow to try]]
[[http://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html][another workflow example]]

see Love,16 section "Rich visualization and reporting of results"

****** Pipeline Comparisons
:PROPERTIES:
:ID:       7B9AD35F-1511-4CB2-BB5A-0989EFEAADF3
:END:
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110

[[file:~/Box%20Sync/library/Fonseca,14%20RNA%20seq%20gene%20profiling%20a%20systematic%20empirical%20comparison.PDF][Fonseca,14 RNA seq gene profiling a systematic empirical comparison]]
[[file:~/Box%20Sync/library/Sigurgeirsson,14%20Analysis%20of%20stranded%20information%20using%20an%20automated%20procedure%20for%20strand%20specific%20RNA%20sequencing.pdf][Sigurgeirsson,14 Analysis of stranded information using an automated procedure for strand specific RNA sequencing]]
***** cell enumeration through bulk RNA-seq
:PROPERTIES:
:CREATED:  [2020-08-05 Wed 08:36]
:ID:       1a1d5b73-8070-402a-9f26-e147d66ca081
:END:
****** CIBERSORTx
:PROPERTIES:
:ID:       bb55c105-5977-4a01-802e-4bc1838f58bb
:END:

https://www.biostars.org/p/428905/
    - raw counts will be converted to CPM in cibersortx- see https://cibersortx.stanford.edu/tutorial.php
- cibersortx
  - cite:newman2015
  - [[mu4e:msgid:BYAPR02MB5592ACF7A79C1137A6F0A974B77B0@BYAPR02MB5592.namprd02.prod.outlook.com][Re: cibersortx from docker]]
  - docker pull lyronctk/cibersortxfractions
- have access token for docker: https://cibersortx.stanford.edu/getoken.php
**** Spatial transcriptomics
:PROPERTIES:
:ID:       053daf71-9f9f-4b5e-b400-222c4d24afea
:END:
https://mail.google.com/mail/u/0/#inbox/FMfcgxwLtQMfThxFrGwWlHQFgpTTDrJJ
https://mail.google.com/mail/u/0/#inbox/FMfcgxwLtGthWWRrdltpWVLTXhQKnZDs
https://mail.google.com/mail/u/0/#inbox/FMfcgxwLtGthWWRrdltpWVLTXhQKnZDs
http://labs.pathology.jhu.edu/halushka/science/
https://science.sciencemag.org/content/360/6389/660
- https://support.10xgenomics.com/spatial-gene-expression/sequencing/doc/specifications-sequencing-requirements-for-visium-spatial-gene-expression
**** Reference
:PROPERTIES:
:ID:       45a1d8fe-cdfb-4dce-8cf7-c7325c83e2d3
:END:
- http://gepia.cancer-pku.cn/
**** Transcription factor binding and perturbation 
https://genome.cshlp.org/content/29/8/1363.full
https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-018-3856-x
https://www.nature.com/articles/ncomms15089#Sec2
- cite:hu2019 
*** Epigenetics
:PROPERTIES:
:CREATED:  [2019-10-30 Wed 10:52]
:ID:       079e398f-e3b0-49c4-92da-9b4f6a73684c
:END:
https://www.illumina.com/destination/epigenetics-webinar-series.html
- methylation
  - https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/medip-seq-dip-seq.html
  - cite:shen2018
- epigenetics
    - ATAC-seq for chromatin accessiblity
    - ChIP-seq
    - differential peak analysis-
      - epigen-specific tools DiffBind, MAnorm
      - uses RNA-seq tools for differential expression 
    - peak-calling: identifying enriched signal from ATAC-seq and ChIP-seq, see cite:liu2019a ref 40
      - MACS/MACS2
      - measures of signal-to-noise by ENCODE
        - FRiP fraction of reads in peaks
        - cross-correlation profiles
    - methylation profiling
      - BSmooth
      - Merman
      - single cell
        - Minifi
      - bisulfite- unmethylated cytosines converted to uralcil, uracil converted to thyamine during amp- ratio T to C
      - methylcap-seq
      - differential peak
        - DiffBind
        - MAnorm
        - DBChIP
        - DAStk
      - Minfi
      - BSmooth
      - methylKit
      - mehtySig
      - metilene
**** ATAC-seq
:PROPERTIES:
:ID:       22e31d06-f5df-427e-bd70-3a2ccd3f47ec
:END:
:LOGBOOK:
CLOCK: [2021-05-26 Wed 07:53]--[2021-05-26 Wed 08:00] =>  0:07
:END:
- Design
  - Read length
    - https://www.biostars.org/p/418363/
- Bench
  - cite:corces2017 
  - cite:buenrostro2015
- QC
  - https://www.cell.com/cell-systems/pdfExtended/S2405-4712(20)30079-X
  - https://bioconductor.org/packages/release/bioc/vignettes/ATACseqQC/inst/doc/ATACseqQC.html
- Applications
  - ATAC-seq and miRNA-seq cite:abernathy2017 
  - Transcription factor binding sites- cite:li2019atac
- General method guides
  - cite:yan2020 
  - cite:corces2017 
  - cite:buenrostro2015
  - https://informatics.fas.harvard.edu/atac-seq-guidelines.html
- [[https://www.scopus.com/results/results.uri?cc=10&sort=cp-f&src=s&nlo=&nlr=&nls=&sid=e3c13d36864e217e2edcf12df7a64b0a&sot=b&sdt=b&sl=59&s=%28TITLE-ABS-KEY%28atac-seq%29+AND+TITLE-ABS-KEY%28bioinformatics%29%29&ss=cp-f&ps=r-f&editSaveSearch=&origin=resultslist&zone=resultslist][scopus atac-seq bioinformatics (42)]]
- atac-seq and mirna-seq https://www.sciencedirect.com/science/article/pii/S193459091730320X?via%3Dihub
- bioinformatics
  - cite:yan2020
- method-establishing papers
  - cite:buenrostro2013
  - cite:buenrostro2015
- for tissue
  - cite:corces2017 
- ENCODE pipeline
  - https://docs.google.com/document/d/1f0Cm4vRyDQDu0bMehHD7P7KOMxTOP-HiNoIvL1VcBt8/edit
  - https://github.com/ENCODE-DCC/atac-seq-pipeline
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5170597/
- https://www.illumina.com/techniques/popular-applications/epigenetics/atac-seq-chromatin-accessibility.html
- https://www.activemotif.com/blog-single-cell-atac-seq
**** methylation                                                    :biopipe:
:PROPERTIES:
:CREATED:  [2019-05-29 Wed 14:46]
:ID:       b8db46d3-8329-4fef-82b1-4d99c2c9312b
:END:
:LOGBOOK:
CLOCK: [2019-06-26 Wed 14:06]--[2019-06-26 Wed 14:45] =>  0:39
:END:
- Bisulfite sequencing
  - methylation protects/blocks cytosine conversion to uracil
  - PCR amp can produce forward/reverse with and without bisulfite block
  - BSMAP, Bismark, BWA-meth, biscuit
  - VCF > bedgraph > bigwig > IGV
  - metilene segments methylation
  - visualization
    - heatmap
    - canyon plot- methylation proportion vs distance from target locus
- cite:liu2020 
- https://www.nature.com/articles/s41591-020-0933-1
- https://www.nature.com/articles/nm.4045
- [[id:6ad7cfcb-1e8b-4d72-b18a-40f80a00f242][nih liq biopsy]] see [2021-01-26 Tue]
- "All the cell line datasets generated during and/or analysed during the current study are available in the GEO repository under accession code GSE79838. The
cfMeDIP-seq NGS data for patient samples that support the findings of this study are available upon request from the corresponding author (D.D.C) to comply with
institutional ethics regulation. Source data for Fig. 1d and e are provided in Supplementary Table 9, and for Figure 1G are provided in Supplementary Table 10.
Additional source data can be found on Zenodo (Supplementary Table 13)."
- to add
  - R
    - MethylKit
    - DESeq2
    - caret
    - minfi
    - Rtsne
  - SAMtools
  - Bowtie
  - MEDIPS
  - BWA
  - GATK IndelRealigner
  - MuTect
  - TrimGalore
  - Bismark6
  - Bowtie
  - 
- from cite:shen2018
  - "The generated SAM files from hg19 alignment were converted to BAM format, ensuring the removal of duplicate reads, and the reads were then sorted and indexed using SAMtools21 before subsequent analysis with the R package MEDIPS22. The CpG enrichment score, as a quality control measure for the immunoprecipitation reaction, was calculated as part of the MEDIPS package."
  - Shell scripts were used to process MeDIP sequencing data to produce RPKM estimates for downstream analysis. Custom R scripts with a collection of R/Bioconductor packages were used to analyse the resulting dataset. The code used for analysis is available in the form of R markdowns deposited onto Zenodo as detailed in Supplementary Table 13.
  - 
| Description                                                                                    | Zenodo DOI for Data    | Zenodo DOI for Code    |
| <25>                                                                                           | <25>                   | <25>                   |
| IHEC Figure                                                                                    | 10.5281/zenodo.1205714 |                        |
| Simulations                                                                                    | 10.5281/zenodo.1205730 | 10.5281/zenodo.1205728 |
| Extended Data Fig. 6 violin plots                                                              | 10.5281/zenodo.1205744 | 10.5281/zenodo.1205735 |
| Bed file of genomic windows that contain CpGs                                                  | 10.5281/zenodo.1205752 |                        |
| TCGA and COSMIC tSNEs                                                                          | 10.5281/zenodo.1205754 |                        |
| Machine learning                                                                               |                        | 10.5281/zenodo.1242697 |
| Markdown with ensemble of non-machine learning scripts including Extended Data Fig. 6 heatmaps |                        | 10.5281/zenodo.1205756 |
| Stand alone scripts called by non-machine learning markdown                                    |                        | 10.5281/zenodo.1205746 |

- Open source code/packages used for data analysis: R, SAMtools (v. 1.3.1), Bowtie (v. 0.12.9), MEDIPS (v. 1.22.0), BWA (v. 0.7.1220),
- Genome Analysis ToolKit (GATK) IndelRealigner, MuTect (v.1.1.5) , Trim Galore! (v. 0.4.4), Bismark (v. 0.10.1), Bowtie2 (v. 2.0.5),
- MethylKit (v 0.99.2), DESeq2 (v. 1.4.5), caret R package, minfi bioconductor package, Rtsne
- Commercial software used for data analysis: Microsoft Excel for Mac (v. 16.16.1)
=======
:PROPERTIES:
:ID:       413414a5-651d-4d43-82c8-a6bb13de8ca3
:END:
**** protein-DNA interactions
:PROPERTIES:
:CREATED:  [2021-01-18 Mon 09:45]
:ID:       b612a1b1-2c01-49c3-a20c-5a890fa1a6e5
:END:
- cut and run - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6302505/

**** transcription factor enrichment
:PROPERTIES:
:ID:       809e1a39-2a8e-46ec-863c-d7b03511fcfb
:END:
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4009616/
- https://science.sciencemag.org/content/352/6288/aad3018
- motif-based- high false pos
- benchtop with ChIP-seq 
*** Cytometry
:PROPERTIES:
:ID:       b7b36cba-0d4d-41b0-ae5d-79b0f0f67d2e
:END:
**** CyTOF
:PROPERTIES:
:ID:       881d23a0-c943-47ac-9bf5-ef965f8a80e3
:END:
- CATALYST
  - https://www.bioconductor.org/packages/release/bioc/html/CATALYST.html
  - https://www.bioconductor.org/packages/release/bioc/vignettes/CATALYST/inst/doc/differential.html
*** immunoprofiling
:PROPERTIES:
:ID:       7bb87c84-6b06-44ef-9a1d-1bb9b9e57295
:END:
- https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0213684
- methods
  - [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- cite:saloura2019 
- https://en.wikipedia.org/wiki/Immunological_Genome_Project#Gene-expression_compendium
**** immunoprofiling/ immunophenotyping, cell type ennumeriation with rnaseq
:PROPERTIES:
:CREATED:  [2019-10-03 Thu 10:22]
:ID:       e07fee1b-d7fd-42ce-9639-ff803cfc9c8e
:END:
:LOGBOOK:
CLOCK: [2020-02-13 Thu 11:25]--[2020-02-13 Thu 11:33] =>  0:08
CLOCK: [2020-02-13 Thu 10:41]--[2020-02-13 Thu 11:05] =>  0:24
:END:
- https://www.ncbi.nlm.nih.gov/pubmed/32036085
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1257-4
- cite:newman2015
- for rna-seq data, see cite:newman2019 suppliment, do transcript TPM (salmon) and sum to gene 
- The following metrics are provided for each mixture sample:
  - P-value: Statistical significance of the deconvolution result across all cell subsets; useful for filtering out results with a poor "goodness of fit." Increase the number of permutations (input page) to increase the number of significant digits.
  - Correlation: Pearson's correlation coefficient (R), generated from comparing the original mixture with the estimated mixture, the latter of which is calculated using imputed cell fractions and corresponding expression profiles from the signature genes file. Of note, the correlation is restricted to signature genes.
  - RMSE: Root mean squared error between the original mixture and the imputed mixture, restricted to genes in the signature gene file.
- in mouse
  - generating a reference from published data
    - single cell 
    - array
      - cite:chen2017
        - https://github.com/wuaipinglab/ImmuCC/tree/master/tissue_immucc
  - using pre-compiled reference
    - cite:wang2017a
      - https://github.com/lintian0616/bica
    - convert to human and use lm22
- enrichment- ID at pathway or gene set
  - by ssGSEA
    - ESTIMATE
    - xCell
- cite:liu2019a
  - immune repetoire sequencing- see liu2019a ref83, table 4, ref88-92
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5996037/
- https://www.nature.com/articles/s43018-019-0022-x
- https://www.sciencedirect.com/science/article/pii/S2589004218301871
- https://www.nature.com/articles/nbt.3724
- cite:yalcin2012
- https://mixcr.readthedocs.io/en/master/index.html
- https://www.nature.com/articles/s41592-019-0529-1
- https://www.biostars.org/p/354696/
- cite:vonstechow2017
- cell type enumeration with RNA-seq
  - https://www.nature.com/articles/s41586-019-1773-3

***** immunoprofiling / immunophenotyping
:PROPERTIES:
:CREATED:  [2019-10-30 Wed 10:52]
:ID:       a143e3d0-35b3-4579-8831-9710791e0a6a
:END:
- cite:liu2019a
- cite:papalexi2018 - immune cell heterogeneity by scRNAseq  

*** Proteomics
:PROPERTIES:
:ID:       e01aba61-2ea4-4235-a237-b0b90b49f153
:END:
https://www.degruyter.com/document/doi/10.1515/hsz-2017-0104/html
https://www.annualreviews.org/doi/10.1146/annurev-anchem-071015-041535
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201800247
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/epdf/10.1002/pmic.200800417
https://pubs.acs.org/doi/10.1021/pr400132j
https://www.embopress.org/doi/full/10.15252/msb.20156297
https://pubs.acs.org/doi/10.1021/acs.jproteome.5b0090
1https://www.sciencedirect.com/science/article/pii/S2212968513000044?via%3Dihub#fig0005
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201300377
https://www.mdpi.com/2227-7382/1/2/109/htm
https://www.olink.com/data-you-can-trust/technology/
Mass Spectrometry 

 MASCOT peptide mass fingerprint
 SILAC (stable isotope labeling by amino acids in cell culture) 

 LC/MS- 1995- 166. Appella, E., Padlan, E. A. & Hunt, D. F. Analysis of the
 structure of naturally processed peptides bound by
 class I and class II major histocompatibility complex
 molecules. Exs 73, 105–119 (1995).

 iTraq
 By Technique
 MALDI-TOF
 MS/MS
 By Experimental Design	


- [[id:179b243b-e6c0-484c-81ef-e8bff57e7f46][Circulating biomarkers]]
- proteome dynamic range is ~10^12, machine limit is usually ~10^3
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096388/
- https://link.springer.com/article/10.1007%2Fs00216-012-6203-4#Sec14
- https://www.nature.com/articles/nprot.2017.040  
- label-free
  - bioinformatics
    - MaxQuant https://www.maxquant.org/
    - Perseus
      - https://link.springer.com/protocol/10.1007/978-1-4939-7493-1_7
      - http://coxdocs.org/doku.php?id=perseus:start
**** SOMAscan
:PROPERTIES:
:ID:       89b4ec6d-992d-4551-81da-19011597732a
:END:
- somascan examples
    - https://www.ncbi.nlm.nih.gov/pubmed/27327800
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6277005/
- https://www.somalogic.com/wp-content/uploads/2017/01/SSM-046-Rev-2-Verification-and-Validation-of-the-SOMAscan-Assay-1.3k.pdf
- https://www.agilent.com/cs/library/eseminars/public/SomaLogic_Agilent_081816.pdf
- http://mohanlab.bme.uh.edu/wp-content/uploads/2015/09/HOU-SOMAscan-Methods-2016.pdf

- https://somalogic.com/wp-content/uploads/2017/06/SSM-002-Technical-White-Paper_010916_LSM1.pdf
- https://somalogic.com/wp-content/uploads/2017/06/SSM-071-Rev-0-Technical-Note-SOMAscan-Data-Standardization.pdf
- https://link.springer.com/protocol/10.1007%2F978-1-4939-9706-0_13
- [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd][email: GTAC SOMAscan notes]]
- cite:candia2017 
- https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd
- https://www.sciencedirect.com/science/article/pii/S1874391916304006
- For protein, SOMAscan has a higher dynamic range and lower limit of detection than comparable Luminex antibody-based assays (cite doi.org/10.1038/s41591-019-0665-2). Moreover, SOMAscan includes targets such as Jagged-1 and -2 not available for multiplexed antibody assays.
*** Metabolomics
:PROPERTIES:
:CREATED:  [2020-10-08 Thu 09:15]
:ID:       fce92cb5-cf68-4944-b2f4-eff55996ec5f
:END:
https://www.sciencedirect.com/science/article/pii/S1097276515003032
https://www.medrxiv.org/content/10.1101/2021.02.05.21251173v1
- http://www.visantnet.org/visantnet.html
*** Systems biology
:PROPERTIES:
:ID:       228D364B-48C8-49E4-9B00-FBFD52384623
:END:

- http://bioportal.bioontology.org/ontologies/GO/?p=classes&conceptid=root
- pathways
  - pathways
   Pathway Browsers and Visualizers 
   GenMAPP
   http://cytoscapeweb.cytoscape.org/
   NCBI BioStstems (pathway diagrams) 
   NCI Pathway Interaction Database
   http://www.reactome.org/ReactomeGWT/entrypoint.html
   NIH Molecular Interaction Maps
   Pathvisio
   Pathvisio pathway styles
   Wikipathways
- Geurts,09 Supervised learning with decision tree-based methods in computational.pdf
- Breker,09 Explorations in topology–delving underneath the surface of genetic.pdf
- Lima-Mendez,09 The powerful law of the power law and other myths in network biologyw.pdf
- - [[http://science.sciencemag.org/content/353/6306/aaf1420][A global genetic interaction network maps a wiring diagram of cellular function]]
- cite:chicco2017
- cite:sen2014
- cytoscape
  - https://github.com/mohd-akram/languages
  - [[http://www.nature.com/nmeth/journal/v9/n11/full/nmeth.2212.html][cytoscape plugins]]
- http://acacia.atspace.eu/Tutorial/Tutorial.html
- - http://amp.pharm.mssm.edu/Enrichr/#stats
- - https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btr656
- https://toppgene.cchmc.org/
- Pathway analysis
  - https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  - cite:Khatri_2012
- GEDI (Gene Expression Dynamics Inspector)
 Website
 Software v.2.1 Manual
 Invented Eichler,03
 Traditional hierarchical clustering and self organizing maps do not idnetify genome-wide emergent phenotypic profiles 
 Uses self organizing map to reduce dimensionality into miniclusters of 1-9 genes and create mosic patterns
- Kitano H (2002) Looking beyond the details: A rise in system-oriented approaches in genetics and molecular biology. Curr Genet 41: 1–10.
- Kitano H (2002) Systems biology: A brief overview. Science 295: 1662–1664.
- Ehrenberg M, Elf J, Aurell E, Sandberg R, Tegner J (2003) Systems biology is taking off. Genome Res 13: 2377–2380.
- http://www.springer.com/us/book/9780387989921
**** [[id:abb6fc6c-07d9-4375-a8f6-a87370f5c0ea][Multivariate biomarker discovery methods]]
:PROPERTIES:
:ID:       34a99004-090f-44c6-a721-f5f1054e5b1f
:END:
**** pathways
:PROPERTIES:
:CREATED:  [2019-09-28 Sat 10:37]
:ID:       e5089cac-0ade-48ea-9417-58bfb67a6338
:END:
- https://github.com/Adrienrougny/sbgntikz/
- pathways
  - https://github.com/PathwayCommons/chibe

***** decoding pt pathways ideker
:PROPERTIES:
:ID:       4CAA5252-B978-4A24-BED0-7A476E35FB44
:CREATED:  [2019-04-26 Fri 10:30]
:END:
:LOGBOOK:
CLOCK: [2018-05-25 Fri 14:00]--[2018-05-25 Fri 15:00] =>  1:00
:END:

- [[file:~/Box%20Sync/data/written.notes/180525.14.note.pdf][180525.14.note.pdf]]

**** pathway analysis
:PROPERTIES:
:CREATED:  [2020-07-20 Mon 10:05]
:ID:       e4351ef3-6d41-4121-a2cc-c63d89f997ac
:END:
- https://www.seriouseats.com/recipes/2019/11/sichuan-bang-bang-turkey-salad.html
- http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3618897&blobtype=pdf
- https://www.sciencedirect.com/science/article/pii/S0022519314001969?via%3Dihub
- https://link.springer.com/article/10.1186/s40246-019-0226-2
- methods papers
  - cite:reimand2019
  - cite:creixell2015 
- https://academic.oup.com/bioinformatics/article/33/3/414/2875813
- tools
  - https://old.reddit.com/r/bioinformatics/comments/jbmw6t/which_software_to_use_for_gene_functionalpathway/
  - http://www.visantnet.org/visantnet.html
  - https://www.nature.com/articles/ncomms13427#Sec10
    - https://github.com/varnivey/ipanda/blob/master/MANUAL.txt
  - [[id:D68E1182-1F7B-4160-A6DD-BD2635F8085F][GSEA]]
- tutorials
  - https://sbc.shef.ac.uk/workshops/2019-01-14-rna-seq-r/rna-seq-gene-set-testing.nb.html
  - https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2019/RNAseq/html/06_Gene_set_testing.html
- https://academic.oup.com/nar/article/48/D1/D479/5626534#190997651
- - http://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-2-r14
- https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btt285
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1790-4
- https://www.nature.com/articles/s41598-019-46059-1
- https://www.nature.com/articles/ncomms13427#Sec10
- cite:amadoz2019
- https://www.nature.com/articles/s41598-019-46059-1
- visualization
  - heatmaps
    - https://www.frontiersin.org/files/Articles/425394/fgene-10-00858-HTML/image_m/fgene-10-00858-g002.jpg
    - https://www.bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
  - wire diagrams
  - bar and lolipop for p-value and set size
    - https://www.nature.com/articles/s41598-018-29420-8
    - https://www.frontiersin.org/articles/10.3389/fgene.2019.00858/full
**** Omics integration
:PROPERTIES:
:ID:       00e4056a-e9b7-48e2-ba0e-3b37e8e36c69
:END:
- https://workflows.omnipathdb.org/
- example papers
  - https://academic.oup.com/nar/article/47/19/10010/5573547
- https://academic.oup.com/bioinformatics/article/36/8/2632/5689817
- cite:higdon2015 
- https://jme.bioscientifica.com/view/journals/jme/62/1/JME-18-0055.xml
- http://mixomics.org/webinar-pls-methods/
- http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3618897&blobtype=pdf
- https://danielhomola.com/assets/DanielHomola_PhD.pdf
- cite:bersanelli2016
- cite:rappoport2018
- cite:subramanian2020
- cite:wu2019
- https://link.springer.com/protocol/10.1007%2F978-1-0716-0849-4_12

*** Bioinformatics visualization 
:PROPERTIES:
:ID:       DB8463CB-97CB-49E6-8EDD-A5ECA2F140BF
:END:
- heatmaps
  - https://jokergoo.github.io/2020/05/06/translate-from-pheatmap-to-complexheatmap/
  - cite:lucas2020 fig3c
- https://cmdcolin.github.io/awesome-genome-visualization/
- https://sites.google.com/view/williamgilks/computational-analysis-of-whole-genome-sequencing-data/flow-chart-in-latex

- https://github.com/Adrienrougny/sbgntikz/
- https://bioinfogp.cnb.csic.es/tools/venny/
- heatmaps
  - https://www.biostars.org/p/287512/
  - https://support.bioconductor.org/p/93424/
- https://www.photopea.com/
- https://observablehq.com/@petulla/statistical-power-explorable
- https://www.fusioncharts.com/resources/chart-primers/radar-chart
- http://miso.readthedocs.io/en/fastmiso/sashimi.html
- https://genome.sph.umich.edu/wiki/LocusZoom_Standalone
- http://genviz.org/
- https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  [[http://svviz.readthedocs.io/en/latest/install.html][documentation]]
**** <<<Integrated Genomics Viewer>>> (<<<IGV>>>)
:PROPERTIES:
:ID:       917062AC-32E8-41F5-A511-84AA7B1504B0
:END:

- cite:robinson2017
https://www.france-bioinformatique.fr/sites/default/files/EBA/V3-2014/Elodie_Giraud_visualisation-IGV_TP_EBA-2014-10.pdf
[[http://software.broadinstitute.org/software/igv/UserGuide][User Guide]]

*** Special contexts
**** Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)
:PROPERTIES:
:ID:       0f1f4aa4-c755-42ff-8f81-8a32c14293a0
:END:
- fragment size
  - https://www.nature.com/articles/s41586-020-2140-0
  - https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-06848-9#Sec9
***** ctDNA WES
:PROPERTIES:
:ID:       e387037a-85c0-4eda-a26f-9b6386e95a34
:END:
- https://clincancerres.aacrjournals.org/content/26/11/2556#sec-6
- cite:parikh2019
** Tools
:PROPERTIES:
:ID:       1b8bb7db-2622-47bd-bd6f-052cbd5d4fdb
:END:
*** table of tools
:PROPERTIES:
:ID:       a0923938-4021-44cd-a8f3-94c0eb10cdb5
:END:
| tool           | function          | input     | repo   | doc      | paper | paper(local)           |
|----------------+-------------------+-----------+--------+----------+-------+------------------------|
| annotationDb   |                   |           |        | [[https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf][doc]]      |       |                        |
| annovar        |                   |           |        | [[http://annovar.openbioinformatics.org/en/latest/][doc]]      |       |                        |
| Ballgown       |                   |           |        |          | [[https://www.nature.com/articles/nprot.2016.095][paper]] |                        |
| BEDtools       |                   |           |        | [[http://bedtools.readthedocs.io/en/latest/][doc]]      |       | cite:quinlan2010       |
| BLAT           |                   |           |        |          |       |                        |
| BLAT           |                   |           |        |          |       |                        |
| bowtie2        |                   |           | [[https://sourceforge.net/projects/bowtie-bio/files/][src]]    | [[http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#the-bowtie2-build-indexer][doc]]      |       |                        |
| bwa            | align             | fastq     |        | [[http://bio-bwa.sourceforge.net/bwa.shtml][site]]     |       |                        |
| CNVkit         | copy#             | bam       | [[https://github.com/etal/cnvkit/tree/fd3555256d2a5aa3978c475d19c88065be68c6c4][repo]]   | [[https://cnvkit.readthedocs.io/en/stable/index.html][doc]]      | [[https://doi.org/10.1371/journal.pcbi.1004873][paper]] |                        |
| complexheatmap |                   |           |        |          |       |                        |
| data-slicer    |                   |           |        | [[http://www.1000genomes.org/data-slicer][site]]     |       |                        |
| deepTools      |                   |           |        | [[https://deeptools.readthedocs.io/en/latest/content/example_usage.html][site]]     |       |                        |
| DESeq2         | diff express      |           |        |          |       | cite:love2014          |
| edgeR          | diff express      |           |        | [[https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf][vignette]] |       | cite:robinson2010      |
| fastp          | [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]]            | fastq     | [[https://github.com/OpenGene/fastp][repo]]   |          | [[https://watermark.silverchair.com/bty560.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAuEwggLdBgkqhkiG9w0BBwagggLOMIICygIBADCCAsMGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMUQnq5IYxgkrFnt0WAgEQgIIClChzxtpO28QYuBn1woUVAMnQOwkPRZWSY20cwnzbVuF3DlOcpwjeNjNrO_ixNp6csW5pOdBZ81CHjV65BMcqsZmP9KOT-yVGy4r_7mWKLuzm7Siid24pM9-xF3k7y1RoBK4-PrnZgzp-J1svrUvOZAphbBPmsezuOowSuj9o9EZyTciDJbUQ2klUbYSRDO_InWNOVu23eOzGpJQwer81BL2iS8TsCfeWF-eZtzNBRuZZr1UZ2PhX_mVQGu4bHvKn1Bmoe64CFYkEJGLoka87zVVfo--J_hAledjiW1_-wEj8kZSyb2diZoP9ABDGOJA6aQ7fzx8qNGgNAb_oK-uAmEdfAfepI_0LECDdlk9EIxIJITC-Z59gk_vNVophJOydBimGZHFK0iACl-N-nZNALjIqRXeRmqOxcEdRFmNy6onzLalfCV0wyqFae42JIVzEYFSLW1oQVC0ugMfKTTmOfysRZ8-4bboBo1UbskpGVlKvrSBTFuDyvuBlPoUTECbOrO6bsdeNk8yfTMTeMQzNzOzZYZ1mQVCULZtuE_YDCuKmwukT3EeUy8iOJbp7pD2Ew5kphJ9q0sV-_io3h0jBZEdJmUr6dTPSRcv56pLP6gtqso6ZdB9bWk3sFNt--tBwIGhQvnLXgyV7xGqRYNPrDeMbs2tPLlK-v4D8XKj-BHGi1sOVuzL1rCvFve0knpL4btsqw_BDp8P2XuFDyp0OOXtmE3dKJktLR0QyQZ3u4-aKksDMfkqrglbPoBjEhInpdpfiDQbEDTFO9qiaL9iajdsV8T636x_KHXpdGkFqO3GpcFyJv1QJyPW2qhAXuneiqWx6t60pJ8f0IyAzRV9BIUoQ4B3soV9TXIuSP171Tf0JYmcY5Q][paper]] | cite:chen2018b         |
| [[id:0a319950-3215-4846-a020-a08b78f3323d][FastQC]]         | [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]]            |           |        | [[https://www.bioinformatics.babraham.ac.uk/projects/fastqc/INSTALL.txt][site]]     |       |                        |
| [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][FASTX-Toolkit]]  |                   |           |        |          |       |                        |
| featureCounts  |                   |           |        |          | [[https://watermark.silverchair.com/btt656.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAApgwggKUBgkqhkiG9w0BBwagggKFMIICgQIBADCCAnoGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMofurkoV2NWsQqlelAgEQgIICS7iktDnfv1X3fe1vdQQcXCS09zT549UGvBjDCLHoNebBGfiQi_Eo7ZjUHVH_Yay_p1DlFd4eDeXU4WEX6HTQ-MzPqmtZ6RMGegyEptsqxaLnvr3QfpGO6WRT01dViqp6xya9-CqCRGV3E7F37VruIYzZEFjr-2svIgoRydPyF4pkSzA2hkvwV_g-JT2YSnqboZ5DOGtwZXeMaLKf48_aEpuYemRUx8VqOfM8xnOm4_jumLGd_BBqCCv6brLX8z3WVJpWfJ4YfpJXhemFRt8mSvIsAe9pYXV52W7-psxHlM2ocXzdD_7vLSD7iy1NocDXWLFT5HRH_Cq8NyiRq1s4Mq-QYuvQJFlTxWArVC6vdpE40bDr9slGD6KTJ0DpUQeC52q-aoKJXr9Y7Pi2SDo881eFuJxTWuSuMjxcXStn9MkPKZfcKa10Vu4CdJSSnvJAKAvd4GsshsfCVGCvkgy4ZfEabo5nOZaB82UYC-8AZEpdbD3RtJg1FUQJHDz0tlY-2gZYf2VB3KU1JK4wZcHzn8T8R5BSChB6MjFrRdTZeo3M8-ZIK3eWQGH2vLj8_bPsYlBF-wfj2JzSG3zJGgUUa_rSBzoK8Vvw8k_BPxCq63h6u-k49l-X0o-MO6f8mOdm_YliTT0dWu_ujvlbXMw9AIBrKjwBJncq27EwMOX0shm8nnzPMoPTfOzoekZR2nYTDPayGjHhqkAs6haFAQrI8XAAd0PfbMtW4l54ArULWEH6hCVxzDX8_h8F8oWgOc_3h2ktpbTLNsiX6Bxj][paper]] |                        |
| G2P            |                   |           |        |          | [[https://www.nature.com/articles/s41467-019-10016-3][paper]] |                        |
| GAGE           | enrichment        | express   |        |          | [[https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-161][paper]] |                        |
| GATK           |                   |           |        | [[https://software.broadinstitute.org/gatk/documentation/][doc]]      |       |                        |
| GISTIC2        | copy# diff        |           |        | [[http://portals.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&paper_id=216&p=t][site]]     |       |                        |
| GSEA           |                   |           |        |          | [[https://www.pnas.org/content/102/43/15545.full][paper]] |                        |
| GSNAP          |                   | fastq     |        | [[http://research-pub.gene.com/gmap/src/README][doc]]      |       | cite:mathe2016         |
| histat2        |                   | fastq     |        | [[https://ccb.jhu.edu/software/hisat2/manual.html][doc]]      |       |                        |
| ichorCNA       | copy#             | wig       | [[https://github.com/broadinstitute/ichorCNA][repo]]   | [[https://github.com/broadinstitute/ichorCNA/wiki][doc]]      | [[https://www.nature.com/articles/s41467-017-00965-y][paper]] | cite:adalsteinsson2017 |
| Integrate      | rna fusion        | bam       |        |          |       | cite:zhang2014a        |
| limma          | diff express      | count     |        |          | [[https://dx.doi.org/10.1093%2Fnar%2Fgkv007][paper]] |                        |
| MultiQC        | seq QC agg        |           | [[https://github.com/ewels/MultiQC][github]] | [[https://multiqc.info/docs/][doc]]      |       |                        |
| oligocalc      |                   |           |        | [[http://www.basic.northwestern.edu/biotools/oligocalc.html][site]]     |       |                        |
| picard         |                   |           |        | [[https://broadinstitute.github.io/picard/index.html][doc]]      |       |                        |
| polyester      | RNA simulation    | fasta,gtf |        | [[http://bioconductor.org/packages/release/bioc/vignettes/polyester/inst/doc/polyester.html][vignette]] |       |                        |
| rnaseqc        | rnaseq qc         | bam       | [[https://github.com/broadinstitute/rnaseqc][repo]]   | [[https://github.com/broadinstitute/rnaseqc][repo]]     |       |                        |
| rsem           | transcript quant  | bam       | [[https://github.com/deweylab/RSEM][repo]]   |          | [[https://doi.org/10.1186/1471-2105-12-323][paper]] | cite:li2011            |
| RSeQC          | rnaseq qc         | bam       |        | [[http://rseqc.sourceforge.net/][repo]]     | [[https://doi.org/10.1093/bioinformatics/bts356][paper]] | cite:wang2012          |
| salmon         | transcript quat   |           |        |          |       |                        |
| sambamba       | sam processing    | bam       |        | [[http://lomereiter.github.io/sambamba/docs/sambamba-view.html][online]]   |       |                        |
| samblaster     | single-pass dedup | bam       | [[https://github.com/GregoryFaust/samblaster][repo]]   | [[https://github.com/GregoryFaust/samblaster][repo]]     | [[https://academic.oup.com/bioinformatics/article/30/17/2503/2748175][paper]] |                        |
| samtools       |                   | bam       |        | [[http://www.htslib.org/doc/samtools.html][online]]   |       |                        |
| scVelo         | RNA velocity      |           |        | [[https://scvelo.readthedocs.io/][site]]     |       |                        |
| [[https://github.com/vsbuffalo/scythe][Scythe]]         |                   |           |        |          |       |                        |
| seqtk          |                   |           | [[https://github.com/lh3/seqtk][github]] | [[https://github.com/lh3/seqtk/blob/master/README.md][README]]   | NA    | NA                     |
| seqTools       | [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]]            |           |        | [[https://www.bioconductor.org/packages/release/bioc/vignettes/seqTools/inst/doc/seqTools.pdf][seqTools]] |       |                        |
| simNGS         | sim reads         | none      |        | [[https://www.ebi.ac.uk/goldman-srv/simNGS/][website]]  |       |                        |
| STAR           |                   |           |        | [[https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf][doc]]      |       | cite:dobin2019         |
| tophat2        |                   |           |        | [[https://ccb.jhu.edu/software/tophat/manual.html][doc]]      |       |                        |
| tximport       |                   |           |        | [[https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem][doc]]      | [[http://dx.doi.org/10.12688/f1000research.7563.1][paper]] |                        |
| WGCNA          |                   |           |        |          | [[https://doi.org/10.1186/1471-2105-9-559][paper]] |                        |
*** bowtie
:PROPERTIES:
:ID:       8883a37b-867c-4c19-8e9c-2b46004125a1
:END:
- [[https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471250953.bi1107s32][protocol paper]]
- [[http://bowtie-bio.sourceforge.net/index.shtml][doc]]
*** BBmap / BBSplit 
:PROPERTIES:
:ID:       52363054-5138-419c-8d45-4ec9b0b02b24
:END:
- http://seqanswers.com/forums/archive/index.php/t-87619.html
- https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbmap-guide/
- outu = unmapped reads
- snakemake example https://www.biostars.org/p/424706/
- http://seqanswers.com/forums/showthread.php?t=41288
- [ ] BBSplit in BBmap - [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwJZJWQrGSPgdPGNfStWbxXkmhP][email]]]]
- https://www.google.com/search?q=bbmap+docker use bioconda
*** BLAT
:PROPERTIES:
:ID:       6e44c130-0eaa-4e44-8c12-c7c8125ab9ec
:END:
  - https://www.youtube.com/watch?v=E6DDNGDXH8A
*** canvas
:PROPERTIES:
:ID:       74ef57cd-c44a-4f28-b4a1-5a54a82ce5a3
:END:
https://github.com/Illumina/canvas


  - https://github.com/chrisamiller/copyCat
  - for cah https://bioconductor.org/packages/release/bioc/html/DNAcopy.html


*** CIBERSORTx and cytotrace
:PROPERTIES:
:ID:       6fdb7ece-efbb-4523-a7a5-7dd594d199dd
:END:
- https://cytotrace.stanford.edu/
*** cnvkit
:PROPERTIES:
:ID:       b471102b-2e5a-4d67-97e6-b03cf4d923b9
:END:
- modify autobin https://github.com/etal/cnvkit/blob/fa86632a5731ebbefb0092069c8031f806c8b611/cnvlib/autobin.py
oldcode from mpnst

# tumor_bam="/duo4/capp-runs/HiSeqW11/demultiplexed/Sample_PC3-N1_Normal/Sample_PC3-N1_Normal.sorted.samtools-deduped.sorted.bam"
# normal_bam="/duo4/capp-runs/HiSeqW11/demultiplexed/Sample_69-C1_cfDNA/Sample_69-C1_cfDNA.sorted.samtools-deduped.sorted.bam"
# bait_bed="/duo4/users/jszymanski/projects/mpnst/170907_HG38_CCDS1X_REZ_capture_targets.bed"
# fasta="/duo4/users/jszymanski/data/ref/hg19.fa"

# cd $workdir/analysis/


# nohup cnvkit.py batch \
# 	  $tumor_bam \
# 	  --normal $normal_bam \
# 	  --targets $bait_bed \
#           --fasta $fasta \
#           --output-reference my_reference.cnn \
# 	  --output-dir results/ \
#           --diagram --scatter

# # Reusing a reference for additional samples
# cnvkit.py batch *Tumor.bam -r Reference.cnn -d results/

# # Reusing targets and antitargets to build a new reference, but no analysis
# cnvkit.py batch -n *Normal.bam --output-reference new_reference.cnn \
#     -t my_targets.bed -a my_antitargets.bed \
#     -f hg19.fasta -g data/access-5kb-mappable.hg19.bed
# https://cnvkit.readthedocs.io/en/stable/nonhybrid.html
## considers WGS as jsut no off-target regions
## so no target file
## NEED TO ANNOTATE
#wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/refFlat.txt.gz
#mv refFlat.txt.gz /duo4/users/jszymanski/data/ref/
#cd /duo4/users/jszymanski/data/ref/
#gunzip refFlat.txt.gz

source ~/repos/mpnst/bin/local-setup.sh
docker_interactive
biotools
mkdir -p /duo4/mpnst/downsample-cnn
cd /duo4/mpnst/downsample-cnn

cnvkit.py batch /duo4/mpnst/downsample-bam/pn*.sorted.bam /duo4/mpnst/downsample-bam/mp*.sorted.bam\
          -n /duo4/mpnst/downsample-bam/nm*.sorted.bam \
          -m wgs -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa \
          --annotate /drive3/users/jszymanski/data/ref/refFlat.txt -p 32


for sample in /duo4/mpnst/downsample-cnn/*.cnr;
do
    filename=$(basename -- "$sample")
    extension="${filename##*.}"
    filename="${filename%.*}"
    cnvkit.py scatter -s ${filename}.cns ${filename}.cnr -o /duo4/mpnst/downsample-cnn/$filename-scatter.pdf 
    cnvkit.py diagram -s ${filename}.cns ${filename}.cnr -o /duo4/mpnst/downsample-cnn/$filename-diagram.pdf 
    cnvkit.py heatmap ${filename}.cns ${filename}.cnr -o /duo4/mpnst/downsample-cnn/$filename-heatmap.pdf 
done

# washu plex ref
# 11 normal reference 
cnvkit.py reference \
          new_HiSeqW31_Undetermined_R6000232_L005001_ATCG.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_ATCG.sorted.targetcoverage.cnn \
          new_HiSeq11GTGT.sorted.antitargetcoverage.cnn \
          new_HiSeq11GTGT.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GAAG.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GAAG.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GCTA.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GCTA.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_CTTC.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_CTTC.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_TAGC.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_TAGC.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_AGGT.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_AGGT.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GCTA.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GCTA.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_CACA.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_CACA.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GTGT.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GTGT.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GAAG.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GAAG.sorted.targetcoverage.cnn \
          -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa -o normal-reference.cnn

# my best mpnst vs normals
cnvkit.py fix --no-edge new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.targetcoverage.cnn new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.antitargetcoverage.cnn normal-reference.cnn -o testMPNST.cnr
cnvkit.py segment -t 1e-6 testMPNST.cnr -o testMPNST.cns
cnvkit.py scatter -s testMPNST.cn{s,r} -o test2.png
# vs. nci plex ref
cnvkit.py fix --no-edge new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.targetcoverage.cnn new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.antitargetcoverage.cnn reference.cnn -o testMPNST_nciPlexRef.cnr
cnvkit.py segment -t 1e-6 testMPNST_nciPlexRef.cnr -o testMPNST_nciPlexRef.cns
cnvkit.py scatter -s testMPNST_nciPlexRef.cn{s,r} -o testMPNST_nciPlexRef.png


#NOTE           new_HiSeqW31_Undetermined_R6000232_L005001_TGTG. d/n exist- mislabel? 
# testing nci mpnst vs our normals 
for fullfile in MPNST_*antitargetcoverage.cnn; do
    filename=$(basename -- "$fullfile")
    filename2="${filename%%.*}"
    echo $filename2
    cnvkit.py fix --no-edge /duo4/mpnst/cnn/${filename2}.sorted.targetcoverage.cnn \
              $fullfile normal-reference.cnn -o /duo4/mpnst/cnn/${filename2}.normref.cnr
done
#
# normal ref vs all cnns
for fullfile in *antitargetcoverage.cnn; do
    filename=$(basename -- "$fullfile")
    filename2="${filename%%.*}"
    echo $filename2
    cnvkit.py fix --no-edge /duo4/mpnst/cnn/${filename2}.sorted.targetcoverage.cnn \
              $fullfile normal-reference.cnn -o /duo4/mpnst/cnn/${filename2}.normref.cnr
done

#
for file in *.normref.cnr; do
    filename="${file%.*}"
    cnvkit.py segment -t 1e-6 $file -o ${filename}.cns     
done
#
for file in *.normref.cnr; do
    filename="${file%.*}"
    cnvkit.py scatter $file -s ${filename}.cns -o ${filename}.png
done

# renaming
##MOVE TO bam consolidation script
for file in new_HiSeq-19*; do mv $file NORMAL_${file}; done
for file in new_HiSeq-20*AGGT*; do mv $file PN_${file}; done
for file in new_HiSeq-20*ATCG*; do mv $file MPNST_${file}; done
for file in new_HiSeq-20*CTTC*; do mv $file PN_${file}; done
for file in new_HiSeq-20*GTGT*; do mv $file MPNST_${file}; done
for file in new_HiSeq-20*TAGC*; do mv $file MPNST_${file}; done
for file in new_HiSeq-20*TCCT*; do mv $file MPNST_${file}; done
for file in new_HiSeq15*ACAC*; do mv $file PN_${file}; done
for file in new_HiSeq15*ATCG*; do mv $file MPNST_${file}; done
for file in new_HiSeq15*CGAT*; do mv $file PN_${file}; done
for file in new_HiSeq15*GAAG*; do mv $file MPNST_${file}; done
for file in new_HiSeq15*GTGT*; do mv $file MPNST_${file}; done
for file in new_HiSeq15*TAGC*; do mv $file MPNST_${file}; done
for file in new_HiSeq28_*ACAC*; do mv $file PN_${file}; done
for file in new_HiSeq28_*AGGT*; do mv $file PN_${file}; done
for file in new_HiSeq28_*ATCG*; do mv $file MPNST_${file}; done
for file in new_HiSeq28_*CACA*; do mv $file PN_${file}; done
for file in new_HiSeq28_*CTTC*; do mv $file PN_${file}; done
for file in new_HiSeq28_*GAAG*; do mv $file PN_${file}; done
for file in new_HiSeq28_*GCTA*; do mv $file PN_${file}; done
for file in new_HiSeq28_*GTGT*; do mv $file MPNST_${file}; done
for file in new_HiSeq28_*TAGC*; do mv $file MPNST_${file}; done
for file in new_HiSeq28_*TCCT*; do mv $file MPNST_${file}; done
for file in new_HiSeq28_*TGTG*; do mv $file PN_${file}; done
for file in new_HiSeq38_*; do mv $file MPNST_${file}; done
for file in new_HiSeq39_*; do mv $file MPNST_${file}; done
for file in new_HiSeq40_*; do mv $file MPNST_${file}; done
for file in new_HiSeq41_*; do mv $file MPNST_${file}; done
for file in new_HiSeq43_*; do mv $file MPNST_${file}; done
for file in new_HiSeqW30_*; do mv $file MPNST_${file}; done
for file in new_HiSeqW31_*; do mv $file NORMAL_${file}; done
for file in new_HiSeqW34_*; do mv $file MPNST_${file}; done
for file in new_HiSeqW35_*; do mv $file MPNST_${file}; done
for file in new_HiSeqW36_*; do mv $file PN_${file}; done
#
=======

# washu plex ref
# 11 normal reference 
cnvkit.py reference \
          new_HiSeqW31_Undetermined_R6000232_L005001_ATCG.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_ATCG.sorted.targetcoverage.cnn \
          new_HiSeq11GTGT.sorted.antitargetcoverage.cnn \
          new_HiSeq11GTGT.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GAAG.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GAAG.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GCTA.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GCTA.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_CTTC.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_CTTC.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_TAGC.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_TAGC.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_AGGT.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_AGGT.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GCTA.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GCTA.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_CACA.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_CACA.sorted.targetcoverage.cnn \
          new_HiSeq-19_L006001_GTGT.sorted.antitargetcoverage.cnn \
          new_HiSeq-19_L006001_GTGT.sorted.targetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GAAG.sorted.antitargetcoverage.cnn \
          new_HiSeqW31_Undetermined_R6000232_L005001_GAAG.sorted.targetcoverage.cnn \
          -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa -o normal-reference.cnn

# my best mpnst vs normals
cnvkit.py fix --no-edge new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.targetcoverage.cnn new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.antitargetcoverage.cnn normal-reference.cnn -o testMPNST.cnr
cnvkit.py segment -t 1e-6 testMPNST.cnr -o testMPNST.cns
cnvkit.py scatter -s testMPNST.cn{s,r} -o test2.png
# vs. nci plex ref
cnvkit.py fix --no-edge new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.targetcoverage.cnn new_HiSeq40_Undetermined_R6000281_L006001_GTGT.sorted.antitargetcoverage.cnn reference.cnn -o testMPNST_nciPlexRef.cnr
cnvkit.py segment -t 1e-6 testMPNST_nciPlexRef.cnr -o testMPNST_nciPlexRef.cns
cnvkit.py scatter -s testMPNST_nciPlexRef.cn{s,r} -o testMPNST_nciPlexRef.png


#NOTE           new_HiSeqW31_Undetermined_R6000232_L005001_TGTG. d/n exist- mislabel? 
# testing nci mpnst vs our normals 
for fullfile in MPNST_*antitargetcoverage.cnn; do
    filename=$(basename -- "$fullfile")
    filename2="${filename%%.*}"
    echo $filename2
    cnvkit.py fix --no-edge /duo4/mpnst/cnn/${filename2}.sorted.targetcoverage.cnn \
              $fullfile normal-reference.cnn -o /duo4/mpnst/cnn/${filename2}.normref.cnr
done
#
for file in *.normref.cnr; do
    filename="${file%.*}"
    cnvkit.py segment -t 1e-6 $file -o ${filename}.cns     
done
#
for file in *.normref.cnr; do
    filename="${file%.*}"
    cnvkit.py scatter $file -s ${filename}.cns -o ${filename}.png
done

>>>>>>> 0ec20f73fda99d63d7625bd25a8c38d6952566dd
# our mpnst vs nci plex ref
declare -a washu_mpnst=("new_HiSeq41_Undetermined_R6000281_L007001_AGGT.sorted.antitargetcoverage.cnn" "new_HiSeq39_Undetermined_R6000281_L005001_TCCT.sorted.antitargetcoverage.cnn")
for fullfile in "${washu_mpnst[@]}"; do
    filename=$(basename -- "$fullfile")
    filename2="${filename%%.*}"
    echo $filename2       
    cnvkit.py fix --no-edge /duo4/mpnst/cnn/${filename2}.sorted.targetcoverage.cnn \
              $fullfile reference.cnn -o /duo4/mpnst/cnn/${filename2}.nciPnRef.cnr
    cnvkit.py segment -t 1e-6 /duo4/mpnst/cnn/${filename2}.nciPnRef.cnr -o /duo4/mpnst/cnn/${filename2}.nciPnRef.cns
    cnvkit.py scatter /duo4/mpnst/cnn/${filename2}.nciPnRef.cnr -s /duo4/mpnst/cnn/${filename2}.nciPnRef.cns -o /duo4/mpnst/cnn/${filename2}.nciPnRef.png
done
#

# nci plex vs normal ref
for fullfile in PN_*_*.antitargetcoverage.cnn; do
    filename=$(basename -- "$fullfile")
    filename2="${filename%%.*}"
    echo $filename2
    cnvkit.py fix --no-edge /duo4/mpnst/cnn/${filename2}.sorted.targetcoverage.cnn \
              $fullfile normal-reference.cnn -o /duo4/mpnst/cnn/${filename2}.normRef.cnr
    cnvkit.py segment -t 1e-6 /duo4/mpnst/cnn/${filename2}.normRef.cnr -o /duo4/mpnst/cnn/${filename2}.normRef.cns
    cnvkit.py scatter /duo4/mpnst/cnn/${filename2}.normRef.cnr -s /duo4/mpnst/cnn/${filename2}.normRef.cns -o /duo4/mpnst/cnn/${filename2}.normRef.png
done

#



#


cnvkit.py batch /duo4/mpnst/bam-nci/MPNST_14_JS0108CD040119.sorted.bam -n /duo4/mpnst/bam-nci/PN_13_JS0028CD111918.sorted.bam -m wgs -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa --annotate /drive3/users/jszymanski/data/ref/refFlat.txt --targets /drive3/users/jszymanski/data/ref/genic_hg19.bed -p 36

cnvkit.py batch /duo4/mpnst/bam-nci/MPNST_*.sorted.bam -n /duo4/mpnst/bam-nci/PN_*.sorted.bam -m wgs -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa --annotate /drive3/users/jszymanski/data/ref/refFlat.txt --targets /drive3/users/jszymanski/data/ref/genic_hg19.bed -p 36

cnvkit.py batch /duo4/mpnst/bam-washu/*.sorted.bam -n /duo4/mpnst/bam-nci/PN_*.sorted.bam -m wgs -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa --annotate /drive3/users/jszymanski/data/ref/refFlat.txt --targets /drive3/users/jszymanski/data/ref/genic_hg19.bed -p 36

cnvkit.py scatter -s new_HiSeq28_Undetermined_R6000204_L006001_TCCT.sorted.cnr -o test2.png

cnvkit.py coverage /duo4/mpnst/bam-nci/*.sorted.bam \
          /drive3/users/jszymanski/data/ref/genic_hg19.bed \
          
          

          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF487C1_cfDNA/Sample_NF487C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF561C1_cfDNA/Sample_NF561C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF564C1_cfDNA/Sample_NF564C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF570C1_cfDNA/Sample_NF570C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF536C1_cfDNA/Sample_NF536C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          -n \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF544C1_cfDNA/Sample_NF544C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF574C1_cfDNA/Sample_NF574C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF575C1_cfDNA/Sample_NF575C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF576C1_cfDNA/Sample_NF576C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF531C1_cfDNA/Sample_NF531C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          /duo4/capp-runs/HiSeqW28/demultiplexed/Sample_NF571C1_cfDNA/Sample_NF571C1_cfDNA.sorted.samtools-deduped.sorted.bam \
          -m wgs -f /duo4/users/jszymanski/data/ref/hg19.fa --annotate /duo4/users/jszymanski/data/ref/refFlat.txt

cd $basedir
cnvkit.py batch $basedir/nci-bam/MPNST*.sorted.bam \
          -n $basedir/nci-bam/PN*.sorted.bam \
          -m wgs -f /media/jeszyman/4edbbc27-7e29-4667-9b24-f5a56d49d94e/users/jeszyman/data/ref/bwa-hg19/hg19.fa --annotate /media/jeszyman/4edbbc27-7e29-4667-9b24-f5a56d49d94e/users/jeszyman/data/ref/refFlat.txt


# start copy-number docker
bsub -Is -q docker-interactive -a 'docker(jeszyman/copy-number)' /bin/bash
workdir="/gscmnt/gc2561/pfeifer/ad_hoc/szymanski/projects/mpnst"
fasta="/gscmnt/gc2561/pfeifer/ad_hoc/szymanski/data/ref/Homo_sapiens.GRCh37.dna.toplevel.fa"
baits="/gscmnt/gc2561/pfeifer/ad_hoc/szymanski/projects/mpnst/data/ref/trim_bait.bed"
cnvkit.py target $baits --split -o $workdir/data/ref/my_targets.bed
# batch run
## NOHUP- takes order of days to run 
cd $workdir
rm nohup.out 
nohup cnvkit.py batch $workdir/data/*_tumor.bam \
          --normal $workdir/data/*_normal.bam \
          --targets $workdir/data/ref/my_targets.bed \
          --fasta $fasta \
          --access $workdir/data/ref/access.hg37.bed \
          --output-reference my_reference.cnn \
          --output-dir $workdir/analysis/batch/ \
          -p 8 \
          --diagram --scatter
# step-by-step cnvkit copy number pipeline
cd $workdir/data/ref
## target
cnvkit.py target $baits --split -o $workdir/data/ref/my_targets.bed
## access: calculate the sequence-accessible coordinates in chromosomes from the given reference genome, output as a BED file
cnvkit.py access $fasta -o $workdir/data/ref/access.hg37.bed
## prebuild with low-complexity excluded: 
wget https://raw.githubusercontent.com/etal/cnvkit/master/data/access-5k-mappable.hg19.bed > $workdir/data/ref/access-5k-mappable.hg19.bed

## antitarget
### using 5k mappable: 
cnvkit.py antitarget $workdir/data/ref/my_targets.bed -g $workdir/data/ref/access-5k-mappable.hg19.bed -o $workdir/data/ref/my_antitargets.bed
## autobin
cnvkit.py autobin $workdir/data/*.bam -t $workdir/data/ref/my_targets.bed -g $workdir/data/ref/access-5k-mappable.hg19.bed
## coverage
### NEEDS NOHUP AND ~days to run
## reference
## fix
## segment
## call




# autobin will get bam depths and estimate on and off target bin sizes as bed files
#  in batch, autobin is run for normal samples
# nohup cnvkit.py autobin $workdir/data/*_normal.bam -t $workdir/data/ref/trim_bait.bed -g ./access-5k-mappable.hg19.bed
# the above failed starting with SK436_normal.bam, maybe a bad bam file? 
### current command here ####

# nohup cnvkit.py batch $workidr/data/*tumor.bam \
# --normal $workdir/data/*normal.bam --targets $workdir/data/ref/trim_bait.bed \
# --fasta /gscmnt/gc2561/pfeifer/ad_hoc/szymanski/data/ref/Homo_sapiens/GRCh38.dna.toplevel.fa

# # NOT WORKING cnvkit.py scatter BI386_tumor.cnr -s BI386_tumor.cns


# # szymanski@virtual-workstation4:~$ tail -n +2 ~/projects/mpnst/data/ref/trim_bait.bed > ~/projects/mpnst/data/ref/trim_bait.bed 
# # ADD SAMTOOLS THEN FOR
# # ValueError: BED file './bait.target.bed' chromosome names don't match any in BAM file         'BI386_normal.bam'
# # do samtools idxstats <>.bam | head -n 3 



# # Generate Plots
# ## Scatter

# ########1########2########3########4########5########6########7########8
mkdir -p /duo4/users/jszymanski/repos/mpnst/analysis/cfDNA-WGS/
cd /duo4/users/jszymanski/data/mpnst/

for sample in /duo4/users/jszymanski/data/mpnst/*.cnr;
do
    filename=$(basename -- "$sample")
    extension="${filename##*.}"
    filename="${filename%.*}"
    cnvkit.py scatter -s ${filename}.cns ${filename}.cnr -o /duo4/users/jszymanski/repos/mpnst/analysis/cfDNA-WGS/$filename-scatter.pdf 
    cnvkit.py diagram -s ${filename}.cns ${filename}.cnr -o /duo4/users/jszymanski/repos/mpnst/analysis/cfDNA-WGS/$filename-diagram.pdf 
    cnvkit.py heatmap ${filename}.cns ${filename}.cnr -o /duo4/users/jszymanski/repos/mpnst/analysis/cfDNA-WGS/$filename-heatmap.pdf 
done




for sample in /duo4/users/jszymanski/data/mpnst/*.cnr;
do
    filename=$(basename -- "$sample")
    extension="${filename##*.}"
    filename="${filename%.*}"
    cnvkit.py scatter -s ${filename}.cns ${filename}.cnr -c chr8 -o /duo4/users/jszymanski/repos/mpnst/analysis/cfDNA-WGS/$filename-scatter-chr8.pdf 
done


*** complexheatmap
:PROPERTIES:
:ID:       af78e4e0-a184-480f-95ac-ce9a0db35246
:END:

- https://www.biostars.org/p/286187/#286507
*** Cytoscape
:PROPERTIES:
:ID:       46cf789e-d185-46da-bb32-8cf6113f3425
:END:
- [[http://manual.cytoscape.org/en/stable/index.html][documentation]]
- https://stackoverflow.com/questions/tagged/cytoscape
*** DESeq2
:PROPERTIES:
:ID:       2F642F69-A9C8-46DA-BE31-783AC8D4F4B6
:CREATED:  [2021-01-27 Wed 07:00]
:END:
:LOGBOOK:
CLOCK: [2021-01-26 Tue 11:53]--[2021-01-26 Tue 13:00] =>  1:07
:END:
- design
  - https://support.bioconductor.org/p/110163/
- [[file:~/Box%20Sync/library/Love,16%20DESeq2%20manual.pdf][Manual]]
- vignettes
  - [[https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#exploratory-analysis-and-visualization][gene workflow 19]]
  - [[http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#using-parallelization][package vignette]]
- handling nested designs https://support.bioconductor.org/p/64480/
- https://genviz.org/module-04-expression/0004/02/01/DifferentialExpression/
- https://bioc.ism.ac.jp/packages/2.14/bioc/vignettes/DESeq2/inst/doc/beginner.pdf
- [[file:~/Box%20Sync/library/Love,14%20Moderated%20estimation%20of%20fold%20change%20and%20dispersion%20for%20RNA%20seq%20data%20with%20DESeq2.pdf][Love,14 Moderated estimation of fold change and dispersion for RNA seq data with DESeq2]]
- a negative binomial GLM 
- http://www.nathalievilla.org/doc/html/solution-DESeq2-rnaseq.html
- import functions
  | function          | package           | framework | output               | DESeq2 input function    |
  | summarizeOverlaps | GenomicAlignments | R         | SummarizedExperiment | DESeqDataSet             |
  | featureCounts     | Rsubread          | R         | matrix               | DESeqDataSetFromMatrix   |
  | tximport          | tximport          | R         | list of matrices     | DESeqDataSetFromTximport |
  | htseq-count       | HTSeq             | Python    | files                | DESeqDataSetFromHTSeq    |
**** Examples- see [[file:~/Box%20Sync/comp_bio/deseq2_template/][deseq2_template]]
:PROPERTIES:
:ID:       5C8E2A3B-2C16-4FB3-9F86-5A1D8AD4802F
:END:

***** DONE Example 1: [[file:~/Box%20Sync/comp_bio/deseq2_template/Love,15%20RNA%20seq%20workflow%20gene%20level%20exploratory%20analysis%20and%20differential%20expression.pdf][Love,15 Airway]]
CLOSED: [2016-06-27 Mon 14:45]
:PROPERTIES:
:ID:       791A1ABC-4138-4CD2-82AE-63905C053571
:END:
:LOGBOOK:  
CLOCK: [2016-06-27 Mon 10:22]--[2016-06-27 Mon 10:37] =>  0:15
CLOCK: [2016-06-23 Thu 09:22]--[2016-06-23 Thu 10:19] =>  0:57
CLOCK: [2016-06-22 Wed 17:56]--[2016-06-22 Wed 18:30] =>  0:34
CLOCK: [2016-06-22 Wed 14:50]--[2016-06-22 Wed 16:02] =>  1:12
CLOCK: [2016-06-22 Wed 12:09]--[2016-06-22 Wed 12:57] =>  0:48
CLOCK: [2016-06-22 Wed 11:40]--[2016-06-22 Wed 11:46] =>  0:06
:END:      

***** DONE Example 2: [[file:~/Box%20Sync/comp_bio/deseq2_template/https___software.rc.fas.harvard.edu_ngsdata_workshops_2015_March_DESeq2_fruitfly.pdf][Drosophila]]
CLOSED: [2016-06-27 Mon 14:45]
:PROPERTIES:
:ID:       CC047BA0-35A3-420B-BB60-CA1AA7AE275B
:END:

https://software.rc.fas.harvard.edu/ngsdata/workshops/2015_March/DESeq2_fruitfly.html

***** DONE Example 3: Hox
CLOSED: [2016-06-28 Tue 13:48]
:PROPERTIES:
:ID:       97126EA4-E1CD-4CD5-9746-A7C9393DF712
:END:

http://www.gettinggeneticsdone.com/2015/12/tutorial-rna-seq-differential.html

***** Example 4 [[file:~/Box%20Sync/comp_bio/Love,16%20Differential%20analysis%20of%20RNA%20seq%20data%20using%20the%20DESeq2%20package.pdf][Love,16 Differential analysis of RNA seq data the DESeq2 package]]
:PROPERTIES:
:ID:       61BDAC0E-3F02-482A-868E-93AC26B33361
:END:
:LOGBOOK:  
CLOCK: [2016-06-22 Wed 11:15]--[2016-06-22 Wed 11:40] =>  0:25
CLOCK: [2016-06-21 Tue 16:41]--[2016-06-21 Tue 16:59] =>  0:18
CLOCK: [2016-06-21 Tue 15:33]--[2016-06-21 Tue 16:31] =>  0:58
CLOCK: [2016-06-20 Mon 13:48]--[2016-06-20 Mon 14:21] =>  0:33
CLOCK: [2016-06-20 Mon 12:21]--[2016-06-20 Mon 12:23] =>  0:02
:END:      

****** Standard workflow
:PROPERTIES:
:ID:       ABBDC255-8B16-4598-AA75-6D0ADAD3E48E
:END:

******* Input data
:PROPERTIES:
:ID:       71704A3E-FEE2-4363-BA67-4CA68D516375
:END:

******** SummarizedExperiment input
:PROPERTIES:
:ID:       F4CCA76E-3FBE-4E20-B6DE-69FB848D7123
:END:

- DESeqDataSet
  - is an extension of the RangedSummarizedExperiment class
  - requires design formula
    - ~ followed by variables w/ plus sign between
    - put variable of interest at end of formula
- Preferred count algorithm is summarizedOverlaps w/ mode="Union"

******** Pre-filtering
:PROPERTIES:
:ID:       1D88E488-4AFA-4A0F-A994-3C1664B5D156
:END:

- by DESeq2
  - on mean of normalized counts per gene- maximized genes w/ adjusted p value below a FDR of alpha 
- (by user)
  - increases processing speed and reduces memory
  - b/ not necessary for DESeq2 function

******* Exploring and exporting results
:PROPERTIES:
:ID:       B87063F9-4578-440B-BF94-9DF134AF438A
:END:

******** MA-plot
:PROPERTIES:
:ID:       BBEB1ADB-F328-4B7B-807D-F8C158D60083
:END:

- useful to look at how the DESeq2 function shrinkage performed
- after calling plotMA, the identify function can interactively select dots
- for unshrunken results, add addMLE=T to get a column called lfcMLE 

******** Plot counts
:PROPERTIES:
:ID:       CB3647FE-90B2-47AD-8E89-AA206A6C207F
:END:

- plotCounts will give counts normalized by sequencing depth for a specific gene 

******** p values set to NA
:PROPERTIES:
:ID:       F28D686F-0E61-42C9-81DB-6BD581223377
:END:

- all samples w/ zero counts
- rows w/ extreme count outlier by Cook's distance
- low mean normalized count (here only the p value will be NA) 

******** Rich visualization and reporting of results- checkout Bioconductor packages [[http://bioconductor.org/packages/release/bioc/html/ReportingTools.html][ReportingTools]], regionReport, [[http://bioconductor.org/packages/release/bioc/html/Glimma.html][Glimma]], and pcaExplorer
:PROPERTIES:
:ID:       8CE69F91-1AFB-477A-996A-95212336AD6D
:END:

****** Data transformations and visualization
:PROPERTIES:
:ID:       99C0CCAA-43A0-4E37-8C72-98617C51D928
:END:

******* Count data transformations
:PROPERTIES:
:ID:       3E7192A1-175C-4D3A-A5E7-08D6CC64DCEC
:END:

- transformations for EDA
  - pseudocounts: y=log2(N+No)
  - log2-based alternatives
    - find experiment-wide trend in variabnce/mean and remove dependnece of variance on mean 
    - rlog
    - VST (variance stabilizing transformations) 

******** Blind dispersion estimation
:PROPERTIES:
:ID:       BFA60D58-A2E6-49F4-9912-2CF15D0BD356
:END:

- blind to sample information is default
- change if many/most genes will have large differences- can be checked with MA over-shrinking 

****** Variations to the standard workflow
:PROPERTIES:
:ID:       7A2DB9FE-8DE5-481A-9C1A-4F26DC867582
:END:

******* Contrasts
:PROPERTIES:
:ID:       A6AC8C5D-8828-4B96-BDB4-760B1F0DE0BA
:END:

- Contrast
  - i.e. w/ levels A,B,C, test all possible differences of A vs B, A vs C, B vs C, ... 
  - linear combination of estimated log2 fold changes

******* Interactions- tests whether log2 fold change attributable to one condition is different based on another condition
:PROPERTIES:
:ID:       E2EE7973-FFB1-4842-83E8-665DCCA8F2C9
:END:

******* Likelihood ratio test
:PROPERTIES:
:ID:       F15FCEE5-B88F-4712-B3B1-03C576F8E44F
:END:

- Wald test asks if log2 fold change b/w groups is zero based on mean and SE
- LR
  - asks whether full model (w/ groups) and reduced model (w/ some terms removed) are different
  - like a ANOVA except this is analysis of deviance

******* Approach to count outliers
:PROPERTIES:
:ID:       682697F8-D316-4A04-9B6D-8BCB0261DE15
:END:

- Cook's distance- for how much single sample is influencing fitting coefficients
  - automatic in results function
  - requires at least 3 replicates
  - @ >6 samples, instead of eliminating gene, the outlier is replaced w/ the trimmed mean over all samples
  - not run when continuous variables are in the design 

******* Independent filtering of results
:PROPERTIES:
:ID:       0778E394-2A3A-4952-9126-668AC8162D3A
:END:

results function maximizes p values less than * over quantiles of mean normalized counts

******* Tests of log2 fold change above or below a threshold
:PROPERTIES:
:ID:       70E0F04B-9D6B-4C11-A34E-6EE2FB008185
:END:

******* Access to all calculated values
:PROPERTIES:
:ID:       60580A3B-7A73-45A0-984F-0E83E5B6F2EC
:END:

******* Sample-/gene-dependent normalization factors
:PROPERTIES:
:ID:       BB95C4F2-763B-447D-A840-372588D43C79
:END:

******* “Model matrix not full rank”
:PROPERTIES:
:ID:       7F14D356-858D-46DD-9044-C1A6CD2F8A08
:END:

******** Linear combinations
:PROPERTIES:
:ID:       D9625DD4-D506-4E43-836D-9E0CDDCB9A79
:END:

******** Levels without samples
:PROPERTIES:
:ID:       E65BA44B-FBA9-4A05-93B0-1246608C3754
:END:

****** Theory behind DESeq2
:PROPERTIES:
:ID:       C0765E52-718B-4771-9ECC-3689EE7CAA21
:END:

******* The DESeq2 model
:PROPERTIES:
:ID:       08949A97-2A7C-4D21-A9EB-811440E003A8
:END:

******* Changes compared to the DESeq package
:PROPERTIES:
:ID:       41E50955-4ADF-4560-ACD0-86F58F544033
:END:

******* Methods changes since the 2014 DESeq2 paper
:PROPERTIES:
:ID:       1D28C489-FF28-4112-B9B9-70C09D382608
:END:

******* Count outlier detection
:PROPERTIES:
:ID:       14989EE3-4903-4FCB-A270-E3BAFFCD6EFA
:END:

******* Contrasts
:PROPERTIES:
:ID:       45C7C99A-DC18-4293-9381-878DE892D84A
:END:

******* Expanded model matrices
:PROPERTIES:
:ID:       FBAAB9AE-7D25-43A5-9363-766F84F32198
:END:

******* Independent filtering and multiple testing
:PROPERTIES:
:ID:       EB54BD27-FD99-45DD-B671-8BBEB06857FE
:END:

******** Filtering criteria
:PROPERTIES:
:ID:       4705CA0F-C007-4503-A898-74E946304812
:END:

******** Why does it work?
:PROPERTIES:
:ID:       BA1FC087-AAF6-4116-9B86-0B0B4BBC5FDB
:END:

****** Frequently asked questions
:PROPERTIES:
:ID:       7A2CF9E1-0C06-45FF-9637-C085CC6F15A0
:END:

******* How can I get support for DESeq2?
:PROPERTIES:
:ID:       B5ADEA5D-15C6-4449-B15F-908E63118B76
:END:

******* Why are some pvalues set to NA?
:PROPERTIES:
:ID:       3D889B88-07E4-4D2C-91F1-58CCF54C9C2C
:END:

******* How can I get unfiltered DESeq results?
:PROPERTIES:
:ID:       A4107221-A74A-4E79-AC6E-057F5CAB86DF
:END:

******* How do I use the variance stabilized or rlog transformed data for differential testing?
:PROPERTIES:
:ID:       F6C4F4CE-03E9-46FC-8F9A-EF915DF7FEF2
:END:

******* Can I use DESeq2 to analyze paired samples?
:PROPERTIES:
:ID:       DA9671DE-23CE-4603-9077-0F8576CEECBF
:END:

******* If I have multiple groups, should I run all together or split into pairs of groups?
:PROPERTIES:
:ID:       5384671F-9FF2-4589-80E4-7B6E7AA681FF
:END:

******* Can I run DESeq2 to contrast the levels of 100 groups?
:PROPERTIES:
:ID:       355A5DC0-1A35-4315-8DEB-4C2275EA9A7C
:END:

******* Can I use DESeq2 to analyze a dataset without replicates?
:PROPERTIES:
:ID:       4AF6174D-0E98-4139-89B6-2E2AB1B13BA4
:END:

******* How can I include a continuous covariate in the design formula?
:PROPERTIES:
:ID:       BBFFFE10-B8CF-4E66-9BA9-B78B0DA2FAF9
:END:

******* Will the log fold change shrinkage “overshrink” large differences?
:PROPERTIES:
:ID:       01210C6E-221E-403D-AE9C-63946B83F9A4
:END:

******* I ran a likelihood ratio test, but results() only gives me one comparison
:PROPERTIES:
:ID:       76AC2A68-59B8-4E92-B384-822B8CE0DD7B
:END:

******* What are the exact steps performed by DESeq()?
:PROPERTIES:
:ID:       74C55B19-7DAA-4357-8F15-179AF2645CFF
:END:

******* Is there an official Galaxy tool for DESeq2
:PROPERTIES:
:ID:       5A4CB62B-A91F-467F-AA52-B8D1C66D9D28
:END:

******* I want to benchmark DESeq2 comparing to other DE tools.
:PROPERTIES:
:ID:       9B564504-8E91-4921-87D8-3AE89F5F6E2F
:END:

****** SessionInfo
:PROPERTIES:
:ID:       626CE66C-E3D7-4D09-BEDC-27812DB3689B
:END:

[[file:~/Box%20Sync/comp_bio/Love,14%20Beginners%20guide%20to%20using%20the%20DESeq2%20package.pdf][Love,14 Beginners guide to using the DESeq2 package]]
[[file:~/Box%20Sync/comp_bio/Love,15%20DESeq2%20User%20Guide.pdf][User Guide]]
[[https://support.bioconductor.org/local/search/page/?q=deseq2][Bioconductor Support]]
[[file:~/Box%20Sync/library/Love,16%20DESeq2%20manual.pdf][Manual]] 
A negative binomial GLM 
[[https://www.youtube.com/results?search_query=deseq2&page=&utm_source=opensearch][Youtube]]
[[file:~/Box%20Sync/library/Love,14%20Moderated%20estimation%20of%20fold%20change%20and%20dispersion%20for%20RNA%20seq%20data%20with%20DESeq2.pdf][Love,14 Moderated estimation of fold change and dispersion for RNA seq data with DESeq2]] 
[[ftp://ftp.jax.org/dgatti/ShortCourse2015/tutorials/Differential-Expression.html][Workflow example]]
[[http://www.sthda.com/english/wiki/rna-seq-differential-expression-work-flow-using-deseq2][Another workflow example]]
[[https://dwheelerau.com/2014/02/17/how-to-use-deseq2-to-analyse-rnaseq-data/][And another workflow example]]


*** edgeR
:PROPERTIES:
:ID:       32BB77F2-0283-4A52-86AC-CDF2CB6A7315
:END:
- EdgeR
  - GLM asking whether exons are used proportionally across experiment conditions
  - DEXSeq is an extension to exons
[[file:~/Box%20Sync/comp_bio/RNA-seq/Rueda,15%20Differential%20expression%20analysis%20using%20edgeR.pdf][Rueda,15 Differential expression analysis using edgeR]]
[[file:~/Box%20Sync/comp_bio/RNA-seq/edgeRUsersGuide.pdf][User Guide]]
[[file:~/Box%20Sync/comp_bio/RNA-seq/edgeR_manual.pdf][Manual]]

- cite:robinson2010
[[file:~/Box%20Sync/library/Robinson,09%20edgeR%20a%20Bioconductor%20package%20for%20differential%20expression%20analysis%20of%20digitial%20gene%20expression%20data.pdf][Robinson,09 edgeR a Bioconductor package for differential expression analysis of digitial gene expression data]]

**** edgeR full example
:PROPERTIES:
:ID:       846943FC-BA4B-4658-9B1E-85FBA536D9EE
:END:
*** exceRpt
:PROPERTIES:
:ID:       1db0ffb6-25b7-4c97-9aa0-263b8c96373b
:END:
- cite:rozowsky2019 
- http://github.gersteinlab.org/exceRpt/
- used in
  - cite:godoy2019 
  - cite:giraldez2018 
*** fastqc
:PROPERTIES:
:ID:       0a319950-3215-4846-a020-a08b78f3323d
:END:
- FastQC
  - [[https://www.youtube.com/watch?v=bz93ReOv87Y][Intro video]]
  - interpretation
    - Per sequence GC content- spikes mean contaminants
    - will only track first 200000
    - base quality
      - Phred vs. read position (goal is >20 or 1 error in 100 bases)
      - Phred histogram- look for single population or filter second population
      - Base position vs. frequency of nucleotide
    - per nucleotide    
      - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/5%20Per%20Sequence%20GC%20Content.html][Per sequence GC content]]
      - Secondary peak = contamination- see overrepresent sequences 
    - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html][Duplicate sequences]]
      - Broad oversequencing -> high flat line
      - Specific enrichments - spikes
        - Possible reasons
          - Technical (PCR) duplication
          - Biological duplication- common in RNA-seq where you are re-sequencing the common sequences 
      - Warning at 20%, failure @ 50%
    - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html][Kmer content]]
      - for partial sequences in a variety of longer sequences 
- https://sequencing.qcfail.com/software/fastqc/
- https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/#:~:text=The%20output%20from%20FastQC%2C%20after,or%20%E2%80%9CFail%E2%80%9D%20is%20assigned.
- cite:williams2014 for a FastQC aggregator Perl script 
- [[https://www.biostars.org/p/56088/#56102][Fastq Quality Read And Score Length Check]]

*** gage
:PROPERTIES:
:ID:       9b222d1c-ab61-450c-8b77-a98ec3213867
:END:
- https://www.bioconductor.org/packages/devel/data/experiment/manuals/gageData/man/gageData.pdf
- https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
- http://amp.pharm.mssm.edu/Enrichr/help#basics
*** gistic2 
:PROPERTIES:
:ID:       4f0b84ab-1dec-45c9-a7d4-bdce41e13150
:END:
https://www.genepattern.org/file-formats-guide

segmentation file column 6 is log2(x)-1 of the copy number, so
| interp  | copy number |   log2(x)-1 |
|---------+-------------+-------------|
| neutral |           2 |           0 |
| gain    |           3 |        0.58 |
| amp     |           4 |           1 |
| loss    |           1 |          -1 |
| del     |           0 | - inf -> NA |
|         |             |             |

- https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#gistic-2-0-data
- https://blindwith.science/2018/10/442.html/
- http://crazyhottommy.blogspot.com/2017/11/run-gistic2-with-sequenza-segmentation.html
*** GSEA
:PROPERTIES:
:ID:       D68E1182-1F7B-4160-A6DD-BD2635F8085F
:END:
:LOGBOOK:
CLOCK: [2020-11-17 Tue 09:56]--[2020-11-17 Tue 12:07] =>  2:11
CLOCK: [2020-10-26 Mon 13:01]--[2020-10-26 Mon 13:55] =>  0:54
CLOCK: [2020-10-26 Mon 12:09]--[2020-10-26 Mon 12:59] =>  0:50
:END:
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6607905/
- leading edge https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5066188/
- [[https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideTEXT.htm#_Consistent_Feature_Identifiers_Acro][documentation]]
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/FAQ#Why_does_GSEA_use_a_false_discovery_rate_.28FDR.29_of_0.25_rather_than_the_more_classic_0.05.3F][FDR discussion]]
  - FAQ
    - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/FAQ#What_array_platforms_and_organism_species_does_GSEA_support.3F][using for mouse data]]
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Using_RNA-seq_Datasets_with_GSEA][RNA-seq data]]
    - with RNA-seq, use TMM
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats][file formats]]
    - GMT: gene matrix transposed file (*.gmt)
    - chip files ftp://ftp.broadinstitute.org/pub/gsea/annotations (use ENSEMBL)
  - [[https://www.gsea-msigdb.org/gsea/msigdb/search.jsp][gene sets]]
- gene sets for mouse GSEA
  - https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html
  - recent, high impact papers citing MSigDB and aligning to mm10
    - https://science.sciencemag.org/content/352/6288/aad3018
    - http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC5924693&blobtype=pdf
    - https://www.cell.com/cell/pdf/S0092-8674(17)31385-5.pdf
    - https://stm.sciencemag.org/content/scitransmed/suppl/2018/01/12/10.424.eaan5488.DC1/aan5488_SM.pdf
- used in 
  - [[id:e4351ef3-6d41-4121-a2cc-c63d89f997ac][pathway analysis]]
  - [[id:91d46e60-eec3-48c7-97f1-b53eb1559b3f][RNA-seq coexpression]]
- processes
  - download data
  - file creation and formatting
    - see to create .rnk file https://github.com/BaderLab/EM-tutorials-docker/blob/master/R_scripts/supplemental_protocol2_rnaseq.R
- [[https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp][MSigDB]] [[https://data.broadinstitute.org/gsea-msigdb/msigdb/][data]]
- [[https://enrichmentmap.readthedocs.io/en/docs-2.2/QuickStartGuide.html][enrichement map]]
  - https://enrichmentmap.readthedocs.io/en/latest/Parameters.html
- platforms
  - online http://www.webgestalt.org/
  - R implimentation: [[https://bioconductor.org/packages/release/bioc/html/fgsea.html][fast GSEA]] 
    #+name:
    #+begin_src R :tangle no
  setwd("~/repos/biotools")
  source("./src/setup.R")
  #+end_src
- [[http://software.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection/4][ssGSEA]] calculates enrichment for each sample-gene set pair
- cite:mootha2003 
*** ichorcna
:PROPERTIES:
:ID:       8c535c38-9310-4fd0-a8fe-c300639493da
:END:
https://www.google.com/search?q=ichorcna&sxsrf=ALeKk02zeu_rSdEZVMYYCyip37Gt2XFzEg:1617895629440&source=lnms&tbm=vid&sa=X&ved=2ahUKEwjZx5ry-u7vAhVJbs0KHXyIAGkQ_AUoAnoECAEQBA&biw=1596&bih=789
- cite:guo2018
- cite:klega2018
- cite:manier2018 
- cite:mouliere2018
- cite:stover2018
- cite:adalsteinsson2017
- [[https://github.com/broadinstitute/ichorCNA/wiki][documentation]]
  - https://github.com/broadinstitute/ichorCNA/wiki/Parameter-tuning-and-settings
- [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGDDjrSfkvnNDnxMlVJNFgfnfk][email from:Sundby re:ichor method]]
- https://www.biostars.org/local/search/page/?q=ichorcna
- for install see [[file:/ssh:chaudhuri-roche:/home/jszymanski/repos/biotools/Dockerfile][biotools Dockerfile]]  
*** INTEGRATE
:PROPERTIES:
:ID:       7519d47f-18e8-4a28-be11-9d081b68b52d
:END:
  - cite:zhang2016
    - [[http://apps.webofknowledge.com/summary.do?product=WOS&parentProduct=WOS&search_mode=CitingArticles&qid=2&SID=6ERcSCYD7iSBCZMtAM7&&page=1&action=changePageSize&pageSize=50][WOS citing]]- no cfRNA yet 
  - integrate
    - cite:zhang2017c
    - cite:zhang2016
    - cite:zhang2016a
    - cite:zhang2016a
    - - cite:zhang2014a
    - https://github.com/ChrisMaherLab/INTEGRATE-Vis/blob/master/README.md
    for integrate, see - cite:li2020
  
*** limma
:PROPERTIES:
:ID:       3283b8b2-fc42-4a71-a8e9-7dbdd7ab62ce
:END:
https://www.coursera.org/lecture/bioconductor/limma-aoNwW
*** limma voom
:PROPERTIES:
:ID:       F9684168-A438-4F5E-81D6-D645B52B4074
:END:
  - https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/decideTests
  - https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/ebayes
  - https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/toptable

[[file:~/Box%20Sync/comp_bio/RNA-seq/Ritchie,15%20limma%20powers%20differential%20expression%20analysis%20for%20RNA%20sequencing%20and%20microarray%20results.pdf][Ritchie,15 limma powers differential expression analysis for RNA sequencing and microarray results]]
:LOGBOOK:  
CLOCK: [2016-05-09 Mon 12:24]--[2016-05-09 Mon 12:25] =>  0:01
:END:      
- raw counts transformed to continuous values w/ confidence weight
- [[file:~/Box%20Sync/comp_bio/RNA-seq/limma%20Manual.pdf][manual]]

- Math, concept- [[file:~/Box%20Sync/comp_bio/RNA-seq/Law,14%20voom%20precision%20weights%20unlock%20linear%20model%20analysis%20tools%20for%20RNA%20seq.pdf][Law,14 voom precision weights unlock linear model analysis tools for RNA seq]]
- voom transformation- from counts to log-counts per million w/ associated weight (just think of it as if microarray data)

**** [[file:~/Box%20Sync/comp_bio/RNA-seq/limma%20User%20Guide.pdf][User Guide]]
:PROPERTIES:
:ID:       15C1AE68-19D2-4FBD-B51E-DA37C605F0D7
:END:

- concepts
  - borrows information across genes for emperical Bayes
  - read counts -> continuous expression data like microarray
  - required matrices for linear models approach
    - design- which RNA to which sample (i.e. specify groups)
    - contrast- which group comparions to make

**** [[file:~/Box%20Sync/comp_bio/limma_voom_template/][Examples and Template]]
:PROPERTIES:
:ID:       D1B28C0A-9AAD-4245-BE85-61300CE4A02B
:END:

***** WEHI example ([[file:~/Box%20Sync/comp_bio/RNA-seq/wehi_limma_example/wehi_limma_example.Rproj][Rproject]], [[file:~/Box%20Sync/comp_bio/RNA-seq/wehi_limma_example/wehi_limma_example.R][Rscript]]), [[http://bioinf.wehi.edu.au/RNAseqCaseStudy/][link]]
:PROPERTIES:
:ID:       1E96E985-8BD2-4D8F-AC88-FB509B99F513
:END:
:LOGBOOK:  
CLOCK: [2016-05-10 Tue 15:46]--[2016-05-10 Tue 16:05] =>  0:19
CLOCK: [2016-05-10 Tue 15:07]--[2016-05-10 Tue 15:20] =>  0:13
CLOCK: [2016-05-09 Mon 12:25]--[2016-05-09 Mon 13:13] =>  0:48
:END:      

***** User Guide Chapter 18
:PROPERTIES:
:ID:       817998EF-31AD-4A22-BF91-ACF72A1C29E5
:END:

****** 18.1 [[file:~/Box%20Sync/comp_bio/limma_voom_template/yoruba_hapmap/][Profiles of Yoruba HapMap]]
:PROPERTIES:
:ID:       1E8CAB25-19C2-4745-B157-FA46D84549CD
:END:

******* DONE Read
CLOSED: [2016-06-30 Thu 11:22]
:PROPERTIES:
:ID:       91CBDFB8-17EB-4F2D-B2A4-58F656C64332
:END:

******* Work through code
:PROPERTIES:
:ID:       05EC5C9D-CE60-4DD5-9F5D-6EED8C229478
:END:

Venn diagrams with limma]]

Another use of the voom plot-  what you are failing to do completely in the voom analysis is to remove the rows of your count matrix that are lowly expressed, because these hose the mean/variance estimation, which you can see/diagnose on the plot generated with you call to voom(..., plot=TRUE). In brief, you want to ensure that the red fit line doesn't "slope/hook back down" (I'm talking as if you are "reading" the graph from right to left here) due to the drop/discretization in the observed variance from the black dots on the left side of the plot.

[[https://www.biostars.org/p/131488/][subsetting from topTable]]
*** multiqc
:PROPERTIES:
:ID:       3e585de3-3b67-4fe3-9f15-ed4e07778d09
:END:
  - - see for better multiqc handling- https://github.com/nextflow-io/rnaseq-nf/blob/3b5b49f/main.nf#L41-L48
*** nextflow
:PROPERTIES:
:CREATED:  [2019-12-27 Fri 14:48]
:ID:       6116d262-9ca8-430b-b385-65e1a7fc0a70
:END:
:LOGBOOK:
CLOCK: [2020-01-30 Thu 13:00]--[2020-01-30 Thu 14:04] =>  1:04
CLOCK: [2020-01-30 Thu 09:35]--[2020-01-30 Thu 12:45] =>  3:10
CLOCK: [2020-01-21 Tue 12:45]--[2020-01-21 Tue 14:30] =>  1:45
CLOCK: [2020-01-15 Wed 14:47]--[2020-01-15 Wed 15:39] =>  0:52
CLOCK: [2020-01-14 Tue 18:38]--[2020-01-14 Tue 19:04] =>  0:26
CLOCK: [2020-01-14 Tue 16:24]--[2020-01-14 Tue 16:44] =>  0:20
CLOCK: [2020-01-14 Tue 13:49]--[2020-01-14 Tue 15:52] =>  2:03
CLOCK: [2020-01-14 Tue 11:11]--[2020-01-14 Tue 12:42] =>  1:31
CLOCK: [2020-01-13 Mon 21:10]--[2020-01-13 Mon 22:00] =>  0:50
CLOCK: [2020-01-10 Fri 17:51]--[2020-01-10 Fri 18:51] =>  1:00
CLOCK: [2020-01-08 Wed 13:06]--[2020-01-08 Wed 14:12] =>  1:06
CLOCK: [2020-01-08 Wed 11:19]--[2020-01-08 Wed 12:31] =>  1:12
CLOCK: [2020-01-07 Tue 17:27]--[2020-01-07 Tue 18:27] =>  1:00
CLOCK: [2020-01-02 Thu 13:32]--[2020-01-02 Thu 14:32] =>  1:00
CLOCK: [2019-12-27 Fri 19:15]--[2019-12-27 Fri 21:20] =>  2:05
CLOCK: [2019-12-27 Fri 14:50]--[2019-12-27 Fri 15:12] =>  0:22
:END:
https://github.com/NYU-Molecular-Pathology
https://www.pnas.org/content/115/4/E743.long
- github searhing e.g. https://github.com/search?utf8=%E2%9C%93&q=followLinks++extension%3A.nf&type=Code&ref=advsearch&l=&l=
- see for inputs from csv- https://github.com/nf-core/atacseq/blob/master/main.nf
- raw data should reside in protected dirs
- https://www.reddit.com/search?q=nextflow
- cite federico2019
- useful tutorials
  - https://nextflow-io.github.io/hack17-varcall/#_input_parameters associated github at radonc-cancerbio:~/data/tmp
- https://sciwiki.fredhutch.org/compdemos/nextflow/
- https://learn.gencore.bio.nyu.edu/nextflow/
- https://www.ebi.ac.uk/goldman-srv/simNGS/runfiles5/HiSeq/
- https://github.com/genome/gms/wiki/HCC1395-WGS-Exome-RNA-Seq-Data
- http://nextflow-io.github.io/patterns/index.html#_process_per_file_pairs
- [[https://stackoverflow.com/questions/tagged/nextflow][nextflow stack questions]]
- https://github.com/nextflow-io/patterns
- [[https://github.com/nextflow-io/nf-hack18][2018 workshop github]]
- [[https://nextflow-io.github.io/nf-hack18/training.html][2018 workshop course]]
- [[https://github.com/search?utf8=%25E2%259C%2593&q=language%253Anextflow+&type=][github repos using nextflow]]
- https://github.com/cbcrg/piper-nf/blob/master/piper.nf
- [[https://github.com/search?utf8=%25E2%259C%2593&q=language%253Anextflow+&type=][github nextflow repositories]]
- https://groups.google.com/forum/m/#!forum/nextflow
- https://www.nextflow.io/presentations.html
- http://groovy-lang.org/single-page-documentation.html
- https://dzone.com/refcardz/groovy?chapter=10
- https://www.manning.com/books/groovy-in-action-second-edition
- https://github.com/MaxUlysse/compile-latex
- https://nextflow-io.github.io/nf-hack18/training.html
- https://github.com/nextflow-io/patterns
- https://nextflow-io.github.io/hack17-varcall/
- [[https://medium.com/@alaincoletta/how-to-create-a-nextflow-pipeline-with-all-necessary-tools-for-a-series-of-repetitive-35cf5b3b19a8][Coletta tutorial]]
- processes
  - check if file exists
- commands
  |----------+---------|
  | function | command |
  |----------+---------|
  | <35>     | <35>    |
  |          |         |
- standards and best practice
- [[https://www.nextflow.io/docs/latest/index.html][documentation]]
- applications
- reference
  - [[file:~/.emacs.d/snippets/org-mode/software]]

*** pathvisio
:PROPERTIES:
:ID:       17c865dc-765c-496d-b5c7-60d3d3439de3
:END:
https://pathvisio.github.io/
- to switch to java 8 https://askubuntu.com/questions/1133216/downgrading-java-11-to-java-8
*** picard
:PROPERTIES:
:ID:       6992d240-4089-49b0-8587-7b84202932eb
:END:
  - [[https://broadinstitute.github.io/picard/command-line-overview.html][Broad Picard Command Line Overview]]
  - [[https://www.broadinstitute.org/videos/broade-mapping-processing-and-duplicate-marking-picard-tools][Broad Processing and Duplicate mapping with Picard]]
  - [[https://broadinstitute.github.io/picard/explain-flags.html][picard explain sam flags]]
*** polyester
:PROPERTIES:
:ID:       065725b4-43d3-48d4-a4e2-3ca5c1e1e6fd
:END:
  - https://www.bioconductor.org/packages/release/bioc/vignettes/polyester/inst/doc/polyester.html

*** rsem
:PROPERTIES:
:ID:       85db5c34-b5fd-4b6c-bdb7-88263e553ba3
:END:
- cite:li2011
- https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem
- documentation
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323
  - https://github.com/ENCODE-DCC/long-rna-seq-pipeline/blob/master/DAC/STAR_RSEM.sh
  - http://deweylab.biostat.wisc.edu/rsem/rsem-calculate-expression.html
  - http://deweylab.biostat.wisc.edu/rsem/rsem-prepare-reference.html
  - https://deweylab.github.io/RSEM/rsem-calculate-expression.html
  - https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem
*** samtools
:PROPERTIES:
:ID:       0a3acec7-f4d1-4cd1-b3c8-7ec769715cc1
:END:
- [[http://biobits.org/samtools_primer.html][another Samtools tutorial]]
**** commands
:PROPERTIES:
:CREATED:  [2019-04-26 Fri 10:32]
:ID:       31C15E78-2EAE-4D54-94FF-91A5AF05190A
:END:

- commands
  - sequence inspection and quality
    | function                              | command                                       |
    |---------------------------------------+-----------------------------------------------|
    | <35>                                  | <35>                                          |
    | see     bam header                    | view -H                                       |
    | see bam header and top seq in sam/bam | samtools view -h <FILE.bam> \vert head -n 100 |
    | see ?                                 | idxstats                                      |
    |                                       |                                               |
  - sequence file manipulations
    |-------------------+---------------------------------------------------|
    | function          | command                                           |
    |-------------------+---------------------------------------------------|
    | check bam EOF     | samtools quickcheck <.bam>                        |
    | merge bams        | samtools merge all.bam *.bam                      |
    | random sample bam | samtools view -s 0.5 -b file.bam >random_half.bam |
    | split by chrom    | samtools view -b <FULL.bam> chr1 > chr1.bam       |
    | merge             | samtools merge <OUT.bam> <IN1.bam> <IN2.bam>      |
    | trim by quality   | -q INT                                            |
    | bam to sam        | samtools view -h -o out.sam in.bam                |
    | sam to bam        | samtools view -bS test.sam > test.bam             |
    | remove unmapped   | samtools view -h -F 4 file.bam                    |

*** samtools 
:PROPERTIES:
:ID:       e998389a-84e1-4d27-bb8e-a711350daa08
:END:
  - [[http://www.htslib.org/workflow/#mapping_to_cram][using cram]]
  - [[https://www.samformat.info/sam-format-header][header tool]]

*** snakemake
:PROPERTIES:
:ID:       a8b27140-4b9e-40c5-86ab-ee5afeae723f
:END:
https://github.com/EnvGen/snakemake-workflows
#+name: 
#+begin_src bash :tangle no
source ~/repos/biotools/src/functions.sh
docker_interactive
biotools

#
mkdir -p /tmp/snakemake-tutorial
cd /tmp/snakemake-tutorial
#
wget https://github.com/snakemake/snakemake-tutorial-data/archive/v5.24.1.tar.gz
#
tar --wildcards -xf v5.24.1.tar.gz --strip 1 "*/data" "*/environment.yaml"
ls

#
snakemake --help

snakemake -np mapped_reads/A.bam
#+end_src
- https://reproducibility.sschmeier.com/index.html
- https://snakemake.readthedocs.io/en/stable/executor_tutorial/google_lifesciences.html
- [[https://snakemake.readthedocs.io/en/stable/index.html][documentation]]
  - [[Snakemake Tutorial][Snakemake Tutorial]]
  #+name: snakemake_tutorial
#+begin_src bash :tangle no
sudo apt-get install singularity-container

cd ~/snakemake-tutorial/

conda activate snakemake-tutorial


snakemake -p --cores 2 sorted_reads/{A,B}.bam
#    --dry-run \

snakemake --dag calls/all.vcf | dot -Tsvg > dag.svg

expand("sorted_reads/{sample}.bam", sample=SAMPLES)

snakemake -p --cores 2 plots/quals.svg

snakemake --cores 2 -p --dag | dot -Tsvg > dag.svg

snakemake --cores 20 --forceall --summary

snakemake --cores 20 --forceall

conda install pygraphviz

snakemake --report report.html

open report.html

snakemake --cores 20 --forceall
#+end_src
  - https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-1-mapping-reads
- docker https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html?highlight=docker#running-jobs-in-containers
- https://snakemake.readthedocs.io/en/stable/index.html
*** STAR
:PROPERTIES:
:ID:       d1d5d554-57c4-4b9c-8385-2bb72369375e
:END:
- used in [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][alignment]]
- star-rsem pipeline
  - cite:djebali2017
*** tool collections
:PROPERTIES:
:ID:       7cce7be9-b3ac-4127-9a92-71559617588e
:END:
- [[https://www.ebi.ac.uk/services][EMBL-EBI Tools]]
*** tophat2
:PROPERTIES:
:ID:       db678422-a644-4298-b962-33657a5dee7e
:END:
notes- https://www.biostars.org/p/251018/
*** Trim Galore 
:PROPERTIES:
:ID:       3e4ca528-a604-4bfe-b235-2052e6d2d4a9
:END:
- 
  - wrapper for Cutadapt and FastQC
  - [[https://github.com/FelixKrueger/TrimGalore][github]]
    - [[https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md][user guide]]
  - on CHPC cluster - http://mgt2.chpc.wustl.edu/wiki119/index.php/Software
- https://github.com/FelixKrueger/TrimGalore

*** UCSC table browser
:PROPERTIES:
:ID:       be612cfb-2a75-4935-92fd-85620dcabf92
:END:
- ucsc
  - table browser
    - https://genomebrowser.wustl.edu/goldenPath/help/hgTextHelp.html#Introduction
  - https://www.biostars.org/p/19029/
  - http://genome.ucsc.edu/FAQ/FAQreleases.html
  - cite:karolchik2011
*** wgcna
:PROPERTIES:
:ID:       948289d8-3a77-471a-b068-0507285e0bac
:END:
:LOGBOOK:
CLOCK: [2020-07-26 Sun 09:40]--[2020-07-26 Sun 10:43] =>  1:03
:END:
- https://www.nature.com/articles/nature16965#Sec1
- [[https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/][Horvath tutorials]]
  #+name:tutorial1-input
  #+begin_src R 
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored. 
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir); 
# Load the package
library(WGCNA);
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
#Read in the female liver data set
femData = read.csv("LiverFemale3600.csv");
# Read in the male liver data set
maleData = read.csv("LiverMale3600.csv");
# Take a quick look at what is in the data sets (caution, longish output):
dim(femData)
names(femData)
dim(maleData)
names(maleData)


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# We work with two sets:
nSets = 2;
# For easier labeling of plots, create a vector holding descriptive names of the two sets.
setLabels = c("Female liver", "Male liver")
shortLabels = c("Female", "Male")
# Form multi-set expression data: columns starting from 9 contain actual expression data.
multiExpr = vector(mode = "list", length = nSets)

multiExpr[[1]] = list(data = as.data.frame(t(femData[-c(1:8)])));
names(multiExpr[[1]]$data) = femData$substanceBXH;
rownames(multiExpr[[1]]$data) = names(femData)[-c(1:8)];
multiExpr[[2]] = list(data = as.data.frame(t(maleData[-c(1:8)])));
names(multiExpr[[2]]$data) = maleData$substanceBXH;
rownames(multiExpr[[2]]$data) = names(maleData)[-c(1:8)];
# Check that the data has the correct format for many functions operating on multiple sets:
exprSize = checkSets(multiExpr)


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


# Check that all genes and samples have sufficiently low numbers of missing values.
gsg = goodSamplesGenesMS(multiExpr, verbose = 3);
gsg$allOK


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


if (!gsg$allOK)
{
  # Print information about the removed genes:
  if (sum(!gsg$goodGenes) > 0)
    printFlush(paste("Removing genes:", paste(names(multiExpr[[1]]$data)[!gsg$goodGenes], 
                                              collapse = ", ")))
  for (set in 1:exprSize$nSets)
  {
    if (sum(!gsg$goodSamples[[set]]))
      printFlush(paste("In set", setLabels[set], "removing samples",
                       paste(rownames(multiExpr[[set]]$data)[!gsg$goodSamples[[set]]], collapse = ", ")))
    # Remove the offending genes and samples
    multiExpr[[set]]$data = multiExpr[[set]]$data[gsg$goodSamples[[set]], gsg$goodGenes];
  }
  # Update exprSize
  exprSize = checkSets(multiExpr)
}


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


sampleTrees = list()
for (set in 1:nSets)
{
  sampleTrees[[set]] = hclust(dist(multiExpr[[set]]$data), method = "average")
}


#=====================================================================================
#
#  Code chunk 6
#
## #=====================================================================================


## pdf(file = "Plots/SampleClustering.pdf", width = 12, height = 12);
## par(mfrow=c(2,1))
## par(mar = c(0, 4, 2, 0))
## for (set in 1:nSets)
##   plot(sampleTrees[[set]], main = paste("Sample clustering on all genes in", setLabels[set]),
##        xlab="", sub="", cex = 0.7);
## dev.off();


#=====================================================================================
#
#  Code chunk 7
#
#=====================================================================================


# Choose the "base" cut height for the female data set
baseHeight = 16
# Adjust the cut height for the male data set for the number of samples
cutHeights = c(16, 16*exprSize$nSamples[2]/exprSize$nSamples[1]);
# Re-plot the dendrograms including the cut lines
## pdf(file = "Plots/SampleClustering.pdf", width = 12, height = 12);
## par(mfrow=c(2,1))
## par(mar = c(0, 4, 2, 0))
## for (set in 1:nSets)
## {
##   plot(sampleTrees[[set]], main = paste("Sample clustering on all genes in", setLabels[set]),
##        xlab="", sub="", cex = 0.7);
##   abline(h=cutHeights[set], col = "red");
## }
## dev.off();


#=====================================================================================
#
#  Code chunk 8
#
#=====================================================================================


for (set in 1:nSets)
{
  # Find clusters cut by the line
  labels = cutreeStatic(sampleTrees[[set]], cutHeight = cutHeights[set])
  # Keep the largest one (labeled by the number 1)
  keep = (labels==1)
  multiExpr[[set]]$data = multiExpr[[set]]$data[keep, ]
}
collectGarbage();
# Check the size of the leftover data
exprSize = checkSets(multiExpr)
exprSize


#=====================================================================================
#
#  Code chunk 9
#
#=====================================================================================


traitData = read.csv("ClinicalTraits.csv");
dim(traitData)
names(traitData)
# remove columns that hold information we do not need.
allTraits = traitData[, -c(31, 16)];
allTraits = allTraits[, c(2, 11:36) ];
# See how big the traits are and what are the trait and sample names
dim(allTraits)
names(allTraits)
allTraits$Mice
# Form a multi-set structure that will hold the clinical traits.
Traits = vector(mode="list", length = nSets);
for (set in 1:nSets)
{
  setSamples = rownames(multiExpr[[set]]$data);
  traitRows = match(setSamples, allTraits$Mice);
  Traits[[set]] = list(data = allTraits[traitRows, -1]);
  rownames(Traits[[set]]$data) = allTraits[traitRows, 1];
}
collectGarbage();
# Define data set dimensions
nGenes = exprSize$nGenes;
nSamples = exprSize$nSamples;


#=====================================================================================
#
#  Code chunk 10
#
#=====================================================================================


save(multiExpr, Traits, nGenes, nSamples, setLabels, shortLabels, exprSize, 
     file = "Consensus-dataInput.RData");


#+end_src

  #+name:tutorial2-auto-network-construct
  #+begin_src R 
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored. 
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir); 
# Load the WGCNA package
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Allow multi-threading within WGCNA. 
# Caution: skip this line if you run RStudio or other third-party R environments.
# See note above.
enableWGCNAThreads()
# Load the data saved in the first part
lnames = load(file = "Consensus-dataInput.RData");
# The variable lnames contains the names of loaded variables.
lnames
# Get the number of sets in the multiExpr structure.
nSets = checkSets(multiExpr)$nSets


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# Choose a set of soft-thresholding powers
powers = c(seq(4,10,by=1), seq(12,20, by=2));
# Initialize a list to hold the results of scale-free analysis
powerTables = vector(mode = "list", length = nSets);
# Call the network topology analysis function for each set in turn
for (set in 1:nSets)
  powerTables[[set]] = list(data = pickSoftThreshold(multiExpr[[set]]$data, powerVector=powers,
                                                     verbose = 2)[[2]]);
collectGarbage();
# Plot the results:
colors = c("black", "red")
# Will plot these columns of the returned scale free analysis tables
plotCols = c(2,5,6,7)
colNames = c("Scale Free Topology Model Fit", "Mean connectivity", "Median connectivity",
"Max connectivity");
# Get the minima and maxima of the plotted points
ylim = matrix(NA, nrow = 2, ncol = 4);
for (set in 1:nSets)
{
  for (col in 1:length(plotCols))
  {
    ylim[1, col] = min(ylim[1, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
    ylim[2, col] = max(ylim[2, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
  }
}
# Plot the quantities in the chosen columns vs. the soft thresholding power
## sizeGrWindow(8, 6)
## pdf(file = "Plots/scaleFreeAnalysis.pdf", wi = 8, he = 6)
## par(mfcol = c(2,2));
## par(mar = c(4.2, 4.2 , 2.2, 0.5))
## cex1 = 0.7;
## for (col in 1:length(plotCols)) for (set in 1:nSets)
## {
##   if (set==1)
##   {
##     plot(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
##          xlab="Soft Threshold (power)",ylab=colNames[col],type="n", ylim = ylim[, col],
##          main = colNames[col]);
##     addGrid();
##   }
##   if (col==1)
##   {
##     text(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
##          labels=powers,cex=cex1,col=colors[set]);
##   } else
##     text(powerTables[[set]]$data[,1], powerTables[[set]]$data[,plotCols[col]],
##          labels=powers,cex=cex1,col=colors[set]);
##   if (col==1)
##   {
##     legend("bottomright", legend = setLabels, col = colors, pch = 20) ;
##   } else
##     legend("topright", legend = setLabels, col = colors, pch = 20) ;
## }
## dev.off();


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


net = blockwiseConsensusModules(
        multiExpr, power = 6, minModuleSize = 30, deepSplit = 2,
        pamRespectsDendro = FALSE, 
        mergeCutHeight = 0.25, numericLabels = TRUE,
        minKMEtoStay = 0,
        saveTOMs = TRUE, verbose = 5)


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


consMEs = net$multiMEs;
moduleLabels = net$colors;
# Convert the numeric labels to color labels
moduleColors = labels2colors(moduleLabels)
consTree = net$dendrograms[[1]]; 


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


## sizeGrWindow(8,6);
## pdf(file = "Plots/ConsensusDendrogram-auto.pdf", wi = 8, he = 6)
## plotDendroAndColors(consTree, moduleColors,
##                     "Module colors",
##                     dendroLabels = FALSE, hang = 0.03,
##                     addGuide = TRUE, guideHang = 0.05,
##                     main = "Consensus gene dendrogram and module colors")

## dev.off()


#=====================================================================================
#
#  Code chunk 6
#
#=====================================================================================


save(consMEs, moduleLabels, moduleColors, consTree, file = "Consensus-NetworkConstruction-auto.RData")


#+end_src

  #+name:tutorial3-relate-clinical-data
  #+begin_src R 
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored. 
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir); 
# Load the WGCNA package
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Load the expression and trait data saved in the first part
lnames = load(file = "FemaleLiver-01-dataInput.RData");
#The variable lnames contains the names of loaded variables.
lnames
# Load network data saved in the second part.
lnames = load(file = "FemaleLiver-02-networkConstruction-auto.RData");
lnames


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, datTraits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix =  paste(signif(moduleTraitCor, 2), "\n(",
                           signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
               xLabels = names(datTraits),
               yLabels = names(MEs),
               ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = greenWhiteRed(50),
               textMatrix = textMatrix,
               setStdMargins = FALSE,
               cex.text = 0.5,
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


# Define variable weight containing the weight column of datTrait
weight = as.data.frame(datTraits$weight_g);
names(weight) = "weight"
# names (colors) of the modules
modNames = substring(names(MEs), 3)

geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));

names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");

geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = "p"));
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));

names(geneTraitSignificance) = paste("GS.", names(weight), sep="");
names(GSPvalue) = paste("p.GS.", names(weight), sep="");


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


module = "brown"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)


#=====================================================================================
#
#  Code chunk 6
#
#=====================================================================================


names(datExpr)


#=====================================================================================
#
#  Code chunk 7
#
#=====================================================================================


names(datExpr)[moduleColors=="brown"]


#=====================================================================================
#
#  Code chunk 8
#
#=====================================================================================


annot = read.csv(file = "GeneAnnotation.csv");
dim(annot)
names(annot)
probes = names(datExpr)
probes2annot = match(probes, annot$substanceBXH)
# The following is the number or probes without annotation:
sum(is.na(probes2annot))
# Should return 0.


#=====================================================================================
#
#  Code chunk 9
#
#=====================================================================================


# Create the starting data frame
geneInfo0 = data.frame(substanceBXH = probes,
                      geneSymbol = annot$gene_symbol[probes2annot],
                      LocusLinkID = annot$LocusLinkID[probes2annot],
                      moduleColor = moduleColors,
                      geneTraitSignificance,
                      GSPvalue)
# Order modules by their significance for weight
modOrder = order(-abs(cor(MEs, weight, use = "p")));
# Add module membership information in the chosen order
for (mod in 1:ncol(geneModuleMembership))
{
  oldNames = names(geneInfo0)
  geneInfo0 = data.frame(geneInfo0, geneModuleMembership[, modOrder[mod]], 
                         MMPvalue[, modOrder[mod]]);
  names(geneInfo0) = c(oldNames, paste("MM.", modNames[modOrder[mod]], sep=""),
                       paste("p.MM.", modNames[modOrder[mod]], sep=""))
}
# Order the genes in the geneInfo variable first by module color, then by geneTraitSignificance
geneOrder = order(geneInfo0$moduleColor, -abs(geneInfo0$GS.weight));
geneInfo = geneInfo0[geneOrder, ]


#=====================================================================================
#
#  Code chunk 10
#
#=====================================================================================


write.csv(geneInfo, file = "geneInfo.csv")


#+end_src
- https://rstudio-pubs-static.s3.amazonaws.com/90587_39301ea4b9514d20880b4c9cda81ca4f.html
- https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/
- https://pdfs.semanticscholar.org/7a4a/c56e7d90b48ffda2ff267f9a84072940726b.pdf?_ga=2.194670125.1809056038.1595426853-2099730291.1595426853
- [[https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/][author tutorials]]
- [[https://www.biostars.org/local/search/page/?q=wgcna][wgcna biostars]]
- oldcode
  #+name: 
  #+begin_src bash 
wget \
    -P /tmp/ \
    -c \
    - N \
    https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip
unzip /tmp/FemaleLiver-Data.zip 

wget \
    -P /tmp/ \
    -c \
    - N \
    https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/MaleLiver-Data.zip
unzip /tmp/MaleLiver-Data.zip
#+end_src

  #+begin_src R 
# Tutorial 1
#
setwd("/tmp")
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
#Read in the female liver data set
femData = read.csv("LiverFemale3600.csv");
# Read in the male liver data set
maleData = read.csv("LiverMale3600.csv");
# Take a quick look at what is in the data sets (caution, longish output):
dim(femData)
names(femData)
dim(maleData)
names(maleData)


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# We work with two sets:
nSets = 2;
# For easier labeling of plots, create a vector holding descriptive names of the two sets.
setLabels = c("Female liver", "Male liver")
shortLabels = c("Female", "Male")
# Form multi-set expression data: columns starting from 9 contain actual expression data.
multiExpr = vector(mode = "list", length = nSets)

multiExpr[[1]] = list(data = as.data.frame(t(femData[-c(1:8)])));
names(multiExpr[[1]]$data) = femData$substanceBXH;
rownames(multiExpr[[1]]$data) = names(femData)[-c(1:8)];
multiExpr[[2]] = list(data = as.data.frame(t(maleData[-c(1:8)])));
names(multiExpr[[2]]$data) = maleData$substanceBXH;
rownames(multiExpr[[2]]$data) = names(maleData)[-c(1:8)];
# Check that the data has the correct format for many functions operating on multiple sets:
exprSize = checkSets(multiExpr)


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


# Check that all genes and samples have sufficiently low numbers of missing values.
gsg = goodSamplesGenesMS(multiExpr, verbose = 3);
gsg$allOK


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


if (!gsg$allOK)
{
  # Print information about the removed genes:
  if (sum(!gsg$goodGenes) > 0)
    printFlush(paste("Removing genes:", paste(names(multiExpr[[1]]$data)[!gsg$goodGenes], 
                                              collapse = ", ")))
  for (set in 1:exprSize$nSets)
  {
    if (sum(!gsg$goodSamples[[set]]))
      printFlush(paste("In set", setLabels[set], "removing samples",
                       paste(rownames(multiExpr[[set]]$data)[!gsg$goodSamples[[set]]], collapse = ", ")))
    # Remove the offending genes and samples
    multiExpr[[set]]$data = multiExpr[[set]]$data[gsg$goodSamples[[set]], gsg$goodGenes];
  }
  # Update exprSize
  exprSize = checkSets(multiExpr)
}


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


sampleTrees = list()
for (set in 1:nSets)
{
  sampleTrees[[set]] = hclust(dist(multiExpr[[set]]$data), method = "average")
}


#=====================================================================================
#
#  Code chunk 6
#
#=====================================================================================


## pdf(file = "Plots/SampleClustering.pdf", width = 12, height = 12);
## par(mfrow=c(2,1))
## par(mar = c(0, 4, 2, 0))
## for (set in 1:nSets)
##   plot(sampleTrees[[set]], main = paste("Sample clustering on all genes in", setLabels[set]),
##        xlab="", sub="", cex = 0.7);
## dev.off();


#=====================================================================================
#
#  Code chunk 7
#
#=====================================================================================


# Choose the "base" cut height for the female data set
baseHeight = 16
# Adjust the cut height for the male data set for the number of samples
cutHeights = c(16, 16*exprSize$nSamples[2]/exprSize$nSamples[1]);
# Re-plot the dendrograms including the cut lines
pdf(file = "Plots/SampleClustering.pdf", width = 12, height = 12);
par(mfrow=c(2,1))
par(mar = c(0, 4, 2, 0))
for (set in 1:nSets)
{
  plot(sampleTrees[[set]], main = paste("Sample clustering on all genes in", setLabels[set]),
       xlab="", sub="", cex = 0.7);
  abline(h=cutHeights[set], col = "red");
}
dev.off();


#=====================================================================================
#
#  Code chunk 8
#
#=====================================================================================


for (set in 1:nSets)
{
  # Find clusters cut by the line
  labels = cutreeStatic(sampleTrees[[set]], cutHeight = cutHeights[set])
  # Keep the largest one (labeled by the number 1)
  keep = (labels==1)
  multiExpr[[set]]$data = multiExpr[[set]]$data[keep, ]
}
collectGarbage();
# Check the size of the leftover data
exprSize = checkSets(multiExpr)
exprSize


#=====================================================================================
#
#  Code chunk 9
#
#=====================================================================================


traitData = read.csv("ClinicalTraits.csv");
dim(traitData)
names(traitData)
# remove columns that hold information we do not need.
allTraits = traitData[, -c(31, 16)];
allTraits = allTraits[, c(2, 11:36) ];
# See how big the traits are and what are the trait and sample names
dim(allTraits)
names(allTraits)
allTraits$Mice
# Form a multi-set structure that will hold the clinical traits.
Traits = vector(mode="list", length = nSets);
for (set in 1:nSets)
{
  setSamples = rownames(multiExpr[[set]]$data);
  traitRows = match(setSamples, allTraits$Mice);
  Traits[[set]] = list(data = allTraits[traitRows, -1]);
  rownames(Traits[[set]]$data) = allTraits[traitRows, 1];
}
collectGarbage();
# Define data set dimensions
nGenes = exprSize$nGenes;
nSamples = exprSize$nSamples;


#=====================================================================================
#
#  Code chunk 10
#
#=====================================================================================


save(multiExpr, Traits, nGenes, nSamples, setLabels, shortLabels, exprSize, 
     file = "Consensus-dataInput.RData");

femData = read.csv("./LiverFemale3600.csv", stringsAsFactors = F)
datExpr0 = as.data.frame(t(femData[, -c(1:8)]))
names(datExpr0) = femData$substanceBXH
rownames(datExpr0) = names(femData)[-c(1:8)]
#
gsg=goodSamplesGenes(datExpr0)
if (!gsg$allOK)
{
# Optionally, print the gene and sample names that were removed:
if (sum(!gsg$goodGenes)>0)
printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
if (sum(!gsg$goodSamples)>0)
printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
# Remove the offending genes and samples from the data:
datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
}
sampleTree = hclust(dist(datExpr0), method = "average");
clust = cutreeStatic(sampleTree, cutHeight = 15, minSize = 10)
table(clust)
# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
datExpr = datExpr0[keepSamples, ]
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
#
traitData = read.csv("./ClinicalTraits.csv")
allTraits = traitData[,-c(31,16)]
allTraits = allTraits[, c(2,11:36)]
#
femaleSamples = rownames(datExpr)
traitRows= match(femaleSamples, allTraits$Mice)
datTraits = allTraits[traitRows, -1]
rownames(datTraits) = allTraits[traitRows, 1]
collectGarbage()
#
#
save(datExpr, datTraits, file = "FemaleLiver-01-dataInput.RData")
#
traitData = read.csv("ClinicalTraits.csv");
dim(traitData)
names(traitData)
# remove columns that hold information we do not need.
allTraits = traitData[, -c(31, 16)];
allTraits = allTraits[, c(2, 11:36) ];
# See how big the traits are and what are the trait and sample names
dim(allTraits)
names(allTraits)
allTraits$Mice
# Form a multi-set structure that will hold the clinical traits.
Traits = vector(mode="list", length = nSets);
for (set in 1:nSets)
{
setSamples = rownames(multiExpr[[set]]$data);
traitRows = match(setSamples, allTraits$Mice);
Traits[[set]] = list(data = allTraits[traitRows, -1]);
rownames(Traits[[set]]$data) = allTraits[traitRows, 1];
}
collectGarbage();
# Define data set dimensions
nGenes = exprSize$nGenes;
nSamples = exprSize$nSamples;
save(multiExpr, Traits, nGenes, nSamples, setLabels, shortLabels, exprSize,
file = "Consensus-dataInput.RData");
# Tutorial 2
options(stringsAsFactors = F)
enableWGCNAThreads()
lnames=load(file="./FemaleLiver-01-dataInput.RData")
lnames
#
(powers=c(c(1:10), seq(from = 12, to = 20, by = 2)))
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
#
net = blockwiseModules(datExpr, power = 6,
TOMType = "unsigned", minModuleSize = 30,
reassignThreshold = 0, mergeCutHeight = 0.25,
numericLabels = TRUE, pamRespectsDendro = FALSE,
saveTOMs = TRUE,
saveTOMFileBase = "femaleMouseTOM",
verbose = 3)
#
# open a graphics window
sizeGrWindow(12, 9)
# Convert labels to colors for plotting
mergedColors = labels2colors(net$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(net$dendrograms[[1]], mergedColors[net$blockGenes[[1]]],
"Module colors",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)
#
moduleLabels = net$colors
moduleColors = labels2colors(net$colors)
MEs = net$MEs;
geneTree = net$dendrograms[[1]];
save(MEs, moduleLabels, moduleColors, geneTree,
file = "FemaleLiver-02-networkConstruction-auto.RData")
#
consMEs = net$multiMEs
moduleLabels = net$colors
moduleColors = labels2colors(moduleLabels)
consTree = net$dendrograms[[1]]
save(consMEs, moduleLabels, moduleColors, consTree, file = "Consensus-NetworkConstruction-auto.RData")

#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored. 
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir); 
# Load the WGCNA package
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Allow multi-threading within WGCNA. 
# Caution: skip this line if you run RStudio or other third-party R environments.
# See note above.
enableWGCNAThreads()
# Load the data saved in the first part
lnames = load(file = "Consensus-dataInput.RData");
#The variable lnames contains the names of loaded variables.
lnames
# Get the number of sets in the multiExpr structure.
nSets = checkSets(multiExpr)$nSets


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# Choose a set of soft-thresholding powers
powers = c(seq(4,10,by=1), seq(12,20, by=2));
# Initialize a list to hold the results of scale-free analysis
powerTables = vector(mode = "list", length = nSets);
# Call the network topology analysis function for each set in turn
for (set in 1:nSets)
  powerTables[[set]] = list(data = pickSoftThreshold(multiExpr[[set]]$data, powerVector=powers,
                                                     verbose = 2)[[2]]);
collectGarbage();
# Plot the results:
colors = c("black", "red")
# Will plot these columns of the returned scale free analysis tables
plotCols = c(2,5,6,7)
colNames = c("Scale Free Topology Model Fit", "Mean connectivity", "Median connectivity",
"Max connectivity");
# Get the minima and maxima of the plotted points
ylim = matrix(NA, nrow = 2, ncol = 4);
for (set in 1:nSets)
{
  for (col in 1:length(plotCols))
  {
    ylim[1, col] = min(ylim[1, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
    ylim[2, col] = max(ylim[2, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
  }
}
# Plot the quantities in the chosen columns vs. the soft thresholding power
sizeGrWindow(8, 6)
par(mfcol = c(2,2));
par(mar = c(4.2, 4.2 , 2.2, 0.5))
cex1 = 0.7;
for (col in 1:length(plotCols)) for (set in 1:nSets)
{
  if (set==1)
  {
    plot(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
         xlab="Soft Threshold (power)",ylab=colNames[col],type="n", ylim = ylim[, col],
         main = colNames[col]);
    addGrid();
  }
  if (col==1)
  {
    text(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
         labels=powers,cex=cex1,col=colors[set]);
  } else
    text(powerTables[[set]]$data[,1], powerTables[[set]]$data[,plotCols[col]],
         labels=powers,cex=cex1,col=colors[set]);
  if (col==1)
  {
    legend("bottomright", legend = setLabels, col = colors, pch = 20) ;
  } else
    legend("topright", legend = setLabels, col = colors, pch = 20) ;
}


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


softPower = 6;
# Initialize an appropriate array to hold the adjacencies
adjacencies = array(0, dim = c(nSets, nGenes, nGenes));
# Calculate adjacencies in each individual data set
for (set in 1:nSets)
  adjacencies[set, , ] = abs(cor(multiExpr[[set]]$data, use = "p"))^softPower;


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


# Initialize an appropriate array to hold the TOMs
TOM = array(0, dim = c(nSets, nGenes, nGenes));
# Calculate TOMs in each individual data set
for (set in 1:nSets)
  TOM[set, , ] = TOMsimilarity(adjacencies[set, , ]);


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


# Define the reference percentile
scaleP = 0.95
# Set RNG seed for reproducibility of sampling
set.seed(12345)
# Sample sufficiently large number of TOM entries
nSamples = as.integer(1/(1-scaleP) * 1000);
# Choose the sampled TOM entries
scaleSample = sample(nGenes*(nGenes-1)/2, size = nSamples)
TOMScalingSamples = list();
# These are TOM values at reference percentile
scaleQuant = rep(1, nSets)
# Scaling powers to equalize reference TOM values
scalePowers = rep(1, nSets)
# Loop over sets
for (set in 1:nSets)
{
  # Select the sampled TOM entries
  TOMScalingSamples[[set]] = as.dist(TOM[set, , ])[scaleSample]
  # Calculate the 95th percentile
  scaleQuant[set] = quantile(TOMScalingSamples[[set]],
                             probs = scaleP, type = 8);
  # Scale the male TOM
  if (set>1)
  {
    scalePowers[set] = log(scaleQuant[1])/log(scaleQuant[set]);
    TOM[set, ,] = TOM[set, ,]^scalePowers[set];
  }
}


#=====================================================================================
#
#  Code chunk 6
#
#=====================================================================================


# For plotting, also scale the sampled TOM entries
scaledTOMSamples = list();
for (set in 1:nSets)
  scaledTOMSamples[[set]] = TOMScalingSamples[[set]]^scalePowers[set]
# Open a suitably sized graphics window
sizeGrWindow(6,6)
pdf(file = "Plots/TOMScaling-QQPlot.pdf", wi = 6, he = 6);
# qq plot of the unscaled samples
qqUnscaled = qqplot(TOMScalingSamples[[1]], TOMScalingSamples[[2]], plot.it = TRUE, cex = 0.6,
                    xlab = paste("TOM in", setLabels[1]), ylab = paste("TOM in", setLabels[2]),
                    main = "Q-Q plot of TOM", pch = 20)
# qq plot of the scaled samples
qqScaled = qqplot(scaledTOMSamples[[1]], scaledTOMSamples[[2]], plot.it = FALSE)
points(qqScaled$x, qqScaled$y, col = "red", cex = 0.6, pch = 20);
abline(a=0, b=1, col = "blue")
legend("topleft", legend = c("Unscaled TOM", "Scaled TOM"), pch = 20, col = c("black", "red"))
dev.off();


#=====================================================================================
#
#  Code chunk 7
#
#=====================================================================================


consensusTOM = pmin(TOM[1, , ], TOM[2, , ]);


#=====================================================================================
#
#  Code chunk 8
#
#=====================================================================================


# Clustering
consTree = hclust(as.dist(1-consensusTOM), method = "average");
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 30;
# Module identification using dynamic tree cut:
unmergedLabels = cutreeDynamic(dendro = consTree, distM = 1-consensusTOM,
                deepSplit = 2, cutHeight = 0.995,
                minClusterSize = minModuleSize,
                pamRespectsDendro = FALSE );
unmergedColors = labels2colors(unmergedLabels)


#=====================================================================================
#
#  Code chunk 9
#
#=====================================================================================


sizeGrWindow(8,6)
plotDendroAndColors(consTree, unmergedColors, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)


#=====================================================================================
#
#  Code chunk 10
#
#=====================================================================================


# Calculate module eigengenes
unmergedMEs = multiSetMEs(multiExpr, colors = NULL, universalColors = unmergedColors)
# Calculate consensus dissimilarity of consensus module eigengenes
consMEDiss = consensusMEDissimilarity(unmergedMEs);
# Cluster consensus modules
consMETree = hclust(as.dist(consMEDiss), method = "average");
# Plot the result
sizeGrWindow(7,6)
par(mfrow = c(1,1))
plot(consMETree, main = "Consensus clustering of consensus module eigengenes",
     xlab = "", sub = "")
abline(h=0.25, col = "red")


#=====================================================================================
#
#  Code chunk 11
#
#=====================================================================================


merge = mergeCloseModules(multiExpr, unmergedLabels, cutHeight = 0.25, verbose = 3)


#=====================================================================================
#
#  Code chunk 12
#
#=====================================================================================


# Numeric module labels
moduleLabels = merge$colors;
# Convert labels to colors
moduleColors = labels2colors(moduleLabels)
# Eigengenes of the new merged modules:
consMEs = merge$newMEs;


#=====================================================================================
#
#  Code chunk 13
#
#=====================================================================================


sizeGrWindow(9,6)
plotDendroAndColors(consTree, cbind(unmergedColors, moduleColors),
                    c("Unmerged", "Merged"),
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)


#=====================================================================================
#
#  Code chunk 14
#
#=====================================================================================


save(consMEs, moduleColors, moduleLabels, consTree, file = "Consensus-NetworkConstruction-man.RData")


#
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored. 
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir); 
# Load the WGCNA package
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Load the data saved in the first part
lnames = load(file = "Consensus-dataInput.RData");
#The variable lnames contains the names of loaded variables.
lnames
# Load the results of network analysis, tutorial part 2.a
lnames = load(file = "Consensus-NetworkConstruction-auto.RData");
lnames


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


lnames = load("../Mouse-Female/FemaleLiver-02-networkConstruction-auto.RData")
lnames
# Rename variables to avoid conflicts
femaleLabels = moduleLabels;
femaleColors = moduleColors;
femaleTree = geneTree;
femaleMEs = orderMEs(MEs, greyName = "ME0");


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


lnames = load("Consensus-NetworkConstruction-auto.RData")
lnames


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


# Isolate the module labels in the order they appear in ordered module eigengenes
femModuleLabels = substring(names(femaleMEs), 3)
consModuleLabels = substring(names(consMEs[[1]]$data), 3)
# Convert the numeric module labels to color labels
femModules = labels2colors(as.numeric(femModuleLabels))
consModules = labels2colors(as.numeric(consModuleLabels))
# Numbers of female and consensus modules
nFemMods = length(femModules)
nConsMods = length(consModules)
# Initialize tables of p-values and of the corresponding counts
pTable = matrix(0, nrow = nFemMods, ncol = nConsMods);
CountTbl = matrix(0, nrow = nFemMods, ncol = nConsMods);
# Execute all pairwaise comparisons
for (fmod in 1:nFemMods)
  for (cmod in 1:nConsMods)
  {
    femMembers = (femaleColors == femModules[fmod]);
    consMembers = (moduleColors == consModules[cmod]);
    pTable[fmod, cmod] = -log10(fisher.test(femMembers, consMembers, alternative = "greater")$p.value);
    CountTbl[fmod, cmod] = sum(femaleColors == femModules[fmod] & moduleColors ==
                      consModules[cmod])
  }


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


# Truncate p values smaller than 10^{-50} to 10^{-50} 
pTable[is.infinite(pTable)] = 1.3*max(pTable[is.finite(pTable)]);
pTable[pTable>50 ] = 50 ;
# Marginal counts (really module sizes)
femModTotals = apply(CountTbl, 1, sum)
consModTotals = apply(CountTbl, 2, sum)
# Actual plotting
sizeGrWindow(10,7 );
pdf(file = "Plots/ConsensusVsFemaleModules.pdf", wi = 10, he = 7);
par(mfrow=c(1,1));
par(cex = 1.0);
par(mar=c(8, 10.4, 2.7, 1)+0.3);
# Use function labeledHeatmap to produce the color-coded table with all the trimmings
labeledHeatmap(Matrix = pTable,
               xLabels = paste(" ", consModules),
               yLabels = paste(" ", femModules),
               colorLabels = TRUE,
               xSymbols = paste("Cons ", consModules, ": ", consModTotals, sep=""),
               ySymbols = paste("Fem ", femModules, ": ", femModTotals, sep=""),
               textMatrix = CountTbl,
               colors = greenWhiteRed(100)[50:100],
               main = "Correspondence of Female set-specific and Female-Male consensus modules",
               cex.text = 1.0, cex.lab = 1.0, setStdMargins = FALSE);
dev.off();


# Tutorial 3: relating consensus modules to set-specific modules
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Load the data saved in the first part
#lnames = load(file = "Consensus-dataInput.RData");
#The variable lnames contains the names of loaded variables.
#lnames
# Load the results of network analysis, tutorial part 2.a
lnames = load(file = "Consensus-NetworkConstruction-auto.RData");
lnames
femaleLabels=moduleLabels
femaleColors = moduleColors
femaleTree = geneTree
femaleMEs = orderMEs(MEs, greyName = "MEO")
#
# Isolate the module labels in the order they appear in ordered module eigengenes
femModuleLabels = substring(names(femaleMEs), 3)
consModuleLabels = substring(names(consMEs[[1]]$data), 3)
# Convert the numeric module labels to color labels
femModules = labels2colors(as.numeric(femModuleLabels))
consModules = labels2colors(as.numeric(consModuleLabels))
# Numbers of female and consensus modules
nFemMods = length(femModules)
nConsMods = length(consModules)
# Initialize tables of p-values and of the corresponding counts
pTable = matrix(0, nrow = nFemMods, ncol = nConsMods);
CountTbl = matrix(0, nrow = nFemMods, ncol = nConsMods);
# Execute all pairwaise comparisons
for (fmod in 1:nFemMods)
for (cmod in 1:nConsMods)
{
femMembers = (femaleColors == femModules[fmod]);
consMembers = (moduleColors == consModules[cmod]);
pTable[fmod, cmod] = -log10(fisher.test(femMembers, consMembers, alternative = "greater")$p.value);
CountTbl[fmod, cmod] = sum(femaleColors == femModules[fmod] & moduleColors ==
consModules[cmod])
}
#+end_src


** Ideas
- Data dictionary 
  #+name: biotools_docker_versions
  | cutadapt_version |    1.18 |
  | mirdeep2_version | 2.0.1.2 |
  | bowtie           |     1.2 |
  #+name: injest_biotools_docker_versions
  #+begin_src python :var biotools_docker_versions=biotools_docker_versions
  import os
  import csv

  os.chdir('/home/jeszyman/repos/biotools')

  with open('./data/biotools_docker_versions.csv', 'w', newline='') as file:
      writer = csv.writer(file, delimiter=',')
      writer.writerows(biotools_docker_versions)
  #+end_src
  #+name: injest_biotools_docker_versions_wrapper
  #+begin_src python :noweb yes
  injest_biotools_docker_versions
  #+end_src
- Ideas
   - make test data
    #+name: make_test_data
    #+begin_src bash :tangle ~/repos/mpnst/bin/make_test_data.sh
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  ### Make small test data files suitable for github push  ###
  #
  # Setup 
  ##
  ## Local
  source ~/repos/mpnst/bin/local-setup.sh
  ##
  ## Docker
  docker_interactive
  biotools
  ##
  ## Functions
  ##
  fastq_test(){
      # Check for parameters, return usage 
      if [[ $# -eq 0 ]] ; then
          printf "\n fastq_test 
                \n Generates fastq files and .gzs of desired file size
                \n \$1 = input (in fastq.gz)
                \n \$2 = FILESIZE (in byte)
                \n \$3 = output directory
                \n \$4 = output filename (END .fastq)
                \n "
      else
          # In-function variable setup
          infile=$1
          filesize=$2
          outdir=$3
          base=`basename -s .fastq.gz $infile`
          # Out file name defaults to <in file name>_test
          outfilename=${4:-${base}_test.fastq}
          readsize=$(zcat $infile | head -n 4 | wc -c)
          lines=$(( 4 * ($filesize / $readsize) ))
          zcat $infile | head -n "$lines" > ${outdir}/$outfilename
          # Get exact same lines for R2
          infile_R2=$(echo "$1" | sed 's/_R1/_R2/g') 
          outfilename_R2=$(echo "$outfilename" | sed 's/_R1/_R2/g') 
          zcat $infile_R2 | head -n "$lines" > ${outdir}/$outfilename_R2
          # Get gzipped files
          gzip -f -k ${outdir}/$outfilename
          gzip -f -k ${outdir}/$outfilename_R2        
      fi
  }
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  #
  rm -rf $repo/data/test
  mkdir -p $repo/data/test
  #
  # WashU WGS fastqs
  ##
  ## cfDNA
  ###
  ### mpnst 
  fastq_test $localdata/.washu-fastq-extract/new_HiSeqW72_Undetermined_R6000482_L001_R1_TCCT_extract.fastq.gz \
             1000000 \
             $repo/data/test \
             washu_mpnst_R1.fastq
  ### healthy
  ###
  fastq_test $localdata/.washu-fastq-extract/new_HiSeqW31_Undetermined_R6000232_L005_R1_001_GAAG_extract.fastq.gz \
             1000000 \
             $repo/data/test \
             washu_healthy_R1.fastq
  ## NCI
  ###
  fastq_test $localdata/fastqs/lib107_R1.fastq.gz \
             1000000 \
             $repo/data/test \
             nci_mpnst_R1.fastq
  ###
  fastq_test $localdata/fastqs/lib088_R1.fastq.gz \
             1000000 \
             $repo/data/test \
             nci_healthy_R1.fastq
  #
  # cappseq demultiplexed but UN-extracted files: 
  ###
  fastq_test $localdata/.washu-demulti-tmp/new_HiSeqW43_Undetermined_R6000292_L004_R1_001_TCCT.fastq.gz \
             1000000 \
             $repo/data/test \
             demulti_TCCT_R1.fastq
  #+end_src
** Reference 
:PROPERTIES:
:CREATED:  [2020-04-23 Thu 08:54]
:ID:       98f8c0e9-cdfc-4acb-b98c-5db1bd6bc3ca
:END:
- https://pages.10xgenomics.com/ugm-2020-10-event-ra_g-user-group-meeting-amr_on-demand.html?src=email&utm_medium=email&lss=internal&utm_source=internal&cnm=ugm-2020-10-event-ra_g-user-group-meeting-amr&utm_campaign=ugm-2020-10-event-ra_g-user-group-meeting-amr&useroffertype=event&userresearcharea=ra_g&userregion=multi&mktouserid=1629122


- http://rafalab.github.io/pages/harvardx.html
- https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga/publications

- https://scilifelab.github.io/courses/
- https://www.reddit.com/r/bioinformatics/comments/m78kqb/what_are_the_best_book_i_could_get_for_a/

- https://www.reddit.com/r/bioinformatics/comments/m2o63g/best_courses_or_material_to_learn_bioinformatics/

- https://learn.gencore.bio.nyu.edu/
- cite:buffalo2015
- biotools bash functions in ./src
  - docker interactive call
    #+name: docker_interactive 
    #+begin_src bash :tangle ./src/bash_functions.sh
#########1#########2#########3#########4#########5#########6#########7#########8
# Interactive docker initialization
docker_interactive() {
    repo="USER INPUT"
    read -p "repo name: " repo
    case $HOSTNAME in
        (radonc-cancerbio) docker run -it \
                                  --env HOME=${HOME} \
                                  --hostname ${HOSTNAME} \
                                  --user $(id -u ${USER}) \
                                  --volume /home/:/home/ \
                                  --volume /mnt/:/mnt/ \
                                  --volume /tmp/:/tmp/ \
                                  jeszyman/$repo \
                                  /bin/bash;;
        (jeszyman-*) docker run -it \
                            --env HOME=/home/jeszyman \
                            --hostname ${HOSTNAME} \
                            --user $(id -u ${USER}) \
                            --volume /home/:/home/ \
                            --volume /mnt/:/mnt/ \
                            --volume /tmp/:/tmp/ \
                            jeszyman/$repo \
                            /bin/bash;;
        (acl*) docker run -it \
                      -v /drive3/:/drive3/ \
                      -v /duo4/:/duo4/ \
                      -v /home/:/home/ \
                      -u $(id -u ${USER}) \
                      jeszyman/$repo \
                      /bin/bash;;
        (ACL*) docker run -it \
                      -v /home/:/home/ \
                      -v /duo4/:/duo4/ \
                      -u $(id -u ${USER}):$(id -g ${USER}) \
                      -h=${HOSTNAME} \
                      jeszyman/$repo \
                      /bin/bash;;
        (virtual-workstation*.gsc.wustl.edu) bsub -Is -q docker-interactive -a 'docker(jeszyman/'"$repo"')' /bin/bash;;
    esac
}
#
#+end_src
- https://www.reddit.com/r/bioinformatics/comments/lfusag/skills_required_for_being_a_proficient/
- cite:peters2018
- https://github.com/jdidion/biotools
- https://support.bioconductor.org/#search-anchor
- https://informatics.fas.harvard.edu/category/tutorials.html
- https://www.youtube.com/channel/UCv4IbnP9j9RC_aZAs8wqdeQ
- https://www.broadinstitute.org/genome-sequencing/broadillumina-genome-analyzer-boot-camp
- https://cloud.genepattern.org/gp/pages/login.jsf
- https://github.com/genome/analysis-workflows
- https://github.com/genome/bfx-workshop
https://github.com/seanken
http://genomewiki.ucsc.edu/index.php/Main_Page
https://genome.ucsc.edu/FAQ/index.html
  - https://basespace.illumina.com/dashboard
    - - dragon aligner 
  - https://www.reddit.com/r/bioinformatics/comments/5wo75h/ben_ward_julia_for_bioinformatics_and/
  - https://platform.dnanexus.com/login
  - https://www.ecseq.com/
    - https://www.ecseq.com/workshops/ngs-data-analysis-courses
  - repos
    - https://github.com/Illumina
    - https://github.com/chrisamiller  
  - https://www.reddit.com/r/bioinformatics/comments/cxlsh5/outdated_software_still_in_use/
  - https://github.com/pditommaso/awesome-pipeline
  - https://coding4medicine.com/cheatsheets/
  - https://github.com/genome/bfx-workshop
  - examples
    - papers
      - https://www.sciencedirect.com/science/article/pii/S009286741830309X#sec4
      - good github repos
        - https://github.com/ding-labrecip
        - https://github.com/pstew
  - https://www.rockefeller.edu/genomics/

  - https://www.reddit.com/r/bioinformatics/comments/iv7xjj/biologist_with_minimal_coding_background/
  - old docker setup code
  #########1#########2#########3#########4#########5#########6#########7#########8
  # DEV
  # check if inside docker
  # if [ -f /.dockerenv ];then 
  #     dockercontainer=yes
  #     echo "Your are in a docker container"    
  # else 
  #     dockercontainer=no
  #     echo "Not in docker"
  #     # setup NON-docker variables 
  #     shopt -s nocasematch
  #     case $HOSTNAME in
  #         (radonc-cancerbio) server_home="/home/jeszyman" ;;
  #         (jeszyman-*) server_home="/home/jeszyman" ;;
  #         (acl*) server_home="/home/jszymanski" ;; 
  #         (virtual-workstation*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
  #         (blade*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
  #     esac
  #     
  #     localdata="$server_home/data/$project/"
  #     unset $SERVER
  # fi
  #
  # case $dockercontainer in
  #     (yes) server_ask ;;
  #     (no) ;;
  # esac
  # #
  # case $SERVER in
  #     (chaudhuri-roche) projectdir=/drive3/users/jszymanski/repos/$project
  #                       localdata=/drive3/users/jszymanski/data/$project;;
  #     (NA) ;;
  # esac
  # #
  # # test data on cancerbio
  #NOT GENERAL
  #basedir=/media/jeszyman/4edbbc27-7e29-4667-9b24-f5a56d49d94e/users/jeszyman/data/mpnst
  #rm -rf $basedir/fastq-test
  #mkdir -p $basedir/fastq-test
  #zcat $basedir/fastq-nci/MPNST/14_JS0108CD040119_R1.fastq.gz | head -n 8000 > $basedir/fastq-test/short_14_JS0108CD040119_R1.fastq
  #zcat $basedir/fastq-nci/MPNST/14_JS0108CD040119_R2.fastq.gz | head -n 8000 > $basedir/fastq-test/short_14_JS0108CD040119_R2.fastq        
  #zcat $basedir/fastq-washu/new_HiSeq-20_L007001_CTTC_R1.fastq.gz | head -n 8000 > $basedir/fastq-test/short_new_HiSeq-20_L007001_CTTC_R1.fastq.gz
  #gzip $basedir/fastq-test/*.fastq
  # setup project repo
- see gnu parallel in [[id:cd560390-aa1e-4c9e-8317-08610e5de42a][linux]]
- https://www.reddit.com/r/bioinformatics/comments/191ykr/resources_for_learning_bioinformatics/
*** education, tutorials, books, etc.
:PROPERTIES:
:CREATED:  [2019-04-26 Fri 10:36]
:ID:       374F30CA-2E80-4CC2-8FD3-5B9BC8083889
:END:
- https://www.reddit.com/r/bioinformatics/comments/esr5mw/any_good_books_about_bioinformatics/
- https://protocolexchange.researchsquare.com/browse?journal=protocol-exchange&subjectArea=Computational%20biology%20and%20bioinformatics

- https://www.melbournebioinformatics.org.au/tutorials/
- https://www.coursera.org/specializations/bioinformatics
- training, tutorials, etc.
  - https://bootcamp.biostars.io/
- https://jef.works/
- https://www.coursera.org/course/bioinformatics
- https://www.ebi.ac.uk/training/online/
- http://egtheory.wordpress.com/2013/06/09/toward-an-algorithmic-theory-of-biology/
- http://rosalind.info/problems/as-graph/
- https://www.nature.com/collections/vzsqzylnvx
- foundations
  - https://www.reddit.com/r/bioinformatics/comments/dckuuk/what_are_some_mathematically_rigorous_intro_books/
- https://wikis.utexas.edu/collector/pages.action?key=bioiteam
- https://www.sh-eahp.org/images/2017_Workshop/1_10ESHAP_2017.pdf
- https://www.reddit.com/r/bioinformatics/comments/bxgp0h/what_do_you_wish_someone_had_just_shown_you_early/
- https://girke.bioinformatics.ucr.edu/GEN242/mydoc_systemPipeRNAseq_06.html
- https://www.illumina.com/company/video-hub.html
- organizations and communities
  - http://www.scilifelab.se/
- [[id:2274F8F7-45D0-4F2F-BA32-4AF38C8A0F80][Molecular Pathology Compendium]]
- [[id:BDC42DA2-A632-4FB7-A586-E100B2A045DC][UNIX]]
- [[id:022FDD07-245F-4231-905E-ED18BE224889][Molecular Biology]]
- organizations
  - [[id:543E3788-28CC-4D16-943B-BC17523D66D9][McDonnell Genome Institute (MGI)]]
  - Broad
    - [[http://software.broadinstitute.org/software/cprg/?q=node/27][Broad ABSOLUTE]]
    - [[https://software.broadinstitute.org/gatk/documentation/topic?name=methods][Broad methods]]
  - [[http://www.sanger.ac.uk/science/tools][Wellcome Sanger]]
- education, tutorials, etc. 
  - https://www.edx.org/learn/bioinformatics
  - https://www.coursera.org/learn/bioinformatics
  - http://bioinformaticsalgorithms.com/
  - https://www.reddit.com/r/bioinformatics/comments/ag7lc8/the_bible_of_bioinformatics/
  - https://bioinformatics.ca/workshops/workshops-2017/
  - [[http://bioinformatics.ucdavis.edu/training/documentation/][UC Davis Bioinformatics Core Workshops]]
  - http://rosalind.info/problems/locations/
  - https://www.reddit.com/r/bioinformatics/comments/6405tf/what_are_some_good_small_bioinformatics_projects/
  - [[http://userweb.eng.gla.ac.uk/umer.ijaz/bioinformatics/linux.html][Linux command line exercises for NGS data processing]]
  - https://biostar.myshopify.com/
  - https://www.reddit.com/r/bioinformatics/comments/7bntke/book_recommendations_to_be_a_better/
  - [[http://area51.stackexchange.com/proposals/109245/bioinformatics][StackExchange Bioinformatics]]
  - Well done sequencing papers http://www.nejm.org/doi/full/10.1056/NEJMoa1502214?rss=mostViewed&
  - http://seqanswers.com/
  - Training
    - https://collections.plos.org/compbiol-education
    - - https://docs.google.com/spreadsheets/d/1hIw66KCfdB0rowN7tJhyVjJv09Qv87r2B0sWLinVeS8/htmlview
    - https://www.broadinstitute.org/partnerships/education/broade/best-practices-variant-calling-gatk-1
    - - https://www.ebi.ac.uk/training/online/
    - - https://www.reddit.com/r/bioinformatics/comments/5miwnr/short_list_of_bioinformatics_skills/
    - https://bioinformatics.ca/workshops/2016 
    - http://rosalind.info/problems/locations/
    - http://griffithlab.org/teaching/
    - http://evomics.org/workshops/2017-workshop-on-transcriptomics-harvard-university/
    - [[https://github.com/nekrut/BMMB554/wiki/7.-NGS-Processing:-QC,-Mapping,-etc][BMMB554 Introduction to data driven life sciences]].
  - https://satijalab.org/scgd/
  - https://www.genomeweb.com/resources/webinars
  - books
    - cite:low2017 
    - cite:sinha2014

**** Notes
:PROPERTIES:
:ID:       90C0571F-2BB4-484B-8C2E-2385AD22A38C
:END:

[[http://onlinelibrary.wiley.com/book/10.1002/0471250953][Current Protocols in Bioinformatics]]

because computation removes redundancy, you will more frequently be at a new, tricky step

**** glossary
:PROPERTIES:
:CREATED:  [2019-05-14 Tue 10:25]
:ID:       80B5A347-33C6-4FC2-A691-FFAFF3C5D352
:END:

- https://software.broadinstitute.org/gatk/documentation/topic?name=dictionary&v=3
- [[http://useast.ensembl.org/info/genome/genebuild/biotypes.html][ensembl biotypes]]
- [[https://useast.ensembl.org/Help/Glossary][Ensembl glossary]]
- [[http://seqanswers.com/forums/showthread.php?t=13098][insert size]]: fragment size + length of both adapters 
- [[http://seqanswers.com/forums/showthread.php?t=13098][fragment size]]: sequence between the paired-end adapters

** Local Variables                                                 :noexport:
:PROPERTIES:
:ID:       0fa73acf-0c14-4f80-8338-192a6007b668
:END:
#+startup: overview
#+property: header-args    :results silent
#+property: header-args+    :cache yes
#+property: header-args+    :eval never-export
#+property: header-args+    :exports code
#+property: header-args+    :tangle no
#+property: header-args:R    :session *mpnst*
#+startup: overview
#+todo: WAITING TODO INPROCESS | DONE
# Local Variables:
# truncate-lines: nil
# End:
