* Bioinformatics                                                     :career:
:PROPERTIES:
:CREATED:  [2020-07-02 Thu 13:14]
:ID:       b03025b7-4b60-4196-8d7f-d947c7d64b28
:header-args:bash: :tangle-mode (identity #o555)
:END:
** Biotools conda yaml
:PROPERTIES:
:ID:       cf13d347-1d0e-4858-a016-0a8a8af22609
:END:
#+begin_src bash :tangle ./resources/conda/biotools_env.yaml
channels:
  - conda-forge
  - bioconda

dependencies:
  - bedtools
  - samtools
  - picard
  - sra-tools
  - pysam
  - fastp
  - fastqc
  - multiqc

#+end_src
#+begin_src bash
source /opt/miniconda3/bin/activate base

mamba env update --name biotools --file ~/repos/biotools/resources/conda/biotools_env.yaml

conda list --name biotools > ~/repos/biotools/resources/full_conda_env.txt
conda list --name cardiac -e > ~/repos/biotools/resources/machine_read_full_conda_env.txt

conda activate biotools

prefetch
#+end_src

** Processes                                                      :hierarchy:
*** 0 Bioinformatics process flat list                                 :flat:
**** Send-out sequencing services
- https://www.reddit.com/r/bioinformatics/comments/13py5j8/folks_who_send_their_samples_out_for_sequencing/
- [[file:~/repos/org/science.org::#e9f3ca18-6066-414e-92a6-268ee5858407][GTAC]]
**** Sequencing depth
https://www.google.com/search?q=%22samtools+depth%22+%22wgs%22&oq=%22samtools+depth%22+%22wgs%22&aqs=chrome..69i57.25710j0j4&sourceid=chrome-mobile&ie=UTF-8

- samtools for depth https://www.biostars.org/p/356937/
**** Sequence reverse compliment in shell
https://bioinformatics.stackexchange.com/questions/7458/what-is-a-quick-way-to-find-the-reverse-complement-in-bash
**** ctDNA burden by cfDNA degragation
https://www.nature.com/articles/s41467-021-22463-y#Sec2
**** Adaptive TCR-beta immunosequencing
:PROPERTIES:
:CREATED:  [2020-12-20 Sun 16:45]
:ID:       61aac189-1770-4d80-a367-e7c39b46e0d1
:END:
**** Adaptive TCR-beta immunosequencing
:PROPERTIES:
:CREATED:  [2020-12-20 Sun 16:45]
:ID:       61aac189-1770-4d80-a367-e7c39b46e0d1
:END:

**** [#Y] Assess library complexity
[[id:6d8cbcc0-2651-4c51-991e-639dbc81ea66][preseq]]


**** ATAC-seq peak calling
:PROPERTIES:
:ID:       1c56c6ad-e0f8-47db-8d69-3782b7ed1e8b
:END:
- https://www.encodeproject.org/software/idr/
- Devon recommending against MACS2 https://www.biostars.org/p/447927/
- https://bioinformatics-core-shared-training.github.io/cruk-autumn-school-2017/ChIP/Materials/Lectures/Lecture5_Peak%20Calling_SS.pdf
- Used in [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
- homer
  - some basic homer commands
    To identify Tn5 hypersensitive sites (THSs), peak calling was performed by findPeak in HOMER software (v4.9) with “-localSize 50000 -size 100 -minDist 50 -fragLength 0 -o auto.” ΔTHS score was calculated with all pairwise comparison of ESC, hCO and hMGEO by dnase_ddhs_scorer.py script with “-A” option in pyDNase library (v0.2.4) (Figure S2)
- [[id:496e62cd-3138-4120-a499-a10246ce77b5][Published MACS2 commands for ATAC-seq]]
**** ATAC-seq QC
:PROPERTIES:
:ID:       fe58011b-ca46-4e45-9c4e-d4780fbfa115
:END:
- [X] 80% unique mapping, 50 mil reads per cite:yan2020atac
- https://docs.google.com/document/d/1f0Cm4vRyDQDu0bMehHD7P7KOMxTOP-HiNoIvL1VcBt8/edit
- https://yiweiniu.github.io/blog/2019/03/Calculate-FRiP-score/
- cite:gontarz2020  https://github.com/Zhang-lab
- atacqv https://github.com/ParkerLab/ataqv/
- mononucleosomal peaks 147 +/- 30 bp
- Get input cell number (can work as low as 500 cells) cite:minnoye2021
- [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
- cite:alvarez_benayas2021
- consider batch correction using limma, see- cite:alvarez_benayas2021
- "A typical way to visually assess the effectiveness of normalization is through an MA plot: a single cloud should be present and the MA distribution should not display an upward or downward shift" cite:minnoye2021
- [[id:a5b73c70-3f1f-4018-92b2-bfdcaed53cf5][ATACseqQC]]
- QC
  - https://www.cell.com/cell-systems/pdfExtended/S2405-4712(20)30079-X
  - https://bioconductor.org/packages/release/bioc/vignettes/ATACseqQC/inst/doc/ATACseqQC.html
  - https://bioconductor.org/packages/release/bioc/html/soGGi.html
**** Batch effect
- batch effect https://evayiwenwang.github.io/Managing_batch_effects/detect.html
https://www.reddit.com/r/bioinformatics/comments/s3twgh/working_with_batch_effects/
**** Bioinformatics project management
- https://www.reddit.com/r/bioinformatics/comments/18alwyn/best_practices_for_putting_a_project_on_github/
- https://www.reddit.com/r/bioinformatics/comments/zxbeqw/how_can_i_have_an_organized_team/
**** Biotools docker containers
- ...rebuild local sing and add checker
- [ ] sing builds to box
- [ ] move all docker to single repo
***** Dockerfiles
- https://github.com/orgs/genome/repositories?type=all
****** Methylation docker
:PROPERTIES:
:header-args:bash: :tangle ./docker/meth_Dockerfile
:END:
:LOGBOOK:
CLOCK: [2022-07-24 Sun 09:05]--[2022-07-24 Sun 09:55] =>  0:50
:END:
******* Preamble
#+begin_src bash
FROM jeszyman/biotools:1.0.2
# Dockerfile to build a bioinformatics methylation toolkit

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/meth
# (https://hub.docker.com/repository/docker/jeszyman/meth)
#+end_src
******* Conda
#+begin_src bash
#################
###   Conda   ###
#################

RUN conda update conda
RUN conda install -c bioconda metilene=0.2.8
#+end_src
******* Custom
#+begin_src bash
###########################
###   Custom Installs   ###
###########################

# Biscuit
RUN curl -OL $(curl -s https://api.github.com/repos/huishenlab/biscuit/releases/latest | grep browser_download_url | grep linux_amd64 | cut -d '"' -f 4)
RUN mv biscuit_* biscuit
RUN chmod +x biscuit
RUN mv /biscuit /bin/
RUN curl -OL $(curl -s https://api.github.com/repos/huishenlab/biscuit/releases/latest | grep browser_download_url | grep build_biscuit_QC_assets.pl | cut -d '"' -f 4)
RUN chmod +x build_biscuit_QC_assets.pl
RUN curl -OL $(curl -s https://api.github.com/repos/huishenlab/biscuit/releases/latest | grep browser_download_url | grep QC.sh | cut -d '"' -f 4)
RUN chmod +x QC.sh
#+end_src

****** [[file:docker/biotools_Dockerfile][Biotools docker]]
:PROPERTIES:
:ID:       7ed344ce-373c-4f9e-b24f-87b57374e1a0
:header-args:bash: :tangle ./docker/biotools_Dockerfile
:END:
:LOGBOOK:
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:19] =>  1:00
CLOCK: [2022-01-04 Tue 09:15]--[2022-01-04 Tue 09:16] =>  0:01
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:15] =>  0:56
CLOCK: [2021-03-03 Wed 19:56]--[2021-03-03 Wed 21:25] =>  1:29
:END:
******** Preamble
#+begin_src bash
FROM jeszyman/basecamp
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build a basic bioinformatics docker image          ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/biotools
# (https://hub.docker.com/repository/docker/jeszyman/atac)

#+end_src
******** Conda
#+begin_src bash

#################
###   Conda   ###
#################

RUN conda install -c bioconda bbmap \
    && conda install -c bioconda bcftools \
    && conda install -c bioconda bedops \
    && conda install -c bioconda bedtools \
    && conda install -c bioconda bowtie \
    && conda install -c bioconda bowtie2 \
    && conda install -c bioconda bwa \
    && conda install -c bioconda cutadapt \
    && conda install -c bioconda deeptools=3.5.1 --force \
    && conda install -c bioconda fastp \
    && conda install -c bioconda fastqc \
    && conda install -c bioconda gatk \
    && conda install -c bioconda goleft \
    && conda install -c bioconda htseq \
    && conda install -c bioconda mosdepth \
    && conda install -c bioconda preseq \
    && conda install -c bioconda pysam \
    && conda install -c bioconda qualimap \
    && conda install -c bioconda sambamba \
    && conda install -c bioconda samblaster \
    && conda install -c bioconda samtools=1.7 \
    && conda install -c bioconda seqkit \
    && conda install -c bioconda seqtk \
    && conda install -c bioconda skewer \
    && conda install -c bioconda snakemake \
    && conda install -c bioconda trimmomatic=0.39 \
    && conda install -c bioconda vcftools
#+end_src
******** R and Bioconductor
#+begin_src bash
# To avoid conflict with jeszyman/basecamp R install
#RUN conda uninstall r-base

##########################
###   R Bioconductor   ###
##########################

RUN R -e "install.packages('BiocManager', repos='http://cran.rstudio.com/')"

RUN R -e "BiocManager::install(version='3.15', update=TRUE, ask=FALSE)"

RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("AnnotationDbi",\
                                 "apeglm"
                                 "BiocParallel", \
                                 "BSgenome.Hsapiens.UCSC.hg19", \
                                 "ComplexHeatmap", \
                                 "EnsDb.Mmusculus.v79", \
                                 "org.Mm.eg.db", \
                                 "readr", \
                                 "Rsamtools", \
                                 "TxDb.Mmusculus.UCSC.mm10.ensGene"));'

#+end_src
******** UCSC
#+begin_src bash

##############################################
###   Select UCSC Command-line utilities   ###
##############################################

RUN mkdir /opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedClip --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCommonRegions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCoverage --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedExtendRanges --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGeneParts --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphPack --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedIntersect --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedItemOverlapCount --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset.py --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedMergeAdjacent --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPartition --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPileUps --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRemoveOverlap --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRestrictToPositions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedSort --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToExons --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigAverageOverBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCat --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCluster --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCorrelate --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigInfo --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigMerge --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/#+end_src --directory-prefix=/opt/ucsc \
    && chmod -R +x /opt/ucsc/

ENV PATH="/opt/ucsc:${PATH}"

#+end_src
******** Custom and manual installs
#+begin_src bash

###########################
###   Custom Installs   ###
###########################

# Multiqc

RUN pip install multiqc

# Flexbar
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated libtbb2
RUN wget https://github.com/seqan/flexbar/releases/download/v3.5.0/flexbar-3.5.0-linux.tar.gz && tar -xzf flexbar-3.5.0-linux.tar.gz && mv flexbar-3.5.0-linux/flexbar /usr/bin/ && mv flexbar-3.5.0-linux/libtbb.so.2 /usr/bin/

# samtools
# https://www.biostars.org/p/9480029/
RUN conda install -c bioconda samtools openssl=1.0

#+end_src
********* Picard
#+begin_src bash

##################
###   Picard   ###
##################

RUN apt-get install -yq --no-install-recommends --allow-unauthenticated openjdk-8-jdk

#########1#########2#########3#########4#########5#########6#########7#########8
#https://github.com/genome/docker-picard-cwl/commit/639ad8e4d8e7bd7e6e10f193df8d5cd16260d2e2
RUN mkdir /opt/picard-2.18.1/ \
    && cd /tmp/ \
    && wget --no-check-certificate https://github.com/broadinstitute/picard/releases/download/2.18.1/picard.jar \
    && mv picard.jar /opt/picard-2.18.1/ \
    && ln -s /opt/picard-2.18.1 /opt/picard \
    && ln -s /opt/picard-2.18.1 /usr/picard
#########1#########2#########3#########4#########5#########6#########7#########8


#+end_src
******** Development                                                    :dev:
:PROPERTIES:
:header-args: :tangle no
:END:
- How to install multiple samtools and call 1.9 vs 1.7?
- give specific versions for all condas
- Old code
    #
  # cnvkit
  RUN pip install -U cython
  #FAILS [2021-02-17] RUN pip install -U future futures pandas pomegranate pyfaidx
  #FAILS [2021-02-17] RUN pip install cnvkit==0.9.6
  #
  # RSeQC
  RUN pip install RSeQC
  #
  #
  RUN cd /opt && \
      git clone https://github.com/shahcompbio/hmmcopy_utils.git && \
      cd hmmcopy_utils && \
      cmake . && \
      make
  ##NEED edger, limma, gage, dseq2, wgcna
  #
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("DNAcopy");'
  #

  #########1#########2#########3#########4#########5#########6#########7######
  #TESTING
  #
  #RUN conda install -c bioconda tophat
  #RUN conda install -c bioconda cnvkit
  #RUN conda install -c bioconda manta
  #RUN conda install -c bioconda lumpy-sv
  #RUN conda install -c bioconda multiqc

  ###8
  # #
  # ### LUMPY
  # # from https://raw.githubusercontent.com/zlskidmore/docker-lumpy/master/Dockerfile
  # # RUN apt-get update -qq
  # # RUN apt-get install -qq --no-install-recommends \
  # # python-pip \
  # # git \
  # # cmake \
  # # build-essential \
  # # libz-dev
  # # RUN cd /opt && git clone https://github.com/hall-lab/lumpy-sv.git && cd /opt/lumpy-sv && make
  # #########1#########2#########3#########4#########5#########6#########7#########8
  # ### Samtools
  # RUN apt-get update -qq
  # RUN apt-get install -qq --no-install-recommends \
  # wget \
  # bzip2 \
  # cmake \
  # gcc \
  # zlib1g-dev \
  # libncurses5-dev
  # ENV SAMTOOLS_INSTALL_DIR=/opt/samtools
  # WORKDIR /tmp
  # RUN wget --no-check-certificate https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2 && \
  # tar --bzip2 -xf samtools-1.9.tar.bz2 && \
  # cd /tmp/samtools-1.9 && \
  # ./configure --prefix=$SAMTOOLS_INSTALL_DIR && \
  # make && \
  # make install && \
  # cd / && \
  # rm -rf /tmp/samtools-1.9 && \
  # ln -s /opt/samtools/bin/* /usr/bin/

  # # # WORKDIR /usr/local/bin/
  # # # RUN curl -SL https://github.com/samtools/samtools/releases/download/${samtools_version}/samtools-${samtools_version}.tar.bz2 \
  # # #     > /usr/local/bin/samtools-${samtools_version}.tar.bz2
  # # # RUN tar -xjf /usr/local/bin/samtools-${samtools_version}.tar.bz2 -C /usr/local/bin/
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && ./configure
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && make
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && make install

  # # # # install lumpy
  # # # WORKDIR /usr/local/bin
  # # # RUN wget https://github.com/arq5x/lumpy-sv/releases/download/${lumpy_version}/lumpy-sv.tar.gz
  # # # RUN tar -xzvf lumpy-sv.tar.gz
  # # # WORKDIR /usr/local/bin/lumpy-sv
  # # # RUN make
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpy /usr/local/bin/lumpy
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpy_filter /usr/local/bin/lumpy_filter
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpyexpress /usr/local/bin/lumpyexpress
  # # # https://github.com/hall-lab/sv-pipeline/blob/master/docker/lumpy/Dockerfile
  # # # Build dependencies
  # # RUN apt-get update -qq \
  # #     && apt-get -y install \
  # #         apt-transport-https \
  # #         g++ \
  # # 	gawk \
  # #         libcurl4-gnutls-dev \
  # #         autoconf \
  # # 	libssl-dev \
  # #         git

  # #########1#########2#########3#########4#########5#########6#########7#########8
  # # ###############
  # # #bam-readcount#
  # # ###############
  # # # ENV SAMTOOLS_ROOT=/opt/samtools
  # # # RUN apt-get update && apt-get install -y --no-install-recommends \
  # # #         cmake \
  # # #         patch && \
  # # #     mkdir /opt/bam-readcount && \
  # # #     cd /opt/bam-readcount && \
  # # #     git clone https://github.com/genome/bam-readcount.git /tmp/bam-readcount-0.7.4 && \
  # # #     git -C /tmp/bam-readcount-0.7.4 checkout v0.7.4 && \
  # # #     cmake /tmp/bam-readcount-0.7.4 && \
  # # #     make && \
  # # #     rm -rf /tmp/bam-readcount-0.7.4 && \
  # # #     ln -s /opt/bam-readcount/bin/bam-readcount /usr/bin/bam-readcount

  # # # #note - this script needs cyvcf - installed in the python secetion!
  # # # COPY bam_readcount_helper.py /usr/bin/bam_readcount_helper.py
  # # # #############
  # # # ## IGV 3.0 ##

  # # # RUN apt-get update && apt-get install -y --no-install-recommends \
  # # #     software-properties-common \
  # # #     glib-networking-common && \
  # # #     mkdir -p /igv && \
  # # #     cd /igv && \
  # # #     wget http://data.broadinstitute.org/igv/projects/downloads/3.0_beta/IGV_3.0_beta.zip && \
  # # #     unzip IGV_3.0_beta.zip && \
  # # #     cd IGV_3.0_beta && \
  # # #     sed -i 's/Xmx4000/Xmx8000/g' igv.sh && \
  # # #     cd /usr/bin && \
  # # #     ln -s /igv/IGV_3.0_beta/igv.sh ./igv

  # # ##############
  # # ## bedtools ##

  # # WORKDIR /usr/local
  # # RUN git clone https://github.com/arq5x/bedtools2.git && \
  # #     cd /usr/local/bedtools2 && \
  # #     git checkout v2.25.0 && \
  # #     make && \
  # #     ln -s /usr/local/bedtools2/bin/* /usr/local/bin/
  # # ##############
  # # ## vcftools ##
  # # ENV ZIP=vcftools-0.1.14.tar.gz
  # # ENV URL=https://github.com/vcftools/vcftools/releases/download/v0.1.14/
  # # ENV FOLDER=vcftools-0.1.14
  # # ENV DST=/tmp

  # # RUN wget $URL/$ZIP -O $DST/$ZIP && \
  # #     tar xvf $DST/$ZIP -C $DST && \
  # #     rm $DST/$ZIP && \
  # #     cd $DST/$FOLDER && \
  # #     ./configure && \
  # #     make && \
  # #     make install && \
  # #     cd / && \
  # #     rm -rf $DST/$FOLDER
  # # ##################
  # # # ucsc utilities #
  # # RUN mkdir -p /tmp/ucsc && \
  # #     cd /tmp/ucsc && \
  # #     wget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigAverageOverBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/wigToBigWig && \
  # #     chmod ugo+x * && \
  # #     mv * /usr/bin/
  # # ############################
  # # # R, bioconductor packages #
  # # # from https://raw.githubusercontent.com/rocker-org/rocker-versioned/master/r-ver/3.4.0/Dockerfile
  # # # we'll pin to 3.4.0 for now

  # # # ARG R_VERSION
  # # # ARG BUILD_DATE
  # # # ENV BUILD_DATE 2017-06-20
  # # # ENV R_VERSION=${R_VERSION:-3.4.0}
  # # # RUN apt-get update && apt-get install -y --no-install-recommends locales && \
  # # #     echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen && \
  # # #     locale-gen en_US.UTF-8 && \
  # # #     LC_ALL=en_US.UTF-8 && \
  # # #     LANG=en_US.UTF-8 && \
  # # #     /usr/sbin/update-locale LANG=en_US.UTF-8 && \
  # # #     TERM=xterm && \
  # # #     apt-get install -y --no-install-recommends \
  # # #     bash-completion \
  # # #     ca-certificates \
  # # #     file \
  # # #     fonts-texgyre \
  # # #     g++ \
  # # #     gfortran \
  # # #     gsfonts \
  # # #     libbz2-1.0 \
  # # #     libcurl3 \
  # # #     libicu55 \
  # # #     libjpeg-turbo8 \
  # # #     libopenblas-dev \
  # # #     libpangocairo-1.0-0 \
  # # #     libpcre3 \
  # # #     libpng12-0 \
  # # #     libtiff5 \
  # # #     liblzma5 \
  # # #     locales \
  # # #     zlib1g \
  # # #     libbz2-dev \
  # # #     libcairo2-dev \
  # # #     libcurl4-openssl-dev \
  # # #     libpango1.0-dev \
  # # #     libjpeg-dev \
  # # #     libicu-dev \
  # # #     libmariadb-client-lgpl-dev \
  # # #     libmysqlclient-dev \
  # # #     libpcre3-dev \
  # # #     libpng-dev \
  # # #     libreadline-dev \
  # # #     libtiff5-dev \
  # # #     liblzma-dev \
  # # #     libx11-dev \
  # # #     libxt-dev \
  # # #     perl \
  # # #     tcl8.5-dev \
  # # #     tk8.5-dev \
  # # #     texinfo \
  # # #     texlive-extra-utils \
  # # #     texlive-fonts-recommended \
  # # #     texlive-fonts-extra \
  # # #     texlive-latex-recommended \
  # # #     x11proto-core-dev \
  # # #     xauth \
  # # #     xfonts-base \
  # # #     xvfb \
  # # #     zlib1g-dev && \
  # # #     cd /tmp/ && \
  # # #     ## Download source code
  # # #     curl -O https://cran.r-project.org/src/base/R-3/R-${R_VERSION}.tar.gz && \
  # # #     ## Extract source code
  # # #     tar -xf R-${R_VERSION}.tar.gz && \
  # # #     cd R-${R_VERSION} && \
  # # #     ## Set compiler flags
  # # #     R_PAPERSIZE=letter && \
  # # #     R_BATCHSAVE="--no-save --no-restore" && \
  # # #     R_BROWSER=xdg-open && \
  # # #     PAGER=/usr/bin/pager && \
  # # #     PERL=/usr/bin/perl && \
  # # #     R_UNZIPCMD=/usr/bin/unzip && \
  # # #     R_ZIPCMD=/usr/bin/zip && \
  # # #     R_PRINTCMD=/usr/bin/lpr && \
  # # #     LIBnn=lib && \
  # # #     AWK=/usr/bin/awk && \
  # # #     CFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g" && \
  # # #     CXXFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g" && \
  # # #     ## Configure options
  # # #     ./configure --enable-R-shlib \
  # # #                --enable-memory-profiling \
  # # #                --with-readline \
  # # #                --with-blas="-lopenblas" \
  # # #                --disable-nls \
  # # #                --without-recommended-packages && \
  # # #     ## Build and install
  # # #     make && \
  # # #     make install && \
  # # #     ## Add a default CRAN mirror
  # # #     echo "options(repos = c(CRAN = 'https://cran.rstudio.com/'), download.file.method = 'libcurl')" >> /usr/local/lib/R/etc/Rprofile.site && \
  # # #     ## Add a library directory (for user-installed packages)
  # # #     mkdir -p /usr/local/lib/R/site-library && \
  # # #     chown root:staff /usr/local/lib/R/site-library && \
  # # #     chmod g+wx /usr/local/lib/R/site-library && \
  # # #     ## Fix library path
  # # #     echo "R_LIBS_USER='/usr/local/lib/R/site-library'" >> /usr/local/lib/R/etc/Renviron && \
  # # #     echo "R_LIBS=\${R_LIBS-'/usr/local/lib/R/site-library:/usr/local/lib/R/library:/usr/lib/R/library'}" >> /usr/local/lib/R/etc/Renviron && \
  # # #     ## install packages from date-locked MRAN snapshot of CRAN
  # # #     [ -z "$BUILD_DATE" ] && BUILD_DATE=$(TZ="America/Los_Angeles" date -I) || true && \
  # # #     MRAN=https://mran.microsoft.com/snapshot/${BUILD_DATE} && \
  # # #     echo MRAN=$MRAN >> /etc/environment && \
  # # #     export MRAN=$MRAN && \
  # # #     echo "options(repos = c(CRAN='$MRAN'), download.file.method = 'libcurl')" >> /usr/local/lib/R/etc/Rprofile.site && \
  # # #     ## Use littler installation scripts
  # # #     Rscript -e "install.packages(c('littler', 'docopt'), repo = '$MRAN')" && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/examples/install2.r /usr/local/bin/install2.r && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/examples/installGithub.r /usr/local/bin/installGithub.r && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/bin/r /usr/local/bin/r

  # # #    ## install r packages, bioconductor, etc ##
  # # #    ADD rpackages.R /tmp/
  # # #    RUN R -f /tmp/rpackages.R && \
  # # #    ## install fishplot ##
  # # #    cd /tmp/ && \
  # # #     wget https://github.com/chrisamiller/fishplot/archive/v0.4.tar.gz && \
  # # #     mv v0.4.tar.gz fishplot_0.4.tar.gz && \
  # # #     R CMD INSTALL fishplot_0.4.tar.gz && \
  # # #     cd && rm -rf /tmp/fishplot_0.4.tar.gz

  # # #    ## Clean up
  # # #    RUN cd / && \
  # # #    rm -rf /tmp/* && \
  # # #    apt-get autoremove -y && \
  # # #    apt-get autoclean -y && \
  # # #    rm -rf /var/lib/apt/lists/* && \
  # # #    apt-get clean

  # # # #################################
  # # # # Python 2 and 3, plus packages

  # # # # Configure environment
  # # # ENV CONDA_DIR /opt/conda
  # # # ENV PATH $CONDA_DIR/bin:$PATH

  # # # # Install conda
  # # # RUN cd /tmp && \
  # # #     mkdir -p $CONDA_DIR && \
  # # #     curl -s https://repo.continuum.io/miniconda/Miniconda3-4.3.21-Linux-x86_64.sh -o miniconda.sh && \
  # # #     /bin/bash miniconda.sh -f -b -p $CONDA_DIR && \
  # # #     rm miniconda.sh && \
  # # #     $CONDA_DIR/bin/conda config --system --add channels conda-forge && \
  # # #     $CONDA_DIR/bin/conda config --system --set auto_update_conda false && \
  # # #     conda clean -tipsy

  # # # # Install Python 3 packages available through pip
  # # # RUN conda install --yes 'pip' && \
  # # #     conda clean -tipsy && \
  # # #     #dependencies sometimes get weird - installing each on it's own line seems to help
  # # #     pip install numpy==1.13.0 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install cruzdb==0.5.6 && \
  # # #     pip install cython==0.25.2 && \
  # # #     pip install pyensembl==1.1.0 && \
  # # #     pip install pyfaidx==0.4.9.2 && \
  # # #     pip install pybedtools==0.7.10 && \
  # # #     pip install cyvcf2==0.7.4 && \
  # # #     pip install intervaltree_bio==1.0.1 && \
  # # #     pip install pandas==0.20.2 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install pysam==0.11.2.2 && \
  # # #     pip install seaborn==0.7.1 && \
  # # #     pip install scikit-learn==0.18.2 && \
  # # #     pip install svviz==1.6.1

  # # # # Install Python 2
  # # # RUN conda create --quiet --yes -p $CONDA_DIR/envs/python2 python=2.7 'pip' && \
  # # #     conda clean -tipsy && \
  # # #     /bin/bash -c "source activate python2 && \
  # # #     #dependencies sometimes get weird - installing each on it's own line seems to help
  # # #     pip install numpy==1.13.0 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install cruzdb==0.5.6 && \
  # # #     pip install cython==0.25.2 && \
  # # #     pip install pyensembl==1.1.0 && \
  # # #     pip install pyfaidx==0.4.9.2 && \
  # # #     pip install pybedtools==0.7.10 && \
  # # #     pip install cyvcf2==0.7.4 && \
  # # #     pip install intervaltree_bio==1.0.1 && \
  # # #     pip install pandas==0.20.2 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install pysam==0.11.2.2 && \
  # # #     pip install seaborn==0.7.1 && \
  # # #     pip install scikit-learn==0.18.2 && \
  # # #     pip install openpyxl==2.4.8 && \
  # # #     source deactivate"

  # # # COPY tsv2xlsx.py /usr/bin/tsv2xlsx.py

  # # # needed for MGI data mounts
  # # RUN apt-get update && apt-get install -y libnss-sss && apt-get clean all

  # # #set timezone to CDT
  # # #LSF: Java bug that need to change the /etc/timezone.
  # # #/etc/localtime is not enough.
  # # # RUN ln -sf /usr/share/zoneinfo/America/Chicago /etc/localtime && \
  # # #     echo "America/Chicago" > /etc/timezone && \
  # # #     dpkg-reconfigure --frontend noninteractive tzdata

  # # #UUID is needed to be set for some applications
  # # RUN apt-get update && apt-get install -y dbus && apt-get clean all
  # # RUN dbus-uuidgen >/etc/machine-id

  # # # WORKS TO HERE
  # # ENV PATH /opt/conda/bin:$PATH

  # # RUN apt-get update --fix-missing && apt-get install -y wget bzip2 ca-certificates \
  # #     libglib2.0-0 libxext6 libsm6 libxrender1 \
  # #     git mercurial subversion

  # # RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda2-4.5.11-Linux-x86_64.sh -O ~/miniconda.sh && \
  # #     /bin/bash ~/miniconda.sh -b -p /opt/conda && \
  # #     rm ~/miniconda.sh && \
  # #     ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
  # #     echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc && \
  # #     echo "conda activate base" >> ~/.bashrc

  # # RUN apt-get install -y curl grep sed dpkg && \
  # #     TINI_VERSION=`curl https://github.com/krallin/tini/releases/latest | grep -o "/v.*\"" | sed 's:^..\(.*\).$:\1:'` && \
  # #     curl -L "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini_${TINI_VERSION}.deb" > tini.deb && \
  # #     dpkg -i tini.deb && \
  # #     rm tini.deb && \
  # #     apt-get clean
  # # RUN conda install -c bioconda sambamba

  # # # WORKDIR /opt

  # # # RUN apt-get update && apt-get install -y \
  # # # 	autoconf \
  # # # 	automake \
  # # # 	make \
  # # # 	g++ \
  # # # 	gcc \
  # # # 	build-essential \
  # # # 	zlib1g-dev \
  # # # 	libgsl0-dev \
  # # # 	perl \
  # # # 	curl \
  # # # 	git \
  # # # 	wget \
  # # # 	unzip \
  # # # 	tabix \
  # # # 	libncurses5-dev

  # # # RUN wget https://github.com/ldc-developers/ldc/releases/download/v0.17.1/ldc2-0.17.1-linux-x86_64.tar.xz && \
  # # #   tar xJf ldc2-0.17.1-linux-x86_64.tar.xz

  # # # ENV PATH=/opt/ldc2-0.17.1-linux-x86_64/bin/:$PATH
  # # # ENV LIBRARY_PATH=/opt/ldc2-0.17.1-linux-x86_64/lib/

  # # # #RUN wget https://github.com/biod/sambamba/releases/download/v0.7.0/sambamba-0.7.0-linux-static.gz && \
  # # # tar xJf sambamba-0.7.0-linux-static.gz

  # # # RUN git clone --recursive https://github.com/lomereiter/sambamba.git && cd WHERE && make
  # # RUN conda install -c bioconda samtools
  # # ## bioconductor R install
  # # # nuke cache dirs before installing pkgs; tip from Dirk E fixes broken img
  # # RUN rm -f /var/lib/dpkg/available && rm -rf  /var/cache/apt/*

  # # # same set of packages for both devel and release
  # # RUN apt-get update && \
  # # 	apt-get -y --no-install-recommends install --fix-missing \
  # # 	gdb \
  # # 	libxml2-dev \
  # # 	python-pip \
  # # 	libz-dev \
  # # 	liblzma-dev \
  # # 	libbz2-dev \
  # # 	libpng-dev \
  # # 	libmariadb-client-lgpl-dev \
  # # 	&& rm -rf /var/lib/apt/lists/*

  # # # issues with '/var/lib/dpkg/available' not found
  # # # this will recreate
  # # RUN dpkg --clear-avail

  # # # # Add bioc user as requested
  # # # RUN useradd -ms /bin/bash -d /home/bioc bioc \
  # # # 	&& echo "bioc:bioc" | chpasswd && adduser bioc sudo
  # # # USER bioc
  # # # RUN mkdir -p /home/bioc/R/library && \
  # # # 	echo "R_LIBS=/usr/local/lib/R/host-site-library:~/R/library" | cat > /home/bioc/.Renviron
  # # # USER root
  # # # RUN echo "R_LIBS=/usr/local/lib/R/host-site-library:\${R_LIBS}" > /usr/local/lib/R/etc/Renviron.site \
  # # # 	&& echo "R_LIBS_USER=''" >> /usr/local/lib/R/etc/Renviron.site \
  # # # 	&& echo "options(defaultPackages=c(getOption('defaultPackages'),'BiocManager'))" >> /usr/local/lib/R/etc/Rprofile.site

  # # # # add R packages test
  # # # RUN R -e "install.packages('methods',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  # # # RUN R -e "install.packages('jsonlite',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  # # # RUN R -e "install.packages('tseries',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  #RSEM
  #RUN cd /opt/
  #RUN git clone https://github.com/deweylab/RSEM.git && cd RSEM && make && make install
  #ENV PATH /opt/RSEM:$PATH

  # # # # install skewer
  # # # RUN \
  # # #   wget -c https://downloads.sourceforge.net/project/skewer/Binaries/skewer-0.2.2-linux-x86_64 && \
  # # #   chmod +x skewer-0.2.2-linux-x86_64 && \
  # # #   cp skewer-0.2.2-linux-x86_64 /usr/local/bin/skewer

  # # # run update and install necessary tools
  # # RUN apt-get update -y && apt-get install -y \
  # #     build-essential \
  # # RUN apt-get install -y software-properties-common
  # # RUN add-apt-repository -y ppa:jonathonf/python-3.6
  # # RUN apt-get update && apt-get install -y python3.6
  #
    ### ^^^ BUILDS INDEPENDENTLY VALIDATED ABOVE THIS POINT ^^^ ###
    ### Last successful build 2020-01-14 15:26 CST              ###
    #
    #
    ### TESTING ###
    ### Installs below this point are not individually validated and my require dependencies from above
    # multiqc
    ## Set the locale
    ## https://stackoverflow.com/questions/28405902/how-to-set-the-locale-inside-a-debian-ubuntu-docker-container#28406007
    #
  RUN pip install -U multiqc

  ##############
  ### simNGS ###
  ##############
  #
  #RUN apt-get update -qq
  #RUN apt-get install -qq --no-install-recommends --allow-unauthenticated \
  #    libblas-dev \
  #    liblapack-dev \
  #    make \
  #    tar \
  #    wget
  #RUN cd /opt && wget --no-check-certificate #https://www.ebi.ac.uk/goldman-srv/simNGS/current/simNGS.tgz && tar -xvzf #simNGS.tgz
  #RUN cd /opt/simNGS/src && make -f Makefile.linux
  #ENV PATH="/opt/simNGS/bin:${PATH}"
  #
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install(
  #
  # WORKS TO HERE [2020-09-21]
  #
  # # GISTIC 2.0
  # RUN mkdir -p /opt/GISTIC2 \
  #     && cd /opt/GISTIC2 \
  #     && wget --no-check-certificate ftp://ftp.broadinstitute.org/pub/GISTIC2.0/GISTIC_2_0_23.tar.gz \
  #     && tar --owner=root --group=root -vxf GISTIC_2_0_23.tar.gz
  # RUN cd /opt/GISTIC2/MCR_Installer \
  #     && unzip MCRInstaller.zip \
  #     && ./install -mode silent -agreeToLicense yes -destinationFolder /opt/GISTIC2
  # RUN cd /opt/GISTIC2 \
  #     && wget --no-check-certificate ftp://ftp.broadinstitute.org/pub/GISTIC2.0/hg19.mat
  # ENV PATH="/opt/GISTIC2/:${PATH}"
  # RUN chown -R root:root /opt/GISTIC2
  # ENV MCR_ROOT=/opt/GISTIC2/MATLAB_Compiler_Runtime
  #

  #
  #RUN conda install -c bioconda mirge3
  #
  #RUN conda create -n py2 python=2.7
  #RUN echo "source activate py2" > ~/.bashrc
  #ENV PATH /opt/conda/envs/py2/bin:$PATH
  #
  #########1#########2#########3#########4#########5#########6#########7#########8
  #

  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  # ichorCNA
  ##
  ## linux dependencies
  RUN apt-get update \
     && apt-get install -y \
     libcurl4-openssl-dev \
     libssl-dev \
     libxml2-dev
  #RUN rm /usr/lib/x86_64-linux-gnu/libcurl.so.4
  #RUN ln -s /usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0 /usr/lib/x86_64-linux-gnu/libcurl.so.4
  ##
  ## R dependencies
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("HMMcopy"); BiocManager::install("GenomeInfoDb"); BiocManager::install("GenomicRanges");'
  ##
  ## git clone install
  RUN cd /opt \
      && git clone https://github.com/broadinstitute/ichorCNA.git \
      && cd ichorCNA \
      && R CMD INSTALL . \
      && cd /opt
  ##
  #RUN echo "source activate py2" > ~/.bashrc
  #ENV PATH /opt/conda/envs/py2/bin:$PATH
****** [[file:docker/rna_Dockerfile][RNA docker]]
:PROPERTIES:
:header-args:bash: :tangle ./docker/rna_Dockerfile
:ID:       1338f10d-c3d0-4b29-9c4a-d3a01758a3d2
:END:
:LOGBOOK:
CLOCK: [2022-06-28 Tue 07:55]--[2022-06-28 Tue 15:55] =>  8:00
:END:
******* Preamble
#+begin_src bash
FROM jeszyman/biotools:1.0.2
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build an RNA-focused docker image                  ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/rna
# (https://hub.docker.com/repository/docker/jeszyman/rna)

#+end_src
******* Conda
#+begin_src bash
RUN conda install -c bioconda kallisto
RUN conda install -c bioconda rsem
RUN conda install -c bioconda star
RUN conda install -c bioconda subread
RUN conda install -c bioconda salmon
#+end_src
******* R and Bioconductor
#+begin_src bash
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("DESeq2", \
                                 "edgeR", \
                                 "fgsea", \
                                 "limma", \
                                 "tximport"));'

#+end_src
******* Custom installs
#+begin_src bash
RUN cd /opt && git clone https://github.com/rajewsky-lab/mirdeep2 && cd /opt/mirdeep2 && perl install.pl

RUN /bin/bash

RUN cd /opt/mirdeep2 && perl install.pl

ENV PATH="/opt/mirdeep2/bin:${PATH}"
#+end_src
****** [[file:docker/atac_Dockerfile][ATAC-seq docker]]
#+begin_src bash :tangle ./docker/atac_Dockerfile
FROM jeszyman/biotools:1.0.2
#########1#########2#########3#########4#########5#########6#########7#########8

###################################
###   Dockerfile for ATAC-seq   ###
###################################

# After build, the image will be pushed to dockerhub as jeszyman/atac
# (https://hub.docker.com/repository/docker/jeszyman/atac)

RUN apt-get -y update && apt-get upgrade -y

# Simple bioconda installs
RUN conda install -c bioconda macs2

# Simple R installs
##
## Via Bioconductor
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("ATACseqQC", \
                                 "ChIPpeakAnno", \
                                 "ChIPseeker", \
                                 "clusterProfiler", \
                                 "csaw"));'


# ATACseqQC
##
## Linux prerequisites
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated --fix-missing \
    libgsl-dev \
    libcurl4-openssl-dev
    #libssl-dev
##
## ATACseqQC
RUN R -e 'install.packages("BiocManager"); \
    BiocManager::install(); \
    BiocManager::install(c("ATACseqQC"));'

# Homer
RUN mkdir /opt/homer
RUN cd /opt/homer
RUN wget http://homer.ucsd.edu/homer/configureHomer.pl -O /opt/homer/configureHomer.pl
RUN perl /opt/homer/configureHomer.pl -install

ENV PATH="${PATH}:/opt/homer/bin/"

RUN perl /opt/homer/.//configureHomer.pl -install mouse-p

RUN conda install -c bioconda samtools=1.7

# R CRAN Installs
RUN R -e "install.packages(c('statmod', repos='http://cran.rstudio.com/'))"

#+end_src
***** Helper shell scripts
****** Built docker container
:PROPERTIES:
:ORDERED:  t
:END:
#+begin_src bash :tangle ./tools/shell/docker_build.sh
#!/usr/bin/env bash

set -o errexit   # abort on nonzero exitstatus
set -o nounset   # abort on unbound variable
set -o pipefail  # don't hide errors within pipes

print_usage(){
    cat <<- EOF

###################################################
###   Docker and Singularity Building Command   ###
###################################################

Steps
1) Builds docker container
2) Pushes container to dockerhub
3) Builds singularity file
4) Pushes singularity file to box

usage: docker_build.sh [options] <DOCKER FILE> <DOCKER TAG> [OPTIONAL - DOCKER BUILD LOC]
  options:
    -n  No cache, use for validation of full build
    -s  A sif directory location
    -v  Docker version, a version appended to docker and singularity tags
        Note: adding a version will push the singularity build to
        shared/singularity-containers/stable

example:

docker_build.sh -n -s ~/sing_containers -v 2.0.0 ~/dockerfiles/atac_Dockerfile jeszyman/atac ~/docker_bulid_dir

EOF
}

# Setup options
while getopts "nt:v:s:" options;
do
    case "${options}" in
        n) usecache="--no-cache" ;;
        v) version=${OPTARG} ;;
        s) sifdir=${OPTARG} ;;
        h | *) print_usage; exit 0 ;;
    esac
done

shift $((OPTIND -1))

[[ $# -eq 0 ]] && print_usage && exit 1

# Scipt variables
dockerfile=$1
tag=$2
if [ -z ${usecache+x} ]; then usecache="cache"; fi
dockerloc="${3:-.}"

main(){
    # Build and push docker container. Append version if provided.
    printf "\nBuilding Docker container now...\n "
    sleep 2s
    if [ -z ${version+x} ]
    then
        build_docker $dockerfile $tag $dockerloc $usecache
        printf "\nPushing docker container to dockerhub...\n "
        docker push $tag
    else
        build_docker $dockerfile "${tag}:${version}" $dockerloc $usecache
        printf "\nPushing docker container to dockerhub...\n "
        docker push $tag:$version
    fi

    # Build and push sif if provided.

    if [ -z ${sifdir+x} ];
    then
        sleep 2s
        printf "\n-s flag is unset, nothing more to do.\n "
    else
        dockertag="$tag"
        if [ -z ${version+x} ];
        then
            siftag=$(echo $tag | sed 's/^.*\///g')
            target="shared/singularity-containers/most-recent/"
        else
            presiftag=$(echo $tag | sed 's/^.*\///g')
            siftag="${presiftag}.${version}"
            target="shared/singularity-containers/stable/"
        fi

        printf "\nBuilding Singularity container now...\n"

        build_singularity $sifdir \
                          $siftag \
                          $dockertag

        sleep 2s
        echo "build_singularity $sifdir $siftag $dockertag"
        printf "\nLocal Singularity build complete.\nPushing singularity to box...\n"

        push_singularity $sifdir \
                         $siftag \
                         $target

    fi
}

# main(){


#     # Build and push sif if provided.

#     if [ -z ${sifdir+x} ];

#     then
#         sleep 2s
#         printf "\n-s flag is unset, nothing more to do.\n "
#     else
#         dockertag="$tag"
#         if [ -z ${version+x} ];
#         then
#             siftag="$tag"
#             target="shared/singularity-containers/most-recent"
#         else
#             siftag="${tag}.${version}"
#             target="shared/singularity-containers/stable"
#         fi
#         printf "\nBuilding Singularity container now...\n"
#         sleep 2s
#         build_singularity $sifdir \
#                           $siftag \
#                           $dockertag
#         printf "\nLocal Singularity build complete.\nPushing singularity to box...\n"
#         push_singularity $sifdir \
#                          $siftag \
#                          $target
#     fi
# }

build_docker(){
    local dockerfile="${1}"
    local dockertag="${2}"
    local dockerloc="${3}"
    if [ $4 == "--no-cache" ];
    then
        docker build \
               --file $dockerfile $4 \
               --pull \
               --tag $dockertag \
               $dockerloc
    else
        docker build \
               --file $dockerfile \
               --pull \
               --tag $dockertag \
               $dockerloc
    fi
}

push_docker(){
    local dockertag="${1}"
    docker push $dockertag
}

build_singularity(){
    local sifdir="${1}"
    local siftag="${2}"
    local dockertag="${3}"
    singularity pull \
                --force \
                "${sifdir}/${siftag}.sif" \
                "docker://${dockertag}"
}

push_singularity(){
    local sifdir="${1}"
    local siftag="${2}"
    local target="${3}"
    rclone copy \
           --update \
           --verbose \
           "${sifdir}/${siftag}.sif" remote:"${target}"
}

main "$@"

#+end_src

****** Make singularity container
#+begin_src bash
#!/usr/bin/env bash
#
###   SCRIPT NAME   ###
#

# Notes
# ...

# Setup

set -e # enable exit on error
set -u # unset variables are errors
set -o pipefail # failures can occur anywhere in pipeline

# Variables

readonly blah=$1
blue=$(tput setaf 4)

# Functions

usage()
{
    printf "
           ${blue}Script to do blah \n
    usage:  blah\n
"
    exit 1
}

main()
{
    [[ $# -ne 2 ]] && usage
    echo "$blah"
    normal=$(tput sgr0)
    printf "%40s\n" "${blue}This text is bluenormal"
}

# Run

main "$@"
#+end_src

#+begin_src bash
singularity pull /mnt/ris/aadel/jeszyman/sing_containers/biotools.sif docker://jeszyman/biotools
#+end_src
****** Run functions
- run docker interactively
  #+name: docker_interactive
  #+begin_src bash
docker_interactive() {
    if [ -f /.dockerenv ]; then
        echo "shell already in docker, exiting"
        exit 1
    else
        account="USER INPUT"
        container="USER INPUT"
        read -p "docker account name: " account
        account="${account:=jeszyman}"
        read -p "container name: " container
        container="${container:=biotools}"
        case $HOSTNAME in
            (radonc-cancerbio) docker run -it \
                                      --env HOME=${HOME} \
                                      --hostname ${HOSTNAME} \
                                      --user $(id -u ${USER}) \
                                      --volume /home/:/home/ \
                                      --volume /mnt/:/mnt/ \
                                      --volume /tmp/:/tmp/ \
                                      $account/$container \
                                      /bin/bash;;
            (acl*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /drive3/:/drive3/ \
                          -v /duo4/:/duo4/ \
                          -v /home/:/home/ \
                          -v /tmp/:/tmp/ \
                          -u $(id -u ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (ACL*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /home/:/home/ \
                          -v /duo4/:/duo4/ \
                          -u $(id -u ${USER}):$(id -g ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (virtual-workstation*.gsc.wustl.edu) bsub -Is -q docker-interactive -a 'docker($account/'"$container"')' /bin/bash;;
            (*) docker run -it \
                       --env HOME=/home/${USER} \
                       --hostname ${HOSTNAME} \
                       --user $(id -u ${USER}) \
                       --volume /home/:/home/ \
                       --volume /mnt/:/mnt/ \
                       --volume /tmp/:/tmp/ \
                       $account/$container \
                       /bin/bash;;
        esac
    fi
}
#+end_src

:LOGBOOK:
CLOCK: [2020-10-08 Thu 10:05]--[2020-10-08 Thu 10:15] =>  0:10
:END:
***** Reference
- [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][Docker software]]

**** cfDNA clonal evolution
https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFWWgNKMCfqmfTbDzSlvgqZDq
**** cfDNA copy number calling
:PROPERTIES:
:ID:       21d876e8-8646-4773-91c1-c4385c3218c7
:END:
- [[#0f1f4aa4-c755-42ff-8f81-8a32c14293a0][Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)]]
- https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-023-01171-w
- has LOH on seq panel w/ custom pipeline  - https://www.sciencedirect.com/science/article/pii/S0923753419454604?pes=vor#ec0010
- https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-020-00735-4#Sec12
- snv based with ABSOLUTE https://www.nature.com/articles/nbt.2203#Sec12
  - https://bioconductor.org/packages/devel/bioc/vignettes/PureCN/inst/doc/PureCN.pdf
- post-CN call processessing and denoising
  - cite:zviran2020
- BIC-seq
  - cite:zviran2020
- Proprietary method
  - cite:chen2019cna
- CNVkit on duplex hybrid capture reads
  - cite:mansukhani2018
- optimal read pre-processing for cfDNA CNV
  - cite:molparia2018
- arm-level by z-score
  - cite:xu2018crc
- review articles
  - cite:cristofanilli2017
    - (p. 119)"In addition, although copy number assessments can be challenging in ptDNA, there are studies that suggest that this is feasible. For example, HER2 amplification for breast and possibly other cancers may one day be assayed through the use of blood [49],"
    - (p. 119)"Wyatt et  al. serially profiled ptDNA in men with castration-resistant metastatic prostate cancer to assess for copy number and mutations in androgen receptor (AR) along with targeted sequencing of 19 genes through subsequent therapies [51]."
  - cite:huang2019
- simulation and power calculation
  - cite:molparia2017
- GISTIC2
  - cite:bujak2020
  - cite:hemming2019
- ichorCNA
  - cite:hemming2019
  - ichorCNA on off-target reads - cite:shah2021
  - Other papers applying ichorCNA
    - cite:bujak2020
    - cite:ge2020
      - tumor vs cfDNA CN see cite:ge2020 fig2C
    - cite:stover2018
    - https://bmccancer.biomedcentral.com/articles/10.1186/s12885-019-6003-8
  - Method-establishing paper cite:adalsteinsson2017
- instability score
  - cite:li2017a
  - cite:feng2020
- by PCR
  - https://www.pnas.org/content/117/9/4858
**** cfDNA fragment end motif analysis
:PROPERTIES:
:ID:       de9c6398-4ffe-48bf-9ef9-31028edf9873
:END:
- [[file:~/data/career/golkaram2022.pdf]]
- [[#0f1f4aa4-c755-42ff-8f81-8a32c14293a0][Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)]]
- cite:an2023
- F-profiles: cite:zhou2023
- cite:guo2022
- cite:zhitnyuk2022
- cite:zhou2022 - fragma
- cite:moldovan2021
- cite:zhou2021 - jaggedness
- cite:jiang2020motif


**** cfDNA fragmentomics
- Fragmentomics from targeted panels [[cite:&helzer2023panelfrag]]
- [[cite:&jiang2015]]
- fragment length NMF [[cite:&renaud2021]]
- https://academic.oup.com/clinchem/article/68/9/1184/6623478#377336919
- fragmentomics from targeted panel [[https://www.annalsofoncology.org/article/S0923-7534(23)00724-X/fulltext]]
- [[file:~/library/alizadeh2023.pptx]]
- cite:lo2021
- Inferred gene expression
  - https://mail.google.com/mail/u/0/#inbox/FMfcgzGsltShrnWQMVgTLMNJhmrjkmXx?projector=1&messagePartId=0.1
- transcription factor binding
  - cite:ulz2019
- https://www.nature.com/articles/s41416-021-01635-z
- https://www.pnas.org/doi/10.1073/pnas.2209852119
- LIQUORICE
  - cite:peneder2021
  - https://jhoonline.biomedcentral.com/articles/10.1186/s13045-022-01342-z#Sec2
- cite:ding2022
- cite:esfahani2022
- http://m.genome.cshlp.org/content/early/2021/12/20/gr.275691.121.full.pdf#page=1&view=FitH
- fragments for open chromatin and tissue of origin
- footprinting
  - https://www.nature.com/articles/s41523-021-00237-5
  - for nucleosome footprinting cite:snyder2016
- fragment size
  - https://www.nature.com/articles/s41586-020-2140-0
  - https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-06848-9#Sec9
- DELFI (DNA evaluation of fragments for early interception)
  - [[file:~/library/alizadeh2023.pptx]]
  - some evidence for increased abberation in non-cancer cirrhosis
    - cite:foda2022
  - short-to-long ratio by genomic bin cite:mathios2021
    - https://github.com/cancer-genomics/reproduce_lucas_wflow
  - https://www.nature.com/articles/s41467-022-31419-9
  - cite:cristiano2019 Delfi
  - https://github.com/Cancer-Genomics/delfi_scripts
- cite:mouliere2018
  - cite:mouliere2018a
- cite:chabon2020 - per individual mutant molecules- see "ctDNA fragment size analysis", R zoo for sliding window, relative enrichment per frag size
- cite:peneder2021
  - [[https://ega-archive.org/studies/EGAS00001003611][peneder data (restricted)]]
**** cfDNA inferred tissue gene expression
:PROPERTIES:
:ID:       dd8b334b-e431-4709-8485-bf0dda1760a5
:END:
https://www.nature.com/articles/ng.3648
cite:foda2022
**** cfDNA from nonterminal mouse blood draws
:PROPERTIES:
:ID:       2e475e89-40a2-4359-903d-6e590f4ab8a0
:END:
  - https://www.frontiersin.org/articles/10.3389/fgene.2018.00006/full#T3
  - https://www.nature.com/articles/cr2017121.pdf?origin=ppub
  - https://clincancerres.aacrjournals.org/content/23/20/6305?ijkey=46368653e37c96739922e66f2f6ae51379b550f3&keytype2=tf_ipsecsha
- https://www.nature.com/articles/s41467-018-06159-4
- cite:campbell2020
- https://www.nejm.org/doi/10.1056/NEJMoa2024534?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub++0pubmed
- duplication rate as a measure of lib complexity
**** cfDNA long read
https://www.biorxiv.org/content/10.1101/2021.10.18.464684v1.full
**** cfDNA methylation
- [[cite:&li2023cfdna]]
- TRACERx - https://www.nature.com/articles/s41586-023-05776-4
- https://mail.google.com/mail/u/0/#inbox/FMfcgzGtxKMwMFsxwrhWpKhfMNbKGtdK
- https://www.annalsofoncology.org/article/S0923-7534(23)00087-X/fulltext
- https://academic.oup.com/nargab/article/5/2/lqad048/7188752
- cite:shen2019cfmedip used in cite:shen2018
- [[id:d9b52bbd-ec6a-4b9e-af4b-17f6efafdcf6][methylation sequencing]]
- https://jhoonline.biomedcentral.com/articles/10.1186/s13045-020-00933-y
- cite:barefoot2022rads
- cite:liu2020
- [[id:0f1f4aa4-c755-42ff-8f81-8a32c14293a0][Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)]]
- cfDNA methylation
  - cite:liu2020
    - https://www.dana-farber.org/newsroom/news-releases/2020/new-blood-test-can-detect-wide-range-of-cancers--now-available-to-at-risk-individuals-in-clinical-study-at-dana-farber/
  - ct methylation
    - cite:widschwendter2017
    - cite:xu2017a
    - cite:heitzer2017
  - cite:guo2017
  - meth for early detection
    - https://www.biorxiv.org/content/10.1101/422675v1
    - https://www.pnas.org/content/113/16/4338.long
    - https://www.nature.com/articles/cr2017106
- https://www.jci.org/articles/view/145973
**** cfDNA methylation marker PCR
https://www.pnas.org/doi/abs/10.1073/pnas.1519286113
**** cfDNA nucleosome profiling
:PROPERTIES:
:ID:       c4dde8d6-1a02-42c7-a6b6-786b757fcad1
:END:
- [[id:2d0d13d9-857a-4285-89e6-4614f749afd3][cfDNA topology]]
- Plasma nucleosome capture and antibody binding [[cite:&fedyuk2022]]
- OCRDetector, for detecting OCRs from whole-genome cfDNA sequencing data   [[cite:&wang2021ocr]]
- nucleosome profiling with griffin
  - [[cite:&doebley2022griffin]]
  - https://github.com/adoebley/Griffin/wiki
  - https://www.medrxiv.org/content/10.1101/2021.08.31.21262867v1.full-text
  - https://github.com/alexcpan/Griffin
- tissue-specific nucleosome depleted regions cite:zhu2021
- inferred gene expression and regulation
  - w/ Griffin, Gavin Ha method cite:sarkar2022
  - cite:ulz2016
- https://github.com/alexcpan/Griffin
**** cfDNA small variant calling
- [[https://www.nature.com/articles/s41591-023-02598-9#:~:text=The%20trial%20met%20its%20primary,CI%3A%2056.5%E2%80%9388.5%25).][ctDNA response after pembrolizumab in non-small cell lung cancer: phase 2 adaptive trial results]]

- [[id:55b77210-fcb6-438d-a546-0c3605f6f5c4][Integrated hybrid capture and genome-wide- MAESTRO]]
- SEQC2 analytical validity [[cite:&deveson2021]]
https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFWNbSMxNkbQTTtVwBpvBltDq
- PhasED-seq- https://www.nature.com/articles/s41587-021-00981-w
  - - phased seq https://www.sciencedirect.com/science/article/pii/S2666634021003202
- cite:cohen2021
- cfDNA and ctDNA sequence variant calling
  - cite:razavi2019
  - cite:chabon2020
    - https://www.theguardian.com/society/2020/mar/25/ai-program-could-check-blood-for-signs-of-lung-cancer
  - tumor mutational burden
    - https://ascopubs.org/doi/full/10.1200/PO.19.00204
**** cfDNA tissue of origin discovery
:PROPERTIES:
:ID:       47c4ed56-1dea-48dc-bb25-ddbb0a4cd359
:END:
- https://www.frontiersin.org/articles/10.3389/fgene.2021.671057/full
- https://onlinelibrary.wiley.com/doi/10.1002/ctm2.177
- https://insight.jci.org/articles/view/153559#SEC4
**** cfDNA window protection score
:PROPERTIES:
:ID:       2b5a4f91-873a-4c79-a265-f528c52f06bc
:END:
- [[file:~/data/career/golkaram2022.pdf]]

**** Compare gene expression across species
https://www.google.com/search?q=biostars+mouse+human+homology+match+genes&oq=biostars+mouse+human+homology+match+genes&aqs=chrome..69i57.9457j0j1&sourceid=chrome&ie=UTF-8
**** Copy-neutral loss of heterozygosity
https://www.biostars.org/p/254848/#:~:text=%22The%20B%2DAllele%20Frequency%20is,alleles%20(e.g.%20AB).%22
**** design matrix
https://seqqc.wordpress.com/2020/11/28/10-tips-tricks-for-complex-model-matrix-designs-in-dge-analysis/
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/
https://stackoverflow.com/questions/59701903/what-does-0-mean-in-the-r-model-matrix
**** Determine sequencing coverage

**** Determine sequencing coverage
:PROPERTIES:
:ID:       7ec3083a-b152-4ab2-b4f8-5fb9bb5ba068
:END:
- https://support.illumina.com/downloads/sequencing_coverage_calculator.html
- idea- manual w/ R, awk
  - https://www.biostars.org/p/104063/
  - https://sarahpenir.github.io/bioinformatics/awk/calculating-mapping-stats-from-a-bam-file-using-samtools-and-awk/
- expected
  - [[mu4e:msgid:33C61570-8F28-4ED8-B061-60E315F153D8@wustl.edu][Coverage for WGS]]
- need depth plots depth (y) vs. 10kb bin x smooth across genome
  - https://www.biostars.org/p/104063/

***** Fusion calling
- https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-021-08094-z
***** Machine learning of whole genome sequencing data

**** Differential chromatin accessibility
:PROPERTIES:
:ID:       17f978bc-f2f2-497d-a8f1-b1d47927380f
:END:
- Reske
  - https://github.com/reskejak/ATAC-seq/blob/a7c3ba9a024030eb985e1a7d17ddb4d089446041/ATACseq_workflow.txt
  - cite:reske2020
- [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
- Run macs2 with a set q value?
- https://www.biostars.org/p/308976/
- cite:gontarz2020
- https://support.bioconductor.org/p/61604/
- cite:reske2020
- Homer
  - http://homer.ucsd.edu/homer/ngs/mergePeaks.html
- Diffbind
  - https://www.cruk.cam.ac.uk/core-facilities/bioinformatics-core/software/diffbind-tool-for-chip-seq-and-atac-seq-analysis
  - https://bioinformatics-core-shared-training.github.io/Quantitative-ChIPseq-Workshop/articles/Quantitative-ChIPseq-Workshop.html#loading-an-experiment-into-diffbind-1
  - https://rdrr.io/bioc/DiffBind/f/inst/doc/DiffBind.pdf
  - https://bioconductor.org/packages/release/bioc/vignettes/DiffBind/inst/doc/DiffBind.pdf
- csaw
  - http://bioconductor.org/books/3.14/csawBook/counting-reads-into-windows.html#
  - https://www.biostars.org/p/320394/
- [[id:ae0363f3-379e-4521-9184-cbc0d46501cd][Chromatin accessibility profiling]]
- General reference
  - cite:minnoye2021
  - Complete list of bulk methods in cite:minnoye2021 box 1
- https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md
**** Differential gene expression
:PROPERTIES:
:ID:       ee030cef-74d4-435d-8228-b85f0e6203db
:END:
https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2021/RNAseq/Markdowns/09_Linear_Models.html
https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html#design-matrices-with-and-without-intercept-term
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/
https://www.nature.com/articles/s42003-023-04791-5

cite:buschmann2016
"y. Even though comparisons of software
performances on small RNA-Seq data are scarce, a num-
ber of independent and extensive evaluations for mRNAs
based on either synthetic data with clearly defined prop-
erties (216), or on biological datasets with validated gene
expressions (217,218) have been made recently. While it
was shown that statistical power of almost all methods is
heavily dependent on the number of samples per condition
and less on sequencing depth, the variability of expression
changes in biological datasets affects each analysis tool dif-
ferently. Outliers, ‘ON/OFF’ expression changes, where a
gene is detected in only one condition, and lopsided ex-
pression patterns, where upregulations drastically outweigh
downregulations or vice versa, influence specificity (false
positive rate) and sensitivity (false negative rate) of each
method unequally. Nonetheless, some methods appear to
capture the true expression status of small RNAs better
than others. Most independent evaluations seem to agree
that calling differential expression with SAMSeq works well
for datasets with sufficient sample sizes of 10 or more. For
smaller datasets, edgeR and especially the more conserva-
tive DESeq (or DESeq2) are found to be the methods of
choice. On"

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/
https://rnabio.org/module-03-expression/0003/06/01/DE_Pathway_Analysis/
https://www.nature.com/articles/s41598-018-29077-3
https://rnabio.org/module-03-expression/0003/05/01/Batch-Correction/
https://compgenomr.github.io/book/clustering-grouping-samples-based-on-their-similarity.html
- https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

- [[https://www.bioconductor.org/packages/devel/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#differential-expression-analysis][easy as 1-2-3]]
https://github.com/hbctraining/DGE_workshop_salmon_online
https://f1000research.com/articles/5-1438/v2
https://www.nature.com/articles/s41598-020-74567-y#Abs1
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6954399/
- Results reporting
  - Examples
    - cite:ward2021 page 4
***** Visualizations
- volcano
- heatmap
-
***** time course
:PROPERTIES:
:ID:       43f1772e-f8ce-4f6b-a4bd-d35d381b89c9
:END:
  - cite:spies2017
  - https://support.bioconductor.org/p/65676/#66860
  - https://www.biostars.org/p/138988/
  - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#group-specific-condition-effects-individuals-nested-within-groups
  - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
  - http://seqanswers.com/forums/showthread.php?t=64039
  - https://www.sciencedirect.com/science/article/pii/S2001037015000392
***** experimental design
:PROPERTIES:
:ID:       4b9eee0d-4dd2-4b31-a5c2-873663443671
:END:
  - cite:mccarthy2012
  - https://angus.readthedocs.io/en/2016/_static/DifferentialExpressionBasics_NGS2016_ID.pdf
***** specialized DE datasets
:PROPERTIES:
:ID:       722cd834-2504-4ae0-84ef-58554fa4da32
:END:
- long non-coding- https://europepmc.org/article/PMC/6058388
- transcript / isoform DE
  - https://davetang.org/muse/2017/10/25/getting-started-hisat-stringtie-ballgown/
  - sleuth https://www.nature.com/articles/nmeth.4324 cite:pachter2015
  - CuffDiff- isoform count-based
***** tutorials
:PROPERTIES:
:ID:       9bd06c29-45ba-42b8-b048-27b83a0fa47d
:END:
- http://master.bioconductor.org/packages/release/workflows/html/rnaseqGene.html
- https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs
- https://bioconductor.org/packages/devel/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#useful-graphical-representations-of-differential-expression-results
- [[https://github.com/griffithlab/rnaseq_tutorial/wiki/Citation][Griffith RNA-seq tutorial]]
***** statistical models
:PROPERTIES:
:ID:       8a788e6c-3899-4e6f-bf09-9c96d110aeaa
:END:

- overdispersion makes working under Poisson assumption prone to high false positives b/c sampling error underestimated


Di,Y., Schafer,D.W., Cumbie,J.S. and Chang,J.H. (2011) The NBP
negative binomial model for assessing differential gene expression
from RNA-Seq. Stat. Appl. Genet. Mol. Biol., 10,
doi:10.2202/1544-6115.1637.



  - cite:huang2015
  - https://angus.readthedocs.io/en/2016/_static/DifferentialExpressionBasics_NGS2016_ID.pdf
  - https://bioinformatics-core-shared-training.github.io/RNAseq-R/slides/LinearModels.pdf
  - dispersion
    - https://support.bioconductor.org/p/75260/
  - no t-test
    - cite:jeanmougin2010
    - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0012336
  - RNA expression varies about 10^5
  - RNA counts follow “overdispersed” poisson ( (large number of events, each event is rare) / negative binomial distribution (additional factor is dispersion)
    expression levels of same gene across tissues follow log-normal distribution
    - base distribution is Poisson (b/c large number of events, each event is rare), variance = mean
    - shot noise- noise from sampling
    - zero inflation- many zero counts
    - negative binomial (overdispersed Poisson), variance = mean + (1/r)*u^2, used by
      - DESeq2 (can also do continuous values like limma)
      - EdgeR
***** Methods and tools
:PROPERTIES:
:ID:       68BB9D3D-5879-4F2E-904B-54F70EAD2B5A
:END:
- count-based
  - negative binomial
    - [[id:32BB77F2-0283-4A52-86AC-CDF2CB6A7315][edgeR]], dispersion by weighted likelihood
    - [[id:2F642F69-A9C8-46DA-BE31-783AC8D4F4B6][DESeq2]], dispersion by modeling all genes via regression
- [[id:F9684168-A438-4F5E-81D6-D645B52B4074][limma voom]], voom transforms read count
- Bayesian
  - BaySeq
  - BitSeq
  - ebSeq
- Non-parametrics- require 5-10 replicates per group
  - SAMSeq
  - NOISeq
- methods comparisons
  - cite:fonseca2014
  - cite:huang2015
  - cite:seyednasrollah2015
  - cite:tang2015
  - cite:rapaport2013
  - https://www.nature.com/articles/s41598-020-59516-z#data-availability
  - comparison
    - https://academic.oup.com/bib/article/16/1/59/240754#26957696
- de novo
  - https://informatics.fas.harvard.edu/best-practices-for-de-novo-transcriptome-assembly-with-trinity.html
***** outputs
:PROPERTIES:
:ID:       48db2c9c-5d17-4438-aa40-43816245c490
:END:
  - log fold change
    - https://www.biostars.org/p/101727/


***** Notes
:PROPERTIES:
:ID:       558011C7-FB50-435E-B130-E3485CC25B90
:END:
- cite:anders2010
- https://pubmed.ncbi.nlm.nih.gov/22931062/
- bench methods
  - cite:parekh2016

**** DNA-seq pre-alignment processing and QC
:PROPERTIES:
:ID:       719b2017-4ec6-44dd-8e91-a677bc6654bd
:END:
- https://github.com/najoshi/sickle
**** Fusion calling
- https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-021-08094-z
**** Get TSS
get tss as annotation https://www.biostars.org/p/68883/
**** Loss of heterozygosity
https://www.sciencedirect.com/science/article/pii/S0016508522010861?via%3Dihub
https://www.biostars.org/p/254848/#:~:text=%22The%20B%2DAllele%20Frequency%20is,alleles%20(e.g.%20AB).%22
**** Machine learning of whole genome sequencing data

:PROPERTIES:
:ID:       38c00ba1-299f-4a0d-959e-58845d12c712
:END:
- cite:mathios2021
- cite:widman2022

***** Make bwa index                                               :smk_rule:
- Snakemake
  #+begin_src snakemake
rule make_bwa_index_:
    input:
        fasta = config["data_dir"] + "/inputs/hg19.fa.gz",
    params:
        out_dir = config["data_dir"] + "/ref/hg19_bwa"
        prefix = config["data_dir"] + "/ref/hg19_bwa/hg19"
    output:
        config["data_dir"] + "/ref/hg19_bwa/hg19.amb"
    shell:
        """
        scripts/make_bwa_index_.sh {params.out_dir} {params.prefix} {input.fasta}
        """
#+end_src
- [[file:./workflow/scripts/make_bwa_index_.sh][Base script]]
  #+begin_src bash :tangle ./workflow/scripts/make_bwa_index_.sh
mkdir -p $1
bwa index -p $2 -a bwtsw $3
# Snakemake variables
# Function
# Run command
#+end_src

***** Multiome ATAC + gene expression
https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression
***** Nucleic acid sequence alignment

**** Make bwa index                                                :smk_rule:
- Snakemake
  #+begin_src snakemake
rule make_bwa_index_:
    input:
        fasta = config["data_dir"] + "/inputs/hg19.fa.gz",
    params:
        out_dir = config["data_dir"] + "/ref/hg19_bwa"
        prefix = config["data_dir"] + "/ref/hg19_bwa/hg19"
    output:
        config["data_dir"] + "/ref/hg19_bwa/hg19.amb"
    shell:
        """
        scripts/make_bwa_index_.sh {params.out_dir} {params.prefix} {input.fasta}
        """
#+end_src
- [[file:./workflow/scripts/make_bwa_index_.sh][Base script]]
  #+begin_src bash :tangle ./workflow/scripts/make_bwa_index_.sh
mkdir -p $1
bwa index -p $2 -a bwtsw $3
# Snakemake variables
# Function
# Run command
#+end_src

**** Mappability
:PROPERTIES:
:ID:       347d4cc0-a25d-4636-96d4-65e6319022df
:END:
- http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign100mer.bigWig
- cite:derrien2012
<<<<<<< HEAD
***** Mappability
=======
**** methylation sequencing
:PROPERTIES:
:ID:       d9b52bbd-ec6a-4b9e-af4b-17f6efafdcf6
:END:
- https://nf-co.re/methylseq/output
- https://clinicalepigeneticsjournal.biomedcentral.com/articles/10.1186/s13148-019-0795-x#Sec8
- https://www.biostars.org/p/164929/
- https://academic.oup.com/gigascience/article/10/5/giab033/6272610
[[file:~/repos/biotools/biotools.org::*cfDNA methylation][cfDNA methylation]]
https://www.nature.com/articles/nature23891
**** Multiome ATAC + gene expression
:PROPERTIES:
:ID:       8859ad56-d367-4000-9895-7dc9baf41756
:END:
https://cdn.10xgenomics.com/image/upload/v1660261285/support-documents/CG000375_DemonstratedProtocol_NucleiIsolationComplexSample_ATAC_GEX_Sequencing_Rev_C.pdf
https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression
**** Mutation and expression signatures
[[cite:&alexandrov2020]]
https://aacrjournals.org/cancerdiscovery/article/10/4/552/2575/Impaired-Death-Receptor-Signaling-in-Leukemia
https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-022-01050-w#Sec25
https://www.nature.com/articles/nature12477

https://www.cell.com/cell-genomics/pdfExtended/S2666-979X(22)00124-0
https://www.nature.com/articles/s41467-022-32598-1#Sec11
https://www.nature.com/articles/s41586-020-1943-3
https://aacrjournals.org/cancerdiscovery/article/10/4/552/2575/Impaired-Death-Receptor-Signaling-in-Leukemia
https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-022-01050-w#Sec25
https://www.nature.com/articles/nature12477
**** NEBNext Enzymatic Methyl-seq kit
https://www.neb.com/products/e7120-nebnext-enzymatic-methyl-seq-kit#Product%20Information
https://www.neb.com/-/media/nebus/files/application-notes/technote_nebnext_enzymatic_methyl-seq.pdf?rev=015e017f782a4bc9b50b61f1b0f3c807&hash=E23FFED5E02C7A282E753B6B8F555A30
**** Nucleic acid sequence alignment
:PROPERTIES:
:ID:       326ecd60-8cd4-4815-a389-967b2c3fef0a
:END:
- fastas for sequence alignment
  - https://lh3.github.io/2017/11/13/which-human-reference-genome-to-use
  - https://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/ "The initial/ subdirectory contains files for the initial release of GRCm38, which has 66 sequences, and no original alternate sequences and no fix sequences. It is the same as the parent download directory. This is probably the best genome file for aligners and most analysis tasks, a version called "analysisSet" for the human genome."
- https://www.biostars.org/p/352354/
- https://www.biostars.org/p/342482/
- https://www.biostars.org/p/388935/

https://www.illumina.com/science/genomics-research/articles/dragen-demystifying-reference-genomes.html

https://gatk.broadinstitute.org/hc/en-us/articles/360041155232-Reference-Genome-Components

73. Li,H. and Homer,N. (2010) A survey of sequence alignment
algorithms for next-generation sequencing. Brief. Bioinform., 11,
473–483.
174. Bao,S., Jiang,R., Kwan,W., Wang,B., Ma,X. and Song,Y.-Q. (2011)
Evaluation of next-generation sequencing software in mapping and
assembly. J. Hum. Genet., 56, 406–414.

- hash table like BLAST or suffic/prefix like Burrows-Wheeler Transform
- https://davetang.org/muse/2017/10/25/getting-started-hisat-stringtie-ballgown/
- prerequisites
  - adaptor identification and trimming
- pseudoaligner
  - kallisto
  - Salmon
- Aligners
  - TopHat2
  - Specialized aligners
    - Fusions
      - TopHat Fusion
        - tophat fusion documentation http://ccb.jhu.edu/software/tophat/fusion_manual.html
      - STAR for chimeric reads / STAR-Fusion
        - [[https://github.com/STAR-Fusion/STAR-Fusion/wiki][website]]
        - From cite:star_2.5_dobin Chapter 5
        - From cite:dobin2015mapping Alternate Protocol 6
          - --chimSegmentMin <N> is minimum allowed length for each of the chimeric segments
        - additional outputs are
          - Chimeric.out.sam
          - Chimeric.out.junction- a STAR-specific format
    - To transcriptome
      -
  - [[*STAR][STAR]]
- HISAT2

**** Omics integration
:PROPERTIES:
:ID:       00e4056a-e9b7-48e2-ba0e-3b37e8e36c69
:END:
https://www-nature-com-s.caas.cn/articles/s41568-021-00408-3?error=cookies_not_supported&code=ad37411b-42f4-4689-9a3b-1605f1f40eaf
https://www.sciencedirect.com/science/article/pii/S2666379122000258?via%3Dihub
https://www.nature.com/articles/s41421-021-00366-y
https://www.nature.com/articles/s41568-021-00408-3
cite:bakhoum2021
https://www.nature.com/articles/s41591-018-0323-0
https://pubmed.ncbi.nlm.nih.gov/34172972/ https://umgear.org/
https://www.nature.com/articles/d41586-021-01994-w
- cite:tarazona2020
- https://www.sciencedirect.com/science/article/pii/S2001037020302816#s0030
- https://workflows.omnipathdb.org/
- https://academic.oup.com/bioinformatics/article/36/8/2632/5689817
- cite:higdon2015
- https://jme.bioscientifica.com/view/journals/jme/62/1/JME-18-0055.xml
- http://mixomics.org/webinar-pls-methods/
- http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3618897&blobtype=pdf
- https://danielhomola.com/assets/DanielHomola_PhD.pdf
- cite:bersanelli2016
- cite:rappoport2018
- cite:subramanian2020
- cite:wu2019
- https://link.springer.com/protocol/10.1007%2F978-1-0716-0849-4_12
***** example papers
- https://www.sciencedirect.com/science/article/pii/S2666379122000258?via%3Dihub
- https://academic.oup.com/nar/article/47/19/10010/5573547
***** Omics integration
:PROPERTIES:
:ID:       00e4056a-e9b7-48e2-ba0e-3b37e8e36c69
:END:
https://www-nature-com-s.caas.cn/articles/s41568-021-00408-3?error=cookies_not_supported&code=ad37411b-42f4-4689-9a3b-1605f1f40eaf
https://www.sciencedirect.com/science/article/pii/S2666379122000258?via%3Dihub
https://www.nature.com/articles/s41421-021-00366-y
https://www.nature.com/articles/s41568-021-00408-3
cite:bakhoum2021
https://www.nature.com/articles/s41591-018-0323-0
https://pubmed.ncbi.nlm.nih.gov/34172972/ https://umgear.org/
https://www.nature.com/articles/d41586-021-01994-w
- cite:tarazona2020
- https://www.sciencedirect.com/science/article/pii/S2001037020302816#s0030
- https://workflows.omnipathdb.org/
- https://academic.oup.com/bioinformatics/article/36/8/2632/5689817
- cite:higdon2015
- https://jme.bioscientifica.com/view/journals/jme/62/1/JME-18-0055.xml
- http://mixomics.org/webinar-pls-methods/
- http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3618897&blobtype=pdf
- https://danielhomola.com/assets/DanielHomola_PhD.pdf
- cite:bersanelli2016
- cite:rappoport2018
- cite:subramanian2020
- cite:wu2019
- https://link.springer.com/protocol/10.1007%2F978-1-0716-0849-4_12
****** example papers
- https://www.sciencedirect.com/science/article/pii/S2666379122000258?via%3Dihub
- https://academic.oup.com/nar/article/47/19/10010/5573547
**** Pathway encoding and diagramming

:PROPERTIES:
:ID:       d84cf649-32f4-46e6-b041-4a8a76b31ffc
:END:
:LOGBOOK:
CLOCK: [2022-02-22 Tue 10:11]--[2022-02-22 Tue 11:59] =>  1:48
CLOCK: [2022-02-22 Tue 08:00]--[2022-02-22 Tue 09:45] =>  1:45
<<<<<<< HEAD
:END:
- https://www.nature.com/articles/nrg3885
- SBGN
  - https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005740
  - https://sbgn.github.io/
  - https://www.cls.uni-konstanz.de/software/vanted/
- https://new.wikipathways.org/academy/
- https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Pathway-Modeling
- 25 years of pathways
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02181-2#:~:text=In%20the%20end%2C%2065k%20pathway,in%20WikiPathways%20or%20Reactome%20databases.
  - https://gladstone-bioinformatics.shinyapps.io/shiny-25years/
- [[id:784895e5-0c94-4458-9cfb-f52759718ab3][NDEx]]
- cite:wikipathways
- [[id:69bfa728-e149-4b8a-9863-347630f86bc4][wikipathways biotool]]
- [[id:17c865dc-765c-496d-b5c7-60d3d3439de3][pathvisio biotool]]
**** RNA-seq for mitochondrial transcripts

**** Sequencing testing, modeling, and read simulation
:PROPERTIES:
:ID:       f44b9dce-d7fa-4b6a-9949-49ff3c02547c
:END:
- Read simulation
  - https://medium.com/computational-biology/a-simple-introduction-to-read-simulators-bbeff4f0c0c6
  - RNA-seq reads https://rdrr.io/bioc/polyester/
  - https://github.com/lh3/wgsim
- variant simulator http://bioinform.github.io/varsim/
- Downsample
  - at fastq
    - https://www.biostars.org/p/6544/
  - at bam
    - subsample function for reads instead of fraction
- Reference
  - [[https://www.reddit.com/r/bioinformatics/comments/c313uk/what_are_some_common_tasks_that_i_should_brush_up/][reddit: What are some common tasks that I should brush up on, that pertain to NGS?]]
# #https://www.biostars.org/p/76791/
# function SubSample {
# FACTOR=$(samtools idxstats $1 | cut -f3 | awk -v COUNT=$2 'BEGIN {total=0} {total += $1} END {print COUNT/total}')
# }
# SubSample $localdata/bams/lib080.dedup.bam
# if [[ $FACTOR > 1 ]]
#   then
#   echo '[ERROR]: Requested number of reads exceeds total read count in' $1 '-- exiting' && exit 1
# fi

# sambamba view -s $FACTOR -t 2 -f bam -l 5 $1

# }

# export -f SubSample

# ls *.bam | parallel "SubSample {} 50000000 > {.}_subsampled.bam"

    - https://bioinformatics.stackexchange.com/questions/402/how-can-i-downsample-a-bam-file-while-keeping-both-reads-in-pairs
    - (no pair preservation) https://www.biostars.org/p/44527/#126428 https://www.biostars.org/p/145820/
    - w/ sambamba # bam_sample(){
#     base=`basename -s .bam $1`
#     sambamba view -f sam -t 30 $1 | shuf -n $2 > $3/${base}_${2}_sample
# }

***** modeling
:PROPERTIES:
:ID:       00983e26-ec16-4f68-be0d-2f59eb31def4
:END:
- http://www.nytimes.com/2012/07/21/science/in-a-first-an-entire-organism-is-simulated-by-software.html?_r=2
- https://www.ebi.ac.uk/biomodels-main/
- https://precision.fda.gov/challenges/
- Synthetic biology- cite:caschera2014
***** Generating Simulated Reads
:PROPERTIES:
:ID:       8877C684-C217-4E7C-8E19-3D972A4015B2
=======
:END:
- https://www.nature.com/articles/nrg3885
- SBGN
  - https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005740
  - https://sbgn.github.io/
  - https://www.cls.uni-konstanz.de/software/vanted/
- https://new.wikipathways.org/academy/
- https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Pathway-Modeling
- 25 years of pathways
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02181-2#:~:text=In%20the%20end%2C%2065k%20pathway,in%20WikiPathways%20or%20Reactome%20databases.
  - https://gladstone-bioinformatics.shinyapps.io/shiny-25years/
- [[id:784895e5-0c94-4458-9cfb-f52759718ab3][NDEx]]
- cite:wikipathways
- [[id:69bfa728-e149-4b8a-9863-347630f86bc4][wikipathways biotool]]
- [[id:17c865dc-765c-496d-b5c7-60d3d3439de3][pathvisio biotool]]
**** Simulate sequencing reads
#+begin_src bash
cd ~/data
source ~/repos/mpnst/bin/setup.sh
docker_interactive
bioinformatics-toolkit
which simNGS
cd /home/jeszyman/data
ls
mkdir -p /home/jeszyman/data/simNGS-runfiles
wget --directory-prefix=/home/jeszyman/data/ref/simNGS-runfiles \
     --no-check-certificate \
     --no-directories \
     --accept "*.runfile" \
     --recursive --level=3 \
     --no-clobber https://www.ebi.ac.uk/goldman-srv/simNGS/runfiles5/HiSeq/
cat /home/jeszyman/data/ref/hg19.fa | simNGS \
                                          --paired paired /home/jeszyman/data/ref/simNGS-runfiles/s_4_4x.runfile > test.fq

<<<<<<< HEAD
- http://www.blopig.com/blog/2018/03/running-jupyter-notebook-on-a-remote-server-via-ssh/
- https://www.reddit.com/r/bioinformatics/comments/7gxsk0/does_anyone_use_cwl_does_it_actually_help_you_get/
- https://cromwell.readthedocs.io/en/stable/
- nextflow
  - https://www.youtube.com/watch?v=jsxTC8pNPUc
    - workflows are fragile
  - https://www.nextflow.io/example1.html
  - https://github.com/nf-core/rnaseq
  - https://www.ncbi.nlm.nih.gov/pubmed/?term=nextflow
- [[https://software.broadinstitute.org/wdl/userguide/][widdle]]
- http://rabix.io/
- http://bioboxes.org/docs/example-biobox-use/
****** Computer clusters
=======
#+end_src
**** Transcription factor binding site prediciton
https://www.nature.com/articles/s41592-019-0511-y
**** Variant interpretation
https://pubmed.ncbi.nlm.nih.gov/36373660/
https://www.nature.com/articles/s43018-022-00379-w
**** Whole-genome sequencing
:PROPERTIES:
:ID:       ccab4617-0399-4ba3-828e-b5fbbb972ba1
:END:
https://singularity.lbl.gov/install-linux
*** Workflow languages
https://www.google.com/search?q=convert+snakemake+to+nextflow&oq=convert+snakemake+to+nextflow&aqs=chrome..69i57j33i160l2.20088j0j7&client=ms-android-hmd-rvo3&sourceid=chrome-mobile&ie=UTF-8
https://www.reddit.com/r/bioinformatics/comments/15gut08/its_almost_2024_what_languageframework_are_you/
*** Biomarkers
https://www.nature.com/articles/s41467-023-40012-7
**** Visualizations
:PROPERTIES:
:ID:       c0359d9c-246a-41fb-a9b4-05ea5cb546ab
:END:
- [[file:~/repos/biotools/biotools.org::*Bioinformatics visualizations][Bioinformatics visualizations]]
https://www.nature.com/articles/s41467-023-38407-7
*** Bioinformatics visualization
:PROPERTIES:
:ID:       DB8463CB-97CB-49E6-8EDD-A5ECA2F140BF
:END:
:LOGBOOK:
CLOCK: [2021-06-05 Sat 09:36]--[2021-06-05 Sat 09:39] =>  0:03
:END:
**** Unfiled
https://www.nature.com/articles/s43018-023-00572-5
https://www.reddit.com/r/bioinformatics/comments/tgzeya/t_sne_graphs_for_data_visualization/
- ideogram /karyotye https://bioinformatics.stackexchange.com/questions/3600/how-to-make-chromosome-color-maps-for-bed-ranges
- GEDI SOMs https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007034#sec002
- https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/cpz1.252
- https://genviz.org/module-03-genvisr/0003/02/01/waterfall_GenVisR/
- upset
    - https://www.nature.com/articles/nmeth.3033
    - https://github.com/hms-dbmi/UpSetR
- https://cran.r-project.org/web/packages/chromoMap/vignettes/chromoMap.html#Data-based_annotation_plots
- https://sites.google.com/view/williamgilks/computational-analysis-of-whole-genome-sequencing-data/flow-chart-in-latex
- https://www.photopea.com/
- General reference
  - http://genviz.org/
- By content
  - pathvisio https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  - svviz [[http://svviz.readthedocs.io/en/latest/install.html][documentation]]
  - https://genome.sph.umich.edu/wiki/LocusZoom_Standalone
  - http://miso.readthedocs.io/en/fastmiso/sashimi.html
  - https://www.fusioncharts.com/resources/chart-primers/radar-chart
  - https://github.com/Adrienrougny/sbgntikz/
- [[id:1DA3ABC0-79B0-4CED-B926-22B37840379D][LINK TO Data Visualization]]
  - Examples
    - https://cmdcolin.github.io/awesome-genome-visualization/
  - https://bioinfogp.cnb.csic.es/tools/venny/
- Experimental design [[./biotools.org_20220817_135339.png]]

https://www.sciencedirect.com/science/article/pii/S1074761320305045?via%3Dihub
- tikz sbgn diagrams: https://github.com/Adrienrougny/sbgntikz/
- https://www.reddit.com/r/bioinformatics/comments/chnt75/guidelines_for_workflow_visualization/
https://www.reddit.com/r/bioinformatics/comments/tgzeya/t_sne_graphs_for_data_visualization/
- ideogram /karyotye https://bioinformatics.stackexchange.com/questions/3600/how-to-make-chromosome-color-maps-for-bed-ranges
- GEDI SOMs https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007034#sec002
- https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/cpz1.252
- https://genviz.org/module-03-genvisr/0003/02/01/waterfall_GenVisR/
- upset
    - https://www.nature.com/articles/nmeth.3033
    - https://github.com/hms-dbmi/UpSetR
- https://cran.r-project.org/web/packages/chromoMap/vignettes/chromoMap.html#Data-based_annotation_plots
- https://sites.google.com/view/williamgilks/computational-analysis-of-whole-genome-sequencing-data/flow-chart-in-latex
- https://www.photopea.com/
- General reference
  - http://genviz.org/
- By content
  - pathvisio https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  - svviz [[http://svviz.readthedocs.io/en/latest/install.html][documentation]]
  - https://genome.sph.umich.edu/wiki/LocusZoom_Standalone
  - http://miso.readthedocs.io/en/fastmiso/sashimi.html
  - https://www.fusioncharts.com/resources/chart-primers/radar-chart
  - https://github.com/Adrienrougny/sbgntikz/
- [[id:1DA3ABC0-79B0-4CED-B926-22B37840379D][LINK TO Data Visualization]]
  - Examples
    - https://cmdcolin.github.io/awesome-genome-visualization/
  - https://bioinfogp.cnb.csic.es/tools/venny/


**** Heatmaps
:PROPERTIES:
:ID:       a259c668-737b-4ca3-9994-a9ce8afa01db
:END:
- https://academic.oup.com/bioinformatics/article/32/18/2847/1743594?login=true
- https://slowkow.com/notes/pheatmap-tutorial/#quantile-breaks
- https://jokergoo.github.io/2020/05/06/translate-from-pheatmap-to-complexheatmap/
- https://slowkow.com/notes/pheatmap-tutorial/
- heatmaps
  - https://www.biostars.org/p/287512/
  - https://support.bioconductor.org/p/93424/
- https://jokergoo.github.io/2020/05/06/translate-from-pheatmap-to-complexheatmap/
- cite:lucas2020 fig3c
**** <<<Integrated Genomics Viewer>>> (<<<IGV>>>)
:PROPERTIES:
:ID:       917062AC-32E8-41F5-A511-84AA7B1504B0
:END:

- cite:robinson2017
https://www.france-bioinformatique.fr/sites/default/files/EBA/V3-2014/Elodie_Giraud_visualisation-IGV_TP_EBA-2014-10.pdf
[[http://software.broadinstitute.org/software/igv/UserGuide][User Guide]]



*** [[id:0f1f4aa4-c755-42ff-8f81-8a32c14293a0][Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)]]
*** Bioinformatics pipelines and project management
- https://osf.io/4pd9n/
- https://www.reddit.com/r/bioinformatics/comments/12cb99z/what_are_some_good_examples_of_wellengineered/
- https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000424
- http://byronjsmith.com/make-bml/
*** Chromatin profiling
- https://bioconductor.org/packages/release/BiocViews.html#___ChIPSeq
*** Common bioinformatics file manipulation
:PROPERTIES:
:ID:       c0c0ee28-2e41-41a7-9a3b-ae195117a93e
:END:
- https://bioinf.shenwei.me/seqkit/usage/#pair
- https://www.google.com/search?q=annotate+bed+file+with+gene+names
- https://biodbnet-abcc.ncifcrf.gov/db/db2db.php
- [[https://www.biostars.org/p/68283/][count total length of regions in bed file]]
- https://gatkforums.broadinstitute.org/gatk/discussion/6484/how-to-generate-an-unmapped-bam-from-fastq-or-aligned-bam
- https://www.biostars.org/p/247903/
- https://gatkforums.broadinstitute.org/gatk/discussion/2909/how-to-fix-a-badly-formatted-bam
- [[https://github.com/lh3/seqtk][seqtk]]
- https://gist.github.com/davfre/8596159
**** Subsetting and splitting
- Fastas
  - https://www.biostars.org/p/13270/
  - Fastqs
    - [[Question: Selecting Random Pairs From Fastq?]]
    - [[https://www.biostars.org/p/6544/][look at all the ways to split a fastq file!]]
- Bams
  - https://bioinformatics.stackexchange.com/questions/402/how-can-i-downsample-a-bam-file-while-keeping-both-reads-in-pairs
  - https://www.google.com/search?q=subset+bam+file
  - samtools view -bo subset.bam -s 123.4 alignments.bam chr1 chr2
  function SubSample {
  #+begin_src bash :tangle ./toosl/shell/subsample_bam.sh
function SubSample {
## Calculate the sampling factor based on the intended number of reads:
FACTOR=$(samtools idxstats $1 | cut -f3 | awk -v COUNT=$2 'BEGIN {total=0} {total += $1} END {print COUNT/total}')

if [[ $FACTOR > 1 ]]
  then
  echo '[ERROR]: Requested number of reads exceeds total read count in' $1 '-- exiting' && exit 1
fi

sambamba view -s $FACTOR -f bam -l 5 $1

}
#+end_src
**** Database downloads
- https://www.google.com/search?q=download+genbank+fasta
*** DNA variants
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890431-The-logic-of-joint-calling-for-germline-short-variants

https://pubmed.ncbi.nlm.nih.gov/35311178/
**** DNA variants
https://pubmed.ncbi.nlm.nih.gov/35311178/
**** variant calling
>>>>>>> bb11343ab3ff683a4ece1c261661becfbc0b872d
:PROPERTIES:
:CREATED:  [2020-06-22 Mon 12:27]
:ID:       a5b408eb-77e2-4634-a5bb-950cefaafe18
:END:
- [[id:57E81A44-3233-479B-8264-2CD5F1904A60][Sequence variation and variant calling]]
***** fusions
:PROPERTIES:
:ID:       0ea6c74e-bcf8-44c2-b4c3-e725d930d4a3
:END:
  - cite:melsted2017
****** Gene Fusion Detection
:PROPERTIES:
:ID:       492e0fad-ced4-407b-8913-80f3a0ac1202
:END:
:PROPERTIES:
:ID:       778C7C81-34E7-4BAE-94F9-5C33318BFDA5
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110
:END:
- [[https://github.com/jeszyman/rna-fusion][repository]]
- [ ] index script
- [ ] align script
- reference
  - https://www.nextflow.io/example4.html
  - gsnap example- - http://www.cureffi.org/2013/11/18/an-mrna-seq-pipeline-using-gsnap-samtools-cufflinks-and-bedtools/
  - https://github.com/genepattern/docker-tophat/blob/master/Dockerfile
  - https://github.com/broadinstitute/gtex-pipeline/blob/master/rnaseq/Dockerfile


********* NEXT INTEGRATE RNA aligners
:PROPERTIES:
:CREATED:  [2019-02-25 Mon 13:16]
:ID:       951F10BC-E844-45C5-8CB4-4FA8756D9F85
:END:

- are combined rna-seq alignment methods just sum of runs?
- cite:zhang2016a
- [-] install
  - [ ] cite:wu2016 in cite:mathe2016
  - [X] hisat2
  - [X] star
  - [X] tophat - but prefer don't use- "TopHat has entered a low maintenance, low support stage as it is now largely superseded by HISAT2" https://ccb.jhu.edu/software/tophat/index.shtml
- [-] create indexes
  - [-] star
  - [ ] hisat2 https://ccb.jhu.edu/software/hisat2/manual.shtml see hista2-build
  -

******** NEXT HCC1395 straight align and integrate and compare to published
:PROPERTIES:
:CREATED:  [2019-03-13 Wed 10:11]
:ID:       58F7BED2-DED1-4911-9E8C-104E12126A73
:END:
:LOGBOOK:
CLOCK: [2019-03-14 Thu 07:49]--[2019-03-14 Thu 08:45] =>  0:56
CLOCK: [2019-03-13 Wed 18:18]--[2019-03-13 Wed 19:54] =>  1:36
CLOCK: [2019-03-13 Wed 11:30]--[2019-03-13 Wed 12:00] =>  0:30
CLOCK: [2019-03-13 Wed 11:00]--[2019-03-13 Wed 11:01] =>  0:01
CLOCK: [2019-03-13 Wed 10:12]--[2019-03-13 Wed 10:47] =>  0:35
:END:
- log
  - from bam to fastq
    - by sample
      - tumor
        - ./sambamba-0.6.8-linux-static sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam /gscmnt/gc2555/spencer/jeff/data/gerald_C1TD1ACXX_8_ACAGTG.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam
      - moved all this to ~data/HCC1395/
      - normal
        - (integrate image, 64gb) sambamba sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal-namesort.bam /gscmnt/gc2555/spencer/jeff/data/HCC1395/gerald_C2DBEACXX_3.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal.namesort.bam
        -
  - alignment
    - bsub -Is -q docker-interactive -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/startest)' /bin/bash
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_2.fastq --outSAMunmapped Within
  - integrate
    - bsub -q research-hpc -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/integrate)' /bin/bash
    - prep each star output:
      -
      - sambamba view -S -o Aligned.out.bam Aligned.out.sam
      - sambamba sort Aligned.out.bam
      - sambamba index Aligned.out.sorted.bam
    - Integrate fusion /gscmnt/gc2555/spencer/jeff/data/reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa /gscmnt/gc2555/spencer/jeff/data/reference/annot-hg38-refeseq.txt /gscmnt/gc2555/spencer/jeff/data/reference/Integrate/bwts /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam
      -

***** Broad Calling variants in RNAseq
:PROPERTIES:
:ID:       DA64BC88-8BD6-40DF-9D80-5596BFA4D643
:END:

https://software.broadinstitute.org/gatk/documentation/article?id=3891
***** RNA-fusion                                                    :biopipe:
:PROPERTIES:
:CREATED:  [2019-06-25 Tue 09:05]
:ID:       96634f32-b19a-4390-8e50-e942abd35f90
:END:
:LOGBOOK:
CLOCK: [2019-07-09 Tue 09:19]--[2019-07-09 Tue 09:25] =>  0:06
CLOCK: [2019-07-02 Tue 08:20]--[2019-07-02 Tue 09:45] =>  1:25
CLOCK: [2019-06-27 Thu 08:50]--[2019-06-27 Thu 09:31] =>  0:41
CLOCK: [2019-06-27 Thu 08:17]--[2019-06-27 Thu 08:31] =>  0:14
CLOCK: [2019-06-26 Wed 13:35]--[2019-06-26 Wed 13:59] =>  0:24
:END:
- [[https://github.com/jeszyman/rna-fusion][repository]]
- [ ] index script
- [ ] align script
- reference
  - https://www.nextflow.io/example4.html
  - gsnap example- - http://www.cureffi.org/2013/11/18/an-mrna-seq-pipeline-using-gsnap-samtools-cufflinks-and-bedtools/
  - https://github.com/genepattern/docker-tophat/blob/master/Dockerfile
  - https://github.com/broadinstitute/gtex-pipeline/blob/master/rnaseq/Dockerfile

********* NEXT INTEGRATE RNA aligners
:PROPERTIES:
:CREATED:  [2019-02-25 Mon 13:16]
:ID:       951F10BC-E844-45C5-8CB4-4FA8756D9F85
:END:

- are combined rna-seq alignment methods just sum of runs?
- cite:zhang2016a
- [-] install
  - [ ] cite:wu2016 in cite:mathe2016
  - [X] hisat2
  - [X] star
  - [X] tophat - but prefer don't use- "TopHat has entered a low maintenance, low support stage as it is now largely superseded by HISAT2" https://ccb.jhu.edu/software/tophat/index.shtml
- [-] create indexes
  - [-] star
  - [ ] hisat2 https://ccb.jhu.edu/software/hisat2/manual.shtml see hista2-build
  -

******** NEXT HCC1395 straight align and integrate and compare to published
:PROPERTIES:
:CREATED:  [2019-03-13 Wed 10:11]
:ID:       58F7BED2-DED1-4911-9E8C-104E12126A73
:END:
:LOGBOOK:
CLOCK: [2019-03-14 Thu 07:49]--[2019-03-14 Thu 08:45] =>  0:56
CLOCK: [2019-03-13 Wed 18:18]--[2019-03-13 Wed 19:54] =>  1:36
CLOCK: [2019-03-13 Wed 11:30]--[2019-03-13 Wed 12:00] =>  0:30
CLOCK: [2019-03-13 Wed 11:00]--[2019-03-13 Wed 11:01] =>  0:01
CLOCK: [2019-03-13 Wed 10:12]--[2019-03-13 Wed 10:47] =>  0:35
:END:
- log
  - from bam to fastq
    - by sample
      - tumor
        - ./sambamba-0.6.8-linux-static sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam /gscmnt/gc2555/spencer/jeff/data/gerald_C1TD1ACXX_8_ACAGTG.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam
      - moved all this to ~data/HCC1395/
      - normal
        - (integrate image, 64gb) sambamba sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal-namesort.bam /gscmnt/gc2555/spencer/jeff/data/HCC1395/gerald_C2DBEACXX_3.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal.namesort.bam
        -
  - alignment
    - bsub -Is -q docker-interactive -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/startest)' /bin/bash
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_2.fastq --outSAMunmapped Within
  - integrate
    - bsub -q research-hpc -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/integrate)' /bin/bash
    - prep each star output:
      -
      - sambamba view -S -o Aligned.out.bam Aligned.out.sam
      - sambamba sort Aligned.out.bam
      - sambamba index Aligned.out.sorted.bam
    - Integrate fusion /gscmnt/gc2555/spencer/jeff/data/reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa /gscmnt/gc2555/spencer/jeff/data/reference/annot-hg38-refeseq.txt /gscmnt/gc2555/spencer/jeff/data/reference/Integrate/bwts /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam
      -

*** Epigenetics
:PROPERTIES:
:CREATED:  [2019-10-30 Wed 10:52]
:ID:       079e398f-e3b0-49c4-92da-9b4f6a73684c
:END:
- [[id:32afbe10-0469-4969-8b8e-1ee14b4101d2][Science of cardiac epigenetics]]
- cite:smith2017
- https://www.nature.com/articles/nature14248?error=cookies_not_supported&code=72ccf3be-3226-467c-b6a1-53bbfe374e27#Sec12
- https://www.illumina.com/destination/epigenetics-webinar-series.html

- [[id:8ca221b7-b51a-4382-9844-5d8ba7193ff6][Chromatin accessibility]]
- epigenetics
    - ATAC-seq for chromatin accessiblity
    - ChIP-seq
    - differential peak analysis-
      - epigen-specific tools DiffBind, MAnorm
      - uses RNA-seq tools for differential expression
    - peak-calling: identifying enriched signal from ATAC-seq and ChIP-seq, see cite:liu2019a ref 40
      - MACS/MACS2
      - measures of signal-to-noise by ENCODE
        - FRiP fraction of reads in peaks
        - cross-correlation profiles
    - methylation profiling
      - BSmooth
      - Merman
      - single cell
        - Minifi
      - bisulfite- unmethylated cytosines converted to uralcil, uracil converted to thyamine during amp- ratio T to C
      - methylcap-seq
      - differential peak
        - DiffBind
        - MAnorm
        - DBChIP
        - DAStk
      - Minfi
      - BSmooth
      - methylKit
      - mehtySig
      - metilene
**** ChIP-seq
:PROPERTIES:
:ID:       4c8cbcad-4523-4883-b041-a8eb548f01ac
:END:
https://rockefelleruniversity.github.io/RU_ChIPseq/
https://github.com/crazyhottommy/ChIP-seq-analysis/blob/master/part2_Preparing-ChIP-seq-count-table.md
https://www.bioconductor.org/help/course-materials/2016/CSAMA/lab-5-chipseq/Epigenetics.html
http://www.guertinlab.org/vignettes/
- https://github.com/crazyhottommy/ChIP-seq-analysis
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5429005/
- https://nbisweden.github.io/workshop-archive/workshop-ChIP-seq/2018-11-07/
https://github.com/crazyhottommy/ChIP-seq-analysis
**** [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
**** Methylation sequencing                                         :biopipe:
:PROPERTIES:
:CREATED:  [2019-05-29 Wed 14:46]
:ID:       b8db46d3-8329-4fef-82b1-4d99c2c9312b
:END:
- https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/medip-seq-dip-seq.html
- cite:shen2018
- cite:giacopelli2021prostate
- cite:liu2020
- https://www.nature.com/articles/s41591-020-0933-1
- https://www.nature.com/articles/nm.4045
- [[id:6ad7cfcb-1e8b-4d72-b18a-40f80a00f242][nih liq biopsy]] see [2021-01-26 Tue]
- "All the cell line datasets generated during and/or analysed during the current study are available in the GEO repository under accession code GSE79838. The
- cfMeDIP-seq NGS data for patient samples that support the findings of this study are available upon request from the corresponding author (D.D.C) to comply with
- to add
  - R
    - MethylKit
    - DESeq2
    - caret
    - minfi
    - Rtsne
  - SAMtools
  - Bowtie
  - MEDIPS
  - BWA
  - GATK IndelRealigner
  - MuTect
  - TrimGalore
  - Bismark6
  - Bowtie
  -
- from cite:shen2018
  - "The generated SAM files from hg19 alignment were converted to BAM format, ensuring the removal of duplicate reads, and the reads were then sorted and indexed using SAMtools21 before subsequent analysis with the R package MEDIPS22. The CpG enrichment score, as a quality control measure for the immunoprecipitation reaction, was calculated as part of the MEDIPS package."
  - Shell scripts were used to process MeDIP sequencing data to produce RPKM estimates for downstream analysis. Custom R scripts with a collection of R/Bioconductor packages were used to analyse the resulting dataset. The code used for analysis is available in the form of R markdowns deposited onto Zenodo as detailed in Supplementary Table 13.
  -
  | Description                                                                                    | Zenodo DOI for Data    | Zenodo DOI for Code    |
  | <25>                                                                                           | <25>                   | <25>                   |
  | IHEC Figure                                                                                    | 10.5281/zenodo.1205714 |                        |
  | Simulations                                                                                    | 10.5281/zenodo.1205730 | 10.5281/zenodo.1205728 |
  | Extended Data Fig. 6 violin plots                                                              | 10.5281/zenodo.1205744 | 10.5281/zenodo.1205735 |
  | Bed file of genomic windows that contain CpGs                                                  | 10.5281/zenodo.1205752 |                        |
  | TCGA and COSMIC tSNEs                                                                          | 10.5281/zenodo.1205754 |                        |
  | Machine learning                                                                               |                        | 10.5281/zenodo.1242697 |
  | Markdown with ensemble of non-machine learning scripts including Extended Data Fig. 6 heatmaps |                        | 10.5281/zenodo.1205756 |
  | Stand alone scripts called by non-machine learning markdown                                    |                        | 10.5281/zenodo.1205746 |
-
- Open source code/packages used for data analysis: R, SAMtools (v. 1.3.1), Bowtie (v. 0.12.9), MEDIPS (v. 1.22.0), BWA (v. 0.7.1220),
- Genome Analysis ToolKit (GATK) IndelRealigner, MuTect (v.1.1.5) , Trim Galore! (v. 0.4.4), Bismark (v. 0.10.1), Bowtie2 (v. 2.0.5),
- MethylKit (v 0.99.2), DESeq2 (v. 1.4.5), caret R package, minfi bioconductor package, Rtsne

***** Bisulfite sequencing
  - methylation protects/blocks cytosine conversion to uracil
  - PCR amp can produce forward/reverse with and without bisulfite block
  - BSMAP, Bismark, BWA-meth, biscuit
  - VCF > bedgraph > bigwig > IGV
  - metilene segments methylation
  - visualization
    - heatmap
    - canyon plot- methylation proportion vs distance from target locus

**** protein-DNA interactions
:PROPERTIES:
:CREATED:  [2021-01-18 Mon 09:45]
:ID:       b612a1b1-2c01-49c3-a20c-5a890fa1a6e5
:END:
- cut and run - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6302505/
  - cite:bakhoum2021
**** transcription factor enrichment
:PROPERTIES:
:ID:       809e1a39-2a8e-46ec-863c-d7b03511fcfb
:END:
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4009616/
- https://science.sciencemag.org/content/352/6288/aad3018
- motif-based- high false pos
- benchtop with ChIP-seq
**** Chromatin accessibility profiling
:PROPERTIES:
:ID:       ae0363f3-379e-4521-9184-cbc0d46501cd
:END:
- cite:minnoye2021
- [[id:8ca221b7-b51a-4382-9844-5d8ba7193ff6][Chromatin accessibility biology]]
- [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
- [[id:17f978bc-f2f2-497d-a8f1-b1d47927380f][Differential chromatin accessibility]]
- chromhhm track annotation
  - https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm10&g=encode3RenChromHmm
  - http://enhancer.sdsc.edu/enhancer_export/ENCODE/chromHMM/pooled/
***** Region or locus-based enrichment analysis
:PROPERTIES:
:ID:       12393aa3-d318-4ba6-8183-6e3a175d3ba4
:END:
- GREAT http://bejerano.stanford.edu/great/public/html/index.php
- Lola
  - https://academic.oup.com/bioinformatics/article/32/4/587/1743969
  - https://bioconductor.org/packages/release/bioc/vignettes/LOLA/inst/doc/usingLOLACore.html
***** Chromatin accessibility profiling methods
:PROPERTIES:
:ID:       f600cd95-723e-4893-8b3f-0634d8905920
:END:
- cite:minnoye2021
- [[id:22e31d06-f5df-427e-bd70-3a2ccd3f47ec][ATAC-seq]]
- [[id:8ca221b7-b51a-4382-9844-5d8ba7193ff6][Chromatin accessibility]]
****** Chromatin accessibility profiling methods

https://www.nature.com/articles/s41588-018-0091-2
************* Chemical- NextBio, see Webster,15ref Kupershmidt,10
:PROPERTIES:
:ID:       f9f486c9-cd9b-401a-a3c0-523a5d09d92b
:END:
  - extracellular
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5426982/
  - stranded
    - https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work
  - ffpe
    - cite:kashofer2013
    - FFPE Pre-analytic variability
      - cite:ahlfen2007rnaqual
        - storage conditions
        - cold ischemia time
        - fixation
          - type
          - time
        - sample thickness
      - biospecimens.cancer.gov/researchnetwork
      - cite:unlockingQiagen
      - https://www.ncbi.nlm.nih.gov/pubmed/25357115
      - By effect
        - Yield
          - - prolonged fix reduces yield- Bresters D, Schipper ME, Reesink HW, Boeser-Nunnink BD, Cuypers HT. The duration of fixation influences the yield of HCV cDNA-PCR products from formalin-fixed, paraffin- embedded liver tissue. J Virol Methods 1994;48:267–72.
          - cite:webster2015mining longer fixation times decrease RNA yields from paraffin-embedded tissues (herein 3 weeks in formalin resulted in half the RNA yield/sec- tion compared with 18 h) and increase RNA input needed during library preparation (see methods). This latter effect is likely due to ‘non-functional’ RNA (eg, adducts or damaged bases) that could be potentially mitigated by treatments like PreCR or other enzymatic and chemical methods.
        - At microarray expression
          - cite:kashofer2013quality
      - "Formalin fixation introduced major changes into microarray gene expression data and led to marked gene-to-gene variations in delta-ct values of qRT-PCR. We found that qRT-PCR efficiency and gene-to- gene variations were mainly attributed to differences in the efficiency of cDNA synthesis as the most sensitive step. These differences could not be reliably detected by quality assessment of total RNA isolated from formalin-fixed tissues by electrophoresis or spectrophotometry. "
      - At sequencing
    - cite:graw2015
    - Past expression tech
      - goals were biomarker discovery and patient stratification
      - cite:Zhao_2014 manifests its superiority in terms of the dynamic range, sampling depth, and has independence from pre-existing sequence information [1,2].
      - cite:webster2015mining When compared with conventional DNA microarrays, RNA-seq enriches for many additional fragments as it is not restricted to predefined probes and has (in principle) no limitations to dynamic range (Li et al., 2014b; SEQC/MAQC-III Consortium, 2014).
      - - https://www.ncbi.nlm.nih.gov/pubmed/14695316
      - - [[https://www.ncbi.nlm.nih.gov/pubmed/15509548][DASL]]
      - https://www.ncbi.nlm.nih.gov/pubmed/21347257
    - FFPE microarray and RT-PCR was useful
      - AlloMap is a non-invasive gene expression-based blood test that is used to manage the clinical care of heart transplant recipients, providing a quantified score for the risk of rejection based on the measurement of expression of 20 genes, a subset of which are related to immune system activation and signalling15,16.
      - - cite:penland2007rna
      - - - https://www.ncbi.nlm.nih.gov/pubmed/11568892/
      - Mittempergher, L., de Ronde, J.J., Nieuwland, M., Kerkhoven, R.M., Simon, I., Th Rutgers, E.J., Wessels, L.F., Van't Veer, L.J., 2011. Gene expression profiles from formalin fixed paraffin embedded breast cancer tissue are largely comparable to fresh fro- zen matched tissue. PLoS One 6, e17163.
    - But this expression tech is inferior
      - In other work14, systematic evalu- ation of RNA-seq-based and microarray-based classifiers found that RNA-seq outperformed arrays in characteriz- ing the transcriptome of cancer and performed similarly to arrays in clinical endpoint prediction.
      - does not depend on genome annotation, or on predesigned species-specific or tran- script-specific probes, for transcript measurement.
      - microarray limited application to clinic- lthough powerful, microarray-based assays can have limitations in some environments, such as those related to laboratory-to-laboratory variation in sample preparation that can affect reproducibility. Moreover, for some applications, microarray signal-to-noise ratios can affect the limit of detection. Interestingly, a number of additional cancer multi-gene profile tests are clinically available, such as OncoTypeDX (Genome Health)135 for breast cancer recurrence risk and Prolaris (Myriad)136 for prostate cancer aggressiveness. These tests are based on qRT-PCR technologies, rather than microarrays, largely owing to the belief that qRT-PCR is more reliable, reproducible, sensitive and accurate.
      - " qRT-PCR is generally deemed the ‘gold standard’ method for measuring transcript levels, particularly in the clinical space; however, there are a number of limitations, including the fact that although it is a high sample throughput technology, relatively few markers or measurements can be made in a single assay. "
      - rnaseq superior for
        - snvs
        - indels
        - isoforms
        - splice vars
        - chimerics
          - fusions
      - semiquan
      - no SNV discovery
      - no fusion
      - "RNA-seq provides much more detailed information on specific transcript expression patterns in clinico-genetic neuroblastoma subgroups than microarrays. " cite:zhang2015comparison
  - small RNA sequencing
    - sample processing
      - circulating miRNA stability
        - serum cite:gautam2016
        - plasma  cite:glinge2017
    - library preparation
      - with low input, use same kit for reproducibility cite:yeri2018
      - cite:dard-dascot2018
    - https://rnajournal.cshlp.org/content/18/6/1279.long
    - qc
      - https://www.agilent.com/en/product/automated-electrophoresis/bioanalyzer-systems/bioanalyzer-rna-kits-reagents/bioanalyzer-small-rna-analysis-228257#howitworks
  - single cell
    - https://www.biorxiv.org/content/biorxiv/early/2019/02/05/541433.full.pdf
    - examples
      - - see fig 3 & methods https://genome.cshlp.org/content/27/2/208.full.pdf
    - https://www.nature.com/articles/s41587-020-0553-9
    - 10x
      - applications - https://pages.10xgenomics.com/nur-gen-research-area-literature.html?src=email&utm_medium=email&lss=nurture&utm_source=nurture&cnm=es-nurture-ra_g-general&utm_campaign=es-nurture-ra_g-general&utm_content=&utm_term=&useroffertype=marketing-literature&userresearcharea=ra_g&userregion=&userrecipient=customer&mktouserid=1265915&cid=&usercampaignid=
    - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0212031
    - http://www.nature.com/nbt/journal/v34/n2/full/nbt.3445.html
  - Clinical applications of RNA-seq
    - cite:byron2016translating
    - cite:Kalari_2014
    - cite:Van_Keuren_Jensen_2014
  - deconvolution
    - tools
      - ISOpure
        - https://link.springer.com/article/10.1186/s12859-015-0597-x#Sec1
    - cite:newman2015
  - time course
    - cite:spies2017
    - https://support.bioconductor.org/p/65676/#66860
    - https://www.biostars.org/p/138988/
    - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#group-specific-condition-effects-individuals-nested-within-groups
    - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
    - http://seqanswers.com/forums/showthread.php?t=64039
    - https://www.sciencedirect.com/science/article/pii/S2001037015000392
- [[*cell enumeration through bulk RNA-seq][cell enumeration through bulk RNA-seq]]
- https://www.rna-seqblog.com/optimization-of-mirna-seq-data-preprocessing/
- [[id:e4351ef3-6d41-4121-a2cc-c63d89f997ac][pathway analysis]]
- [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- [[id:57E81A44-3233-479B-8264-2CD5F1904A60][Sequence variation and variant calling]]


- [[*omics integration][omics integration]]
************* Allele-specific Expression
:PROPERTIES:
:ID:       96C367A3-FEEB-4F6F-975A-A08A172FD277
:END:


************* RNA-fusion                                            :biopipe:
:PROPERTIES:
:CREATED:  [2019-06-25 Tue 09:05]
:ID:       96634f32-b19a-4390-8e50-e942abd35f90
:END:
:LOGBOOK:
CLOCK: [2019-07-09 Tue 09:19]--[2019-07-09 Tue 09:25] =>  0:06
CLOCK: [2019-07-02 Tue 08:20]--[2019-07-02 Tue 09:45] =>  1:25
CLOCK: [2019-06-27 Thu 08:50]--[2019-06-27 Thu 09:31] =>  0:41
CLOCK: [2019-06-27 Thu 08:17]--[2019-06-27 Thu 08:31] =>  0:14
CLOCK: [2019-06-26 Wed 13:35]--[2019-06-26 Wed 13:59] =>  0:24

http://101.datascience.community/tag/octave/
http://www.cbioportal.org/cgds_r.jsp
https://precision.fda.gov/
http://www.russpoldrack.org/?m=1
https://www.biostars.org/u/3737/
https://bioinformatics.ca/workshops/2014/informatics-rna-sequence-analysis-bc-2014
[[https://www.bioconductor.org/help/course-materials/][Bioconductor courses and conferences]]
[[https://www.ebi.ac.uk/training/online/course/embo-practical-course-analysis-high-throughput-seq/introduction-r-and-bioconductor][EMBO introduction to R and Biocinductor]]
[[https://www.edx.org/course/introduction-bioconductor-annotation-harvardx-ph525-5x][edX Intro to bioconductor]]
[[https://www.edx.org/course/high-dimensional-data-analysis-harvardx-ph525-4x][edX high-dimensional data analysis]]
[[https://www.coursera.org/learn/bioconductor][coursera bioconductor for genomic data science]]
http://bcb.dfci.harvard.edu/~aedin/courses/Bioconductor/

https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html

************ Gene Set Enrichment
:PROPERTIES:
:ID:       87734999-3139-4343-A323-925416811B12
:END:
https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/#:~:text=The%20GSEA%20enrichment%20score%20(S,sets%20one%20at%20a%20time.
- https://www.reddit.com/r/bioinformatics/comments/66374x/gsea_vs_go_enrichment/

************* [[http://www.bioconductor.org/packages/release/bioc/html/gage.html][GAGE]] (generally applicable gene-set enrichment for pathway analysis)
:PROPERTIES:
:ID:       441F7686-9969-4F8D-8092-C4429377214C
:END:

[[file:~/Box%20Sync/comp_bio/gsea_template/][Folder]]
[[file:~/Box%20Sync/comp_bio/gsea_template/gage.pdf][Manual]]
[[file:~/Box%20Sync/comp_bio/gsea_template/gsea_template.R][Rscript]]

************** RNA-Seq Workflow Example
:PROPERTIES:
:ID:       B687F931-3A5E-4CD0-B2B7-30DF026E0AB9
:END:

[[file:~/Box%20Sync/comp_bio/gsea_template/][Folder]]
[[file:~/Box%20Sync/comp_bio/gsea_template/RNA-seqWorkflow.pdf][pdf]]


************** RNA-Seq Workflow Example
:PROPERTIES:
:ID:       B687F931-3A5E-4CD0-B2B7-30DF026E0AB9
:END:

[[file:~/Box%20Sync/comp_bio/gsea_template/][Folder]]
[[file:~/Box%20Sync/comp_bio/gsea_template/RNA-seqWorkflow.pdf][pdf]]

************* Commercial
:PROPERTIES:
:ID:       417416B5-0169-41CA-8FCE-4D6E440C7E3A
:END:

http://www.ingenuity.com/
[[http://genome.ucsc.edu/cgi-bin/hgTables?command=start][create your own GTF]]
[[http://bedops.readthedocs.io/en/latest/index.html][BEDOPS]]
https://www.datacamp.com/

************ Genotype-Phenotype Associations
:PROPERTIES:
:ID:       69F82D91-6DE6-4332-AD0A-7ADF25263858
:END:

https://www.nature.com/articles/s41467-019-11591-1
- cite:stuart2019
[[id:992de4dc-775e-41f0-a70a-e38c99b87265][Single cell methylation sequencing]]
https://www.reddit.com/r/bioinformatics/comments/uy9g1j/where_to_start_with_scrnaseq_data/
https://www.embopress.org/doi/full/10.15252/msb.20188746
- cite:storrs2022
- https://bioengineeringcommunity.nature.com/posts/dialogue-maps-multicellular-programs-in-tissue-from-single-cell-or-spatial-transcriptomics-data
https://www.nature.com/articles/nmeth.4612
https://satijalab.org/signac/articles/peak_calling.html
https://www.nature.com/articles/s41467-021-21583-9#Sec2
https://mobile.twitter.com/nomad421/status/1522554692202598403?s=20&t=UObG4oqigCp6gJIIgmyhPA
- https://www.sciencedirect.com/science/article/pii/S1097276516301861?via%3Dihub
***** ATAC-seq
:PROPERTIES:
:ID:       22e31d06-f5df-427e-bd70-3a2ccd3f47ec
:END:
****** [[id:6ca433ff-59ed-4f60-889f-c43e1f945064][Cis-regulatory elements]]
****** Method-establishing papers
- cite:buenrostro2015
- cite:corces2017
****** Method guides, tutorials, and good practice
- https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/cphg.101
- https://fonseca.lab.mcgill.ca/resources/20220810_ATAC_Analysis_Demo/guide.html
- https://rockefelleruniversity.github.io/RU_ATAC_Workshop.html
- https://github.com/maxplanck-ie/snakepipes/tree/master/snakePipes/workflows/ATAC-seq
- cite:reske20s20
  - https://github.com/reskejak/ATAC-seq
- https://github.com/crazyhottommy/pyflow-ATACseq/blob/master/Snakefile
- https://www.sciencedirect.com/science/article/pii/S2666166720302768
- http://www.guertinlab.org/vignettes/
- https://github.com/epigen-UCSD/atac_seq_pipeline/blob/master/pipeline.md
- https://www.hdsu.org/chipatac2020/04_CHIP_Alignment.html
- https://informatics.fas.harvard.edu/atac-seq-guidelines.html
- pipeline graphic https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-020-00342-y/figures/4
- cite:yan2020atac
- cite:corces2017
- [[https://informatics.fas.harvard.edu/atac-seq-guidelines.html][Harvard FAS]]
- [[https://www.scopus.com/results/results.uri?cc=10&sort=cp-f&src=s&nlo=&nlr=&nls=&sid=e3c13d36864e217e2edcf12df7a64b0a&sot=b&sdt=b&sl=59&s=%28TITLE-ABS-KEY%28atac-seq%29+AND+TITLE-ABS-KEY%28bioinformatics%29%29&ss=cp-f&ps=r-f&editSaveSearch=&origin=resultslist&zone=resultslist][scopus atac-seq bioinformatics (42)]]
- https://snakepipes.readthedocs.io/en/latest/content/workflows/ATAC-seq.html
- https://bcbio-nextgen.readthedocs.io/en/latest/contents/atac.html
- https://rockefelleruniversity.github.io/RU_ATAC_Workshop.html
- https://www.illumina.com/techniques/popular-applications/epigenetics/atac-seq-chromatin-accessibility.html
- http://biocworkshops2019.bioconductor.org.s3-website-us-east-1.amazonaws.com/page/ATACseqQCWorkshop__ATACseqQC_workshop/
- cite:minnoye2021
- https://www.encodeproject.org/atac-seq/
- https://yiweiniu.github.io/blog/2019/03/ATAC-seq-data-analysis-from-FASTQ-to-peaks/
- https://rockefelleruniversity.github.io/RU_ATACseq/
- cite:sun2019atac
- https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html
- https://nf-co.re/atacseq/1.1.0/output
- https://yiweiniu.github.io/blog/2019/03/ATAC-seq-data-analysis-from-FASTQ-to-peaks/
- https://github.com/nf-core/atacseq/blob/master/main.nf
- [[https://www.scopus.com/results/results.uri?sort=plf-f&src=s&sid=1b5013f3a0a4db522d602fa164281aa0&sot=a&sdt=a&cluster=scopubyr%2c%222021%22%2ct%2c%222020%22%2ct%2c%222019%22%2ct%2c%222018%22%2ct%2c%222017%22%2ct%2c%222016%22%2ct&sl=262&s=TITLE-ABS-KEY%28atac-seq%29+AND+NOT+TITLE-ABS-KEY%28chip%29+AND+NOT+DOI%2810.1016%2fj.cell.2017.12.014%29+AND+NOT+DOI%2810.1016%2fj.molmet.2016.01.002%29+AND+NOT+DOI%2810.1038%2fnature23283%29+AND+NOT+DOI%2810.1038%2fnature18606%29+AND+NOT+TITLE-ABS-KEY%28single-cell%29+AND+NOT+TITLE-ABS-KEY%28Hi-C%29&origin=searchadvanced&editSaveSearch=&txGid=38c8cceffee3c655f632c5736cf123cd][top atac-seq cites w/ good methods writeup]]
  - https://www.pnas.org/content/114/13/E2776
  - https://science.sciencemag.org/content/sci/suppl/2018/10/24/362.6413.eaav1898.DC1/aav1898_Corces_SM.pdf
  - https://ars.els-cdn.com/content/image/1-s2.0-S1074761316304393-mmc1.pdf
- https://link.springer.com/article/10.1186/s43556-020-00009-w
****** Bench DNA prep
- cite:corces2017
- cite:buenrostro2015
- cite:buenrostro2015
- for tissue
  - cite:corces2017
****** [[id:fe58011b-ca46-4e45-9c4e-d4780fbfa115][Quality control]]
****** Alignment parameters
  - https://www.nature.com/articles/ng.3844#Sec14https://www.sciencedirect.com/science/article/pii/S1074761316304393?via%3Dihub#mmc1
  - https://www.pnas.org/content/114/13/E2776
  - https://informatics.fas.harvard.edu/atac-seq-guidelines.html#alignments
  - https://ars.els-cdn.com/content/image/1-s2.0-S1074761316304393-mmc1.pdf
- Devon recommending filtering to 120 bp or lower  https://www.biostars.org/p/447927/
****** [[id:1c56c6ad-e0f8-47db-8d69-3782b7ed1e8b][Peak calling]]
****** Peak annotation
- [[id:a4cf29e3-7fa6-4ea2-a538-f59ec450cc7b][ChIPseeker]]
****** [[id:17f978bc-f2f2-497d-a8f1-b1d47927380f][Differential chromatin accessibility]]
****** Data integration
https://www.nature.com/articles/s41590-019-0312-6/figures/2
- Integration
  - atac-seq and mirna-seq https://www.sciencedirect.com/science/article/pii/S193459091730320X?via%3Dihub
  - ATAC-seq and miRNA-seq cite:abernathy2017
****** Footprinting
- https://github.com/Zhang-lab/ATAC-seq_QC_analysis/blob/master/documents/ifr_documentation.md
- cite:bentsen2020
  - https://github.com/loosolab/TOBIAS
- cite:vierstra2020
#+begin_src R
source("./src/setup.R")
dataDIR = file.path("/mnt/ris/jschwarz/cardiac-radiobiology")
peaks = dir(file.path(dataDIR,"macs2"), pattern = "*narrowPeak", full.names = T)


if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("ChIPQC")
library(ChIPQC)

peaks_range = lapply(peaks, ChIPQC:::GetGRanges, simple = TRUE)

peaks_range = GRangesList(peaks_range)
peaks_range

names(peaks_range) = c("ir_m01.1",
                       "ct_f02.1",
                       "ct_f03.1",
                       "ir_m03.1",
                       "ct_m01.1",
                       "ir_f04.1",
                       "ir_f02.1")

seqlevelsStyle(peaks_range) = "UCSC"

peaks_range

peakCombo = groupByOverlaps(peaks_range)

class(peakCombo)

group = factor(c("ir","ct","ct","ir","ct","ir","ir"))
peaks_range$group = group








reduced = reduce(unlist(peaks_range))

reduced

consensusIDs <- paste0("consensus_", seq(1, length(reduced)))

mcols(reduced) <- do.call(cbind, lapply(peaks_range, function(x) (reduced %over% x) + 0))

reducedConsensus <- reduced
mcols(reducedConsensus) <- cbind(as.data.frame(mcols(reducedConsensus)), consensusIDs)
consensusIDs <- paste0("consensus_", seq(1, length(reducedConsensus)))
return(reducedConsensus)

reducedConsensus


names(reducedConsensus) = c("ir_m01.1",
                       "ct_f02.1",
                       "ct_f03.1",
                       "ir_m03.1",
                       "ct_m01.1",
                       "ir_f04.1",
                       "ir_f02.1")

group = factor(c("ir","ct","ct","ir","ct","ir","ir"))
reducedConsensus$group = group
reducedConsensus

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("soGGi")

library(soGGi)

consensusToCount = soGGi:::runConsensusRegions(GRangesList(reducedConsensus), "none")


consensusToCount
#+end_src

#+begin_src R
source("./src/setup.R")
dataDIR = file.path("/mnt/ris/jschwarz/cardiac-radiobiology")
peaks = dir(file.path(dataDIR,"macs2"), pattern = "*narrowPeak", full.names = T)


if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("ChIPQC")
library(ChIPQC)

peaks_range = lapply(peaks, ChIPQC:::GetGRanges, simple = TRUE)

peaks_range = GRangesList(peaks_range)
peaks_range

names(peaks_range) = c("ir_m01.1",
                       "ct_f02.1",
                       "ct_f03.1",
                       "ir_m03.1",
                       "ct_m01.1",
                       "ir_f04.1",
                       "ir_f02.1")

seqlevelsStyle(peaks_range) = "UCSC"

peaks_range

peakCombo = groupByOverlaps(peaks_range)

class(peakCombo)

group = factor(c("ir","ct","ct","ir","ct","ir","ir"))
peaks_range$group = group








reduced = reduce(unlist(peaks_range))

reduced

consensusIDs <- paste0("consensus_", seq(1, length(reduced)))

mcols(reduced) <- do.call(cbind, lapply(peaks_range, function(x) (reduced %over% x) + 0))

reducedConsensus <- reduced
mcols(reducedConsensus) <- cbind(as.data.frame(mcols(reducedConsensus)), consensusIDs)
consensusIDs <- paste0("consensus_", seq(1, length(reducedConsensus)))
return(reducedConsensus)

reducedConsensus


names(reducedConsensus) = c("ir_m01.1",
                       "ct_f02.1",
                       "ct_f03.1",
                       "ir_m03.1",
                       "ct_m01.1",
                       "ir_f04.1",
                       "ir_f02.1")

group = factor(c("ir","ct","ct","ir","ct","ir","ir"))
reducedConsensus$group = group
reducedConsensus

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("soGGi")

library(soGGi)

consensusToCount = soGGi:::runConsensusRegions(GRangesList(reducedConsensus), "none")


consensusToCount
#+end_src

****** Motifs
- [[./biotools.org_20220906_122531.png]]
- Motif analysis
  - cite:minnoye2021
  - homer
    - https://www.biostars.org/p/493498/
  - https://rockefelleruniversity.github.io/RU_ATACseq/
    - https://rockefelleruniversity.github.io/RU_ATACseq/presentations/singlepage/RU_ATAC_part3.html
  - cite:yan2020atac
  - https://ccg.epfl.ch/pwmtools/pwmscan.php
  - https://meme-suite.org/meme/
  - example of homer-jaspar method https://www.sciencedirect.com/science/article/pii/S1535610817302040?via%3Dihub
  - cite:sun2019atac
****** Identifying transcription factor binding sites
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1642-2
****** Tables
- [[https://gdc.cancer.gov/about-data/publications/ATACseq-AWG][outputs example]]
****** [[id:12393aa3-d318-4ba6-8183-6e3a175d3ba4][Region or locus-based enrichment analysis]]
****** Visualizations / figures                                      :scifig:
- highlight visualizations on ucsc or igv
  - dump beds and narrowpeaks directly into ucsc
    - https://genome.ucsc.edu/cgi-bin/hgCustom
    - https://groups.google.com/a/soe.ucsc.edu/g/genome/c/JzQtbjGbTnc
  - http://www.guertinlab.org/wp-content/uploads/2021/05/Genomic_ATAC-seq_analysis_190912.pdf
  - https://www.biostars.org/p/365263/
  - https://genome.ucsc.edu/goldenPath/help/bigWig.html
- tf or tf-group specific workup
  - motif position vs condition
    - file:resources/monroe2019fig4d.png
  - motif vs condition
    - [[file:resources/monroe2019fig4c.png]]
  - - TSS distance map- maybe easy in chipseeker?
- https://nf-co.re/atacseq/1.1.0/output
- atac vs expression signal
  - 2d heatmap [[file:resources/wang2018amdfig4e.png]]
- 3-state location-specific normalized counts
  - [[file:resources/wang2018amdfig2a.png]]
- Differential peak matrix
  - [[file:resources/wang2018amdfig2b.png]]
- global accessibility change
  - bar graph of counts of peak change cite:wang2018amd fig3d
  - line graph of signal intensity through time cite:wang2018 fig3e
  - cite:wang2018amd fig2d
    - [[file:resources/wang2018amdfig2d.png]]
- accessibility difference at gene
  - cite:wang2018amd fig2a
- peak heatmaps
  - cite:wang2018amd fig1c
- peak annotation across cohorts
  - [[./biotools.org_20220906_113352.png]] cite:merrill2022 fig1D
- annotation correlation to gene expression
  - [[./biotools.org_20220906_113853.png]] cite:merrill2022 fig2
- https://tobiasrausch.com/courses/atac/atac-seq-data-analysis.html#data-exploration
- peak to feature proportion  https://nf-co.re/atacseq/1.1.0/output
- fragment depth vs genome-wide feature
  - [[file:resources/monroe2019fig4f.png]]
- consensus peak upset plot  https://nf-co.re/atacseq/1.1.0/output
- global read distribution  https://nf-co.re/atacseq/1.1.0/output
- qc
  - deeptools fingerprint https://nf-co.re/atacseq/1.1.0/output https://deeptools.readthedocs.io/en/develop/content/tools/plotFingerprint.html
  - insert size density by sample https://nf-co.re/atacseq/1.1.0/output
  - complexity curve by sample https://nf-co.re/atacseq/1.1.0/output
- accessibility by chromosome https://nf-co.re/atacseq/1.1.0/output
- feature-aligned heatmaps
  - https://www.bioconductor.org/packages/devel/bioc/vignettes/ChIPpeakAnno/inst/doc/ChIPpeakAnno.html
- cite:wilson2019
- cite:wang2018amd
******* [[id:29821fa1-acfe-4a74-9c5d-a24adddf2969][Coordinate-based genomics visualization tracks]]
****** Part of [[id:ae0363f3-379e-4521-9184-cbc0d46501cd][Chromatin accessibility profiling]]
****** General reference
- https://github.com/Zhang-lab/ATACseq_benchmarking/blob/master/simulated_ATAC_tests_limma.R
- https://www.sciencedirect.com/science/article/pii/S2666166720302768
- https://bioinformatics-core-shared-training.github.io/cruk-autumn-school-2017/ChIP/Materials/Lectures/Lecture5_Peak%20Calling_SS.pdf
- Tn5 insertion
  - See for T5 shift visualization https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1642-2#Sec13
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8557372/
- [[https://bioconductor.org/packages/release/BiocViews.html#___ATACSeq][ATAC-seq related bioconductor packages]]
- Reference maps
  - https://academic.oup.com/nar/article/49/D1/D55/5943839
- Alternative uses https://www.science.org/doi/10.1126/sciadv.aba2489
- https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/atac-seq.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5170597/
******* Encode
- Encode reference sequences - https://www.encodeproject.org/data-standards/reference-sequences/
- Encode ATAC-seq https://www.encodeproject.org/atac-seq/
  - v1 pipeline
    - as pdf https://www.encodeproject.org/documents/c008d7bd-5d60-4a23-a833-67c5dfab006a/@@download/attachment/ATACSeqPipeline.pdf
    - as google doc https://docs.google.com/document/d/1f0Cm4vRyDQDu0bMehHD7P7KOMxTOP-HiNoIvL1VcBt8/edit
  - https://github.com/ENCODE-DCC/atac-seq-pipeline
  - example qc report- https://storage.googleapis.com/encode-pipeline-test-samples/encode-atac-seq-pipeline/ENCSR889WQX/example_output/qc.html
- https://www.encodeproject.org/search/?type=Pipeline&accession=ENCPL787FUN&accession=ENCPL344QWT
- [[id:585e27b3-3c79-46d5-aeab-9430251d4318][ENCODE]]
******* [[id:8ca221b7-b51a-4382-9844-5d8ba7193ff6][Chromatin accessibility biology]]

  - cite:buenrostro2013



**** Enhancers
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4666684/
*** Single-cell methods
:PROPERTIES:
:ID:       2597b54c-972a-43c3-9630-0e542ba90eeb
:END:
- https://www.nature.com/articles/s41591-020-0844-1
- cite:griffith2015
- http://bioinformatics.sdstate.edu/idep/
- cite:qi2019
- applications
  - [[id:7bb87c84-6b06-44ef-9a1d-1bb9b9e57295][immunoprofiling]]
  - intra-tumor heterogeneity / [[id:12cfb78c-7912-4c68-b0f4-e2f90c112185][tumor microenvironment]]
    - cite:lawson2018
- Single cell data integration cite:stuart2019
- [[id:1acf73b6-ca1e-458c-91c1-b93f17b80a02][single-cell illumina workshop [2019-09-10 Tue 09:00-15:30]​]]
- inputs and pre-processing
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02048-6
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1171-9
**** Good practice
:PROPERTIES:
:ID:       f6418486-9a29-4a65-9145-2a2a3ac788a7
:END:
- [[cite:&luecken2019]]
- cite:heumos2023
- cite:slyper2020
- https://www.nature.com/articles/s41576-023-00586-w
**** single-nucleus RNA-seq
:PROPERTIES:
:ID:       1149b091-eabe-4137-a3e9-222f21bfdd57
:END:
- https://www.nature.com/articles/s41591-020-0844-1
**** Jointly defining cell states from singl-cell multi-omic and spatial tanscriptomic datasest
:PROPERTIES:
:ID:       76cd9647-825f-46b4-9344-b7b95049f2f6
:END:
:LOGBOOK:
CLOCK: [2021-11-02 Tue 11:01]--[2021-11-02 Tue 11:26] =>  0:25
:END:
- https://event.on24.com/eventRegistration/console/EventConsoleApollo.jsp?&eventid=3478709&sessionid=1&username=&partnerref=&format=fhaudio&mobile=&flashsupportedmobiledevice=&helpcenter=&key=87791679E10838FA9DCF16C96D7F22F8&newConsole=true&nxChe=true&newTabCon=true&consoleEarEventConsole=false&text_language_id=en&playerwidth=748&playerheight=526&eventuserid=488203953&contenttype=A&mediametricsessionid=422205446&mediametricid=4883132&usercd=488203953&mode=launch
- liger cell welch 2019 linked inference of henomic experimental relationships
- scalable, iteratifve integration with online iNMF gai nature biotechnology 2021
-
**** [[id:99ee4bd0-060e-4109-9901-29aa7fafd816][Single nucleus and single cell methods applied to heart]]
**** [[id:992de4dc-775e-41f0-a70a-e38c99b87265][Single cell methylation sequencing]]
**** General
- https://www.sciencedirect.com/science/article/pii/S1097276516301861?via%3Dihub
- Tutorials, guides, good practice
  - https://www.reddit.com/r/bioinformatics/comments/uy9g1j/where_to_start_with_scrnaseq_data/
  - https://www.embopress.org/doi/full/10.15252/msb.20188746
  - https://mobile.twitter.com/nomad421/status/1522554692202598403?s=20&t=UObG4oqigCp6gJIIgmyhPA
- Reviews
  - https://www.nature.com/articles/nmeth.4612
  - cite:stuart2019
- Multi-cell pathway analysis - https://bioengineeringcommunity.nature.com/posts/dialogue-maps-multicellular-programs-in-tissue-from-single-cell-or-spatial-transcriptomics-data
- Examples of well-implimented applications
  - cite:storrs2022
  - Li Ding pancreas - https://www.nature.com/articles/s41588-022-01157-1
- Integrating multiple single cell data sets - Harmony - https://www.nature.com/articles/s41592-019-0619-0
- RNA velocity - scVelo RNA velocity  [[https://scvelo.readthedocs.io/][site]]
**** Single cell methylation sequencing
:PROPERTIES:
:ID:       992de4dc-775e-41f0-a70a-e38c99b87265
:END:
https://www.nature.com/articles/nmeth.3035
https://www.nature.com/articles/s41467-018-03149-4
[[id:2597b54c-972a-43c3-9630-0e542ba90eeb][Single-cell methods]]
**** Single cell ATAC-seq
- https://www.nature.com/articles/s41467-021-21583-9#Sec2
**** single-nucleus RNA-seq
:PROPERTIES:
:ID:       1149b091-eabe-4137-a3e9-222f21bfdd57
:END:
- cite:grindberg2013
- https://www.nature.com/articles/s41591-020-0844-1

**** single-cell RNA sequencing (scRNA-seq)
:PROPERTIES:
:ID:       2bbeb644-af54-4f07-80ad-3584e1e21b04
:END:
- clustering and identification algorithms - https://www.nature.com/articles/s41467-023-36383-6
- trajectory inference
  - Monocle 2
  - Slingshot
  - TSCAN
  - Wishbone
- https://cancerdiscovery.aacrjournals.org/content/early/2021/12/22/2159-8290.CD-RW2021-181

https://www.nature.com/articles/s41587-022-01452-6
https://www.embopress.org/doi/full/10.15252/msb.20188746
- https://www.biorxiv.org/content/biorxiv/early/2019/02/05/541433.full.pdf
- examples
  - - see fig 3 & methods https://genome.cshlp.org/content/27/2/208.full.pdf
- https://www.nature.com/articles/s41587-020-0553-9
- 10x
  - applications - https://pages.10xgenomics.com/nur-gen-research-area-literature.html?src=email&utm_medium=email&lss=nurture&utm_source=nurture&cnm=es-nurture-ra_g-general&utm_campaign=es-nurture-ra_g-general&utm_content=&utm_term=&useroffertype=marketing-literature&userresearcharea=ra_g&userregion=&userrecipient=customer&mktouserid=1265915&cid=&usercampaignid=
- https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0212031
- http://www.nature.com/nbt/journal/v34/n2/full/nbt.3445.html
- https://www.10xgenomics.com/videos
- https://www.nature.com/articles/s41591-021-01276-y
- [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- https://bioconductor.org/packages/release/bioc/vignettes/FlowSOM/inst/doc/FlowSOM.pdf
- https://chanzuckerberg.github.io/scRNA-python-workshop/analysis/05-diffexp.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6914766/
- https://www.nature.com/articles/s41467-019-12235-0
- http://www.kumc.edu/genomics/genome-sequencing-facility/services/chromium-single-cell-5-and-v(d)j-enrichment.html
- https://www.10xgenomics.com/solutions/vdj/
- https://research.medgenome.com/ngs-services/single-cell-sequencing/
- https://www.sciencedirect.com/science/article/pii/S1097276517300497#fig1
- https://www.biorxiv.org/content/10.1101/615013v1
- https://www.youtube.com/watch?v=Wbusqowyu6g&t=147s
- cite:monaco2019
- cite:see2018
- toolkits
  - Seurat
  - Monocle 2
  - scVI
- clustering
  - SC3
  - SIMLR
  - CIDR
  - SNN-Cliq
  - Biscuit
- imputing dropouts
  - MAGIC
  - scImpute
  - SAVER
- https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/mkfastq
- http://www.partek.com/free-trial/
- https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger
- https://github.com/koash/cellranger-docker
- https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html
- http://core-genomics.blogspot.com/2016/07/10x-genomics-single-cell-3mrna-seq.html
- https://www.frontiersin.org/articles/10.3389/fcell.2018.00108/full
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6292934/
- https://www.sciencedirect.com/science/article/pii/S1097276515002610?via%3Dihub
- https://www.ncbi.nlm.nih.gov/pubmed/26000846/
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5561556/
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6131190/
- https://www.nature.com/articles/s41598-019-46932-z
- https://www.biorxiv.org/content/biorxiv/early/2018/03/05/275859.full.pdf
- cite:wohnhaas2019
- cite:papalexi2018
- https://support.10xgenomics.com/single-cell-gene-expression/datasets
- https://www.cell.com/cell-reports/pdf/S2211-1247(19)30059-2.pdf
- https://www.nature.com/articles/s41591-020-0844-1
- https://www.workcast.com/AuditoriumAuthenticator.aspx?cpak=8475927829006248&pak=1763506853003757

- cite:qi2019
- applications
  - [[id:7bb87c84-6b06-44ef-9a1d-1bb9b9e57295][immunoprofiling]]
  - intra-tumor heterogeneity / [[id:12cfb78c-7912-4c68-b0f4-e2f90c112185][tumor microenvironment]]
    - cite:lawson2018
- cite:stuart2019
- [[id:1acf73b6-ca1e-458c-91c1-b93f17b80a02][single-cell illumina workshop [2019-09-10 Tue 09:00-15:30]​]]
- inputs and pre-processing
  - - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02048-6
  - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1171-9
  - https://www.nature.com/articles/s41591-020-0844-1
- trajectory inference
  - Monocle 2
  - Slingshot
  - TSCAN
  - Wishbone
*** Sequence motif analysis
https://meme-suite.org/meme/
*** Gene expression
https://www.science.org/doi/10.1126/sciadv.ade2812
overlap plus combined DEGs https://wustl.app.box.com/file/1235774979545 fig2c
**** Microarray
:PROPERTIES:
:ID:       294edcbb-f306-4f94-9eba-e8a1c7c512f1
:END:
- cite:lee2004
- cite:nguyen2002
**** Bulk RNA-seq                                                    :career:
:PROPERTIES:
:ID:       4206be38-8d3c-43c4-a5c7-ca6a8bd8361a
:CREATED:  [2021-02-17 Wed 09:08]
:END:
https://www.nature.com/articles/s41598-020-76881-x
https://www.cell.com/ajhg/fulltext/S0002-9297(18)30401-4
***** RNA-seq experimental design
:PROPERTIES:
:ID:       19780d7c-afa7-4195-b1f9-df9c563de303
:END:
https://www.biostars.org/p/9549394/
https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/
- https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html#design-matrices-with-and-without-intercept-term
- https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html#design-matrices-with-and-without-intercept-term
- https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2021/RNAseq/Markdowns/09_Linear_Models.html
- spike-ins
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4538800/
- validation technologies
  - cannot distinguish isomiRs
  - qPCR expected to correlate >.9 w/ seq expression > 2^8 cite:yeri2018
  - qPCR more concordant than EdgeSeq or FirePlex cite:yeri2018
- what depth? https://www.researchgate.net/post/how_many_million_reads_for_miRNA_discovery
- reads per sample
  - https://www.qiagen.com/us/resources/faq?id=a119d5f0-0a2a-4612-828e-d8fa3cc37891&lang=en
- cite:garve2012
- cite:williams2014
- https://www.degruyter.com/view/journals/sagmb/18/1/article-20180021.xml?language=en
- https://ucdavis-bioinformatics-training.github.io/2018-June-RNA-Seq-Workshop/tuesday/ExperimentalDesign.pdf
- http://scotty.genetics.utah.edu/help.html
- Depth
  - - determine directly using technical replicates
  - expression estimate correspondence among technical replicates (no biological variation) for genes with above-median coverage stabilized at about 10 million reads per sample (Wang et al. 2011).
- Replicates and power
  - sufficient replication has to occur at every level of comparison. In a fully "factorial" design, for example, more than one experimental "factor" is of interest, each with two or more treatments, such that any individual receives one of multiple possible treatments at each factor. In this case, every possible combination of treatments across factors needs to be replicated sufficiently,
  - biological variation is more important than technical variation- See Bullard,10 Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments, fig. 4
  - adding depth vs. replicate- replicates preferred, gives less uncertainty
  - remember, more comparisons require more replicates
  - cite:schurch2016
  - Power
    - https://cofactorgenomics.com/6-changes-thatll-make-big-difference-rna-seq-part-5/
    - cite:peixoto2015
    - the rewards in statistical power due to additional biological replicates will surpass the improved parameterization of technical variation garnered from additional technical replicates
    - low counts mean low power- see rna seqlopedia fig 1.1
  - [[http://scotty.genetics.utah.edu/help.html][Scotty]]
  - consider a "bank" of more sample replicates than are acutally sequenced
- By specific experimental question
  - Differential expression
    - [[http://rnaseq.uoregon.edu/#exp-design][RNA-seqlopedia]]
      - qualitative or quantitative (DE, ect.) design- usually quant
      - need bio replicates
      - stranded and paired library prep not required
- Prefer “balanced block” design w/ barcode-> mixed prep-> all samples on all lanes
- cite:auer2010

****** General reference
- http://www.slideshare.net/YaoyuEWang/rnaseq-analysis-ngsapplication2
- http://michelebusby.tumblr.com/post/26913184737/thinking-about-designing-rna-seq-experiments-to
****** Parameters
- cite:williams2014
  - contains the 30-50 mil read for DE recommendation
******* Parameters affecting sequencing
  - Variance is bimodal
    - low @ low counts b/c Poisson
    - low @ high counts b/c?
  - biological
    - transcript abundence 10^5-10^7 see cite:Griffith_2015 refs 85,86
  - technical @ platform
    - specialized
      - On target efficiency with bedtools
    - read length
    - read depth
    - strandedness problem[[file:org.org_imgs/20170224_132238_567aRU.png]]
  - Strandedness
    - cite:Zhao_2015

****** Validation
- validation technologies
  - cannot distinguish isomiRs
  - qPCR expected to correlate >.9 w/ seq expression > 2^8 cite:yeri2018
  - qPCR more concordant than EdgeSeq or FirePlex cite:yeri2018
***** [[id:386d899a-9208-4d5c-93d8-3750b22d9748][RNA bench methods]]
***** [[id:7BF989EC-4DA7-4B0C-AAA9-0175156447FF][RNA extraction, solubilization, and isolation]]
:PROPERTIES:
:ID:       b477ff77-85e8-4613-9bea-a703de899817
:END:
***** [[id:4E23127E-4ED6-4802-A03C-4103AA8CAF29][RNA Quality and Quantification]]
:PROPERTIES:
:ID:       f8fb2eb4-e4dd-40f4-9ca6-f0582f4a3aac
:END:
***** Library preparation
:PROPERTIES:
:ID:       2111e30d-9183-40de-948e-3f76b70905d8
:END:
  - common kits
    - RNA exome
    - pico
  - From input to library
    - cite:van_Dijk_2014
    - By workflow
      - Isolation
      - Enrichment
        - Methods
          - Hybridization
            - For selection of targets
              - via oligo-dT
                - will include some mitochondial
            - For removal of non-targets
              - rRNA depletion- rely on specific oligo set, d/n work well if rRNA is degraded
          - Size-selection enrichment- for ncRNA
          - DSN (duplex-specific nuclease digestion)- @ cDNA level, allow abundant transcripts to renature and are degraded by duplex-specific nuclease
      - Fragmentation- RNase III, metal ion, heat, or sonication
      - Library preparation
        - FFPE
          - However, formalin treatment degrades RNA through cross-linking and fragmentation, which significantly impairs molecular analyses (Bass et al., 2014; Farragher et al., 2008; Klopfleisch et al., 2011). T
          - check webster 15 cites- (Auerbach et al., 2014; Hedegaard et al., 2014; Linton et al., 2012; Spencer et al., 2013; Zhao et al., 2014b).
        - [[http://support.illumina.com/content/dam/illumina-support/courses/truseq-rna-access/story.html?iframe?iframe][TruSeq: RNA Access Library Prep]]
        - [[http://support.illumina.com/content/dam/illumina-support/courses/TruSeq_Sample_Prep_Best_Practices/index.html?iframe?iframe][TruSeq Sample Preparation Best Practices]]
        - First-strand synthesis
          - Priming options
            - oligo-dT
            - "random" hex
        - Second-strand synthesis
          - Also requires a primer
          - Usually via RNA displacement- RNase H and T4 ligase
        - cDNA fragmentation
          - accounting for fragment bias- https://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-3-r22
        - Sequencing adaptors
          - Y-adaptor PCR: blunt cDNA ends, add adenosine overhangs
- cite:oneil2013
- stranded
  - https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work
***** [[id:3BC41E6A-00D3-4498-A48C-463F68CFD76B][Massively parallel sequencing]]
:PROPERTIES:
:ID:       68056b25-1413-4bda-a1f2-6610e92c6765
:END:
- what depth? https://www.researchgate.net/post/how_many_million_reads_for_miRNA_discovery
***** [[id:1EE731A3-83DD-46AF-B988-2311307C335F][Raw sequence quality control]]
:PROPERTIES:
:ID:       7cc53256-8b7b-4c93-aaa5-2a35e41b7319
:END:
***** Alignment and pseudoalignment
****** [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][Nucleic acid sequence alignment]]
:PROPERTIES:
:ID:       542988fa-307f-454e-927e-ee6a17187265
:END:
****** Pseudoalignment
- https://www.reddit.com/r/bioinformatics/comments/uv1hu9/what_are_the_pros_and_cons_of_pseudo_alignment/
****** Read annotation
:PROPERTIES:
:ID:       8e435b88-6895-4e75-a728-cfdaa272bddb
:END:
- https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1308-8
- https://github.com/griffithlab/rnaseq_tutorial/wiki/Annotation
***** Quantification and normalization
:PROPERTIES:
:ID:       853dd34c-27be-4be8-9794-19f04a1e22b0
:END:
- via spike or replicate with RUVs https://www.nature.com/articles/nbt.2931?error=cookies_not_supported&code=a01fb01e-6702-4e1e-96da-1dbad752e672
- https://www.reneshbedre.com/blog/expression_units.html
- http://luisvalesilva.com/datasimple/rna-seq_units.html
https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
by spike-in  Jiang,L., Schlesinger,F., Davis,C.A., Zhang,Y., Li,R., Salit,M., Gingeras,T.R. and Oliver,B. (2011) Synthetic spike-in standards for RNA-seq experiments. Genome Res., 21, 1543–1551.

- choosing a strategy- Zyprych-Walczak,J., Szabelska,A., Handschuh,L., Górczak,K., Klamecka,K., Figlerowicz,M. and Siatkowski,I. (2015) The impactof normalization methods on RNA-Seq data analysis. Biomed Res. Int., 2015, doi:10.1155/2015/621690.
- what differences are controlled?
  - library size
  - batch effect
  - gene length
  - GC content
- methods
  - upper quartile- remove zero counts, scale based on 75th quartile of remaining reads Bullard,J.H., Purdom,E., Hansen,K.D. and Dudoit,S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics, 11, doi:10.1186/1471-2105-11-94.
  - quantile- read distribution assumed identical, counts adjusted by median of each quantile across all samples
  - TMM
  - pseudoalignment
    - kallisto, used with sleuth for DE
      - https://www.nature.com/articles/nbt.3519
      - https://pachterlab.github.io/sleuth/about
  - by spike-in  Jiang,L., Schlesinger,F., Davis,C.A., Zhang,Y., Li,R., Salit,M., Gingeras,T.R. and Oliver,B. (2011) Synthetic spike-in standards for RNA-seq experiments. Genome Res., 21, 1543–1551.
  - count-based
    - analysis only at the gene level
      - cite:pachter2015
    - edgeR
  - TPM
  - FPKM
  - https://europepmc.org/article/PMC/6058388
  - abundance
    - http://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#introduction
  - cite:pachter2015
- Bullard,J.H., Purdom,E., Hansen,K.D. and Dudoit,S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics, 11, doi:10.1186/1471-2105-11-94.
- cite:peixoto2015
- https://www.the-scientist.com/news-opinion/technical-bias-widespread-in-rna-seq-datasets-66766?utm_campaign=TS_DAILY%20NEWSLETTER_2019&utm_source=hs_email&utm_medium=email&utm_content=79783848&_hsenc=p2ANqtz-_TvsXNNQSHgNWR9yE8Q6vZP04TRSx3T-W0rLk0NLucSkrSI0wQ0GDQ2L1Vs_Jf1x1_21ou8yVG1ur9nXtXbBYl91OsUg&_hsmi=79783848
- normalization to control for systemic error / bias
- shrinkage: pooling information across genes to obtain reliable variance estimates for small sample sizes, cite:pachter2015
- feature counting
  - HTSeq
  - featureCounts
- https://www.slideshare.net/mobile/jakonix/part-1-of-rnaseq-for-de-defining-the-goal
- https://www.biostars.org/p/404211/#:~:text=What%20is%20the%20difference%20between,as%20fragment%20length%20distribution%20etc.
- https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
- [[*rsem][rsem]]
- https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/
- https://github.com/griffithlab/rnaseq_tutorial/wiki/Transcript-Assembly-Merge
- https://www.reddit.com/r/bioinformatics/comments/i6qhrx/samples_from_different_studies_de_analysis_help/
- https://www.nature.com/articles/ng1032z.pdf
- evaluation
  - linearity
    - https://link.springer.com/article/10.1186/s12859-017-1526-y
  - https://bioinformatics.stackexchange.com/questions/92/confirm-success-or-failure-of-rna-seq-normalization
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/
- https://groups.google.com/forum/m/#!topic/rsem-users/GRyJfEOK1BQ
- https://www.biostars.org/p/68126/
- https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25
- https://cgatoxford.wordpress.com/2016/08/17/why-you-should-stop-using-featurecounts-htseq-or-cufflinks2-and-start-using-kallisto-salmon-or-sailfish/
- isoform abundance
  - https://cgatoxford.wordpress.com/2016/08/17/why-you-should-stop-using-featurecounts-htseq-or-cufflinks2-and-start-using-kallisto-salmon-or-sailfish/
- https://www.biostars.org/p/329625/
- sources of bias
  -
- https://htseq.readthedocs.io/en/release_0.11.1/tour.html#counting-reads-by-genes
- RSEM
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323
****** featureCounts in Rsubread
:PROPERTIES:
:ID:       0FB3EE5E-DD17-4438-BF35-011555A41DF9
:END:

[[http://www.rdocumentation.org/packages/Rsubread/functions/featureCounts][nice featureCounts manual]]

****** [#Y] summarizeOverlaps in GenomicAlignments
:PROPERTIES:
:ID:       16408005-913F-4675-BA61-B7594A3FB4EC
:END:

****** HTseq-count
:PROPERTIES:
:ID:       E2262D77-424F-4943-8DA4-5649B2F0B4B2
:END:

[[file:~/Box%20Sync/comp_bio/Anders,15%20HTSeq%20a%20python%20framework%20to%20work%20with%20high%20throughput%20sequencing%20data.pdf][Anders,15 HTSeq a python framework to work with high throughput
sequencing data]]

Not on cluster
- create R script to run stats on generic output file
- % no feature, ambiguous, too low qual, not aligned, alignment not
  unique

******* Prerequisites
:PROPERTIES:
:ID:       D5102869-0833-4513-A716-69D44892FDDF
:END:

- samtools sort (to postion)

******* [[http://www-huber.embl.de/HTSeq/doc/count.html#count][website]]
:PROPERTIES:
:ID:       6B7F3881-DA7D-4700-BC30-9EF83150D916
:END:

htseq-count [options] <alignment_file> <gff_file>

simple code: htseq-count -m union -r pos -i gene_name -a 10 --stranded=no agcgccg.bam Mus_musculus.GRCm38.84.gtf > output.counts

Important options
--stranded=no (defualt is yes)
--format bam (defualt is sam)
  - for paired end, order is important
    - do samtools sort (default) to put in position order
    - in HTSeq-counts, specify -r pos
    -a 10 (minimum quality, 10 is default)
    -m default is union, alternate intersection-strict and intersection-nonempty

    Why union is default: consider two genes, which share a stretch of common sequence such that for a read mapping to this stretch, the aligner cannot decide which of the two genes the read originated from and hence reports a multiple alignment. If we discard all such reads, we undercount the total output of the genes, but the ratio of expression strength (the “fold change”) between samples or experimental condition will still be correct, because we discard the same fratcion of reads in all samples. On the other hand, if we counted these reads for both genes, a subsequent diffential-expression analysis might find false positives:


  followed by the special counters, which count reads that were not counted for any feature for various reasons

  [[http://www-huber.embl.de/HTSeq/doc/counting.html][algorithm details]]


****** Expression Level Abundance, Normalization Units
:PROPERTIES:
:ID:       D4BFCCCA-8BBC-4D23-B71B-F8C2ED6287E3
:END:
https://ro-che.info/articles/2016-11-28-rna-seq-normalization
R/FPKM and C/TPM are "local" to the sample while TMM must be recalculated with different sample set

******* Raw- required input for DE analyses
:PROPERTIES:
:ID:       82A81565-9333-4E40-806E-474399BD73C4
:END:

******* R/FPKM
:PROPERTIES:
:ID:       9D6324B7-D9A2-44ED-A54F-B0C0C8DCECE2
:END:
https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/
- normalization for library size (i.e. sequencing depth) and transcript length
- use to
  - compare a gene across samples or
  - compare different genes within a sample
- see Mortazavi A., Williams B.A., McCue K., Schaeffer L., and Wold B. Mapping and quantifying mammalian transcriptomes by RNA-seq. Nat Methods 5(7):621–628, 2008.
- RPKM (reads per thousand nucleotides in transcript per million reads)
  - “For example, if a 2 kb transcript has 1000 reads and the total number of reads is 25 million, then RPKM = (1000/2)/25 = 20”
- FPKM (fragments per thousand nucleotides per million mapped reads)- equivalent of RPKM for paired-end (two reads per fragment)

******* CPM (counts per million) / TPM
:PROPERTIES:
:ID:       8C3481E5-B350-4D9D-A44C-DAB17FAEE1C6
:END:

  -   - see Wagner G.P., Kin K., and Lynch V.J. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. Theory Biosci 131(4):281–285, 2012.=
  - limma voom uses a logCPM
  - corrects for transcript length distribution of different sample pools
  - TPM (transcripts per million)- divides by sum of transcript-length-normalized reads
    i.e. normalizes to transcripts as opposed to reads
    useful if average transcript length differs b/w samples

******* TMM (trimmed mean of M values)
:PROPERTIES:
:ID:       544B0EAB-4594-404B-9C83-B951E2244CFE
:END:

-attempts to normalize this problem:
[[file:work.org_imgs/20160419_132550_469_bk.png]]
- corrects for less "sequencing real estate" due to high expression per tissue
***** RNA-seq quality control
:PROPERTIES:
:ID:       e6cbde1b-22c3-4a23-b99e-d9d4fe0a6b95
:END:
- https://www.reddit.com/r/bioinformatics/comments/ufuk46/question_transcriptome_quality_assessments/

- https://bioconductor.org/packages/release/bioc/html/RUVSeq.html
- http://rseqc.sourceforge.net/
- https://www.reddit.com/r/bioinformatics/comments/flqb8e/how_do_i_differentiate_outliers_from_ingroup/
- https://www.rna-seqblog.com/an-rna-seq-qc-overview/
- https://www.biostars.org/p/368420/
- https://bioinformatics.stackexchange.com/questions/3572/strange-per-sequence-gc-content-results
- cite:wang2012


- samtools sort (to postion)

******* [[http://www-huber.embl.de/HTSeq/doc/count.html#count][website]]
:PROPERTIES:
:ID:       6B7F3881-DA7D-4700-BC30-9EF83150D916
:END:

htseq-count [options] <alignment_file> <gff_file>

simple code: htseq-count -m union -r pos -i gene_name -a 10 --stranded=no agcgccg.bam Mus_musculus.GRCm38.84.gtf > output.counts

Important options
--stranded=no (defualt is yes)
--format bam (defualt is sam)
  - for paired end, order is important
    - do samtools sort (default) to put in position order
    - in HTSeq-counts, specify -r pos
    -a 10 (minimum quality, 10 is default)
    -m default is union, alternate intersection-strict and intersection-nonempty

    Why union is default: consider two genes, which share a stretch of common sequence such that for a read mapping to this stretch, the aligner cannot decide which of the two genes the read originated from and hence reports a multiple alignment. If we discard all such reads, we undercount the total output of the genes, but the ratio of expression strength (the “fold change”) between samples or experimental condition will still be correct, because we discard the same fratcion of reads in all samples. On the other hand, if we counted these reads for both genes, a subsequent diffential-expression analysis might find false positives:


  followed by the special counters, which count reads that were not counted for any feature for various reasons

  [[http://www-huber.embl.de/HTSeq/doc/counting.html][algorithm details]]

****** Basic counting script
:PROPERTIES:
:ID:       AE616985-EA77-4CF0-8FEA-60FF56A04AC3
:END:

counting is computationally intensive- script HTseq and
summarizeOverlaps on the cluster
variabilize RNA seq workflow- make into function
***** RNA-seq exploratory data analysis
:PROPERTIES:
:ID:       77d2a559-9858-453a-83c5-a646bbf473a3
:END:
- https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
- https://www.reddit.com/r/bioinformatics/comments/u1hsjy/great_thread_on_the_importance_of_eda_and/
- A more fundamental problem, however, arises from the high dimensional-
ity of the expression state. As the number of dimensions increases,
traditional distance metrics, such as Euclidean distance, gradually
lose their ability to distinguish the closest and most distant points18. cite:kharchenko2021


https://bookdown.org/rdpeng/exdata/k-means-clustering.html
- cite:vidman2019
Bergkvist,A., Rusnakova,V., Sindelka,R., Garda,J.M., Sjogreen,B.,
Lindh,D., Forootan,A. and Kubista,M. (2010) Gene expression
profiling–clusters of possibilities. Methods, 50, 323–335.
228. Beyene,J., Tritchler,D., Bull,S.B., Cartier,K.C., Jonasdottir,G.,
Kraja,A.T., Li,N., Nock,N.L., Parkhomenko,E., Rao,J.S. et al.
(2007) Multivariate analysis of complex gene expression and clinical
phenotypes with genetic marker data. Genet. Epidemiol., 31(Suppl.
1), S103–S109.

- cite:buschmann2016 pg 6011
- hierarchical clustering
  - eucledian
  - pearson
- partitioning
  - k-means
  - self-organizing maps
- latent variable
  - PCA
    - add scores and loadings for initial variable
- [[id:588f6536-9db9-4279-9e67-800890c2de23][LINK TO High dimension data transformation]]
****** Clustering
:PROPERTIES:
:ID:       22a6e4a4-60c2-44cf-8e5a-3ff351a02aa3
:END:
https://old.reddit.com/r/AskStatistics/comments/qcxb6c/evaluation_of_clustering_rand_index_and_nmi_i_do/
https://developers.google.com/machine-learning/clustering/clustering-algorithms
https://www.analyticsvidhya.com/blog/2016/11/an-introduction-to-clustering-and-different-methods-of-clustering/
https://www.datanovia.com/en/lessons/cluster-analysis-example-quick-start-r-code/
http://statweb.stanford.edu/~susan/courses/stats305c/examplesClusters.html
- https://mubaris.com/posts/kmeans-clustering/
https://uc-r.github.io/hc_clustering
https://www.biostars.org/p/287512/
- https://www.datanovia.com/en/blog/types-of-clustering-methods-overview-and-quick-start-r-code/

- cite:vidman2019
***** [[id:ee030cef-74d4-435d-8228-b85f0e6203db][Differential expression]]
***** [[id:91d46e60-eec3-48c7-97f1-b53eb1559b3f][Co-expression]]
***** RNA-seq visualization
:PROPERTIES:
:ID:       b1351d51-3ac1-4193-b944-850c44a903fd
:END:
- http://gepia.cancer-pku.cn/
- [[id:DB8463CB-97CB-49E6-8EDD-A5ECA2F140BF][visualization tools]]
- examples
  - [[https://www.cell.com/cell-reports/pdf/S2211-1247(20)30669-0.pdf][good seq analysis and visual example]]
- volcano
  - https://www.biostars.org/p/203876/
- heatmaps
  - http://compbio.ucsd.edu/making-heat-maps-r/
  - https://www.rdocumentation.org/packages/gplots/versions/3.0.3/topics/heatmap.2
- mapped reads were visualized by SeqMonk[[file:org.org_imgs/20161220_180037_4849srR.png]]
***** Methods by RNA population
****** exRNA
- [[cite:&hulstaert2021]]
- [[cite:&hulstaert2020]]
- [[cite:&exrnaqc2021]]
- spike-in use
  - cite:godoy2019
  - cite:giraldez2018

- cite:godoy2018
- cite:alexander2022
- spike ins use
  - cite:godoy2019
  - cite:giraldez2018
- [[id:fd462ab0-6aaf-4732-953b-141ad5e2ee72][small RNA sequencing]]
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9358052/
- https://www.biorxiv.org/content/10.1101/113050v1.full
- cite:vorperian2022
- https://www.sciencedirect.com/science/article/pii/S0092867419301679?via%3Dihub#sec4
- https://www.sciencedirect.com/science/article/pii/S2211124720315412?via%3Dihub
- cite:yeri2018


******* [[id:80cff175-c41d-42d4-aaa0-39546e0a6f1d][Plasma exRNA (cfRNA) extraction and isolation]]
******* [[id:51e5d70c-4a62-4960-880e-56f050194a0a][exRNA Biology]]
******* [[id:b2873d1a-837b-4927-b8cf-8a27e42d7bea][small RNA extraction]]
******* [[id:71289eeb-a929-44bd-a46e-a0c3dcd7a8f4][small RNA library preparation]]
******* [[id:fd462ab0-6aaf-4732-953b-141ad5e2ee72][small RNA sequencing]]
******* QC
- cite:lee2016exrna
******* Complete pipelines
- https://exrna.org/resources/software/
- pipelines- [[id:1db0ffb6-25b7-4c97-9aa0-263b8c96373b][exceRpt]]
- https://github.com/oncornalab/rnaexome from [[cite:&hulstaert2021]]
******* By transcript and sequencing
- single cell- https://www.nature.com/articles/s41587-021-01188-9
******* By fluid
- cite:godoy2018
******** Plasma
- non-human plasma small RNA
  - cite:vliet2017
- human plasma small RNA
  - cite:danielson2018
  - cite:godoy2018
  - cite:max2018
  - cite:yeri2018
  - cite:danielson2017
  - cite:juzenas2017
  - cite:yeri2017
  - cite:freedman2016
- plasma transcriptome RNA
  - cite:zhao2018
  - cite:chen2019exrna
******** urine
  - https://pubmed.ncbi.nlm.nih.gov/28303895/
******** salivary
  - https://pubmed.ncbi.nlm.nih.gov/28303895/
  - - https://www.ncbi.nlm.nih.gov/pubmed/28961734
******* General reference
- Extracellular RNA Communication Consortium
  - https://www.sciencedirect.com/science/article/pii/S0092867419302855?via%3Dihub
- - cell of origin -
  - https://www.biorxiv.org/content/10.1101/2021.05.05.441859v1.full
  - https://www.nature.com/articles/s41587-021-01188-9
- cite:alexander2022
- pipelines- [[id:1db0ffb6-25b7-4c97-9aa0-263b8c96373b][exceRpt]]

****** small RNA sequencing
:PROPERTIES:
:ID:       fd462ab0-6aaf-4732-953b-141ad5e2ee72
:END:
- cite:giraldez2018
- https://link.springer.com/protocol/10.1007%2F978-1-4939-9042-9_4
- cite:buschmann2016
******* General reference
:PROPERTIES:
:ID:       65a85f59-afd5-4c6b-a785-9b6269cc7a51
:END:
- miRQC cite:mestdagh2014
- cite:mehta2014
- [[https://www.biostars.org/t/smallrna/?sort=votes&limit=all%20time][biostars small RNA]]
- cite:korpelainen2015 chapter 12
- best practices
  - https://www.encodeproject.org/data-standards/rna-seq/small-rnas/


******* QC
:PROPERTIES:
:ID:       fc7aeeb1-d387-4379-be37-46ebf1244cd7
:END:
https://exrna.org/resources/data/data-quality-control-standards/
******* Pre-processing
:PROPERTIES:
:ID:       0c7d62b9-6ef4-43d1-8e21-c3aa536125bd
:END:
- read filtering
  - adapter trimming
  - quality filtering
  - read length trimming
    - remove at less than 16 nt (see cite:buschmann2016 ref 157,158)
    - these are adapter dimers and degraded RNA
- QC
  - filtered reads should have miRNA 21-23 nt peak and piRNA 30-32 nt peak
  - overrepresented sequence- k-mer distribution- see cite:buschmann2016 pg 6007
- [[id:1EE731A3-83DD-46AF-B988-2311307C335F][LINK TO Raw sequence QC and processing]]
******* [[id:51e5d70c-4a62-4960-880e-56f050194a0a][LINK TO Extracellular RNA biology]]
:PROPERTIES:
:ID:       49a0e52b-0bda-4237-8bab-ff2db54cf26d
:END:
******* Design
:PROPERTIES:
:ID:       4e8a8446-e84d-4c4f-8bd8-45ee611337b5
:END:
- read length
- spike-ins
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4538800/
- validation technologies
  - cannot distinguish isomiRs
  - qPCR expected to correlate >.9 w/ seq expression > 2^8 cite:yeri2018
  - qPCR more concordant than EdgeSeq or FirePlex cite:yeri2018
- what depth? https://www.researchgate.net/post/how_many_million_reads_for_miRNA_discovery
- reads per sample
  - https://www.qiagen.com/us/resources/faq?id=a119d5f0-0a2a-4612-828e-d8fa3cc37891&lang=en
******* [[id:b2873d1a-837b-4927-b8cf-8a27e42d7bea][LINK TO small RNA extraction]]
:PROPERTIES:
:ID:       dfc3a8a2-e4c4-4043-af71-2db93128c18e
:END:
******* [[id:4E23127E-4ED6-4802-A03C-4103AA8CAF29][LINK TO RNA Quality and Quantification]]
:PROPERTIES:
:ID:       4df90313-39e4-44ed-83cd-b98453e18358
:END:
******* small RNA library preparation
:PROPERTIES:
:ID:       71289eeb-a929-44bd-a46e-a0c3dcd7a8f4
:END:
- cite:wong2019
- cite:coenen-stass2018
- cite:qiagen2019biofluid_mirna
- Truseq small finds more tRNA fragments than other kits cite:yeri2018
- Bioscientific NextFlex maps more reads to miRNA than other kits cite:yeri2018 (~75% vs 65% for truseq in plasma)
- read diversity correlates to input volume, but b/w 550 and 650 at >1 RPM at volumes 200 uL to 5 mL cite:yeri2018 8c
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6161688/
- cite:buschmann2016
- cite:illumina2016truseq_small
- 10-50 ng isolated or 1 ug total https://support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/samplepreps_truseq/truseqsmallrna/truseq-small-rna-library-prep-kit-reference-guide-15004197-02.pdf
- library prep - cite:wright2019
- https://dx.doi.org/10.1186%2Fs12864-018-4726-6
- https://dx.doi.org/10.1186%2Fs12864-018-4491-6
- small RNA library preparation cite:illumina2016
- kit consistency required- see cite:yeri2018 8b
- with low input, use same kit for reproducibility cite:yeri2018
- cite:dard-dascot2018
******* [[id:3BC41E6A-00D3-4498-A48C-463F68CFD76B][LINK TO Massively parallel sequencing]]
:PROPERTIES:
:ID:       6584c294-78b9-42e5-a25b-28a069e112d8
:END:
******* Bioinformatics
:PROPERTIES:
:ID:       f46e278c-dd38-4603-9fa2-a973823e1333
:END:
- alignment to miRbase needs to be more conservative to avoid mismatches
- cite:buschmann2016
- https://link.springer.com/protocol/10.1007%2F978-1-4939-9042-9_4
******** Pre-processing
:PROPERTIES:
:ID:       0c7d62b9-6ef4-43d1-8e21-c3aa536125bd
:END:
- read filtering
  - adapter trimming
  - quality filtering
  - read length trimming
    - remove at less than 16 nt (see cite:buschmann2016 ref 157,158)
    - these are adapter dimers and degraded RNA
- QC
  - filtered reads should have miRNA 21-23 nt peak and piRNA 30-32 nt peak
  - overrepresented sequence- k-mer distribution- see cite:buschmann2016 pg 6007
- [[id:1EE731A3-83DD-46AF-B988-2311307C335F][LINK TO Raw sequence QC and processing]]
******** Alignment
:PROPERTIES:
:ID:       d657a691-e338-44ad-96e4-47b4b18a03d2
:END:
- alignment to miRbase needs to be more conservative to avoid mismatches
******** QC
:PROPERTIES:
:ID:       fc7aeeb1-d387-4379-be37-46ebf1244cd7
:END:
https://exrna.org/resources/data/data-quality-control-standards/
******** [[id:853dd34c-27be-4be8-9794-19f04a1e22b0][LINK TO Quantification and normalization]]
:PROPERTIES:
:ID:       9a349ccd-3d29-46b7-b4fa-71693d32c674
:END:
******** [[id:ee030cef-74d4-435d-8228-b85f0e6203db][LINK TO Differential expression]]
:PROPERTIES:
:ID:       56d53b46-2a62-4777-b5c7-eedbc306d4b4
:END:
cite:buschmann2016
"y. Even though comparisons of software
performances on small RNA-Seq data are scarce, a num-
ber of independent and extensive evaluations for mRNAs
based on either synthetic data with clearly defined prop-
erties (216), or on biological datasets with validated gene
expressions (217,218) have been made recently. While it
was shown that statistical power of almost all methods is
heavily dependent on the number of samples per condition
and less on sequencing depth, the variability of expression
changes in biological datasets affects each analysis tool dif-
ferently. Outliers, ‘ON/OFF’ expression changes, where a
gene is detected in only one condition, and lopsided ex-
pression patterns, where upregulations drastically outweigh
downregulations or vice versa, influence specificity (false
positive rate) and sensitivity (false negative rate) of each
method unequally. Nonetheless, some methods appear to
capture the true expression status of small RNAs better
than others. Most independent evaluations seem to agree
that calling differential expression with SAMSeq works well
for datasets with sufficient sample sizes of 10 or more. For
smaller datasets, edgeR and especially the more conserva-
tive DESeq (or DESeq2) are found to be the methods of
choice. On"
******** Functional analysis
:PROPERTIES:
:ID:       e597d527-5538-4f69-bd28-a24a70969317
:END:
cite:buschmann2016 pg6011
******** Integrated tools
:PROPERTIES:
:ID:       07055c47-6e24-4f59-bce4-0ab91a7f5377
:END:
- miRDeep2 / miRDeep
  - cite:friedlander2012
- exceRpt / genboree
  - http://github.gersteinlab.org/exceRpt/
  - using
    - cite:godoy2018
    - cite:freedman2016
  - http://genboree.org/java-bin/workbench.jsp
- miRge / miRge2 WOS 29 times cited
  - https://mirge3.readthedocs.io/en/latest/quick_start.html
  - https://github.com/mhalushka/miRge3.0
  - cite:lu2018mirge
  - https://github.com/mhalushka/miRge
    - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2287-y
- sRNAbench
  - https://academic.oup.com/nar/article/47/W1/W530/5494756
  - https://bioinfo2.ugr.es/ceUGR/srnabench/
  - sRNAtoolbox
    - https://arn.ugr.es/srnatoolbox/srnabench/
    - https://arn.ugr.es/srnatoolbox/static/sRNAtoolbox_manual.pdf
******* Applications
:PROPERTIES:
:ID:       55212621-170f-41aa-b9b5-b1c00385ffe6
:END:
******** Extracellular RNA bioinformatics
:PROPERTIES:
:ID:       cbf67ef7-39ec-45ec-b599-cdc0e8590cf7
:END:
- spike ins use
  - cite:godoy2019
  - cite:giraldez2018
- pipelines- [[id:1db0ffb6-25b7-4c97-9aa0-263b8c96373b][exceRpt]]
********* [[id:71289eeb-a929-44bd-a46e-a0c3dcd7a8f4][small RNA library preparation]]
********* [[id:b2873d1a-837b-4927-b8cf-8a27e42d7bea][small RNA extraction]]
********* By transcript and sequencing
- single cell- https://www.nature.com/articles/s41587-021-01188-9
********* By fluid
- cite:godoy2018
********** Plasma
- non-human plasma small RNA
  - cite:vliet2017
- human plasma small RNA
  - cite:danielson2018
  - cite:godoy2018
  - cite:max2018
  - cite:yeri2018
  - cite:danielson2017
  - cite:juzenas2017
  - cite:yeri2017
  - cite:freedman2016
- plasma transcriptome RNA
  - cite:zhao2018
  - cite:chen2019exrna
********** urine
  - https://pubmed.ncbi.nlm.nih.gov/28303895/
********** salivary
  - https://pubmed.ncbi.nlm.nih.gov/28303895/
  - - https://www.ncbi.nlm.nih.gov/pubmed/28961734
******** miRNA-seq
:PROPERTIES:
:ID:       903f32d8-bd13-496c-8d5c-26a63455936a
:END:
:LOGBOOK:
CLOCK: [2021-01-27 Wed 13:53]--[2021-01-27 Wed 13:57] =>  0:04
:END:
- spike-ins
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4538800/
- reads per sample
  - https://www.qiagen.com/us/resources/faq?id=a119d5f0-0a2a-4612-828e-d8fa3cc37891&lang=en
- cite:tam2015
- https://www.nature.com/articles/srep05150#Sec11
- http://www.mircode.org/faq.php
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4652620/
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6161688/
- [[id:5d9593fd-f083-450f-a510-449aa15ae670][LINK TO miRNA biology]]
- [[id:95b3c7f3-3b84-45b1-b8bf-5e2e6e2a6f88][LINK TO miRNA data]]
- spike-ins
  - cite:qiagen2018mirnaspike cite:exiqon2016exiseq
  - cite:locati2015
  - https://genome.cshlp.org/content/21/9/1543.long
- processes
  - preprocessing
    - read collapse just speads up downstream
      - see for read collapse https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0143066
  - QC
    - library complexity
    - https://arn.ugr.es/mirnaqc/
  - normalization
    - https://academic.oup.com/bib/article/16/6/950/226343
    - https://rnajournal.cshlp.org/content/18/6/1279.long
  - functional analysis
    - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4489228/
  - target identification
    - http://ophid.utoronto.ca/mirDIP/
  - [[id:82aafb6f-63a8-4ab7-867a-9add0e0c224a][LINK TO Integrated miRNA-mRNA analysis]]
- tools
  - manatee cite:handzlik2020
  - tool comparisons
    - https://www.frontiersin.org/articles/10.3389/fgene.2018.00022/full
    - cite:chen2019mirna
- general reference
  - cite:pritchard2012
    - citing, times cited WOS:
      - cite:braunwald2013
      - cite:dong2013
      - cite:schwarzenbach2014
      - " Whereas high-speed centrifugation (14,000–16,000 g) can remove living cells and cell debris, separating exosomes and small shedding vesicles requires ultracentrifugation (>110,000 g).30 Blood cell count has a significant effect on miRNA expression levels in plasma;31 therefore, samples from patients with an inflammatory process, who should have high white blood cell count, should be excluded from analysis. Haemolysis is also known to markedly influence miRNA expression patterns, because of a contamination of abundant intracellular miRNAs.31"
  - [[http://apps.pathology.jhu.edu/mirnaprogram/links/http://apps.pathology.jhu.edu/mirnaprogram/links/][Johns Hopkins microRNA Bio lab]]
  - https://www.biostars.org/t/mirna/
- protocols
  - cite:fu2018mirna
  - https://www.sciencedirect.com/science/article/pii/S266591312030131X#sec3
******* General reference
:PROPERTIES:
:ID:       65a85f59-afd5-4c6b-a785-9b6269cc7a51
:END:
- miRQC cite:mestdagh2014
- cite:mehta2014
- [[https://www.biostars.org/t/smallrna/?sort=votes&limit=all%20time][biostars small RNA]]
- cite:korpelainen2015 chapter 12
- best practices
  - https://www.encodeproject.org/data-standards/rna-seq/small-rnas/


******* Integrated miRNA-mRNA analysis
:PROPERTIES:
:ID:       82aafb6f-63a8-4ab7-867a-9add0e0c224a
:END:
- [[id:903f32d8-bd13-496c-8d5c-26a63455936a][LINK TO miRNA-seq]]
- example papers
  - https://bmccancer.biomedcentral.com/articles/10.1186/s12885-019-5983-8#Sec2
  - https://www.oncotarget.com/article/8713/text/
  - cite:li2017mirna
***** Methods by tissue source or isolation
:PROPERTIES:
:ID:       4d4a89d1-e039-4a89-b968-81c80f38d884
:END:
****** Salivary
  - https://pubmed.ncbi.nlm.nih.gov/28303895/
  - - https://www.ncbi.nlm.nih.gov/pubmed/28961734

****** Urine
  - https://pubmed.ncbi.nlm.nih.gov/28303895/

****** Plasma
- non-human plasma small RNA
  - cite:vliet2017
- human plasma small RNA
  - cite:danielson2018
  - cite:godoy2018
  - cite:max2018
  - cite:yeri2018
  - cite:danielson2017
  - cite:juzenas2017
  - cite:yeri2017
  - cite:freedman2016
- plasma transcriptome RNA
  - cite:zhao2018
  - cite:chen2019exrna

****** ffpe
  - cite:kashofer2013
  - FFPE Pre-analytic variability
    - cite:ahlfen2007rnaqual
      - storage conditions
      - cold ischemia time
      - fixation
        - type
        - time
      - sample thickness
    - biospecimens.cancer.gov/researchnetwork
    - cite:unlockingQiagen
    - https://www.ncbi.nlm.nih.gov/pubmed/25357115
    - By effect
      - Yield
        - - prolonged fix reduces yield- Bresters D, Schipper ME, Reesink HW, Boeser-Nunnink BD, Cuypers HT. The duration of fixation influences the yield of HCV cDNA-PCR products from formalin-fixed, paraffin- embedded liver tissue. J Virol Methods 1994;48:267–72.
        - cite:webster2015mining longer fixation times decrease RNA yields from paraffin-embedded tissues (herein 3 weeks in formalin resulted in half the RNA yield/sec- tion compared with 18 h) and increase RNA input needed during library preparation (see methods). This latter effect is likely due to ‘non-functional’ RNA (eg, adducts or damaged bases) that could be potentially mitigated by treatments like PreCR or other enzymatic and chemical methods.
      - At microarray expression
        - cite:kashofer2013quality
    - "Formalin fixation introduced major changes into microarray gene expression data and led to marked gene-to-gene variations in delta-ct values of qRT-PCR. We found that qRT-PCR efficiency and gene-to- gene variations were mainly attributed to differences in the efficiency of cDNA synthesis as the most sensitive step. These differences could not be reliably detected by quality assessment of total RNA isolated from formalin-fixed tissues by electrophoresis or spectrophotometry. "
    - At sequencing
  - cite:graw2015
  - Past expression tech
    - goals were biomarker discovery and patient stratification
    - cite:Zhao_2014 manifests its superiority in terms of the dynamic range, sampling depth, and has independence from pre-existing sequence information [1,2].
    - cite:webster2015mining When compared with conventional DNA microarrays, RNA-seq enriches for many additional fragments as it is not restricted to predefined probes and has (in principle) no limitations to dynamic range (Li et al., 2014b; SEQC/MAQC-III Consortium, 2014).
    - - https://www.ncbi.nlm.nih.gov/pubmed/14695316
    - - [[https://www.ncbi.nlm.nih.gov/pubmed/15509548][DASL]]
    - https://www.ncbi.nlm.nih.gov/pubmed/21347257
  - FFPE microarray and RT-PCR was useful
    - AlloMap is a non-invasive gene expression-based blood test that is used to manage the clinical care of heart transplant recipients, providing a quantified score for the risk of rejection based on the measurement of expression of 20 genes, a subset of which are related to immune system activation and signalling15,16.
    - - cite:penland2007rna
    - - - https://www.ncbi.nlm.nih.gov/pubmed/11568892/
    - Mittempergher, L., de Ronde, J.J., Nieuwland, M., Kerkhoven, R.M., Simon, I., Th Rutgers, E.J., Wessels, L.F., Van't Veer, L.J., 2011. Gene expression profiles from formalin fixed paraffin embedded breast cancer tissue are largely comparable to fresh fro- zen matched tissue. PLoS One 6, e17163.
  - But this expression tech is inferior
    - In other work14, systematic evalu- ation of RNA-seq-based and microarray-based classifiers found that RNA-seq outperformed arrays in characteriz- ing the transcriptome of cancer and performed similarly to arrays in clinical endpoint prediction.
    - does not depend on genome annotation, or on predesigned species-specific or tran- script-specific probes, for transcript measurement.
    - microarray limited application to clinic- lthough powerful, microarray-based assays can have limitations in some environments, such as those related to laboratory-to-laboratory variation in sample preparation that can affect reproducibility. Moreover, for some applications, microarray signal-to-noise ratios can affect the limit of detection. Interestingly, a number of additional cancer multi-gene profile tests are clinically available, such as OncoTypeDX (Genome Health)135 for breast cancer recurrence risk and Prolaris (Myriad)136 for prostate cancer aggressiveness. These tests are based on qRT-PCR technologies, rather than microarrays, largely owing to the belief that qRT-PCR is more reliable, reproducible, sensitive and accurate.
    - " qRT-PCR is generally deemed the ‘gold standard’ method for measuring transcript levels, particularly in the clinical space; however, there are a number of limitations, including the fact that although it is a high sample throughput technology, relatively few markers or measurements can be made in a single assay. "
    - rnaseq superior for
      - snvs
      - indels
      - isoforms
      - splice vars
      - chimerics
        - fusions
    - semiquan
    - no SNV discovery
    - no fusion
    - "RNA-seq provides much more detailed information on specific transcript expression patterns in clinico-genetic neuroblastoma subgroups than microarrays. " cite:zhang2015comparison
***** General
- my RNA-seq work
  - [[file:~/org/career.org::*high-grade head and neck characterization][high-grade head and neck characterization]]
  - [[id:e7a8362d-f4cf-45ec-a3f8-91b0ffbdc500][ffpe rna-seq project]]
  - [[file:~/data/career/pathology/molecular-pathology/ffpe-rna/ffpe-rnaseq.org]]
- [[https://www.nature.com/subjects/rna-sequencing/nmeth][Nature Methods RNA sequencing]]
- tool comparisons cite:sahraeian2017
- data
  - public RNA sequencing data
    - [[https://webshare.bioinf.unc.edu/public/mRNAseq_TCGA/UNC_mRNAseq_summary.pdf][TCGA RSEM]]
      - https://support.bioconductor.org/p/131066/
      - http://zyxue.github.io/2017/06/02/understanding-TCGA-mRNA-Level3-analysis-results-files-from-firebrose.html
      - https://bsbludwig.com/post/94066296740/what-do-tcgas-rnaseq-files-actually-show
      - http://seqanswers.com/forums/showthread.php?t=42911
      - https://support.bioconductor.org/p/65890/
  - see [[http://www.cbil.upenn.edu/BEERS/][BEERS]] for generating fake RNA seq data

- cite:korpelainen2015
  - [[http://rnaseq-book.blogspot.fi/][Korpelainen,15 web companion]]
- [[http://rnaseq.uoregon.edu/][RNAcyclopedia]]
- comparisons cite:poplawski2015
- [[https://hub.docker.com/r/mgibio/rnaseq/dockerfile][mgi rna-seq dockerfile]]
***** Nacent RNA sequencing
:PROPERTIES:
:ID:       7b2235be-1f01-4e51-9de9-adbdfc016b36
:END:
:LOGBOOK:
CLOCK: [2022-06-06 Mon 10:45]--[2022-06-06 Mon 11:25] =>  0:40
CLOCK: [2022-06-03 Fri 08:56]--[2022-06-03 Fri 09:31] =>  0:35
:END:
- https://www.scopus.com/results/results.uri?sort=plf-f&src=s&st1=%22precision+run-on+sequencing%22+OR+%22PRO-seq%22&sid=eeff59adf884bc66d7f41fb65fee374c&sot=b&sdt=b&sl=57&s=TITLE-ABS-KEY%28%22precision+run-on+sequencing%22+OR+%22PRO-seq%22%29&origin=searchbasic&editSaveSearch=&yearFrom=Before+1960&yearTo=Present
  - cite:mahat2016
    - https://www.scopus.com/results/citedbyresults.uri?sort=cp-f&cite=2-s2.0-84979882531&src=s&nlo=&nlr=&nls=&imp=t&sid=83fb718ed42783d656150cb229690a68&sot=cite&sdt=a&sl=0&origin=resultslist&offset=1&txGid=80fce121791a099b206e9017b82f705f&isValidNewDocSearchRedirection=false
      - https://www.nature.com/articles/nature20149
      - https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1870-5
      - https://www.nature.com/articles/s41467-017-00151-0
  - https://www.nature.com/articles/nprot.2016.086
- https://genome.cshlp.org/content/27/11/1816
- https://www.scopus.com/results/results.uri?cc=10&sort=cp-f&src=s&st1=%22nascent+RNA+sequencing%22&nlo=&nlr=&nls=&sid=a567d38581c262627b32ded722d56613&sot=b&sdt=b&sl=39&s=TITLE-ABS-KEY%28%22nascent+RNA+sequencing%22%29&ss=cp-f&ps=r-f&editSaveSearch=&origin=resultslist&zone=resultslist
  - cite:core2008
  - cite:henriques2018
***** Integrating results across RNA-seq datasets
- cite:rajkumar2015
- https://www.nature.com/articles/s42003-023-04791-5
***** Writeup
- https://www.nature.com/articles/s42003-023-04791-5
***** General reference
- cite:li2014
- cite:su2014
***** Complete pipelines
:PROPERTIES:
:ID:       3b95fbca-3d9a-479c-9285-f8f328c5114f
:END:
- [[https://github.com/zhxiaokang/RASflow/tree/master][RASflow]]
- [[https://bitbucket.org/cfce/viper/src/master/viper.snakefile][viper]]
- [[http://bioinf.wehi.edu.au/marray/jsm2005/lab4/lab4.html][A limma workflow to try]]
- [[http://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html][another workflow example]]
- https://github.com/BIMSBbioinfo/pigx_rnaseq
****** Pipeline Comparisons
:PROPERTIES:
:ID:       7B9AD35F-1511-4CB2-BB5A-0989EFEAADF3
:END:
[[file:~/Box%20Sync/library/Fonseca,14%20RNA%20seq%20gene%20profiling%20a%20systematic%20empirical%20comparison.PDF][Fonseca,14 RNA seq gene profiling a systematic empirical comparison]]
[[file:~/Box%20Sync/library/Sigurgeirsson,14%20Analysis%20of%20stranded%20information%20using%20an%20automated%20procedure%20for%20strand%20specific%20RNA%20sequencing.pdf][Sigurgeirsson,14 Analysis of stranded information using an automated procedure for strand specific RNA sequencing]]
***** Methods for non-expression analysis
****** cell enumeration through bulk RNA-seq
:PROPERTIES:
:CREATED:  [2020-08-05 Wed 08:36]
:ID:       1a1d5b73-8070-402a-9f26-e147d66ca081
:END:
******* CIBERSORTx
:PROPERTIES:
:ID:       bb55c105-5977-4a01-802e-4bc1838f58bb
:END:

https://www.biostars.org/p/428905/
    - raw counts will be converted to CPM in cibersortx- see https://cibersortx.stanford.edu/tutorial.php
- cibersortx
  - cite:newman2015
  - [[mu4e:msgid:BYAPR02MB5592ACF7A79C1137A6F0A974B77B0@BYAPR02MB5592.namprd02.prod.outlook.com][Re: cibersortx from docker]]
  - docker pull lyronctk/cibersortxfractions
- have access token for docker: https://cibersortx.stanford.edu/getoken.php
****** cell enumeration through bulk RNA-seq
:PROPERTIES:
:CREATED:  [2020-08-05 Wed 08:36]
:ID:       1a1d5b73-8070-402a-9f26-e147d66ca081
:END:
******* CIBERSORTx
:PROPERTIES:
:ID:       bb55c105-5977-4a01-802e-4bc1838f58bb
:END:

https://www.biostars.org/p/428905/
    - raw counts will be converted to CPM in cibersortx- see https://cibersortx.stanford.edu/tutorial.php
- cibersortx
  - cite:newman2015
  - [[mu4e:msgid:BYAPR02MB5592ACF7A79C1137A6F0A974B77B0@BYAPR02MB5592.namprd02.prod.outlook.com][Re: cibersortx from docker]]
  - docker pull lyronctk/cibersortxfractions
- have access token for docker: https://cibersortx.stanford.edu/getoken.php
**** Allele-specific expression

******* Applications for bulk RNA-seq

:PROPERTIES:
:ID:       a9e9bbf7-8012-456a-9b45-0c86aacd84c7
:END:
- [[*cell enumeration through bulk RNA-seq][cell enumeration through bulk RNA-seq]]
- deconvolution
  - http://labs.pathology.jhu.edu/halushka/science/
  - tools
    - ISOpure
      - https://link.springer.com/article/10.1186/s12859-015-0597-x#Sec1
  - cite:newman2015
- Clinical applications of RNA-seq
  - cite:byron2016translating
  - cite:Kalari_2014
  - cite:Van_Keuren_Jensen_2014
******** eQTL (expression quantitative trait loci) mapping
:PROPERTIES:
:ID:       2CA85A6B-7B1E-4A38-96AA-717847A59168
:END:
- identifies genetic variants that explain variation in gene expression
- w/ RNA-seq via splAdder
  - https://github.com/ratschlab/spladder
  - https://academic.oup.com/bioinformatics/article/32/12/1840/1744052?login=true
**** Spatial transcriptomics
:PROPERTIES:
:ID:       053daf71-9f9f-4b5e-b400-222c4d24afea
:END:
- scopus
- https://www.nature.com/articles/s41592-019-0548-y
- https://www.nature.com/articles/s41586-022-05202-1
- Platforms
  | Platform   | Areas per slide | Barcodes per area | Detection                     |
  |------------+-----------------+-------------------+-------------------------------|
  | 10x Visium |               4 |              5000 | in situ capture, seq by synth |
  | GeoMx      |                 |                   | in situ hyb w/ oligo detect   |
  | ish        |                 |                   | in sity hyb                   |
  - Visium
    - https://mail.google.com/mail/u/0/#inbox/FMfcgxwLtQMfThxFrGwWlHQFgpTTDrJJ
    - https://mail.google.com/mail/u/0/#inbox/FMfcgxwLtGthWWRrdltpWVLTXhQKnZDs https://support.10xgenomics.com/spatial-gene-expression/sequencing/doc/specifications-sequencing-requirements-for-visium-spatial-gene-expression
  - GeoMx
    - https://www.scopus.com/results/results.uri?sort=plf-f&src=s&st1=geomx&sid=cceac18bcbdba4f776ce2a6635ba19bc&sot=b&sdt=b&sl=20&s=TITLE-ABS-KEY%28geomx%29&origin=searchbasic&editSaveSearch=&yearFrom=Before+1960&yearTo=Present
      - Only covers protein- cite:van2019
      - cite:nielsen2020
    - https://mail.google.com/mail/u/0/#inbox/FMfcgzGmvffBFCTQTwfXgXkvZNcgdlrN
  - Ish https://science.sciencemag.org/content/360/6389/660
- Reference
  - https://www.scopus.com/results/results.uri?cc=10&sort=cp-f&src=s&st1=%22spatial+transcriptomics%22&nlo=&nlr=&nls=&sid=f8602d6245cf86196aeef7c88b5ab0dd&sot=b&sdt=cl&cluster=scosubtype%2c%22re%22%2ct&sl=40&s=TITLE-ABS-KEY%28%22spatial+transcriptomics%22%29&ss=cp-f&ps=r-f&editSaveSearch=&origin=resultslist&zone=resultslist
    - https://www.sciencedirect.com/science/article/pii/S0958166916302397?via%3Dihub
    - Proof of concept https://www.science.org/doi/10.1126/science.aaf2403
  - [[file:~/repos/org/career.org::*Zander spatial transcriptomics zoom][Zander spatial transcriptomics zoom]]



 MERFISH2 or SeqFISH3 to averages across cells spanning dozens of micrometers in spot-based protocols such as spatial transcriptomics4. Methods also vary in the number of molecular features acquired: from tens in fluorescence in situ hybridization (FISH), cyclic immunofluorescence (CyCIF)5 and imaging mass cytometry (IMC)6 to hundreds or thousands in specialized probe-based spatial transcriptomics methods (MERFISH2 or SeqFISH3) or imaging mass spectrometry7, and tens of thousands in spot-based spatial transcriptomics, such as Slide-Seq8,9, Visium10, DBiT-seq11 and high-definition spatial transcriptomics (HDST)12. Suc


| Platform    | Description                 | ffpe | multiplex | resolution | species |
|-------------+-----------------------------+------+-----------+------------+---------|
| MERFISH2    | DNA probe amp               | no?  |           |            |         |
| SeqFISH     |                             |      |           |            |         |
| IMC         |                             |      |           |            |         |
| Slide-Seq2  |                             |      |           |            |         |
| Visium ffpe | probe                       | yes  |           |            |         |
| Visium      | onslide cDNA barcode to seq | no   |           |            |         |
| DBiT-seq    |                             |      |           |            |         |
| HDST        |                             |      |           |            |         |
| RNAscope    | DNA probe amp               |      |           |            |         |
| DSP         | Probe cleave to array       | yes  |           |            |         |
**** Expression flux
- https://journals.physiology.org/doi/full/10.1152/physiolgenomics.00128.2018
**** Trajectories
[[https://www.cell.com/cell-reports/pdf/S2211-1247(20)30367-3.pdf]]
**** Co-expression / pathway enrichment
:PROPERTIES:
:ID:       91d46e60-eec3-48c7-97f1-b53eb1559b3f
:END:
https://uclouvain-cbio.github.io/WSBIM2122/sec-gsea.html
[[https://www.reddit.com/r/bioinformatics/comments/13vk8r5/comparing_gsea_results_between_different_species/][clusterprofiler term to gene]]
- cite:reimand2019
- https://old.reddit.com/r/bioinformatics/comments/r7hzoq/annual_ingenuity_subscription_sticker_shock_and/
- https://www.sciencedirect.com/science/article/pii/S2666166721004755
- https://hbctraining.github.io/Training-modules/DGE-functional-analysis/lessons/02_functional_analysis.html

- cite:sanchez-vega2018
https://www.sciencedirect.com/science/article/pii/S2666166721004755
https://www.reddit.com/r/bioinformatics/comments/tu61pw/doubt_over_pathway_analysis_approach/
https://yulab-smu.top/biomedical-knowledge-mining-book/index.html
https://girke.bioinformatics.ucr.edu/GEN242/tutorials/rfea/rfea/
https://bioinformaticsbreakdown.com/how-to-gsea/
https://metascape.org/gp/index.html#/main/step1
https://yulab-smu.top/biomedical-knowledge-mining-book/index.html

- https://www.biorxiv.org/content/10.1101/2021.12.19.473368v1
- variancePartition https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1323-z

***** Methods
- https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1482-6
- [[id:9b222d1c-ab61-450c-8b77-a98ec3213867][gage]]
  - https://genviz.org/module-04-expression/0004/03/01/pathwayAnalysis/
- Aggregate - clusterProfiler
  - cite:yu2012
  - http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html
- iPANDA https://www.nature.com/articles/ncomms13427#Sec10
****** Over-representation / hypergeometric test
- https://david.ncifcrf.gov/gene2gene.jsp
- https://maayanlab.cloud/Enrichr/
- Fisher exact
  - https://maayanlab.cloud/Enrichr/
- Proposed in cite:boyle2004
****** [[id:D68E1182-1F7B-4160-A6DD-BD2635F8085F][GSEA]] (Gene set enrichment)
****** Methods and associated tools
:PROPERTIES:
:ID:       d11cda99-d3a3-4549-a828-d065c6979c5d
:END:
- [[id:948289d8-3a77-471a-b068-0507285e0bac][wgcna]]
  - wgcna
    - https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
    - cite:vanDam2018
      #+begin_src bash
    mkdir -p /tmp/wgnca-example
    cd /tmp/wgnca-example/
    wget https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip
    unzip FemaleLiver-Data.zip
    #+end_src
      #+begin_src R
  setwd("/tmp/wgnca-example")
  source("~/repos/basecamp/bin/setup.R")
  #
  # ez install of wgcna and dependencies for R 3.6
  if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
  BiocManager::install("fastLiquidAssociation")
  #
  library(WGCNA)
  options(stringsAsFactors=F)
  femData=read.csv("./LiverFemale3600.csv")
  datExpr0=as.data.frame(t(femData[,-c(1:8)]))
  head(datExpr0)
  gsg=goodSamplesGenes(datExpr0, verbose=3)
  gsg$allOK
  if (!gsg$allOK)
  {
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
  printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
  printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
  }
  #
  sampleTree=hclust(dist(datExpr0),method="average")
  plot(sampleTree)
  #
  # Plot a line to show the cut
  abline(h = 15, col = "red");
  # Determine cluster under the line
  clust = cutreeStatic(sampleTree, cutHeight = 15, minSize = 10)
  table(clust)
  # clust 1 contains the samples we want to keep.
  keepSamples = (clust==1)
  datExpr = datExpr0[keepSamples, ]
  nGenes = ncol(datExpr)
  nSamples = nrow(datExpr)
  #
  traitData = read.csv("ClinicalTraits.csv")
  dim(traitData)
  names(traitData)
  # remove columns that hold information we do not need.
  allTraits = traitData[, -c(31, 16)]
  allTraits = allTraits[, c(2, 11:36) ]
  dim(allTraits)
  names(allTraits)
  # Form a data frame analogous to expression data that will hold the clinical traits.
  femaleSamples = rownames(datExpr)
  traitRows = match(femaleSamples, allTraits$Mice)
  datTraits = allTraits[traitRows, -1]
  rownames(datTraits) = allTraits[traitRows, 1]
  collectGarbage()
  #
  save(datExpr, datTraits, file = "FemaleLiver-01-dataInput.RData")
  #
  enableWGCNAThreads()
  lnames=load(file="FemaleLiver-01-dataInput.RData")
  #
  lnames
    #+end_src
***** Databases
:PROPERTIES:
:ID:       f7eed3be-f505-451c-b191-1a58373fb180
:END:
- [[id:c79bfe84-0f15-474e-8a22-1f210605ed5a][MSigDB]]
- https://academic.oup.com/bioinformatics/article/33/3/414/2875813
- https://metascape.org/gp/index.html#/main/step1
- gene sets for mouse GSEA
  - https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html
  - recent, high impact papers citing MSigDB and aligning to mm10
    - https://science.sciencemag.org/content/352/6288/aad3018
    - http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC5924693&blobtype=pdf
    - https://www.cell.com/cell/pdf/S0092-8674(17)31385-5.pdf
    - https://stm.sciencemag.org/content/scitransmed/suppl/2018/01/12/10.424.eaan5488.DC1/aan5488_SM.pdf
***** Applciation Examples
- cite:sanchez-vega2018
***** Gene regulatory network inference
- tools
  - GENIE
    - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0012776
    - https://www.bioconductor.org/packages/devel/bioc/vignettes/GENIE3/inst/doc/GENIE3.html
***** Gene co-expression visualization
:PROPERTIES:
:ID:       fe3baa6c-d28e-48c4-91b1-0539e01a4cb1
:END:
- Need ssGSEA / GSVA for pathway heatmap https://bioconductor.org/packages/release/bioc/vignettes/GSVA/inst/doc/GSVA.html
- [[https://enrichmentmap.readthedocs.io/en/docs-2.2/QuickStartGuide.html][enrichement map]]
  - https://enrichmentmap.readthedocs.io/en/latest/Parameters.html
- https://kiharalab.org/web/navigo/views/goset.php
- http://www.visantnet.org/visantnet.html
- [[https://bioconductor.org/packages/release/bioc/vignettes/pathview/inst/doc/pathview.pdf][Pathview]]
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9653450/
- visualization
  - heatmaps
    - https://www.frontiersin.org/files/Articles/425394/fgene-10-00858-HTML/image_m/fgene-10-00858-g002.jpg
    - https://www.bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
  - wire diagrams
  - bar and lolipop for p-value and set size
    - https://www.nature.com/articles/s41598-018-29420-8
    - https://www.frontiersin.org/articles/10.3389/fgene.2019.00858/full
- https://github.com/PathwayCommons/chibe
- https://github.com/Adrienrougny/sbgntikz/
***** Tutorials
- https://bioinformaticsbreakdown.com/how-to-gsea/
- Nature Protocol: cite:reimand2019
- tutorial: https://girke.bioinformatics.ucr.edu/GEN242/tutorials/rfea/rfea/
***** General reference
:PROPERTIES:
:ID:       98048a64-6bd5-4107-ab7e-573b57758f12
:END:
- cite:maleki2020
- [[id:2CA85A6B-7B1E-4A38-96AA-717847A59168][QTL mapping]]
- tutorial on transcriptional regulatory networks from total RNA-seq cite:chouvarine2021
- cite:vandam2018
- https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html
- pathway annotation and conservation
  - https://academic.oup.com/nar/article/48/D1/D479/5626534#190997651
- method and tool comparisons
  - cite:amadoz2019
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1790-4
- cite:reimand2019
- https://link.springer.com/article/10.1186/s40246-019-0226-2
- https://www.sciencedirect.com/science/article/pii/S0022519314001969?via%3Dihub
- cite:creixell2015
- cite:rahmatallah2016

***** Methods and associated tools
:PROPERTIES:
:ID:       d11cda99-d3a3-4549-a828-d065c6979c5d
:END:
- https://maayanlab.cloud/Enrichr/
- [[id:2CA85A6B-7B1E-4A38-96AA-717847A59168][QTL mapping]]
- iPANDA
  - https://www.nature.com/articles/ncomms13427#Sec10
    - https://github.com/varnivey/ipanda/blob/master/MANUAL.txt
- Aggregate - clusterProfiler
  - http://yulab-smu.top/clusterProfiler-book/index.html
  - http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html
- Gene set enrichment
  - https://david.ncifcrf.gov/gene2gene.jsp
  - https://genviz.org/module-04-expression/0004/03/01/pathwayAnalysis/
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - http://software.broadinstitute.org/gsea/msigdb/collections.jsp
  - https://monashbioinformaticsplatform.github.io/RNAseq-DE-analysis-with-R/RNAseq_DE_analysis_with_R.html#gene-set-enrichment
  - pathway enrichment
    - - https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
    - - https://www.nature.com/articles/s41596-018-0103-9.pdf?proof=true19
    - https://bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
  - [[id:9b222d1c-ab61-450c-8b77-a98ec3213867][gage]]
  - cite:rahmatallah2014
  - https://academic.oup.com/bioinformatics/article/33/3/414/2875813
  - [[id:D68E1182-1F7B-4160-A6DD-BD2635F8085F][GSEA]]
- Fisher exact
  - https://maayanlab.cloud/Enrichr/
- [[id:948289d8-3a77-471a-b068-0507285e0bac][wgcna]]
  - wgcna
    - https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
    - cite:vanDam2018
      #+begin_src bash
    mkdir -p /tmp/wgnca-example
    cd /tmp/wgnca-example/
    wget https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip
    unzip FemaleLiver-Data.zip
    #+end_src
      #+begin_src R
  setwd("/tmp/wgnca-example")
  source("~/repos/basecamp/bin/setup.R")
  #
  # ez install of wgcna and dependencies for R 3.6
  if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
  BiocManager::install("fastLiquidAssociation")
  #
  library(WGCNA)
  options(stringsAsFactors=F)
  femData=read.csv("./LiverFemale3600.csv")
  datExpr0=as.data.frame(t(femData[,-c(1:8)]))
  head(datExpr0)
  gsg=goodSamplesGenes(datExpr0, verbose=3)
  gsg$allOK
  if (!gsg$allOK)
  {
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
  printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
  printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
  }
  #
  sampleTree=hclust(dist(datExpr0),method="average")
  plot(sampleTree)
  #
  # Plot a line to show the cut
  abline(h = 15, col = "red");
  # Determine cluster under the line
  clust = cutreeStatic(sampleTree, cutHeight = 15, minSize = 10)
  table(clust)
  # clust 1 contains the samples we want to keep.
  keepSamples = (clust==1)
  datExpr = datExpr0[keepSamples, ]
  nGenes = ncol(datExpr)
  nSamples = nrow(datExpr)
  #
  traitData = read.csv("ClinicalTraits.csv")
  dim(traitData)
  names(traitData)
  # remove columns that hold information we do not need.
  allTraits = traitData[, -c(31, 16)]
  allTraits = allTraits[, c(2, 11:36) ]
  dim(allTraits)
  names(allTraits)
  # Form a data frame analogous to expression data that will hold the clinical traits.
  femaleSamples = rownames(datExpr)
  traitRows = match(femaleSamples, allTraits$Mice)
  datTraits = allTraits[traitRows, -1]
  rownames(datTraits) = allTraits[traitRows, 1]
  collectGarbage()
  #
  save(datExpr, datTraits, file = "FemaleLiver-01-dataInput.RData")
  #
  enableWGCNAThreads()
  lnames=load(file="FemaleLiver-01-dataInput.RData")
  #
  lnames
    #+end_src
***** General reference
:PROPERTIES:
:ID:       98048a64-6bd5-4107-ab7e-573b57758f12
:END:
- https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html
- pathway annotation and conservation
  - https://academic.oup.com/nar/article/48/D1/D479/5626534#190997651
- method and tool comparisons
  - cite:amadoz2019
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1790-4
- cite:reimand2019
- https://link.springer.com/article/10.1186/s40246-019-0226-2
- https://www.sciencedirect.com/science/article/pii/S0022519314001969?via%3Dihub
- cite:creixell2015
- cite:rahmatallah2016
**** Gene expression dynamics
- [[https://www.scopus.com/results/citedbyresults.uri?sort=cp-f&cite=2-s2.0-0344464733&src=s&nlo=&nlr=&nls=&imp=t&sid=1aade77c6b42db34d11646993e38a758&sot=cite&sdt=a&sl=0&origin=resultslist&offset=1&txGid=3af857e8bcb60fab910d99917b405130][scopus citing GEDI]]
***** Time course
:PROPERTIES:
:ID:       43f1772e-f8ce-4f6b-a4bd-d35d381b89c9
:END:
  - cite:spies2017
  - https://support.bioconductor.org/p/65676/#66860
  - https://www.biostars.org/p/138988/
  - https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#group-specific-condition-effects-individuals-nested-within-groups
  - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
  - http://seqanswers.com/forums/showthread.php?t=64039
  - https://www.sciencedirect.com/science/article/pii/S2001037015000392
**** General reference
:PROPERTIES:
:ID:       70491293-d832-4947-b1bd-a0f4ae066c17
:END:
- https://www.cell.com/fulltext/S0092-8674%2812%2901226-3
- http://gepia.cancer-pku.cn/
- Regulation of gene expression https://www.reddit.com/r/bioinformatics/comments/sjp16n/database_of_differential_gene_expression_based_on/

- my RNA-seq work
  - [[file:~/org/career.org::*high-grade head and neck characterization][high-grade head and neck characterization]]
  - [[id:e7a8362d-f4cf-45ec-a3f8-91b0ffbdc500][ffpe rna-seq project]]
  - [[file:~/data/career/pathology/molecular-pathology/ffpe-rna/ffpe-rnaseq.org]]
- [[https://www.nature.com/subjects/rna-sequencing/nmeth][Nature Methods RNA sequencing]]
- tool comparisons cite:sahraeian2017
- data
  - public RNA sequencing data
    - [[https://webshare.bioinf.unc.edu/public/mRNAseq_TCGA/UNC_mRNAseq_summary.pdf][TCGA RSEM]]
      - https://support.bioconductor.org/p/131066/
      - http://zyxue.github.io/2017/06/02/understanding-TCGA-mRNA-Level3-analysis-results-files-from-firebrose.html
      - https://bsbludwig.com/post/94066296740/what-do-tcgas-rnaseq-files-actually-show
      - http://seqanswers.com/forums/showthread.php?t=42911
      - https://support.bioconductor.org/p/65890/
  - see [[http://www.cbil.upenn.edu/BEERS/][BEERS]] for generating fake RNA seq data
- cite:korpelainen2015
  - [[http://rnaseq-book.blogspot.fi/][Korpelainen,15 web companion]]
- [[http://rnaseq.uoregon.edu/][RNAcyclopedia]]
- https://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools
- [[id:386d899a-9208-4d5c-93d8-3750b22d9748][RNA methods]]
***** Ideas
:PROPERTIES:
:ID:       B5E9166C-CB80-4578-BD90-1E694B46F192
:END:
- http://gepia.cancer-pku.cn/
[[http://bioinf.wehi.edu.au/marray/jsm2005/lab4/lab4.html][A limma workflow to try]]
[[http://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html][another workflow example]]

see Love,16 section "Rich visualization and reporting of results"

****** Pipeline Comparisons
:PROPERTIES:
:ID:       7B9AD35F-1511-4CB2-BB5A-0989EFEAADF3
:END:
>>>>>>> e60be068dcbc8c18477fc28e00ee3b742dcf0110

[[file:~/Box%20Sync/library/Fonseca,14%20RNA%20seq%20gene%20profiling%20a%20systematic%20empirical%20comparison.PDF][Fonseca,14 RNA seq gene profiling a systematic empirical comparison]]
[[file:~/Box%20Sync/library/Sigurgeirsson,14%20Analysis%20of%20stranded%20information%20using%20an%20automated%20procedure%20for%20strand%20specific%20RNA%20sequencing.pdf][Sigurgeirsson,14 Analysis of stranded information using an automated procedure for strand specific RNA sequencing]]




*** Immunoprofiling and immunophenotyping
:PROPERTIES:
:ID:       7bb87c84-6b06-44ef-9a1d-1bb9b9e57295
:END:
- https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0213684
- methods
  - [[id:2bbeb644-af54-4f07-80ad-3584e1e21b04][single-cell RNA sequencing (scRNA-seq)]]
- cite:saloura2019
- https://en.wikipedia.org/wiki/Immunological_Genome_Project#Gene-expression_compendium
**** Deconvolution, immunoprofiling/ immunophenotyping, cell type ennumeriation with rnaseq
:PROPERTIES:
:CREATED:  [2019-10-03 Thu 10:22]
:ID:       e07fee1b-d7fd-42ce-9639-ff803cfc9c8e
:END:
:LOGBOOK:
CLOCK: [2020-02-13 Thu 11:25]--[2020-02-13 Thu 11:33] =>  0:08
CLOCK: [2020-02-13 Thu 10:41]--[2020-02-13 Thu 11:05] =>  0:24
:END:
- https://www.nature.com/articles/s43018-022-00356-3#code-availability
- https://www.ncbi.nlm.nih.gov/pubmed/32036085
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1257-4
- cite:newman2015
- for rna-seq data, see cite:newman2019 suppliment, do transcript TPM (salmon) and sum to gene
- The following metrics are provided for each mixture sample:
  - P-value: Statistical significance of the deconvolution result across all cell subsets; useful for filtering out results with a poor "goodness of fit." Increase the number of permutations (input page) to increase the number of significant digits.
  - Correlation: Pearson's correlation coefficient (R), generated from comparing the original mixture with the estimated mixture, the latter of which is calculated using imputed cell fractions and corresponding expression profiles from the signature genes file. Of note, the correlation is restricted to signature genes.
  - RMSE: Root mean squared error between the original mixture and the imputed mixture, restricted to genes in the signature gene file.
- in mouse
  - generating a reference from published data
    - single cell
    - array
      - cite:chen2017
        - https://github.com/wuaipinglab/ImmuCC/tree/master/tissue_immucc
  - using pre-compiled reference
    - cite:wang2017a
      - https://github.com/lintian0616/bica
    - convert to human and use lm22
- enrichment- ID at pathway or gene set
  - by ssGSEA
    - ESTIMATE
    - xCell
- cite:liu2019a
  - immune repetoire sequencing- see liu2019a ref83, table 4, ref88-92
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5996037/
- https://www.nature.com/articles/s43018-019-0022-x
- https://www.sciencedirect.com/science/article/pii/S2589004218301871
- https://www.nature.com/articles/nbt.3724
- cite:yalcin2012
- https://mixcr.readthedocs.io/en/master/index.html
- https://www.nature.com/articles/s41592-019-0529-1
- https://www.biostars.org/p/354696/
- cite:vonstechow2017
- cell type enumeration with RNA-seq
  - https://www.nature.com/articles/s41586-019-1773-3

***** immunoprofiling / immunophenotyping
:PROPERTIES:
:CREATED:  [2019-10-30 Wed 10:52]
:ID:       a143e3d0-35b3-4579-8831-9710791e0a6a
:END:
- cite:liu2019a
- cite:papalexi2018 - immune cell heterogeneity by scRNAseq

**** Cytometry
:PROPERTIES:
:ID:       b7b36cba-0d4d-41b0-ae5d-79b0f0f67d2e
:END:
***** Mass cytometry, cytometry by time of flight (CyTOF)
:PROPERTIES:
:ID:       881d23a0-c943-47ac-9bf5-ef965f8a80e3
:END:
- https://onlinelibrary.wiley.com/doi/10.1002/cyto.a.23621
- CATALYST
  - https://www.bioconductor.org/packages/release/bioc/html/CATALYST.html
  - https://www.bioconductor.org/packages/release/bioc/vignettes/CATALYST/inst/doc/differential.html
***** Cytometry


- cite:Denny_2011
- cite:Kho_2011


**** Immune cell sequencing
- cite:bentham2021
*** Liquid biopsy
- liquid biopsy somascan https://www.cell.com/cell/fulltext/S0092-8674(23)01033-4#secsectitle0090

https://www.nature.com/articles/s43018-022-00498-4
https://www.nature.com/articles/s41416-022-01881-9
- cite:illumina2022
**** Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)
:PROPERTIES:
:ID:       0f1f4aa4-c755-42ff-8f81-8a32c14293a0
:END:
***** Integrated hybrid capture and genome-wide- MAESTRO
:PROPERTIES:
:ID:       55b77210-fcb6-438d-a546-0c3605f6f5c4
:END:
cite:gydush2022
***** cfDNA long read sequencing
- cfDNA nanopore for meth and COO [[cite:&katsman2022cfdnananopore]]
***** Promenent research groups
****** Adalsteinsson group
[[id:55b77210-fcb6-438d-a546-0c3605f6f5c4][Integrated hybrid capture and genome-wide- MAESTRO]]
****** Landau Group
:PROPERTIES:
:ID:       b4b2a35f-e59a-4e84-995e-ca16d7fd190d
:END:
cite:widman2022
***** [[id:dd8b334b-e431-4709-8485-bf0dda1760a5][cfDNA inferred tissue gene expression]]
***** [[id:21d876e8-8646-4773-91c1-c4385c3218c7][cfDNA copy number calling]]
***** [[id:47c4ed56-1dea-48dc-bb25-ddbb0a4cd359][cfDNA tissue of origin discovery]]
***** [[id:c4dde8d6-1a02-42c7-a6b6-786b757fcad1][cfDNA nucleosome profiling]]
***** [[*cfDNA methylation][cfDNA methylation]]
***** [[*cfDNA fragmentomics][cfDNA fragmentomics]]
***** cfDNA topology
:PROPERTIES:
:ID:       2d0d13d9-857a-4285-89e6-4614f749afd3
:END:
- [[id:c4dde8d6-1a02-42c7-a6b6-786b757fcad1][cfDNA nucleosome profiling]]
- https://insight.jci.org/articles/view/159590/pdf
***** [[*cfDNA small variant calling][cfDNA small variant calling]]
***** Tumor-informed methods
- phylogenic for cancer evolution cite:abbosh2017
***** WGS cfDNA methods
- phased seq of multiple muts https://www.nature.com/articles/s41587-021-00981-w
- mutational signatures for LP WGS [[cite:&wan2022]]
- [[id:55b77210-fcb6-438d-a546-0c3605f6f5c4][Integrated hybrid capture and genome-wide- MAESTRO]]
- https://www.medrxiv.org/content/10.1101/2022.12.06.22283116v1
- mutation profiling in low coverage wgs https://www.nature.com/articles/s41467-022-32598-1
- https://www.biorxiv.org/content/10.1101/2022.01.17.476508v1.full.pdf
- cite:zviran2020
***** cfDNA and ctDNA-specific visualizations
- [[file:resources/moding2020fig4b.png]] cite:moding2020 fig4b
- tumor fraction ratios [[./biotools.org_20220817_135512.png]]
- https://d0dd8a80-1a24-4cc5-a00a-3f42615181d7.filesusr.com/ugd/ad98ab_846561db683443d3af0664c558849464.pdf
***** [[id:2e475e89-40a2-4359-903d-6e590f4ab8a0][cfDNA from nonterminal mouse blood draws]]
***** General reference
- https://www.sitcancer.org/research/jitc-2/jitc/collections/liquid-biopsies
- cite:keller2020
- https://www.nature.com/collections/gcdaehehhf/webinars
- https://www.nature.com/articles/s41551-021-00837-3
- https://www.mdpi.com/2072-6694/13/14/3460/htm
- https://www.nature.com/articles/s41416-021-01696-0
- https://royalsocietypublishing.org/doi/10.1098/rsob.200119
- cfDNA datasets-  https://generegulation.org/cfdna/
- see  [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGDNNMGQwVvHMjSKjBHhrkMLrS][jake's ulpwgs cfdna preprocessing]]
- cite:heberts2021
*** Machine learning in genomics and bioinformatics
- https://colab.research.google.com/github/isb-cgc/Community-Notebooks/blob/master/MachineLearning/How_to_build_an_RNAseq_logistic_regression_classifier.ipynb
- [[id:38c00ba1-299f-4a0d-959e-58845d12c712][Machine learning of whole genome sequencing data]]
- cite:chiorean2022
- https://www.science.org/doi/10.1126/scitranslmed.abk2756
- https://www.nature.com/articles/s41568-021-00408-3
- https://www.nature.com/articles/s41576-021-00434-9
- [[id:1768b182-858a-46f8-97a8-4cd9747a2361][Probability and statistics related to bioinformatics]]
*** Massively parallel sequencing
https://www.nature.com/articles/nrg3642/tables/2
**** Sequence alignment
https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/
**** [[id:728e8984-696f-4531-8e76-bf6d78bfa3b7][DNA-seq]]
**** Nucleic acid sequening processes
https://www.reddit.com/r/bioinformatics/comments/te1f5w/what_is_important_to_know_about_your_ngs_library/
***** Massively parallel sequencing
=======
cite:mitelman2007impact

cite:lieber2016mechanisms

********* Human Pathological Translocations
:PROPERTIES:
:ID:       92C7145A-9535-4E74-B4FF-88A90EE6A78F
:END:

cite:lipson2012identification

cite:bunting2013end

cite:zheng2013oncogenic

cite:greaves14_janet_rowley

cite:wheeler2013human

********* [[Clinical Fusion Protein Databases]]
:PROPERTIES:
:ID:       4AE77A12-C491-4590-9F8B-6C6EC4077242
:END:

********* Clinical RNA-seq Assays to Detect Fusion Proteins
:PROPERTIES:
:ID:       9786E312-B8BC-4F14-BAF2-C9B682703FB7
:END:

cite:beadling16_multip_amplic_approac_detec_gene

cite:qadir2014childseq


cite:scolnick2015efficient

********** Specific Cancers
:PROPERTIES:
:ID:       E44CCF9D-6D9A-4F01-B90F-929D038CF4F4
:END:

cite:alves2015next

cite:smallridge2013rna

cite:edgren2011identification

cite:pekar2015fluorescence

cite:duncavage2012targeted

********* [[FISH]]
:PROPERTIES:
:ID:       F54BB35F-3BAA-4AA9-928C-18FAB543D7C6
:END:


********* Post-Alignment Analysis Programs
:PROPERTIES:
:ID:       EC14669D-1B71-45FD-9D4D-62E405CE6917
:END:

https://rdrr.io/bioc/chimera/man/starReads.html
http://www.bioconductor.org/packages/release/bioc/vignettes/chimera/inst/doc/chimera.pdf
https://github.com/STAR-Fusion/STAR-Fusion/wiki

cite:chimera_r

******* Variant Calling
:PROPERTIES:
:ID:       E3637686-6560-4088-9784-0A601F39C34D
:END:
https://pubmed.ncbi.nlm.nih.gov/36063163/
https://pubmed.ncbi.nlm.nih.gov/33580225/
- tumor evolution
  - fishplot https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-3195-z
- incidental acquired variant detection in germline https://www.nature.com/articles/s41436-021-01138-5?proof=t+target%3D#Sec3
- clonal evolution- https://cancerdiscovery.aacrjournals.org/content/8/11/1390.long
- HRDtools for homologous recombination index
  - https://github.com/eyzhao/hrdtools
- cite:ainscough2018
- - cite:barnell2018
- [[https://software.broadinstitute.org/gatk/best-practices/bp_3step.php?case=GermShortWGS][GATK Best Practices Germline SNP and Indel]]
- [[http://www.htslib.org/workflow/][WGS/WES Mapping to Variant Calls - Version 1.0]]
- [[https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.php][Broad UnifiedGenotyper]]
- [[https://software.broadinstitute.org/gatk/documentation/article?id=3238][Broad best practices DNAseq variant]]
- cite:vcf
http://swog.org/Visitors/Fall13GpMtg/plenary/03-Corless-SWOG-Genomics-TM-ver1.pdf

******** GATK Tool Set
:PROPERTIES:
:ID:       B4100085-BD16-442F-9AB8-586241CD897A
:END:

********* GATK UnifiedGenotyper
:PROPERTIES:
:ID:       0D7356A4-E139-4BDF-BD6B-4170C15531F9
:END:

- one of several GATK variant discovery tools

********* DONE Setup GATK
CLOSED: [2016-12-05 Mon 14:22]
:PROPERTIES:
:ID:       E38AAAE9-DB89-42C8-9CE3-C18003F720D3
:END:

- resuem here https://software.broadinstitute.org/gatk/documentation/quickstart.php

- [[https://software.broadinstitute.org/gatk/documentation/quickstart.php][Getting Started]]
  - "Unlike C-compiled programs such as Samtools, GATK cannot simply be added to your PATH, so we recommend setting up an environment variable to act as a shortcut."

******* sequence variation
:PROPERTIES:
:ID:       ec437673-e697-4c0d-a65f-46aec2641ed2
:END:
- https://ai.googleblog.com/2020/09/improving-accuracy-of-genomic-analysis.html?m=1
******** variant calling
:PROPERTIES:
:ID:       743b5990-53e6-4cce-8adc-a9d766d1c5e2
:END:
********* SCNA variants
:PROPERTIES:
:ID:       179ce37c-f8cf-44b3-872b-e32677bcb49c
:END:
- SCNAs
  - https://pmbio.org/module-05-somatic/0005/04/02/Somatic_CNV_FilteringAnnotationReview/
  - https://www.annualreviews.org/doi/full/10.1146/annurev-genom-082410-101404?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed
  - cite:quigley2018
  - https://bioconductor.org/packages/release/bioc/html/DNAcopy.html
  - cite:talevich2016
  - [[file:~/repos/copy-number/]]
  - [[file:~/repos/copy-number/copy-number.org]]
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-S11-S1
  - https://www.youtube.com/watch?v=m9XPWJrMThY&t=8s
  - https://deeptools.readthedocs.io/en/latest/content/installation.html
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5737511/
- WGS callers
  - https://github.com/GavinHaLab
******* copy-number
:PROPERTIES:
:CREATED:  [2019-05-29 Wed 14:46]
:ID:       0206fb96-8785-47fb-87ca-1f597de0ee85
:END:
:LOGBOOK:
CLOCK: [2019-10-10 Thu 15:45]--[2019-10-10 Thu 15:53] =>  0:08
CLOCK: [2019-05-15 Wed 13:09]--[2019-05-15 Wed 13:24] =>  0:15
CLOCK: [2019-06-25 Tue 07:51]--[2019-06-25 Tue 08:00] =>  0:09
CLOCK: [2019-06-23 Sun 07:49]--[2019-06-23 Sun 07:53] =>  0:04
CLOCK: [2019-06-19 Wed 14:28]--[2019-06-19 Wed 14:49] =>  0:21
CLOCK: [2019-06-19 Wed 12:38]--[2019-06-19 Wed 13:33] =>  0:55
CLOCK: [2019-06-12 Wed 14:40]--[2019-06-12 Wed 14:58] =>  0:18
CLOCK: [2019-05-29 Wed 15:23]--[2019-05-29 Wed 15:34] =>  0:11
CLOCK: [2019-05-29 Wed 14:46]--[2019-05-29 Wed 14:47] =>  0:01
:END:
- https://link.springer.com/protocol/10.1007/978-1-4939-9004-7_8
- https://bioinformaticsdotca.github.io/bicg_2017_module4_lab
- cite:hehir-kwa2015
- https://www.nature.com/scitable/topicpage/copy-number-variation-and-human-disease-741737/
- https://www.frontiersin.org/articles/10.3389/fonc.2020.00321/full
- [[id:8c535c38-9310-4fd0-a8fe-c300639493da][ichorcna]]
- copy number classifiers
  - https://www.sciencedirect.com/science/article/pii/S0016508521002778?via%3Dihub
- https://www.sciencedirect.com/science/article/pii/S1535610818301119?via%3Dihub
- allele-specific copy number: the number of copies of each inherited allele, allows for the detection of copy-neutral loss of heterozygosity https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5627665/
- [[https://www.bioconductor.org/packages/devel/bioc/vignettes/TitanCNA/inst/doc/TitanCNA.pdf][see ploidy purity correction math]]
- [[https://www.biostars.org/p/383522/][good CN,purity,ploid discussion]]
******* [[id:5738AF1F-8621-40C3-AF80-0C396CDC889B][LINK TO Clinical NGS]]
:PROPERTIES:
:ID:       413414a5-651d-4d43-82c8-a6bb13de8ca3
:END:
******* DNA break mapping
>>>>>>> bb11343ab3ff683a4ece1c261661becfbc0b872d
:PROPERTIES:
:ID:       8a2ff484-d234-48fa-88c9-c4994f2e0d14
:END:
<<<<<<< HEAD
https://teichlab.github.io/scg_lib_structs/methods_html/Illumina.html
http://enseqlopedia.com/2018/01/illumina-index-sequencing-sample/
https://www.illumina.com/content/dam/illumina-marketing/documents/products/illumina_sequencing_introduction.pdf
http://tucf.com/htseq_GenomicSeq_DataSheet.pdf
****** [[id:7ec3083a-b152-4ab2-b4f8-5fb9bb5ba068][Determine sequencing coverage]]
****** General reference
=======
https://febs.onlinelibrary.wiley.com/doi/10.1111/febs.15568
https://www.nature.com/articles/s41467-019-10332-8#Sec13

****** Analysis of cell-free nucleic acids
>>>>>>> bb11343ab3ff683a4ece1c261661becfbc0b872d
:PROPERTIES:
:ID:       906150c9-910e-464c-9d7f-d8608abf6c07
:END:
******* Cell-free DNA (cfDNA) and circulating-tumor DNA (ctDNA)
:PROPERTIES:
:ID:       0f1f4aa4-c755-42ff-8f81-8a32c14293a0
:END:
- https://www.nature.com/collections/haffgaicaf
- https://www.sciencedirect.com/science/article/pii/S0890850819303883#fig1
- https://www.nature.com/articles/s41467-022-32598-1#Sec11
- https://www.nature.com/articles/s41416-020-01047-5
- https://www.nature.com/articles/s41551-022-00855-9

******** SEQC2
https://www.nature.com/articles/s41587-021-00857-z
https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02529-2#Sec15
******** Hybrid capture
https://mail.google.com/mail/u/0/#inbox/FMfcgzGrblWQxQHrNZrMjzpGmzjqDNGX
******** Landau Group
:PROPERTIES:
:ID:       b4b2a35f-e59a-4e84-995e-ca16d7fd190d
:END:
cite:widman2022

******** [[id:21d876e8-8646-4773-91c1-c4385c3218c7][cfDNA copy number calling]]
******** [[id:47c4ed56-1dea-48dc-bb25-ddbb0a4cd359][cfDNA tissue of origin discovery]]
******** [[id:c4dde8d6-1a02-42c7-a6b6-786b757fcad1][cfDNA nucleosome profiling]]
******** [[*cfDNA methylation][cfDNA methylation]]
******** [[*cfDNA fragmentomics][cfDNA fragmentomics]]
******** [[*cfDNA small variant calling][cfDNA small variant calling]]
******** Tumor-informed methods
- phylogenic for cancer evolution cite:abbosh2017
******** WGS cfDNA methods
- https://www.biorxiv.org/content/10.1101/2022.01.17.476508v1.full.pdf
- cite:zviran2020
******** cfDNA and ctDNA-specific visualizations
  - cite:moding2020
  - cite:everett2020
  - - https://d0dd8a80-1a24-4cc5-a00a-3f42615181d7.filesusr.com/ugd/ad98ab_846561db683443d3af0664c558849464.pdf

******** cfDNA from nonterminal mouse blood draws
  - https://www.frontiersin.org/articles/10.3389/fgene.2018.00006/full#T3
  - https://www.nature.com/articles/cr2017121.pdf?origin=ppub
  - https://clincancerres.aacrjournals.org/content/23/20/6305?ijkey=46368653e37c96739922e66f2f6ae51379b550f3&keytype2=tf_ipsecsha
******** General reference
- https://www.nature.com/collections/gcdaehehhf/webinars
- https://www.nature.com/articles/s41551-021-00837-3
- https://www.mdpi.com/2072-6694/13/14/3460/htm
- https://www.nature.com/articles/s41416-021-01696-0
- https://royalsocietypublishing.org/doi/10.1098/rsob.200119
- cfDNA datasets-  https://generegulation.org/cfdna/
- see  [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGDNNMGQwVvHMjSKjBHhrkMLrS][jake's ulpwgs cfdna preprocessing]]
- cite:heberts2021
******* cfRNA
:PROPERTIES:
:ID:       f4d64349-0202-41e5-9143-39de3dd63d90
:END:
- https://directorsblog.nih.gov/2023/09/12/new-approach-to-liquid-biopsy-relies-on-repetitive-rna-in-the-bloodstream/
- https://www.nature.com/articles/s41551-023-01081-7
- https://oaepublishstorage.blob.core.windows.net/b918d521-af58-4cf0-bc57-db03c5c43f33/4446.pdf
- https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFgqmWTzSTxFbqlqjBLnzFsWT
- cite:rozowsky2019
- https://www.nature.com/articles/s41586-022-04410-z
- https://www.nature.com/articles/d41586-021-03801-y
- [[cite:&larson2021]]
- [[cite:&griffith2016a]]
- https://mail.google.com/mail/u/0/#inbox/FMfcgzGlksDcrLMrdlPVPsVnzpdCNTsx
 [[#larson2021][larson2021]]
https://www.frontiersin.org/articles/10.3389/fgene.2021.742496/full
****** Allele-specific expression
:PROPERTIES:
:ID:       a52b4ca3-c3ca-4aec-b68a-01c1de39fc8a
:END:

- https://github.com/secastel/allelecounter
- https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0762-6#Sec11
****** Alternative splicing
:PROPERTIES:
:ID:       eba855e5-ce0e-4318-894d-4942f7e2fe45
:END:
- w/ RNA-seq via splAdder
  - https://github.com/ratschlab/spladder
  - https://academic.oup.com/bioinformatics/article/32/12/1840/1744052?login=true
****** Setup common reference files
:PROPERTIES:
:ID:       813ed980-1c55-44d7-84d8-67afdad2e5bc
:END:
  #+name: setup_common_references
  #+begin_src bash
#
# Setup
/bin/bash
alias wget_common=wget --noclobber
refdata=
#
# Download reference genomes
## Human
### UCSC Hg38
wget_common https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz -O $refdata/hg38.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.toplevel.fa.gz
#### Primary assembly
wget --no-clobber ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
##
## Rattus norvegicus
### UCSC
wget_common ftp://hgdownload.cse.ucsc.edu/goldenPath/rn6/bigZips/rn6.fa.gz -O $refdata/rn6.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz -O $refdata/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz
#### NOTE: No Rattus primary assembly
#### cDNA
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/cdna/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
gunzip /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
#### GTF
wget_common ftp://ftp.ensembl.org/pub/release-102/gtf/rattus_norvegicus/Rattus_norvegicus.Rnor_6.0.102.gtf.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.102.gtf.gz
#
# Unzip
for file in $refdata/*.fa.gz $refdata/*.gtf.gz;
do
    gunzip_common $file
done
#

#########1#########2#########3#########4#########5#########6#########7#########8
# ideas
#in/out from csv
# get readmes
#########1#########2#########3#########4#########5#########6#########7#########8
#+end_src
***** [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][Nucleic acid sequence alignment]]
***** Nucleic acid sequening processes
https://www.reddit.com/r/bioinformatics/comments/te1f5w/what_is_important_to_know_about_your_ngs_library/
****** [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][Nucleic acid sequence alignment]]
****** Massively parallel sequencing
:PROPERTIES:
:ID:       3BC41E6A-00D3-4498-A48C-463F68CFD76B
:END:
:LOGBOOK:
CLOCK: [2022-06-03 Fri 08:33]--[2022-06-03 Fri 08:40] =>  0:07
CLOCK: [2021-05-27 Thu 11:25]--[2021-05-27 Thu 11:38] =>  0:13
:END:
https://www.illumina.com/content/dam/illumina-marketing/documents/products/illumina_sequencing_introduction.pdf
http://tucf.com/htseq_GenomicSeq_DataSheet.pdf
******* [[id:7ec3083a-b152-4ab2-b4f8-5fb9bb5ba068][Determine sequencing coverage]]
******* General reference
:PROPERTIES:
:ID:       8438f135-bfc1-4d5e-8a02-f594a1d0773a
:END:
- https://www.youtube.com/user/IlluminaInc/playlists
- https://docs.nvidia.com/clara/parabricks/v3.0/index.html#
- https://support.illumina.com/training.html
- cite:goodwin2016
- cite:lee2016
******* High Throughput Sequencing Platforms
:PROPERTIES:
:ID:       DF6FFA19-8867-49AC-947D-FD57B403761A
:END:
- https://www.broadinstitute.org/genome-sequencing/broadillumina-genome-analyzer-boot-camp
- commercial sequencing entities
  - medgeneome
    - https://research.medgenome.com/qc-reports/
    - https://research.medgenome.com/ngs-services/whole-genome-exome-sequencing/
- https://www.illumina.com/systems/sequencing-platforms/comparison-tool.html
- https://dnatech.genomecenter.ucdavis.edu/
- [[https://www.youtube.com/watch?v=v1DbcJD4Ry0][youtube Mardis NGS]]
- [[http://www.broadinstitute.org/scientific-community/science/platforms/genome-sequencing/broadillumina-genome-analyzer-boot-camp][Broad / Illumina Genome Analyzer Boot Camp]]
- [[file:~/Box%20Sync/library/Mardis,08%20Next-Generation%20DNA%20%20Sequencing%20Methods.pdf][Mardis,08 Next-Generation DNA  Sequencing Methods]]
- 150924.1600Mardis talk
  - Illumina read length limit is signal to noise
  - Ion torrent difficult for indels with runs of single nucleotide
  - Paired end allows placement of repetitive sequences where one side might be repetitive, the other not
  - Evaluate coverage by comparison to snp array
  - Clinical sequencing false positive if variants have patterns
- Sequencing platform overview and limitations
  - cite:payton2016
- cite:glenn2011
******** Ion Torrent
:PROPERTIES:
:ID:       EC631BCD-1A9B-4B8B-877E-7861C27D948B
:END:

******** Illumina
:PROPERTIES:
:ID:       D1A3EF94-7933-4854-99ED-C63DE8F7741D
:END:
- https://www.reddit.com/r/bioinformatics/comments/uxm1s7/novaseq_2dye_v_hiseq_4dye_sequencing_differences/
- adapters
  - only 3' adapters need trim https://support.illumina.com/bulletins/2016/04/adapter-trimming-why-are-adapter-sequences-trimmed-from-only-the--ends-of-reads.html
  - indexing goes on outside of adapter seq- so routine trim will find and clip
  - https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html
- cite:illumina2015nextseq
- table
    | series  | platform |
    |---------+----------|
    | NovaSeq |     6000 |
    | HiSeq   |     3000 |
    | HiSeq   |     2500 |
    | MiSeq   |          |
    | Nexseq  |          |
    - https://www.illumina.com/systems/sequencing-platforms/comparison-tool.html
https://www.illumina.com/systems/sequencing-platforms/novaseq/specifications.html
output- *.bcl files- base call and quality score per cycle
convert *.bcl to .fastq.gz w/ CASAVA
also demultiplexes
- [[https://www.illumina.com/systems.html][system specs]]
- PhiX phasing and cluster ctrl https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/hiseq-phix-control-v3-technical-note.pdf
- library preparation kits
  - [[https://www.illumina.com/library-prep-array-kit-selector.html][selector]]
  - https://support.illumina.com/sequencing/sequencing_kits/truseq-small-rna-kit.html
********* Phix
https://support.illumina.com/bulletins/2017/02/what-is-the-phix-control-v3-library-and-what-is-its-function-in-.html
https://support.illumina.com/bulletins/2017/02/how-much-phix-spike-in-is-recommended-when-sequencing-low-divers.html
https://support.illumina.com/bulletins/2016/07/what-is-nucleotide-diversity-and-why-is-it-important.html
********* Illumina TruSeq
:PROPERTIES:
:ID:       3CE663C4-BF37-4E2E-9A0D-0E822241D05E
:END:
- https://www.google.com/search?q=illumina+truseq+indexes
- See Kukurba,15
- Steps
  - RNA isolation
  - Poly-A purification
  - Fragmentation
  - cDNA synthesis
  - Adaptor ligation
  - Size selection
  - PCR amplification

********** Removal of rRNA and tRNA
:PROPERTIES:
:ID:       D509473A-536B-46D2-8639-22DC19DF41E2
:END:
******* [[id:f44b9dce-d7fa-4b6a-9949-49ff3c02547c][Sequencing testing, modeling, and read simulation]]
******* Adapter and probe design and processing
:PROPERTIES:
:ID:       8f0a09c6-2a68-425c-91a9-4d09866c8ae2
:END:
- https://en.wikipedia.org/wiki/Edit_distance
https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4428-5#Sec5
- https://www.nature.com/articles/s41587-018-0006-x
- https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/user-guide-manual/xgen-prism-dna-library-prep-kit-processing-sequencing-data-with-umis.pdf
- https://github.com/pughlab/ConsensusCruncher
- https://rochesequencingstore.com/wp-content/uploads/2017/10/KAPA-DI-Adapter-Kit_KR1318-%E2%80%93-v3.19.pdf
******* Library preparation
:PROPERTIES:
:ID:       cae4df04-eb04-4148-88a3-798be945a9fd
:END:
- https://www.future-science.com/doi/10.2144/000114133
- [[http://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-2-r18][amplification bias]]
******* Sequencing errors
:PROPERTIES:
:ID:       008673aa-98af-4ffb-89d9-cde43ededa83
:END:
- seq error correction
  - as an information model, propogation of errors, random vs. nonrandom
  - random start-stop- allows ID of PCR duplicates- by start/stop regardless of internal seq
  - ligation umi
    - umi strategies
    - umi actaully adds back 'false' pcr duplicates which can now be positively IDed as unique molecules
  - duplexing and asymmetric strand artifacts
    - distinguishing strands
- 1st exon GC-rich, poorly seq by illumina
- https://www.nature.com/articles/s41598-018-29325-6
******* Raw sequence QC and processing
:PROPERTIES:
:ID:       1EE731A3-83DD-46AF-B988-2311307C335F
:CREATED:  [2020-10-14 Wed 08:47]
:END:
:LOGBOOK:
CLOCK: [2020-10-13 Tue 14:00]--[2020-10-13 Tue 14:47] =>  0:47
CLOCK: [2016-04-05 Tue 17:21]--[2016-04-05 Tue 17:32] =>  0:11
CLOCK: [2016-04-05 Tue 17:06]--[2016-04-05 Tue 17:15] =>  0:09
CLOCK: [2016-04-05 Tue 10:44]--[2016-04-05 Tue 11:30] =>  0:46
CLOCK: [2019-07-23 Tue 20:35]--[2019-07-23 Tue 21:13] =>  0:38
CLOCK: [2019-07-23 Tue 19:27]--[2019-07-23 Tue 19:40] =>  0:13
CLOCK: [2019-07-16 Tue 14:46]--[2019-07-16 Tue 15:03] =>  0:17
CLOCK: [2019-07-07 Sun 11:18]--[2019-07-07 Sun 11:34] =>  0:16
CLOCK: [2019-07-06 Sat 08:11]--[2019-07-06 Sat 09:07] =>  0:56
CLOCK: [2019-07-05 Fri 21:00]--[2019-07-05 Fri 22:04] =>  1:04
CLOCK: [2019-06-05 Wed 09:07]--[2019-06-05 Wed 09:47] =>  0:40
CLOCK: [2019-06-03 Mon 18:40]--[2019-06-03 Mon 20:30] =>  1:22
CLOCK: [2019-06-04 Tue 12:39]--[2019-06-04 Tue 13:12] =>  0:33
CLOCK: [2020-10-13 Tue 13:38]--[2020-10-13 Tue 14:00] =>  0:22
:END:
- https://www.google.com/search?q=riboerase+adapter+trimming
- https://genomics.sschmeier.com/ngs-qc/index.html?highlight=phix#phix-genome
- 161. Lo,C.-C. and Chain,P.S.G. (2014) Rapid evaluation and quality
control of next generation sequencing data with FaQCs. BMC
Bioinformatics, 15, doi:10.1186/s12859-014-0366-2.
- . Chen,C., Khaleel,S.S., Huang,H. and Wu,C.H. (2014) Software for
pre-processing Illumina next-generation sequencing short read
sequences. Source Code Biol. Med., 9, doi:10.1186/1751-0473-9-8.
- 165. Del Fabbro,C., Scalabrin,S., Morgante,M. and Giorgi,F.M. (2013)
An extensive evaluation of read trimming effects on Illumina NGS
data analysis. PLoS One, 8, e85024.
- raw read trimming and filtering
  - https://www.biostars.org/p/368420/
  - adapter
    - https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary
    - https://www.biostars.org/p/267388/
    - http://seqanswers.com/forums/showthread.php?t=51226
    - https://github.com/griffithlab/rnaseq_tutorial/wiki/Adapter-Trim
    - do all aligners need adaptor trimming? at least common to do so for all
    - https://www.biostars.org/p/368420/
  - trimming- removing bad ends
    - adaptors
    - low quality bases
  - filering- removing bad reads
    - low quality reads
    - contamination
    - low complexity reads
    - short reads
  - https://www.biostars.org/p/212136/
  - https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary
- https://sequencing.qcfail.com/software/fastqc/
- RNA
  - Do not
    - quality-trim RNA- cite:williams2016a
      - see also - http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0085024
        - w/ commentary here- http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/
      - - https://www.michaelchimenti.com/2016/06/trim-rna-seq-reads/
    - remove duplicate- no necessarly PCR artifact
      - https://www.biostars.org/p/55648/
      - https://www.biostars.org/p/14283/
  - https://www.slideshare.net/joachimjacob/2rna-seqpart2investigatin-rawdata
  - https://sequencing.qcfail.com/applications/mrna-seq/
- [[id:c0c0ee28-2e41-41a7-9a3b-ae195117a93e][LINK TO common sequence file manipulation]]
- QC outputs explained- https://sequencing.qcfail.com/
- see - cite:chen2013 for GC bias and effects on assembly
- [[http://bioinformatics-core-shared-training.github.io/cruk-bioinf-sschool/Day1/fastqc_sweave.pdf][de Santiago,15]]
- [[https://www.danielecook.com/double-checking-fastqs/][nice fastq python]]
- https://wikis.utexas.edu/display/bioiteam/Evaluating+your+raw+sequencing+data
- [[id:1165bc2b-9373-4313-94bd-2a1d7be34fb8][LINK TO aligned sequence qc]]
- reference
  - yas
  - [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][docker]]
    - dockerfile - include current build status in comments
    - sequencing dockerfile repos
      - https://github.com/broadinstitte/gtex-pipeline.git
      - https://hub.docker.com/u/mgibio/
      - https://hub.docker.com/u/biocontainers
  - https://github.com/genome/gms
  - best practices
    - each tool install as standalone
    - Version control
    - File organization
      - https://libraries.mit.edu/data-management/
      - See [[file:~/maintenance.org::Information Maintenance][information maintenance]] Organization -> File tree structure
    - http://www.broadinstitute.org/gatk/guide/best-practices
      best practices include recording everything
      record of all commands
      record of all software versions
      see Emacs, rpubs.com, sessionInfo and knitR
    - [[http://www.nature.com/sdata/for-authors/submission-guidelines][Nature data submission guidelines]]
    - https://kieranhealy.org/resources/
    - http://kieranhealy.org/files/misc/workflow-apps.pdf
    - see Anders,13
    - no docker in docker for dev- confuses vols, etc. instead add emacs to each container and have local repos of .emacs and org
  - associated mods
    - [[id:43331BB8-4234-48FE-B510-E92F631D5D7B][Code]]
  - compendia
    - [[file:~/Box%20Sync/org/compend.org::*UNIX][UNIX]]
    - [[file:~/.emacs.d/settings.org::*Emacs][Emacs]]
    - git
  - [ ] information management- see cite:griffith2015b
  - https://github.com/cbcrg/lncrna-annotation-nf
  - set alias https://emacs.stackexchange.com/questions/34022/error-initialization-user-has-no-home-directory
  - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
  - https://hub.docker.com/r/mgibio/mark_duplicates-cwl/dockerfile
  - https://hub.docker.com/r/broadinstitute/picard/dockerfile
  - stack
    - ubuntu
    - git and github
    - docker
      - https://github.com/bioinformatics-core-shared-training/docker-4-bioinformatics
      - https://gatkforums.broadinstitute.org/firecloud/discussion/6886/guidelines-for-working-with-docker-images-and-dockerfiles-broadies-only
    - nextflow
      - https://www.youtube.com/watch?v=jsxTC8pNPUc&t=579s
      - https://github.com/cbcrg/lncrna-annotation-nf
      - https://www.nextflow.io/example1.html
      - https://www.nextflow.io/blog/2016/docker-and-nextflow.html
      - commands
        |----------+---------|
        | function | command |
        |----------+---------|
        | <35>     | <35>    |
        |          |         |
      - processes
      - documentation
- ideas
  - Consider more emacs shell work:
  - add to gatk docker
    - https://hub.docker.com/u/broadinstitute
    - https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php
  - ideas
    - link to documentation- maybe an org and/or markdown file?
    - script comment design
  - https://gatkforums.broadinstitute.org/gatk/discussion/2909/how-to-fix-a-badly-formatted-bam

******* DNA-seq
:PROPERTIES:
:ID:       728e8984-696f-4531-8e76-bf6d78bfa3b7
:END:
- https://www.nature.com/articles/s41586-021-03894-5
- index, adapter, barcode
- https://www.genscript.com/molecular-biology-glossary/11941/genomic-equivalent
- DNA
  - for immuno regulatory regions- https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-016-0220-7
  - for chromatin accessibility
    - ATAC-seq
      - cite:buenrostro2015
      - FAST ATAC- ATAC-Seq optimized for blood
  - DNA for sequence variation
    - variation
      - visualizations
        - - https://mail.google.com/mail/u/0/#inbox/FMfcgxwHNCtwvpXGcckxJfDWPTPKHTst
      - clingen
        - https://www.youtube.com/channel/UCsn4nEVUTpVQz70rClgMMsQ?view_as=subscriber
    - https://gatk.broadinstitute.org/hc/en-us/articles/360035535892-Somatic-copy-number-variant-discovery-CNVs-
    - https://gatk.broadinstitute.org/hc/en-us/articles/360035535912-Data-pre-processing-for-variant-discovery
    - https://gatk.broadinstitute.org/hc/en-us/sections/360007226651-Best-Practices-Workflows
    - https://mail.google.com/mail/u/0/#inbox/FMfcgxwHNWHJNLRlZwfGkGJHwJnBxFlh
  - for sequence homology- cite:stoeger2018
  - cfDNA
    - cite:huang2019
- marking duplicates :process:
  - consider picard markduplicates - https://gatkforums.broadinstitute.org/gatk/discussion/6945/picard-tools-markduplicates-using-cram-format-how-to-pass-a-valid-cram-reference
  - samtools
    - https://samtools.github.io/hts-specs/SAMv1.pdf
    - sort
      - https://www.biostars.org/p/247588/
    - http://www.htslib.org/algorithms/duplicate.html
  - https://qcb.ucla.edu/wp-content/uploads/sites/14/2016/03/GATKwr12-2-Marking_duplicates.pdf
******** [[id:719b2017-4ec6-44dd-8e91-a677bc6654bd][DNA-seq pre-alignment processing and QC]]
******** Paired tumor-normal sequencing
:PROPERTIES:
:ID:       0496ac54-3250-4fd3-9681-c880aaa296b2
:END:
- https://github.com/mskcc/facets
******** Tumor-only whole exome sequencing
:PROPERTIES:
:CREATED:  [2021-05-01 Sat 07:26]
:ID:       eed43327-3a0c-4f7b-9886-855a88afa02d
:END:
- https://ascopubs.org/doi/full/10.1200/CCI.19.00130
- https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)31635-8
- cite:carter2012 ABSOLUTE
  - https://software.broadinstitute.org/cancer/cga/absolute
  - https://www.biostars.org/p/175131/
******** DNA Bioinformatics
:PROPERTIES:
:ID:       1C24E3B6-CE16-47C8-8D7B-95EB530560F1
:END:

- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4669575/
- https://github.com/kevinblighe/ClinicalGradeDNAseq
- https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html

- https://software.broadinstitute.org/gatk/documentation/tooldocs/current/picard_illumina_IlluminaBasecallsToFastq.php

- ffpe
  - cite:munchel2015
  - cite:spencer2013
- Fonseca,12 Tools for mapping high throughput sequencing data(2)
- [[file:~/Box%20Sync/scientific_technique/Cruchaga%20lab%20eQTLs.pptx][Cruchaga eQTLs]]

********* Post-alignment Samtools
:PROPERTIES:
:ID:       2D153E08-9F4A-4564-ACA7-5B4932D1022B
:END:

- nice header review https://sarahpenir.github.io/bioinformatics/awk/calculating-mapping-stats-from-a-bam-file-using-samtools-and-awk/
- Post-alignment
  - samtools
    - [[http://www.htslib.org/][website]]
      - [[http://www.htslib.org/workflow/][workflows]]
      - [[http://www.htslib.org/doc/][documentation]]
        - [[http://www.htslib.org/doc/#publications][publications]]
    - [[https://github.com/samtools/][github]]
    - cite:li2009
    - samtools view sorted.bam | awk '{sum+=$5} END { print "Mean MAPQ =",sum/NR}'
    - [[https://www.biostars.org/p/15953/][biostars: calculate MAPQ]]
    - calc mapq w/ samtools- https://www.biostars.org/p/15953/][Malachi Griffith biostars
- [[http://www.htslib.org/doc/samtools-1.2.html][Documentation]]
- Subset a bam
  - By chromosome
    - e.g.: ​samtools view -b in.bam chr1 > in_chr1.bam
- [[https://www.biostars.org/p/17361/][Picard is Samtools!]]
- Find ummapped reads by flags
- [[http://davetang.org/wiki/tiki-index.php?page=SAMTools#Converting_a_SAM_file_to_a_BAM_file][Sam to Bam, indexing ]]

********* Ideas
:PROPERTIES:
:ID:       F6511EA3-A5B4-455E-99F1-D536EE81F22D
:END:

- [[http://zvfak.blogspot.com/2012/02/samtools-in-parallel.html][run in parallel]]

- incorporate samtools in R workflow via Rsamtools

********* Exome Sequencing Pipeline
:PROPERTIES:
:ID:       57992D53-D767-4E44-9E3F-B4A05E5B4EF1
:END:

********* Specialized Applications
:PROPERTIES:
:ID:       7A9FC47A-2E9E-46D5-8E3E-E7C7C5C5AD9E
:END:

********** Detecting Internal Tandem Duplications
:PROPERTIES:
:ID:       4B340D6D-E38D-42E5-AD1F-3CB0BDF62545
:END:

[[file:~/Box%20Sync/library/Spencer,12%20Detection%20of%20FLT3%20ITD%20in%20short%20read%20length%20NGS.pdf][Spencer,12 Detection of FLT3 ITD in short read length NGS]]

********* DNA Alignment
:PROPERTIES:
:ID:       B2F17B06-C1BA-487A-93EC-D9737584BC0E
:END:

- https://www.youtube.com/watch?v=4n7NPk5lwbI
- - https://www.youtube.com/watch?v=hpb-mH-yjLc&list=PL2mpR0RYFQsBiCWVJSvVAO3OJ2t7DzoHA
- tutorials
  - http://seqinformatics.com/?page_id=160
    - near perfect matches (BLAST not specific)
  - https://www.youtube.com/watch?v=zMAa9gFd2Gs
  - http://www.ngscourse.org/Course_Materials/alignment/tutorial/example.html
  - https://wikis.utexas.edu/display/bioiteam/Mapping+tutorial

********** Novoalign
:PROPERTIES:
:ID:       22097A02-1E18-40BD-BA2F-0D3BC645F0D5
:END:

- cite:Novocraft
- [[http://www.novocraft.com/support/download/][Novoalign]]

********** BWA
:PROPERTIES:
:ID:       83F91393-2108-4785-9696-BDE8EC09FA88
:END:

https://bioinformatics.stackexchange.com/questions/2955/is-there-a-more-elegant-solution-to-the-bwa-mem-paired-reads-have-different-nam

-"bwa concatenates all the references sequences together before aligning. So if a read hangs off of one sequence onto the next, it's given the appropriate mapping position, and the unmapped flag is also set, as a sign that something is off about the alignment."

- sequences should be <3% difference from reference
- A block sorting lossless data compression algorithm
- Compresses the genome as a BWT index
- Identical sequences (genome repeats) only occur once in the index, reads that match the genome in multiple locations get a mapping quality of 0
- output is SAM
- [[http://bio-bwa.sourceforge.net/][Burrows-Wheeler Aligner]]
- [[https://www.biostars.org/t/bwa%20mem/][Biostars BWA Posts]]
- [[http://bio-bwa.sourceforge.net/bwa.shtml][bwa manual]]
  - algorithms
    - backtrack illumina to 100 bp
    - 70 bp to 1 mb
      - sw
      - mem
    - mem is generally recommended
    -

********* Post-Alignment QC
:PROPERTIES:
:ID:       F689D6FA-071C-4167-BC21-85A7EC20DA41
>>>>>>> bb11343ab3ff683a4ece1c261661becfbc0b872d
:END:

filtering
noncoding, known mis-spliced, mitochondrial, x/y in mixed sex, unannotated

********** Inspection
:PROPERTIES:
:ID:       07602EB8-5EFD-4179-9F55-320190ECE670
:END:

********** Processing
:PROPERTIES:
:ID:       2A4B2B42-C088-4479-BC65-670CC8377D39
:END:

********** FASTX-Toolkit
:PROPERTIES:
:ID:       8169C9DC-FDB6-473E-9B54-E154261186EA
:END:

  - [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][documentation]]
  - brew installed on mac, but says, "cannot execute binary file", will have to run from cluster
  - [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][Command-line Usage]]
  - the clip can be a .fa file
  - [[http://mgt2.chpc.wustl.edu/wiki119/index.php/FASTX_Toolkit][on cluster]]
  - Clipper
    https://github.com/NCBI-Hackathons/RNA_mapping

********* Broad Best Practices for Variant Discovery in DNAseq
:PROPERTIES:
:ID:       850AA38C-75E9-42E2-8DEE-61F4950BCFDE
:END:

https://software.broadinstitute.org/gatk/documentation/topic?name=methods

******** aligned sequence qc
:PROPERTIES:
:ID:       1165bc2b-9373-4313-94bd-2a1d7be34fb8
:END:
- https://sarahpenir.github.io/bioinformatics/awk/calculating-mapping-stats-from-a-bam-file-using-samtools-and-awk/
- coverage
  - cite:illumina2014
  - Picard CollectWgsMetrics
********* coverage and depth
:PROPERTIES:
:ID:       a4dcf0d6-1619-4f82-989d-2bd599cdf061
:END:
- https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/hiseq-x-30x-coverage-technical-note-770-2014-042.pdf
- https://www.biostars.org/p/331957/
- https://www.htslib.org/doc/samtools-depth.1.html#OPTIONS
- for coverage https://www.biostars.org/p/104063/
******** capture probes and barcoding
:PROPERTIES:
:ID:       675eba72-76e5-4b49-81e9-37bdc02cce9b
:END:
https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFWNbSMxNkbQTTtVwBpvBltDq
https://support.illumina.com/bulletins/2018/08/understanding-unique-dual-indexes--udi--and-associated-library-p.html
https://www.illumina.com/content/dam/illumina-marketing/documents/products/whitepapers/index-hopping-white-paper-770-2017-004.pdf
https://www.google.com/search?q=calculation+probability+of+sequence+mismatch&oq=calculation+probability+of+sequence+mismatch&aqs=chrome..69i57j33.12480j0j7&client=ubuntu&sourceid=chrome&ie=UTF-8
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245947/
- capture probe design
  - https://ascopubs.org/doi/full/10.1200/CCI.19.00077?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed
   Contigs: group of sequences representing overlapping regions from a genome or transcriptome.
- [[https://wikis.utexas.edu/display/GSAF/Illumina+-+all+flavors][illumina adaptors explained]]
- adaptor strategies
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/case-study/idt_improved-coverage-uniformity-with-xgen-lockdown-panels-case-study.pdf?sfvrsn=c33d1907_6
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/protocol/idt_xgen-prism-dna-library-prep-kit-quick-reference-guide.pdf?sfvrsn=6fe71207_4
  - https://sfvideo.blob.core.windows.net/sitefinity/docs/default-source/user-guide-manual/xgen-prism-dna-library-prep-kit-processing-sequencing-data-with-umis.pdf?sfvrsn=22991407_10
  - https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4428-5#MOESM1
- UMI
  - - [[http://www.genomics.agilent.com/article.jsp?pageId=3081][HaloPlex]]
  - - https://cgatoxford.wordpress.com/2015/08/14/unique-molecular-identifiers-the-problem-the-solution-and-the-proof/
- barcoding
  - 10x
- https://pdfs.semanticscholar.org/310b/3bac42989485c98406848217418ff22c22e7.pdf
- barcodes and adaptors for increased sensitivity
  - cite:kennedy2014
  - cite:newman2016
  - cite:wang2019
  - https://www.youtube.com/watch?v=HIVCQ7iaowk
  - newman2016
    - adaptor informatics
      - consensus building
        - set "family size"
        -
    - error base signature
      - C>A C>T
  - wang2019
    - IDT xGen
      - blocks adaptor to limit off-target
      - Cot-1 blocks repetitve regions
    - method explanation followed by experiments to validate
      - present experiments as condition vs. condition

- see edit distance in [[id:A91BF233-F14E-4E91-A1E9-B39802A9ED2D][Computers, Computer Science, and Information Technology]]
******* [[id:61aac189-1770-4d80-a367-e7c39b46e0d1][Adaptive TCR-beta immunosequencing]]
******* [[id:ccab4617-0399-4ba3-828e-b5fbbb972ba1][Whole-genome sequencing]]



****** Code associated with sequence reference builds
:PROPERTIES:
:ID:       767e5e30-8490-4cda-8018-8b0bd91c4ef3
:END:
- need wget for github-             'https://raw.githubusercontent.com/Boyle-Lab/Blacklist/blob/master/lists/mm10-blacklist.v2.bed.gz')
#+begin_src bash
ref_dir=/mnt/local/users/jeszyman/data/ref/raw_downloads

assemblies=('http://ftp.ensembl.org/pub/release-104/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz'
            'http://ftp.ensembl.org/pub/release-104/fasta/rattus_norvegicus/cdna/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz'
            'https://ftp.ncbi.nlm.nih.gov/genomes/genbank/vertebrate_mammalian/Mus_musculus/latest_assembly_versions/GCA_000001635.9_GRCm39/GCA_000001635.9_GRCm39_genomic.fna.gz'
            'https://ftp.ncbi.nlm.nih.gov/genomes/genbank/vertebrate_mammalian/Mus_musculus/latest_assembly_versions/GCA_000001635.9_GRCm39/GCA_000001635.9_GRCm39_genomic.gtf.gz'
            'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/635/GCA_000001635.8_GRCm38.p6/GCA_000001635.8_GRCm38.p6_genomic.fna.gz'
            'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/635/GCA_000001635.8_GRCm38.p6/GCA_000001635.8_GRCm38.p6_genomic.gtf.gz'
            'http://hgdownload.cse.ucsc.edu/goldenpath/mm10/bigZips/latest/mm10.fa.gz'
            'https://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/genes/mm10.knownGene.gtf.gz')

for i in "${assemblies[@]}";
do
    wget \
        --continue \
        --directory-prefix $ref_dir \
        --execute robots=off \
        --no-check-certificate \
        --no-parent \
        --timestamping $i
    file=$(echo $i | sed "s/^.*\///g")
    base=$(basename -s .gz "$file")
done


    # if [ "${file: -3}" == ".gz" ] && [ $ref_dir/$base -ot $ref_dir/$file ];
    # then
    #     gunzip --to-stdout $ref_dir/$file > $ref_dir/$base
    # fi
#+end_src

#+begin_src bash
bwa index \
    --algoType bwtsw \
    --prefix $prefix \
    $fasta
#+end_src
- pull down common reference sequences
  #+name: reference_pull
  #+begin_src bash
#########1#########2#########3#########4#########5#########6#########7#########8
#
wget --no-clobber https://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.ensGene.gtf.gz -O /mnt/users/jeszyman/data/rn6.ensGene.gtf.gz
#
gunzip -c /mnt/users/jeszyman/data/rn6.ensGene.gtf.gz > /mnt/users/jeszyman/data/rn6.ensGene.gtf
#

cd /mnt/users/jeszyman/data
rsem-prepare-reference --gtf ./rn6.ensGene.gtf \
#                       --bowtie \
                       ./rn6.fa \
                       rsem_ref

#+end_src
****** Sequencing target data types and methods
******* Motifs
https://academic.oup.com/genetics/article/216/2/353/6066178
******* Sequence similarity and patterns
:PROPERTIES:
:ID:       508ac0df-9d93-4455-a451-cb876f83498d
:END:
- https://www.nature.com/articles/nature12477

<<<<<<< HEAD
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  # ichorCNA
  ##
  ## linux dependencies
  RUN apt-get update \
     && apt-get install -y \
     libcurl4-openssl-dev \
     libssl-dev \
     libxml2-dev
  #RUN rm /usr/lib/x86_64-linux-gnu/libcurl.so.4
  #RUN ln -s /usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0 /usr/lib/x86_64-linux-gnu/libcurl.so.4
  ##
  ## R dependencies
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("HMMcopy"); BiocManager::install("GenomeInfoDb"); BiocManager::install("GenomicRanges");'
  ##
  ## git clone install
  RUN cd /opt \
      && git clone https://github.com/broadinstitute/ichorCNA.git \
      && cd ichorCNA \
      && R CMD INSTALL . \
      && cd /opt
  ##
  #RUN echo "source activate py2" > ~/.bashrc
  #ENV PATH /opt/conda/envs/py2/bin:$PATH
****** [[file:docker/rna_Dockerfile][RNA docker]]
=======
******* Sequence variation and variant calling
>>>>>>> bb11343ab3ff683a4ece1c261661becfbc0b872d
:PROPERTIES:
:ID:       57E81A44-3233-479B-8264-2CD5F1904A60
:END:
:LOGBOOK:
CLOCK: [2020-07-07 Tue 12:17]--[2020-07-07 Tue 12:52] =>  0:35
CLOCK: [2019-04-26 Fri 11:28]--[2019-04-26 Fri 11:44] =>  0:16
CLOCK: [2019-04-26 Fri 11:03]--[2019-04-26 Fri 11:13] =>  0:10
CLOCK: [2019-04-26 Fri 10:43]--[2019-04-26 Fri 10:57] =>  0:14
:END:
<<<<<<< HEAD
******* Preamble
#+begin_src bash
FROM jeszyman/biotools:1.0.2
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build an RNA-focused docker image                  ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/rna
# (https://hub.docker.com/repository/docker/jeszyman/rna)

#+end_src
******* Conda
#+begin_src bash
RUN conda install -c bioconda kallisto
RUN conda install -c bioconda rsem
RUN conda install -c bioconda star
RUN conda install -c bioconda subread
RUN conda install -c bioconda salmon
#+end_src
******* R and Bioconductor
#+begin_src bash
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("DESeq2", \
                                 "edgeR", \
                                 "fgsea", \
                                 "limma", \
                                 "tximport"));'

#+end_src
******* Custom installs
#+begin_src bash
RUN cd /opt && git clone https://github.com/rajewsky-lab/mirdeep2 && cd /opt/mirdeep2 && perl install.pl

RUN /bin/bash

RUN cd /opt/mirdeep2 && perl install.pl

ENV PATH="/opt/mirdeep2/bin:${PATH}"
#+end_src
****** [[file:docker/atac_Dockerfile][ATAC-seq docker]]
#+begin_src bash :tangle ./docker/atac_Dockerfile
FROM jeszyman/biotools:1.0.2
#########1#########2#########3#########4#########5#########6#########7#########8

###################################
###   Dockerfile for ATAC-seq   ###
###################################

# After build, the image will be pushed to dockerhub as jeszyman/atac
# (https://hub.docker.com/repository/docker/jeszyman/atac)

RUN apt-get -y update && apt-get upgrade -y

# Simple bioconda installs
RUN conda install -c bioconda macs2

# Simple R installs
##
## Via Bioconductor
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("ATACseqQC", \
                                 "ChIPpeakAnno", \
                                 "ChIPseeker", \
                                 "clusterProfiler", \
                                 "csaw"));'


# ATACseqQC
##
## Linux prerequisites
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated --fix-missing \
    libgsl-dev \
    libcurl4-openssl-dev
    #libssl-dev
##
## ATACseqQC
RUN R -e 'install.packages("BiocManager"); \
    BiocManager::install(); \
    BiocManager::install(c("ATACseqQC"));'

# Homer
RUN mkdir /opt/homer
RUN cd /opt/homer
RUN wget http://homer.ucsd.edu/homer/configureHomer.pl -O /opt/homer/configureHomer.pl
RUN perl /opt/homer/configureHomer.pl -install

ENV PATH="${PATH}:/opt/homer/bin/"

RUN perl /opt/homer/.//configureHomer.pl -install mouse-p

RUN conda install -c bioconda samtools=1.7

# R CRAN Installs
RUN R -e "install.packages(c('statmod', repos='http://cran.rstudio.com/'))"

#+end_src
*** Metabolomics
:PROPERTIES:
:CREATED:  [2020-10-08 Thu 09:15]
:ID:       fce92cb5-cf68-4944-b2f4-eff55996ec5f
:END:
https://www.sciencedirect.com/science/article/pii/S1097276515003032
https://www.medrxiv.org/content/10.1101/2021.02.05.21251173v1
- http://www.visantnet.org/visantnet.html
**** Metabolomics
:PROPERTIES:
:CREATED:  [2020-10-08 Thu 09:15]
:ID:       fce92cb5-cf68-4944-b2f4-eff55996ec5f
:END:
https://www.sciencedirect.com/science/article/pii/S1097276515003032
https://www.medrxiv.org/content/10.1101/2021.02.05.21251173v1
- http://www.visantnet.org/visantnet.html
*** Bioinformatics visualizations
https://www.reddit.com/r/bioinformatics/comments/12ddk5r/what_genomics_plot_types_are_appealing_to_wet/
- [[id:c0359d9c-246a-41fb-a9b4-05ea5cb546ab][Biomarker visualizations]]
*** Probability and statistics related to bioinformatics
:PROPERTIES:
:ID:       1768b182-858a-46f8-97a8-4cd9747a2361
:END:
- https://www.reddit.com/r/bioinformatics/comments/teeywx/advicebooks_for_getting_started_on_mathematical/
[[id:ee66da5b-aa64-4535-af72-5d98976871ff][clinical prediction models]]
- https://www.nature.com/articles/s41576-019-0130-6
- https://www.pnas.org/content/107/21/9546
- cite:tarazona2020
- https://cran.r-project.org/web/packages/NMF/vignettes/NMF-vignette.pdf
- Pavlou M, Ambler G, Seaman SR, Guttmann O, Elliott P, King M, et al. How to develop a more accurate risk prediction model when there are few events. BMJ 2015;351:h3868.
- https://www.liebertpub.com/doi/abs/10.1089/cmb.2004.11.714

**** Clustering and cluster analysis
:PROPERTIES:
:ID:       5372195f-b5fd-41da-8020-a87533c6e74a
:END:
https://stackoverflow.com/questions/42674102/extracting-members-of-cluster-pheatmap
https://www.datanovia.com/en/lessons/determining-the-optimal-number-of-clusters-3-must-know-methods/

https://www.reddit.com/r/statistics/comments/t3ok40/q_is_there_any_difference_between_latent_profile/
- https://www.reddit.com/r/statistics/comments/t6gudh/q_how_to_evaluate_clustering_in_dimensionality/
- http://statweb.stanford.edu/~susan/courses/stats305c/examplesClusters.html
- https://bookdown.org/rdpeng/exdata/k-means-clustering.html
- https://www.datanovia.com/en/lessons/cluster-analysis-example-quick-start-r-code/
- https://academic.oup.com/biostatistics/article/19/1/71/3852318
**** [[*Machine learning in genomics and bioinformatics][LINK TO Machine learning in genomics and bioinformatics]]
*** Proteomics
:PROPERTIES:
:ID:       e01aba61-2ea4-4235-a237-b0b90b49f153
:END:
- liquid biopsy somascan https://www.cell.com/cell/fulltext/S0092-8674(23)01033-4#secsectitle0090

https://www.sciencedirect.com/science/article/pii/S1535610823002192#sec1.3
https://bioone.org/journals/radiation-research/volume-198/issue-6/RADE-22-00074.1/Prediction-of-Total-Body-and-Partial-Body-Exposures-to-Radiation/10.1667/RADE-22-00074.1.short
https://www.kavrakilab.org/nci-1u01ca258512-01.html
https://www.science.org/doi/10.1126/science.abj1541
https://www.science.org/doi/10.1126/science.abj1541
https://www.degruyter.com/document/doi/10.1515/hsz-2017-0104/html
https://www.annualreviews.org/doi/10.1146/annurev-anchem-071015-041535
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201800247
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/epdf/10.1002/pmic.200800417
https://pubs.acs.org/doi/10.1021/pr400132j
https://www.embopress.org/doi/full/10.15252/msb.20156297
https://pubs.acs.org/doi/10.1021/acs.jproteome.5b0090
1https://www.sciencedirect.com/science/article/pii/S2212968513000044?via%3Dihub#fig0005
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201300377
https://www.mdpi.com/2227-7382/1/2/109/htm
https://www.olink.com/data-you-can-trust/technology/
Mass Spectrometry

 MASCOT peptide mass fingerprint
 SILAC (stable isotope labeling by amino acids in cell culture)

 LC/MS- 1995- 166. Appella, E., Padlan, E. A. & Hunt, D. F. Analysis of the
 structure of naturally processed peptides bound by
 class I and class II major histocompatibility complex
 molecules. Exs 73, 105–119 (1995).

 iTraq
 By Technique
 MALDI-TOF
 MS/MS
 By Experimental Design


- [[id:179b243b-e6c0-484c-81ef-e8bff57e7f46][Circulating biomarkers]]
- proteome dynamic range is ~10^12, machine limit is usually ~10^3
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096388/
- https://link.springer.com/article/10.1007%2Fs00216-012-6203-4#Sec14
- https://www.nature.com/articles/nprot.2017.040
- label-free
  - bioinformatics
    - MaxQuant https://www.maxquant.org/
    - Perseus
      - https://link.springer.com/protocol/10.1007/978-1-4939-7493-1_7
      - http://coxdocs.org/doku.php?id=perseus:start
**** SOMAscan
:PROPERTIES:
:ID:       89b4ec6d-992d-4551-81da-19011597732a
:END:
:LOGBOOK:
CLOCK: [2021-08-30 Mon 13:28]--[2021-08-30 Mon 14:10] =>  0:42
:END:
- somascan examples
  - https://www.ncbi.nlm.nih.gov/pubmed/27327800
- cite:yang2020
- cite:joshi2018
- cite:candia2017
- cite:wilcox2016
- isolation for somascan
  - http://mohanlab.bme.uh.edu/wp-content/uploads/2015/09/HOU-SOMAscan-Methods-2016.pdf
- per-reviewed protocols
  - https://link.springer.com/protocol/10.1007%2F978-1-4939-9706-0_13
- [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd][email: GTAC SOMAscan notes]]
- https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd
- https://www.sciencedirect.com/science/article/pii/S1874391916304006
- For protein, SOMAscan has a higher dynamic range and lower limit of detection than comparable Luminex antibody-based assays (cite doi.org/10.1038/s41591-019-0665-2). Moreover, SOMAscan includes targets such as Jagged-1 and -2 not available for multiplexed antibody assays.
**** Proteomics
:PROPERTIES:
:ID:       e01aba61-2ea4-4235-a237-b0b90b49f153
:END:
https://ashpublications.org/blood/article/doi/10.1182/blood.2022016033/486036/Proteomic-and-Phosphoproteomic-Landscapes-of-Acute
https://www.science.org/doi/10.1126/science.abj1541
https://www.science.org/doi/10.1126/science.abj1541
https://www.degruyter.com/document/doi/10.1515/hsz-2017-0104/html
https://www.annualreviews.org/doi/10.1146/annurev-anchem-071015-041535
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201800247
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/epdf/10.1002/pmic.200800417
https://pubs.acs.org/doi/10.1021/pr400132j
https://www.embopress.org/doi/full/10.15252/msb.20156297
https://pubs.acs.org/doi/10.1021/acs.jproteome.5b0090
1https://www.sciencedirect.com/science/article/pii/S2212968513000044?via%3Dihub#fig0005
https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/pmic.201300377
https://www.mdpi.com/2227-7382/1/2/109/htm
https://www.olink.com/data-you-can-trust/technology/
Mass Spectrometry

 MASCOT peptide mass fingerprint
 SILAC (stable isotope labeling by amino acids in cell culture)

 LC/MS- 1995- 166. Appella, E., Padlan, E. A. & Hunt, D. F. Analysis of the
 structure of naturally processed peptides bound by
 class I and class II major histocompatibility complex
 molecules. Exs 73, 105–119 (1995).

 iTraq
 By Technique
 MALDI-TOF
 MS/MS
 By Experimental Design


- [[id:179b243b-e6c0-484c-81ef-e8bff57e7f46][Circulating biomarkers]]
- proteome dynamic range is ~10^12, machine limit is usually ~10^3
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096388/
- https://link.springer.com/article/10.1007%2Fs00216-012-6203-4#Sec14
- https://www.nature.com/articles/nprot.2017.040
- label-free
  - bioinformatics
    - MaxQuant https://www.maxquant.org/
    - Perseus
      - https://link.springer.com/protocol/10.1007/978-1-4939-7493-1_7
      - http://coxdocs.org/doku.php?id=perseus:start
***** SOMAscan
:PROPERTIES:
:ID:       89b4ec6d-992d-4551-81da-19011597732a
:END:
:LOGBOOK:
CLOCK: [2021-08-30 Mon 13:28]--[2021-08-30 Mon 14:10] =>  0:42
:END:
- somascan examples
  - https://www.ncbi.nlm.nih.gov/pubmed/27327800
- cite:yang2020
- cite:joshi2018
- cite:candia2017
- cite:wilcox2016
- isolation for somascan
  - http://mohanlab.bme.uh.edu/wp-content/uploads/2015/09/HOU-SOMAscan-Methods-2016.pdf
- per-reviewed protocols
  - https://link.springer.com/protocol/10.1007%2F978-1-4939-9706-0_13
- [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd][email: GTAC SOMAscan notes]]
- https://mail.google.com/mail/u/0/#inbox/FMfcgxwGCtQssVccPJsjlFklhgFHqJGd
- https://www.sciencedirect.com/science/article/pii/S1874391916304006
- For protein, SOMAscan has a higher dynamic range and lower limit of detection than comparable Luminex antibody-based assays (cite doi.org/10.1038/s41591-019-0665-2). Moreover, SOMAscan includes targets such as Jagged-1 and -2 not available for multiplexed antibody assays.
*** Systems biology
:PROPERTIES:
:ID:       228D364B-48C8-49E4-9B00-FBFD52384623
:END:
https://www.reddit.com/r/bioinformatics/comments/11ym1x9/multiomics_integration_and_analysis_tutorials/
https://www.reddit.com/r/bioinformatics/comments/11ym1x9/multiomics_integration_and_analysis_tutorials/
- http://xena.ucsc.edu/
- https://www.mdpi.com/1099-4300/22/6/627
- https://www.sciencedirect.com/science/article/pii/S2589004222015486
- cite:sanchez-vega2018
- https://www.sciencedirect.com/science/article/pii/S2666166721004755
- https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3016-x
- http://bioportal.bioontology.org/ontologies/GO/?p=classes&conceptid=root
- pathways
  - pathways
   Pathway Browsers and Visualizers
   GenMAPP
   http://cytoscapeweb.cytoscape.org/
   NCBI BioStstems (pathway diagrams)
   NCI Pathway Interaction Database
   http://www.reactome.org/ReactomeGWT/entrypoint.html
   NIH Molecular Interaction Maps
   Pathvisio
   Pathvisio pathway styles
   Wikipathways
- Geurts,09 Supervised learning with decision tree-based methods in computational.pdf
- Breker,09 Explorations in topology–delving underneath the surface of genetic.pdf
- Lima-Mendez,09 The powerful law of the power law and other myths in network biologyw.pdf
- - [[http://science.sciencemag.org/content/353/6306/aaf1420][A global genetic interaction network maps a wiring diagram of cellular function]]
- cite:chicco2017
- cite:sen2014
- cytoscape
  - https://github.com/mohd-akram/languages
  - [[http://www.nature.com/nmeth/journal/v9/n11/full/nmeth.2212.html][cytoscape plugins]]
- http://acacia.atspace.eu/Tutorial/Tutorial.html
- - http://amp.pharm.mssm.edu/Enrichr/#stats
- - https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btr656
- https://toppgene.cchmc.org/
- Pathway analysis
  - https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  - cite:Khatri_2012
- GEDI (Gene Expression Dynamics Inspector)
 Website
 Software v.2.1 Manual
 Invented Eichler,03
 Traditional hierarchical clustering and self organizing maps do not idnetify genome-wide emergent phenotypic profiles
 Uses self organizing map to reduce dimensionality into miniclusters of 1-9 genes and create mosic patterns
- Kitano H (2002) Looking beyond the details: A rise in system-oriented approaches in genetics and molecular biology. Curr Genet 41: 1–10.
- Kitano H (2002) Systems biology: A brief overview. Science 295: 1662–1664.
- Ehrenberg M, Elf J, Aurell E, Sandberg R, Tegner J (2003) Systems biology is taking off. Genome Res 13: 2377–2380.
- http://www.springer.com/us/book/9780387989921
**** [[id:91d46e60-eec3-48c7-97f1-b53eb1559b3f][Co-expression]]
:PROPERTIES:
:ID:       632d1c7d-aecc-4ad5-85ad-0f26571e454d
:END:
**** [[id:abb6fc6c-07d9-4375-a8f6-a87370f5c0ea][Multivariate biomarker discovery methods]]
:PROPERTIES:
:ID:       34a99004-090f-44c6-a721-f5f1054e5b1f
:END:
- iPANDA
  - https://www.nature.com/articles/ncomms13427#Sec10
    - https://github.com/varnivey/ipanda/blob/master/MANUAL.txt
- JCO top cited biomarker studies
      - with cfDNA copy number
        - cite:stover2018
        - cite:slaven2018
      - past 1 year
        - innocenti2019.pdf
        - loi2019.pdf
        - ott2019.pdf
      - past 5 years
        - oxnard2016.pdf
        - roemer2016.pdf
        - fribbens2016.pdf
        - rizvi2018.pdf
**** Systems biology
:PROPERTIES:
:ID:       228D364B-48C8-49E4-9B00-FBFD52384623
:END:
- cite:sanchez-vega2018
- https://www.sciencedirect.com/science/article/pii/S2666166721004755
- https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3016-x
- http://bioportal.bioontology.org/ontologies/GO/?p=classes&conceptid=root
- pathways
  - pathways
   Pathway Browsers and Visualizers
   GenMAPP
   http://cytoscapeweb.cytoscape.org/
   NCBI BioStstems (pathway diagrams)
   NCI Pathway Interaction Database
   http://www.reactome.org/ReactomeGWT/entrypoint.html
   NIH Molecular Interaction Maps
   Pathvisio
   Pathvisio pathway styles
   Wikipathways
- Geurts,09 Supervised learning with decision tree-based methods in computational.pdf
- Breker,09 Explorations in topology–delving underneath the surface of genetic.pdf
- Lima-Mendez,09 The powerful law of the power law and other myths in network biologyw.pdf
- - [[http://science.sciencemag.org/content/353/6306/aaf1420][A global genetic interaction network maps a wiring diagram of cellular function]]
- cite:chicco2017
- cite:sen2014
- cytoscape
  - https://github.com/mohd-akram/languages
  - [[http://www.nature.com/nmeth/journal/v9/n11/full/nmeth.2212.html][cytoscape plugins]]
- http://acacia.atspace.eu/Tutorial/Tutorial.html
- - http://amp.pharm.mssm.edu/Enrichr/#stats
- - https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btr656
- https://toppgene.cchmc.org/
- Pathway analysis
  - https://www.pathvisio.org/documentation/tutorials/tutorial-1/
  - cite:Khatri_2012
- GEDI (Gene Expression Dynamics Inspector)
 Website
 Software v.2.1 Manual
 Invented Eichler,03
 Traditional hierarchical clustering and self organizing maps do not idnetify genome-wide emergent phenotypic profiles
 Uses self organizing map to reduce dimensionality into miniclusters of 1-9 genes and create mosic patterns
- Kitano H (2002) Looking beyond the details: A rise in system-oriented approaches in genetics and molecular biology. Curr Genet 41: 1–10.
- Kitano H (2002) Systems biology: A brief overview. Science 295: 1662–1664.
- Ehrenberg M, Elf J, Aurell E, Sandberg R, Tegner J (2003) Systems biology is taking off. Genome Res 13: 2377–2380.
- http://www.springer.com/us/book/9780387989921
***** LINK TO [[id:91d46e60-eec3-48c7-97f1-b53eb1559b3f][Co-expression]]
:PROPERTIES:
:ID:       632d1c7d-aecc-4ad5-85ad-0f26571e454d
:END:
***** [[id:abb6fc6c-07d9-4375-a8f6-a87370f5c0ea][Multivariate biomarker discovery methods]]
:PROPERTIES:
:ID:       34a99004-090f-44c6-a721-f5f1054e5b1f
:END:
- JCO top cited biomarker studies
      - with cfDNA copy number
        - cite:stover2018
        - cite:slaven2018
      - past 1 year
        - innocenti2019.pdf
        - loi2019.pdf
        - ott2019.pdf
      - past 5 years
        - oxnard2016.pdf
        - roemer2016.pdf
        - fribbens2016.pdf
        - rizvi2018.pdf
*** General reference                                                   :ref:
- [[id:D2FAA88D-70CC-4450-BBB3-8A446CB43AF3][Docker software]]
- https://github.com/orgs/genome/repositories?type=all
*** Coordinate-based genomics visualization tracks
:PROPERTIES:
:ID:       29821fa1-acfe-4a74-9c5d-a24adddf2969
:END:
- https://cmdcolin.github.io/awesome-genome-visualization/?citations=-1&tag=Coverage
- https://rockefelleruniversity.github.io/RU_VisualizingGenomicsData/exercises/answers/Viz_part1_answers.html

- motif tracks https://bioconductor.org/packages/release/bioc/vignettes/motifStack/inst/doc/motifStack_HTML.html
- plotgardener
- soGGi https://bioconductor.org/packages/release/bioc/vignettes/soGGi/inst/doc/soggi.pdf
- aggregate signal based
  - https://bioconductor.org/packages/release/bioc/vignettes/metagene2/inst/doc/metagene2.html
  - https://bioconductor.org/packages/release/bioc/vignettes/chromstaR/inst/doc/chromstaR.pdf
  - https://bioconductor.org/packages/release/bioc/vignettes/CoverageView/inst/doc/CoverageView.pdf
  - https://bioconductor.org/packages/release/bioc/vignettes/profileplyr/inst/doc/profileplyr.html
- specific signal-based
  - https://www.bioconductor.org/packages/release/bioc/vignettes/wiggleplotr/inst/doc/wiggleplotr.html
  - gviz
    - https://bioconductor.org/packages/devel/bioc/vignettes/Gviz/inst/doc/Gviz.html
    - https://rockefelleruniversity.github.io/RU_VisualizingGenomicsData/exercises/answers/Viz_part1_answers.html
    - https://bioconductor.org/packages/release/bioc/vignettes/GenVisR/inst/doc/Intro.html
  - https://bioconductor.org/packages/release/bioc/vignettes/soGGi/inst/doc/soggi.pdf
  - https://phanstiellab.github.io/plotgardener/reference/plotSignal.html
** Variant databases
https://www.oncokb.org/
https://civicdb.org/welcome
** Bioinformatics consortia, data portals and commons
:PROPERTIES:
:ID:       0e7d0c19-22cc-4502-a3c2-4be5b537f375
:END:
- [[https://portal.gdc.cancer.gov/][Genomic data commons]]
- [[https://datascience.cancer.gov/data-commons][NCI Cancer Research Data Commons]]
- [[https://www.cancer.gov/research/infrastructure/bioinformatics/cancer-research-data-ecosystem-infographic][NCI Cancer Research Data Ecosystem]]
- GENIE - https://aacrjournals.org/cancerdiscovery/article/7/8/818/6485/AACR-Project-GENIE-Powering-Precision-Medicine
- https://www.science.org/doi/full/10.1126/scisignal.2004088
- https://www.nature.com/articles/s41588-021-00791-5#Sec2
- [[cite:&langmead2018]]
- [[cite:&grossman2019]]
- [[cite:&bernasconi2020]]
- [[cite:&grossman2018]]
** Scripts
*** R
**** [[file:~/repos/basecamp/basecamp.org::*R scripts][R scripts in basecamp]]
*** Bash
**** Check run conditions for integration testing

#+begin_src bash
#!/usr/bin/env bash

repo=$1
mntpt=$2
sif_dir=$3

# Check for parameters, return usage if empty
if [ $# -ne 3 ];
then
    printf "\n usage: repo_startup.sh <REPO PATH> <RIS MOUNT PT> <SINGULARITY CONTAINER DIR>
    \n Repo development helper script to check run conditions for integration testing
    \n "
else

    # Check mount point
    if grep -qs $mntpt /proc/mounts; then
        echo "RIS storage mounted."
    else
        echo "RIS storage NOT mounted, exiting."
        exit 1
    fi

    # Check singularity container
    if [ -r $sif_dir/atac.sif ]; then
        echo "Local SIF file present"
    else
        echo "No local SIF file found"
        exit 1
    fi

    # Check singularity container up-to-date
    if [ /mnt/ris/jschwarz/cardiac-radiobiology/atac.sif -nt $sif_dir/atac.sif ]; then
        echo "Local SIF is out of date. Updating ..."
        cp /mnt/ris/jschwarz/cardiac-radiobiology/atac.sif $sif_dir/atac.sif
    else
        echo "Local SIF file is up to date"
    fi

    cur_branch=$(git branch | head -n 1)
    echo "Current branch is $cur_branch"
fi
#+end_src

**** Dry run
:PROPERTIES:
:ID:       c6c494f0-5dbe-4cc4-bf99-8bcd7b12de24
:END:

#+begin_src bash :tangle ./tools/shell/smk_dry_run.sh
#  Note: This code block transcluded from code blocks in the Emacs Org-mode
#  file at https://github.com/jeszyman/biotools/blob/master/biotools.org.
#  Changes made directly to this region will be overwritten by transclusion from
#  the source blocks in that file.

# Check for parameters and return usage
if [ "$#" -ne 2 ];
then
    printf "\n usage: smk_dry_run.sh <SMK CONFIG YAML> <SMK FILE>
    \n Script for snakemake dry run.
    \n Will not test singularity container is working
    \n "
else
    # Necessary to run conda snakemake command in shell script
    eval "$(command conda 'shell.bash' 'hook' 2> /dev/null)"
    #
    conda activate snakemake
    #
    snakemake \
        --configfile $1 \
        --cores 1 \
        --dry-run \
        --forceall \
        --printshellcmds \
        --use-singularity \
        --snakefile $2
fi
#+end_src

**** Draw rulegraph
:PROPERTIES:
:ID:       78a99954-1d5d-4e09-87a4-6690f28f4519
:CUSTOM_ID: 78a99954-1d5d-4e09-87a4-6690f28f4519
:END:

#+begin_src bash :tangle ./tools/shell/smk_draw.sh
#  Note: This code block transcluded from code blocks in the Emacs Org-mode
#  file at https://github.com/jeszyman/biotools/blob/master/biotools.org.
#  Changes made directly to this region will be overwritten by transclusion from
#  the source blocks in that file.

# Check for parameters and return usage
if [ "$#" -ne 3 ];
then
    printf "\n usage: smk_dry_run.sh config_file smk_file pdf_loc
    \n Draws the rulegraph for a snakemake file at pdf_loc
    \n "
else
    # Necessary to run conda snakemake command in shell script
    eval "$(command conda 'shell.bash' 'hook' 2> /dev/null)"
    #
    png=$(echo $3 | sed 's/.pdf/.png/g')
    conda activate snakemake
    #
    snakemake \
        --configfile $1 \
        --cores 1 \
        --rulegraph \
        --snakefile $2 | dot -Tpdf > $3
    #
    snakemake \
        --configfile $1 \
        --cores 1 \
        --rulegraph \
        --snakefile $2 | dot -Tpng > $png
fi
#+end_src

**** TODO [#D] Normal run
:PROPERTIES:
:effort:   00:30
:ID:       3adf72d3-aa35-4a3d-8eba-0df075c52e26
:CUSTOM_ID: 3adf72d3-aa35-4a3d-8eba-0df075c52e26
:END:

- [ ] needs a cores option w/ default
#+begin_src bash :tangle ./tools/shell/smk_run.sh
#  Note: This code block transcluded from code blocks in the Emacs Org-mode
#  file at https://github.com/jeszyman/biotools/blob/master/biotools.org.
#  Changes made directly to this region will be overwritten by transclusion from
#  the source blocks in that file.

# Check for parameters, return usage if empty
if [ "$#" -ne 2 ];
then
    printf "\n usage: smk_forced_run.sh <SMK CONFIG YAML> <SMK FILE>
    \n Script to complete an normal run of a snakefile
    \n Assumes
    - Singularity container specified in config
    - a mount point at /mnt
    \n "
else
    # Necessary to run conda snakemake command in shell script
    eval "$(command conda 'shell.bash' 'hook' 2> /dev/null)"
    #
    conda activate snakemake
    #
    snakemake \
        --configfile $1 --cores 4 \
        --use-singularity \
        --printshellcmds \
        --singularity-args "--bind ${HOME}:${HOME} --bind /mnt:/mnt" \
        --rerun-incomplete \
        --resources mem_mb=6000 \
        --snakefile $2 \
        --verbose
fi
#+end_src

**** Forced run
:PROPERTIES:
:ID:       c2a25d2f-05bb-4eda-920f-e2d9fa768de6
:CUSTOM_ID: c2a25d2f-05bb-4eda-920f-e2d9fa768de6
:END:

#+begin_src bash :tangle ./tools/shell/smk_forced_run.sh
#  Note: This code block transcluded from code blocks in the Emacs Org-mode
#  file at https://github.com/jeszyman/biotools/blob/master/biotools.org.
#  Changes made directly to this region will be overwritten by transclusion from
#  the source blocks in that file.

# Check for parameters and return usage
if [ "$#" -ne 2 ];
then
    printf "\n usage: smk_forced_run.sh config_file smk_file
    \n Script to complete forced run of snakefile.
    \n Assumes
    - Singularity container specified in config
    - a mount point at /mnt
    \n "
else
    # Necessary to run conda snakemake command in shell script
    eval "$(command conda 'shell.bash' 'hook' 2> /dev/null)"
    #
    conda activate snakemake
    #
    snakemake \
        --configfile $1 --cores 4 \
        --use-singularity \
        --singularity-args "--bind ${HOME}:${HOME} --bind /mnt:/mnt" \
        --forceall \
        --printshellcmds \
        --rerun-incomplete \
        --snakefile $2 \
        --verbose
fi
#+end_src

**** Development
:PROPERTIES:
:header-args: :tangle no
:END:

- https://unix.stackexchange.com/questions/10646/repeat-a-unix-command-every-x-seconds-forever
- ?
  #+begin_src bash
#!/bin/bash
#
#############################################################################
###              BIOINFORMATICS PROJECT COMMON SETUP SCRIPT               ###
#############################################################################
#
###############
### SCRIPTS ###
###############
#
# Interactive docker initialization
docker_interactive() {
  repo="USER INPUT"
  read -p "repo name: " repo
  case $HOSTNAME in
      (radonc-cancerbio) docker run -it \
                                -v /media/:/media/ \
                                -v /home/:/home/ \
                          -u $(id -u ${USER}) \
                          jeszyman/$repo \
                          /bin/bash;;
      (jeszyman-*) docker run -it \
                          -v /home/:/home/ \
                          -u $(id -u ${USER}) \
                          jeszyman/$repo \
                          /bin/bash;;
      (acl*) docker run -it \
                    -v /drive3/:/drive3/ \
                    -v /duo4/:/duo4/ \
                    -v /home/:/home/ \
                    -u $(id -u ${USER}) \
                    jeszyman/$repo \
                    /bin/bash;;
      (ACL*) docker run -it \
                    -v /home/:/home/ \
                    -v /duo4/:/duo4/ \
                    -u $(id -u ${USER}):$(id -g ${USER}) \
                    -h=${HOSTNAME} \
                    jeszyman/$repo \
                    /bin/bash;;
      (virtual-workstation*.gsc.wustl.edu) bsub -Is -q docker-interactive -a 'docker(jeszyman/'"$repo"')' /bin/bash;;
  esac
}
#
#
echo "Biotools setup completed successfully"
#
############################################################################
#
#############
### IDEAS ###
#############
# # identifies server in docker containers
# server_ask() {
#     echo -n "Choose your server: "
#     read SERVER
#     SERVER="${SERVER:=chaudhuri-roche}"
# }
# # check if inside docker
# if [ -f /.dockerenv ];then
#     dockercontainer=yes
#     echo "Your are in a docker container"
# else
#     dockercontainer=no
#     echo "Not in docker"
#     # setup NON-docker variables
#     shopt -s nocasematch
#     case $HOSTNAME in
#         (radonc-cancerbio) server_home="/home/jeszyman" ;;
#         (jeszyman-*) server_home="/home/jeszyman" ;;
#         (acl*) server_home="/home/jszymanski" ;;
#         (virtual-workstation*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
#         (blade*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
#     esac
#     projectdir=$server_home"/repos/$project"
#     localdata="$server_home/data/$project/"
#     unset $SERVER
# fi
# #
# case $dockercontainer in
#     (yes) server_ask ;;
#     (no) ;;
# esac
# #
# case $SERVER in
#     (chaudhuri-roche) projectdir=/drive3/users/jszymanski/repos/$project
#                       localdata=/drive3/users/jszymanski/data/$project;;
#     (NA) ;;
# esac
#
#IDEAS
### DEPENDENCIES ###
#declare -a software=(
#"rclone"
# #DOES NOT RETURN ON MGI                     "docker"
# #"git"
# "bash"
# )
# for i in "${software[@]}"; do
# if command -v $i >/dev/null 2>&1 ; then
#     echo "$i installed"
# else
#     echo "$i not found, exiting"
#     exit 1
# fi
# done

#+end_src
- example [[id:8263e3b1-3ba4-4c4f-8d39-9dd9bb06e2bb][publish to box]]
  #+begin_src bash
#!/usr/bin/env bash
git pull &&
    git add -A &&
    git commit -m "minor changes" &&
    git push

git branch
git branch stable4.1
git checkout stable4.1
git merge master
git push origin stable4.1
git checkout master
#+end_src

***** Dev

#+begin_src bash
#!/bin/bash
#########1#########2#########3#########4#########5#########6#########7#########8

####################################
###   Choose and Run Snakefile   ###
####################################

# Setup
#set -euxov pipefail

source config/${HOSTNAME}.sh
echo "The following `*.smk` archives were found; select one:"

# set the prompt used by select, replacing "#?"
PS3="Use number to select an option"

select filename in ./workflow/*.smk

do
    if [[ "$filename" == "" ]]
    then
        echo "'$REPLY' is not a valid number"
        continue
    fi
    echo $filename
    select run_option in dry_run normal force_final force_all
    do
        echo selected $run_option
        case $run_option in
            dry_run)
                source activate snakemake
                snakemake \
                    --configfile config/${HOSTNAME}.yaml \
                    --cores $threads \
                    --directory ${repo} \
                    --dry-run \
                    --rerun-incomplete \
                    --singularity-args "--bind $mntpt:$mntpt" \
                    --use-singularity \
                    --snakefile $filename
                ;;
            normal)
                source activate snakemake
                select nohup_option in no yes
                do
                    case $nohup_option in
                        no)
                            snakemake \
                                --configfile config/${HOSTNAME}.yaml \
                                --cores $threads \
                                --directory ${repo} \
                                --singularity-args "--bind $mntpt:$mntpt" \
                                --use-singularity \
                                --snakefile $filename
                            ;;
                        yes)
                            nohup snakemake \
                                  --configfile config/${HOSTNAME}.yaml \
                                  --cores $threads \
                                  --directory ${repo} \
                                  --singularity-args "--bind $mntpt:$mntpt" \
                                  --use-singularity \
                                  --snakefile $filename
                            ;;
                    esac
                done
                ;;
            force_final)
                source activate snakemake
                select nohup_option in no yes
                do
                    case $nohup_option in
                        no)
                            snakemake \
                                --configfile config/${HOSTNAME}.yaml \
                                --cores $threads \
                                --directory ${repo} \
                                --force \
                                --singularity-args "--bind $mntpt:$mntpt" \
                                --use-singularity \
                                --snakefile $filename
                            ;;
                        yes)
                            nohup snakemake \
                                  --configfile config/${HOSTNAME}.yaml \
                                  --cores $threads \
                                  --directory ${repo} \
                                  --force \
                                  --singularity-args "--bind $mntpt:$mntpt" \
                                  --use-singularity \
                                  --snakefile $filename
                            ;;
                    esac
                done
                ;;
            force_all)
                source activate snakemake
                select nohup_option in no yes
                do
                    case $nohup_option in
                        no)

                snakemake \
                    --configfile config/${HOSTNAME}.yaml \
                    --cores $threads \
                    --directory ${repo} \
                    -F \
                    --singularity-args "--bind $mntpt:$mntpt" \
                    --use-singularity \
                    --snakefile $filename
                                            ;;
                        yes)
                nohup snakemake \
                    --configfile config/${HOSTNAME}.yaml \
                    --cores $threads \
                    --directory ${repo} \
                    -F \
                    --singularity-args "--bind $mntpt:$mntpt" \
                    --use-singularity \
                    --snakefile $filename
                            ;;
                    esac
                done
                ;;
        esac
        break
    done
    break
done
#+end_src

**** Examples

- https://github.com/jeszyman/exrna
- https://github.com/jeszyman/cfdna-wgs
- https://github.com/jeszyman/cappseq
- https://github.com/jeszyman/cfdna-frag
- https://github.com/jeszyman/atac-seq

**** [[https://snakemake.readthedocs.io/en/stable/index.html][Documentation]]

- https://snakemake-wrappers.readthedocs.io/en/stable/wrappers.html
- [[Snakemake Tutorial][Snakemake Tutorial]]

**** General reference

- https://github.com/aomlomics/tourmaline
- https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/
- https://swcarpentry.github.io/r-novice-inflammation/05-cmdline/index.html
- https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/
- #+PROPERTY: header-args:bash :tangle-mode (identity #o777)
- ideas
  `(make-string (+ 2 (org-current-level)) ?*)` Run commands
  ,#+begin_src bash
  conda activate snakemake

  snakemake \
    --cores $threads \
    --printshellcmds \
    --snakefile workflow/$1
  ,#+end_src
- Standards and best practice
  - https://snakemake.readthedocs.io/en/stable/snakefiles/best_practices.html#snakefiles-best-practices
- Wildcards
  - https://edwards.flinders.edu.au/wildcards-in-snakemake/
- https://endrebak.gitbooks.io/the-snakemake-book/content/
- https://academic.oup.com/bioinformatics/article/28/19/2520/290322
- https://www.biostars.org/p/9469533/
- https://stackoverflow.com/questions/64949149/is-it-possible-to-add-a-conditional-statement-in-snakemakes-rule-all
- https://stackoverflow.com/questions/54400171/how-to-use-list-in-snakemake-tabular-configuration-for-describing-of-sequencing
- https://github.com/ngs-docs/2020-ggg-201b-rnaseq/blob/latest/rnaseq-workflow.Rmd
- https://angus.readthedocs.io/en/2019/snakemake_for_automation.html
- https://pages.charlesreid1.com/how-do-i-snakemake/converting/
- http://ivory.idyll.org/blog/2020-snakemake-hacks-collections-files.html
- http://origin.research.jetbrains.org/files/material/5ec245b4e907c.pdf
- https://f1000research.com/articles/10-33/v2
- [ ] https://lachlandeer.github.io/snakemake-econ-r-tutorial/config-files.html
- https://github.com/snakemake-workflows/rna-seq-star-deseq2/blob/master/workflow/rules/diffexp.smk
- https://github.com/csoneson/ARMOR
- https://github.com/slowkow/snakefiles/blob/master/kallisto/Snakefile
- https://lachlandeer.github.io/snakemake-econ-r-tutorial/initial-steps-with-snakemake.html
- for repos, git ignore the singularity build
- https://github.com/EnvGen/snakemake-workflows

***** Tutorials and guides

- https://reproducibility.sschmeier.com/index.html
- a tutorial
  #+begin_src bash
source ~/repos/biotools/src/functions.sh
docker_interactive
biotools

#
mkdir -p /tmp/snakemake-tutorial
cd /tmp/snakemake-tutorial
#
wget https://github.com/snakemake/snakemake-tutorial-data/archive/v5.24.1.tar.gz
#
tar --wildcards -xf v5.24.1.tar.gz --strip 1 "*/data" "*/environment.yaml"
ls

#
snakemake --help

snakemake -np mapped_reads/A.bam
#+end_src
- another tutorial
  #+begin_src bash
sudo apt-get install singularity-container

cd ~/snakemake-tutorial/

conda activate snakemake-tutorial


snakemake -p --cores 2 sorted_reads/{A,B}.bam
#    --dry-run \

snakemake --dag calls/all.vcf | dot -Tsvg > dag.svg

expand("sorted_reads/{sample}.bam", sample=SAMPLES)

snakemake -p --cores 2 plots/quals.svg

snakemake --cores 2 -p --dag | dot -Tsvg > dag.svg

snakemake --cores 20 --forceall --summary

snakemake --cores 20 --forceall

conda install pygraphviz

snakemake --report report.html

open report.html

snakemake --cores 20 --forceall
#+end_src
- https://snakemake.readthedocs.io/en/stable/executor_tutorial/google_lifesciences.html

**** Examples

- https://github.com/vari-bbc/rnaseq_workflow/blob/master/Snakefile
- https://github.com/ohsu-cedar-comp-hub/cfRNA-seq-pipeline-Ngo-manuscript-2019
- https://github.com/yztxwd/rna-seq-standard-pipeline/tree/c19a9e27216b86d3dbfdaed7228aeff59f869383
- https://github.com/porchard/ATACseq-Snakemake/blob/master/src/Snakefile
- https://github.com/zhxiaokang/RASflow
- https://github.com/EthanHolleman
- https://www.biostars.org/p/406385/
- https://github.com/chrishah/samtools-docker/blob/master/Dockerfile
- https://github.com/lachlandeer/snakemake-econ-r

**** Config


:PROPERTIES:
:ID:       5120c9dd-8e0b-4b2c-8c77-a89f271a48e4
:END:

#+name: bioinfo.config.common.sh
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.config.common.sh
# -*- mode: snippet -*-
# key: bioinfo.config.common.sh
# expand-env: ((yas-indent-line 'fixed))
# --
,#+name: bash_common_config
,#+begin_src bash :noweb yes :tangle ./config/common.sh
#!/bin/bash
#########1#########2#########3#########4#########5#########6#########7#########8
#

# Check git file hook is read-able
if [ -r "${repo}/.git/hooks/precommit" ]; then
   echo "Git size check is read-able"
else
    echo
    "Git size check is not read-able"
    exit 1
fi

# Check mount point
if grep -qs $mntpt /proc/mounts; then
    echo "RIS storage mounted."
else
    echo "RIS storage NOT mounted, exiting."
    exit 1
fi

# Check singularity container
if [ -r $sif_dir/atac.sif ]; then
    echo "Local SIF file present"
else
    echo "No local SIF file found"
fi
    fi
}
#+end_src

**** Bioinformatic shell functions
:PROPERTIES:
:header-args: :tangle ./lib/bioinformatics_function.sh :tangle-mode (identity #o555) :mkdirp yes :noweb yes :comments org
:ID:       2e51c010-1e25-4fc3-acc5-aefd490164be
:END:

** Local YAS
*** r.bioc :yas:
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/ess-mode/r.bioc
# -*- mode: snippet -*-
# key: r.bioc
# expand-env: ((yas-indent-line 'none))
# --
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
#+end_src
*** UPDATE_ON_USE bioinfo.setup
CLOSED: [2021-12-09 Thu 12:38]
:PROPERTIES:
:ID:       dbd3c67c-e33b-42d8-800d-705292d3c372
:CUSTOM_ID: dbd3c67c-e33b-42d8-800d-705292d3c372
:END:

#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.setup
# -*- mode: snippet -*-
# key: bioinfo.setup
# expand-env: ((yas-indent-line 'none))
# --
`(concat "*" (make-string (org-current-level) ?*))` Setup
`(concat "**" (make-string (org-current-level) ?*))` Bash
`(concat "***" (make-string (org-current-level) ?*))` TODO Configurations
- Local (by $HOSTNAME)
- Common
  ,#+begin_src bash :tangle ./config/common.sh
,#+end_src
`(concat "***" (make-string (org-current-level) ?*))` TODO Preambles
- Command line
  ,#+name: bash_preamble_cmd
  ,#+begin_src bash
if [[ -f ./config/\${HOSTNAME}.sh ]];
then source ./config/\${HOSTNAME}.sh
else
    echo "No config file found for \${HOSTNAME}"
    exit 1
fi
,#+end_src
- Snakemake
`(concat "**" (make-string (org-current-level) ?*))` TODO Box and rclone
`(concat "**" (make-string (org-current-level) ?*))` TODO Docker and singularity
`(concat "***" (make-string (org-current-level) ?*))` Processes
1. Write
   ,#+begin_src fundamental :tangle ./config/DOCKER_IMAGE_NAME_Dockerfile
,#+end_src
2. Build
3. Make singularity container from docker container
`(concat "**" (make-string (org-current-level) ?*))` TODO Emacs and Org-mode
- [ ] Run minimal YAS
  bioinfo.setup
  bioinfo.inputs
  bioinfo.data
  locvars
- [ ] Add
  - to main reference section
    - Bioinformatics project [[6acdcddf-d693-40f2-a3ae-9703bad56043][module headline]] and [[tag:{bioinfo}][instances]]
  - to Local variables
    ,#+TODO: WAITING(w@) TODO(t) INPROCESS(p) | CLOSEOUT DONE(d!) DELEGATED(@) CANCELED(@)
    ,#+property: header-args:R    :session *PROJECT_NAME*
`(concat "**" (make-string (org-current-level) ?*))` TODO Local data directory and WashU RIS Storage processes
- Mount
- Create directory structure
  ,#+begin_src bash
  <<bash_preamble>>
  ,#+end_src
`(concat "**" (make-string (org-current-level) ?*))` R
`(concat "***" (make-string (org-current-level) ?*))` TODO Configurations
`(concat "***" (make-string (org-current-level) ?*))` TODO Preambles
`(concat "**" (make-string (org-current-level) ?*))` TODO Repository, git, and github
- [ ] ensure [[file:.git/hooks/pre-commit]]
- [ ] Create repository structure
  ,#+begin_src bash
mkdir -p "\${repo}/config"
mkdir -p "\${repo}/resources"
mkdir -p "\${repo}/results"
mkdir -p "\${repo}/src"
mkdir -p "\${repo}/workflow/scripts"
,#+end_src
`(concat "***" (make-string (org-current-level) ?*))` README
`(concat "**" (make-string (org-current-level) ?*))` Snakemake
`(concat "***" (make-string (org-current-level) ?*))` TODO Configurations
`(concat "***" (make-string (org-current-level) ?*))` TODO Run commands
#+end_src

*** bioinfo.launch.docker

#+name: bioinfo.launch.docker
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.launch.docker
  # -*- mode: snippet -*-
  # key: bioinfo.launch.docker
  # expand-env: ((yas-indent-line 'fixed))
  # --
  ,#+begin_src bash
  #!/bin/bash
  #########1#########2#########3#########4#########5#########6#########7#########8
  #

  # Notes: This block build from bioinfo.launch.docker at
  #  ce85a21e-466f-43d0-ad13-3cd21497bfc3

  # Function to launch an interactive docker session

  ,#+end_src
#+end_src

*** bioinfo.setup.bash

#+name: bioinfo.setup.bash
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.setup.bash
  # -*- mode: snippet -*-
  # key: bioinfo.setup.bash
  # expand-env: ((yas-indent-line 'fixed))
  # --
  `(make-string (1+ (org-current-level)) ?*)` Bash
  Potential YAS to run:
  bioinfo.config.host.sh
  bioinfo.config.main.sh
  ris.mount
  $0
#+end_src

*** bioinfo.config.main.R

#+name: bioinfo.config.main.R
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.config.main.R
  # -*- mode: snippet -*-
  # key: bioinfo.config.main.R
  # expand-env: ((yas-indent-line 'fixed))
  # --
  ,#+begin_src R :noweb yes :tangle ./config/main.R
  #!/usr/bin/env Rscript
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  # Repository-local, relative variables
  $0

  # Pull in from other repositories
  ##
  ## Basecamp
  <<./basecamp/basecamp.org:basecamp_r_setup_wrapper()>>

  ##
  ## Biotools
  <<./biotools/biotools.org:biotools_r_setup_wrapper()>>

  ,#+end_src
#+end_src

*** bioinfo.config.local.R

#+name: bioinfo.config.local.R
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.config.local.R
  # -*- mode: snippet -*-
  # key: bioinfo.config.local.R
  # expand-env: ((yas-indent-line 'fixed))
  # --
  ,#+begin_src R :noweb yes :tangle ./config/${1:HOSTNAME}.R
  #!/usr/bin/env Rscript
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  # Repository-local, host-absolute variables
  data_dir = file.path("${2:DATA_DIR}")
  source("./config/main.R")
  $0
  ,#+end_src
#+end_src

*** bioinfo.bash.script

#+name: bioinfo.bash.script
#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.bash.script
  # -*- mode: snippet -*-
  # key: bioinfo.bash.script
  # expand-env: ((yas-indent-line 'fixed))
  # --
  #!/bin/bash
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  ### ${1:SCRIPT NAME} ###
  #
  # Notes:
  ## This script assumes your your bash sessin is running within the repository docker container (see ./)
  ## This script calls relative varibles and assumes you have sourced an appriopriate configuration file (see ./config)
#+end_src

*** bioinfo.inputs
:PROPERTIES:
:ID:       d6bb61c9-8b10-4460-b987-0a6c678d2ddb
:CUSTOM_ID: d6bb61c9-8b10-4460-b987-0a6c678d2ddb
:END:

#+begin_src fundamental :tangle ~/repos/emacs.d/snippets/org-mode/bioinfo.inputs
# -*- mode: snippet -*-
# key: bioinfo.inputs
# expand-env: ((yas-indent-line 'none))
# --
`(make-string (1+ (org-current-level)) ?*)` Inputs
,#+name: input_table
|basename|dirname|get_class|model_class|origin
`(concat "**" (make-string (org-current-level) ?*))` Manually created input tables
- (names must end in _input)
-
#+end_src
** ENCODE analysis pipelines
- https://github.com/ENCODE-DCC/atac-seq-pipeline
https://www.genome.gov/sites/default/files/Multimedia/Slides/ENCODE2016-ResearchAppsUsers/ENCODE2016_strattan_slides.pdf
https://www.biorxiv.org/content/10.1101/2023.04.04.535623v1.full.pdf
** Snakemake demo                                                      :demo:
** Glossary
-
** Bioinformatics software tools
*** 0 Bioinformatics software flat list
:PROPERTIES:
:ID:       1b8bb7db-2622-47bd-bd6f-052cbd5d4fdb
:END:
**** Ecotyper
https://www.nature.com/articles/s42003-023-05117-1#Sec2
**** annotationDb
|   |   | [[https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf][doc]] |   |   |
**** annovar
[[http://annovar.openbioinformatics.org/en/latest/][doc]]
**** ATACseqQC                                                      :biotool:
:PROPERTIES:
:ID:       a5b73c70-3f1f-4018-92b2-bfdcaed53cf5
:CREATED:  [2021-09-13 Mon 07:29]
:function: qc
:input: bam
:repo:
:doc:
:paper:
:END:
- ATAC fraction of reads in peaks
- http://biocworkshops2019.bioconductor.org.s3-website-us-east-1.amazonaws.com/page/ATACseqQCWorkshop__ATACseqQC_workshop/
- https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4559-3#Sec15
- https://haibol2016.github.io/ATACseqQCWorkshop/articles/ATACseqQC_workshop.html
- https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2019/ChIPSeq/Materials/Practicals/Day5/Practical01_ATAC-seq_analysis_SS.html
**** BBmap / BBSplit                                                :biotool:
:PROPERTIES:
:ID:       52363054-5138-419c-8d45-4ec9b0b02b24
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- http://seqanswers.com/forums/archive/index.php/t-87619.html
- https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbmap-guide/
- outu = unmapped reads
- snakemake example https://www.biostars.org/p/424706/
- http://seqanswers.com/forums/showthread.php?t=41288
- [ ] BBSplit in BBmap - [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwJZJWQrGSPgdPGNfStWbxXkmhP][email]]]]
- https://www.google.com/search?q=bbmap+docker use bioconda
**** bcl2fastq
https://support.illumina.com/content/dam/illumina-support/documents/documentation/software_documentation/bcl2fastq/bcl2fastq2-v2-20-software-guide-15051736-03.pdf
**** bcl2fastq
https://support.illumina.com/content/dam/illumina-support/documents/documentation/software_documentation/bcl2fastq/bcl2fastq2-v2-20-software-guide-15051736-03.pdf
**** BEDOPS                                                         :biotool:
:PROPERTIES:
:ID:       0be280ad-47b6-42f4-8a65-4fadadf4940b
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- https://bedops.altius.org/forum/index.php?topic=20.0
- https://bedops.readthedocs.io/en/latest/index.html
**** bedtools                                                       :biotool:
:PROPERTIES:
:doc: [[http://bedtools.readthedocs.io/en/latest/][doc]]
:paper: cite:quinlan2010       |
:CREATED:  [2021-09-01 Wed 13:42]
:ID:       05f314e8-f402-45cb-ac1f-e3dece4f3114
:END:

**** BiocParallel :biotool:
https://bioconductor.org/packages/devel/bioc/vignettes/BiocParallel/inst/doc/Introduction_To_BiocParallel.pdf
**** biscuit
:PROPERTIES:
:ID:       5e9e8bfa-ac9e-4103-9cc5-7123337b4e24
:END:
https://github.com/genome/docker-biscuit
**** bismark
https://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf
- https://github.com/FelixKrueger/Bismark/issues/241
- https://github.com/FelixKrueger/Bismark
https://www.nature.com/articles/nmeth.1828
**** BLAT                                                           :biotool:
:PROPERTIES:
:ID:       6e44c130-0eaa-4e44-8c12-c7c8125ab9ec
:CREATED:  [2021-09-13 Mon 07:29]
:END:
  - https://www.youtube.com/watch?v=E6DDNGDXH8A
**** bowtie2                                                        :biotool:
:PROPERTIES:
:ID:       8883a37b-867c-4c19-8e9c-2b46004125a1
:CREATED:  [2021-09-13 Mon 07:29]
:END:
| bowtie2 |   |   | [[https://sourceforge.net/projects/bowtie-bio/files/][src]] | [[http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#the-bowtie2-build-indexer][doc]] |   |   |
- [[https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471250953.bi1107s32][protocol paper]]
- [[http://bowtie-bio.sourceforge.net/index.shtml][doc]]
**** bwa :biotools:
https://www.biostars.org/p/270090/
| bwa | align | fastq |   | [[http://bio-bwa.sourceforge.net/bwa.shtml][site]] |   |   |
**** canvas                                                         :biotool:
:PROPERTIES:
:ID:       74ef57cd-c44a-4f28-b4a1-5a54a82ce5a3
:CREATED:  [2021-09-13 Mon 07:29]
:END:
https://github.com/Illumina/canvas


  - https://github.com/chrisamiller/copyCat
  - for cah https://bioconductor.org/packages/release/bioc/html/DNAcopy.html


**** cappSeq
- cappseq https://mail.google.com/mail/u/0/?ui=2&ik=ff50768eb7&view=lg&permmsgid=msg-f:1658523263188957345
**** ChIPseeker :biotool:
:PROPERTIES:
:ID:       a4cf29e3-7fa6-4ea2-a538-f59ec450cc7b
:END:
- [[https://rdrr.io/bioc/ChIPseeker/man/annotatePeak.html][rdrr ChIPseeker]]
- [[https://bioconductor.org/packages/devel/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html#:~:text=ChIPseeker%20is%20an%20R%20package,peaks%20binding%20to%20TSS%20regions.&text=Moreover%2C%20it%20supports%20evaluating%20significant%20overlap%20among%20ChIP%2Dseq%20datasets][ChIPseeker vignette]].

**** CIBERSORTx and cytotrace                                       :biotool:
:PROPERTIES:
:ID:       6fdb7ece-efbb-4523-a7a5-7dd594d199dd
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- https://www.cell.com/cell/fulltext/S0092-8674(21)01061-8?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867421010618%3Fshowall%3Dtrue
- https://ecotyper.stanford.edu/
- https://cytotrace.stanford.edu/
**** clusterProfiler
:PROPERTIES:
:ID:       8e4c1fb8-cd52-4fac-900a-4b6e6eb5a28a
:END:
- https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/
- https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html
**** cnvkit                                                         :biotool:
:PROPERTIES:
:ID:       b471102b-2e5a-4d67-97e6-b03cf4d923b9
:CREATED:  [2021-09-13 Mon 07:29]
:END:


| CNVkit | copy# | bam | [[https://github.com/etal/cnvkit/tree/fd3555256d2a5aa3978c475d19c88065be68c6c4][repo]] | [[https://cnvkit.readthedocs.io/en/stable/index.html][doc]] | [[https://doi.org/10.1371/journal.pcbi.1004873][paper]] |   |
- modify autobin https://github.com/etal/cnvkit/blob/fa86632a5731ebbefb0092069c8031f806c8b611/cnvlib/autobin.py
oldcode from mpnst

# tumor_bam="/duo4/capp-runs/HiSeqW11/demultiplexed/Sample_PC3-N1_Normal/Sample_PC3-N1_Normal.sorted.samtools-deduped.sorted.bam"
# normal_bam="/duo4/capp-runs/HiSeqW11/demultiplexed/Sample_69-C1_cfDNA/Sample_69-C1_cfDNA.sorted.samtools-deduped.sorted.bam"
# bait_bed="/duo4/users/jszymanski/projects/mpnst/170907_HG38_CCDS1X_REZ_capture_targets.bed"
# fasta="/duo4/users/jszymanski/data/ref/hg19.fa"

# cd $workdir/analysis/


# nohup cnvkit.py batch \
# 	  $tumor_bam \
# 	  --normal $normal_bam \
# 	  --targets $bait_bed \
#           --fasta $fasta \
#           --output-reference my_reference.cnn \
# 	  --output-dir results/ \
#           --diagram --scatter

# # Reusing a reference for additional samples
# cnvkit.py batch *Tumor.bam -r Reference.cnn -d results/

# # Reusing targets and antitargets to build a new reference, but no analysis
# cnvkit.py batch -n *Normal.bam --output-reference new_reference.cnn \
#     -t my_targets.bed -a my_antitargets.bed \
#     -f hg19.fasta -g data/access-5kb-mappable.hg19.bed
# https://cnvkit.readthedocs.io/en/stable/nonhybrid.html
## considers WGS as jsut no off-target regions
## so no target file
## NEED TO ANNOTATE
#wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/refFlat.txt.gz
#mv refFlat.txt.gz /duo4/users/jszymanski/data/ref/
#cd /duo4/users/jszymanski/data/ref/
#gunzip refFlat.txt.gz

source ~/repos/mpnst/bin/local-setup.sh
docker_interactive
biotools
mkdir -p /duo4/mpnst/downsample-cnn
cd /duo4/mpnst/downsample-cnn


cnvkit.py batch /duo4/mpnst/downsample-bam/pn*.sorted.bam /duo4/mpnst/downsample-bam/mp*.sorted.bam\
          -n /duo4/mpnst/downsample-bam/nm*.sorted.bam \
          -m wgs -f /drive3/users/jszymanski/data/ref/bwa-hg19/hg19.fa \
          --annotate /drive3/users/jszymanski/data/ref/refFlat.txt -p 32

- https://www.biostars.org/p/247903/
- https://gatkforums.broadinstitute.org/gatk/discussion/2909/how-to-fix-a-badly-formatted-bam
- [[https://github.com/lh3/seqtk][seqtk]]
- sequence file manipulation
  - https://gist.github.com/davfre/8596159
  - [[https://www.biostars.org/p/6544/][look at all the ways to split a bam file!]]
**** complexheatmap                                                 :biotool:
:PROPERTIES:
:ID:       af78e4e0-a184-480f-95ac-ce9a0db35246
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- https://www.biostars.org/p/286187/#286507
**** Cytoscape                                                      :biotool:
:PROPERTIES:
:ID:       46cf789e-d185-46da-bb32-8cf6113f3425
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- [[http://manual.cytoscape.org/en/stable/index.html][documentation]]
- https://stackoverflow.com/questions/tagged/cytoscape
**** deeptools
| deepTools |   |   |   | [[https://deeptools.readthedocs.io/en/latest/content/example_usage.html][site]] |   |   |
**** DEGreport
https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
https://bioconductor.org/packages/release/bioc/vignettes/DEGreport/inst/doc/DEGreport.html
https://www.biostars.org/p/497108/
**** DESeq2                                                         :biotool:
:PROPERTIES:
:ID:       2F642F69-A9C8-46DA-BE31-783AC8D4F4B6
:CREATED:  [2021-01-27 Wed 07:00]
:END:
- https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html
- remove factor levels https://support.bioconductor.org/p/61875/
| DESeq2 | diff express |   |   |   |   | cite:love2014 |
- Notes
  - a negative binomial GLM
- https://genviz.org/module-04-expression/0004/02/01/DifferentialExpression/

- https://bioc.ism.ac.jp/packages/2.14/bioc/vignettes/DESeq2/inst/doc/beginner.pdf
- [[file:~/Box%20Sync/library/Love,14%20Moderated%20estimation%20of%20fold%20change%20and%20dispersion%20for%20RNA%20seq%20data%20with%20DESeq2.pdf][Love,14 Moderated estimation of fold change and dispersion for RNA seq data with DESeq2]]
- http://www.nathalievilla.org/doc/html/solution-DESeq2-rnaseq.html


***** [[https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html][documentation]]
***** Processes
- Statistical design
  - https://support.bioconductor.org/p/110163/
  - handling nested designs https://support.bioconductor.org/p/64480/
  - LRT
    - https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html
  - Methods writeup
    - LRT is evaluated akin to ANOVA with Wald tests serving as post-hoc significance of specific group comparisons.
- import functions
  | function          | package           | framework | output               | DESeq2 input function    |
  | summarizeOverlaps | GenomicAlignments | R         | SummarizedExperiment | DESeqDataSet             |
  | featureCounts     | Rsubread          | R         | matrix               | DESeqDataSetFromMatrix   |
  | tximport          | tximport          | R         | list of matrices     | DESeqDataSetFromTximport |
  | htseq-count       | HTSeq             | Python    | files                | DESeqDataSetFromHTSeq    |
- reset design https://support.bioconductor.org/p/9135927/
- Subset https://www.google.com/search?q=subset+deseqdataset
  - https://www.biostars.org/p/442188/
***** Examples- see [[file:~/Box%20Sync/comp_bio/deseq2_template/][deseq2_template]]
:PROPERTIES:
:ID:       5C8E2A3B-2C16-4FB3-9F86-5A1D8AD4802F
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- vignettes
  - [[https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#exploratory-analysis-and-visualization][gene workflow 19]]
  - [[http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#using-parallelization][package vignette]]
****** DONE Example 1: [[file:~/Box%20Sync/comp_bio/deseq2_template/Love,15%20RNA%20seq%20workflow%20gene%20level%20exploratory%20analysis%20and%20differential%20expression.pdf][Love,15 Airway]]
CLOSED: [2016-06-27 Mon 14:45]
:PROPERTIES:
:ID:       791A1ABC-4138-4CD2-82AE-63905C053571
:CREATED:  [2021-09-13 Mon 07:29]
:END:
:LOGBOOK:
CLOCK: [2016-06-27 Mon 10:22]--[2016-06-27 Mon 10:37] =>  0:15
CLOCK: [2016-06-23 Thu 09:22]--[2016-06-23 Thu 10:19] =>  0:57
CLOCK: [2016-06-22 Wed 17:56]--[2016-06-22 Wed 18:30] =>  0:34
CLOCK: [2016-06-22 Wed 14:50]--[2016-06-22 Wed 16:02] =>  1:12
CLOCK: [2016-06-22 Wed 12:09]--[2016-06-22 Wed 12:57] =>  0:48
CLOCK: [2016-06-22 Wed 11:40]--[2016-06-22 Wed 11:46] =>  0:06
:END:

****** DONE Example 2: [[file:~/Box%20Sync/comp_bio/deseq2_template/https___software.rc.fas.harvard.edu_ngsdata_workshops_2015_March_DESeq2_fruitfly.pdf][Drosophila]]
CLOSED: [2016-06-27 Mon 14:45]
:PROPERTIES:
:ID:       CC047BA0-35A3-420B-BB60-CA1AA7AE275B
:CREATED:  [2021-09-13 Mon 07:29]
:END:

https://software.rc.fas.harvard.edu/ngsdata/workshops/2015_March/DESeq2_fruitfly.html

****** DONE Example 3: Hox
CLOSED: [2016-06-28 Tue 13:48]
:PROPERTIES:
:ID:       97126EA4-E1CD-4CD5-9746-A7C9393DF712
:CREATED:  [2021-09-13 Mon 07:29]
:END:

http://www.gettinggeneticsdone.com/2015/12/tutorial-rna-seq-differential.html

****** Example 4 [[file:~/Box%20Sync/comp_bio/Love,16%20Differential%20analysis%20of%20RNA%20seq%20data%20using%20the%20DESeq2%20package.pdf][Love,16 Differential analysis of RNA seq data the DESeq2 package]]
:PROPERTIES:
:ID:       61BDAC0E-3F02-482A-868E-93AC26B33361
:CREATED:  [2021-09-13 Mon 07:29]
:END:
:LOGBOOK:
CLOCK: [2016-06-22 Wed 11:15]--[2016-06-22 Wed 11:40] =>  0:25
CLOCK: [2016-06-21 Tue 16:41]--[2016-06-21 Tue 16:59] =>  0:18
CLOCK: [2016-06-21 Tue 15:33]--[2016-06-21 Tue 16:31] =>  0:58
CLOCK: [2016-06-20 Mon 13:48]--[2016-06-20 Mon 14:21] =>  0:33
CLOCK: [2016-06-20 Mon 12:21]--[2016-06-20 Mon 12:23] =>  0:02
:END:

******* Standard workflow
:PROPERTIES:
:ID:       ABBDC255-8B16-4598-AA75-6D0ADAD3E48E
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Input data
:PROPERTIES:
:ID:       71704A3E-FEE2-4363-BA67-4CA68D516375
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* SummarizedExperiment input
:PROPERTIES:
:ID:       F4CCA76E-3FBE-4E20-B6DE-69FB848D7123
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- DESeqDataSet
  - is an extension of the RangedSummarizedExperiment class
  - requires design formula
    - ~ followed by variables w/ plus sign between
    - put variable of interest at end of formula
- Preferred count algorithm is summarizedOverlaps w/ mode="Union"

********* Pre-filtering
:PROPERTIES:
:ID:       1D88E488-4AFA-4A0F-A994-3C1664B5D156
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- by DESeq2
  - on mean of normalized counts per gene- maximized genes w/ adjusted p value below a FDR of alpha
- (by user)
  - increases processing speed and reduces memory
  - b/ not necessary for DESeq2 function

******** Exploring and exporting results
:PROPERTIES:
:ID:       B87063F9-4578-440B-BF94-9DF134AF438A
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* MA-plot
:PROPERTIES:
:ID:       BBEB1ADB-F328-4B7B-807D-F8C158D60083
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- useful to look at how the DESeq2 function shrinkage performed
- after calling plotMA, the identify function can interactively select dots
- for unshrunken results, add addMLE=T to get a column called lfcMLE

********* Plot counts
:PROPERTIES:
:ID:       CB3647FE-90B2-47AD-8E89-AA206A6C207F
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- plotCounts will give counts normalized by sequencing depth for a specific gene

********* p values set to NA
:PROPERTIES:
:ID:       F28D686F-0E61-42C9-81DB-6BD581223377
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- all samples w/ zero counts
- rows w/ extreme count outlier by Cook's distance
- low mean normalized count (here only the p value will be NA)

********* Rich visualization and reporting of results- checkout Bioconductor packages [[http://bioconductor.org/packages/release/bioc/html/ReportingTools.html][ReportingTools]], regionReport, [[http://bioconductor.org/packages/release/bioc/html/Glimma.html][Glimma]], and pcaExplorer
:PROPERTIES:
:ID:       8CE69F91-1AFB-477A-996A-95212336AD6D
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******* Data transformations and visualization
:PROPERTIES:
:ID:       99C0CCAA-43A0-4E37-8C72-98617C51D928
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Count data transformations
:PROPERTIES:
:ID:       3E7192A1-175C-4D3A-A5E7-08D6CC64DCEC
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- transformations for EDA
  - pseudocounts: y=log2(N+No)
  - log2-based alternatives
    - find experiment-wide trend in variabnce/mean and remove dependnece of variance on mean
    - rlog
    - VST (variance stabilizing transformations)

********* Blind dispersion estimation
:PROPERTIES:
:ID:       BFA60D58-A2E6-49F4-9912-2CF15D0BD356
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- blind to sample information is default
- change if many/most genes will have large differences- can be checked with MA over-shrinking

******* Variations to the standard workflow
:PROPERTIES:
:ID:       7A2DB9FE-8DE5-481A-9C1A-4F26DC867582
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Contrasts
:PROPERTIES:
:ID:       A6AC8C5D-8828-4B96-BDB4-760B1F0DE0BA
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- Contrast
  - i.e. w/ levels A,B,C, test all possible differences of A vs B, A vs C, B vs C, ...
  - linear combination of estimated log2 fold changes

******** Interactions- tests whether log2 fold change attributable to one condition is different based on another condition
:PROPERTIES:
:ID:       E2EE7973-FFB1-4842-83E8-665DCCA8F2C9
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Likelihood ratio test
:PROPERTIES:
:ID:       F15FCEE5-B88F-4712-B3B1-03C576F8E44F
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- Wald test asks if log2 fold change b/w groups is zero based on mean and SE
- LR
  - asks whether full model (w/ groups) and reduced model (w/ some terms removed) are different
  - like a ANOVA except this is analysis of deviance

******** Approach to count outliers
:PROPERTIES:
:ID:       682697F8-D316-4A04-9B6D-8BCB0261DE15
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- Cook's distance- for how much single sample is influencing fitting coefficients
  - automatic in results function
  - requires at least 3 replicates
  - @ >6 samples, instead of eliminating gene, the outlier is replaced w/ the trimmed mean over all samples
  - not run when continuous variables are in the design

******** Independent filtering of results
:PROPERTIES:
:ID:       0778E394-2A3A-4952-9126-668AC8162D3A
:CREATED:  [2021-09-13 Mon 07:29]
:END:

results function maximizes p values less than * over quantiles of mean normalized counts

******** Tests of log2 fold change above or below a threshold
:PROPERTIES:
:ID:       70E0F04B-9D6B-4C11-A34E-6EE2FB008185
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Access to all calculated values
:PROPERTIES:
:ID:       60580A3B-7A73-45A0-984F-0E83E5B6F2EC
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Sample-/gene-dependent normalization factors
:PROPERTIES:
:ID:       BB95C4F2-763B-447D-A840-372588D43C79
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** “Model matrix not full rank”
:PROPERTIES:
:ID:       7F14D356-858D-46DD-9044-C1A6CD2F8A08
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* Linear combinations
:PROPERTIES:
:ID:       D9625DD4-D506-4E43-836D-9E0CDDCB9A79
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* Levels without samples
:PROPERTIES:
:ID:       E65BA44B-FBA9-4A05-93B0-1246608C3754
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******* Theory behind DESeq2
:PROPERTIES:
:ID:       C0765E52-718B-4771-9ECC-3689EE7CAA21
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** The DESeq2 model
:PROPERTIES:
:ID:       08949A97-2A7C-4D21-A9EB-811440E003A8
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Changes compared to the DESeq package
:PROPERTIES:
:ID:       41E50955-4ADF-4560-ACD0-86F58F544033
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Methods changes since the 2014 DESeq2 paper
:PROPERTIES:
:ID:       1D28C489-FF28-4112-B9B9-70C09D382608
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Count outlier detection
:PROPERTIES:
:ID:       14989EE3-4903-4FCB-A270-E3BAFFCD6EFA
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Contrasts
:PROPERTIES:
:ID:       45C7C99A-DC18-4293-9381-878DE892D84A
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Expanded model matrices
:PROPERTIES:
:ID:       FBAAB9AE-7D25-43A5-9363-766F84F32198
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Independent filtering and multiple testing
:PROPERTIES:
:ID:       EB54BD27-FD99-45DD-B671-8BBEB06857FE
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* Filtering criteria
:PROPERTIES:
:ID:       4705CA0F-C007-4503-A898-74E946304812
:CREATED:  [2021-09-13 Mon 07:29]
:END:

********* Why does it work?
:PROPERTIES:
:ID:       BA1FC087-AAF6-4116-9B86-0B0B4BBC5FDB
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******* Frequently asked questions
:PROPERTIES:
:ID:       7A2CF9E1-0C06-45FF-9637-C085CC6F15A0
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** How can I get support for DESeq2?
:PROPERTIES:
:ID:       B5ADEA5D-15C6-4449-B15F-908E63118B76
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Why are some pvalues set to NA?
:PROPERTIES:
:ID:       3D889B88-07E4-4D2C-91F1-58CCF54C9C2C
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** How can I get unfiltered DESeq results?
:PROPERTIES:
:ID:       A4107221-A74A-4E79-AC6E-057F5CAB86DF
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** How do I use the variance stabilized or rlog transformed data for differential testing?
:PROPERTIES:
:ID:       F6C4F4CE-03E9-46FC-8F9A-EF915DF7FEF2
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Can I use DESeq2 to analyze paired samples?
:PROPERTIES:
:ID:       DA9671DE-23CE-4603-9077-0F8576CEECBF
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** If I have multiple groups, should I run all together or split into pairs of groups?
:PROPERTIES:
:ID:       5384671F-9FF2-4589-80E4-7B6E7AA681FF
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Can I run DESeq2 to contrast the levels of 100 groups?
:PROPERTIES:
:ID:       355A5DC0-1A35-4315-8DEB-4C2275EA9A7C
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Can I use DESeq2 to analyze a dataset without replicates?
:PROPERTIES:
:ID:       4AF6174D-0E98-4139-89B6-2E2AB1B13BA4
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** How can I include a continuous covariate in the design formula?
:PROPERTIES:
:ID:       BBFFFE10-B8CF-4E66-9BA9-B78B0DA2FAF9
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Will the log fold change shrinkage “overshrink” large differences?
:PROPERTIES:
:ID:       01210C6E-221E-403D-AE9C-63946B83F9A4
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** I ran a likelihood ratio test, but results() only gives me one comparison
:PROPERTIES:
:ID:       76AC2A68-59B8-4E92-B384-822B8CE0DD7B
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** What are the exact steps performed by DESeq()?
:PROPERTIES:
:ID:       74C55B19-7DAA-4357-8F15-179AF2645CFF
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Is there an official Galaxy tool for DESeq2
:PROPERTIES:
:ID:       5A4CB62B-A91F-467F-AA52-B8D1C66D9D28
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** I want to benchmark DESeq2 comparing to other DE tools.
:PROPERTIES:
:ID:       9B564504-8E91-4921-87D8-3AE89F5F6E2F
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******* SessionInfo
:PROPERTIES:
:ID:       626CE66C-E3D7-4D09-BEDC-27812DB3689B
:CREATED:  [2021-09-13 Mon 07:29]
:END:

[[file:~/Box%20Sync/comp_bio/Love,14%20Beginners%20guide%20to%20using%20the%20DESeq2%20package.pdf][Love,14 Beginners guide to using the DESeq2 package]]
[[file:~/Box%20Sync/comp_bio/Love,15%20DESeq2%20User%20Guide.pdf][User Guide]]
[[https://support.bioconductor.org/local/search/page/?q=deseq2][Bioconductor Support]]
[[file:~/Box%20Sync/library/Love,16%20DESeq2%20manual.pdf][Manual]]
A negative binomial GLM
[[https://www.youtube.com/results?search_query=deseq2&page=&utm_source=opensearch][Youtube]]
[[file:~/Box%20Sync/library/Love,14%20Moderated%20estimation%20of%20fold%20change%20and%20dispersion%20for%20RNA%20seq%20data%20with%20DESeq2.pdf][Love,14 Moderated estimation of fold change and dispersion for RNA seq data with DESeq2]]
[[ftp://ftp.jax.org/dgatti/ShortCourse2015/tutorials/Differential-Expression.html][Workflow example]]
[[http://www.sthda.com/english/wiki/rna-seq-differential-expression-work-flow-using-deseq2][Another workflow example]]
[[https://dwheelerau.com/2014/02/17/how-to-use-deseq2-to-analyse-rnaseq-data/][And another workflow example]]


**** DRIMSeq                                                        :biotool:
:PROPERTIES:
:CREATED:  [2021-09-13 Mon 07:29]
:ID:       7cbebba6-1707-41ea-aad4-ca697b99b03a
:END:
https://bioconductor.org/packages/release/bioc/vignettes/DRIMSeq/inst/doc/DRIMSeq.pdf
**** edgeR                                                          :biotool:
:PROPERTIES:
:ID:       32BB77F2-0283-4A52-86AC-CDF2CB6A7315
:CREATED:  [2021-09-13 Mon 07:29]
:END:
| edgeR | diff express |   |   | [[https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf][vignette]] |   | cite:robinson2010 |
https://academic.oup.com/nar/article/40/10/4288/2411520?login=true
- GLM asking whether exons are used proportionally across experiment conditions
- DEXSeq is an extension to exons
- [[file:~/Box%20Sync/comp_bio/RNA-seq/Rueda,15%20Differential%20expression%20analysis%20using%20edgeR.pdf][Rueda,15 Differential expression analysis using edgeR]]
- [[file:~/Box%20Sync/comp_bio/RNA-seq/edgeRUsersGuide.pdf][User Guide]]
- [[file:~/Box%20Sync/comp_bio/RNA-seq/edgeR_manual.pdf][Manual]]
- cite:robinson2010
- [[file:~/Box%20Sync/library/Robinson,09%20edgeR%20a%20Bioconductor%20package%20for%20differential%20expression%20analysis%20of%20digitial%20gene%20expression%20data.pdf][Robinson,09 edgeR a Bioconductor package for differential expression analysis of digitial gene expression data]]


**** epicseq :biotools:
https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFgtGDXpvjWgzHdQjhXHVlSvf
**** exceRpt                                                        :biotool:
:PROPERTIES:
:ID:       1db0ffb6-25b7-4c97-9aa0-263b8c96373b
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- [[https://www.scopus.com/results/citedbyresults.uri?sort=cp-f&cite=2-s2.0-85063257542&src=s&nlo=&nlr=&nls=&imp=t&sid=f566de7056a65b814af39a1c988abbfb&sot=cite&sdt=a&sl=0&origin=resultslist&offset=21&txGid=1f850b1319595ca2e1efdc578c6f60bb&isValidNewDocSearchRedirection=false][scopus citing exceRpt]]
- cite:rozowsky2019
- http://github.gersteinlab.org/exceRpt/
- used in
  - cite:godoy2019
  - cite:giraldez2018
**** faSplit
https://www.biostars.org/p/13270/
**** fastp
| fastp | [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]] | fastq | [[https://github.com/OpenGene/fastp][repo]] |   | [[https://watermark.silverchair.com/bty560.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAuEwggLdBgkqhkiG9w0BBwagggLOMIICygIBADCCAsMGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMUQnq5IYxgkrFnt0WAgEQgIIClChzxtpO28QYuBn1woUVAMnQOwkPRZWSY20cwnzbVuF3DlOcpwjeNjNrO_ixNp6csW5pOdBZ81CHjV65BMcqsZmP9KOT-yVGy4r_7mWKLuzm7Siid24pM9-xF3k7y1RoBK4-PrnZgzp-J1svrUvOZAphbBPmsezuOowSuj9o9EZyTciDJbUQ2klUbYSRDO_InWNOVu23eOzGpJQwer81BL2iS8TsCfeWF-eZtzNBRuZZr1UZ2PhX_mVQGu4bHvKn1Bmoe64CFYkEJGLoka87zVVfo--J_hAledjiW1_-wEj8kZSyb2diZoP9ABDGOJA6aQ7fzx8qNGgNAb_oK-uAmEdfAfepI_0LECDdlk9EIxIJITC-Z59gk_vNVophJOydBimGZHFK0iACl-N-nZNALjIqRXeRmqOxcEdRFmNy6onzLalfCV0wyqFae42JIVzEYFSLW1oQVC0ugMfKTTmOfysRZ8-4bboBo1UbskpGVlKvrSBTFuDyvuBlPoUTECbOrO6bsdeNk8yfTMTeMQzNzOzZYZ1mQVCULZtuE_YDCuKmwukT3EeUy8iOJbp7pD2Ew5kphJ9q0sV-_io3h0jBZEdJmUr6dTPSRcv56pLP6gtqso6ZdB9bWk3sFNt--tBwIGhQvnLXgyV7xGqRYNPrDeMbs2tPLlK-v4D8XKj-BHGi1sOVuzL1rCvFve0knpL4btsqw_BDp8P2XuFDyp0OOXtmE3dKJktLR0QyQZ3u4-aKksDMfkqrglbPoBjEhInpdpfiDQbEDTFO9qiaL9iajdsV8T636x_KHXpdGkFqO3GpcFyJv1QJyPW2qhAXuneiqWx6t60pJ8f0IyAzRV9BIUoQ4B3soV9TXIuSP171Tf0JYmcY5Q][paper]] | cite:chen2018b |

**** fastqc                                                         :biotool:
:PROPERTIES:
:CREATED:  [2021-09-13 Mon 07:29]
:END:
| [[id:0a319950-3215-4846-a020-a08b78f3323d][FastQC]] | [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]] |   |   | [[https://www.bioinformatics.babraham.ac.uk/projects/fastqc/INSTALL.txt][site]] |   |   |
- [[https://www.youtube.com/watch?v=bz93ReOv87Y][Intro video]]
- Only non-truncated fastqs can be processed by fastqc.
- Interpretation
  - Per sequence GC content- spikes mean contaminants
  - will only track first 200000
  - base quality
    - Phred vs. read position (goal is >20 or 1 error in 100 bases)
    - Phred histogram- look for single population or filter second population
    - Base position vs. frequency of nucleotide
  - per nucleotide
    - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/5%20Per%20Sequence%20GC%20Content.html][Per sequence GC content]]
    - Secondary peak = contamination- see overrepresent sequences
  - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html][Duplicate sequences]]
    - Broad oversequencing -> high flat line
    - Specific enrichments - spikes
      - Possible reasons
        - Technical (PCR) duplication
        - Biological duplication- common in RNA-seq where you are re-sequencing the common sequences
    - Warning at 20%, failure @ 50%
  - [[http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html][Kmer content]]
    - for partial sequences in a variety of longer sequences
  - https://sequencing.qcfail.com/software/fastqc/
  - https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/#:~:text=The%20output%20from%20FastQC%2C%20after,or%20%E2%80%9CFail%E2%80%9D%20is%20assigned.
- cite:williams2014 for a FastQC aggregator Perl script
- [[https://www.biostars.org/p/56088/#56102][Fastq Quality Read And Score Length Check]]
-
**** fastx
| [[http://hannonlab.cshl.edu/fastx_toolkit/commandline.html][FASTX-Toolkit]] |   |   |   |   |   |   |
**** featureCounts
| featureCounts |   |   |   |   | [[https://watermark.silverchair.com/btt656.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAApgwggKUBgkqhkiG9w0BBwagggKFMIICgQIBADCCAnoGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMofurkoV2NWsQqlelAgEQgIICS7iktDnfv1X3fe1vdQQcXCS09zT549UGvBjDCLHoNebBGfiQi_Eo7ZjUHVH_Yay_p1DlFd4eDeXU4WEX6HTQ-MzPqmtZ6RMGegyEptsqxaLnvr3QfpGO6WRT01dViqp6xya9-CqCRGV3E7F37VruIYzZEFjr-2svIgoRydPyF4pkSzA2hkvwV_g-JT2YSnqboZ5DOGtwZXeMaLKf48_aEpuYemRUx8VqOfM8xnOm4_jumLGd_BBqCCv6brLX8z3WVJpWfJ4YfpJXhemFRt8mSvIsAe9pYXV52W7-psxHlM2ocXzdD_7vLSD7iy1NocDXWLFT5HRH_Cq8NyiRq1s4Mq-QYuvQJFlTxWArVC6vdpE40bDr9slGD6KTJ0DpUQeC52q-aoKJXr9Y7Pi2SDo881eFuJxTWuSuMjxcXStn9MkPKZfcKa10Vu4CdJSSnvJAKAvd4GsshsfCVGCvkgy4ZfEabo5nOZaB82UYC-8AZEpdbD3RtJg1FUQJHDz0tlY-2gZYf2VB3KU1JK4wZcHzn8T8R5BSChB6MjFrRdTZeo3M8-ZIK3eWQGH2vLj8_bPsYlBF-wfj2JzSG3zJGgUUa_rSBzoK8Vvw8k_BPxCq63h6u-k49l-X0o-MO6f8mOdm_YliTT0dWu_ujvlbXMw9AIBrKjwBJncq27EwMOX0shm8nnzPMoPTfOzoekZR2nYTDPayGjHhqkAs6haFAQrI8XAAd0PfbMtW4l54ArULWEH6hCVxzDX8_h8F8oWgOc_3h2ktpbTLNsiX6Bxj][paper]] |   |
**** fgsea :biotool:
- https://www.reddit.com/r/bioinformatics/comments/1703dz0/gsea_are_authors_just_cherry_picking_enriched/
- https://www.sciencedirect.com/science/article/pii/S1874939919302287?via%3Dihub#s0110
- scoreType
  - https://github.com/ctlab/fgsea/issues/87
- ranking metric
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1674-0#Sec8
- time course data
  - https://bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/geseca-tutorial.html#analysis-of-time-course-data
- documentation
  - tutorial https://bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html
- Gene set enrichment analysis :: coexpression method that determines enrichment of gene sets by the set's genes ranked differential expression positions
- Historical development
  - Developed in cite:mootha2003
  - Proposed in cite:subramanian2005
- Outputs
  - ES (enrichment score)- maximum deviantion from zero in random walk of running sums of genes in a list. Corresponds to Kolmogorov-Smirnov
  - p-value of ES- versus null of permutated gene labels
  - Leading edge analysis :: "A subset of genes found in the ranking at or just before the maximal ES in a GSEA analysis. This subset of genes often accounts for a pathway being defined as enriched." cite:reimand2019
- Used in
  - [[id:91d46e60-eec3-48c7-97f1-b53eb1559b3f][Used in Co-expression / pathway enrichment process]]
  - Used in [[id:8e4c1fb8-cd52-4fac-900a-4b6e6eb5a28a][clusterProfiler]]
- fgsea
  - https://bioconductor.org/packages/devel/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html
- Ranking metric
  - -log10(p-value)
    - https://bioinformatics.stackexchange.com/questions/13443/ranking-metric-in-gsea
  - sign(logFC) * -log10(pval)
    - https://support.bioconductor.org/p/85681/
- [[https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideTEXT.htm#_Consistent_Feature_Identifiers_Acro][documentation]]
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/FAQ#Why_does_GSEA_use_a_false_discovery_rate_.28FDR.29_of_0.25_rather_than_the_more_classic_0.05.3F][FDR discussion]]
  - FAQ
    - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/FAQ#What_array_platforms_and_organism_species_does_GSEA_support.3F][using for mouse data]]
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Using_RNA-seq_Datasets_with_GSEA][RNA-seq data]]
    - with RNA-seq, use TMM
  - [[http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats][file formats]]
    - GMT: gene matrix transposed file (*.gmt)
    - chip files ftp://ftp.broadinstitute.org/pub/gsea/annotations (use ENSEMBL)
  - [[https://www.gsea-msigdb.org/gsea/msigdb/search.jsp][gene sets]]
- [[id:f7eed3be-f505-451c-b191-1a58373fb180][Co-expression gene set databases]]
  - [[id:c79bfe84-0f15-474e-8a22-1f210605ed5a][MSigDB]]
- file creation and formatting
  - see to create .rnk file https://github.com/BaderLab/EM-tutorials-docker/blob/master/R_scripts/supplemental_protocol2_rnaseq.R
- [[http://software.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection/4][ssGSEA]] calculates enrichment for each sample-gene set pair
- https://academic.oup.com/bib/article/17/3/393/1744776

**** flexbar                                                        :biotool:
:PROPERTIES:
:doc: https://github.com/seqan/flexbar/wiki/Manual
:ID:       5c20daf7-d02e-4447-bd1e-f991e1366c48
:END:
| flexbar |   |   |   | [[https://github.com/seqan/flexbar/wiki/Manual][doc]] |   | cite:roehr2017 |

- [[https://github.com/seqan/flexbar/wiki/Manual][Documentation]]
- https://www.google.com/search?q=flexbar+sequencing+manual
**** gage                                                           :biotool:
:PROPERTIES:
:ID:       9b222d1c-ab61-450c-8b77-a98ec3213867
:CREATED:  [2021-09-13 Mon 07:29]
:END:
#+begin_src R
#########1#########2#########3#########4#########5#########6#########7#########8
#
library(gage)
library(gageData)

data(kegg.sets.rn)
data(go.sets.rn)
data(sigmet.idx.rn)
data(go.subs.rn)

# Subset gene set data to biological processes (BP) and molecular functions, without disease states
kg_rat = kegg.sets.rn[sigmet.idx.rn]
go_rat = do.call(c, list(go.sets.rn[go.subs.rn$MF],go.sets.rn[go.subs.rn$BP]))

rat = useMart("ensembl", dataset = "rnorvegicus_gene_ensembl")
rat_entrez =
  getBM(attributes = c("ensembl_gene_id", "entrezgene_id"),
        values = rownames(res_BN3_IW_sham),
        mart = rat)

#########1#########2#########3#########4#########5#########6#########7#########8
#
# Functions
##
##
make_gage_object = function(deseq_res, biomat){
  df = as.data.frame(deseq_res) %>%
    rownames_to_column(var = "ensembl_gene_id") %>%
    left_join(biomat, by = "ensembl_gene_id") %>%
    filter(!is.na(entrezgene_id))
  df2 = df %>% pull(log2FoldChange)
  names(df2) = df$entrezgene_id
  return(df2)
  }

run_gage = function(gage_object){
  go = gage(gage_object,
            gsets = go_rat,
            ref = NULL,
            sample = NULL,
            same.dir = TRUE,
            rank.test = TRUE)
  go_up = as.data.frame(go$greater)
  go_down = as.data.frame(go$less)
  #
  kegg = gage(gage_object,
              gsets = kg_rat,
              ref = NULL,
              sample = NULL,
              same.dir = FALSE,
              rank.test = TRUE)
  kegg = as.data.frame(kegg$greater)
  #
  gage_all = list(go_up, go_down, kegg)
  return(gage_all)
}

#
deseq_res_ch3_enrich_list = list(res_BN3_tx_sham, res_SS_tx_sham, res_BN3_SS_sham, res_BN3_SS_tx)
gage_objects = lapply(deseq_res_ch3_enrich_list, make_gage_object, biomat = rat_entrez)
gage_results = lapply(gage_objects, run_gage)
names(gage_results) = c("gage_BN3_tx_sham", "gage_SS_tx_sham", "gage_BN3_SS_sham", "gage_BN3_SS_tx")

save(gage_results,
     file = "./data/gage_results.RData")
#+end_src
| GAGE | enrichment | express |   |   | [[https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-161][paper]] |   |
- https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
- https://genviz.org/module-04-expression/0004/03/01/pathwayAnalysis/
- https://www.bioconductor.org/packages/devel/data/experiment/manuals/gageData/man/gageData.pdf
- https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf
- cite:luo2009
**** GATK
         |   |   |   | [[https://software.broadinstitute.org/gatk/documentation/][doc]] |   |   |
**** Genomic Alignments
:PROPERTIES:
:ID:       608b8cb2-4c77-4a9b-bc6a-24470bb572b0
:END:
- see [[id:78b22803-f843-41f2-9390-d621caae82c0][Genomic Ranges]]
- summarizeOverlaps https://bioconductor.org/packages/release/bioc/vignettes/GenomicAlignments/inst/doc/summarizeOverlaps.pdf
**** Genomic Ranges                                                 :biotool:
:PROPERTIES:
:CREATED:  [2021-09-13 Mon 07:29]
:ID:       78b22803-f843-41f2-9390-d621caae82c0
:END:
- https://bioconductor.github.io/BiocWorkshops/solving-common-bioinformatic-challenges-using-genomicranges.html
- https://rstudio-pubs-static.s3.amazonaws.com/142838_0374b144401c449eb17ed5c2c70cb513.html#grangeslist-multiple-interval-range-features
- see [[id:608b8cb2-4c77-4a9b-bc6a-24470bb572b0][Genomic Alignments]]
- https://seandavi.github.io/ITR/RangesAndSignal.html
- https://rdrr.io/bioc/GenomicRanges/
- https://kasperdanielhansen.github.io/genbioconductor/html/GenomicRanges_Lists.html
- https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html
- https://www.google.com/search?q=convert+bed+to+granges
- Granges manipulations https://rockefelleruniversity.github.io/Bioconductor_Introduction/exercises/answers/GenomicFeatures_answers.html

**** giggle
:PROPERTIES:
:ID:       a4e2adcd-f921-4aef-a808-866147b6db8a
:END:
- https://github.com/ryanlayer/giggle
- https://www.nature.com/articles/nmeth.4556
-
**** gistic2                                                        :biotool:
:PROPERTIES:
:doc:  [[http://portals.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&paper_id=216&p=t][site]]
:ID:       4f0b84ab-1dec-45c9-a7d4-bdce41e13150
:CREATED:  [2021-09-13 Mon 07:29]
:END:
https://www.genepattern.org/file-formats-guide

segmentation file column 6 is log2(x)-1 of the copy number, so
| interp  | copy number |   log2(x)-1 |
|---------+-------------+-------------|
| neutral |           2 |           0 |
| gain    |           3 |        0.58 |
| amp     |           4 |           1 |
| loss    |           1 |          -1 |
| del     |           0 | - inf -> NA |
|         |             |             |

- https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#gistic-2-0-data
- https://blindwith.science/2018/10/442.html/
- http://crazyhottommy.blogspot.com/2017/11/run-gistic2-with-sequenza-segmentation.html
**** GSEA                                                           :biotool:
:PROPERTIES:
:ID:       D68E1182-1F7B-4160-A6DD-BD2635F8085F
:CREATED:  [2021-09-13 Mon 07:29]
:END:
-
- see [[*fgsea][fgsea]]
**** Gviz
https://rdrr.io/bioc/Gviz/man/DataTrack-class.html
https://bioconductor.org/packages/release/bioc/vignettes/Gviz/inst/doc/Gviz.html#7_Composite_plots_for_multiple_chromosomes
**** histat2
:PROPERTIES:
:doc: |         |              | fastq   |      | [[https://ccb.jhu.edu/software/hisat2/manual.html][doc]]      |       |                        |
:END:

**** homer :biotool:
:PROPERTIES:
:ID:       33a4b4a6-f22f-4e12-8c27-7170bb1e1a9e
:END:
- Reference
  - [[https://www.biostars.org/tag/homer/?order=views][homer on Biostars]]
- Processes
  - Motifs in genomic regions
    - http://homer.ucsd.edu/homer/ngs/peakMotifs.html
    - https://www.biostars.org/p/138407/
  - Results interpretation
    - http://homer.ucsd.edu/homer/motif/index.html
  - Gene-based promoter motif enrichment
    - http://homer.ucsd.edu/homer/microarray/index.html
**** ichorcna                                                       :biotool:
:PROPERTIES:
:use: genome-wide copy number alteration detection
:substrate: bam
:repo: [[https://github.com/broadinstitute/ichorCNA][repo]]
:doc:  [[https://github.com/broadinstitute/ichorCNA/wiki][doc]]
:paper: | [[https://www.nature.com/articles/s41467-017-00965-y][paper]] | cite:adalsteinsson2017 |
:ID:       8c535c38-9310-4fd0-a8fe-c300639493da
:CREATED:  [2021-09-13 Mon 07:29]
:END:
"Briefly, the genome was divided into T non-overlapping windows, or bins, of 1 Mb. Aligned reads were counted based on overlap within each bin using the tools in HMMcopy Suite (http://compbio.bccrc.ca/software/hmmcopy/). The read counts were then normalized to correct for GC content and mappability biases, and then CNAs and tumor fraction were estimated using HMMcopy R package18. "


- https://github.com/gavinha/TitanCNA/issues/86
  - [X] does hmmcopy need blacklist? - no mappability score part of hmmcopy The mappability file, which was generated using ENCODE Duke Uniqueness of 35bp sequences track from UCSC (http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability), provides scores indicating how mappable the reference sequence at each bin is to aligners; the higher the score, the more mappable the sequence.

- https://www.google.com/search?q=viterbi+algorithm&oq=Viterbi+algorithm&aqs=chrome.0.0l8.356j0j7&client=ubuntu&sourceid=chrome&ie=UTF-8

https://www.google.com/search?q=ichorcna&sxsrf=ALeKk02zeu_rSdEZVMYYCyip37Gt2XFzEg:1617895629440&source=lnms&tbm=vid&sa=X&ved=2ahUKEwjZx5ry-u7vAhVJbs0KHXyIAGkQ_AUoAnoECAEQBA&biw=1596&bih=789
- cite:guo2018
- cite:klega2018
- cite:manier2018
- cite:mouliere2018
- cite:stover2018
- cite:adalsteinsson2017
- [[https://github.com/broadinstitute/ichorCNA/wiki][documentation]]
  - https://github.com/broadinstitute/ichorCNA/wiki/Parameter-tuning-and-settings
- [[https://mail.google.com/mail/u/0/#inbox/FMfcgxwGDDjrSfkvnNDnxMlVJNFgfnfk][email from:Sundby re:ichor method]]
- https://www.biostars.org/local/search/page/?q=ichorcna
- for install see [[file:/ssh:chaudhuri-roche:/home/jszymanski/repos/biotools/Dockerfile][biotools Dockerfile]]
**** INTEGRATE                                                      :biotool:
:PROPERTIES:
:use: detect rna fusions
:substrate: bam
:paper:  cite:zhang2014a        |
:ID:       7519d47f-18e8-4a28-be11-9d081b68b52d
:CREATED:  [2021-09-13 Mon 07:29]
:END:
  - cite:zhang2016
    - [[http://apps.webofknowledge.com/summary.do?product=WOS&parentProduct=WOS&search_mode=CitingArticles&qid=2&SID=6ERcSCYD7iSBCZMtAM7&&page=1&action=changePageSize&pageSize=50][WOS citing]]- no cfRNA yet
  - integrate
    - cite:zhang2017c
    - cite:zhang2016
    - cite:zhang2016a
    - cite:zhang2016a
    - - cite:zhang2014a
    - https://github.com/ChrisMaherLab/INTEGRATE-Vis/blob/master/README.md
    for integrate, see - cite:li2020
**** liftover :biotool:
https://genome.ucsc.edu/cgi-bin/hgLiftOver
**** limma                                                          :biotool:
:PROPERTIES:
:ID:       3283b8b2-fc42-4a71-a8e9-7dbdd7ab62ce
:CREATED:  [2021-09-13 Mon 07:29]
:use: differential expression
:paper:  | count   |      |          | [[https://dx.doi.org/10.1093%2Fnar%2Fgkv007][paper]] |                        |
:END:
- https://www.coursera.org/lecture/bioconductor/limma-aoNwW
- [[file:~/Box%20Sync/comp_bio/RNA-seq/Ritchie,15%20limma%20powers%20differential%20expression%20analysis%20for%20RNA%20sequencing%20and%20microarray%20results.pdf][Ritchie,15 limma powers differential expression analysis for RNA sequencing and microarray results]]
- raw counts transformed to continuous values w/ confidence weight
- [[file:~/Box%20Sync/comp_bio/RNA-seq/limma%20Manual.pdf][manual]]
- Math, concept- [[file:~/Box%20Sync/comp_bio/RNA-seq/Law,14%20voom%20precision%20weights%20unlock%20linear%20model%20analysis%20tools%20for%20RNA%20seq.pdf][Law,14 voom precision weights unlock linear model analysis tools for RNA seq]]
- voom transformation- from counts to log-counts per million w/ associated weight (just think of it as if microarray data)
***** [[file:~/Box%20Sync/comp_bio/RNA-seq/limma%20User%20Guide.pdf][User Guide]]
:PROPERTIES:
:ID:       15C1AE68-19D2-4FBD-B51E-DA37C605F0D7
:CREATED:  [2021-09-13 Mon 07:29]
:END:

- concepts
  - borrows information across genes for emperical Bayes
  - read counts -> continuous expression data like microarray
  - required matrices for linear models approach
    - design- which RNA to which sample (i.e. specify groups)
    - contrast- which group comparions to make

***** [[file:~/Box%20Sync/comp_bio/limma_voom_template/][Examples and Template]]
:PROPERTIES:
:ID:       D1B28C0A-9AAD-4245-BE85-61300CE4A02B
:CREATED:  [2021-09-13 Mon 07:29]
:END:

****** WEHI example ([[file:~/Box%20Sync/comp_bio/RNA-seq/wehi_limma_example/wehi_limma_example.Rproj][Rproject]], [[file:~/Box%20Sync/comp_bio/RNA-seq/wehi_limma_example/wehi_limma_example.R][Rscript]]), [[http://bioinf.wehi.edu.au/RNAseqCaseStudy/][link]]
:PROPERTIES:
:ID:       1E96E985-8BD2-4D8F-AC88-FB509B99F513
:CREATED:  [2021-09-13 Mon 07:29]
:END:
:LOGBOOK:
CLOCK: [2016-05-10 Tue 15:46]--[2016-05-10 Tue 16:05] =>  0:19
CLOCK: [2016-05-10 Tue 15:07]--[2016-05-10 Tue 15:20] =>  0:13
CLOCK: [2016-05-09 Mon 12:25]--[2016-05-09 Mon 13:13] =>  0:48
:END:

****** User Guide Chapter 18
:PROPERTIES:
:ID:       817998EF-31AD-4A22-BF91-ACF72A1C29E5
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******* 18.1 [[file:~/Box%20Sync/comp_bio/limma_voom_template/yoruba_hapmap/][Profiles of Yoruba HapMap]]
:PROPERTIES:
:ID:       1E8CAB25-19C2-4745-B157-FA46D84549CD
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** DONE Read
CLOSED: [2016-06-30 Thu 11:22]
:PROPERTIES:
:ID:       91CBDFB8-17EB-4F2D-B2A4-58F656C64332
:CREATED:  [2021-09-13 Mon 07:29]
:END:

******** Work through code
:PROPERTIES:
:ID:       374ef56d-7a1a-4c40-ade8-f4acafd37602
:END:

:PROPERTIES:
:ID:       05EC5C9D-CE60-4DD5-9F5D-6EED8C229478
:CREATED:  [2021-09-13 Mon 07:29]
:END:


Venn diagrams with limma]]

Another use of the voom plot-  what you are failing to do completely in the voom analysis is to remove the rows of your count matrix that are lowly expressed, because these hose the mean/variance estimation, which you can see/diagnose on the plot generated with you call to voom(..., plot=TRUE). In brief, you want to ensure that the red fit line doesn't "slope/hook back down" (I'm talking as if you are "reading" the graph from right to left here) due to the drop/discretization in the observed variance from the black dots on the left side of the plot.

[[https://www.biostars.org/p/131488/][subsetting from topTable]]
**** MACS2                                                          :biotool:
:PROPERTIES:
:CREATED:  [2021-09-13 Mon 07:29]
:ID:       75ca7513-8f01-4819-a813-f9459f447483
:END:
- google
- macs2 shift - https://www.biostars.org/p/207318/
- https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md
- https://hbctraining.github.io/Intro-to-ChIPseq/lessons/05_peak_calling_macs.html
- https://ucdavis-bioinformatics-training.github.io/2020-Epigenetics_Workshop/data_reduction/chip_atac/04-peaks

a _peaks.xls file looks like
# This file is generated by MACS version 2.2.6
# Command line: callpeak --treatment /mnt/ris/jschwarz/cardiac-radiobiology/bam/ct01_aut_blk_ddp_open.bam --format BAMPE --gsize mm --keep-dup all --outdir /mnt/ris/jschwarz/cardiac-radiobiology/macs2 --name ct01_aut_blk_ddp_open.bam --bdg --SPMR --extsize 150 --nomodel --call-summits --nolambda
# ARGUMENTS LIST:
# name = ct01_aut_blk_ddp_open.bam
# format = BAMPE
# ChIP-seq file = ['/mnt/ris/jschwarz/cardiac-radiobiology/bam/ct01_aut_blk_ddp_open.bam']
# control file = None
# effective genome size = 1.87e+09
# band width = 300
# model fold = [5
# qvalue cutoff = 5.00e-02
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 10000 bps
# Broad region calling is off
# Paired-End mode is on
# Searching for subpeak summits is on
# MACS will save fragment pileup signal per million reads

# fragment size is determined as 90 bps
# total fragments in treatment: 5942028
# d = 90
# local lambda is disabled!
chr start end length abs_summit pileup -log10(pvalue) fold_enrichment -log10(qvalue) name
chr1 3120443 3120560 118 3120499 6.00 7.58942 5.43209 5.03290 ct01_aut_blk_ddp_open.bam_peak_1
- https://www.google.com/search?q=macs2
- https://hbctraining.github.io/Intro-to-ChIPseq/lessons/05_peak_calling_macs.html
***** Published MACS2 commands for ATAC-seq
:PROPERTIES:
:ID:       496e62cd-3138-4120-a499-a10246ce77b5
:END:
- cite:corces2017 macs2 callpeak --nomodel --nolambda --keep-dup all --call-summits
- cite:corces2018 macs2 callpeak --shift -75--extsize 150 --nomodel --call-summits --nolambda --keep-dup all -p 0.01
- cite:hendrickson2017 macs2 callpeak --f BAMPE --B --SPMR
- cite:scott-browne2016 macs2 callpeak --nomodel -q 0.01 --keep-dup all --call-summits
**** msigdbr
:PROPERTIES:
:ID:       edf39ef6-df91-40c2-a71a-c612d60b5582
:END:
[[id:c79bfe84-0f15-474e-8a22-1f210605ed5a][MSigDB]]
https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html

**** multiqc                                                        :biotool:
:PROPERTIES:
:ID:       066d50fc-5407-4c80-a6ca-2534f37ae043
:CREATED:  [2021-09-13 Mon 07:29]
:use: sequencing QC
:repo: [[https://github.com/ewels/MultiQC][github]]
:doc: | [[https://multiqc.info/docs/][doc]]      |       |                        |
:END:
https://cran.r-project.org/web/packages/TidyMultiqc/index.html
**** NCBI Genome Workbench
https://www.ncbi.nlm.nih.gov/tools/gbench/
**** NCBI Genome Workbench
https://www.ncbi.nlm.nih.gov/tools/gbench/
**** NDEx :biotool:
:PROPERTIES:
:ID:       784895e5-0c94-4458-9cfb-f52759718ab3
:END:
- [[id:d84cf649-32f4-46e6-b041-4a8a76b31ffc][Pathway encoding]]
- https://www.ndexbio.org/#/
- https://www.ndexbio.org/iquery/
**** nextflow                                                       :biotool:
:PROPERTIES:
:CREATED:  [2019-12-27 Fri 14:48]
:ID:       6116d262-9ca8-430b-b385-65e1a7fc0a70
:END:
:LOGBOOK:
CLOCK: [2020-01-30 Thu 13:00]--[2020-01-30 Thu 14:04] =>  1:04
CLOCK: [2020-01-30 Thu 09:35]--[2020-01-30 Thu 12:45] =>  3:10
CLOCK: [2020-01-21 Tue 12:45]--[2020-01-21 Tue 14:30] =>  1:45
CLOCK: [2020-01-15 Wed 14:47]--[2020-01-15 Wed 15:39] =>  0:52
CLOCK: [2020-01-14 Tue 18:38]--[2020-01-14 Tue 19:04] =>  0:26
CLOCK: [2020-01-14 Tue 16:24]--[2020-01-14 Tue 16:44] =>  0:20
CLOCK: [2020-01-14 Tue 13:49]--[2020-01-14 Tue 15:52] =>  2:03
CLOCK: [2020-01-14 Tue 11:11]--[2020-01-14 Tue 12:42] =>  1:31
CLOCK: [2020-01-13 Mon 21:10]--[2020-01-13 Mon 22:00] =>  0:50
CLOCK: [2020-01-10 Fri 17:51]--[2020-01-10 Fri 18:51] =>  1:00
CLOCK: [2020-01-08 Wed 13:06]--[2020-01-08 Wed 14:12] =>  1:06
CLOCK: [2020-01-08 Wed 11:19]--[2020-01-08 Wed 12:31] =>  1:12
CLOCK: [2020-01-07 Tue 17:27]--[2020-01-07 Tue 18:27] =>  1:00
CLOCK: [2020-01-02 Thu 13:32]--[2020-01-02 Thu 14:32] =>  1:00
CLOCK: [2019-12-27 Fri 19:15]--[2019-12-27 Fri 21:20] =>  2:05
CLOCK: [2019-12-27 Fri 14:50]--[2019-12-27 Fri 15:12] =>  0:22
:END:
https://github.com/NYU-Molecular-Pathology
https://www.pnas.org/content/115/4/E743.long
- github searhing e.g. https://github.com/search?utf8=%E2%9C%93&q=followLinks++extension%3A.nf&type=Code&ref=advsearch&l=&l=
- see for inputs from csv- https://github.com/nf-core/atacseq/blob/master/main.nf
- raw data should reside in protected dirs
- https://www.reddit.com/search?q=nextflow
- cite federico2019
- useful tutorials
  - https://nextflow-io.github.io/hack17-varcall/#_input_parameters associated github at radonc-cancerbio:~/data/tmp
- https://sciwiki.fredhutch.org/compdemos/nextflow/
- https://learn.gencore.bio.nyu.edu/nextflow/
- https://www.ebi.ac.uk/goldman-srv/simNGS/runfiles5/HiSeq/
- https://github.com/genome/gms/wiki/HCC1395-WGS-Exome-RNA-Seq-Data
- http://nextflow-io.github.io/patterns/index.html#_process_per_file_pairs
- [[https://stackoverflow.com/questions/tagged/nextflow][nextflow stack questions]]
- https://github.com/nextflow-io/patterns
- [[https://github.com/nextflow-io/nf-hack18][2018 workshop github]]
- [[https://nextflow-io.github.io/nf-hack18/training.html][2018 workshop course]]
- [[https://github.com/search?utf8=%25E2%259C%2593&q=language%253Anextflow+&type=][github repos using nextflow]]
- https://github.com/cbcrg/piper-nf/blob/master/piper.nf
- [[https://github.com/search?utf8=%25E2%259C%2593&q=language%253Anextflow+&type=][github nextflow repositories]]
- https://groups.google.com/forum/m/#!forum/nextflow
- https://www.nextflow.io/presentations.html
- http://groovy-lang.org/single-page-documentation.html
- https://dzone.com/refcardz/groovy?chapter=10
- https://www.manning.com/books/groovy-in-action-second-edition
- https://github.com/MaxUlysse/compile-latex
- https://nextflow-io.github.io/nf-hack18/training.html
- https://github.com/nextflow-io/patterns
- https://nextflow-io.github.io/hack17-varcall/
- [[https://medium.com/@alaincoletta/how-to-create-a-nextflow-pipeline-with-all-necessary-tools-for-a-series-of-repetitive-35cf5b3b19a8][Coletta tutorial]]
- processes
  - check if file exists
- commands
  |----------+---------|
  | function | command |
  |----------+---------|
  | <35>     | <35>    |
  |          |         |
- standards and best practice
- [[https://www.nextflow.io/docs/latest/index.html][documentation]]
- applications
- reference
  - [[file:~/.emacs.d/snippets/org-mode/software]]
**** oligocalc
:PROPERTIES:
:doc:  [[http://www.basic.northwestern.edu/biotools/oligocalc.html][site]]
:END:
**** pathvisio                                                      :biotool:
:PROPERTIES:
:ID:       17c865dc-765c-496d-b5c7-60d3d3439de3
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004085
- [[id:d84cf649-32f4-46e6-b041-4a8a76b31ffc][Pathway encoding]]
- https://github.com/PathVisio/GPML
- https://github.com/PathVisio
- cite:wikipathways
- used in [[id:69bfa728-e149-4b8a-9863-347630f86bc4][wikipathways]]
- https://pathvisio.github.io/
- to switch to java 8 https://askubuntu.com/questions/1133216/downgrading-java-11-to-java-8
**** picard                                                         :biotool:
:PROPERTIES:
:ID:       9e48626d-47e1-477a-8b79-eff41f7974b8
:CREATED:  [2021-09-13 Mon 07:29]
:doc:  [[https://broadinstitute.github.io/picard/index.html][doc]]
:END:
**** polyester                                                      :biotool:
:PROPERTIES:
:ID:       065725b4-43d3-48d4-a4e2-3ca5c1e1e6fd
:CREATED:  [2021-09-13 Mon 07:29]
:substrate: fasta,gtf
:use: rna-seq simulation
:doc:  [[http://bioconductor.org/packages/release/bioc/vignettes/polyester/inst/doc/polyester.html][vignette]]
:END:
  - https://www.bioconductor.org/packages/release/bioc/vignettes/polyester/inst/doc/polyester.html

**** preseq :biotool:
:PROPERTIES:
:ID:       6d8cbcc0-2651-4c51-991e-639dbc81ea66
:END:
- http://smithlabresearch.org/manuals/preseqmanual.pdf
- library complexity- preseqr http://smithlabresearch.org/manuals/preseqmanual.pdf

**** rGREAT
https://www.bioconductor.org/packages/release/bioc/vignettes/rGREAT/inst/doc/local-GREAT.html
**** rnaseqc
:PROPERTIES:
:use: rna-seq qc
:substrate: bam
:repo:  [[https://github.com/broadinstitute/rnaseqc][repo]]
:doc: [[https://github.com/broadinstitute/rnaseqc][repo]]
:END:
**** rsem                                                           :biotool:
:PROPERTIES:
:ID:       85db5c34-b5fd-4b6c-bdb7-88263e553ba3
:CREATED:  [2021-09-13 Mon 07:29]
:paper:  cite:li2011
:use: transcript quantification
:substrate: bam
:repo: [[https://github.com/deweylab/RSEM][repo]]
:paper: |          | [[https://doi.org/10.1186/1471-2105-12-323][paper]] | cite:li2011            |
:END:
-
- https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem
- documentation
  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323
  - https://github.com/ENCODE-DCC/long-rna-seq-pipeline/blob/master/DAC/STAR_RSEM.sh
  - http://deweylab.biostat.wisc.edu/rsem/rsem-calculate-expression.html
  - http://deweylab.biostat.wisc.edu/rsem/rsem-prepare-reference.html
  - https://deweylab.github.io/RSEM/rsem-calculate-expression.html
  - https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem
**** RSeQC
:PROPERTIES:
:use: rna-seq qc
:substrate: bam
:doc:  [[http://rseqc.sourceforge.net/][repo]]
:paper:  [[https://doi.org/10.1093/bioinformatics/bts356][paper]] | cite:wang2012
:END:
**** salmon                                                         :biotool:
:PROPERTIES:
:substrate: fastq
:use: RNA-seq pseudoalignment
:repo:  [[https://github.com/COMBINE-lab/salmon][repo]]
:doc:    [[https://salmon.readthedocs.io/en/latest/][doc]]
:paper:    [[https://doi.org/10.1038/nmeth.4197][paper]] | cite:patro2017
:ID:       c0d79be1-f887-460e-9142-4d43c6632126
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- see [[id:a0923938-4021-44cd-a8f3-94c0eb10cdb5][table of tools]]
- https://support.bioconductor.org/p/9140398/
- https://www.biostars.org/p/397086/
- https://www.scopus.com/results/citedbyresults.uri?sort=cp-f&cite=2-s2.0-85014549629&src=s&nlo=&nlr=&nls=&imp=t&sid=b3adb06908021bf68bd2e4cbb741bd78&sot=cite&sdt=a&sl=0&origin=resultslist&offset=1&txGid=8f3b391583118470ef66b945cc173d58
**** sambamba
:PROPERTIES:
:use: bam processing
:substrate: bam
:doc:  [[http://lomereiter.github.io/sambamba/docs/sambamba-view.html][online]]
:END:
**** samblaster
:PROPERTIES:
:doc:  [[https://github.com/GregoryFaust/samblaster][repo]]
:paper:   [[https://academic.oup.com/bioinformatics/article/30/17/2503/2748175][paper]]
:repo: [[https://github.com/GregoryFaust/samblaster][repo]]
:substrate: bam
:utility: single-pass dedup
:END:
**** samtools                                                       :biotool:
:PROPERTIES:
:ID:       0a3acec7-f4d1-4cd1-b3c8-7ec769715cc1
:CREATED:  [2021-09-13 Mon 07:29]
:doc:  [[http://www.htslib.org/doc/samtools.html][online]]
:END:
- https://bioinformatics.stackexchange.com/questions/18538/samtools-sort-most-efficient-memory-and-thread-settings-for-many-samples-on-a-c
- [[http://barcwiki.wi.mit.edu/wiki/SOPs/miningSAMBAM][some common Samtools tasks]]
- [[http://davetang.org/wiki/tiki-index.php?page=SAMTools][a Samtools tutorial]]
- [[http://biobits.org/samtools_primer.html][another Samtools tutorial]]
- [[http://broadinstitute.github.io/picard/explain-flags.html][SAM flags]]
- https://github.com/merenlab/anvio/issues/1479
- [[http://biobits.org/samtools_primer.html][another Samtools tutorial]]
- [[http://www.htslib.org/workflow/#mapping_to_cram][using cram]]
- [[https://www.samformat.info/sam-format-header][header tool]]
- commands
  - sequence inspection and quality
    | function                              | command                                       |
    |---------------------------------------+-----------------------------------------------|
    | <35>                                  | <35>                                          |
    | see     bam header                    | view -H                                       |
    | see bam header and top seq in sam/bam | samtools view -h <FILE.bam> \vert head -n 100 |
    | see ?                                 | idxstats                                      |
    |                                       |                                               |
  - sequence file manipulations
    |-------------------+---------------------------------------------------|
    | function          | command                                           |
    |-------------------+---------------------------------------------------|
    | check bam EOF     | samtools quickcheck <.bam>                        |
    | merge bams        | samtools merge all.bam *.bam                      |
    | random sample bam | samtools view -s 0.5 -b file.bam >random_half.bam |
    | split by chrom    | samtools view -b <FULL.bam> chr1 > chr1.bam       |
    | merge             | samtools merge <OUT.bam> <IN1.bam> <IN2.bam>      |
    | trim by quality   | -q INT                                            |
    | bam to sam        | samtools view -h -o out.sam in.bam                |
    | sam to bam        | samtools view -bS test.sam > test.bam             |
    | remove unmapped   | samtools view -h -F 4 file.bam                    |
**** Scythe
 [[https://github.com/vsbuffalo/scythe][Scythe]]
**** SeqKit

https://bioinf.shenwei.me/seqkit/usage/#fx2tab-tab2fx
https://www.biostars.org/p/9465609/
**** seqtk
:PROPERTIES:
:repo:  [[https://github.com/lh3/seqtk][github]]
:doc: [[https://github.com/lh3/seqtk/blob/master/README.md][README]]

:END:
**** seqTools
:PROPERTIES:
:function:  [[id:1EE731A3-83DD-46AF-B988-2311307C335F][seq qc]]
:doc:  [[https://www.bioconductor.org/packages/release/bioc/vignettes/seqTools/inst/doc/seqTools.pdf][seqTools]]
:END:
**** simNGS
:PROPERTIES:
:doc: [[https://www.ebi.ac.uk/goldman-srv/simNGS/][website]]

:END:
**** [[id:a8b27140-4b9e-40c5-86ab-ee5afeae723f][snakemake]]
**** STAR                                                           :biotool:
:PROPERTIES:
:ID:       d1d5d554-57c4-4b9c-8385-2bb72369375e
:CREATED:  [2021-09-13 Mon 07:29]
:paper:  cite:dobin2019
:doc:  [[https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf][doc]]
:END:
 UPDATE- STAR should handle adaptors (and maybe low quality reads as well, although there aren't many of those).
 see- https://groups.google.com/forum/#!searchin/rna-star/adaptor/rna-star/T8zBqtlxopU/8kVi17eDGAAJ

- used in [[id:326ecd60-8cd4-4815-a389-967b2c3fef0a][alignment]]
- star-rsem pipeline
  - cite:djebali2017
**** tophat2                                                        :biotool:
:PROPERTIES:
:doc:  [[https://ccb.jhu.edu/software/tophat/manual.html][doc]]
:ID:       db678422-a644-4298-b962-33657a5dee7e
:CREATED:  [2021-09-13 Mon 07:29]
:END:
notes- https://www.biostars.org/p/251018/
**** Trim Galore                                                    :biotool:
:PROPERTIES:
:ID:       3e4ca528-a604-4bfe-b235-2052e6d2d4a9
:CREATED:  [2021-09-13 Mon 07:29]
:END:
-
  - wrapper for Cutadapt and FastQC
  - [[https://github.com/FelixKrueger/TrimGalore][github]]
    - [[https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md][user guide]]
  - on CHPC cluster - http://mgt2.chpc.wustl.edu/wiki119/index.php/Software
- https://github.com/FelixKrueger/TrimGalore
**** trimmomatic :biotool:
http://www.usadellab.org/cms/?page=trimmomatic
https://www.biostars.org/p/478810/
**** trimmomatic :biotool:
http://www.usadellab.org/cms/?page=trimmomatic
https://www.biostars.org/p/478810/
**** tximport                                                       :biotool:
:PROPERTIES:
:doi:  [[http://dx.doi.org/10.12688/f1000research.7563.1][paper]]
:ID:       a5d4a32f-764b-4930-80e9-2a6e8d562ae8
:CREATED:  [2021-09-13 Mon 07:29]
:END:
#+begin_src R
saveDb()
#+end_src
- offsets handled directly in DESeq2 command - https://bioconductor.org/packages/devel/bioc/vignettes/tximport/inst/doc/tximport.html#Downstream_DGE_in_Bioconductor
"tximport can alternatively produce a matrix of data that is roughly on the scale of counts, by scaling transcript-per-million (TPM) abundances to add up to the total number of mapped reads. This counts-from-abundance approach directly corrects for technical biases and differential transcript usage across samples, obviating the need for the accompanying offset matrix"
**** wgcna                                                          :biotool:
:PROPERTIES:
:paper: [[https://doi.org/10.1186/1471-2105-9-559][paper]]
:ID:       948289d8-3a77-471a-b068-0507285e0bac
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- visualization
  - example
    - https://www.nature.com/articles/s41467-018-08126-5#Sec2
- https://www.nature.com/articles/nature16965#Sec1
- [[https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/][Horvath tutorials]]
  #+name:tutorial1-input
  #+begin_src R
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored.
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir);
# Load the package
library(WGCNA);
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
#Read in the female liver data set
femData = read.csv("LiverFemale3600.csv");
# Read in the male liver data set
maleData = read.csv("LiverMale3600.csv");
# Take a quick look at what is in the data sets (caution, longish output):
dim(femData)
names(femData)
dim(maleData)
names(maleData)


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# We work with two sets:
nSets = 2;
# For easier labeling of plots, create a vector holding descriptive names of the two sets.
setLabels = c("Female liver", "Male liver")
shortLabels = c("Female", "Male")
# Form multi-set expression data: columns starting from 9 contain actual expression data.
multiExpr = vector(mode = "list", length = nSets)

multiExpr[[1]] = list(data = as.data.frame(t(femData[-c(1:8)])));
names(multiExpr[[1]]$data) = femData$substanceBXH;
rownames(multiExpr[[1]]$data) = names(femData)[-c(1:8)];
multiExpr[[2]] = list(data = as.data.frame(t(maleData[-c(1:8)])));
names(multiExpr[[2]]$data) = maleData$substanceBXH;
rownames(multiExpr[[2]]$data) = names(maleData)[-c(1:8)];
# Check that the data has the correct format for many functions operating on multiple sets:
exprSize = checkSets(multiExpr)


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


# Check that all genes and samples have sufficiently low numbers of missing values.
gsg = goodSamplesGenesMS(multiExpr, verbose = 3);
gsg$allOK


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


if (!gsg$allOK)
{
  # Print information about the removed genes:
  if (sum(!gsg$goodGenes) > 0)
    printFlush(paste("Removing genes:", paste(names(multiExpr[[1]]$data)[!gsg$goodGenes],
                                              collapse = ", ")))
  for (set in 1:exprSize$nSets)
  {
    if (sum(!gsg$goodSamples[[set]]))
      printFlush(paste("In set", setLabels[set], "removing samples",
                       paste(rownames(multiExpr[[set]]$data)[!gsg$goodSamples[[set]]], collapse = ", ")))
    # Remove the offending genes and samples
    multiExpr[[set]]$data = multiExpr[[set]]$data[gsg$goodSamples[[set]], gsg$goodGenes];
  }
  # Update exprSize
  exprSize = checkSets(multiExpr)
}


#=====================================================================================
#
#  Code chunk 5
#
#=====================================================================================


sampleTrees = list()
for (set in 1:nSets)
{
  sampleTrees[[set]] = hclust(dist(multiExpr[[set]]$data), method = "average")
}


#=====================================================================================
#
#  Code chunk 6
#
## #=====================================================================================



#=====================================================================================
#
#  Code chunk 7
#
#=====================================================================================


# Choose the "base" cut height for the female data set
baseHeight = 16
# Adjust the cut height for the male data set for the number of samples
cutHeights = c(16, 16*exprSize$nSamples[2]/exprSize$nSamples[1]);
# Re-plot the dendrograms including the cut lines
## pdf(file = "Plots/SampleClustering.pdf", width = 12, height = 12);
## par(mfrow=c(2,1))
## par(mar = c(0, 4, 2, 0))
## for (set in 1:nSets)
## {
##   plot(sampleTrees[[set]], main = paste("Sample clustering on all genes in", setLabels[set]),
##        xlab="", sub="", cex = 0.7);
##   abline(h=cutHeights[set], col = "red");
## }
## dev.off();


#=====================================================================================
#
#  Code chunk 8
#
#=====================================================================================


for (set in 1:nSets)
{
  # Find clusters cut by the line
  labels = cutreeStatic(sampleTrees[[set]], cutHeight = cutHeights[set])
  # Keep the largest one (labeled by the number 1)
  keep = (labels==1)
  multiExpr[[set]]$data = multiExpr[[set]]$data[keep, ]
}
collectGarbage();
# Check the size of the leftover data
exprSize = checkSets(multiExpr)
exprSize


#=====================================================================================
#
#  Code chunk 9
#
#=====================================================================================


traitData = read.csv("ClinicalTraits.csv");
dim(traitData)
names(traitData)
# remove columns that hold information we do not need.
allTraits = traitData[, -c(31, 16)];
allTraits = allTraits[, c(2, 11:36) ];
# See how big the traits are and what are the trait and sample names
dim(allTraits)
names(allTraits)
allTraits$Mice
# Form a multi-set structure that will hold the clinical traits.
Traits = vector(mode="list", length = nSets);
for (set in 1:nSets)
{
  setSamples = rownames(multiExpr[[set]]$data);
  traitRows = match(setSamples, allTraits$Mice);
  Traits[[set]] = list(data = allTraits[traitRows, -1]);
  rownames(Traits[[set]]$data) = allTraits[traitRows, 1];
}
collectGarbage();
# Define data set dimensions
nGenes = exprSize$nGenes;
nSamples = exprSize$nSamples;


#=====================================================================================
#
#  Code chunk 10
#
#=====================================================================================


save(multiExpr, Traits, nGenes, nSamples, setLabels, shortLabels, exprSize,
     file = "Consensus-dataInput.RData");


#+end_src

  #+name:tutorial2-auto-network-construct
  #+begin_src R
#=====================================================================================
#
#  Code chunk 1
#
#=====================================================================================


# Display the current working directory
getwd();
# If necessary, change the path below to the directory where the data files are stored.
# "." means current directory. On Windows use a forward slash / instead of the usual \.
workingDir = ".";
setwd(workingDir);
# Load the WGCNA package
library(WGCNA)
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE);
# Allow multi-threading within WGCNA.
# Caution: skip this line if you run RStudio or other third-party R environments.
# See note above.
enableWGCNAThreads()
# Load the data saved in the first part
lnames = load(file = "Consensus-dataInput.RData");
# The variable lnames contains the names of loaded variables.
lnames
# Get the number of sets in the multiExpr structure.
nSets = checkSets(multiExpr)$nSets


#=====================================================================================
#
#  Code chunk 2
#
#=====================================================================================


# Choose a set of soft-thresholding powers
powers = c(seq(4,10,by=1), seq(12,20, by=2));
# Initialize a list to hold the results of scale-free analysis
powerTables = vector(mode = "list", length = nSets);
# Call the network topology analysis function for each set in turn
for (set in 1:nSets)
  powerTables[[set]] = list(data = pickSoftThreshold(multiExpr[[set]]$data, powerVector=powers,
                                                     verbose = 2)[[2]]);
collectGarbage();
# Plot the results:
colors = c("black", "red")
# Will plot these columns of the returned scale free analysis tables
plotCols = c(2,5,6,7)
colNames = c("Scale Free Topology Model Fit", "Mean connectivity", "Median connectivity",
"Max connectivity");
# Get the minima and maxima of the plotted points
ylim = matrix(NA, nrow = 2, ncol = 4);
for (set in 1:nSets)
{
  for (col in 1:length(plotCols))
  {
    ylim[1, col] = min(ylim[1, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
    ylim[2, col] = max(ylim[2, col], powerTables[[set]]$data[, plotCols[col]], na.rm = TRUE);
  }
}
# Plot the quantities in the chosen columns vs. the soft thresholding power
## sizeGrWindow(8, 6)
## pdf(file = "Plots/scaleFreeAnalysis.pdf", wi = 8, he = 6)
## par(mfcol = c(2,2));
## par(mar = c(4.2, 4.2 , 2.2, 0.5))
## cex1 = 0.7;
## for (col in 1:length(plotCols)) for (set in 1:nSets)
## {
##   if (set==1)
##   {
##     plot(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
##          xlab="Soft Threshold (power)",ylab=colNames[col],type="n", ylim = ylim[, col],
##          main = colNames[col]);
##     addGrid();
##   }
##   if (col==1)
##   {
##     text(powerTables[[set]]$data[,1], -sign(powerTables[[set]]$data[,3])*powerTables[[set]]$data[,2],
##          labels=powers,cex=cex1,col=colors[set]);
##   } else
##     text(powerTables[[set]]$data[,1], powerTables[[set]]$data[,plotCols[col]],
##          labels=powers,cex=cex1,col=colors[set]);
##   if (col==1)
##   {
##     legend("bottomright", legend = setLabels, col = colors, pch = 20) ;
##   } else
##     legend("topright", legend = setLabels, col = colors, pch = 20) ;
## }
## dev.off();


#=====================================================================================
#
#  Code chunk 3
#
#=====================================================================================


net = blockwiseConsensusModules(
        multiExpr, power = 6, minModuleSize = 30, deepSplit = 2,
        pamRespectsDendro = FALSE,
        mergeCutHeight = 0.25, numericLabels = TRUE,
        minKMEtoStay = 0,
        saveTOMs = TRUE, verbose = 5)


#=====================================================================================
#
#  Code chunk 4
#
#=====================================================================================


************* NEXT HCC1395 straight align and integrate and compare to published
:PROPERTIES:
:CREATED:  [2019-03-13 Wed 10:11]
:ID:       58F7BED2-DED1-4911-9E8C-104E12126A73
:END:
:LOGBOOK:
CLOCK: [2019-03-14 Thu 07:49]--[2019-03-14 Thu 08:45] =>  0:56
CLOCK: [2019-03-13 Wed 18:18]--[2019-03-13 Wed 19:54] =>  1:36
CLOCK: [2019-03-13 Wed 11:30]--[2019-03-13 Wed 12:00] =>  0:30
CLOCK: [2019-03-13 Wed 11:00]--[2019-03-13 Wed 11:01] =>  0:01
CLOCK: [2019-03-13 Wed 10:12]--[2019-03-13 Wed 10:47] =>  0:35
:END:
- log
  - from bam to fastq
    - by sample
      - tumor
        - ./sambamba-0.6.8-linux-static sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam /gscmnt/gc2555/spencer/jeff/data/gerald_C1TD1ACXX_8_ACAGTG.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395-anmesort.bam
      - moved all this to ~data/HCC1395/
      - normal
        - (integrate image, 64gb) sambamba sort -n -o /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal-namesort.bam /gscmnt/gc2555/spencer/jeff/data/HCC1395/gerald_C2DBEACXX_3.bam
        - samtools fastq -1 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq -2 /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/normal.namesort.bam
        -
  - alignment
    - bsub -Is -q docker-interactive -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/startest)' /bin/bash
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-bl_2.fastq --outSAMunmapped Within
      - STAR --runThreadN 4 --genomeDir /gscmnt/gc2555/spencer/jeff/data/reference/star-index/ --readFilesIn /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_1.fastq /gscmnt/gc2555/spencer/jeff/data/HCC1395/HCC1395-all_2.fastq --outSAMunmapped Within
  - integrate
    - bsub -q research-hpc -R "select[ncpus >=4 && mem>64000] span[hosts=1] rusage[mem=64000]" -a 'docker(jeszyman/integrate)' /bin/bash
    - prep each star output:
      -
      - sambamba view -S -o Aligned.out.bam Aligned.out.sam
      - sambamba sort Aligned.out.bam
      - sambamba index Aligned.out.sorted.bam
    - Integrate fusion /gscmnt/gc2555/spencer/jeff/data/reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa /gscmnt/gc2555/spencer/jeff/data/reference/annot-hg38-refeseq.txt /gscmnt/gc2555/spencer/jeff/data/reference/Integrate/bwts /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam /gscmnt/gc2555/spencer/jeff/dev/starhcc/Aligned.out.sorted.bam
      -
**** wikipathways :biotool:
:PROPERTIES:
:ID:       69bfa728-e149-4b8a-9863-347630f86bc4
:END:
- uses [[id:17c865dc-765c-496d-b5c7-60d3d3439de3][pathvisio]]
- cite:wikipathways
*** Table
#+BEGIN: propview :id "1b8bb7db-2622-47bd-bd6f-052cbd5d4fdb" :cols (DOC)
|                                                                                                  DOC |                                     |   |       |   |                                                          |   |   |   |
|------------------------------------------------------------------------------------------------------+-------------------------------------+---+-------+---+----------------------------------------------------------+---+---+---|
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                   "" |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                [[http://bedtools\.readthedocs\.io/en/latest/] [doc]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                       "https://github.com/seqan/flexbar/wiki/Manual" |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
| [[http://portals\.broadinstitute\.org/cgi-bin/cancer/publications/pub_paper\.cgi\?mode=view&paper_id=216&p=t] [site]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    " |                                     |   | fastq |   | [[https://ccb.jhu.edu/software/hisat2/manual.html][doc]] |   |   | " |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                           [[https://github\.com/broadinstitute/ichorCNA/wiki] [doc]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    " | [[https://multiqc.info/docs/][doc]] |   |       | " |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                             [[http://www\.basic\.northwestern\.edu/biotools/oligocalc\.html] [site]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                      [[https://broadinstitute\.github\.io/picard/index\.html] [doc]] |                                     |   |       |   |                                                          |   |   |   |
| [[http://bioconductor\.org/packages/release/bioc/vignettes/polyester/inst/doc/polyester\.html] [vignette]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                [[https://github\.com/broadinstitute/rnaseqc] [repo]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                           [[http://rseqc\.sourceforge\.net/] [repo]] |                                     |   |       |   |                                                          |   |   |   |
|                                                 [[https://salmon\.readthedocs\.io/en/latest/] [doc]] |                                     |   |       |   |                                                          |   |   |   |
|                         [[http://lomereiter\.github\.io/sambamba/docs/sambamba-view\.html] [online]] |                                     |   |       |   |                                                          |   |   |   |
|                                               [[https://github\.com/GregoryFaust/samblaster] [repo]] |                                     |   |       |   |                                                          |   |   |   |
|                                              [[http://www\.htslib\.org/doc/samtools\.html] [online]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
| [[https://www\.bioconductor\.org/packages/release/bioc/vignettes/seqTools/inst/doc/seqTools\.pdf] [seqTools]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                         [[https://github\.com/alexdobin/STAR/blob/master/doc/STARmanual\.pdf] [doc]] |                                     |   |       |   |                                                          |   |   |   |
|                                         [[https://ccb\.jhu\.edu/software/tophat/manual\.html] [doc]] |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|                                                                                                    0 |                                     |   |       |   |                                                          |   |   |   |
|------------------------------------------------------------------------------------------------------+-------------------------------------+---+-------+---+----------------------------------------------------------+---+---+---|
|                                                                                                      |                                     |   |       |   |                                                          |   |   |   |
#+END:
*** tool collections
:PROPERTIES:
:ID:       7cce7be9-b3ac-4127-9a92-71559617588e
:CREATED:  [2021-09-13 Mon 07:29]
:END:
- [[https://www.ebi.ac.uk/services][EMBL-EBI Tools]]
**** UCSC tools                                                     :biotool:
:PROPERTIES:
:ID:       be612cfb-2a75-4935-92fd-85620dcabf92
:CREATED:  [2021-09-13 Mon 07:29]
:END:
#+begin_src bash
ls
wget -r --no-parent --reject "index.html*" -nH --cut-dirs=3 http://genome.ucsc.edu/goldenPath/help/examples/hubDirectory/

#+end_src
- hosting track hubs- try cyverse
  - https://cyverse.org/about
  - https://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Hosting
  - https://genome.ucsc.edu/goldenPath/help/hubQuickStart.html
  - https://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Hosting
- ucsc
  - table browser
    - https://genomebrowser.wustl.edu/goldenPath/help/hgTextHelp.html#Introduction
  - https://www.biostars.org/p/19029/
  - http://genome.ucsc.edu/FAQ/FAQreleases.html
  - cite:karolchik2011
*** <<<Integrated Genomics Viewer>>> (<<<IGV>>>)
:PROPERTIES:
:ID:       917062AC-32E8-41F5-A511-84AA7B1504B0
:END:

- cite:robinson2017
https://www.france-bioinformatique.fr/sites/default/files/EBA/V3-2014/Elodie_Giraud_visualisation-IGV_TP_EBA-2014-10.pdf
[[http://software.broadinstitute.org/software/igv/UserGuide][User Guide]]
** [[id:d482d49c-9dbe-43cd-a188-b1662a9ce348][Bioinformatics research project module]]
** Biotools dockerfiles
:LOGBOOK:
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:19] =>  1:00
CLOCK: [2022-01-04 Tue 09:15]--[2022-01-04 Tue 09:16] =>  0:01
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:15] =>  0:56
CLOCK: [2021-03-03 Wed 19:56]--[2021-03-03 Wed 21:25] =>  1:29
:END:
*** [[id:79e01a67-5ba1-4257-8353-8c5cf1ac8c4a][Basecamp docker]]
*** [[file:docker/biotools_Dockerfile][Biotools docker]]
:PROPERTIES:
:ID:       7ed344ce-373c-4f9e-b24f-87b57374e1a0
:header-args:bash: :tangle ./docker/biotools_Dockerfile
:END:
:LOGBOOK:
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:19] =>  1:00
CLOCK: [2022-01-04 Tue 09:15]--[2022-01-04 Tue 09:16] =>  0:01
CLOCK: [2021-12-09 Thu 16:19]--[2021-12-09 Thu 17:15] =>  0:56
CLOCK: [2021-03-03 Wed 19:56]--[2021-03-03 Wed 21:25] =>  1:29
:END:
***** Preamble
#+begin_src bash
FROM jeszyman/basecamp:2.0.0
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build a basic bioinformatics docker image          ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/biotools
# (https://hub.docker.com/repository/docker/jeszyman/atac)

#+end_src
***** Conda
#+begin_src bash

#################
###   Conda   ###
#################

RUN conda install -c bioconda bbmap=39.00 \
    && conda install -c bioconda bcftools=1.15.1 \
    && conda install -c bioconda bedops=2.4.41 \
    && conda install -c bioconda bedtools=2.30.0 \
    && conda install -c bioconda bowtie=1.3.1 \
    && conda install -c bioconda bowtie2=2.4.5 \
    && conda install -c bioconda bwa=0.7.17 \
    && conda install -c bioconda cutadapt=4.1 \
    && conda install -c bioconda deeptools=3.5.1 --force \
    && conda install -c bioconda fastp=0.23.2 \
    && conda install -c bioconda fastqc=0.11.9 \
    && conda install -c bioconda gatk=4.2.6.0 \
    && conda install -c bioconda goleft=0.2.4 \
    && conda install -c bioconda htseq=2.0.2 \
    && conda install -c bioconda mosdepth=0.3.3 \
    && conda install -c bioconda preseq=3.2.0 \
    && conda install -c bioconda pysam=0.19.1 \
    && conda install -c bioconda qualimap=2.2.2d \
    && conda install -c bioconda sambamba=0.8.2 \
    && conda install -c bioconda samblaster=0.1.26 \
    && conda install -c bioconda samtools=1.7 \
    && conda install -c bioconda seqkit=2.3.1 \
    && conda install -c bioconda seqtk=1.3 \
    && conda install -c bioconda skewer=0.2.2 \
    && conda install -c bioconda trimmomatic=0.39 \
    && conda install -c bioconda vcftools=0.1.16-7
#+end_src
***** R and Bioconductor
#+begin_src bash
# To avoid conflict with jeszyman/basecamp R install
#RUN conda uninstall r-base

##########################
###   R Bioconductor   ###
##########################

RUN R -e "install.packages('BiocManager', repos='http://cran.rstudio.com/')"

RUN R -e "BiocManager::install(version='3.15', update=TRUE, ask=FALSE)"

RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("AnnotationDbi",\
                                 "apeglm"
                                 "BiocParallel", \
                                 "biomaRt", \
                                 "BSgenome.Hsapiens.UCSC.hg38", \
                                 "BSgenome.Mmusculus.UCSC.mm10", \
                                 "ComplexHeatmap", \
                                 "DEGreport", \
                                 "DESeq2", \
                                 "edgeR", \
                                 "EnsDb.Mmusculus.v79", \
                                 "GenomicAlignments", \
                                 "org.Hs.eg.db", \
                                 "org.Mm.eg.db", \
                                 "readr", \
                                 "Rsamtools", \
                                 "TxDb.Hsapiens.UCSC.hg38.knownGene", \
                                 "TxDb.Hsapiens.UCSC.hg38.ensGene", \
                                 "TxDb.Mmusculus.UCSC.mm10.knownGene", \
                                 "TxDb.Mmusculus.UCSC.mm10.ensGene"));'

RUN R -e "install.packages(c('seqinr'), repos='http://cran.rstudio.com/')"

#+end_src
***** UCSC
#+begin_src bash

##############################################
###   Select UCSC Command-line utilities   ###
##############################################

RUN mkdir /opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedClip --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCommonRegions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCoverage --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedExtendRanges --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGeneParts --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphPack --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedIntersect --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedItemOverlapCount --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset.py --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedMergeAdjacent --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPartition --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPileUps --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRemoveOverlap --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRestrictToPositions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedSort --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToExons --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigAverageOverBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCat --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCluster --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCorrelate --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigInfo --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigMerge --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/#+end_src --directory-prefix=/opt/ucsc \
    && chmod -R +x /opt/ucsc/

ENV PATH="/opt/ucsc:${PATH}"

#+end_src
***** Custom and manual installs
#+begin_src bash

###########################
###   Custom Installs   ###
###########################

# Multiqc

RUN pip install multiqc

# Flexbar
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated libtbb2
RUN wget https://github.com/seqan/flexbar/releases/download/v3.5.0/flexbar-3.5.0-linux.tar.gz && tar -xzf flexbar-3.5.0-linux.tar.gz && mv flexbar-3.5.0-linux/flexbar /usr/bin/ && mv flexbar-3.5.0-linux/libtbb.so.2 /usr/bin/

# samtools
# https://www.biostars.org/p/9480029/
RUN conda install -c bioconda samtools openssl=1.0

#+end_src
****** Picard
#+begin_src bash

##################
###   Picard   ###
##################

RUN apt-get install -yq --no-install-recommends --allow-unauthenticated openjdk-8-jdk

#########1#########2#########3#########4#########5#########6#########7#########8
#https://github.com/genome/docker-picard-cwl/commit/639ad8e4d8e7bd7e6e10f193df8d5cd16260d2e2
RUN mkdir /opt/picard-2.18.1/ \
    && cd /tmp/ \
    && wget --no-check-certificate https://github.com/broadinstitute/picard/releases/download/2.18.1/picard.jar \
    && mv picard.jar /opt/picard-2.18.1/ \
    && ln -s /opt/picard-2.18.1 /opt/picard \
    && ln -s /opt/picard-2.18.1 /usr/picard
#########1#########2#########3#########4#########5#########6#########7#########8


#+end_src
***** Development                                                       :dev:
:PROPERTIES:
:header-args: :tangle no
:END:
- give specific versions for all condas
- Old code
    #
  # cnvkit
  RUN pip install -U cython
  #FAILS [2021-02-17] RUN pip install -U future futures pandas pomegranate pyfaidx
  #FAILS [2021-02-17] RUN pip install cnvkit==0.9.6
  #
  # RSeQC
  RUN pip install RSeQC
  #
  #
  RUN cd /opt && \
      git clone https://github.com/shahcompbio/hmmcopy_utils.git && \
      cd hmmcopy_utils && \
      cmake . && \
      make
  ##NEED edger, limma, gage, dseq2, wgcna
  #
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("DNAcopy");'
  #

  #########1#########2#########3#########4#########5#########6#########7######
  #TESTING
  #
  #RUN conda install -c bioconda tophat
  #RUN conda install -c bioconda cnvkit
  #RUN conda install -c bioconda manta
  #RUN conda install -c bioconda lumpy-sv
  #RUN conda install -c bioconda multiqc

  ###8
  # #
  # ### LUMPY
  # # from https://raw.githubusercontent.com/zlskidmore/docker-lumpy/master/Dockerfile
  # # RUN apt-get update -qq
  # # RUN apt-get install -qq --no-install-recommends \
  # # python-pip \
  # # git \
  # # cmake \
  # # build-essential \
  # # libz-dev
  # # RUN cd /opt && git clone https://github.com/hall-lab/lumpy-sv.git && cd /opt/lumpy-sv && make
  # #########1#########2#########3#########4#########5#########6#########7#########8
  # ### Samtools
  # RUN apt-get update -qq
  # RUN apt-get install -qq --no-install-recommends \
  # wget \
  # bzip2 \
  # cmake \
  # gcc \
  # zlib1g-dev \
  # libncurses5-dev
  # ENV SAMTOOLS_INSTALL_DIR=/opt/samtools
  # WORKDIR /tmp
  # RUN wget --no-check-certificate https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2 && \
  # tar --bzip2 -xf samtools-1.9.tar.bz2 && \
  # cd /tmp/samtools-1.9 && \
  # ./configure --prefix=$SAMTOOLS_INSTALL_DIR && \
  # make && \
  # make install && \
  # cd / && \
  # rm -rf /tmp/samtools-1.9 && \
  # ln -s /opt/samtools/bin/* /usr/bin/

  # # # WORKDIR /usr/local/bin/
  # # # RUN curl -SL https://github.com/samtools/samtools/releases/download/${samtools_version}/samtools-${samtools_version}.tar.bz2 \
  # # #     > /usr/local/bin/samtools-${samtools_version}.tar.bz2
  # # # RUN tar -xjf /usr/local/bin/samtools-${samtools_version}.tar.bz2 -C /usr/local/bin/
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && ./configure
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && make
  # # # RUN cd /usr/local/bin/samtools-${samtools_version}/ && make install

  # # # # install lumpy
  # # # WORKDIR /usr/local/bin
  # # # RUN wget https://github.com/arq5x/lumpy-sv/releases/download/${lumpy_version}/lumpy-sv.tar.gz
  # # # RUN tar -xzvf lumpy-sv.tar.gz
  # # # WORKDIR /usr/local/bin/lumpy-sv
  # # # RUN make
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpy /usr/local/bin/lumpy
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpy_filter /usr/local/bin/lumpy_filter
  # # # RUN ln -s /usr/local/bin/lumpy-sv/bin/lumpyexpress /usr/local/bin/lumpyexpress
  # # # https://github.com/hall-lab/sv-pipeline/blob/master/docker/lumpy/Dockerfile
  # # # Build dependencies
  # # RUN apt-get update -qq \
  # #     && apt-get -y install \
  # #         apt-transport-https \
  # #         g++ \
  # # 	gawk \
  # #         libcurl4-gnutls-dev \
  # #         autoconf \
  # # 	libssl-dev \
  # #         git

  # #########1#########2#########3#########4#########5#########6#########7#########8
  # # ###############
  # # #bam-readcount#
  # # ###############
  # # # ENV SAMTOOLS_ROOT=/opt/samtools
  # # # RUN apt-get update && apt-get install -y --no-install-recommends \
  # # #         cmake \
  # # #         patch && \
  # # #     mkdir /opt/bam-readcount && \
  # # #     cd /opt/bam-readcount && \
  # # #     git clone https://github.com/genome/bam-readcount.git /tmp/bam-readcount-0.7.4 && \
  # # #     git -C /tmp/bam-readcount-0.7.4 checkout v0.7.4 && \
  # # #     cmake /tmp/bam-readcount-0.7.4 && \
  # # #     make && \
  # # #     rm -rf /tmp/bam-readcount-0.7.4 && \
  # # #     ln -s /opt/bam-readcount/bin/bam-readcount /usr/bin/bam-readcount

  # # # #note - this script needs cyvcf - installed in the python secetion!
  # # # COPY bam_readcount_helper.py /usr/bin/bam_readcount_helper.py
  # # # #############
  # # # ## IGV 3.0 ##

  # # # RUN apt-get update && apt-get install -y --no-install-recommends \
  # # #     software-properties-common \
  # # #     glib-networking-common && \
  # # #     mkdir -p /igv && \
  # # #     cd /igv && \
  # # #     wget http://data.broadinstitute.org/igv/projects/downloads/3.0_beta/IGV_3.0_beta.zip && \
  # # #     unzip IGV_3.0_beta.zip && \
  # # #     cd IGV_3.0_beta && \
  # # #     sed -i 's/Xmx4000/Xmx8000/g' igv.sh && \
  # # #     cd /usr/bin && \
  # # #     ln -s /igv/IGV_3.0_beta/igv.sh ./igv

  # # ##############
  # # ## bedtools ##

  # # WORKDIR /usr/local
  # # RUN git clone https://github.com/arq5x/bedtools2.git && \
  # #     cd /usr/local/bedtools2 && \
  # #     git checkout v2.25.0 && \
  # #     make && \
  # #     ln -s /usr/local/bedtools2/bin/* /usr/local/bin/
  # # ##############
  # # ## vcftools ##
  # # ENV ZIP=vcftools-0.1.14.tar.gz
  # # ENV URL=https://github.com/vcftools/vcftools/releases/download/v0.1.14/
  # # ENV FOLDER=vcftools-0.1.14
  # # ENV DST=/tmp

  # # RUN wget $URL/$ZIP -O $DST/$ZIP && \
  # #     tar xvf $DST/$ZIP -C $DST && \
  # #     rm $DST/$ZIP && \
  # #     cd $DST/$FOLDER && \
  # #     ./configure && \
  # #     make && \
  # #     make install && \
  # #     cd / && \
  # #     rm -rf $DST/$FOLDER
  # # ##################
  # # # ucsc utilities #
  # # RUN mkdir -p /tmp/ucsc && \
  # #     cd /tmp/ucsc && \
  # #     wget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigAverageOverBed http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/wigToBigWig && \
  # #     chmod ugo+x * && \
  # #     mv * /usr/bin/
  # # ############################
  # # # R, bioconductor packages #
  # # # from https://raw.githubusercontent.com/rocker-org/rocker-versioned/master/r-ver/3.4.0/Dockerfile
  # # # we'll pin to 3.4.0 for now

  # # # ARG R_VERSION
  # # # ARG BUILD_DATE
  # # # ENV BUILD_DATE 2017-06-20
  # # # ENV R_VERSION=${R_VERSION:-3.4.0}
  # # # RUN apt-get update && apt-get install -y --no-install-recommends locales && \
  # # #     echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen && \
  # # #     locale-gen en_US.UTF-8 && \
  # # #     LC_ALL=en_US.UTF-8 && \
  # # #     LANG=en_US.UTF-8 && \
  # # #     /usr/sbin/update-locale LANG=en_US.UTF-8 && \
  # # #     TERM=xterm && \
  # # #     apt-get install -y --no-install-recommends \
  # # #     bash-completion \
  # # #     ca-certificates \
  # # #     file \
  # # #     fonts-texgyre \
  # # #     g++ \
  # # #     gfortran \
  # # #     gsfonts \
  # # #     libbz2-1.0 \
  # # #     libcurl3 \
  # # #     libicu55 \
  # # #     libjpeg-turbo8 \
  # # #     libopenblas-dev \
  # # #     libpangocairo-1.0-0 \
  # # #     libpcre3 \
  # # #     libpng12-0 \
  # # #     libtiff5 \
  # # #     liblzma5 \
  # # #     locales \
  # # #     zlib1g \
  # # #     libbz2-dev \
  # # #     libcairo2-dev \
  # # #     libcurl4-openssl-dev \
  # # #     libpango1.0-dev \
  # # #     libjpeg-dev \
  # # #     libicu-dev \
  # # #     libmariadb-client-lgpl-dev \
  # # #     libmysqlclient-dev \
  # # #     libpcre3-dev \
  # # #     libpng-dev \
  # # #     libreadline-dev \
  # # #     libtiff5-dev \
  # # #     liblzma-dev \
  # # #     libx11-dev \
  # # #     libxt-dev \
  # # #     perl \
  # # #     tcl8.5-dev \
  # # #     tk8.5-dev \
  # # #     texinfo \
  # # #     texlive-extra-utils \
  # # #     texlive-fonts-recommended \
  # # #     texlive-fonts-extra \
  # # #     texlive-latex-recommended \
  # # #     x11proto-core-dev \
  # # #     xauth \
  # # #     xfonts-base \
  # # #     xvfb \
  # # #     zlib1g-dev && \
  # # #     cd /tmp/ && \
  # # #     ## Download source code
  # # #     curl -O https://cran.r-project.org/src/base/R-3/R-${R_VERSION}.tar.gz && \
  # # #     ## Extract source code
  # # #     tar -xf R-${R_VERSION}.tar.gz && \
  # # #     cd R-${R_VERSION} && \
  # # #     ## Set compiler flags
  # # #     R_PAPERSIZE=letter && \
  # # #     R_BATCHSAVE="--no-save --no-restore" && \
  # # #     R_BROWSER=xdg-open && \
  # # #     PAGER=/usr/bin/pager && \
  # # #     PERL=/usr/bin/perl && \
  # # #     R_UNZIPCMD=/usr/bin/unzip && \
  # # #     R_ZIPCMD=/usr/bin/zip && \
  # # #     R_PRINTCMD=/usr/bin/lpr && \
  # # #     LIBnn=lib && \
  # # #     AWK=/usr/bin/awk && \
  # # #     CFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g" && \
  # # #     CXXFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g" && \
  # # #     ## Configure options
  # # #     ./configure --enable-R-shlib \
  # # #                --enable-memory-profiling \
  # # #                --with-readline \
  # # #                --with-blas="-lopenblas" \
  # # #                --disable-nls \
  # # #                --without-recommended-packages && \
  # # #     ## Build and install
  # # #     make && \
  # # #     make install && \
  # # #     ## Add a default CRAN mirror
  # # #     echo "options(repos = c(CRAN = 'https://cran.rstudio.com/'), download.file.method = 'libcurl')" >> /usr/local/lib/R/etc/Rprofile.site && \
  # # #     ## Add a library directory (for user-installed packages)
  # # #     mkdir -p /usr/local/lib/R/site-library && \
  # # #     chown root:staff /usr/local/lib/R/site-library && \
  # # #     chmod g+wx /usr/local/lib/R/site-library && \
  # # #     ## Fix library path
  # # #     echo "R_LIBS_USER='/usr/local/lib/R/site-library'" >> /usr/local/lib/R/etc/Renviron && \
  # # #     echo "R_LIBS=\${R_LIBS-'/usr/local/lib/R/site-library:/usr/local/lib/R/library:/usr/lib/R/library'}" >> /usr/local/lib/R/etc/Renviron && \
  # # #     ## install packages from date-locked MRAN snapshot of CRAN
  # # #     [ -z "$BUILD_DATE" ] && BUILD_DATE=$(TZ="America/Los_Angeles" date -I) || true && \
  # # #     MRAN=https://mran.microsoft.com/snapshot/${BUILD_DATE} && \
  # # #     echo MRAN=$MRAN >> /etc/environment && \
  # # #     export MRAN=$MRAN && \
  # # #     echo "options(repos = c(CRAN='$MRAN'), download.file.method = 'libcurl')" >> /usr/local/lib/R/etc/Rprofile.site && \
  # # #     ## Use littler installation scripts
  # # #     Rscript -e "install.packages(c('littler', 'docopt'), repo = '$MRAN')" && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/examples/install2.r /usr/local/bin/install2.r && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/examples/installGithub.r /usr/local/bin/installGithub.r && \
  # # #     ln -s /usr/local/lib/R/site-library/littler/bin/r /usr/local/bin/r

  # # #    ## install r packages, bioconductor, etc ##
  # # #    ADD rpackages.R /tmp/
  # # #    RUN R -f /tmp/rpackages.R && \
  # # #    ## install fishplot ##
  # # #    cd /tmp/ && \
  # # #     wget https://github.com/chrisamiller/fishplot/archive/v0.4.tar.gz && \
  # # #     mv v0.4.tar.gz fishplot_0.4.tar.gz && \
  # # #     R CMD INSTALL fishplot_0.4.tar.gz && \
  # # #     cd && rm -rf /tmp/fishplot_0.4.tar.gz

  # # #    ## Clean up
  # # #    RUN cd / && \
  # # #    rm -rf /tmp/* && \
  # # #    apt-get autoremove -y && \
  # # #    apt-get autoclean -y && \
  # # #    rm -rf /var/lib/apt/lists/* && \
  # # #    apt-get clean

  # # # #################################
  # # # # Python 2 and 3, plus packages

  # # # # Configure environment
  # # # ENV CONDA_DIR /opt/conda
  # # # ENV PATH $CONDA_DIR/bin:$PATH

  # # # # Install conda
  # # # RUN cd /tmp && \
  # # #     mkdir -p $CONDA_DIR && \
  # # #     curl -s https://repo.continuum.io/miniconda/Miniconda3-4.3.21-Linux-x86_64.sh -o miniconda.sh && \
  # # #     /bin/bash miniconda.sh -f -b -p $CONDA_DIR && \
  # # #     rm miniconda.sh && \
  # # #     $CONDA_DIR/bin/conda config --system --add channels conda-forge && \
  # # #     $CONDA_DIR/bin/conda config --system --set auto_update_conda false && \
  # # #     conda clean -tipsy

  # # # # Install Python 3 packages available through pip
  # # # RUN conda install --yes 'pip' && \
  # # #     conda clean -tipsy && \
  # # #     #dependencies sometimes get weird - installing each on it's own line seems to help
  # # #     pip install numpy==1.13.0 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install cruzdb==0.5.6 && \
  # # #     pip install cython==0.25.2 && \
  # # #     pip install pyensembl==1.1.0 && \
  # # #     pip install pyfaidx==0.4.9.2 && \
  # # #     pip install pybedtools==0.7.10 && \
  # # #     pip install cyvcf2==0.7.4 && \
  # # #     pip install intervaltree_bio==1.0.1 && \
  # # #     pip install pandas==0.20.2 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install pysam==0.11.2.2 && \
  # # #     pip install seaborn==0.7.1 && \
  # # #     pip install scikit-learn==0.18.2 && \
  # # #     pip install svviz==1.6.1

  # # # # Install Python 2
  # # # RUN conda create --quiet --yes -p $CONDA_DIR/envs/python2 python=2.7 'pip' && \
  # # #     conda clean -tipsy && \
  # # #     /bin/bash -c "source activate python2 && \
  # # #     #dependencies sometimes get weird - installing each on it's own line seems to help
  # # #     pip install numpy==1.13.0 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install cruzdb==0.5.6 && \
  # # #     pip install cython==0.25.2 && \
  # # #     pip install pyensembl==1.1.0 && \
  # # #     pip install pyfaidx==0.4.9.2 && \
  # # #     pip install pybedtools==0.7.10 && \
  # # #     pip install cyvcf2==0.7.4 && \
  # # #     pip install intervaltree_bio==1.0.1 && \
  # # #     pip install pandas==0.20.2 && \
  # # #     pip install scipy==0.19.0 && \
  # # #     pip install pysam==0.11.2.2 && \
  # # #     pip install seaborn==0.7.1 && \
  # # #     pip install scikit-learn==0.18.2 && \
  # # #     pip install openpyxl==2.4.8 && \
  # # #     source deactivate"

  # # # COPY tsv2xlsx.py /usr/bin/tsv2xlsx.py

  # # # needed for MGI data mounts
  # # RUN apt-get update && apt-get install -y libnss-sss && apt-get clean all

  # # #set timezone to CDT
  # # #LSF: Java bug that need to change the /etc/timezone.
  # # #/etc/localtime is not enough.
  # # # RUN ln -sf /usr/share/zoneinfo/America/Chicago /etc/localtime && \
  # # #     echo "America/Chicago" > /etc/timezone && \
  # # #     dpkg-reconfigure --frontend noninteractive tzdata

  # # #UUID is needed to be set for some applications
  # # RUN apt-get update && apt-get install -y dbus && apt-get clean all
  # # RUN dbus-uuidgen >/etc/machine-id

  # # # WORKS TO HERE
  # # ENV PATH /opt/conda/bin:$PATH

  # # RUN apt-get update --fix-missing && apt-get install -y wget bzip2 ca-certificates \
  # #     libglib2.0-0 libxext6 libsm6 libxrender1 \
  # #     git mercurial subversion

  # # RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda2-4.5.11-Linux-x86_64.sh -O ~/miniconda.sh && \
  # #     /bin/bash ~/miniconda.sh -b -p /opt/conda && \
  # #     rm ~/miniconda.sh && \
  # #     ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
  # #     echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc && \
  # #     echo "conda activate base" >> ~/.bashrc

  # # RUN apt-get install -y curl grep sed dpkg && \
  # #     TINI_VERSION=`curl https://github.com/krallin/tini/releases/latest | grep -o "/v.*\"" | sed 's:^..\(.*\).$:\1:'` && \
  # #     curl -L "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini_${TINI_VERSION}.deb" > tini.deb && \
  # #     dpkg -i tini.deb && \
  # #     rm tini.deb && \
  # #     apt-get clean
  # # RUN conda install -c bioconda sambamba

  # # # WORKDIR /opt

  # # # RUN apt-get update && apt-get install -y \
  # # # 	autoconf \
  # # # 	automake \
  # # # 	make \
  # # # 	g++ \
  # # # 	gcc \
  # # # 	build-essential \
  # # # 	zlib1g-dev \
  # # # 	libgsl0-dev \
  # # # 	perl \
  # # # 	curl \
  # # # 	git \
  # # # 	wget \
  # # # 	unzip \
  # # # 	tabix \
  # # # 	libncurses5-dev

  # # # RUN wget https://github.com/ldc-developers/ldc/releases/download/v0.17.1/ldc2-0.17.1-linux-x86_64.tar.xz && \
  # # #   tar xJf ldc2-0.17.1-linux-x86_64.tar.xz

  # # # ENV PATH=/opt/ldc2-0.17.1-linux-x86_64/bin/:$PATH
  # # # ENV LIBRARY_PATH=/opt/ldc2-0.17.1-linux-x86_64/lib/

  # # # #RUN wget https://github.com/biod/sambamba/releases/download/v0.7.0/sambamba-0.7.0-linux-static.gz && \
  # # # tar xJf sambamba-0.7.0-linux-static.gz

  # # # RUN git clone --recursive https://github.com/lomereiter/sambamba.git && cd WHERE && make
  # # RUN conda install -c bioconda samtools
  # # ## bioconductor R install
  # # # nuke cache dirs before installing pkgs; tip from Dirk E fixes broken img
  # # RUN rm -f /var/lib/dpkg/available && rm -rf  /var/cache/apt/*

  # # # same set of packages for both devel and release
  # # RUN apt-get update && \
  # # 	apt-get -y --no-install-recommends install --fix-missing \
  # # 	gdb \
  # # 	libxml2-dev \
  # # 	python-pip \
  # # 	libz-dev \
  # # 	liblzma-dev \
  # # 	libbz2-dev \
  # # 	libpng-dev \
  # # 	libmariadb-client-lgpl-dev \
  # # 	&& rm -rf /var/lib/apt/lists/*

  # # # issues with '/var/lib/dpkg/available' not found
  # # # this will recreate
  # # RUN dpkg --clear-avail

  # # # # Add bioc user as requested
  # # # RUN useradd -ms /bin/bash -d /home/bioc bioc \
  # # # 	&& echo "bioc:bioc" | chpasswd && adduser bioc sudo
  # # # USER bioc
  # # # RUN mkdir -p /home/bioc/R/library && \
  # # # 	echo "R_LIBS=/usr/local/lib/R/host-site-library:~/R/library" | cat > /home/bioc/.Renviron
  # # # USER root
  # # # RUN echo "R_LIBS=/usr/local/lib/R/host-site-library:\${R_LIBS}" > /usr/local/lib/R/etc/Renviron.site \
  # # # 	&& echo "R_LIBS_USER=''" >> /usr/local/lib/R/etc/Renviron.site \
  # # # 	&& echo "options(defaultPackages=c(getOption('defaultPackages'),'BiocManager'))" >> /usr/local/lib/R/etc/Rprofile.site

  # # # # add R packages test
  # # # RUN R -e "install.packages('methods',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  # # # RUN R -e "install.packages('jsonlite',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  # # # RUN R -e "install.packages('tseries',dependencies=TRUE, repos='http://cran.rstudio.com/')"
  #RSEM
  #RUN cd /opt/
  #RUN git clone https://github.com/deweylab/RSEM.git && cd RSEM && make && make install
  #ENV PATH /opt/RSEM:$PATH

  # # # # install skewer
  # # # RUN \
  # # #   wget -c https://downloads.sourceforge.net/project/skewer/Binaries/skewer-0.2.2-linux-x86_64 && \
  # # #   chmod +x skewer-0.2.2-linux-x86_64 && \
  # # #   cp skewer-0.2.2-linux-x86_64 /usr/local/bin/skewer

  # # # run update and install necessary tools
  # # RUN apt-get update -y && apt-get install -y \
  # #     build-essential \
  # # RUN apt-get install -y software-properties-common
  # # RUN add-apt-repository -y ppa:jonathonf/python-3.6
  # # RUN apt-get update && apt-get install -y python3.6
  #
    ### ^^^ BUILDS INDEPENDENTLY VALIDATED ABOVE THIS POINT ^^^ ###
    ### Last successful build 2020-01-14 15:26 CST              ###
    #
    #
    ### TESTING ###
    ### Installs below this point are not individually validated and my require dependencies from above
    # multiqc
    ## Set the locale
    ## https://stackoverflow.com/questions/28405902/how-to-set-the-locale-inside-a-debian-ubuntu-docker-container#28406007
    #
  RUN pip install -U multiqc

  ##############
  ### simNGS ###
  ##############
  #
  #RUN apt-get update -qq
  #RUN apt-get install -qq --no-install-recommends --allow-unauthenticated \
  #    libblas-dev \
  #    liblapack-dev \
  #    make \
  #    tar \
  #    wget
  #RUN cd /opt && wget --no-check-certificate #https://www.ebi.ac.uk/goldman-srv/simNGS/current/simNGS.tgz && tar -xvzf #simNGS.tgz
  #RUN cd /opt/simNGS/src && make -f Makefile.linux
  #ENV PATH="/opt/simNGS/bin:${PATH}"
  #
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install(
  #
  # WORKS TO HERE [2020-09-21]
  #
  # # GISTIC 2.0
  # RUN mkdir -p /opt/GISTIC2 \
  #     && cd /opt/GISTIC2 \
  #     && wget --no-check-certificate ftp://ftp.broadinstitute.org/pub/GISTIC2.0/GISTIC_2_0_23.tar.gz \
  #     && tar --owner=root --group=root -vxf GISTIC_2_0_23.tar.gz
  # RUN cd /opt/GISTIC2/MCR_Installer \
  #     && unzip MCRInstaller.zip \
  #     && ./install -mode silent -agreeToLicense yes -destinationFolder /opt/GISTIC2
  # RUN cd /opt/GISTIC2 \
  #     && wget --no-check-certificate ftp://ftp.broadinstitute.org/pub/GISTIC2.0/hg19.mat
  # ENV PATH="/opt/GISTIC2/:${PATH}"
  # RUN chown -R root:root /opt/GISTIC2
  # ENV MCR_ROOT=/opt/GISTIC2/MATLAB_Compiler_Runtime
  #

  #
  #RUN conda install -c bioconda mirge3
  #
  #RUN conda create -n py2 python=2.7
  #RUN echo "source activate py2" > ~/.bashrc
  #ENV PATH /opt/conda/envs/py2/bin:$PATH
  #
  #########1#########2#########3#########4#########5#########6#########7#########8
  #

  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  # ichorCNA
  ##
  ## linux dependencies
  RUN apt-get update \
     && apt-get install -y \
     libcurl4-openssl-dev \
     libssl-dev \
     libxml2-dev
  #RUN rm /usr/lib/x86_64-linux-gnu/libcurl.so.4
  #RUN ln -s /usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0 /usr/lib/x86_64-linux-gnu/libcurl.so.4
  ##
  ## R dependencies
  RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("HMMcopy"); BiocManager::install("GenomeInfoDb"); BiocManager::install("GenomicRanges");'
  ##
  ## git clone install
  RUN cd /opt \
      && git clone https://github.com/broadinstitute/ichorCNA.git \
      && cd ichorCNA \
      && R CMD INSTALL . \
      && cd /opt
  ##
  #RUN echo "source activate py2" > ~/.bashrc
  #ENV PATH /opt/conda/envs/py2/bin:$PATH
*** [[file:docker/rna_Dockerfile][RNA docker]]
:PROPERTIES:
:header-args:bash: :tangle ./docker/rna_Dockerfile
:ID:       1338f10d-c3d0-4b29-9c4a-d3a01758a3d2
:END:
**** Preamble
#+begin_src bash
FROM jeszyman/biotools:2.0.0
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build an RNA-focused docker image                  ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/rna
# (https://hub.docker.com/repository/docker/jeszyman/rna)

#+end_src
**** Conda
#+begin_src bash
RUN conda install -c bioconda kallisto
RUN conda install -c bioconda rsem
RUN conda install -c bioconda star
RUN conda install -c bioconda subread
RUN conda install -c bioconda salmon
#+end_src
**** R and Bioconductor
#+begin_src bash
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("DESeq2", \
                                 "edgeR", \
                                 "fgsea", \
                                 "limma", \
                                 "tximport"));'

#+end_src
#+begin_src bash
RUN R -e "install.packages(c('statmod', repos='http://cran.rstudio.com/'))"

#+end_src
**** Custom installs
#+begin_src bash
RUN cd /opt && git clone https://github.com/rajewsky-lab/mirdeep2 && cd /opt/mirdeep2 && perl install.pl

RUN /bin/bash

RUN cd /opt/mirdeep2 && perl install.pl

ENV PATH="/opt/mirdeep2/bin:${PATH}"
#+end_src
*** [[file:docker/atac_Dockerfile][ATAC-seq docker]]
:PROPERTIES:
:header-args:bash: :tangle ./docker/atac_Dockerfile :tangle-mode (identity #o555)
:ID:       57458bd3-005f-4342-ada7-58c55a74d7d0
:END:
:LOGBOOK:
CLOCK: [2022-08-30 Tue 11:43]--[2022-08-30 Tue 11:57] =>  0:14
:END:
**** Preamble
#+begin_src bash
FROM jeszyman/biotools:1.0.2
#########1#########2#########3#########4#########5#########6#########7#########8
###                                                                          ###
###                  Dockerfile for ATAC-seq Related Software                ###
###                                                                          ###
#########1#########2#########3#########4#########5#########6#########7#########8

#################
###   Notes   ###
#################

# After build, the image will be pushed to dockerhub as jeszyman/atac
# (https://hub.docker.com/repository/docker/jeszyman/atac)

#+end_src
**** Apt
#+begin_src bash

###############
###   APT   ###
###############

# Update
RUN apt-get -y update && apt-get upgrade -y

# Linux prerequisites for ATACseqQC
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated --fix-missing \
    libgsl-dev \
    libcurl4-openssl-dev
#+end_src
**** Conda
#+begin_src bash
#################
###   Conda   ###
#################

RUN conda install -c bioconda macs2

#+end_src
**** R and bioconductor
#+begin_src bash
##############################
###   R and Bioconductor   ###
##############################

# CRAN
RUN R -e "install.packages(c('statmod', repos='http://cran.rstudio.com/'))"

# Bioconductor
RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("ATACseqQC", \
                                 "ChIPpeakAnno", \
                                 "ChIPseeker", \
                                 "clusterProfiler", \
                                 "csaw"));'

#+end_src
**** Custom installs
#+begin_src bash
###########################
###   Custom Installs   ###
###########################

# Homer
RUN mkdir /opt/homer
RUN cd /opt/homer
RUN wget http://homer.ucsd.edu/homer/configureHomer.pl -O /opt/homer/configureHomer.pl
RUN perl /opt/homer/configureHomer.pl -install
ENV PATH="${PATH}:/opt/homer/bin/"
RUN perl /opt/homer/.//configureHomer.pl -install mouse-p

#+end_src
****                                                                    :dev:
get updated bowtie 2 with multithread indexing (>2.2.6?)
https://www.biostars.org/p/219541/

*** Truseq small adapters
- supercite:illumina2016truseq_small
** Technologies, kits, etc
*** Truseq RNA Exome
:PROPERTIES:
:ID:       5fe9c572-c692-4aad-bb48-8b2fb2b86760
:END:
- [[cite:&larson2021]]
- [[https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/truseq-rna-access.html][Illumina link]]
  - TruSeq RNA Exome kit (Illumina, 20020189) 2x @ 2480
  - enrichments 2x @ 2995
  - indexes 2x @ 250
** References, databases, standards, file types, and data sets
:PROPERTIES:
:ID:       FCBA0863-010A-448C-A33C-941D2AA45AD5
:END:
- https://www.biostars.org/p/181014/
- https://www.genomicinformationcommons.org/
- https://ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Mus_musculus/GRCm38.p3/seqs_for_alignment_pipelines/
- https://www.ncbi.nlm.nih.gov/grc/help/faq/#format-reference-data-for-read-alignment
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19
- https://www.illumina.com/science/genomics-research/articles/dragen-demystifying-reference-genomes.html
- methylation https://www.nature.com/articles/s41586-022-05580-6
- Cytobands- # https://www.biostars.org/p/383786/
- GENIE
  - https://www.synapse.org/#!Synapse:syn7222066/wiki/410922
  - https://aacrjournals.org/cancerdiscovery/article/12/9/2044/708766/AACR-Project-GENIE-100-000-Cases-and-BeyondGENIE
- https://www.biostars.org/p/383786/
- https://biodbnet-abcc.ncifcrf.gov/
- https://snpedia.com/index.php/SNPedia
- https://academic.oup.com/nar/article/43/D1/D1117/2439489
- Uniprot
  - https://www.uniprot.org/help/query-fields
*** MSigDB
:PROPERTIES:
:ID:       c79bfe84-0f15-474e-8a22-1f210605ed5a
:END:
- [[id:edf39ef6-df91-40c2-a71a-c612d60b5582][msigdbr]]
- [[https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp][MSigDB]] [[https://data.broadinstitute.org/gsea-msigdb/msigdb/][data]]
- http://software.broadinstitute.org/gsea/msigdb/collections.jsp
*** Chromosome sizes
:PROPERTIES:
:ID:       8a50f360-6043-4e19-9620-61f3b957b28a
:END:
  - http://hgdownload.cse.ucsc.edu/goldenpath/mm10/bigZips/mm10.chrom.sizes
*** Genome reference builds and associated sequences
:PROPERTIES:
:ID:       e9e201b5-4a94-4861-8de8-803676c3a37f
:END:
- UCSC is chr, ensembl is NOT
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19
- https://www.ncbi.nlm.nih.gov/tools/vecscreen/univec/
*** By substrate
:PROPERTIES:
:ID:       ffe7a0f8-5e6d-45ff-adec-efbba7c44783
:END:
**** RNA
:PROPERTIES:
:ID:       59c43aa9-cfa1-4e03-b5fb-25c74f626352
:END:
cite:consortium2020
*** Adapter sequences
:PROPERTIES:
:ID:       cf2ea58e-8751-4d90-ab08-a63445925c89
:END:
- http://www.science.smith.edu/cmbs/wp-content/uploads/sites/36/2020/01/illumina-adapter-sequences-1000000002694-11.pdf
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5759201/
- https://support.illumina.com/downloads/illumina-adapter-sequences-document-1000000002694.html
- https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html
*** [[id:c0c0ee28-2e41-41a7-9a3b-ae195117a93e][LINK TO common bioinformatics file manipulation]]
:PROPERTIES:
:ID:       fc44f2e8-c32f-4169-ba9c-5032a35327a6
:END:
*** [[id:767e5e30-8490-4cda-8018-8b0bd91c4ef3][LINK TO Code associated with sequence reference builds]]
:PROPERTIES:
:ID:       a7a6c48c-f2a5-478e-858d-c99ebd9be027
:END:
*** By organism
:PROPERTIES:
:CREATED:  [2020-10-26 Mon 12:18]
:ID:       33638cdf-113c-4248-aff4-b0d8ea7cbc09
:END:
**** Rat
:PROPERTIES:
:ID:       f4ff99c8-9149-4183-89ff-56a09e68943b
:END:
- Genome assemblies
  - https://www.ncbi.nlm.nih.gov/assembly/organism/10116/latest/
- https://useast.ensembl.org/Rattus_norvegicus/Info/Index
- most recent assembly is Rnor_6.0
  - https://rgd.mcw.edu/rgdweb/report/genomeInformation/genomeInformation.html?species=Rat&mapKey=360&details=true
- https://rgd.mcw.edu/
**** human
:PROPERTIES:
:ID:       8a910eca-5f96-4a74-ad3e-efc6eb748b7c
:END:
- http://browser.1000genomes.org/index.html
**** mouse
:PROPERTIES:
:ID:       23fc30cd-7e45-4333-8740-1d8ec0755dab
:END:
- https://github.com/czbiohub/tabula-muris
- [[http://www.informatics.jax.org/vocab/gene_ontology][jax mouse gene ontology browser]]
- cite:lai2016
- https://www.biorxiv.org/content/10.1101/082511v1#disqus_thread
*** by data type
:PROPERTIES:
:ID:       5f2913ae-1f4a-44d5-90e2-9ee045c37b78
:END:
- extracellular RNA
  - https://exrna.org/resources/data/
- protein - http://www.proteinatlas.org/about
- integrated
  - https://www.nature.com/articles/s41586-019-1629-x
- [[id:333ebad6-9ed4-4d5b-a50e-f9fe9a8c3a92][gene expression]]
- [[id:629e512e-60ac-4283-bc0e-028b8cad3dcc][sequence variation]]
- miRNA
  - http://www.mirbase.org/cgi-bin/sentences.pl?acc=MI0001445
*** reference data
:PROPERTIES:
:ID:       4dc2f252-12fe-409e-bef0-aab2d765de7a
:END:
- homology
  - http://bioinf.wehi.edu.au/MSigDB/
  - cite:lai2016
    - https://bioconductor.org/packages/release/data/experiment/vignettes/gskb/inst/doc/gskb.pdf
    - http://ge-lab.org/gskb/
  - http://www.informatics.jax.org/homology.shtml
- gene sets and pathways
  - cite:liberzon2015
  - cite:lai2016
  - KEGG- note mostly orthology mapping for non-human organisms
  - http://pathguide.org/
  - cite:chowdhury2015
  - https://www.bioconductor.org/packages/devel/data/experiment/vignettes/gskb/inst/doc/gskb.pdf
  - - https://www.bioconductor.org/packages/release/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
  - MSigDB
  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4707969/
  - [[https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/MSigDB_v7.0_Release_Notes#Changes_to_MSigDB_Gene_Symbol_Mapping_Procedures][orthology mapping]]
- https://github.com/genome/gms/wiki/HCC1395-WGS-Exome-RNA-Seq-Data
- [[https://github.com/genome-in-a-bottle][Genome-in-a-bottle]]
  - https://github.com/genome-in-a-bottle
- http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/
- http://www.mirbase.org/
- annotaiton
  - http://daid.abcc.ncifcrf.gov/
- alignment
  - terminology
    - abinitio is predicted genes
  - organizations
    - Ensembl
  - [[https://support.illumina.com/sequencing/sequencing_software/igenome.html][illumina iGenomes refrence sequence and annotation]]
  - [[https://software.broadinstitute.org/gatk/documentation/article?id=11010][gatk:  Human genome reference builds - GRCh38/hg38 - b37 - hg19]]
  - https://www.ncbi.nlm.nih.gov/grc/help/faq/
  - https://www.ncbi.nlm.nih.gov/grc/help/definitions/
  - https://www.ccdatalab.org/blog/2019/3/29/gene-expression-repos-explained
  - https://useast.ensembl.org/info/data/ftp/index.html
  - Genome Reference Consortium
    - https://www.slideshare.net/ShaojunXie/hg19-grch37-vs-hg38-grch38
  - Ensembl
    - [[https://useast.ensembl.org/info/data/ftp/index.html][Ensembl reference FTP]]
    - https://useast.ensembl.org/info/genome/variation/
*** individual subject / sample level data
:PROPERTIES:
:ID:       55d196dd-c99f-4ce8-b8e6-290b778e15b3
:END:
- [[https://www.ebi.ac.uk/ena][European Nucleotide Archive]]
- https://www.ncbi.nlm.nih.gov/geo/
- [[http://www.ebi.ac.uk/gwas/][The NHGRI-EBI Catalog of published genome-wide association studies]]
- https://www.ebi.ac.uk/gxa/experiments/E-MTAB-513/Results
- SRA (sequence read archive)
  - https://edwards.sdsu.edu/research/fastq-dump/
- dbGaP
  - https://www.ncbi.nlm.nih.gov/books/NBK99225/#DArequest.could_you_give_a_demo_about_th
- [[https://www.ccdatalab.org/blog/2019/3/29/gene-expression-repos-explained][gene expression repos explained]]
- [[https://www.encodeproject.org/][ENCODE]]
  - http://www.reddit.com/r/askscience/comments/znlk6/askscience_special_ama_we_are_the_encyclopedia_of/

- cite:beck2015
- [[https://www.reddit.com/r/bioinformatics/comments/601fzr/where_can_i_access_free_sequencing_data/][Where can I access free sequencing data]]
- [[https://www.aacr.org/professionals/research/aacr-project-genie/][GENIE]]
**** GDC: Genomic Data Commons
:PROPERTIES:
:ID:       99eace6a-6b4e-4acf-9381-1b48a59a4b61
:END:
- https://gdc.cancer.gov/about-data/data-types-and-file-formats
- https://github.com/NCI-GDC/gdc-workflow-overview
- https://gdc.cancer.gov/about-data/data-standards
- https://docs.gdc.cancer.gov/Data/Data_Model/GDC_Data_Model/
- https://gdc.cancer.gov/about-data/data-dictionary
- [[https://gdc.cancer.gov/][NCI Genomic Data Commons]]
  - [[https://docs.gdc.cancer.gov/][GDC documentation]]
  - [[https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization/genomic-data-alignment/rna-seq-pipeline][GDC RNA-seq pipeline]]
  - [[https://cbiit.nci.nih.gov/ncip/nci-cancer-genomics-cloud-pilots/nci-cloud-initiative][GDC cloud pilots]]
  - - https://gdc.cancer.gov/developers
  - https://docs.gdc.cancer.gov/
  - - https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization/genomic-data-alignment/rna-seq-pipeline
  - https://cbiit.nci.nih.gov/ncip/nci-cancer-genomics-cloud-pilots/nci-cloud-initiative
- http://www.cbioportal.org/tutorials
*** sequence variation
:PROPERTIES:
:ID:       629e512e-60ac-4283-bc0e-028b8cad3dcc
:END:
- [[http://firebrowse.org/?cohort=ACC&download_dialog=true][firehose]]
- http://browser.1000genomes.org/index.html
- Variance
  - gnomAD
    - - https://mail.google.com/mail/u/0/#inbox/FMfcgxwBWKXrqjHPFNdbSqJRRRdhzdGf
    - https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad/
    - (Subsumes ExAC)
  - ClinVar
    - [[https://www.ncbi.nlm.nih.gov/projects/clinvar/ClinVarDataDictionary.pdf][ClinVar Data Dictionary]]
  - phenotype association
    - https://genematcher.org/?tab=submissionsTab
    - [[https://www.ncbi.nlm.nih.gov/gap][dbGaP: database of Genotypes and Phenotypes]]
  - cBioPortal
    - http://www.cbioportal.org/tutorials
    - http://www.cbioportal.org/rmatlab
  - [[https://www.ncbi.nlm.nih.gov/projects/SNP/index.html][dbSNP]]
  - [[http://exac.broadinstitute.org/][ExAC Exome Aggregation Consortium]]
- polymorphism
  - [[http://www.internationalgenome.org/data#download][IGSR at 1000 Genomes]]
- [[https://civic.genome.wustl.edu/#/home][CIViC]]
  - cite:griffith2017civic
*** gene expression
:PROPERTIES:
:ID:       333ebad6-9ed4-4d5b-a50e-f9fe9a8c3a92
:END:
https://www.nature.com/articles/s41598-020-72664-6
- https://github.com/czbiohub/tabula-muris
- [[http://firebrowse.org/?cohort=ACC&download_dialog=true][firehose]]
- [[https://gtexportal.org/home/datasets][GTEx]]
  - [[https://biospecimens.cancer.gov/resources/sops/][GTEx SOPs]]
  - GTEx currently uses the
    - GRCh37/Hg19 genome
    - GENCODE v19 annotation
      - The GENCODE annotation should be patched to use Ensembl chromosome names: zcat gencode.v19.annotation.gtf.gz | sed 's/chrM/chrMT/;s/chr//' > gencode.v19.annotation.patched_contigs.gtf
  - sample ID
    - such as GTEX-14753-1626-SM-5NQ9L?
    - GTEX-YYYYY GTEx donor ID
    - 2nd number mostly refers to the tissue site, BUT we do not recommend using it for tissue site designation.
    - 3rd number is the RNA or DNA aliquot ID used for sequencing.
  - Unstranded
  - GTEx
    - [[https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000424.v7.p2][at dbGAP]]
*** public data, data mapping
:PROPERTIES:
:CREATED:  [2020-07-02 Thu 13:18]
:ID:       7b386a21-a6b8-49d1-ad4b-f5976bf30666
:END:
:LOGBOOK:
CLOCK: [2020-07-02 Thu 13:19]--[2020-07-02 Thu 13:29] =>  0:10
:END:
- https://www.r-bloggers.com/converting-mouse-to-human-gene-names-with-biomart-package/
- geo submission
  - https://www.ncbi.nlm.nih.gov/geo/info/seq.html
  - [[http://yeolab.github.io/onboarding/geo.html][see for useful ftp commands]]
  - [[https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE138253][example from Rentschler lab]]
*** Sequence Databases and Data Sets
:PROPERTIES:
:ID:       C137CB28-0517-4A98-8FB4-6D312DDBC325
:END:
- https://tabula-muris.ds.czbiohub.org/
- [[http://www.mitomap.org/MITOMAP][MITOMAP human mitochondrial genome]]
- https://portal.gdc.cancer.gov/
- [[https://www.ebi.ac.uk/ega/datasets/EGAD00001000078][EBI Data Browser]]
- [[http://www.internationalgenome.org/sites/1000genomes.org/files/documents/1000genomes_browser_quickstart.pdf][100s genomes browser]]
- [[https://genome.ucsc.edu/training/vids/index.html#vid08][ucsc training videos]]
- [[http://useast.ensembl.org/info/website/tutorials/sequence.html][ensembl sequence tutorials]]
- General Feature Format (GFF) / GTF Annotation Files
  - [[http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=494995959_n1VrbO9xpH51JE8qLga6fCvOyV2R&clade=mammal&org=Mouse&db=mm9&hgta_group=genes&hgta_track=refGene&hgta_table=0&hgta_regionType=genome&position=chr12%3A57795963-57815592&hgta_outputType=primaryTable&hgta_outFileName=][make your own]]
  - a tab-separated value file
  - see for standard: http://mblab.wustl.edu/GTF22.html

**** Stored on neuro server at G:/reference
:PROPERTIES:
:ID:       F08101C9-F675-4A35-B9A8-D0159FD74AFF
:END:

- mouse
  - gencode fa.gz
- human
  - hg38.fa and

**** Genomes
:PROPERTIES:
:ID:       277AAE94-CFE5-400E-B5B9-291677A513D0
:END:
- https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19#:~:text=The%20latest%20build%20of%20the,the%20repertoire%20of%20ALT%20contigs.
- [[https://www.biostars.org/p/4405/][reasons to avoid soft-masked genomes for alignment]]
- [[http://hgdownload.cse.ucsc.edu/downloads.html][UCSC goldenPath bigZips]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/][Mouse mm10]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/mm9/bigZips/][Mouse mm9]]]]
***** Custom Genomes
:PROPERTIES:
:ID:       F5506BF8-8A6E-4949-9F5E-15B21511E87E
:END:

- No hits for "custom" in cite:star_2.5_dobin or cite:dobin2015mapping



https://www.biostars.org/p/111012/
http://mblab.wustl.edu/GTF2.html
http://useast.ensembl.org/info/website/upload/gff.html#fields
https://www.biostars.org/p/182230/
https://wiki.galaxyproject.org/Learn/CustomGenomes
http://wiki.bits.vib.be/index.php/Create_a_valid_gtf_file_for_tophat
https://support.bioconductor.org/p/66379/

****** Generating a Custom Genome
:PROPERTIES:
:ID:       11F1FF38-7F23-4B81-AEA3-AA760DC62C9F
:END:

******* Trial w/ EWSR1-FLLI1
:PROPERTIES:
:ID:       D293BED1-C843-476E-891B-E8FEFB4917F3
:END:

- Looking for help
  - no info
    - cite:beadling16_multip_amplic_approac_detec_gene
  - bad ideas
    - don't align to transcriptome- this doesn't save any time
    - don't try to make a gtf from a fa- the info isn't there
- Sequence sources
  - Genome- Downloaded GRCh37.85rm fasta from [[http://ftp.ensembl.org/pub/grch37/release-85/fasta/homo_sapiens/dna/][here]]
  - Fusions
    - from ensembl
      - [[http://useast.ensembl.org/Homo_sapiens/Transcript/Exons?db=core;g=ENSG00000182944;r=22:29268009-29300522;t=ENST00000332050][EWSR1 Exon 7]]
        - coord- 29,286,923	29,287,134	2	1	212
        - seq- CTATTCCTCTACACAGCCGACTAGTTATGATCAGAGCAGTTACTCTCAGCAGAACACCTATGGGCAACCGAGCAGCTATGGACAGCAGAGTAGCTATGGTCAACAAAGCAGCTATGGGCAGCAGCCTCCCACTAGTTACCCACCCCAAACTGGATCCTACAGCCAAGCTCCAAGTCAATATAGCCAACAGAGCAGCAGCTACGGGCAGCAGA
      - FLI1 exon - _complicated by multiple splice variants_
    - failed
      - not from UCSC (yet), it will output the whole gene sequence
      - ?from bed files?
      - EWSR1-FLI1 fusion sequence- COSMIC-Gene-Analysis-Fusions-by prevalance : http://cancer.sanger.ac.uk/cosmic/fusion/summary?id=166
- Thought about splitting this to make a little genome to play with, but splits just get me way too many files, a lot with NNNNN only
  - [[Split text file by percentage  the file from 533,942,310 lines to 10,000 lines- 53394 files
- Adding sequences to a fasta][Added fusion sequence programmatically

***** Human genome builds
:PROPERTIES:
:ID:       D0BD8791-0AE3-472E-A346-C0DD0D79AFD9
:END:

- [[http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/][Human hg38]]
- [[http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/][Human hg19]]
- [[http://useast.ensembl.org/info/website/tutorials/grch37.html][for converting GRCh37 to GRCh38]]
- https://www.biostars.org/p/113100/
  - The actual sequences you'll get from NCBI/UCSC/Ensembl will be identical, but their annotations will be different and (importantly) updated at different frequencies. NCBI's annotation is the "refseq" dataset (the "refGene" track in UCSC), which is essentially a subset of the UCSC and Ensembl annotations. UCSC's annotations are kind of a mess.
****** hg19
:PROPERTIES:
:ID:       5a7cdb22-c982-4daa-b0ef-799f7347a683
:END:
:LOGBOOK:
CLOCK: [2020-10-05 Mon 13:24]--[2020-10-05 Mon 13:31] =>  0:07
:END:
- hg19 copy-number accessible whitelist
  - subset to autosomes
  - subset against duke
****** Ensembl GRCh37 fastas
:PROPERTIES:
:ID:       B6066026-81E3-49F3-9A0E-C5E610CB86AB
:END:

[[ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/README][Readme]]

hard masked- low complexity regions by RepeatMasker replaced with N's
soft masked- nucleotide replaced w/ lower case
https://www.biostars.org/p/122305/
[[https://groups.google.com/forum/#!topic/rna-star/2wdHXaPv_vU][Dobin suggests hard masking?]]

***** by Institute
:PROPERTIES:
:ID:       48D2EFA9-39A0-4D5F-8AFA-4726E9488D60
:END:

    - genome reference consortium
      - [[http://www.slideshare.net/GenomeRef][genome reference consortium presentations]]
      - [[http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/human/][human genome overview]]

        GRC distributes the reference genome in two versions: one as contigs and the other as assembled chromosomes. The latter is in the "assembled_chromosome" directory. I do not know who are using the contigs, but nearly everyone I know use assembled chromosomes only.

        GRCh37/hg19 and GRCh38 are genome builds rather than annotations, which describe where features are in a given genome build. The actual sequences you'll get from NCBI/UCSC/Ensembl will be identical, but their annotations will be different and (importantly) updated at different frequencies. NCBI's annotation is the "refseq" dataset (the "refGene" track in UCSC), which is essentially a subset of the UCSC and Ensembl annotations. UCSC's annotations are kind of a mess. You'll find genes with the same ID on multiple strand and multiple chromosomes, which makes them a bit useless. Ensembl's annotations typically contain more features than UCSC (so a bit more noise), but they're otherwise much better put together (e.g., you'll never find a gene ID on different strand or different chromosomes) and their IDs are typically easier to map to other things (e.g., gene names, GO and pathway memberships). Ensembl also updates its annotation fairly often and versions everything nicely, so it's quite convenient to report what version you used in a paper (reproducibility is always a good thing). Given the choice, use the Ensembl annotation.

        BTW, don't forget that the various sources can use different names for chromosomes (e.g., chr1 in UCSC is just 1 in Ensembl), so don't mix and match them.

***** 37=hg19 sort of
:PROPERTIES:
:ID:       604441C8-6BCE-426F-9DE9-5D960F75C3D3
:END:

https://www.biostars.org/p/123767/
I believe the genomic content for the two is identical, except for the mitochondrial contig.

The contig names are also different. GRCh37 names them `chr1`, `chr2`,,`chr3`, etc, while hg19 just has `1`, `2`, `3`.

Thus you can use the same GTF file for both (excluding mitochondrial, of course) if you do a simple replace operation for the contig names.

https://wiki.dnanexus.com/Scientific-Notes/human-genome

***** [[http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/][GRC GR/scratch/jeszyman/f/scratch/jeszyman/fishseq/set1/S14-43147_2_sequence.fastq ishseq/set1/S14-43147_2_sequence.fastq Cm38]]
:PROPERTIES:
:ID:       A57F0012-6870-4F79-92A3-C580DB1D33AD
:END:

***** [[http://useast.ensembl.org/info/data/ftp/index.html?redirect=no][Ensembl]]
:PROPERTIES:
:ID:       D18360E5-0B07-4AB6-92F6-D4B717A2E159
:END:

******
:PROPERTIES:
:ID:       B94A8093-3229-41A6-92A6-104D03128BAE
:END:

*******
:PROPERTIES:
:ID:       D47888B2-1210-4C9D-B30C-81672143987A
:END:

*******
:PROPERTIES:
:ID:       68A22065-D788-4653-9BBC-2B5F67AD218A
:END:

*******
:PROPERTIES:
:ID:       687708C8-78BF-4EB9-8C48-F5456E9CF63D
:END:

*******
:PROPERTIES:
:ID:       B0449C36-57B8-4A2A-AE44-67C96C51706D
:END:

*** Clinical Fusion Protein Databases
:PROPERTIES:
:ID:       51EC0C78-7164-43CA-97B1-1FF2205B46A7
:END:

chimer-
cite:Kim01012010

cite:wang2015fusioncancer

cite:korla2015fare

see Wellcome Trust db and Chimerdb 2.0- cite:scolnick2015efficient

*** Other Databases and Data Sets
:PROPERTIES:
:ID:       396F7A49-ACB4-4F81-8C73-ACC907207BB8
:END:
- DGIdb
  - - https://pubmed.ncbi.nlm.nih.gov/33237278/
- mouse
  - [[http://www.informatics.jax.org/][Jackson Lab mouse genome informatics]]
  - [[http://web.stanford.edu/group/barres_lab/brain_rnaseq.html][mouse cerebral cortex RNA Seq database]]
  - mouse Ensembl genome files
- any
  - for datasets
  - GEO and SRA
  - [[http://www.ebi.ac.uk/arrayexpress/about.html][ArrayExpress]]
*** Bioinformatics File Formats
:PROPERTIES:
:ID:       5e2c4e39-99bc-4cc8-86a6-c39bea5ac2d8
:END:
- https://genome.ucsc.edu/FAQ/FAQformat.html#format4
- [[https://genome.ucsc.edu/FAQ/FAQformat.html#format1][Nucleic acid data formats]]
- https://help.basespace.illumina.com/articles/descriptive/fastq-files/
- alignment formats
  - [[https://software.broadinstitute.org/gatk/documentation/article?id=11014][GATK explaination of alignment formats]]
  - https://samtools.github.io/hts-specs/CRAMv3.pdf
- https://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp_v2/Content/Vault/Informatics/Sequencing_Analysis/BS/swSEQ_mBS_FASTQFiles.htm
- bcl file format?
- mutation annotation format
  - maf format viewer https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html
- gff / gtf
  - [[http://www.gencodegenes.org/gencodeformat.html][Gencode gtf fields]]
  - [[https://www.ensembl.org/info/genome/genebuild/genome_annotation.html#Names][Ensembl annotation]]
    - note that chr.gtf and gtf are same https://www.biostars.org/p/217700/
  - [[https://useast.ensembl.org/info/website/upload/gff.html][Ensembl GFF/GTF format]]
- [[https://genome.ucsc.edu/goldenpath/help/bigWig.html][bigWig format]]
- Variant call format (VCF)
  - http://samtools.github.io/hts-specs/VCFv4.3.pdf
  - http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/VCF%20(Variant%20Call%20Format)%20version%204.0/encoding-structural-variants
  - [[http://samtools.github.io/hts-specs/VCFv4.3.pdf][Current documentation]]
  - [[http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/][VCF page at IGSR]]
- [[https://www.biostars.org/p/84686/][Obi on 1 vs 0 based coordinates]]
- cite:li2009
**** Sequence file formats
:PROPERTIES:
:ID:       4a86acd3-db60-4a63-979e-43e34070e7c7
:END:

- fasta
  - ("genomic" fasta .fna == .fa)
***** fastq
:PROPERTIES:
:ID:       59bc8f81-cf57-4e56-a5ff-3b0ff4095a91
:END:
- https://en.wikipedia.org/wiki/FASTQ_format
- [[https://www.biostars.org/p/198143/][get sequencer type from fastq header]]
- https://github.com/10XGenomics/supernova/blob/master/tenkit/lib/python/tenkit/illumina_instrument.py#L12-L45
- https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/FileFormat_FASTQ-files_swBS.htm

**** .bed
:PROPERTIES:
:ID:       6DB65B82-D88C-43EA-9D2B-169D9184C9D5
:END:
**** SAM
https://www.biostars.org/p/356811/
**** CIGAR- a subformat in SAM w/ three operations
:PROPERTIES:
:ID:       A664A1C3-9130-400D-BE32-D7D58CF80537
:END:

Original CIGAR
match/mismatch
insertion
deletion
Extended CIGAR
N for skipped bases on the reference- how is this different than an insertion?
S for soft clipping
H for hard clipping
P for padding


read in IGV
parameters
reads up to 128 Mbp
tab delimited
per-alignment fields
mandatory

**** Sequence alignment/Map format specification (.sam/.bam)
:PROPERTIES:
:ID:       44889954-63D8-4416-9F28-0691ECF99A9B
:END:
- [[https://genome.sph.umich.edu/wiki/SAM][nice visualization of fields]]
- cite:samspec2017
  - 1
    - 1.3 The header section
      - read groups
        -
- See [[Samtools]]
- sam- fastq with alignment column
  - BAM- companion to SAM in compressed binary
    - .bai is a companion file which contains the index
      - index acts like an external table of contents, and allows programs to jump directly to specific parts of the bam file without reading through all of the sequences. Without the corresponding bam file, your bai file is useless, since it doesn't actually contain any sequence data.
- read groups
  - https://gatkforums.broadinstitute.org/gatk/discussion/6472/read-groups
  - effectively treated as a separate run of the NGS instrument in tools like base quality score recalibration -- all reads within a read group are assumed to come from the same instrument run and to therefore share the same error model
  - GATK tools treat all read groups with the same SM value as containing sequencing data for the same sample
  - https://gatkforums.broadinstitute.org/gatk/discussion/6472/read-groups

** Scripts :hierarchy:
*** Bash helper scripts [[file:tools/shell/]]
**** Built docker container
:PROPERTIES:
:ORDERED:  t
:END:
#+begin_src bash :tangle ./tools/shell/docker_build.sh
#!/usr/bin/env bash

set -o errexit   # abort on nonzero exitstatus
set -o nounset   # abort on unbound variable
set -o pipefail  # don't hide errors within pipes

print_usage(){
    cat <<- EOF

###################################################
###   Docker and Singularity Building Command   ###
###################################################

Steps
1) Builds docker container
2) Pushes container to dockerhub
3) Builds singularity file
4) Pushes singularity file to box

usage: docker_build.sh [options] <DOCKER FILE> <DOCKER TAG> [OPTIONAL - DOCKER BUILD LOC]
  options:
    -n  No cache, use for validation of full build
    -s  A sif directory location
    -v  Docker version, a version appended to docker and singularity tags
        Note: adding a version will push the singularity build to
        shared/singularity-containers/stable

example:

docker_build.sh -n -s ~/sing_containers -v 2.0.0 ~/dockerfiles/atac_Dockerfile jeszyman/atac ~/docker_bulid_dir

EOF
}

# Setup options
while getopts "nt:v:s:" options;
do
    case "${options}" in
        n) usecache="--no-cache" ;;
        v) version=${OPTARG} ;;
        s) sifdir=${OPTARG} ;;
        h | *) print_usage; exit 0 ;;
    esac
done

shift $((OPTIND -1))

[[ $# -eq 0 ]] && print_usage && exit 1

# Scipt variables
dockerfile=$1
tag=$2
if [ -z ${usecache+x} ]; then usecache="cache"; fi
dockerloc="${3:-.}"

main(){
    # Build and push docker container. Append version if provided.
    printf "\nBuilding Docker container now...\n "
    sleep 2s
    if [ -z ${version+x} ]
    then
        build_docker $dockerfile $tag $dockerloc $usecache
        printf "\nPushing docker container to dockerhub...\n "
        docker push $tag
    else
        build_docker $dockerfile "${tag}:${version}" $dockerloc $usecache
        printf "\nPushing docker container to dockerhub...\n "
        docker push $tag:$version
    fi

    # Build and push sif if provided.

    if [ -z ${sifdir+x} ];
    then
        sleep 2s
        printf "\n-s flag is unset, nothing more to do.\n "
    else
        dockertag="$tag"
        if [ -z ${version+x} ];
        then
            siftag=$(echo $tag | sed 's/^.*\///g')
            target="shared/singularity-containers/most-recent/"
        else
            presiftag=$(echo $tag | sed 's/^.*\///g')
            siftag="${presiftag}.${version}"
            target="shared/singularity-containers/stable/"
        fi

        printf "\nBuilding Singularity container now...\n"

        build_singularity $sifdir \
                          $siftag \
                          $dockertag

        sleep 2s
        echo "build_singularity $sifdir $siftag $dockertag"
        printf "\nLocal Singularity build complete.\nPushing singularity to box...\n"

        push_singularity $sifdir \
                         $siftag \
                         $target

    fi
}

# main(){


#     # Build and push sif if provided.

#     if [ -z ${sifdir+x} ];

#     then
#         sleep 2s
#         printf "\n-s flag is unset, nothing more to do.\n "
#     else
#         dockertag="$tag"
#         if [ -z ${version+x} ];
#         then
#             siftag="$tag"
#             target="shared/singularity-containers/most-recent"
#         else
#             siftag="${tag}.${version}"
#             target="shared/singularity-containers/stable"
#         fi
#         printf "\nBuilding Singularity container now...\n"
#         sleep 2s
#         build_singularity $sifdir \
#                           $siftag \
#                           $dockertag
#         printf "\nLocal Singularity build complete.\nPushing singularity to box...\n"
#         push_singularity $sifdir \
#                          $siftag \
#                          $target
#     fi
# }

build_docker(){
    local dockerfile="${1}"
    local dockertag="${2}"
    local dockerloc="${3}"
    if [ $4 == "--no-cache" ];
    then
        docker build \
               --file $dockerfile $4 \
               --pull \
               --tag $dockertag \
               $dockerloc
    else
        docker build \
               --file $dockerfile \
               --pull \
               --tag $dockertag \
               $dockerloc
    fi
}

push_docker(){
    local dockertag="${1}"
    docker push $dockertag
}

build_singularity(){
    local sifdir="${1}"
    local siftag="${2}"
    local dockertag="${3}"
    singularity pull \
                --force \
                "${sifdir}/${siftag}.sif" \
                "docker://${dockertag}"
}

push_singularity(){
    local sifdir="${1}"
    local siftag="${2}"
    local target="${3}"
    rclone copy \
           --update \
           --verbose \
           "${sifdir}/${siftag}.sif" remote:"${target}"
}

main "$@"

#+end_src

*****                                                                   :ref:
- https://github.com/orgs/genome/repositories?type=all
*****                                                                   :dev:
- old mpnst dockerfile
  #+begin_src bash
FROM jeszyman/biotools
#########1#########2#########3#########4#########5#########6#########7#########8

##############################################
###      Dockerfile for MPNST Projects     ###
##############################################

# After build, this image will be pushed to dockerhub as jeszyman/mpnst.
#  See (https://hub.docker.com/repository/docker/jeszyman/mpnst)

#
RUN cd /opt && \
    git clone https://github.com/shahcompbio/hmmcopy_utils.git && \
    cd hmmcopy_utils && \
    cmake . && \
    make

#
# ichorCNA
##
## linux dependencies
RUN apt-get update \
   && apt-get install -y \
   libcurl4-openssl-dev \
   libssl-dev \
   libxml2-dev
#RUN rm /usr/lib/x86_64-linux-gnu/libcurl.so.4
#RUN ln -s /usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0 /usr/lib/x86_64-linux-gnu/libcurl.so.4
##
## R dependencies
RUN R -e 'install.packages("BiocManager"); BiocManager::install(); BiocManager::install("HMMcopy"); BiocManager::install("GenomeInfoDb"); BiocManager::install("GenomicRanges");'
##
## git clone install
RUN cd /opt \
    && git clone https://github.com/broadinstitute/ichorCNA.git \
    && cd ichorCNA \
    && R CMD INSTALL . \
    && cd /opt
##
#+end_src
- ...rebuild local sing and add checker
- [ ] sing builds to box
- [ ] move all docker to single repo

***:dev:
**** Run functions
- run docker interactively
  #+name: docker_interactive
  #+begin_src bash
docker_interactive() {
    if [ -f /.dockerenv ]; then
        echo "shell already in docker, exiting"
        exit 1
    else
        account="USER INPUT"
        container="USER INPUT"
        read -p "docker account name: " account
        account="${account:=jeszyman}"
        read -p "container name: " container
        container="${container:=biotools}"
        case $HOSTNAME in
            (radonc-cancerbio) docker run -it \
                                      --env HOME=${HOME} \
                                      --hostname ${HOSTNAME} \
                                      --user $(id -u ${USER}) \
                                      --volume /home/:/home/ \
                                      --volume /mnt/:/mnt/ \
                                      --volume /tmp/:/tmp/ \
                                      $account/$container \
                                      /bin/bash;;
            (acl*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /drive3/:/drive3/ \
                          -v /duo4/:/duo4/ \
                          -v /home/:/home/ \
                          -v /tmp/:/tmp/ \
                          -u $(id -u ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (ACL*) docker run -it \
                          --env HOME=${HOME} \
                          --hostname ${HOSTNAME} \
                          -v /home/:/home/ \
                          -v /duo4/:/duo4/ \
                          -u $(id -u ${USER}):$(id -g ${USER}) \
                          $account/$container \
                          /bin/bash;;
            (virtual-workstation*.gsc.wustl.edu) bsub -Is -q docker-interactive -a 'docker($account/'"$container"')' /bin/bash;;
            (*) docker run -it \
                       --env HOME=/home/${USER} \
                       --hostname ${HOSTNAME} \
                       --user $(id -u ${USER}) \
                       --volume /home/:/home/ \
                       --volume /mnt/:/mnt/ \
                       --volume /tmp/:/tmp/ \
                       $account/$container \
                       /bin/bash;;
        esac
    fi
}
#+end_src
**** [[*Built docker container][Built docker container]]
*** Make singularity container
#+begin_src bash
#!/usr/bin/env bash
#
###   SCRIPT NAME   ###
#

# Notes
# ...

# Setup

set -e # enable exit on error
set -u # unset variables are errors
set -o pipefail # failures can occur anywhere in pipeline

# Variables

readonly blah=$1
blue=$(tput setaf 4)

# Functions

usage()
{
    printf "
           ${blue}Script to do blah \n
    usage:  blah\n
"
    exit 1
}

main()
{
    [[ $# -ne 2 ]] && usage
    echo "$blah"
    normal=$(tput sgr0)
    printf "%40s\n" "${blue}This text is bluenormal"
}

# Run

main "$@"
#+end_src

#+begin_src bash
singularity pull /mnt/ris/aadel/jeszyman/sing_containers/biotools.sif docker://jeszyman/biotools
#+end_src
https://pubs.rsc.org/en/content/articlelanding/2020/cs/d0cs00647e
*** Setup common reference files
:PROPERTIES:
:ID:       813ed980-1c55-44d7-84d8-67afdad2e5bc
:END:
  #+name: setup_common_references
  #+begin_src bash
#
# Setup
/bin/bash
alias wget_common=wget --noclobber
refdata=
#
# Download reference genomes
## Human
### UCSC Hg38
wget_common https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz -O $refdata/hg38.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.toplevel.fa.gz
#### Primary assembly
wget --no-clobber ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz -O $refdata/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
##
## Rattus norvegicus
### UCSC
wget_common ftp://hgdownload.cse.ucsc.edu/goldenPath/rn6/bigZips/rn6.fa.gz -O $refdata/rn6.fa.gz
### Ensembl
#### TopLevel
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz -O $refdata/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz
#### NOTE: No Rattus primary assembly
#### cDNA
wget_common ftp://ftp.ensembl.org/pub/release-102/fasta/rattus_norvegicus/cdna/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
gunzip /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.cdna.all.fa.gz
#### GTF
wget_common ftp://ftp.ensembl.org/pub/release-102/gtf/rattus_norvegicus/Rattus_norvegicus.Rnor_6.0.102.gtf.gz -O /home/jeszyman/data/Rattus_norvegicus.Rnor_6.0.102.gtf.gz
#
# Unzip
for file in $refdata/*.fa.gz $refdata/*.gtf.gz;
do
    gunzip_common $file
done
#

#########1#########2#########3#########4#########5#########6#########7#########8
# ideas
#in/out from csv
# get readmes
#########1#########2#########3#########4#########5#########6#########7#########8
#+end_src
** [[id:271b4d5f-727e-496e-b835-8fe9f8655655][Bioinformatics pipeline]]

** My bioinformatics pipelines
*** meth seq

****** Preamble
#+begin_src bash
<<<<<<< HEAD
FROM jeszyman/basecamp
################################################################################
################################################################################
###                                                                          ###
###         Dockerfile to build a basic bioinformatics docker image          ###
###                                                                          ###
################################################################################
################################################################################

#################
###   Notes   ###
#################
#
# After build, the image will be pushed to the dockerhub as
# jeszyman/biotools
# (https://hub.docker.com/repository/docker/jeszyman/atac)

#+end_src
****** Conda
#+begin_src bash

#################
###   Conda   ###
#################

RUN conda install -c bioconda bbmap \
    && conda install -c bioconda bcftools \
    && conda install -c bioconda bedops \
    && conda install -c bioconda bedtools \
    && conda install -c bioconda bowtie \
    && conda install -c bioconda bowtie2 \
    && conda install -c bioconda bwa \
    && conda install -c bioconda cutadapt \
    && conda install -c bioconda deeptools=3.5.1 --force \
    && conda install -c bioconda fastp \
    && conda install -c bioconda fastqc \
    && conda install -c bioconda gatk \
    && conda install -c bioconda goleft \
    && conda install -c bioconda htseq \
    && conda install -c bioconda mosdepth \
    && conda install -c bioconda preseq \
    && conda install -c bioconda pysam \
    && conda install -c bioconda qualimap \
    && conda install -c bioconda sambamba \
    && conda install -c bioconda samblaster \
    && conda install -c bioconda samtools=1.7 \
    && conda install -c bioconda seqkit \
    && conda install -c bioconda seqtk \
    && conda install -c bioconda skewer \
    && conda install -c bioconda snakemake \
    && conda install -c bioconda trimmomatic=0.39 \
    && conda install -c bioconda vcftools

#+end_src
****** R and Bioconductor
#+begin_src bash
# To avoid conflict with jeszyman/basecamp R install
#RUN conda uninstall r-base


##########################
###   R Bioconductor   ###
##########################

RUN R -e "install.packages('BiocManager', repos='http://cran.rstudio.com/')"

RUN R -e "BiocManager::install(version='3.15', update=TRUE, ask=FALSE)"

RUN R -e 'install.packages("BiocManager"); \
          BiocManager::install(); \
          BiocManager::install(c("AnnotationDbi",\
                                 "apeglm"
                                 "BiocParallel", \
                                 "BSgenome.Hsapiens.UCSC.hg19", \
                                 "ComplexHeatmap", \
                                 "EnsDb.Mmusculus.v79", \
                                 "org.Mm.eg.db", \
                                 "readr", \
                                 "Rsamtools", \
                                 "TxDb.Mmusculus.UCSC.mm10.ensGene"));'

#+end_src
****** UCSC
#+begin_src bash

##############################################
###   Select UCSC Command-line utilities   ###
##############################################

RUN mkdir /opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedClip --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCommonRegions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedCoverage --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedExtendRanges --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGeneParts --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphPack --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedIntersect --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedItemOverlapCount --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedJoinTabOffset.py --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedMergeAdjacent --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPartition --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedPileUps --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRemoveOverlap --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedRestrictToPositions --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedSort --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToExons --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigAverageOverBed --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCat --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCluster --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigCorrelate --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigInfo --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigMerge --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToWig --directory-prefix=/opt/ucsc \
    && wget --no-parent --no-directories https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/#+end_src --directory-prefix=/opt/ucsc \
    && chmod -R +x /opt/ucsc/

ENV PATH="/opt/ucsc:${PATH}"

#+end_src
****** Custom and manual installs
#+begin_src bash

###########################
###   Custom Installs   ###
###########################

# Multiqc

RUN pip install multiqc

# Flexbar
RUN apt-get install -yq --no-install-recommends --allow-unauthenticated libtbb2
RUN wget https://github.com/seqan/flexbar/releases/download/v3.5.0/flexbar-3.5.0-linux.tar.gz && tar -xzf flexbar-3.5.0-linux.tar.gz && mv flexbar-3.5.0-linux/flexbar /usr/bin/ && mv flexbar-3.5.0-linux/libtbb.so.2 /usr/bin/

# samtools
# https://www.biostars.org/p/9480029/
RUN conda install -c bioconda samtools openssl=1.0

#+end_src
******* Picard
#+begin_src bash

##################
###   Picard   ###
##################

RUN apt-get install -yq --no-install-recommends --allow-unauthenticated openjdk-8-jdk

#########1#########2#########3#########4#########5#########6#########7#########8
#https://github.com/genome/docker-picard-cwl/commit/639ad8e4d8e7bd7e6e10f193df8d5cd16260d2e2
RUN mkdir /opt/picard-2.18.1/ \
    && cd /tmp/ \
    && wget --no-check-certificate https://github.com/broadinstitute/picard/releases/download/2.18.1/picard.jar \
    && mv picard.jar /opt/picard-2.18.1/ \
    && ln -s /opt/picard-2.18.1 /opt/picard \
    && ln -s /opt/picard-2.18.1 /usr/picard
#########1#########2#########3#########4#########5#########6#########7#########8


#+end_src
****** Development                                                      :dev:
:PROPERTIES:
:header-args: :tangle no

*** [[file:~/repos/exrna/exrna.org][exRNA]]
:LOGBOOK:
CLOCK: [2022-06-08 Wed 14:22]--[2022-06-08 Wed 14:44] =>  0:22
CLOCK: [2022-06-08 Wed 13:41]--[2022-06-08 Wed 13:59] =>  0:18
:END:
*** [[file:~/repos/cappseq/cappseq.org][CAPP-seq]]
:LOGBOOK:
CLOCK: [2022-06-07 Tue 08:19]--[2022-06-07 Tue 08:21] =>  0:02
CLOCK: [2022-04-29 Fri 09:12]--[2022-04-29 Fri 10:00] =>  0:48
CLOCK: [2022-04-26 Tue 08:34]--[2022-04-26 Tue 09:08] =>  0:34
CLOCK: [2022-04-26 Tue 12:00]--[2022-04-26 Tue 13:31] =>  1:31
CLOCK: [2022-04-26 Tue 09:51]--[2022-04-26 Tue 10:10] =>  0:19
:END:
*** [[file:~/repos/cfdna-wgs/cfdna-wgs.org][cfDNA WGS biopipe]]
:PROPERTIES:
:ID:       13120759-71db-497c-8ed3-1c58e47a7840
:END:
:LOGBOOK:
CLOCK: [2022-05-04 Wed 08:12]--[2022-05-04 Wed 08:16] =>  0:04
CLOCK: [2022-05-03 Tue 10:19]--[2022-05-03 Tue 11:04] =>  0:45
CLOCK: [2022-05-03 Tue 09:46]--[2022-05-03 Tue 10:15] =>  0:29
CLOCK: [2022-04-29 Fri 10:34]--[2022-04-29 Fri 10:56] =>  0:22
CLOCK: [2022-06-27 Mon 11:13]--[2022-06-27 Mon 13:40] =>  2:27
CLOCK: [2022-06-27 Mon 09:11]--[2022-06-27 Mon 10:47] =>  1:36
CLOCK: [2022-06-27 Mon 07:07]--[2022-06-27 Mon 09:11] =>  2:04
CLOCK: [2022-06-24 Fri 08:02]--[2022-06-24 Fri 10:19] =>  2:17
CLOCK: [2022-06-23 Thu 10:20]--[2022-06-23 Thu 11:18] =>  0:58
CLOCK: [2022-06-16 Thu 12:48]--[2022-06-16 Thu 13:42] =>  0:54
CLOCK: [2022-06-16 Thu 08:08]--[2022-06-16 Thu 08:39] =>  0:31
CLOCK: [2022-06-07 Tue 08:29]--[2022-06-07 Tue 10:56] =>  2:27
CLOCK: [2022-06-02 Thu 08:38]--[2022-06-02 Thu 08:40] =>  0:02
CLOCK: [2022-05-31 Tue 11:03]--[2022-05-31 Tue 12:44] =>  1:41
:END:
**** frag
:LOGBOOK:
CLOCK: [2022-04-29 Fri 13:52]--[2022-04-29 Fri 14:07] =>  0:15
CLOCK: [2022-04-29 Fri 11:17]--[2022-04-29 Fri 11:40] =>  0:23
CLOCK: [2021-12-07 Tue 08:55]--[2021-12-07 Tue 09:08] =>  0:13
:END:
*** [[file:~/repos/atac-seq/atac_seq.org][ATAC-seq biopipe]]
:LOGBOOK:
CLOCK: [2022-08-29 Mon 12:18]--[2022-08-29 Mon 13:00] =>  0:42
CLOCK: [2022-05-01 Sun 12:17]--[2022-05-01 Sun 13:47] =>  1:30
CLOCK: [2021-12-09 Thu 12:47]--[2022-05-01 Sun 13:57] => 3432:10
:END:
*** cfDNA WGS fragmentomics biopipe
:LOGBOOK:
CLOCK: [2022-08-31 Wed 08:20]--[2022-08-31 Wed 09:11] =>  0:51
:END:
*** See [[id:271b4d5f-727e-496e-b835-8fe9f8655655][Bioinformatics project module]]
** [[id:39F11977-BE11-47CF-98B1-14CE6D9B97C4][Information Science / Data Science]]
:PROPERTIES:
:ID:       014218e1-bb8b-4282-84ec-84354c29b8f5
:END:
** [[file:~/repos/basecamp/basecamp.org::*Basecamp][Basecamp]]
:PROPERTIES:
:ID:       76100c2b-a077-4c21-a398-16d88722e1c6
:END:
** Development :dev:
*** v0.1.0
**** sifs to szymanski RIS
*** Ideas
:PROPERTIES:
:ID:       1a00532b-6854-48e5-9860-17d62fe1100e
:END:
- molecular disease signatures
- https://www.reddit.com/r/commandline/comments/vcs92o/scitopdf_quickly_fetch_science_from_scihub/
- https://slowkow.com/publications/
- Data dictionary
  #+name: biotools_docker_versions
  | cutadapt_version |    1.18 |
  | mirdeep2_version | 2.0.1.2 |
  | bowtie           |     1.2 |
  #+name: injest_biotools_docker_versions
  ,#+begin_src python :var biotools_docker_versions=biotools_docker_versions
  import os
  import csv

  os.chdir('/home/jeszyman/repos/biotools')

  with open('./data/biotools_docker_versions.csv', 'w', newline='') as file:
      writer = csv.writer(file, delimiter=',')
      writer.writerows(biotools_docker_versions)
  .#+end_src
  #+name: injest_biotools_docker_versions_wrapper
  .#+begin_src python :noweb yes
  injest_biotools_docker_versions
  .#+end_src
- Ideas
   - make test data
    #+name: make_test_data
    .#+begin_src bash
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  ### Make small test data files suitable for github push  ###
  #
  # Setup
  ##
  ## Local
  source ~/repos/mpnst/bin/local-setup.sh
  ##
  ## Docker
  docker_interactive
  biotools
  ##
  ## Functions
  ##
  fastq_test(){
      # Check for parameters, return usage
      if [[ $# -eq 0 ]] ; then
          printf "\n fastq_test
                \n Generates fastq files and .gzs of desired file size
                \n \$1 = input (in fastq.gz)
                \n \$2 = FILESIZE (in byte)
                \n \$3 = output directory
                \n \$4 = output filename (END .fastq)
                \n "
      else
          # In-function variable setup
          infile=$1
          filesize=$2
          outdir=$3
          base=`basename -s .fastq.gz $infile`
          # Out file name defaults to <in file name>_test
          outfilename=${4:-${base}_test.fastq}
          readsize=$(zcat $infile | head -n 4 | wc -c)
          lines=$(( 4 * ($filesize / $readsize) ))
          zcat $infile | head -n "$lines" > ${outdir}/$outfilename
          # Get exact same lines for R2
          infile_R2=$(echo "$1" | sed 's/_R1/_R2/g')
          outfilename_R2=$(echo "$outfilename" | sed 's/_R1/_R2/g')
          zcat $infile_R2 | head -n "$lines" > ${outdir}/$outfilename_R2
          # Get gzipped files
          gzip -f -k ${outdir}/$outfilename
          gzip -f -k ${outdir}/$outfilename_R2
      fi
  }
  #########1#########2#########3#########4#########5#########6#########7#########8
  #
  #
  rm -rf $repo/data/test
  mkdir -p $repo/data/test
  #
  # WashU WGS fastqs
  ##
  ## cfDNA
  ###
  ### mpnst
  fastq_test $localdata/.washu-fastq-extract/new_HiSeqW72_Undetermined_R6000482_L001_R1_TCCT_extract.fastq.gz \
             1000000 \
             $repo/data/test \
             washu_mpnst_R1.fastq
  ### healthy
  ###
  fastq_test $localdata/.washu-fastq-extract/new_HiSeqW31_Undetermined_R6000232_L005_R1_001_GAAG_extract.fastq.gz \
             1000000 \
             $repo/data/test \
             washu_healthy_R1.fastq
  ## NCI
  ###
  fastq_test $localdata/fastqs/lib107_R1.fastq.gz \
             1000000 \
             $repo/data/test \
             nci_mpnst_R1.fastq
  ###
  fastq_test $localdata/fastqs/lib088_R1.fastq.gz \
             1000000 \
             $repo/data/test \
             nci_healthy_R1.fastq
  #
  # cappseq demultiplexed but UN-extracted files:
  ###
  fastq_test $localdata/.washu-demulti-tmp/new_HiSeqW43_Undetermined_R6000292_L004_R1_001_TCCT.fastq.gz \
             1000000 \
             $repo/data/test \
             demulti_TCCT_R1.fastq
  #+end_src
** Reference
:PROPERTIES:
:CREATED:  [2020-04-23 Thu 08:54]
:ID:       98f8c0e9-cdfc-4acb-b98c-5db1bd6bc3ca
:END:
- https://hgserver1.amc.nl/cgi-bin/r2/main.cgi?open_page=login
- https://www.reddit.com/r/bioinformatics/comments/17svhsq/genome_in_a_bottle_just_released_hg002_v10_new/
- https://github.com/fgomez02
- https://ai.tinybio.cloud/chat?rdt_cid=3878222416981413620
- https://www.reddit.com/r/bioinformatics/comments/154v2k6/resources_for_learning_best_practices_for_writing/

- https://www.cancergenomics.org/cgc_webinars_and_special_semin.php
- https://cancergenomics.org/meetings/2022_annual_meeting_presentati.php
- https://www.reddit.com/r/bioinformatics/comments/12402xq/what_are_the_struggles_that_bioinformatician_face/
- https://www.reddit.com/r/bioinformatics/comments/13i5mgf/we_have_got_a_pin_up_board_in_our_bioinfo_lab_i/
- https://hpc.nih.gov/apps/
- https://wikis.utexas.edu/display/bioiteam/MultiQC+-+fastQC+summary+tool+--+GVA2021
- https://www.youtube.com/playlist?list=PLEEE2A91B09B77B4A
- https://www.reddit.com/r/bioinformatics/comments/11sx6vo/whats_the_best_way_to_catch_up_on_mathematics_for/
- https://mobile.twitter.com/jsantoyo/status/1139238326454669312
- [[https://www.cell.com/cancer-cell/fulltext/S1535-6108(22)]]
- https://www.cell.com/cell/fulltext/S0092-8674(22)01379-4
- https://rnnh.github.io/bioinfo-notebook/;
- https://yulab-smu.top/biomedical-knowledge-mining-book/index.html
- https://www.nature.com/articles/s43018-022-00467-x#Sec8
- https://bioinfocore.com/blog/
- https://www.reddit.com/r/bioinformatics/comments/ys65uy/lessons_learned_from_a_bioinformatics_msc/
https://www.reddit.com/r/bioinformatics/comments/xoltse/golden_rules_of_data_analysis/
https://www.nature.com/articles/s41588-022-01179-9
https://aacrjournals.org/cancerdiscovery/article/doi/10.1158/2159-8290.CD-22-0200/708156/Proteogenomic-markers-of-chemotherapy-resistance
- multi-omics population disease characterizations
  - https://www.frontiersin.org/articles/10.3389/fonc.2022.932337/full
https://www.reddit.com/r/bioinformatics/comments/wtnu0f/staying_up_to_date_with_processestechnology/
- https://www.omicsdi.org/
- https://hpc.nih.gov/training/gatk_tutorial/preproc.html
- https://www.ncbi.nlm.nih.gov/assembly/model/
- https://github.com/pfields8/Fields_et_al_2018
- https://darwinsark.org/forums/topic/open-source-database/
- https://www.reddit.com/r/bioinformatics/comments/uf4q2m/what_was_your_bioinformatics_success_story_of_the/
- https://www.reddit.com/r/bioinformatics/comments/rc9901/how_do_you_see_the_future_of_computational/
- https://www.reddit.com/r/bioinformatics/comments/r941br/does_this_learning_path_seem_good_to_become_a/
- https://docs.gdc.cancer.gov/Data/Introduction/
- https://www.reddit.com/r/bioinformatics/comments/u0ppfq/can_you_recommend_a_textbook_that_gives_a/
- https://bioinformatics.chat/
- http://labs.pathology.jhu.edu/halushka/science/
- https://www.reddit.com/r/ProgrammerHumor/comments/tpo157/after_every_scrum_meeting/
- https://www.nature.com/articles/s41592-022-01412-7
- [[https://bioinformaticsresourceportal.wustl.edu/][Washington University Computational Biology & Bioinformatics Resource Portal]]
- https://www.reddit.com/r/bioinformatics/comments/skiod8/whats_your_setup_like_to_make_your_life_easier/
- https://mail.google.com/mail/u/0/#inbox/FMfcgzGmtrSDsSwZKDbRkXhVfzJPStMW
- https://www.sciencedirect.com/science/article/pii/S2666979X21000367?via%3Dihub
- https://github.com/genome/bfx-workshop
- https://www.biostars.org/p/190366/
- https://wustl.app.box.com/s/00qniw32ng670650r1cz5bljq2uja4q1
- https://pubmed.ncbi.nlm.nih.gov/34935906/
- https://www.cancergenomicscloud.org/
- https://www.reddit.com/r/bioinformatics/comments/rkqci9/whats_something_cool_and_exiting_going_on_in/
- https://med.stanford.edu/dbds/education/workshops-in-biostatistics/workshop-archives.html
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5302158/
- https://www.reddit.com/r/bioinformatics/comments/pt7ecx/to_all_the_seasoned_bioinformaticians_out_there/
- https://github.com/genome/bfx-workshop/blob/master/README.md#communications
- https://rockefelleruniversity.github.io/Bioconductor_Introduction/
- https://github.com/Bioinformatics-Research-Network/training-requirements

- https://www.reddit.com/r/bioinformatics/comments/n2fwz4/what_is_something_that_you_realized_that_entirely/
- https://www.reddit.com/r/bioinformatics/comments/n2fwz4/what_is_something_that_you_realized_that_entirely/

- https://www.reddit.com/r/bioinformatics/comments/nawxng/bioinformaticianswhat_do_you_wish_wet_lab/?sort=controversial
- http://bioconductor.org/packages/stats/
  - top 75
    - http://bioconductor.org/packages/release/bioc/html/fgsea.html
    - http://bioconductor.org/packages/release/bioc/html/DOSE.html
- https://project-awesome.org/danielecook/Awesome-Bioinformatics
- [[https://www.youtube.com/channel/UCQqQpIx3zQPaifBj67ocv1w][BabrahamBioinf youtube]]
- https://www.freecodecamp.org/news/python-for-bioinformatics-use-machine-learning-and-data-analysis-for-drug-discovery/
- https://pages.10xgenomics.com/ugm-2020-10-event-ra_g-user-group-meeting-amr_on-demand.html?src=email&utm_medium=email&lss=internal&utm_source=internal&cnm=ugm-2020-10-event-ra_g-user-group-meeting-amr&utm_campaign=ugm-2020-10-event-ra_g-user-group-meeting-amr&useroffertype=event&userresearcharea=ra_g&userregion=multi&mktouserid=1629122
- https://slowkow.com/
- http://rafalab.github.io/pages/harvardx.html
- https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga/publications

- https://scilifelab.github.io/courses/
- https://www.reddit.com/r/bioinformatics/comments/m78kqb/what_are_the_best_book_i_could_get_for_a/

- https://www.reddit.com/r/bioinformatics/comments/m2o63g/best_courses_or_material_to_learn_bioinformatics/

- https://learn.gencore.bio.nyu.edu/
- cite:buffalo2015
- biotools bash functions in ./src
- https://www.reddit.com/r/bioinformatics/comments/lfusag/skills_required_for_being_a_proficient/
- cite:peters2018
- https://github.com/jdidion/biotools
- https://support.bioconductor.org/#search-anchor
- https://informatics.fas.harvard.edu/category/tutorials.html
- https://www.youtube.com/channel/UCv4IbnP9j9RC_aZAs8wqdeQ
- https://www.broadinstitute.org/genome-sequencing/broadillumina-genome-analyzer-boot-camp
- https://cloud.genepattern.org/gp/pages/login.jsf
- https://github.com/genome/analysis-workflows
- https://github.com/genome/bfx-workshop
https://github.com/seanken
http://genomewiki.ucsc.edu/index.php/Main_Page
https://genome.ucsc.edu/FAQ/index.html
  - https://basespace.illumina.com/dashboard
    - - dragon aligner
  - https://www.reddit.com/r/bioinformatics/comments/5wo75h/ben_ward_julia_for_bioinformatics_and/
  - https://platform.dnanexus.com/login
  - https://www.ecseq.com/
    - https://www.ecseq.com/workshops/ngs-data-analysis-courses
  - repos
    - https://github.com/Illumina
    - https://github.com/chrisamiller
  - https://www.reddit.com/r/bioinformatics/comments/cxlsh5/outdated_software_still_in_use/
  - https://github.com/pditommaso/awesome-pipeline
  - https://coding4medicine.com/cheatsheets/
  - https://github.com/genome/bfx-workshop
  - examples
    - papers
      - https://www.sciencedirect.com/science/article/pii/S009286741830309X#sec4
      - good github repos
        - https://github.com/ding-labrecip
        - https://github.com/pstew
  - https://www.rockefeller.edu/genomics/

  - https://www.reddit.com/r/bioinformatics/comments/iv7xjj/biologist_with_minimal_coding_background/
  - old docker setup code
  #########1#########2#########3#########4#########5#########6#########7#########8
  # DEV
  # check if inside docker
  # if [ -f /.dockerenv ];then
  #     dockercontainer=yes
  #     echo "Your are in a docker container"
  # else
  #     dockercontainer=no
  #     echo "Not in docker"
  #     # setup NON-docker variables
  #     shopt -s nocasematch
  #     case $HOSTNAME in
  #         (radonc-cancerbio) server_home="/home/jeszyman" ;;
  #         (jeszyman-*) server_home="/home/jeszyman" ;;
  #         (acl*) server_home="/home/jszymanski" ;;
  #         (virtual-workstation*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
  #         (blade*.gsc.wustl.edu) server_home="/gscuser/szymanski" ;;
  #     esac
  #
  #     localdata="$server_home/data/$project/"
  #     unset $SERVER
  # fi
  #
  # case $dockercontainer in
  #     (yes) server_ask ;;
  #     (no) ;;
  # esac
  # #
  # case $SERVER in
  #     (chaudhuri-roche) projectdir=/drive3/users/jszymanski/repos/$project
  #                       localdata=/drive3/users/jszymanski/data/$project;;
  #     (NA) ;;
  # esac
  # #
  # # test data on cancerbio
  #NOT GENERAL
  #basedir=/media/jeszyman/4edbbc27-7e29-4667-9b24-f5a56d49d94e/users/jeszyman/data/mpnst
  #rm -rf $basedir/fastq-test
  #mkdir -p $basedir/fastq-test
  #zcat $basedir/fastq-nci/MPNST/14_JS0108CD040119_R1.fastq.gz | head -n 8000 > $basedir/fastq-test/short_14_JS0108CD040119_R1.fastq
  #zcat $basedir/fastq-nci/MPNST/14_JS0108CD040119_R2.fastq.gz | head -n 8000 > $basedir/fastq-test/short_14_JS0108CD040119_R2.fastq
  #zcat $basedir/fastq-washu/new_HiSeq-20_L007001_CTTC_R1.fastq.gz | head -n 8000 > $basedir/fastq-test/short_new_HiSeq-20_L007001_CTTC_R1.fastq.gz
  #gzip $basedir/fastq-test/*.fastq
  # setup project repo
- see gnu parallel in [[id:cd560390-aa1e-4c9e-8317-08610e5de42a][linux]]
- https://www.reddit.com/r/bioinformatics/comments/191ykr/resources_for_learning_bioinformatics/
*** education, tutorials, books, etc.
:PROPERTIES:
:CREATED:  [2019-04-26 Fri 10:36]
:ID:       374F30CA-2E80-4CC2-8FD3-5B9BC8083889
:END:
- https://www.reddit.com/r/bioinformatics/comments/rc9901/how_do_you_see_the_future_of_computational/
- https://www.reddit.com/r/bioinformatics/comments/r941br/does_this_learning_path_seem_good_to_become_a/
- https://www.reddit.com/r/bioinformatics/comments/esr5mw/any_good_books_about_bioinformatics/
- https://protocolexchange.researchsquare.com/browse?journal=protocol-exchange&subjectArea=Computational%20biology%20and%20bioinformatics

- https://www.melbournebioinformatics.org.au/tutorials/
- https://www.coursera.org/specializations/bioinformatics
- training, tutorials, etc.
  - https://bootcamp.biostars.io/
- https://jef.works/
- https://www.coursera.org/course/bioinformatics
- https://www.ebi.ac.uk/training/online/
- http://egtheory.wordpress.com/2013/06/09/toward-an-algorithmic-theory-of-biology/
- http://rosalind.info/problems/as-graph/
- https://www.nature.com/collections/vzsqzylnvx
- foundations
  - https://www.reddit.com/r/bioinformatics/comments/dckuuk/what_are_some_mathematically_rigorous_intro_books/
- https://wikis.utexas.edu/collector/pages.action?key=bioiteam
- https://www.sh-eahp.org/images/2017_Workshop/1_10ESHAP_2017.pdf
- https://www.reddit.com/r/bioinformatics/comments/bxgp0h/what_do_you_wish_someone_had_just_shown_you_early/
- https://girke.bioinformatics.ucr.edu/GEN242/mydoc_systemPipeRNAseq_06.html
- https://www.illumina.com/company/video-hub.html
- organizations and communities
  - http://www.scilifelab.se/
- [[id:2274F8F7-45D0-4F2F-BA32-4AF38C8A0F80][Molecular Pathology Compendium]]
- [[id:BDC42DA2-A632-4FB7-A586-E100B2A045DC][UNIX]]
- [[id:022FDD07-245F-4231-905E-ED18BE224889][Molecular Biology]]
- organizations
  - [[id:543E3788-28CC-4D16-943B-BC17523D66D9][McDonnell Genome Institute (MGI)]]
  - Broad
    - [[http://software.broadinstitute.org/software/cprg/?q=node/27][Broad ABSOLUTE]]
    - [[https://software.broadinstitute.org/gatk/documentation/topic?name=methods][Broad methods]]
  - [[http://www.sanger.ac.uk/science/tools][Wellcome Sanger]]
- education, tutorials, etc.
  - https://www.edx.org/learn/bioinformatics
  - https://www.coursera.org/learn/bioinformatics
  - http://bioinformaticsalgorithms.com/
  - https://www.reddit.com/r/bioinformatics/comments/ag7lc8/the_bible_of_bioinformatics/
  - https://bioinformatics.ca/workshops/workshops-2017/
  - [[http://bioinformatics.ucdavis.edu/training/documentation/][UC Davis Bioinformatics Core Workshops]]
  - http://rosalind.info/problems/locations/
  - https://www.reddit.com/r/bioinformatics/comments/6405tf/what_are_some_good_small_bioinformatics_projects/
  - [[http://userweb.eng.gla.ac.uk/umer.ijaz/bioinformatics/linux.html][Linux command line exercises for NGS data processing]]
  - https://biostar.myshopify.com/
  - https://www.reddit.com/r/bioinformatics/comments/7bntke/book_recommendations_to_be_a_better/
  - [[http://area51.stackexchange.com/proposals/109245/bioinformatics][StackExchange Bioinformatics]]
  - Well done sequencing papers http://www.nejm.org/doi/full/10.1056/NEJMoa1502214?rss=mostViewed&
  - http://seqanswers.com/
  - Training
    - https://collections.plos.org/compbiol-education
    - - https://docs.google.com/spreadsheets/d/1hIw66KCfdB0rowN7tJhyVjJv09Qv87r2B0sWLinVeS8/htmlview
    - https://www.broadinstitute.org/partnerships/education/broade/best-practices-variant-calling-gatk-1
    - - https://www.ebi.ac.uk/training/online/
    - - https://www.reddit.com/r/bioinformatics/comments/5miwnr/short_list_of_bioinformatics_skills/
    - https://bioinformatics.ca/workshops/2016
    - http://rosalind.info/problems/locations/
    - http://griffithlab.org/teaching/
    - http://evomics.org/workshops/2017-workshop-on-transcriptomics-harvard-university/
    - [[https://github.com/nekrut/BMMB554/wiki/7.-NGS-Processing:-QC,-Mapping,-etc][BMMB554 Introduction to data driven life sciences]].
  - https://satijalab.org/scgd/
  - https://www.genomeweb.com/resources/webinars
  - books
    - cite:low2017
    - cite:sinha2014

**** Notes
:PROPERTIES:
:ID:       90C0571F-2BB4-484B-8C2E-2385AD22A38C
:END:

[[http://onlinelibrary.wiley.com/book/10.1002/0471250953][Current Protocols in Bioinformatics]]

because computation removes redundancy, you will more frequently be at a new, tricky step

**** glossary
:PROPERTIES:
:CREATED:  [2019-05-14 Tue 10:25]
:ID:       80B5A347-33C6-4FC2-A691-FFAFF3C5D352
:END:

- https://software.broadinstitute.org/gatk/documentation/topic?name=dictionary&v=3
- [[http://useast.ensembl.org/info/genome/genebuild/biotypes.html][ensembl biotypes]]
- [[https://useast.ensembl.org/Help/Glossary][Ensembl glossary]]
- [[http://seqanswers.com/forums/showthread.php?t=13098][insert size]]: fragment size + length of both adapters
- [[http://seqanswers.com/forums/showthread.php?t=13098][fragment size]]: sequence between the paired-end adapters

*** ENCODE
:PROPERTIES:
:ID:       585e27b3-3c79-46d5-aeab-9430251d4318
:END:
- cite:becker2011
- https://www.annualreviews.org/doi/10.1146/annurev-genom-090413-025448#_i21
